diff -Naur qtractor-0.2.2.orig/qtractor.desktop.in qtractor-0.2.2/qtractor.desktop.in
--- qtractor-0.2.2.orig/qtractor.desktop.in	2007-07-03 12:05:21.000000000 +0300
+++ qtractor-0.2.2/qtractor.desktop.in	2008-11-02 23:53:30.000000000 +0200
@@ -1,8 +1,12 @@
 [Desktop Entry]
 Name=Qtractor
+Name[tr]=Qtractor
+GenericName=Audio/MIDI multi-track sequencer
+GenericName[tr]=MIDI düzenleme uygulaması
 Comment=Qtractor is an Audio/MIDI multi-track sequencer application
-Exec=@ac_prefix@/bin/qtractor
-Icon=@ac_prefix@/share/pixmaps/qtractor.png
+Comment[tr]=Çok kanallı Ses/MIDI kayıt ve düzenleme uygulaması
+Exec=qtractor
+Icon=qtractor
 Categories=Applications;Multimedia;AudioVideo;MIDI;JACK;ALSA;Qt;
 Terminal=false
 Type=Application
