diff -Naur qtractor-0.3.0.orig/qtractor.desktop.in qtractor-0.3.0/qtractor.desktop.in
--- qtractor-0.3.0.orig/qtractor.desktop.in	2008-12-15 14:05:39.000000000 +0200
+++ qtractor-0.3.0/qtractor.desktop.in	2008-12-25 23:00:44.000000000 +0200
@@ -1,10 +1,12 @@
 [Desktop Entry]
 Name=Qtractor
+Name[tr]=Qtractor
 GenericName=Multi-track Sequencer
+GenericName[tr]=Çok kanallı ses düzenleme yazılımı
 Comment=Qtractor is an Audio/MIDI multi-track sequencer application
-Exec=@ac_prefix@/bin/qtractor
-Icon=@ac_prefix@/share/pixmaps/qtractor.png
-Categories=Audio;AudioVideo;JACK;ALSA;Qt;
+Comment[tr]=Çok kanallı ses düzenleme yazılımı
+Exec=qtractor
+Icon=qtractor.png
+Categories=Audio;Multimedia;AudioVideo;JACK;ALSA;Qt;
 Terminal=false
 Type=Application
-X-SuSE-translate=true
