diff -Naur fotoxx-9.6.orig/Makefile fotoxx-9.6/Makefile
--- fotoxx-9.6.orig/Makefile	2010-02-23 23:24:20.163771370 +0200
+++ fotoxx-9.6/Makefile	2010-02-23 23:33:10.379524040 +0200
@@ -5,31 +5,31 @@
 SOURCE = $(PROGRAM)-$(VERSION).cpp
 
 # replace CFLAGS and LFLAGS -O3 with -g for GDB debugging
-CFLAGS = -O3 -Wall -c `pkg-config --cflags gtk+-2.0`
-LFLAGS = -O3 -Wall -rdynamic `pkg-config --libs gtk+-2.0 gthread-2.0`
+CFLAGS = -g -Wall -c `pkg-config --cflags gtk+-2.0`
+LFLAGS = -g -Wall -rdynamic -lfreeimage `pkg-config --libs gtk+-2.0 gthread-2.0`
 
 # use $PREFIX if defined, else assume /usr/local
 ifeq "$(PREFIX)" ""
-	PREFIX = /usr/local
+	PREFIX = /usr
 endif
 
 BINDIR = $(PREFIX)/bin
 DATADIR = $(PREFIX)/share/$(PROGRAM)
 ICONDIR = $(DATADIR)/icons
-DESKTOP = $(DATADIR)/kornelix-$(PROGRAM).desktop
+DESKTOP = $(PREFIX)/share/applications/$(PROGRAM).desktop
 DOCDIR = $(PREFIX)/share/doc/$(PROGRAM)
 MANDIR = $(PREFIX)/share/man/man1
 
 $(PROGRAM): $(PROGRAM).o zfuncs.o
 	@ ./dependencies.sh
-	@ g++ $(LFLAGS) -lfreeimage -o $(PROGRAM) $(PROGRAM).o zfuncs.o
+	@ $(CXX) $(LDFLAGS) -lfreeimage $(PROGRAM).o zfuncs.o -o $(PROGRAM) $(LFLAGS)
 
 $(PROGRAM).o: $(SOURCE)
-	@ g++ $(CFLAGS) -o $(PROGRAM).o $(SOURCE)		\
+	@ $(CXX) $(CFLAGS) -o $(PROGRAM).o $(SOURCE)        \
 	  -D "DATADIR=\"$(DATADIR)\"" -D "DOCDIR=\"$(DOCDIR)\""
 
 zfuncs.o: zfuncs.cpp
-	@ g++ $(CFLAGS) zfuncs.cpp	\
+	@ $(CXX) $(CFLAGS) zfuncs.cpp   \
 	  -D "DATADIR=\"$(DATADIR)\"" -D "DOCDIR=\"$(DOCDIR)\"" -D "BINDIR=\"$(BINDIR)\""
 
 install: $(PROGRAM)
