diff -Naur icewm-1.2.26/lib/menu icewm-1.2.26-yeni/lib/menu
--- icewm-1.2.26/lib/menu	2006-04-16 19:12:56.000000000 +0300
+++ icewm-1.2.26-yeni/lib/menu	2006-06-18 18:28:27.000000000 +0300
@@ -4,16 +4,21 @@
 # since modifications to this file will be discarded when you
 # (re)install icewm.
 #
-prog xterm xterm xterm
-prog rxvt xterm rxvt -bg black -cr green -fg white -C -fn 9x15 -sl 500
-prog fte fte fte
-prog NEdit nedit nedit
-prog Mozilla mozilla mozilla
-prog XChat xchat xchat
-prog Gimp gimp gimp
+#prog xterm xterm xterm
+#prog rxvt xterm rxvt -bg black -cr green -fg white -C -fn 9x15 -sl 500
+#prog fte fte fte
+#prog NEdit nedit nedit
+#prog Mozilla mozilla mozilla
+#prog XChat xchat xchat
+#prog Gimp gimp gimp
+prog konsole /usr/share/icons/Tulliana-1.0/48x48/apps/konsole.png konsole
+prog firefox /usr/share/icons/Tulliana-1.0/48x48/apps/firefox firefox
+prog konversation /usr/kde/3.5/share/icons/hicolor/32x32/apps/konversation.png konversation
+prog mplayer /usr/share/icons/Tulliana-1.0/32x32/apps/mplayer.png gmplayer
+prog tasma tasma tasma
 separator
-menuprog Gnome folder icewm-menu-gnome1 --list 
-menuprog Gnome folder icewm-menu-gnome2 --list 
-menuprog KDE folder icewm-menu-gnome --list /usr/share/applnk/
+#menuprog Gnome folder icewm-menu-gnome1 --list 
+#menuprog Gnome folder icewm-menu-gnome2 --list 
+menuprog KDE folder icewm-menu-gnome --list /usr/kde/3.5/share/applnk/
 menufile Programs folder programs
 menufile Tool_bar folder toolbar
diff -Naur icewm-1.2.26/lib/menu.in icewm-1.2.26-yeni/lib/menu.in
--- icewm-1.2.26/lib/menu.in	2006-04-16 19:12:31.000000000 +0300
+++ icewm-1.2.26-yeni/lib/menu.in	2006-06-18 18:28:42.000000000 +0300
@@ -4,16 +4,21 @@
 # since modifications to this file will be discarded when you
 # (re)install icewm.
 #
-prog xterm xterm xterm
-prog rxvt xterm rxvt -bg black -cr green -fg white -C -fn 9x15 -sl 500
-prog fte fte fte
-prog NEdit nedit nedit
-prog Mozilla mozilla mozilla
-prog XChat xchat xchat
-prog Gimp gimp gimp
+#prog xterm xterm xterm
+#prog rxvt xterm rxvt -bg black -cr green -fg white -C -fn 9x15 -sl 500
+#prog fte fte fte
+#prog NEdit nedit nedit
+#prog Mozilla mozilla mozilla
+#prog XChat xchat xchat
+#prog Gimp gimp gimp
+prog konsole /usr/share/icons/Tulliana-1.0/48x48/apps/konsole.png konsole
+prog firefox /usr/share/icons/Tulliana-1.0/48x48/apps/firefox firefox
+prog konversation /usr/kde/3.5/share/icons/hicolor/32x32/apps/konversation.png konversation
+prog mplayer /usr/share/icons/Tulliana-1.0/32x32/apps/mplayer.png gmplayer
+prog tasma tasma tasma
 separator
-menuprog Gnome folder icewm-menu-gnome1 --list @CONFIG_GNOME1_MENU_DIR@
-menuprog Gnome folder icewm-menu-gnome2 --list @CONFIG_GNOME2_MENU_DIR@
+#menuprog Gnome folder icewm-menu-gnome1 --list @CONFIG_GNOME1_MENU_DIR@
+#menuprog Gnome folder icewm-menu-gnome2 --list @CONFIG_GNOME2_MENU_DIR@
 menuprog KDE folder icewm-menu-gnome@GNOME_VER@ --list @CONFIG_KDE_MENU_DIR@
 menufile Programs folder programs
 menufile Tool_bar folder toolbar
