--- settings/xfce-mixer-settings.desktop.in.orig	2008-07-13 15:22:08.000000000 +0300
+++ settings/xfce-mixer-settings.desktop.in	2008-07-13 15:49:48.000000000 +0300
@@ -1,11 +1,14 @@
 [Desktop Entry]
 Encoding=UTF-8
-_Name=Xfce 4 Mixer Settings
-_Comment=Mixer Settings
+Name=Xfce 4 Mixer Settings
+Name[tr]=Ses Ayarları
+Comment=Mixer Settings
+Comment[tr]=Ses Ayarları
 Exec=xfce-setting-show sound
 Icon=xfce-sound
 Terminal=false
 Type=Application
 Categories=X-XFCE;Settings;DesktopSettings;
 OnlyShowIn=XFCE;
-_GenericName=Mixer Settings
+GenericName=Mixer Settings
+GenericName[tr]=Ses Ayarları
