diff -u servicemenus/amarok_append.desktop servicemenus2/amarok_append.desktop
--- servicemenus/amarok_append.desktop	2006-10-29 00:05:30.000000000 +0300
+++ src/servicemenus/amarok_append.desktop	2007-04-09 01:01:27.000000000 +0300
@@ -4,8 +4,10 @@
 X-KDE-Submenu=Amarok
 X-KDE-Submenu[bn]=আমারক
 X-KDE-Submenu[pa]=ਅਮਰੋਕ
+X-KDE-Submenu[tr]=Amarok
 X-KDE-Submenu[xx]=xxAmarokxx
 
+
 [Desktop Action appendToPlaylist]
 Name=Append to Playlist
 Name[bg]=Добавяне към списъка с песни
@@ -30,6 +32,7 @@
 Name[sv]=Lägg till i spellistan
 Name[uk]=Додати до списку композицій
 Name[uz]=Қўшиқ рўйхатига қўшиш
+Name[tr]=Çalma Listesine Ekle
 Name[xx]=xxAppend to Playlistxx
 Name[zh_CN]=追加到播放列表
 Icon=amarok
@@ -59,6 +62,7 @@
 Name[sv]=Lägg till och spela
 Name[uk]=Додати і програти
 Name[uz]=Қўшиш ва ўйнаш
+Name[tr]=Ekle ve Çal
 Name[xx]=xxAppend & Playxx
 Name[zh_CN]=追加并播放
 Icon=amarok
@@ -87,6 +91,7 @@
 Name[sv]=Köa spår
 Name[uk]=Додати доріжку в чергу
 Name[uz]=Навбатга қўйиш
+Name[tr]=Parçayı Kuyruğa Ekle
 Name[xx]=xxQueue Trackxx
 Name[zh_CN]=音轨排队
 Icon=amarok
diff -u servicemenus/ark_compress.desktop servicemenus2/ark_compress.desktop
--- servicemenus/ark_compress.desktop	2006-10-22 19:03:05.000000000 +0300
+++ src/servicemenus/ark_compress.desktop	2007-04-09 01:06:45.000000000 +0300
@@ -7,5 +7,6 @@
 [Desktop Action compressHere]
 Name=Compress Here
 Name[de]=Hierher komprimieren
+Name[tr]=Buraya Aç
 Exec=ark --add-to %u %u.tar.gz
 Icon=ark
diff -u servicemenus/ark_extract.desktop servicemenus2/ark_extract.desktop
--- servicemenus/ark_extract.desktop	2006-10-22 19:03:05.000000000 +0300
+++ src/servicemenus/ark_extract.desktop	2007-04-09 01:07:09.000000000 +0300
@@ -7,6 +7,7 @@
 [Desktop Action extractHere]
 Name=Extract Here
 Name[de]=Hierher extrahieren
+Name[tr]=Buraya Aç
 Exec=ark --extract-to %d %u
 Icon=ark
 
diff -u servicemenus/dolphin_su.desktop servicemenus2/dolphin_su.desktop
--- servicemenus/dolphin_su.desktop	2006-10-22 19:03:05.000000000 +0300
+++ src/servicemenus/dolphin_su.desktop	2007-04-09 01:07:46.000000000 +0300
@@ -6,5 +6,6 @@
 ServiceTypes=inode/directory
 Name=Open as Root
 Name[de]=Öffne als Root
+Name[tr]=Root Olarak Aç
 Icon=dolphin
 Exec=kdesu -c 'dolphin %U'
diff -u servicemenus/floppy_format.desktop servicemenus2/floppy_format.desktop
--- servicemenus/floppy_format.desktop	2006-10-22 19:03:05.000000000 +0300
+++ src/servicemenus/floppy_format.desktop	2007-04-09 01:08:24.000000000 +0300
@@ -25,6 +25,7 @@
 Name[sr]=Форматирај
 Name[sr@Latn]=Formatiraj
 Name[sv]=Formatera
+Name[tr]=Biçimlendir
 Name[uk]=Форматування
 Name[xx]=xxFormatxx
 Name[zh_CN]=格式化
diff -u servicemenus/imageconverter.desktop servicemenus2/imageconverter.desktop
--- servicemenus/imageconverter.desktop	2006-10-22 19:03:05.000000000 +0300
+++ src/servicemenus/imageconverter.desktop	2007-04-09 01:09:00.000000000 +0300
@@ -26,6 +26,7 @@
 X-KDE-Submenu[sr]=Претвори у
 X-KDE-Submenu[sr@Latn]=Pretvori u
 X-KDE-Submenu[sv]=Konvertera till
+X-KDE-Submenu[tr]=Dönüştür
 X-KDE-Submenu[uk]=Перетворити в
 X-KDE-Submenu[xx]=xxConvert Toxx
 X-KDE-Submenu[zh_CN]=转换为
diff -u servicemenus/isoservice.desktop servicemenus2/isoservice.desktop
--- servicemenus/isoservice.desktop	2006-10-22 19:03:05.000000000 +0300
+++ src/servicemenus/isoservice.desktop	2007-04-09 01:10:20.000000000 +0300
@@ -7,8 +7,10 @@
 Comment=ISO9660 View
 Comment[hu]=ISO9660 Nézet
 Comment[fr]=Lecteur ISO9660
+Comment[tr]=ISO9660 Göster
 Icon=cd
 Name=ISO9660 View
 Name[hu]=ISO9660 Nézet
 Name[fr]=Lecteur ISO9660
+Name[tr]=ISO9660 Göster
 Exec=
\ Dosya sonunda yenisatır yok.
diff -u servicemenus/jpegorient.desktop servicemenus2/jpegorient.desktop
--- servicemenus/jpegorient.desktop	2006-10-22 19:03:05.000000000 +0300
+++ src/servicemenus/jpegorient.desktop	2007-04-09 01:11:35.000000000 +0300
@@ -22,6 +22,7 @@
 X-KDE-Submenu[sr]=Трансформиши слику
 X-KDE-Submenu[sr@Latn]=Transformiši sliku
 X-KDE-Submenu[sv]=Ändra bild
+X-KDE-Submenu[tr]=Resmi Döndür
 X-KDE-Submenu[uk]=Перетворити зображення
 X-KDE-Submenu[xx]=xxTransform Imagexx
 X-KDE-Submenu[zh_CN]=图像变形
@@ -66,7 +67,7 @@
 Name[sv]=Rotera medurs
 Name[ta]=வலதுபுறமாக சுழற்று
 Name[tg]=Чархиш бо ақрабаки соат
-Name[tr]=Saat Yönünde Döndürme
+Name[tr]=Saat Yönünde Döndür
 Name[uk]=Обернути за годинниковою стрілкою
 Name[uz]=Соат кўрсатгичи бўйича айлантириш
 Name[xx]=xxRotate Clockwisexx
@@ -114,7 +115,7 @@
 Name[sv]=Rotera moturs
 Name[ta]=கடிகார ஓட்டத்திற்கெதிராக சுழற்று
 Name[tg]=Чархиш ба муқобили ақрабаки соат
-Name[tr]=Sayacı Saat Yönünde Döndürme
+Name[tr]=Saat Yönünün Tersine Döndür
 Name[uk]=Обернути проти годинникової стрілки
 Name[uz]=Соат кўрсатгичига қарши айлантириш
 Name[xx]=xxRotate Counter-Clockwisexx
@@ -166,7 +167,7 @@
 Name[sv]=Vänd vertikalt
 Name[ta]=மேல்கீழாக திருப்பு
 Name[tg]=Баргардондан ба амудӣ
-Name[tr]=Dik Olarak Dağıtma
+Name[tr]=Dikey Olarak Ters Çevir
 Name[uk]=Перекинути вертикально
 Name[uz]=Эни бўйича айлантириш
 Name[xx]=xxFlip Verticallyxx
@@ -213,7 +214,7 @@
 Name[sv]=Vänd horisontellt
 Name[ta]=இடம்வலமாக திருப்புக
 Name[tg]=Баргардондан ба уфуқӣ
-Name[tr]=Yatay Olarak Dağıtma
+Name[tr]=Yatay Olarak Ters Çevir
 Name[uk]=Перекинути горизонтально
 Name[uz]=Бўйи бўйича айлантириш
 Name[xx]=xxFlip Horizontallyxx
diff -u servicemenus/mail_as_attachment.desktop servicemenus2/mail_as_attachment.desktop
--- servicemenus/mail_as_attachment.desktop	2006-10-22 19:03:05.000000000 +0300
+++ src/servicemenus/mail_as_attachment.desktop	2007-04-09 01:04:53.000000000 +0300
@@ -6,5 +6,6 @@
 [Desktop Action mail_as_attachment]
 Name=Mail as Attachment...
 Name[de]=Verschicken als Anhang...
+Name[tr]=Eklenti olarak E-posta ile gönder...
 Icon=mail_send
 Exec=kmail --attach %U
diff -u servicemenus/media_realfolder.desktop servicemenus2/media_realfolder.desktop
--- servicemenus/media_realfolder.desktop	2006-10-22 19:03:05.000000000 +0300
+++ src/servicemenus/media_realfolder.desktop	2007-04-09 01:03:31.000000000 +0300
@@ -103,6 +103,7 @@
 Name[sr]=Отвори системску фасциклу медијума
 Name[sr@Latn]=Otvori sistemsku fasciklu medijuma
 Name[sv]=Öppna mediumsystemkatalog
+Name[tr]=Aygıt Sistem Dizini Aç
 Name[uk]=Відкрити системну теку носія
 Name[zh_CN]=打开介质系统文件夹
 Exec=kio_media_realfolder %u
diff -u servicemenus/open_in_digikam.desktop servicemenus2/open_in_digikam.desktop
--- servicemenus/open_in_digikam.desktop	2006-10-22 19:03:05.000000000 +0300
+++ src/servicemenus/open_in_digikam.desktop	2007-04-09 01:01:59.000000000 +0300
@@ -22,6 +22,7 @@
 Name[uk]=Відкрити в Digikam
 Name[zh_CN]=在 Digikam 中打开
 Name[zh_TW]=在 Digikam 中開啟
+Name[tr]=Digikam ile Aç
 [Desktop Entry]
 Actions=OpenDigikam;
 ServiceTypes=media/gphoto2camera,media/camera_mounted,media/camera_unmounted
