diff -Nuar thunar-vcs-plugin-0.1.0.orig/thunar-vcs-plugin/Makefile.in thunar-vcs-plugin-0.1.0/thunar-vcs-plugin/Makefile.in
--- thunar-vcs-plugin-0.1.0.orig/thunar-vcs-plugin/Makefile.in	2009-09-28 11:55:12.812296915 +0300
+++ thunar-vcs-plugin-0.1.0/thunar-vcs-plugin/Makefile.in	2009-09-28 11:58:06.510297657 +0300
@@ -90,7 +90,12 @@
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(extensionsdir)"
 LTLIBRARIES = $(extensions_LTLIBRARIES)
-thunar_vcs_plugin_la_LIBADD =
+thunar_vcs_plugin_la_LIBADD = \
+	$(GLIB_LIBS) \
+	$(THUNARX_LIBS) \
+	$(THUNAR_VFS_LIBS) \
+	$(APR_LIBS)
+
 am__thunar_vcs_plugin_la_SOURCES_DIST = tvp-provider.c tvp-provider.h \
 	thunar-vcs-plugin.c tvp-svn-backend.c tvp-svn-backend.h \
 	tvp-svn-action.c tvp-svn-action.h tvp-svn-property-page.c \
