--- configure.ac	2009-05-07 23:03:29.000000000 +0200
+++ configure.ac.new	2009-05-08 00:34:02.000000000 +0200
@@ -20,18 +20,19 @@
               [pamuserpass=true], [pamuserpass=false])
 AM_CONDITIONAL(SESMAN_PAMUSERPASS, [test x$pamuserpass = xtrue])
 libdir="${libdir}/xrdp/";
+AC_DEFINE_UNQUOTED([XRDP_LIB_PATH], ["${libdir}"], [xrdp lib dir])
 if test "x${prefix}" = "xNONE" ; then
 sysconfdir="/etc/";
 else
-AC_DEFINE_UNQUOTED([XRDP_CFG_PATH], ["${prefix}/etc/xrdp"], [xrdp config dir])
-AC_DEFINE_UNQUOTED([XRDP_CFG_FILE], ["${prefix}/etc/xrdp/xrdp.ini"], [xrdp config file])
-AC_DEFINE_UNQUOTED([XRDP_KEY_FILE], ["${prefix}/etc/xrdp/rsakeys.ini"], [xrdp key file])
-AC_DEFINE_UNQUOTED([XRDP_KEYMAP_FILE], ["${prefix}/etc/xrdp/km-%4.4x.ini"], [xrdp keymap file])
-AC_DEFINE_UNQUOTED([XRDP_PID_FILE], ["${prefix}/var/run/xrdp.pid"], [xrdp pid file])
+AC_DEFINE_UNQUOTED([XRDP_CFG_PATH], ["${sysconfdir}/xrdp"], [xrdp config dir])
+AC_DEFINE_UNQUOTED([XRDP_CFG_FILE], ["${sysconfdir}/xrdp/xrdp.ini"], [xrdp config file])
+AC_DEFINE_UNQUOTED([XRDP_KEY_FILE], ["${sysconfdir}/xrdp/rsakeys.ini"], [xrdp key file])
+AC_DEFINE_UNQUOTED([XRDP_KEYMAP_FILE], ["${sysconfdir}/xrdp/km-%4.4x.ini"], [xrdp keymap file])
+AC_DEFINE_UNQUOTED([XRDP_PID_FILE], ["${localstatedir}/run/xrdp.pid"], [xrdp pid file])
 AC_DEFINE_UNQUOTED([XRDP_SBIN_PATH], ["${prefix}/sbin"], [xrdp sbin dir])
 AC_DEFINE_UNQUOTED([XRDP_SHARE_PATH], ["${prefix}/share/xrdp"], [xrdp share dir])
-AC_DEFINE_UNQUOTED([SESMAN_PID_FILE], ["${prefix}/var/run/sesman.pid"], [sesman pid file])
-AC_DEFINE_UNQUOTED([SESMAN_CFG_FILE], ["${prefix}/etc/xrdp/sesman.ini"], [sesman config file])
+AC_DEFINE_UNQUOTED([SESMAN_PID_FILE], ["${localstatedir}/run/sesman.pid"], [sesman pid file])
+AC_DEFINE_UNQUOTED([SESMAN_CFG_FILE], ["${sysconfdir}/xrdp/sesman.ini"], [sesman config file])
 fi
 AC_CONFIG_FILES([Makefile
                  common/Makefile
