--- instfiles/Makefile.am	2008-08-18 01:05:14.000000000 +0200
+++ instfiles/Makefile.am.new	2009-05-07 01:11:03.000000000 +0200
@@ -4,12 +4,3 @@
 
 startscriptdir=$(sysconfdir)/xrdp
 
-startscript_DATA = \
-  xrdp.sh
-
-# must be tab below
-install-data-hook:
-	chmod 755 $(DESTDIR)$(sysconfdir)/xrdp/xrdp.sh
-	if [ -w /etc/init.d ]; then rm -f /etc/init.d/xrdp.sh; fi
-	if [ -w /etc/init.d ]; then $(LN_S) $(DESTDIR)$(sysconfdir)/xrdp/xrdp.sh /etc/init.d/xrdp.sh; fi
-	$(DESTDIR)$(bindir)/xrdp-keygen xrdp $(DESTDIR)$(sysconfdir)/xrdp/rsakeys.ini
