--- Kino.desktop	2006-01-14 04:18:23.000000000 +0200
+++ Kino-yeni.desktop	2006-07-11 01:36:20.000000000 +0300
@@ -15,6 +15,7 @@
 GenericName[nb]=Videoredigering
 GenericName[no]=Videoredigerer
 GenericName[ru]=Видеоредактор
+GenerıcNAme[tr]=Video Düzenleyicisi
 Comment=Edit DV video
 Comment[ca]=Edita vídeos DV
 Comment[cs]=Nástroj pro úpravy DV videa
@@ -30,6 +31,7 @@
 Comment[nb]=Redigering av DV video
 Comment[no]=Redigere DV video
 Comment[ru]=Редактирование DV видео
+Comment[tr]=DV video düzenleyicisi
 Encoding=UTF-8
 TryExec=kino
 Exec=kino %F
