diff -Nuar itk3.3.orig/configure itk3.3/configure
--- itk3.3.orig/configure	2005-03-25 21:39:53.000000000 +0200
+++ itk3.3/configure	2007-08-05 13:38:38.000000000 +0300
@@ -7288,7 +7288,7 @@
 	    # results, and the version is kept in special file).
 
 	    if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
-		system=MP-RAS-`awk '{print }' /etc/.relid'`
+		system=MP-RAS-`awk '{print }' /etc/.relid`
 	    fi
 	    if test "`uname -s`" = "AIX" ; then
 		system=AIX-`uname -v`.`uname -r`
diff -Nuar itk3.3.orig/tclconfig/tcl.m4 itk3.3/tclconfig/tcl.m4
--- itk3.3.orig/tclconfig/tcl.m4	2005-04-05 21:07:03.000000000 +0300
+++ itk3.3/tclconfig/tcl.m4	2007-08-05 13:38:38.000000000 +0300
@@ -794,7 +794,7 @@
 	    # results, and the version is kept in special file).
 	
 	    if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
-		system=MP-RAS-`awk '{print $3}' /etc/.relid'`
+		system=MP-RAS-`awk '{print $3}' /etc/.relid`
 	    fi
 	    if test "`uname -s`" = "AIX" ; then
 		system=AIX-`uname -v`.`uname -r`
@@ -2208,7 +2208,7 @@
 	    # results, and the version is kept in special file).
 	
 	    if test -r /etc/.relid -a "X`uname -n`" = "X`uname -s`" ; then
-		system=MP-RAS-`awk '{print $3}' /etc/.relid'`
+		system=MP-RAS-`awk '{print $3}' /etc/.relid`
 	    fi
 	    if test "`uname -s`" = "AIX" ; then
 		system=AIX-`uname -v`.`uname -r`
