Index: qbittorrent-2.4.4/src/GUI.cpp
===================================================================
--- qbittorrent-2.4.4.orig/src/GUI.cpp
+++ qbittorrent-2.4.4/src/GUI.cpp
@@ -408,7 +408,7 @@ void GUI::on_actionDocumentation_trigger
 }
 
 void GUI::on_actionBugReport_triggered() const {
-  QDesktopServices::openUrl(QUrl(QString::fromUtf8("http://bugs.qbittorrent.org")));
+  QDesktopServices::openUrl(QUrl(QString::fromUtf8("http://bugs.pardus.org.tr")));
 }
 
 void GUI::tab_changed(int new_tab) {
