--- eric/UI/UserInterface.py.orig	2008-02-24 16:20:17.000000000 +0200
+++ eric/UI/UserInterface.py	2008-02-24 16:23:31.000000000 +0200
@@ -1147,6 +1147,7 @@
             """<b>Check for Updates...</b>"""
             """<p>Checks the internet for updates of eric4.</p>"""
                              ))
+        self.checkUpdateAct.setVisible(False)
         self.checkUpdateAct.connectIt(SIGNAL('triggered()'), self.performVersionCheck)
         self.actions.append(self.checkUpdateAct)
     
@@ -1160,6 +1161,7 @@
             """<p>Shows the eric4 versions available for download """
             """from the internet.</p>"""
                              ))
+        self.showVersionsAct.setVisible(False)
         self.showVersionsAct.connectIt(SIGNAL('triggered()'), 
                 self.showAvailableVersionsInfo)
         self.actions.append(self.showVersionsAct)
