--- desktop/Makefile.am 2006-03-07 20:31:26.000000000 +0000
+++ desktop/Makefile.am 2007-05-02 00:00:11.000000000 +0100
@@ -3,7 +3,6 @@
 
 # We install the .desktop file and we change the icon directory
 install-data-local : 
-	cat briquolo.desktop | sed -e "s#Icon=briquolo.svg#Icon=$(datadir)/pixmaps/briquolo.svg#g" > $(DESTDIR)$(datadir)/applications/briquolo.desktop
 
 uninstall-local :
 	rm -f $(DESTDIR)$(datadir)/applications/briquolo.desktop
