diff -Naur keepassx-0.3.3.orig/src/src.pro keepassx-0.3.3/src/src.pro
--- keepassx-0.3.3.orig/src/src.pro	2008-05-31 23:42:10.000000000 +0300
+++ keepassx-0.3.3/src/src.pro	2008-10-04 19:04:22.000000000 +0300
@@ -146,6 +146,7 @@
                 translations/keepassx-fr_FR.ts \
                 translations/keepassx-cs_CZ.ts \
                 translations/keepassx-ja_JP.ts \
+                translations/keepassx-tr_TR.ts \
-                translations/keepassx-xx_XX.ts
+               #translations/keepassx-xx_XX.ts

 HEADERS += lib/UrlLabel.h \
