diff -Naur keepassx-0.4.0.orig/src/translations/keepassx-tr_TR.ts keepassx-0.4.0/src/translations/keepassx-tr_TR.ts
--- keepassx-0.4.0.orig/src/translations/keepassx-tr_TR.ts	2009-03-28 01:13:07.000000000 +0200
+++ keepassx-0.4.0/src/translations/keepassx-tr_TR.ts	2009-03-28 01:34:26.000000000 +0200
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!DOCTYPE TS><TS version="1.1" language="tr_TR">
+<defaultcodec></defaultcodec>
 <context>
     <name>AboutDialog</name>
     <message>
@@ -1509,7 +1510,7 @@
     <message>
         <location filename="../Kdb3Database.cpp" line="1333"/>
         <source>The database has been opened read-only.</source>
-        <translation type="unfinished"></translation>
+        <translation>Veritabanı salt okunur olarak açık.</translation>
     </message>
 </context>
 <context>
@@ -1914,7 +1915,10 @@
 This means that either someone else has opened the file or KeePassX crashed last time it opened the database.
 
 Do you want to open it anyway?</source>
-        <translation type="unfinished"></translation>
+        <translation>Açmaya çalıştığınız veritabanı kilitli.
+Veritabanı başka bir yerde açık ya da KeePassX kapanma önce çökmüş olabilir.
+
+Yine de bu veritabanını açmak ister misiniz ?</translation>
     </message>
     <message>
         <location filename="../mainwindow.cpp" line="448"/>
@@ -1936,12 +1940,12 @@
     <message>
         <location filename="../mainwindow.cpp" line="414"/>
         <source>Open read-only</source>
-        <translation type="unfinished"></translation>
+        <translation>Salt-okunur olarak aç</translation>
     </message>
     <message>
         <location filename="../mainwindow.cpp" line="1429"/>
         <source>Couldn&apos;t create lock file. Opening the database read-only.</source>
-        <translation type="unfinished"></translation>
+        <translation>Kilit dosyası oluşturulamadı. Veritabanı salt-okunur olarak açılıyor.</translation>
     </message>
 </context>
 <context>
@@ -2302,17 +2306,17 @@
     <message>
         <location filename="../forms/MainWindow.ui" line="172"/>
         <source>&amp;Entries</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Girdiler</translation>
     </message>
     <message>
         <location filename="../forms/MainWindow.ui" line="240"/>
         <source>&amp;Groups</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Gruplar</translation>
     </message>
     <message>
         <location filename="../forms/MainWindow.ui" line="594"/>
         <source>Sort groups</source>
-        <translation type="unfinished"></translation>
+        <translation>Grupları sırala</translation>
     </message>
 </context>
 <context>
@@ -2744,7 +2748,7 @@
     <message>
         <location filename="../forms/SettingsDlg.ui" line="1260"/>
         <source>You can disable several features of KeePassX here according to your needs in order to keep the user interface slim.</source>
-        <translation type="unfinished"></translation>
+        <translation>Buradan istediğiniz özellikleri ekleyip, kaldırarak KeePassX arayüzünü düzenleyebilirsiniz.</translation>
     </message>
     <message>
         <location filename="../forms/SettingsDlg.ui" line="1270"/>
@@ -2844,7 +2848,7 @@
     <message>
         <location filename="../forms/SettingsDlg.ui" line="1116"/>
         <source>Enable this if you want to use your bookmarks and the last opened file independet from their absolute paths. This is especially useful when using KeePassX portably and therefore with changing mount points in the file system.</source>
-        <translation type="unfinished"></translation>
+        <translation>Kendi yer imlerinizi kullanmak istiyorsanız bunu seçin. Bu özellik daha çok taşınabilir KeePassX ya da bağlanma noktası değişken durumlarda  için uygundur.</translation>
     </message>
     <message>
         <location filename="../forms/SettingsDlg.ui" line="1119"/>
@@ -2999,7 +3003,8 @@
         <location filename="../forms/TargetWindowDlg.ui" line="19"/>
         <source>To specify the target window, either select an existing currently-opened window
 from the drop-down list, or enter the window title manually:</source>
-        <translation type="unfinished"></translation>
+        <translation>Hedef pencereyi seçmek için: açılır menüden açık olan bir pencereyi seçin
+ya da pencere başlığını girin:</translation>
     </message>
 </context>
 <context>
