--- gnome/gretl.desktop.orig	2006-11-26 22:35:02.000000000 +0200
+++ gnome/gretl.desktop	2007-09-21 00:59:35.000000000 +0300
@@ -1,9 +1,12 @@
 [Desktop Entry]
-Name=gretl
-Comment=Econometrics program
-Exec=gretl
-Icon=gretl.xpm
-Terminal=0
 Type=Application
+Version=1.0
+Encoding=UTF-8
+Name=Gretl
+GenericName=Econometrics program
+GenericName[tr]=Ekonometri programı
+Exec=gretl
+Icon=gretl.png
+Terminal=false
+StartupNotify=false
 Categories=Application;Science;Econometrics
-
