--- gui2/gretl.c.orig	2007-05-15 15:54:59.000000000 +0300
+++ gui2/gretl.c	2007-09-21 02:20:12.000000000 +0300
@@ -140,10 +140,10 @@
 char calculator[MAXSTR] = "xcalc";
 char latex[MAXSTR] = "latex";
 char viewdvi[MAXSTR] = "xdvi";
-char viewps[MAXSTR] = "gv";
-char viewpdf[MAXSTR] = "acroread";
-char Rcommand[MAXSTR] = "xterm -e R";
-char Browser[MAXSTR] = "mozilla";
+char viewps[MAXSTR] = "kghostview ";
+char viewpdf[MAXSTR] = "kpdf";
+char Rcommand[MAXSTR] = "konsole -e R";
+char Browser[MAXSTR] = "firefox";
 #endif
 
 static void spreadsheet_edit (gpointer p, guint u, GtkWidget *w) 
