--- /dev/null	2007-09-10 18:16:35.844002276 +0300
+++ src/translations/keepass-tr_TR.ts	2007-09-10 20:22:57.000000000 +0300
@@ -0,0 +1,2156 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE TS><TS version="1.1" language="tr_TR">
+<defaultcodec></defaultcodec>
+<context>
+    <name>AboutDlg</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>About</source>
+        <translation>Hakkında</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>License</source>
+        <translation>Lisans</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Translation</source>
+        <translation>Çeviri</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;/head&gt;&lt;body style=&quot; white-space: pre-wrap; font-family:Sans Serif; font-size:9pt; font-weight:400; font-style:normal; text-decoration:none;&quot;&gt;&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:12pt; font-weight:600;&quot;&gt;KeePassX&lt;/span&gt; - Cross Platform Password Manager&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
+        <translation>&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;/head&gt;&lt;body style=&quot; white-space: pre-wrap; font-family:Sans Serif; font-size:9pt; font-weight:400; font-style:normal; text-decoration:none;&quot;&gt;&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:12pt; font-weight:600;&quot;&gt;KeePassX&lt;/span&gt; Parola Yöneticisi&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Copyright (C) 2005 - 2006 Tarek Saidi 
+KeePassX is distributed under the terms of the 
+General Public License (GPL) version 2.</source>
+        <translation>Copyright (C) 2005 - 2006 Tarek Saidi 
+KeePassX Genel Kamu Lisansı (GPL) 2. sürümü
+altında dağıtılmaktadır.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Credits</source>
+        <translation>Katkıda Bulunanlar</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>http://keepassx.sourceforge.net</source>
+        <translation>http://keepassx.sourceforge.net</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>keepassx@gmail.com</source>
+        <translation>keepassx@gmail.com</translation>
+    </message>
+</context>
+<context>
+    <name>AutoType</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>More than one &apos;Auto-Type:&apos; key sequence found.
+Allowed is only one per entry.</source>
+        <translation>Birden fazla &apos;Otomatik-Tip&apos;: anahtar sıralamasıbulundu.
+Her girdi için bir taneye izin veriliyor.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Syntax Error in Auto-Type sequence near character %1
+<byte value="x9"/><byte value="x9"/><byte value="x9"/><byte value="x9"/><byte value="x9"/>Found &apos;{&apos; without closing &apos;}&apos;</source>
+        <translation>Otomatik-Tip sıralamasında %1 karakterinin yanında söz dizimi hatası
+<byte value="x9"/>bulundu &apos;{&apos; without closing &apos;}&apos;</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Error</source>
+        <translation>Hata</translation>
+    </message>
+</context>
+<context>
+    <name>CAboutDialog</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>KeePassX %1</source>
+        <translation>KeePassX %1</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Error</source>
+        <translation>Hata</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>File &apos;%1&apos; could not be found.</source>
+        <translation>&apos;%1&apos; dosyası bulunamadı.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Make sure that the program is installed correctly.</source>
+        <translation>Uygulamanın doğru yüklendiğinden emin olun.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>OK</source>
+        <translation>TAMAM</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Could not open file &apos;%1&apos;</source>
+        <translation>&apos;%1&apos; dosyası açılamadı</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>The following error occured:
+%1</source>
+        <translation>Şu hata oluştu:
+%1</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&lt;b&gt;Current Translation: None&lt;/b&gt;&lt;br&gt;&lt;br&gt;</source>
+        <comment>Please replace &apos;None&apos; with the language of your translation</comment>
+        <translation>&lt;b&gt;Geçerli Çeviri: Türkçe&lt;/b&gt;&lt;br&gt;&lt;br&gt;</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&lt;b&gt;Author:&lt;/b&gt; %1&lt;br&gt;</source>
+        <translation>&lt;b&gt;Çeviren:&lt;/b&gt; %1&lt;br&gt;</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>$TRANSLATION_AUTHOR_EMAIL</source>
+        <comment>Here you can enter your email or homepage if you want.</comment>
+        <translation>tulliana@gmail.com</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Matthias Miller</source>
+        <translation>Matthias Miller</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Eugen Gorschenin</source>
+        <translation>Eugen Gorschenin</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>$TRANSLATION_AUTHOR</source>
+        <translation>Serdar Soytetir</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Team</source>
+        <translation>Takım</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Tarek Saidi</source>
+        <translation>Tarek Saidi</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Developer, Project Admin</source>
+        <translation>Geliştirici, Proje Yöneticisi</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>tariq@users.berlios.de</source>
+        <translation>tariq@users.berlios.de</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Web Designer</source>
+        <translation>Web Sayfası</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>geugen@users.berlios.de</source>
+        <translation>geugen@users.berlios.de</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Thanks To</source>
+        <translation>Teşekkürler</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Patches for better MacOS X support</source>
+        <translation>Daha iyi MacOS X desteği için yamalar</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>www.outofhanwell.com</source>
+        <translation>www.outofhanwell.com</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Information on how to translate KeePassX can be found under:
+http://keepassx.sourceforge.net/</source>
+        <translation>KeePassX uygulamasının nasıl yerelleştirileceği ile ilgili bilgiyi burada bulabilirsiniz:
+http://keepassx.sourceforge.net/</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>James Nicholls</source>
+        <translation>James Nicholls</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Main Application Icon</source>
+        <translation>Ugulamanın Simgesi</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>http://keepassx.sf.net</source>
+        <translation>http://keepassx.sf.net</translation>
+    </message>
+</context>
+<context>
+    <name>CDbSettingsDlg</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>AES(Rijndael):  256 Bit   (default)</source>
+        <translation>AES(Rijndael):  256 Bit   (öntanımlı)</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Twofish:  256 Bit</source>
+        <translation> Twofish:  256 Bit</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Warning</source>
+        <translation>Uyarı</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Please determine the number of encryption rounds.</source>
+        <translation>Lütfen şifrelemenin kaç basamak ile yapılacağını belirtin.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>OK</source>
+        <translation>TAMAM</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Error</source>
+        <translation>Hata</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&apos;%1&apos; is not valid integer value.</source>
+        <translation>&apos;%1&apos; geçerli bir tam sayı değeri değil.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>The number of encryption rounds have to be greater than 0.</source>
+        <translation>Sıfırdan büyük olması gereken ve şifrelemenin kaç basamaklı olacağını belirten sayı.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Settings</source>
+        <translation>Ayarlar</translation>
+    </message>
+</context>
+<context>
+    <name>CEditEntryDlg</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Warning</source>
+        <translation>Uyarı</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Password and password repetition are not equal.
+Please check your input.</source>
+        <translation>Parola ve parola tekrarı birbirini tutmuyor.
+Lütfen girdilerinizi kontrol edin.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>OK</source>
+        <translation>TAMAM</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Save Attachment...</source>
+        <translation>Eklentiyi kaydet...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Overwrite?</source>
+        <translation>Üzerine Yazılsın mı?</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>A file with this name already exists.
+Do you want to replace it?</source>
+        <translation>Aynı isim ile kaydedilmiş bir dosya zaten var.
+O dosyayı silip yerine yazmak ister misininiz?</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Yes</source>
+        <translation>Evet</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>No</source>
+        <translation>Hayır</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Error</source>
+        <translation>Hata</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Could not remove old file.</source>
+        <translation>Eski dosya silinemedi.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Could not create new file.</source>
+        <translation>Yeni dosya oluşturulamadı.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Error while writing the file.</source>
+        <translation>Dosya yazılırken hata.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Delete Attachment?</source>
+        <translation>Eklenti silinsin mi?</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>You are about to delete the attachment of this entry.
+Are you sure?</source>
+        <translation>Bu girdinin eklentisini silmek üzeresiniz.
+Emin misiniz?</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>No, Cancel</source>
+        <translation>Hayır, İptal Et</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Edit Entry</source>
+        <translation>Girdiyi Düzenle</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Could not open file.</source>
+        <translation>Dosya açılamadı.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>%1 Bit</source>
+        <translation>%1 Bit</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Add Attachment...</source>
+        <translation>Eklenti Ekle...</translation>
+    </message>
+</context>
+<context>
+    <name>CGenPwDialog</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Notice</source>
+        <translation>Dikkat</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>You need to enter at least one character</source>
+        <translation>En az bir karakter girmeniz gerekiyor</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>OK</source>
+        <translation>TAMAM</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Error</source>
+        <translation>Hata</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Could not open &apos;/dev/random&apos; or &apos;/dev/urandom&apos;.</source>
+        <translation>&apos;/dev/random&apos; ya da /dev/urandom&apos; açılamadı.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Password Generator</source>
+        <translation>Parola Oluşturucu</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>%1 Bit</source>
+        <translation>%1 Bit</translation>
+    </message>
+</context>
+<context>
+    <name>CPasswordDialog</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>OK</source>
+        <translation>TAMAM</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Error</source>
+        <translation>Hata</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Please enter a Password.</source>
+        <translation>Lütfen bir parola girin.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Please choose a key file.</source>
+        <translation>Lütfen bir anahtar dosyası seçin.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Please enter a Password or select a key file.</source>
+        <translation>Lütfen gir parola girin ya da anahtar dosyası seçin.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Database Key</source>
+        <translation>Veritabanı Anahtarı</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Select a Key File</source>
+        <translation>Anahtar dosyasını Seç</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>*.key</source>
+        <translation>*.key</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unexpected Error: File does not exist.</source>
+        <translation>Beklenmeyen hata: Dosya yok.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>The selected key file or directory does not exist.</source>
+        <translation>Seçilen anahtar dosyası ya da dizin yok.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>The given directory does not contain any key files.</source>
+        <translation>Verilen dizin bir anahtar dosyası içermiyor.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>The given directory contains more then one key file.
+Please specify the key file directly.</source>
+        <translation>Verilen dizin birden fazla anahtar dosyası içeriyor.
+Lütfen anahtar dosyasını direkt olarak belirtin.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>The key file found in the given directory is not readable.
+Please check your permissions.</source>
+        <translation>Verilen dizindeki anahtar dosyası okunabilir değil.
+Lütfen izinleriniz kontrol edin.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Key file could not be found.</source>
+        <translation>Anahtar dosyası bulunamadı.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Key file is not readable.
+Please check your permissions.</source>
+        <translation>Anahtar dosyası okunabilir değil.
+Lütfen izinlerinizi kontrol edin.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Warning</source>
+        <translation>Uyarı</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Password an password repetition are not equal.
+Please check your input.</source>
+        <translation>Parola ve parola tekrarı birbirini tutmuyor.
+Lütfen girdilerinizi kontrol edin.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Please enter a password or select a key file.</source>
+        <translation>Lütfen bir parola girin ya da anahtar dosyası seçin.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>A file with the name &apos;pwsafe.key&apos; already exisits in the given directory.
+Do you want to replace it?</source>
+        <translation>Verilen dizinde &apos;pwsafe.key&apos; isimli bir dosya zaten var.
+Bu dosyayı silip yerine yeni dosyayı yazmak istiyor musunuz?</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Yes</source>
+        <translation>Evet</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>No</source>
+        <translation>Hayır</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>The exisiting file is not writable.</source>
+        <translation>Var olan dosya yazılabilir değil.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>A file with the this name already exisits.
+Do you want to replace it?</source>
+        <translation>Bu ismi kullanan bir dosya zaten var.
+Bu dosyayı silip, yerine yeni dosyayı yazmak ister misiniz?</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>The selected key file or directory is not readable.
+Please check your permissions.</source>
+        <translation>Seçilen dizin ya da dosya yazılabilir değil.
+Lütfen izinlerinizi kontrol edin.</translation>
+    </message>
+</context>
+<context>
+    <name>CSearchDlg</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Notice</source>
+        <translation>Dikkat</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Please enter a search string.</source>
+        <translation>Lütfen aramak için bir ifade girin.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>OK</source>
+        <translation>TAMAM</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Search</source>
+        <translation>ARA</translation>
+    </message>
+</context>
+<context>
+    <name>CSelectIconDlg</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Delete</source>
+        <translation>Sil</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Add Icons...</source>
+        <translation>Simge Ekle...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Images (%1)</source>
+        <translation>Resimler (%1)</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>%1: File could not be loaded.
+</source>
+        <translation>%1: Dosya yüklenemedi.
+</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Error</source>
+        <translation>Hata</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Replace...</source>
+        <translation>Değiştir...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>An error occured while loading the icon(s):
+%1</source>
+        <translation>Simgeler yüklenirken bir hata oluştu:
+%1</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>An error occured while loading the icon.</source>
+        <translation>Simge yüklenirken bir hata oluştu.</translation>
+    </message>
+</context>
+<context>
+    <name>CSettingsDlg</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Settings</source>
+        <translation>Ayarlar</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Select a directory...</source>
+        <translation>Bir dizin seçin...</translation>
+    </message>
+</context>
+<context>
+    <name>EditEntryDialog</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Edit Entry</source>
+        <translation>Girdiyi Düzenle</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Username:</source>
+        <translation>Kullanıcı adı:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Password Repet.:</source>
+        <translation>Parola (Yeniden):</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Title:</source>
+        <translation>Başlık:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>URL:</source>
+        <translation>Adres:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Password:</source>
+        <translation>Parola:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Quality:</source>
+        <translation>Kalite:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Comment:</source>
+        <translation>Açıklama:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Expires:</source>
+        <translation>Süre Dolumu:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Group:</source>
+        <translation>Grup:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Cancel</source>
+        <translation>İpta&amp;l</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+C</source>
+        <translation>Alt+L</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>%1</source>
+        <translation>%1</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Icon:</source>
+        <translation>Simge:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ge&amp;n.</source>
+        <translation>Olu&amp;ştur.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>...</source>
+        <translation>...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>O&amp;K</source>
+        <translation>TAMA&amp;M</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+K</source>
+        <translation>Alt+M</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Never</source>
+        <translation>Asla</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Attachment:</source>
+        <translation>Eklenti:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&gt;</source>
+        <translation>&gt;</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>%1 Bit</source>
+        <translation>%1 Bit</translation>
+    </message>
+</context>
+<context>
+    <name>EditGroupDialog</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Group Properties</source>
+        <translation>Grup Özellikleri</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Title:</source>
+        <translation>Başlık:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Icon:</source>
+        <translation>Simge:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Cancel</source>
+        <translation>İpta&amp;l</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+C</source>
+        <translation>Alt+L</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>O&amp;K</source>
+        <translation>TAMA&amp;M</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+K</source>
+        <translation>Alt+M</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&gt;</source>
+        <translation>&gt;</translation>
+    </message>
+</context>
+<context>
+    <name>Export_Txt</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Could not open file (FileError=%1)</source>
+        <translation>Dosya açılamadı (Dosya Hatası=%1)</translation>
+    </message>
+</context>
+<context>
+    <name>GenPwDlg</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+U</source>
+        <translation>Alt+B</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+N</source>
+        <translation>Alt+M</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+M</source>
+        <translation>Alt+U</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+L</source>
+        <translation>Alt+K</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Password Generator</source>
+        <translation>Parola Oluşturucu</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Accep&amp;t</source>
+        <translation>Kabul &amp;Et</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Cancel</source>
+        <translation>İpta&amp;l</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Generate</source>
+        <translation>Oluştur</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>New Password:</source>
+        <translation>Yeni Parola:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Quality:</source>
+        <translation>Kalite:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Options</source>
+        <translation>Seçenekler</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Upper Letters</source>
+        <translation>&amp;Büyük Harfler</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Lower Letters</source>
+        <translation>&amp;Küçük Harfler</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Numbers</source>
+        <translation>Raka&amp;mlar</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Special Characters</source>
+        <translation>Ö&amp;zel Karakterler</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Minus</source>
+        <translation>Eksi</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>U&amp;nderline</source>
+        <translation>Altı &amp;çizili</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>h&amp;igher ANSI-Characters</source>
+        <translation>Y&amp;üksek ANSI-Karakterler</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+H</source>
+        <translation>Alt+Ü</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Use &amp;only following characters:</source>
+        <translation>Sadece bu karak&amp;terleri kullan:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+O</source>
+        <translation>Alt+T</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Length:</source>
+        <translation>Uzunluk:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Use &quot;/dev/rando&amp;m&quot;</source>
+        <translation>&quot;/dev/random&quot; k&amp;ullan</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Use follo&amp;wing character groups:</source>
+        <translation>&amp;Şu karakter gruplarını kullan:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+W</source>
+        <translation>Alt+Ş</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>White &amp;Spaces</source>
+        <translation>Beyaz Alanla&amp;r</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+S</source>
+        <translation>Alt+R</translation>
+    </message>
+</context>
+<context>
+    <name>KeepassEntryView</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Title</source>
+        <translation>Başlık</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Username</source>
+        <translation>Kullanıcı Adı</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>URL</source>
+        <translation>Adres</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Password</source>
+        <translation>Parola</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Comments</source>
+        <translation>Açıklama</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Expires</source>
+        <translation>Süre Dolumu</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Creation</source>
+        <translation>Oluşturulma</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Last Change</source>
+        <translation>Son Değişiklik</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Last Access</source>
+        <translation>Son Erişim</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Attachment</source>
+        <translation>Eklenti</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>%1 items</source>
+        <translation>%1 öge</translation>
+    </message>
+</context>
+<context>
+    <name>KeepassGroupView</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Search Results</source>
+        <translation>Arama Sonuçları</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Groups</source>
+        <translation>Gruplar</translation>
+    </message>
+</context>
+<context>
+    <name>KeepassMainWindow</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+N</source>
+        <translation>Ctrl+N</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+O</source>
+        <translation>Ctrl+O</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+S</source>
+        <translation>Ctrl+S</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+G</source>
+        <translation>Ctrl+G</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+C</source>
+        <translation>Ctrl+C</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+B</source>
+        <translation>Ctrl+B</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+U</source>
+        <translation>Ctrl+U</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+Y</source>
+        <translation>Ctrl+Y</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+E</source>
+        <translation>Ctrl+E</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+D</source>
+        <translation>Ctrl+D</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+K</source>
+        <translation>Ctrl+K</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+F</source>
+        <translation>Ctrl+F</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+W</source>
+        <translation>Ctrl+W</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Shift+Ctrl+S</source>
+        <translation>Shift+Ctrl+S</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Shift+Ctrl+F</source>
+        <translation>Shift+Ctrl+F</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Error</source>
+        <translation>Hata</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>The following error occured while opening the database:
+%1</source>
+        <translation>Veritabanı açılırken hata oluştu:
+%1</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>OK</source>
+        <translation>TAMAM</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Save modified file?</source>
+        <translation>Değiştirilmiş dosya kaydeilsin mi?</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>The current file was modified. Do you want
+to save the changes?</source>
+        <translation>Geçerli dosya üzerinde değişiklikler yapıldı. Değişiklikleri
+kaydetmek ister misiniz?</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Yes</source>
+        <translation>Evet</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>No</source>
+        <translation>Hayır</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Cancel</source>
+        <translation>İptal</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>KeePassX - %1</source>
+        <translation>KeePassX - %1</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&lt;B&gt;Group: &lt;/B&gt;%1  &lt;B&gt;Title: &lt;/B&gt;%2  &lt;B&gt;Username: &lt;/B&gt;%3  &lt;B&gt;URL: &lt;/B&gt;&lt;a href=%4&gt;%4&lt;/a&gt;  &lt;B&gt;Password: &lt;/B&gt;%5  &lt;B&gt;Creation: &lt;/B&gt;%6  &lt;B&gt;Last Change: &lt;/B&gt;%7  &lt;B&gt;LastAccess: &lt;/B&gt;%8  &lt;B&gt;Expires: &lt;/B&gt;%9</source>
+        <translation>&lt;B&gt;Grup: &lt;/B&gt;%1  &lt;B&gt;Başlık: &lt;/B&gt;%2  &lt;B&gt;Kullanıcı Adı: &lt;/B&gt;%3  &lt;B&gt;Adres: &lt;/B&gt;&lt;a href=%4&gt;%4&lt;/a&gt;  &lt;B&gt;Parola: &lt;/B&gt;%5  &lt;B&gt;Oluşturulma: &lt;/B&gt;%6  &lt;B&gt;Son Değişiklik: &lt;/B&gt;%7  &lt;B&gt;Son Erişim: &lt;/B&gt;%8  &lt;B&gt;Süre Dolumu: &lt;/B&gt;%9</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Clone Entry</source>
+        <translation>Girdiyi İkile</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Delete Entry</source>
+        <translation>Girdiyi Sil</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Clone Entries</source>
+        <translation>Girdileri İkile</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Delete Entries</source>
+        <translation>Girdileri Sil</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>File could not be saved.
+%1</source>
+        <translation>Dosya kaydedilemedi.
+%1</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Save Database As...</source>
+        <translation>Veritabanını Farklı Kaydet...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ready</source>
+        <translation>Hazır</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>[new]</source>
+        <translation>[yeni]</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Open Database...</source>
+        <translation>Veritabanı Aç...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Loading Database...</source>
+        <translation>Veritabanı Yükleniyor...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Loading Failed</source>
+        <translation>Yükleme Başarısız Oldu</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Could not create key file. The following error occured:
+%1</source>
+        <translation>Anahtar dosyası oluşturulamadı. Şu hata oluştu:
+%1</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Export To...</source>
+        <translation>Dışarıya Aktar...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>KeePassX [new]</source>
+        <translation>KeePassX [yeni]</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unknown error in Import_PwManager::importFile()()</source>
+        <translation>Import_PwManager::importFile()() içerisinde bilinmeyen hata</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unknown error in Import_KWalletXml::importFile()</source>
+        <translation>Import_KWalletXml::importFile() içerisnde bilinmeyen hata</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unknown error in PwDatabase::openDatabase()</source>
+        <translation>PwDatabase::openDatabase() içerisinde bilinmeyen hata</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+V</source>
+        <translation>Ctrl+V</translation>
+    </message>
+</context>
+<context>
+    <name>MainWindow</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>KeePassX</source>
+        <translation>KeePassX</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Columns</source>
+        <translation>Sütunlar</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>PwManager File (*.pwm)</source>
+        <translation>PwManager Dosyası (*.pwm)</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>KWallet XML-File (*.xml)</source>
+        <translation>KWallet XML-Dosyası (*.xml)</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Add New Group...</source>
+        <translation>Yeni Grup Ekle...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Edit Group...</source>
+        <translation>Grubu Düzenle...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Delete Group</source>
+        <translation>Grubu Sil</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Copy Password to Clipboard</source>
+        <translation>Parolayı Panoya Kopyala</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Copy Username to Clipboard</source>
+        <translation>Kullanıcı Adını Panoya Kopyala</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Open URL</source>
+        <translation>Adresi Aç</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Save Attachment As...</source>
+        <translation>Eklentiyi Farklı Kaydet...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Add New Entry...</source>
+        <translation>Yeni Girdi Ekle...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>View/Edit Entry...</source>
+        <translation>Girdiyi Göster/Düzenle...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Delete Entry</source>
+        <translation>Girdiyi Sil</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Clone Entry</source>
+        <translation>Girdiyi İkile</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Search In Database...</source>
+        <translation>Veritabanında Ara...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Search in this group...</source>
+        <translation>Bu grup içerisinde ara...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Show Toolbar</source>
+        <translation>Araç Çubuğunu Göster</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Show Entry Details</source>
+        <translation>Girdi Detaylarını Göster</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Hide Usernames</source>
+        <translation>Kullanıcı Adlarını Gizle</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Hide Passwords</source>
+        <translation>Parolaları Gizle</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Title</source>
+        <translation>Başlık</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Username</source>
+        <translation>Kullanıcı Adı</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>URL</source>
+        <translation>Adres</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Password</source>
+        <translation>Parola</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Comment</source>
+        <translation>Açıklama</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Expires</source>
+        <translation>Süre Dolumu</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Creation</source>
+        <translation>Oluşturulma</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Last Change</source>
+        <translation>Son Değişiklik</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Last Access</source>
+        <translation>Son Erişim</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Attachment</source>
+        <translation>Eklenti</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Show Statusbar</source>
+        <translation>Durum Çubuğunu Göster</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Plain Text (*.txt)</source>
+        <translation>Düz Metin (*.txt)</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Hide</source>
+        <translation>Gizle</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Perform AutoType</source>
+        <translation>Otomatik Yaz</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Type Here</source>
+        <translation>Buraya Yaz</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Toolbar Icon Size</source>
+        <translation>Araç Çubuğu Simge Boyutu</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>16x16</source>
+        <translation>16x16</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>22x22</source>
+        <translation>22x22</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>28x28</source>
+        <translation>28x28</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;View</source>
+        <translation>&amp;Göster</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;File</source>
+        <translation>&amp;Dosya</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Import from...</source>
+        <translation>&amp;İçeri aktar...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Export to...</source>
+        <translation>D&amp;ışarı aktar...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Edit</source>
+        <translation>Dü&amp;zenle</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>E&amp;xtras</source>
+        <translation>Ek&amp;stralar</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Help</source>
+        <translation>&amp;Yardım</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;New Database...</source>
+        <translation>Y&amp;eni Veritabanı...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Open Database...</source>
+        <translation>&amp;Veritabanı Aç...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Close Database</source>
+        <translation>Veritabanını &amp;Kapat</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Save Database</source>
+        <translation>Veritabanını Kayde&amp;t</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Save Database &amp;As...</source>
+        <translation>Veritabanını &amp;Farklı Kaydet...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Database Settings...</source>
+        <translation>Veri&amp;tabanı Ayarları...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Change &amp;Master Key...</source>
+        <translation>Uygulama Anahtarını De&amp;ğiştir...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>E&amp;xit</source>
+        <translation>&amp;Çık</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Settings...</source>
+        <translation>&amp;Ayarlar...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;About...</source>
+        <translation>&amp;Hakkında...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;KeePassX Handbook...</source>
+        <translation>&amp;KeePassX El Kitabı...</translation>
+    </message>
+</context>
+<context>
+    <name>PasswordDlg</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>TextLabel</source>
+        <translation>MetinEtiketi</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>OK</source>
+        <translation>TAMAM</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>...</source>
+        <translation>...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Cancel</source>
+        <translation>İptal</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Enter a Password and/or choose a key file.</source>
+        <translation>Bir parola girin ya da bir anahtar dosyası seçin.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Key</source>
+        <translation>Anahtar</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Password:</source>
+        <translation>Parola:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Key file or directory:</source>
+        <translation>Anahtar dosyası ya da dizin:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Browse...</source>
+        <translation>&amp;Gözat...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+B</source>
+        <translation>Alt+G</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Use Password AND Key File</source>
+        <translation>Parola VE Anahtar Dosyası Kullan</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Exit</source>
+        <translation>Çık</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Password Repet.:</source>
+        <translation>Parola tekrarı:</translation>
+    </message>
+</context>
+<context>
+    <name>PwDatabase</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unknown Error</source>
+        <translation>Bilinmeyen Hata</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unexpected file size (DB_TOTAL_SIZE &lt; DB_HEADER_SIZE)</source>
+        <translation>Beklenmeyen dosya boyutu (DB_TOTAL_SIZE &lt; DB_HEADER_SIZE)</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Wrong Signature</source>
+        <translation>Yanlış İmza</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>AES-Init Failed</source>
+        <translation>AES Başlatılamadı</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unexpected error: Offset is out of range. [G1]</source>
+        <translation>Bilinmeyen hata: Offset is out of range. [G1]</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unexpected error: Offset is out of range. [G2]</source>
+        <translation>Bilinmeyen hata: Offset is out of range. [G2]</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unexpected error: Offset is out of range. [E1]</source>
+        <translation>Bilinmeyen hata: Offset is out of range. [E1]</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unexpected error: Offset is out of range. [E2]</source>
+        <translation>Bilinmeyen hata: Offset is out of range. [E2]</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unexpected error: Offset is out of range. [E3]</source>
+        <translation>Bilinmeyen hata: Offset is out of range. [E3]</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Hash test failed.
+The key is wrong or the file is damaged.</source>
+        <translation>Özet kontrolü başarısız.
+Anahtar yanlış ya da dosya bozulmuş.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Could not open key file.</source>
+        <translation>Anahtar dosyası açılamadı.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Key file could not be written.</source>
+        <translation>Anahtar dosyası yazılamadı.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Could not open file.</source>
+        <translation>Dosya açılamadı.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Could not open file for writing.</source>
+        <translation>Yazmak için açılamadı.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unsupported File Version.</source>
+        <translation>Desteklenmeyen Dosya Sürümü.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unknown Encryption Algorithm.</source>
+        <translation>Bilinmeyen Şifreleme Algoritması.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Decryption failed.
+The key is wrong or the file is damaged.</source>
+        <translation>Şifre çözme işlemi başarısız.
+Anahtar yanlış ya da dosya bozuk.</translation>
+    </message>
+</context>
+<context>
+    <name>QObject</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Warning</source>
+        <translation>Uyarı</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Could not save configuration file.
+Make sure you have write access to &apos;~/.keepass&apos;.</source>
+        <translation>yapılandırma dosyası kaydedilemedi.
+ &apos;~/.keepass&apos; dosyasına yazma izninizin olduğundan emin olun.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>OK</source>
+        <translation>TAMAM</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>File &apos;%1&apos; could not be found.</source>
+        <translation>&apos;%1&apos; dosyası bulunamadı.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>File not found.</source>
+        <translation>Dosya bulunamadı.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Could not open file.</source>
+        <translation>Dosya açılamadı.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>File is no valid PwManager file.</source>
+        <translation>Dosya geçerli bir PwManager dosyası değil.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unsupported file version.</source>
+        <translation>Desteklenmeyen dosya sürümü.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unsupported hash algorithm.</source>
+        <translation>Desteklenmeyen özet algoritması.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Unsupported encryption algorithm.</source>
+        <translation>Desteklenmeyen şifreleme algoritması.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Compressed files are not supported yet.</source>
+        <translation>Sıkıştırılmış dosyalar henüz desteklenmiyor.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Wrong password.</source>
+        <translation>Yanlış parola.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>File is damaged (hash test failed).</source>
+        <translation>Dosya bozulmuş (özet testi başarısız oldu).</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Invalid XML data (see stdout for details).</source>
+        <translation>Geçersiz XML verisi (detaylar için stdout çıktısına bakın).</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>File is empty.</source>
+        <translation>Dosya boş.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Invalid XML file (see stdout for details).</source>
+        <translation>Geçersiz XML dosyası (detaylar için stdout çıktısına bakın).</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Invalid XML file.</source>
+        <translation>Geçersiz XML dosyası.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Document does not contain data.</source>
+        <translation>Belge veri içermiyor.</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Error</source>
+        <translation>Hata</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Warning:</source>
+        <translation>Uyarı:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Invalid RGB color value.
+</source>
+        <translation>Geçersiz RGB renk değeri.
+</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Never</source>
+        <translation>Asla</translation>
+    </message>
+</context>
+<context>
+    <name>Search_Dlg</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+T</source>
+        <translation>Alt+B</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+U</source>
+        <translation>Alt+U</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>A&amp;nhang</source>
+        <translation>&amp;Ek</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+N</source>
+        <translation>Alt+E</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+W</source>
+        <translation>Alt+O</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+C</source>
+        <translation>Alt+Y</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Search...</source>
+        <translation>Ara...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Search For:</source>
+        <translation>Şunu Ara:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Regular E&amp;xpression</source>
+        <translation>Düzenli İ&amp;fade</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+X</source>
+        <translation>Alt+F</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Case Sensitive</source>
+        <translation>Bü&amp;yük Küçük Harflere Duyarlı</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Include:</source>
+        <translation>İçersin:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Titles</source>
+        <translation>&amp;Başlıklar</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Usernames</source>
+        <translation>K&amp;ullanıcı Adları</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>C&amp;omments</source>
+        <translation>A&amp;çıklamalar</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+O</source>
+        <translation>Alt+Ç</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>U&amp;RLs</source>
+        <translation>A&amp;dresler</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+R</source>
+        <translation>Alt+D</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Pass&amp;words</source>
+        <translation>Par&amp;olalar</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Search</source>
+        <translation>Ara</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Clo&amp;se</source>
+        <translation>Kapa&amp;t</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+S</source>
+        <translation>Alt+T</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Include Subgroups (recursive)</source>
+        <translation>Alt Grupları İçer (özyinelemeli)</translation>
+    </message>
+</context>
+<context>
+    <name>SelectIconDlg</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Icon Selection</source>
+        <translation>Simge Seçimi</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Add Custom Icon...</source>
+        <translation>Özel Simge Ekle...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Pick</source>
+        <translation>Al</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Cancel</source>
+        <translation>İptal</translation>
+    </message>
+</context>
+<context>
+    <name>SettingsDialog</name>
+    <message encoding="UTF-8">
+        <location filename="" line="137494912"/>
+        <source>Alt+Ö</source>
+        <translation>Alt+Ö</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>O&amp;K</source>
+        <translation>TAMA&amp;M</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+K</source>
+        <translation>Alt+M</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+C</source>
+        <translation>Alt+L</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Settings</source>
+        <translation>Ayarlar</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Cancel</source>
+        <translation>İpta&amp;l</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Clear clipboard after:</source>
+        <translation>Bu süre geçince panoyu temizle:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Seconds</source>
+        <translation>Saniye</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Sh&amp;ow passwords in plain text by default</source>
+        <translation>Öntanımlı olarak parolaları bas&amp;it metin içerisinde göster</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+O</source>
+        <translation>Alt+İ</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Appea&amp;rance</source>
+        <translation>Görün&amp;üm</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Banner Color</source>
+        <translation>Üst Alan Rengi</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Text Color:</source>
+        <translation>Metin Rengi:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Change...</source>
+        <translation>Değiştir...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Color 2:</source>
+        <translation>Renk 2:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>C&amp;hange...</source>
+        <translation>Deği&amp;ştir...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+H</source>
+        <translation>Alt+Ş</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Color 1:</source>
+        <translation>Renk 1:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Expand group tree when opening a database</source>
+        <translation>Bir veritabanını açarken grup ağacını genişlet</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Other</source>
+        <translation>&amp;Diğer</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Browser Command:</source>
+        <translation>Tarayıcı Komutu:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Securi&amp;ty</source>
+        <translation>Güve&amp;nlik</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alternating Row Colors</source>
+        <translation>Satır Renkleri Farklılaştırılsın</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Browse...</source>
+        <translation>Gözat...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Remember last key type and location</source>
+        <translation>Son anahtar tipini ve konumunu hatırla</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Mounting Root:</source>
+        <translation>Bağlanacak kök dizin:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Remember last opened file</source>
+        <translation>Son açılan dosyayı hatırla</translation>
+    </message>
+</context>
+<context>
+    <name>SimplePasswordDialog</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>O&amp;K</source>
+        <translation>TAMA&amp;M</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+K</source>
+        <translation>Alt+M</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Alt+C</source>
+        <translation>Alt+L</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>...</source>
+        <translation>...</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Enter your Password</source>
+        <translation>Parolanızı Girin</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Password:</source>
+        <translation>Parola:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Cancel</source>
+        <translation>İpta&amp;l</translation>
+    </message>
+</context>
+<context>
+    <name>dbsettingdlg_base</name>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Database Settings</source>
+        <translation>Veritabanı Ayarları</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Encryption</source>
+        <translation>Şifreleme</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Algorithm:</source>
+        <translation>Algoritma:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>?</source>
+        <translation>?</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Encryption Rounds:</source>
+        <translation>Şifreleme Basamağı:</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>O&amp;K</source>
+        <translation>TAMA&amp;M</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+K</source>
+        <translation>Ctrl+M</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>&amp;Cancel</source>
+        <translation>İpta&amp;l</translation>
+    </message>
+    <message>
+        <location filename="" line="137494912"/>
+        <source>Ctrl+C</source>
+        <translation>Ctrl+L</translation>
+    </message>
+</context>
+</TS>
