--- ../KDE/cdcat.desktop	2005-08-22 20:03:00.000000000 +0300
+++ ../KDE/cdcat.desktop	2007-07-28 18:20:53.000000000 +0300
@@ -1,12 +1,16 @@
 # KDE Config File  
 [Desktop Entry]
 Type=Application
+Version=1.0
+Encoding=UTF-8
 Exec=cdcat
-Icon=/usr/local/share/cdcat/cdcat.png
+Icon=cdcat
 GenericName[pt]=Catalogador
-GenericName[hu]=Katalogizáló
-Comment[pt]=Um programa de catalogação
-Comment[hu]=Media katalogizáló program
+GenericName[hu]=Katalogizï¿½lï¿½
+GenericName[tr]=Media Katalog uygulamasÄ±
+Comment[pt]=Um programa de catalogaï¿½ï¿½o
+Comment[hu]=Media katalogizï¿½lï¿½ program
+Comment[tr]=CD Katalog UygulamasÄ±
 Terminal=false
 Name=Hyper's CdCatalog
 MimeType=application/x-k3b
