--- comix.desktop	2006-04-09 00:49:28.000000000 +0300
+++ comix-yeni.desktop	2006-06-28 16:35:57.000000000 +0300
@@ -2,12 +2,15 @@
 Encoding=UTF-8
 Name=Comix
 Name[it]=Comix
+Name[tr]=Comix
 GenericName=Comic book viewer
 GenericName[sv]=Serieboksläsare
+GenericName[tr]=Çizgi roman göstericisi
 Comment=A viewer for comic book archives
 Comment[it]=Un visualizzatore di fumetti 
 Comment[es]=Un visor de comics
 Comment[sv]=En serieboksläsare
+Comment[tr]=Çizgi roman arşivleri için gösterici
 Exec=comix %f
 Icon=comix.png
 Terminal=false
