diff -Nuar rpy-1.0.3.orig/setup.py rpy-1.0.3/setup.py
--- rpy-1.0.3.orig/setup.py	2008-04-17 18:01:14.000000000 +0300
+++ rpy-1.0.3/setup.py	2008-08-03 02:41:35.000000000 +0300
@@ -160,7 +160,7 @@
     else: # unix-like systems, this is known to work for Linux and Solaris
         include_dirs = [ os.path.join(RHOME.strip(), 'include'), 
                          'src', '/usr/share/R/include' ]
-        libraries=['R','Rlapack']
+        libraries=['R','lapack']
         library_dirs = r_libs
         runtime_libs = r_libs
         extra_compile_args=["-shared"]
