diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur juffed_0.4-orig/juffed.desktop juffed_0.4/juffed.desktop
--- juffed_0.4-orig/juffed.desktop	2009-01-14 17:26:27.000000000 +0200
+++ juffed_0.4/juffed.desktop	2009-01-14 17:29:18.000000000 +0200
@@ -1,17 +1,13 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=Text editor JuffEd
-Name[ru]=Текстовый редактор JuffEd
-Name[pl]=Edytor tekstu JuffEd
-Name[de]=Texteditor JuffEd
-Name[fr]=Éditeur de texte JuffEd
-Name[pt]=Editor de texto JuffEd
+Name=JuffEd
 Comment=Simple text editor
 Comment[ru]=Простой текстовый редактор
 Comment[pl]=Prosty edytor tekstu
 Comment[de]=Ein einfacher Texteditor
 Comment[fr]=Un simple éditeur de texte
 Comment[pt]=Editor de texto simples
+Comment[tr]=Basit metin editörü
 Exec=juffed
 Terminal=false
 Type=Application
@@ -25,3 +21,4 @@
 GenericName[de]=Ein einfacher Texteditor
 GenericName[fr]=Un simple éditeur de texte
 GenericName[pt]=Editor de texto simples
+GenericName[tr]=Basit metin editörü
