--- project2.pro	2007-06-03 21:48:51.000000000 +0300
+++ project.pro	2007-06-03 23:03:48.000000000 +0300
@@ -158,7 +158,7 @@
 
 ############################## required by "make install" ########################################
 isEmpty(PREFIX) {
-        PREFIX = "/usr/local"
+        PREFIX = "/usr"
 }
 isEmpty(I18NDIR) {
 	I18NDIR = "$${PREFIX}/share/qtpfsgui/i18n"
@@ -172,7 +172,7 @@
 icon.path        = $${PREFIX}/share/icons/hicolor/48x48/apps
 htmls.files      = html
 htmls.path       = $${PREFIX}/share/qtpfsgui
-i18n.files       = i18n/lang_it.qm
+i18n.files       = i18n/lang_tr.qm
 i18n.path        = $$I18NDIR
 
 INSTALLS        += target menu icon htmls i18n
@@ -333,7 +333,7 @@
 
 RESOURCES = icons.qrc
 
-TRANSLATIONS = i18n/lang_it.ts \
+TRANSLATIONS = i18n/lang_tr.ts \
                i18n/lang_no.ts
 
 # Old durand, we use the fftw version now.
