--- i18n/lang_tr.ts	2007-07-14 12:19:48.000000000 +0300
+++ i18n/lang_tr2.ts	2007-08-01 19:45:56.000000000 +0300
@@ -1,166 +1,167 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!DOCTYPE TS><TS version="1.1" language="no">
+<!DOCTYPE TS><TS version="1.1" language="nb">
+<defaultcodec></defaultcodec>
 <context>
     <name>BatchTMDialog</name>
     <message>
         <location filename="../forms/batch_dialog.ui" line="13"/>
         <source>Batch Tone Mapping</source>
-        <translation type="unfinished"></translation>
+        <translation>Ton Haritalama Yığını</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="36"/>
         <source>Input Hdr list</source>
-        <translation type="unfinished"></translation>
+        <translation>Hdr listesini içe aktar</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="64"/>
         <source>Add all the HDRs in a directory to the list.</source>
-        <translation type="unfinished"></translation>
+        <translation>Dizindeki tüm HDRleri listeye ekle.</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="67"/>
         <source>Add &amp;Directory</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Dizin Ekle</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="88"/>
         <source>Add single HDR files to the list.</source>
-        <translation type="unfinished"></translation>
+        <translation>Bir tek HDR dosyasını listeye ekle.</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="91"/>
         <source>&amp;Add Files</source>
-        <translation type="unfinished"></translation>
+        <translation>Dosy&amp;a Ekle</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="112"/>
         <source>Remove single HDR files to the list.</source>
-        <translation type="unfinished"></translation>
+        <translation>Bir tek HDR dosyasını listeden çıkart.</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="115"/>
         <source>&amp;Remove Files</source>
-        <translation type="unfinished"></translation>
+        <translation>Dosyala&amp;rı çıkart</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="143"/>
         <source>List of HDRs that will be tone mapped</source>
-        <translation type="unfinished"></translation>
+        <translation>Ton haritası oluşturulacak HDRlerin listesi</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="159"/>
         <source>Input Tone Mapping Setting list</source>
-        <translation type="unfinished"></translation>
+        <translation>Ton Haritası Ayarlar listesini içe aktar </translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="187"/>
         <source>Add all the Tone Mapping Setting files in a directory to the list.</source>
-        <translation type="unfinished"></translation>
+        <translation>Bir dizindeki tüm Ton Haritalama Ayarlar dosyalarını listeye ekle.</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="190"/>
         <source>Add D&amp;irectory</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Dizin Ekle</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="211"/>
         <source>Add single Tone Mapping Setting files to the list.</source>
-        <translation type="unfinished"></translation>
+        <translation>Tek Ton Haritalama Ayarlar dosyalarını listeye ekle.</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="214"/>
         <source>Add Fi&amp;les</source>
-        <translation type="unfinished"></translation>
+        <translation>Dosya Ek&amp;le</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="235"/>
         <source>Remove single Tone Mapping Setting files to the list.</source>
-        <translation type="unfinished"></translation>
+        <translation>Tek Ton Haritalama Ayarlar dosyalarını listeden çıkart.</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="238"/>
         <source>R&amp;emove Files</source>
-        <translation type="unfinished"></translation>
+        <translation>Dosyala&amp;rı çıkart</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="266"/>
         <source>List of Tone Mapping Setting files that will be used to tone map each HDR.</source>
-        <translation type="unfinished"></translation>
+        <translation>Her Ton Haritası kullanılan HDR dosyası için Ton Haritalama Ayarlar dosyaları listesi.</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="284"/>
         <source>Specify output folder for the tone mapped files.</source>
-        <translation type="unfinished"></translation>
+        <translation>Ton haritalanmış dosyalar için belirlenmiş çıktı klasörü.</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="287"/>
         <source>Output</source>
-        <translation type="unfinished"></translation>
+        <translation>Çıktı</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="299"/>
         <source>Select &amp;Output Folder...</source>
-        <translation type="unfinished"></translation>
+        <translation>Çıktı Klasörünü S&amp;eç...</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="316"/>
         <source>Log:</source>
-        <translation type="unfinished"></translation>
+        <translation>Günlük:</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="342"/>
         <source>Batch operations report.</source>
-        <translation type="unfinished"></translation>
+        <translation>Yığın operasyonu raporu.</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="369"/>
         <source>&amp;Show only:</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Sadece göster:</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="379"/>
         <source>Filter messages based on severity.</source>
-        <translation type="unfinished"></translation>
+        <translation>Önemli süzgeç mesajları.</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="383"/>
         <source>All messages</source>
-        <translation type="unfinished"></translation>
+        <translation>Bütün mesajlar</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="388"/>
         <source>Errors only</source>
-        <translation type="unfinished"></translation>
+        <translation>Sadece Hatalar</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="403"/>
         <source>&amp;Filter Log messages:</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Süzgeç günlük mesajları:</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="419"/>
         <source>Clear filter text</source>
-        <translation type="unfinished"></translation>
+        <translation>Süzgeç metnini temizle</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="434"/>
         <source>Overall completion progress.</source>
-        <translation type="unfinished"></translation>
+        <translation>Tamamlama işlemleri bütünü.</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="465"/>
         <source>&amp;Cancel</source>
-        <translation type="unfinished"></translation>
+        <translation>İ&amp;ptal</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="478"/>
         <source>Start batch tone mapping.</source>
-        <translation type="unfinished"></translation>
+        <translation>Yığın ton haritalamaya başla.</translation>
     </message>
     <message>
         <location filename="../forms/batch_dialog.ui" line="481"/>
         <source>&amp;Start!</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Başla!</translation>
     </message>
     <message>
         <location filename="../src/Batch/batch_dialog_impl.cpp" line="228"/>
@@ -180,7 +181,7 @@
     <message>
         <location filename="../src/Batch/batch_dialog_impl.cpp" line="116"/>
         <source>Load the tonemapping settings text files...</source>
-        <translation type="unfinished"></translation>
+        <translation>Ton haritalama ayarları metin dosyalarını yükle...</translation>
     </message>
     <message>
         <location filename="../src/Batch/batch_dialog_impl.cpp" line="116"/>
@@ -190,17 +191,17 @@
     <message>
         <location filename="../src/Batch/batch_dialog_impl.cpp" line="328"/>
         <source>Processing...</source>
-        <translation type="unfinished"></translation>
+        <translation>İşleniyor...</translation>
     </message>
     <message>
         <location filename="../src/Batch/batch_dialog_impl.cpp" line="345"/>
         <source>&amp;Done</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Bitti</translation>
     </message>
     <message>
         <location filename="../src/Batch/batch_dialog_impl.cpp" line="387"/>
         <source>Done</source>
-        <translation type="unfinished"></translation>
+        <translation>Bitti</translation>
     </message>
 </context>
 <context>
@@ -357,7 +358,7 @@
     <message>
         <location filename="../forms/hdrwizardform.ui" line="310"/>
         <source>Weighting function:</source>
-        <translation type="unfinished"></translation>
+        <translation>Derinlik işlevleri:</translation>
     </message>
     <message>
         <location filename="../src/HdrWizard/hdrwizardform_impl.cpp" line="371"/>
@@ -426,7 +427,7 @@
     <message>
         <location filename="../forms/hdrwizardform.ui" line="646"/>
         <source>Weights:</source>
-        <translation type="unfinished"></translation>
+        <translation>Derinlik:</translation>
     </message>
     <message>
         <location filename="../src/HdrWizard/hdrwizardform_impl.cpp" line="375"/>
@@ -590,7 +591,7 @@
     <message>
         <location filename="../src/HdrWizard/hdrwizardform_impl.cpp" line="364"/>
         <source>(*) Weights: </source>
-        <translation type="unfinished"></translation>
+        <translation>(*) Derinlik: </translation>
     </message>
     <message>
         <location filename="../src/HdrWizard/hdrwizardform_impl.cpp" line="364"/>
@@ -857,12 +858,12 @@
     <message>
         <location filename="../forms/maingui.ui" line="226"/>
         <source>&amp;Tonemap the Hdr...</source>
-        <translation type="unfinished"></translation>
+        <translation>HDR &amp;ton haritası...</translation>
     </message>
     <message>
         <location filename="../forms/maingui.ui" line="229"/>
         <source>Tonemap the Hdr</source>
-        <translation type="unfinished"></translation>
+        <translation>HDR ton haritası</translation>
     </message>
     <message>
         <location filename="../forms/maingui.ui" line="241"/>
@@ -1227,7 +1228,7 @@
     <message>
         <location filename="../forms/maingui.ui" line="79"/>
         <source>&amp;Toolbars</source>
-        <translation type="unfinished"></translation>
+        <translation>A&amp;raç Çubukları</translation>
     </message>
     <message>
         <location filename="../forms/maingui.ui" line="157"/>
@@ -1247,7 +1248,7 @@
     <message>
         <location filename="../forms/maingui.ui" line="238"/>
         <source>Tonemap the Hdr into a common LDR (i.e. JPEG or PNG) format</source>
-        <translation type="unfinished"></translation>
+        <translation>Yaygın LDR (Örneğin JPEG yada PNG) biçimleri içinden HDR ton haritası</translation>
     </message>
     <message>
         <location filename="../forms/maingui.ui" line="284"/>
@@ -1272,7 +1273,7 @@
     <message>
         <location filename="../forms/maingui.ui" line="496"/>
         <source>&amp;Preferences...</source>
-        <translation type="unfinished"></translation>
+        <translation>Te&amp;rcihler...</translation>
     </message>
     <message>
         <location filename="../forms/maingui.ui" line="522"/>
@@ -1292,32 +1293,32 @@
     <message>
         <location filename="../forms/maingui.ui" line="603"/>
         <source>Toolbars</source>
-        <translation type="unfinished"></translation>
+        <translation>Araç Çubukları</translation>
     </message>
     <message>
         <location filename="../forms/maingui.ui" line="611"/>
         <source>Text &amp;Under Icons</source>
-        <translation type="unfinished"></translation>
+        <translation>Sim&amp;ge Altında Metin</translation>
     </message>
     <message>
         <location filename="../forms/maingui.ui" line="619"/>
         <source>&amp;Icons Only</source>
-        <translation type="unfinished"></translation>
+        <translation>Sadece S&amp;imgeler</translation>
     </message>
     <message>
         <location filename="../forms/maingui.ui" line="627"/>
         <source>&amp;Text Only</source>
-        <translation type="unfinished"></translation>
+        <translation>Sa&amp;dece Metin</translation>
     </message>
     <message>
         <location filename="../forms/maingui.ui" line="635"/>
         <source>Text &amp;Alongside Icons</source>
-        <translation type="unfinished"></translation>
+        <translation>Simgelerin &amp;Yanına Metin</translation>
     </message>
     <message>
         <location filename="../forms/maingui.ui" line="640"/>
         <source>&amp;Batch Tone Mapping...</source>
-        <translation type="unfinished"></translation>
+        <translation>Ton Haritalama &amp;Yığını...</translation>
     </message>
     <message>
         <location filename="../src/MainWindow/maingui_impl.cpp" line="105"/>
@@ -1352,12 +1353,12 @@
     <message>
         <location filename="../forms/maingui.ui" line="557"/>
         <source>Copy Exif Data between two set of files</source>
-        <translation type="unfinished"></translation>
+        <translation>İki dosya arasında Exif verisini kopyala</translation>
     </message>
     <message>
         <location filename="../forms/maingui.ui" line="643"/>
         <source>Ctrl+B</source>
-        <translation type="unfinished"></translation>
+        <translation>Ctrl+Y</translation>
     </message>
 </context>
 <context>
@@ -1612,12 +1613,12 @@
     <message>
         <location filename="../forms/options.ui" line="218"/>
         <source>Batch</source>
-        <translation type="unfinished"></translation>
+        <translation>Yığın</translation>
     </message>
     <message>
         <location filename="../forms/options.ui" line="238"/>
         <source>Save using this LDR image &amp;format:</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;LDR görüntü biçimini kullanarak kaydet:</translation>
     </message>
     <message>
         <location filename="../forms/options.ui" line="249"/>
@@ -1648,7 +1649,8 @@
         <location filename="../forms/options.ui" line="287"/>
         <source>Number of &amp;threads:
 (useful on multi-core/SMP machines)</source>
-        <translation type="unfinished"></translation>
+        <translation>İş &amp;Parçacıkları Numarası:
+(Çoklu işlemcili makineler için kullanışlıdır)</translation>
     </message>
     <message>
         <location filename="../forms/options.ui" line="623"/>
@@ -1690,7 +1692,10 @@
 p, li { white-space: pre-wrap; }
 &lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:&apos;Sans Serif&apos;; font-size:9pt; font-weight:400; font-style:normal; text-decoration:none;&quot;&gt;
 &lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:12pt; font-weight:600;&quot;&gt;Hdr Visualization Options&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
-        <translation type="unfinished"></translation>
+        <translation>&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
+p, li { white-space: pre-wrap; }
+&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:&apos;Sans Serif&apos;; font-size:9pt; font-weight:400; font-style:normal; text-decoration:none;&quot;&gt;
+&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:12pt; font-weight:600;&quot;&gt;Hdr Canlandırma Seçenekleri&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
     </message>
 </context>
 <context>
@@ -1703,7 +1708,7 @@
     <message>
         <location filename="../src/main.cpp" line="47"/>
         <source>Cannot find Qt&apos;s JPEG Plugin...&lt;br&gt;Please unzip the DLL package with the option &quot;use folder names&quot; activated.</source>
-        <translation type="unfinished"></translation>
+        <translation>Qt JPEG Eklentisi bulunamadı..&lt;br&gt;Lütfen DLL paketini &quot;klasör isimleri kullanarak&quot; seçeneği aktif hale getirin ve çıkartın.</translation>
     </message>
 </context>
 <context>
@@ -2289,12 +2294,12 @@
     <message>
         <location filename="../forms/tonemappingdialog.ui" line="122"/>
         <source>Adjust &amp;Levels</source>
-        <translation type="unfinished"></translation>
+        <translation>Seviye&amp;leri Uydur</translation>
     </message>
     <message>
         <location filename="../forms/tonemappingdialog.ui" line="125"/>
         <source>Adjust Levels</source>
-        <translation type="unfinished"></translation>
+        <translation>Seviyeleri Uydur</translation>
     </message>
 </context>
 <context>
@@ -2382,7 +2387,7 @@
     <message>
         <location filename="../src/TransplantExif/transplant_impl.cpp" line="293"/>
         <source>Done.</source>
-        <translation type="unfinished"></translation>
+        <translation>Bitti.</translation>
     </message>
 </context>
 </TS>
