diff -Naur mypaint-0.7.1.orig/desktop/mypaint.desktop mypaint-0.7.1/desktop/mypaint.desktop
--- mypaint-0.7.1.orig/desktop/mypaint.desktop	2009-07-22 14:16:25.000000000 +0300
+++ mypaint-0.7.1/desktop/mypaint.desktop	2009-08-01 04:36:02.945298786 +0300
@@ -2,11 +2,13 @@
 Version=1.0
 Encoding=UTF-8
 Name=MyPaint
-Exec=mypaint %U
+Exec=mypaint
 Comment=Paint images from scratch
+Comment[tr]=Görüntüleri Sıfırdan Boyama
 GenericName=Image Editor
+GenericName[tr]=Resim Düzenleyici
 MimeType=image/openraster;image/png;image/jpeg;
 Type=Application
-Icon=mypaint_48
+Icon=mypaint_48
 Categories=Graphics;
 Terminal=false
