diff -Nuar qtiplot-0.8.8.orig/qtiplot.pro qtiplot-0.8.8/qtiplot.pro
--- qtiplot-0.8.8.orig/qtiplot.pro	2006-09-07 10:45:02.000000000 +0300
+++ qtiplot-0.8.8/qtiplot.pro	2006-11-25 17:15:45.000000000 +0200
@@ -4,3 +4,4 @@
 	  fitPlugins/fitRational0\
 	  fitPlugins/fitRational1\
 	  qtiplot
++unix:system(sip -I /usr/include/python2.4 -t Qt_3_3_7 -t WS_X11 -c $${MOC_DIR} src/qti.sip)
\ Dosya sonunda yenisatır yok.
