diff -Nuar scrollkeeper-0.3.14.orig/configure scrollkeeper-0.3.14/configure
--- scrollkeeper-0.3.14.orig/configure	2007-02-15 23:24:56.000000000 +0200
+++ scrollkeeper-0.3.14/configure	2007-02-15 23:26:24.000000000 +0200
@@ -8360,7 +8360,7 @@
 	echo "$as_me:$LINENO: checking for DocBook XML DTD" >&5
 echo $ECHO_N "checking for DocBook XML DTD... $ECHO_C" >&6
 	db_found=
-	for vers in 4.2 4.1.2; do
+	for vers in 4.2 4.1.2 4.4; do
 	   if xmlcatalog $XML_CATALOG "-//OASIS//DTD DocBook XML V$vers//EN" >/dev/null ; then
 	      db_found=$vers
 	      break
