--- conver_submenus/Conversions_submenus.desktop	2006-12-26 20:30:05.000000000 +0200
+++ conver_submenus/Conversions_submenus_yeni.desktop	2007-01-30 14:03:58.000000000 +0200
@@ -21,48 +21,48 @@
 X-KDE-Submenu=Conversion Tools
 
 [Desktop Action Temperature]
-Exec=~/.kde/share/apps/konqueror/servicemenus/conver_submenus.sh Temperature
+Exec=/usr/kde/3.5/share/apps/konqueror/servicemenus/conver_submenus.sh Temperature
 Icon=package_utilities
 Name=Temperature
 
 
 [Desktop Action Currency]
-Exec=~/.kde/share/apps/konqueror/servicemenus/conver_submenus.sh Currency
+Exec=/usr/kde/3.5/share/apps/konqueror/servicemenus/conver_submenus.sh Currency
 Icon=package_utilities
 Name=Currency
 
 
 [Desktop Action Weight]
-Exec=~/.kde/share/apps/konqueror/servicemenus/conver_submenus.sh Weight
+Exec=/usr/kde/3.5/share/apps/konqueror/servicemenus/conver_submenus.sh Weight
 Icon=package_utilities
 Name=Weight
 
 
 [Desktop Action Distance]
-Exec=~/.kde/share/apps/konqueror/servicemenus/conver_submenus.sh Distance
+Exec=/usr/kde/3.5/share/apps/konqueror/servicemenus/conver_submenus.sh Distance
 Icon=package_utilities
 Name=Distance
 
 
 [Desktop Action Velocity]
-Exec=~/.kde/share/apps/konqueror/servicemenus/conver_submenus.sh Velocity
+Exec=/usr/kde/3.5/share/apps/konqueror/servicemenus/conver_submenus.sh Velocity
 Icon=package_utilities
 Name=Velocity
 
 
 [Desktop Action Degrees]
-Exec=~/.kde/share/apps/konqueror/servicemenus/conver_submenus.sh Degrees
+Exec=/usr/kde/3.5/share/apps/konqueror/servicemenus/conver_submenus.sh Degrees
 Icon=package_utilities
 Name=Degrees
 
 
 [Desktop Action Volume]
-Exec=~/.kde/share/apps/konqueror/servicemenus/conver_submenus.sh Volume
+Exec=/usr/kde/3.5/share/apps/konqueror/servicemenus/conver_submenus.sh Volume
 Icon=package_utilities
 Name=Volume
 
 
 [Desktop Action Area]
-Exec=~/.kde/share/apps/konqueror/servicemenus/conver_submenus.sh Area
+Exec=/usr/kde/3.5/share/apps/konqueror/servicemenus/conver_submenus.sh Area
 Icon=package_utilities
 Name=Area
\ Dosya sonunda yenisatır yok.
