diff -Naur keepassx-0.3.3.orig/share/applications/keepassx.desktop keepassx-0.3.3/share/applications/keepassx.desktop
--- keepassx-0.3.3.orig/share/applications/keepassx.desktop	2008-03-04 20:05:57.000000000 +0200
+++ keepassx-0.3.3/share/applications/keepassx.desktop	2008-10-04 20:31:50.000000000 +0300
@@ -1,6 +1,8 @@
 [Desktop Entry]
 Name=KeePassX
+Name[tr]=KeePassX
 GenericName=Cross Platform Password Manager
+GenericName[tr]=Parola Yöneticisi
 GenericName[fr]=Gestionnaire de mot de passe
 GenericName[de]=Passwortverwaltung
 Exec=keepassx
@@ -10,4 +12,6 @@
 Comment[de]=Passwortverwaltung
 Terminal=false
 Type=Application
+MimeType=application/x-keepass;
+Patterns=*.kdb
 Categories=Qt;Utility;
