diff -Naur enblend-3.0.orig/configure enblend-3.0/configure
--- enblend-3.0.orig/configure	2007-01-27 07:01:55.000000000 +0200
+++ enblend-3.0/configure	2007-02-18 12:44:57.000000000 +0200
@@ -5721,8 +5721,8 @@
 
 
 if test "${ac_cv_header_lcms_h+set}" = set; then
-  echo "$as_me:$LINENO: checking for lcms.h" >&5
-echo $ECHO_N "checking for lcms.h... $ECHO_C" >&6
+  echo "$as_me:$LINENO: checking for lcms/lcms.h" >&5
+echo $ECHO_N "checking for lcms/lcms.h... $ECHO_C" >&6
 if test "${ac_cv_header_lcms_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 fi
@@ -5730,8 +5730,8 @@
 echo "${ECHO_T}$ac_cv_header_lcms_h" >&6
 else
   # Is the header compilable?
-echo "$as_me:$LINENO: checking lcms.h usability" >&5
-echo $ECHO_N "checking lcms.h usability... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking lcms/lcms.h usability" >&5
+echo $ECHO_N "checking lcms/lcms.h usability... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -5739,7 +5739,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <lcms.h>
+#include <lcms/lcms.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
@@ -5775,15 +5775,15 @@
 echo "${ECHO_T}$ac_header_compiler" >&6
 
 # Is the header present?
-echo "$as_me:$LINENO: checking lcms.h presence" >&5
-echo $ECHO_N "checking lcms.h presence... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking lcms/lcms.h presence" >&5
+echo $ECHO_N "checking lcms/lcms.h presence... $ECHO_C" >&6
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <lcms.h>
+#include <lcms/lcms.h>
 _ACEOF
 if { (eval echo "$as_me:$LINENO: \"$ac_cpp conftest.$ac_ext\"") >&5
   (eval $ac_cpp conftest.$ac_ext) 2>conftest.er1
@@ -5817,25 +5817,25 @@
 # So?  What about this header?
 case $ac_header_compiler:$ac_header_preproc:$ac_cxx_preproc_warn_flag in
   yes:no: )
-    { echo "$as_me:$LINENO: WARNING: lcms.h: accepted by the compiler, rejected by the preprocessor!" >&5
-echo "$as_me: WARNING: lcms.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
-    { echo "$as_me:$LINENO: WARNING: lcms.h: proceeding with the compiler's result" >&5
-echo "$as_me: WARNING: lcms.h: proceeding with the compiler's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: lcms/lcms.h: accepted by the compiler, rejected by the preprocessor!" >&5
+echo "$as_me: WARNING: lcms/lcms.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
+    { echo "$as_me:$LINENO: WARNING: lcms/lcms.h: proceeding with the compiler's result" >&5
+echo "$as_me: WARNING: lcms/lcms.h: proceeding with the compiler's result" >&2;}
     ac_header_preproc=yes
     ;;
   no:yes:* )
-    { echo "$as_me:$LINENO: WARNING: lcms.h: present but cannot be compiled" >&5
-echo "$as_me: WARNING: lcms.h: present but cannot be compiled" >&2;}
-    { echo "$as_me:$LINENO: WARNING: lcms.h:     check for missing prerequisite headers?" >&5
-echo "$as_me: WARNING: lcms.h:     check for missing prerequisite headers?" >&2;}
-    { echo "$as_me:$LINENO: WARNING: lcms.h: see the Autoconf documentation" >&5
-echo "$as_me: WARNING: lcms.h: see the Autoconf documentation" >&2;}
-    { echo "$as_me:$LINENO: WARNING: lcms.h:     section \"Present But Cannot Be Compiled\"" >&5
-echo "$as_me: WARNING: lcms.h:     section \"Present But Cannot Be Compiled\"" >&2;}
-    { echo "$as_me:$LINENO: WARNING: lcms.h: proceeding with the preprocessor's result" >&5
-echo "$as_me: WARNING: lcms.h: proceeding with the preprocessor's result" >&2;}
-    { echo "$as_me:$LINENO: WARNING: lcms.h: in the future, the compiler will take precedence" >&5
-echo "$as_me: WARNING: lcms.h: in the future, the compiler will take precedence" >&2;}
+    { echo "$as_me:$LINENO: WARNING: lcms/lcms.h: present but cannot be compiled" >&5
+echo "$as_me: WARNING: lcms/lcms.h: present but cannot be compiled" >&2;}
+    { echo "$as_me:$LINENO: WARNING: lcms/lcms.h:     check for missing prerequisite headers?" >&5
+echo "$as_me: WARNING: lcms/lcms.h:     check for missing prerequisite headers?" >&2;}
+    { echo "$as_me:$LINENO: WARNING: lcms/lcms.h: see the Autoconf documentation" >&5
+echo "$as_me: WARNING: lcms/lcms.h: see the Autoconf documentation" >&2;}
+    { echo "$as_me:$LINENO: WARNING: lcms/lcms.h:     section \"Present But Cannot Be Compiled\"" >&5
+echo "$as_me: WARNING: lcms/lcms.h:     section \"Present But Cannot Be Compiled\"" >&2;}
+    { echo "$as_me:$LINENO: WARNING: lcms/lcms.h: proceeding with the preprocessor's result" >&5
+echo "$as_me: WARNING: lcms/lcms.h: proceeding with the preprocessor's result" >&2;}
+    { echo "$as_me:$LINENO: WARNING: lcms/lcms.h: in the future, the compiler will take precedence" >&5
+echo "$as_me: WARNING: lcms/lcms.h: in the future, the compiler will take precedence" >&2;}
     (
       cat <<\_ASBOX
 ## -------------------------------------------- ##
@@ -5846,8 +5846,8 @@
       sed "s/^/$as_me: WARNING:     /" >&2
     ;;
 esac
-echo "$as_me:$LINENO: checking for lcms.h" >&5
-echo $ECHO_N "checking for lcms.h... $ECHO_C" >&6
+echo "$as_me:$LINENO: checking for lcms/lcms.h" >&5
+echo $ECHO_N "checking for lcms/lcms.h... $ECHO_C" >&6
 if test "${ac_cv_header_lcms_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
@@ -5860,8 +5860,8 @@
 if test $ac_cv_header_lcms_h = yes; then
   :
 else
-  { echo "$as_me:$LINENO: WARNING: lcms header files are required to compile enblend." >&5
-echo "$as_me: WARNING: lcms header files are required to compile enblend." >&2;}
+  { echo "$as_me:$LINENO: WARNING: lcms/lcms.header files are required to compile enblend." >&5
+echo "$as_me: WARNING: lcms/lcms.header files are required to compile enblend." >&2;}
 fi
 
 
diff -Naur enblend-3.0.orig/src/enblend.cc enblend-3.0/src/enblend.cc
--- enblend-3.0.orig/src/enblend.cc	2007-01-27 22:05:39.000000000 +0200
+++ enblend-3.0/src/enblend.cc	2007-02-18 12:40:38.000000000 +0200
@@ -75,7 +75,7 @@
 #endif
 
 #include <boost/random/mersenne_twister.hpp>
-#include <lcms.h>
+#include <lcms/lcms.h>
 
 // Globals
 
