diff -Naur arora-0.4.orig/src/data/defaultbookmarks.xbel arora-0.4/src/data/defaultbookmarks.xbel
--- arora-0.4.orig/src/data/defaultbookmarks.xbel	2008-10-01 04:17:22.000000000 +0300
+++ arora-0.4/src/data/defaultbookmarks.xbel	2008-10-09 19:21:57.000000000 +0300
@@ -2,36 +2,30 @@
 <!DOCTYPE xbel>
 <xbel version="1.0">
     <folder folded="yes">
         <title>Bookmarks Bar</title>
-        <bookmark href="http://arora-browser.org/">
-            <title>Arora</title>
-        </bookmark>
-        <bookmark href="http://webkit.org/">
-            <title>WebKit.org</title>
+        <bookmark href="http://www.pardus.org.tr/">
+            <title>Pardus.Org</title>
         </bookmark>
-        <bookmark href="http://doc.trolltech.com/">
-            <title>Qt Documentation</title>
+        <bookmark href="http://www.ozgurlukicin.com/">
+            <title>Özgürlük İçin </title>
         </bookmark>
-        <bookmark href="http://doc.trolltech.com/qq/">
-            <title>Qt Quarterly</title>
+        <bookmark href="http://pardus-wiki.org/">
+            <title>Pardus-Wiki</title>
         </bookmark>
-        <bookmark href="http://labs.trolltech.com/">
-            <title>Trolltech Labs</title>
+        <bookmark href="http://bugs.pardus.org.tr/">
+            <title>Pardus Bugzilla</title>
         </bookmark>
-        <bookmark href="http://www.qtcentre.org/">
-            <title>Qt Centre</title>
+        <bookmark href="http://lists.pardus.org.tr/">
+            <title>Pardus Lists</title>
         </bookmark>
-        <bookmark href="http://qt-apps.org/">
-            <title>Qt-Apps.org</title>
-        </bookmark>
-        <bookmark href="http://xkcd.com/">
-            <title>xkcd</title>
+        <bookmark href="http://planet.pardus.org.tr">
+            <title>Planet Pardus</title>
         </bookmark>
     </folder>
     <folder folded="yes">
         <title>Bookmarks Menu</title>
-        <bookmark href="http://reddit.com/">
-            <title>reddit.com: what's new online!</title>
+        <bookmark href="http://arora-browser.org/">
+            <title>Arora</title>
         </bookmark>
     </folder>
 </xbel>
