--- simple-ccsm.desktop.in.orig	2008-07-09 23:38:19.000000000 +0300
+++ simple-ccsm.desktop.in	2008-07-09 23:39:04.000000000 +0300
@@ -1,8 +1,9 @@
 [Desktop Entry]
 Encoding=UTF-8
-_Name=Simple CompizConfig Settings Manager
+Name=Simple CompizConfig
 _Comment=Configure Compiz with CompizConfig
-Icon=simple-ccsm
+GenericName[tr]=Compiz Fusion Profil Yönetici
+Icon=simpleccsm
 Exec=simple-ccsm
 Terminal=false
 Type=Application
