--- CMakeLists.txt.orig	2008-07-24 13:56:11.000000000 +0300
+++ CMakeLists.txt	2008-09-06 13:19:44.000000000 +0300
@@ -46,9 +46,9 @@
     SET (DOC_DIR ":/doc/")
     SET (ICON_DIR ":/icons/")
 ELSE (WANT_RESOURCES)
-    SET (TRANSLATION_DIR "${CMAKE_INSTALL_PREFIX}/share/sqliteman/")
-    SET (DOC_DIR "${CMAKE_INSTALL_PREFIX}/share/sqliteman/doc/")
-    SET (ICON_DIR "${CMAKE_INSTALL_PREFIX}/share/sqliteman/icons")
+    SET (TRANSLATION_DIR "/usr/share/sqliteman/")
+    SET (DOC_DIR "/usr/share/sqliteman/doc/")
+    SET (ICON_DIR "/usr/share/sqliteman/icons")
 ENDIF (WANT_RESOURCES)
 
 
@@ -154,6 +154,6 @@
     )
     INSTALL(FILES
         sqliteman/icons/sqliteman.png
-        DESTINATION "share/icons"
+        DESTINATION "share/pixmaps"
     )
 ENDIF (WIN32)
