diff -Naur gcstar.orig/share/applications/gcstar.desktop gcstar/share/applications/gcstar.desktop
--- gcstar.orig/share/applications/gcstar.desktop	2008-07-26 14:59:15.000000000 +0300
+++ gcstar/share/applications/gcstar.desktop	2008-10-28 17:57:14.000000000 +0200
@@ -1,10 +1,13 @@
 [Desktop Entry]
 Version=1.0
 Name=GCstar Collections Manager
+Name[tr]=GCstar
 Name[es]=Gestor de colecciones GCstar
 Name[fr]=GCstar gestionnaire de collections
 GenericName=Personal Collections Manager
+GenericName[tr]=Özel Kolleksiyon Yöneticisi
 Comment=Manage your collections of movies, video games, books, music,...
+Comment[tr]=Film, Oyun, Kitap ve Müzik Kolleksiyonunuzu Yönetin...
 Comment[en]=Manage your collections of movies, video games, books, music,...
 Comment[es]=Organice sus colecciones de películas, juegos, vídeos, libros, música...
 Comment[fr]=Gérer vos collections de films, jeux vidéos, livres, musique,...
@@ -14,5 +17,5 @@
 Terminal=false
 StartupNotify=true
 Type=Application
-Categories=Office;
+Categories=Office;Database;GTK;
-MimeType=application/x-gcstar
+MimeType=application/x-gcstar;
