--- lib/gdk-pixbuf.loaders.orig	2007-01-17 21:10:55.000000000 +0200
+++ lib/gdk-pixbuf.loaders	2007-01-17 21:11:24.000000000 +0200
@@ -4,25 +4,25 @@
 #
 #LoaderDir = lib/gdk-pixbuf/loaders
 #
-"/opt/mysql-gui-tools-5.0/lib/gdk-pixbuf/loaders/libpixbufloader-xpm.so"
+"/opt/mysql-query-browser/lib/gdk-pixbuf/loaders/libpixbufloader-xpm.so"
 "xpm" 4 "gtk20" "The XPM image format"
 "image/x-xpixmap" ""
 "xpm" ""
 "/* XPM */" "" 100
 
-"/opt/mysql-gui-tools-5.0/lib/gdk-pixbuf/loaders/libpixbufloader-gif.so"
+"/opt/mysql-query-browser/lib/gdk-pixbuf/loaders/libpixbufloader-gif.so"
 "gif" 4 "gtk20" "The GIF image format"
 "image/gif" ""
 "gif" ""
 "GIF8" "" 100
 
-"/opt/mysql-gui-tools-5.0/lib/gdk-pixbuf/loaders/libpixbufloader-png.so"
+"/opt/mysql-query-browser/lib/gdk-pixbuf/loaders/libpixbufloader-png.so"
 "png" 5 "gtk20" "The PNG image format"
 "image/png" ""
 "png" ""
 "\211PNG\r\n\032\n" "" 100
 
-"/opt/mysql-gui-tools-5.0/lib/gdk-pixbuf/loaders/libpixbufloader-jpeg.so"
+"/opt/mysql-query-browser/lib/gdk-pixbuf/loaders/libpixbufloader-jpeg.so"
 "jpeg" 5 "gtk20" "The JPEG image format"
 "image/jpeg" ""
 "jpeg" "jpe" "jpg" ""
--- lib/pango.modules.orig	2007-01-17 21:12:28.000000000 +0200
+++ lib/pango.modules	2007-01-17 21:12:48.000000000 +0200
@@ -1,2 +1,2 @@
-/opt/mysql-gui-tools-5.0/lib/pango/pango-basic-fc.so BasicScriptEngineFc PangoEngineShape PangoRenderFc armenian:* bopomofo:* cherokee:* coptic:* cyrillic:* deseret:* ethiopic:* georgian:* gothic:* greek:* han:* hiragana:* katakana:* latin:* ogham:* old-italic:* runic:* canadian-aboriginal:* yi:* braille:* cypriot:* limbu:* osmanya:* shavian:* linear-b:* ugaritic:* common:
-/opt/mysql-gui-tools-5.0/lib/pango/pango-basic-x.so BasicScriptEngineX PangoEngineShape PangoRenderX common:
+/opt/mysql-query-browser/lib/pango/pango-basic-fc.so BasicScriptEngineFc PangoEngineShape PangoRenderFc armenian:* bopomofo:* cherokee:* coptic:* cyrillic:* deseret:* ethiopic:* georgian:* gothic:* greek:* han:* hiragana:* katakana:* latin:* ogham:* old-italic:* runic:* canadian-aboriginal:* yi:* braille:* cypriot:* limbu:* osmanya:* shavian:* linear-b:* ugaritic:* common:
+/opt/mysql-query-browser/lib/pango/pango-basic-x.so BasicScriptEngineX PangoEngineShape PangoRenderX common:
--- lib/pangorc.orig	2007-01-17 21:12:14.000000000 +0200
+++ lib/pangorc	2007-01-17 21:13:04.000000000 +0200
@@ -1,2 +1,2 @@
 [Pango]
-ModuleFiles = /opt/mysql-gui-tools-5.0/lib/pango.modules
+ModuleFiles = /opt/mysql-query-browser/lib/pango.modules
