diff -Naur kvm.orig/src/kvm_convert_hq.desktop kvm/src/kvm_convert_hq.desktop
--- kvm.orig/src/kvm_convert_hq.desktop	2007-01-02 01:26:44.000000000 +0200
+++ kvm/src/kvm_convert_hq.desktop	2007-05-25 22:09:34.000000000 +0300
@@ -23,51 +23,60 @@
 Encoding=UTF-8
 X-KDE-Submenu=Kvm - Convert to high quality
 X-KDE-Submenu[fr]=Kvm - Convertir en haute qualité
+X-KDE-Submenu[tr]=Kvm - Yüksek kaliteye çevir
 
 [Desktop Action PalDv]
 Name=Convert to DV (Pal)
 Name[fr]=Convertir en DV (Pal)
+Name[tr]=DV'ye dönüştür (Pal)
 Icon=video
 Exec=kvm "%f" pal-dv
 
 [Desktop Action NtscDv]
 Name=Convert to DV (NTSC)
 Name[fr]=Convertir en DV (NTSC)
+Name[tr]=DV'ye dönüştür (NTSC)
 Icon=video
 Exec=kvm "%f" ntsc-dv
 
 [Desktop Action PalDvd]
 Name=Convert to DVD (Pal)
 Name[fr]=Convertir en DVD (Pal)
+Name[tr]=DVD'ye dönüştür (Pal)
 Icon=video
 Exec=kvm "%f" pal-dvd
 
 [Desktop Action NtscDvd]
 Name=Convert to DVD (NTSC)
 Name[fr]=Convertir en DVD (NTSC)
+Name[tr]=DVD'ye dönüştür (NTSC)
 Icon=video
 Exec=kvm "%f" ntsc-dvd
 
 [Desktop Action PalVcd]
 Name=Convert to VCD (Pal)
 Name[fr]=Convertir en VCD (Pal)
+Name[tr]=VCD'ye dönüştür (Pal)
 Icon=video
 Exec=kvm "%f" pal-vcd
 
 [Desktop Action NtscVcd]
 Name=Convert to VCD (NTSC)
 Name[fr]=Convertir en VCD (NTSC)
+Name[tr]=VCD'ye dönüştür (NTSC)
 Icon=video
 Exec=kvm "%f" ntsc-vcd
 
 [Desktop Action PalSvcd]
 Name=Convert to SVCD (Pal)
 Name[fr]=Convertir en SVCD (Pal)
+Name[tr]=SVCD'ye dönüştür (Pal)
 Icon=video
 Exec=kvm "%f" pal-svcd
 
 [Desktop Action NtscSvcd]
 Name=Convert to SVCD (NTSC)
 Name[fr]=Convertir en SVCD (NTSC)
+Name[tr]=SVCD'ye dönüştür (NTSC)
 Icon=video
 Exec=kvm "%f" ntsc-svcd
\ Dosya sonunda yenisatır yok.
diff -Naur kvm.orig/src/kvm_convert_lq.desktop kvm/src/kvm_convert_lq.desktop
--- kvm.orig/src/kvm_convert_lq.desktop	2007-01-22 00:40:36.000000000 +0200
+++ kvm/src/kvm_convert_lq.desktop	2007-05-25 22:13:25.000000000 +0300
@@ -23,51 +23,60 @@
 Encoding=UTF-8
 X-KDE-Submenu=Kvm - Convert to low quality
 X-KDE-Submenu[fr]=Kvm - Convertir en faible qualité
+X-KDE-Submenu[tr]=Kvm - Düşük kaliteye çevir
 
 [Desktop Action IpodStd]
 Name=Convert to Ipod format (std)
 Name[fr]=Convertir au format Ipod (std)
+Name[tr]=Ipod biçimine dönüştür (std)
 Icon=video
 Exec=kvm "%f" ipod-std
 
 [Desktop Action IpodPerso]
 Name=Convert to Ipod format (perso)
 Name[fr]=Convertir au format Ipod (perso)
+Name[tr]=Ipod biçimine dönüştür (perso)
 Icon=video
 Exec=kvm "%f" ipod-perso
 
 [Desktop Action PspStd]
 Name=Convert to PSP format (std)
 Name[fr]=Convertir au format PSP (std)
+Name[tr]=PSP biçimine dönüştür (std)
 Icon=video
 Exec=kvm "%f" psp-std
 
 [Desktop Action PspPerso]
 Name=Convert to PSP format (perso)
 Name[fr]=Convertir au format PSP (perso)
+Name[tr]=PSP biçimine dönüştür (std)
 Icon=video
 Exec=kvm "%f" psp-perso
 
 [Desktop Action FlvHigh]
 Name=Convert to flv (large)
 Name[fr]=Convertir en flv (grand)
+Name[tr]=flv biçimine dönüştür (büyük)
 Icon=video
 Exec=kvm "%f" flv-high
 
 [Desktop Action FlvMed]
 Name=Convert to flv (medium)
 Name[fr]=Convertir en flv (moyen)
+Name[tr]=flv biçimine dönüştür (orta)
 Icon=video
 Exec=kvm "%f" flv-medium
 
 [Desktop Action FlvLow]
 Name=Convert to flv (small)
 Name[fr]=Convertir en flv (petit)
+Name[tr]=flv biçimine dönüştür (küçük)
 Icon=video
 Exec=kvm "%f" flv-low
 
 [Desktop Action FlvPerso]
 Name=Convert to flv (perso)
 Name[fr]=Convertir en flv (perso)
+Name[tr]=flv biçimine dönüştür (perso)
 Icon=video
 Exec=kvm "%f" flv-perso
\ Dosya sonunda yenisatır yok.
diff -Naur kvm.orig/src/kvm_convert_mq.desktop kvm/src/kvm_convert_mq.desktop
--- kvm.orig/src/kvm_convert_mq.desktop	2007-01-27 18:13:55.000000000 +0200
+++ kvm/src/kvm_convert_mq.desktop	2007-05-25 22:15:34.000000000 +0300
@@ -23,75 +23,88 @@
 Encoding=UTF-8
 X-KDE-Submenu=Kvm - Convert to medium quality
 X-KDE-Submenu[fr]=Kvm - Convertir en moyenne qualité
+X-KDE-Submenu[tr]=Kvm - Orta kaliteye çevir
 
 [Desktop Action MpgHigh]
 Name=Convert to mpg1 (HQ)
 Name[fr]=Convertir en mpg1 (HQ)
+Name[tr]=mpg1 biçimine dönüştür (HQ)
 Icon=video
 Exec=kvm "%f" mpg-high
 
 [Desktop Action MpgMed]
 Name=Convert to mpg1 (MQ)
 Name[fr]=Convertir en mpg1 (MQ)
+Name[tr]=mpg1 biçimine dönüştür (MQ)
 Icon=video
 Exec=kvm "%f" mpg-medium
 
 [Desktop Action MpgLow]
 Name=Convert to mpg1 (LQ)
 Name[fr]=Convertir en mpg1 (LQ)
+Name[tr]=mpg1 biçimine dönüştür (LQ)
 Icon=video
 Exec=kvm "%f" mpg-low
 
 [Desktop Action MpgPerso]
 Name=Convert to mpg1 (perso)
 Name[fr]=Convertir en mpg1 (perso)
+Name[tr]=mpg1 biçimine dönüştür (perso)
 Icon=video
 Exec=kvm "%f" mpg-perso
 
 [Desktop Action Mpg4High]
 Name=Convert to mpg4 (HQ)
 Name[fr]=Convertir en mpg4 (HQ)
+Name[tr]=mpg4 biçimine dönüştür (HQ)
 Icon=video
 Exec=kvm "%f" mpg4-high
 
 [Desktop Action Mpg4Med]
 Name=Convert to mpg4 (MQ)
 Name[fr]=Convertir en mpg4 (MQ)
+Name[tr]=mpg4 biçimine dönüştür (MQ)
 Icon=video
 Exec=kvm "%f" mpg4-medium
 
 [Desktop Action Mpg4Low]
 Name=Convert to mpg4 (LQ)
 Name[fr]=Convertir en mpg4 (LQ)
+Name[tr]=mpg4 biçimine dönüştür (LQ)
 Icon=video
 Exec=kvm "%f" mpg4-low
 
 [Desktop Action Mpg4Perso]
 Name=Convert to mpg4 (perso)
 Name[fr]=Convertir en mpg4 (perso)
+Name[tr]=mpg4 biçimine dönüştür (perso)
 Icon=video
 Exec=kvm "%f" mpg4-perso
 
 [Desktop Action MovHigh]
 Name=Convert to mov (HQ)
 Name[fr]=Convertir en mov (HQ)
+Name[tr]=mov biçimine dönüştür (HQ)
 Icon=video
 Exec=kvm "%f" mov-high
 
 [Desktop Action MovMed]
 Name=Convert to mov (MQ)
 Name[fr]=Convertir en mov (MQ)
+Name[tr]=mov biçimine dönüştür (MQ)
 Icon=video
 Exec=kvm "%f" mov-medium
 
 [Desktop Action MovLow]
 Name=Convert to mov (LQ)
 Name[fr]=Convertir en mov (LQ)
+Name[tr]=mov biçimine dönüştür (LQ)
 Icon=video
 Exec=kvm "%f" mov-low
 
 [Desktop Action MovPerso]
 Name=Convert to mov (perso)
 Name[fr]=Convertir en mov (perso)
+Name[tr]=mov biçimine dönüştür (perso)
 Icon=video
 Exec=kvm "%f" mov-perso
\ Dosya sonunda yenisatır yok.
diff -Naur kvm.orig/src/kvm_modify.desktop kvm/src/kvm_modify.desktop
--- kvm.orig/src/kvm_modify.desktop	2007-01-21 01:25:42.000000000 +0200
+++ kvm/src/kvm_modify.desktop	2007-05-25 22:33:06.000000000 +0300
@@ -23,81 +23,95 @@
 Encoding=UTF-8
 X-KDE-Submenu=Kvm - Crop and Resize
 X-KDE-Submenu[fr]=Kvm - Découper et redimensionner
+X-KDE-Submenu[tr]=Kvm - Kes ve Boyutlandır
 
 [Desktop Action RsizeHD720i]
 Name=Resize (HD720i)
 Name[fr]=Redimensionner (HD720i)
+Name[tr]=Boyutlandır (HD720i)
 Icon=video
 Exec=kvm "%f" size-HD720i
 
 [Desktop Action Rsize720x576]
 Name=Resize (720x576)
 Name[fr]=Redimensionner (720x576)
+Name[tr]=Boyutlandır (720x576)
 Icon=video
 Exec=kvm "%f" size-720x576
 
 [Desktop Action Rsize640x480]
 Name=Resize (640x480)
 Name[fr]=Redimensionner (640x480)
+Name[tr]=Boyutlandır (640x480)
 Icon=video
 Exec=kvm "%f" size-640x480
 
 [Desktop Action Rsize320x240]
 Name=Resize (320x240)
 Name[fr]=Redimensionner (320x240)
+Name[tr]=Boyutlandır (320x240)
 Icon=video
 Exec=kvm "%f" size-320x240
 
 [Desktop Action RsizePerso]
 Name=Resize (perso)
 Name[fr]=Redimensionner (perso)
+Name[tr]=Boyutlandır (perso)
 Icon=video
 Exec=kvm "%f" size-perso
 
 [Desktop Action Crop720x576]
 Name=Convert to 16/9 (720x576)
 Name[fr]=Convertir en 16/9 (720x576)
+Name[tr]=Dönüştür 16/9 (720x576)
 Icon=video
 Exec=kvm "%f" 16/9-720x576
 
 [Desktop Action Crop640x480]
 Name=Convert to 16/9 (640x480)
 Name[fr]=Convertir en 16/9 (640x480)
+Name[tr]=Dönüştür 16/9 (640x480)
 Icon=video
 Exec=kvm "%f" 16/9-640x480
 
 [Desktop Action Crop320x240]
 Name=Convert to 16/9 (320x240)
 Name[fr]=Convertir en 16/9 (320x240)
+Name[tr]=Dönüştür 16/9 (320x240)
 Icon=video
 Exec=kvm "%f" 16/9-320x240
 
 [Desktop Action CropMovie]
 Name=Crop the video
 Name[fr]=Retailler la video
+Name[tr]=Videoyu kes
 Icon=video
 Exec=kvm "%f" crop-movie
 
 [Desktop Action CutMovie]
 Name=Extract a part of the video
 Name[fr]=Extraire une partie de la video
+Name[tr]=Videonun bir parçasını ayır
 Icon=video
 Exec=kvm "%f" cut-movie
 
 [Desktop Action Deinterlace]
 Name=Deinterlace the video
 Name[fr]=Désentrelacer la video
+Name[tr]=Videoyu parçalara ayır
 Icon=video
 Exec=kvm "%f" deinterlace
 
 [Desktop Action Aspect4/3]
 Name=Set the aspect ratio to 4:3
 Name[fr]=Régler l'aspect à 4:3
+Name[tr]=Ekran çözünürlüğünü 4:3 
 Icon=video
 Exec=kvm "%f" aspect-4/3
 
 [Desktop Action Aspect16/9]
 Name=Set the aspect ratio to 16:9
 Name[fr]=Régler l'aspect à 16:9
+Name[tr]=Ekran çözünürlüğünü 16:9
 Icon=video
 Exec=kvm "%f" aspect-16/9
