--- data/filezilla.orig.desktop	2007-09-08 20:16:48.000000000 +0300
+++ data/filezilla.desktop	2007-09-12 11:30:08.000000000 +0300
@@ -1,7 +1,9 @@
 [Desktop Entry]
 Name=FileZilla
 GenericName=FTP and SFTP client
+GenericName[tr]=FTP ve SFTP istemcisi
 Comment=Download and upload files via ftp, sftp and ftps
+Comment[tr]=Dosyaları ftp, sftp, ftps ile gönderin ve indirin
 Comment[fr]=Transférer des fichiers via ftp, sftp et ftps
 Exec=filezilla
 Terminal=false
