--- comix.desktop	2006-08-24 12:32:13.000000000 +0300
+++ comix-yeni.desktop	2006-08-30 10:38:03.000000000 +0300
@@ -2,15 +2,18 @@
 Encoding=UTF-8
 Name=Comix
 Name[it]=Comix
+Name[tr]=Comix
 GenericName=Comic book viewer
 GenericName[sv]=Serieboksläsare
 GenericName[fr]=Visionneur de bandes dessinées
+GenericName[tr]=Çizgi roman göstericisi
 Comment=A viewer for comic book archives
 Comment[it]=Un visualizzatore di fumetti 
 Comment[es]=Un visor de comics
 Comment[sv]=En serieboksläsare
 Comment[fr]=Visionneur d'images spécialisé dans la lecture des bandes dessinées
 Comment[pl]=Przeglądarka komiksów
+Comment[tr]=Çizgi roman arşivleri için gösterici
 Exec=comix %f
 Icon=comix.png
 Terminal=false
