--- src/Makefile.in.orig	2008-11-27 03:25:23.000000000 +0200
+++ src/Makefile.in	2008-11-27 03:27:31.000000000 +0200
@@ -56,7 +56,7 @@
   esac;
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
 gtk_recordMyDesktopPYTHON_INSTALL = $(INSTALL_DATA)
-py_compile = $(top_srcdir)/py-compile
+# py_compile = $(top_srcdir)/py-compile
 desktopDATA_INSTALL = $(INSTALL_DATA)
 pixmapDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(desktop_DATA) $(pixmap_DATA)
@@ -272,13 +272,13 @@
 	    $(gtk_recordMyDesktopPYTHON_INSTALL) "$$b$$p" "$(DESTDIR)$(gtk_recordMyDesktopdir)/$$f"; \
 	  else :; fi; \
 	done; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(gtk_recordMyDesktopdir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(gtk_recordMyDesktopdir)" $$dlist; \
-	  fi; \
-	else :; fi
+#	if test -n "$$dlist"; then \
+#	  if test -z "$(DESTDIR)"; then \
+#	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(gtk_recordMyDesktopdir)" $$dlist; \
+#	  else \
+#	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(gtk_recordMyDesktopdir)" $$dlist; \
+#	  fi; \
+#	else :; fi
 
 uninstall-gtk_recordMyDesktopPYTHON:
 	@$(NORMAL_UNINSTALL)
