--- for-user-to-change-config-settings	2007-07-24 19:08:09.000000000 +0300
+++ for-user-to-change-config-settings1	2007-08-04 00:35:27.000000000 +0300
@@ -14,7 +14,7 @@
 ## software is being built and will be used.
 
 unix {
-    PREFIX = "/usr/local"
+    PREFIX = "/usr"
     BASEBINDIR = "$${PREFIX}/bin"
     PLUGINDIR = "$${PREFIX}/lib/$${TARGET}/plugins"
     LOCALEDIR = "$${PREFIX}/share/$${TARGET}/locales"
