diff -Nuar cinelerra.orig/cinelerra/ffmpeg.C cinelerra/cinelerra/ffmpeg.C
--- cinelerra.orig/cinelerra/ffmpeg.C	2009-08-01 14:56:51.000000000 +0300
+++ cinelerra/cinelerra/ffmpeg.C	2009-08-01 15:27:57.309992478 +0300
@@ -2,7 +2,8 @@
 
 #ifdef HAVE_SWSCALER
 extern "C" {
-#include <swscale.h>
+#include <libavcodec/avcodec.h>
+#include <libswscale/swscale.h>
 }
 #endif
 
diff -Nuar cinelerra.orig/cinelerra/ffmpeg.h cinelerra/cinelerra/ffmpeg.h
--- cinelerra.orig/cinelerra/ffmpeg.h	2009-06-01 16:41:26.000000000 +0300
+++ cinelerra/cinelerra/ffmpeg.h	2009-08-01 15:21:46.490990660 +0300
@@ -2,7 +2,7 @@
 #define FFMPEG_H
 
 extern "C" {
-#include <avcodec.h>
+#include <libavcodec/avcodec.h>
 }
 
 #include "asset.h"
diff -Nuar cinelerra.orig/cinelerra/fileac3.h cinelerra/cinelerra/fileac3.h
--- cinelerra.orig/cinelerra/fileac3.h	2009-06-01 16:41:26.000000000 +0300
+++ cinelerra/cinelerra/fileac3.h	2009-08-01 15:23:16.684994164 +0300
@@ -2,7 +2,7 @@
 #define FILEAC3_H
 
 extern "C" {
-#include <avcodec.h>
+#include <libavcodec/avcodec.h>
 };
 
 #include "filebase.h"
diff -Nuar cinelerra.orig/cinelerra/filedv.h cinelerra/cinelerra/filedv.h
--- cinelerra.orig/cinelerra/filedv.h	2009-06-01 16:41:26.000000000 +0300
+++ cinelerra/cinelerra/filedv.h	2009-08-01 15:25:00.411988285 +0300
@@ -6,7 +6,7 @@
 #include "file.inc"
 
 #ifdef DV_USE_FFMPEG
-#include <avcodec.h>
+#include <libavcodec/avcodec.h>
 #endif
 
 #include <libdv/dv.h>
diff -Nuar cinelerra.orig/cinelerra/Makefile.am cinelerra/cinelerra/Makefile.am
--- cinelerra.orig/cinelerra/Makefile.am	2009-06-01 16:41:25.000000000 +0300
+++ cinelerra/cinelerra/Makefile.am	2009-08-01 15:04:04.989993039 +0300
@@ -687,6 +687,7 @@
 	$(SOUND_LDFLAGS) \
 	$(FIREWIRE_LDFLAGS) \
 	$(LIBDV_LIBS) \
+        $(FFMPEG_LIBS) \
 	-luuid \
 	$(MJPEG_LIBS)
 
diff -Nuar cinelerra.orig/configure.in cinelerra/configure.in
--- cinelerra.orig/configure.in	2009-08-01 14:56:51.000000000 +0300
+++ cinelerra/configure.in	2009-08-01 15:06:15.174991475 +0300
@@ -470,10 +470,6 @@
 					libmpeg3/Makefile libmpeg3/video/Makefile libmpeg3/audio/Makefile \
 					quicktime/Makefile \
 					quicktime/encore50/Makefile \
-	quicktime/ffmpeg/Makefile \
-	quicktime/ffmpeg/libavcodec/Makefile \
-	quicktime/ffmpeg/libavcodec/i386/Makefile \
-	quicktime/ffmpeg/libavcodec/ppc/Makefile \
 					mpeg2enc/Makefile toolame-02l/Makefile \
 					guicast/Makefile cinelerra/Makefile \
 	cinelerra/data/Makefile \
diff -Nuar cinelerra.orig/quicktime/mpeg4.c cinelerra/quicktime/mpeg4.c
--- cinelerra.orig/quicktime/mpeg4.c	2009-06-01 16:41:28.000000000 +0300
+++ cinelerra/quicktime/mpeg4.c	2009-08-01 15:08:30.546989783 +0300
@@ -7,7 +7,7 @@
 
 
 
-#include "avcodec.h"
+#include <libavcodec/avcodec.h>
 #include "colormodels.h"
 #include "funcprotos.h"
 #include "qtffmpeg.h"
diff -Nuar cinelerra.orig/quicktime/qtffmpeg.h cinelerra/quicktime/qtffmpeg.h
--- cinelerra.orig/quicktime/qtffmpeg.h	2009-06-01 16:41:28.000000000 +0300
+++ cinelerra/quicktime/qtffmpeg.h	2009-08-01 15:10:23.174989425 +0300
@@ -17,7 +17,7 @@
 
 
 
-#include "avcodec.h"
+#include <libavcodec/avcodec.h>
 #include "qtprivate.h"
 
 
diff -Nuar cinelerra.orig/quicktime/qth264.c cinelerra/quicktime/qth264.c
--- cinelerra.orig/quicktime/qth264.c	2009-06-01 16:41:28.000000000 +0300
+++ cinelerra/quicktime/qth264.c	2009-08-01 15:11:26.607991646 +0300
@@ -1,4 +1,4 @@
-#include "avcodec.h"
+#include <libavcodec/avcodec.h>
 #include "colormodels.h"
 #include "funcprotos.h"
 #include <pthread.h>
diff -Nuar cinelerra.orig/quicktime/wma.c cinelerra/quicktime/wma.c
--- cinelerra.orig/quicktime/wma.c	2009-06-01 16:41:28.000000000 +0300
+++ cinelerra/quicktime/wma.c	2009-08-01 15:12:37.851988013 +0300
@@ -1,4 +1,4 @@
-#include "avcodec.h"
+#include <libavcodec/avcodec.h>
 #include "funcprotos.h"
 #include "quicktime.h"
 #include <string.h>
