diff -Naur cnr.orig/guicast/bcresources.C cnr/guicast/bcresources.C
--- cnr.orig/guicast/bcresources.C	2010-09-20 10:53:48.000000000 +0400
+++ cnr/guicast/bcresources.C	2010-09-22 14:11:09.000000000 +0400
@@ -73,11 +73,11 @@
 char* BC_Resources::large_fontset = "8x16,*";
 
 char* BC_Resources::small_font_xft = N_("-*-luxi sans-*-r-*-*-12-*-*-*-*-*-*-*");
-char* BC_Resources::small_font_xft2 = N_("-microsoft-verdana-*-*-*-*-*-*-*-*-*-*-*-*");
+char* BC_Resources::small_font_xft2 = N_("-*-verdana-*-*-*-*-*-*-*-*-*-*-*-*");
 char* BC_Resources::medium_font_xft = N_("-*-luxi sans-*-r-*-*-16-*-*-*-*-*-*-*");
-char* BC_Resources::medium_font_xft2 = N_("-microsoft-verdana-*-*-*-*-*-*-*-*-*-*-*-*");
+char* BC_Resources::medium_font_xft2 = N_("-*-verdana-*-*-*-*-*-*-*-*-*-*-*-*");
 char* BC_Resources::large_font_xft = N_("-*-luxi sans-bold-r-*-*-20-*-*-*-*-*-*-*");
-char* BC_Resources::large_font_xft2 = N_("-microsoft-verdana-*-*-*-*-*-*-*-*-*-*-*-*");
+char* BC_Resources::large_font_xft2 = N_("-*-verdana-*-*-*-*-*-*-*-*-*-*-*-*");
 
 suffix_to_type_t BC_Resources::suffix_to_type[] = 
 {
diff -Naur cnr.orig/plugins/titler/title.C cnr/plugins/titler/title.C
--- cnr.orig/plugins/titler/title.C	2010-09-20 10:53:50.000000000 +0400
+++ cnr/plugins/titler/title.C	2010-09-22 14:10:01.000000000 +0400
@@ -55,8 +55,7 @@
 
 #define ZERO (1.0 / 64.0)
 
-#define FONT_SEARCHPATH "fonts"
-//#define FONT_SEARCHPATH "/usr/X11R6/lib/X11/fonts"
+#define FONT_SEARCHPATH "/usr/share/fonts"
 
 
 REGISTER_PLUGIN(TitleMain)
@@ -1082,11 +1081,8 @@
 	if(!fonts)
 	{
 		fonts = new ArrayList<FontEntry*>;
-// Construct path from location of the plugin
 		char search_path[BCTEXTLEN];
-		strcpy(search_path, PluginClient::get_path());
-		char *ptr = strrchr(search_path, '/');
-		strcpy(ptr + 1, FONT_SEARCHPATH);
+		strcpy(search_path, FONT_SEARCHPATH);
 		char command_line[BCTEXTLEN];
 
 		sprintf(command_line, 
