--- plugins/titler/title.C.orig	2009-10-24 12:12:11.000000000 +0300
+++ plugins/titler/title.C	2009-10-25 11:59:21.404893765 +0200
@@ -55,7 +55,7 @@
 
 #define ZERO (1.0 / 64.0)
 
-#define FONT_SEARCHPATH "fonts"
+#define FONT_SEARCHPATH "/usr/share/fonts"
 //#define FONT_SEARCHPATH "/usr/X11R6/lib/X11/fonts"
 
 
@@ -1082,11 +1082,11 @@
 	if(!fonts)
 	{
 		fonts = new ArrayList<FontEntry*>;
-// Construct path from location of the plugin
+// Construct path from location of the plugin - bad idea :  fonts are arch independant.
 		char search_path[BCTEXTLEN];
-		strcpy(search_path, PluginClient::get_path());
-		char *ptr = strrchr(search_path, '/');
-		strcpy(ptr + 1, FONT_SEARCHPATH);
+		//strcpy(search_path, PluginClient::get_path());
+		strcpy(search_path, FONT_SEARCHPATH);
+		//char *ptr = strrchr(search_path, '/');
 		char command_line[BCTEXTLEN];
 
 		sprintf(command_line, 
