diff -up cinelerra-2.1/Makefile.am.libmpeg3_s cinelerra-2.1/Makefile.am
--- cinelerra-2.1/Makefile.am.libmpeg3_s	2008-11-03 18:25:11.000000000 +0100
+++ cinelerra-2.1/Makefile.am	2008-11-04 01:53:16.000000000 +0100
@@ -1,5 +1,9 @@
-SUBDIRS=libmpeg3 quicktime mpeg2enc toolame-02l \
-	guicast cinelerra mplexlo plugins po image m4
+if HAVE_LIBMPEG3_INTERNAL
+  MAYBE_LIBMPEG3 = libmpeg3 mplexlo
+endif
+
+SUBDIRS=$(MAYBE_LIBMPEG3) quicktime mpeg2enc toolame-02l \
+	guicast cinelerra plugins po image m4
 EXTRA_DIST = admin debian doc depcomp README.BUILD LICENSE \
 	cinelerra-cvs-current.spec
 AUTOMAKE_OPTIONS=foreign
diff -up cinelerra-2.1/quicktime/qtmp3.c.libmpeg3_s cinelerra-2.1/quicktime/qtmp3.c
--- cinelerra-2.1/quicktime/qtmp3.c.libmpeg3_s	2008-11-04 01:53:36.000000000 +0100
+++ cinelerra-2.1/quicktime/qtmp3.c	2008-11-04 01:53:47.000000000 +0100
@@ -4,7 +4,6 @@
 
 
 #include "mpeg3private.h"
-#include "mpeg3protos.h"
 #include "quicktime.h"
 #include "qtmp3.h"
 #include <string.h>
diff -up cinelerra-2.1/quicktime/Makefile.am.libmpeg3_s cinelerra-2.1/quicktime/Makefile.am
--- cinelerra-2.1/quicktime/Makefile.am.libmpeg3_s	2008-11-03 18:25:12.000000000 +0100
+++ cinelerra-2.1/quicktime/Makefile.am	2008-11-04 01:53:16.000000000 +0100
@@ -21,7 +21,7 @@ libquicktimehv_la_LIBADD = \
 	$(FAAD_LIBS) \
 	-lfaac \
 	encore50/libencore.la \
-	$(top_builddir)/libmpeg3/libmpeg3hv.la \
+	$(LIBMPEG3_LIBS) \
 	$(LIBX264_LIBS) \
 	$(LIBDV_LIBS) \
 	-ljpeg -lpng \
@@ -119,7 +119,7 @@ noinst_HEADERS = cmodel_permutation.h \
 	
 EXTRA_DIST = docs
 
-INCLUDES = -I$(top_srcdir)/libmpeg3
+INCLUDES = $(LIBMPEG3_CFLAGS)
 
 pkgincludedir=$(includedir)/quicktime
 pkginclude_HEADERS=quicktime.h qtprivate.h
diff -up cinelerra-2.1/configure.in.libmpeg3_s cinelerra-2.1/configure.in
--- cinelerra-2.1/configure.in.libmpeg3_s	2008-11-04 01:53:16.000000000 +0100
+++ cinelerra-2.1/configure.in	2008-11-04 01:53:16.000000000 +0100
@@ -407,6 +407,20 @@ fi
 echo --------------------------------------------------
 ############# END run ffmpeg configure to generates avconfig.h
 
+############ external libmpeg3
+AC_ARG_WITH([external-libmpeg3], AC_HELP_STRING([--with-external-libmpeg3], [use external libmpeg3 library]))
+
+if test "x$with_external_libmpeg3" = "xyes"; then
+  PKG_CHECK_MODULES(LIBMPEG3, libmpeg3,[libmpeg3_system=yes],:)
+else
+  LIBMPEG3_CFLAGS="-I\$(top_srcdir)/libmpeg3"
+  LIBMPEG3_LIBS="\$(top_builddir)/libmpeg3/libmpeg3hv.la"
+  AC_SUBST(LIBMPEG3_CFLAGS)
+  AC_SUBST(LIBMPEG3_LIBS)
+fi
+AM_CONDITIONAL(HAVE_LIBMPEG3_INTERNAL, [test "x$libmpeg3_system" != "xyes"])
+############# END external libmpeg3
+
 ############## XIPH
 PKG_CHECK_MODULES(libogg,ogg >= 1.1,[libogg=yes],:)
 PKG_CHECK_MODULES(libvorbis,vorbis,[libvorbis=yes],:)
diff -up cinelerra-2.1/mpeg2enc/Makefile.am.libmpeg3_s cinelerra-2.1/mpeg2enc/Makefile.am
--- cinelerra-2.1/mpeg2enc/Makefile.am.libmpeg3_s	2008-11-04 10:08:43.000000000 +0100
+++ cinelerra-2.1/mpeg2enc/Makefile.am	2008-11-04 10:08:59.000000000 +0100
@@ -4,7 +4,7 @@
 
 noinst_LTLIBRARIES = libmpeg2enc.la
 
-INCLUDES = -I$(top_srcdir)/quicktime -I$(top_srcdir)/libmpeg3
+INCLUDES = -I$(top_srcdir)/quicktime $(LIBMPEG3_CFLAGS)
 AM_CFLAGS = $(LARGEFILE_CFLAGS)
 
 libmpeg2enc_la_SOURCES = conform.c mpeg2enc.c putseq.c putpic.c puthdr.c putmpg.c \
diff -up cinelerra-2.1/cinelerra/Makefile.am.libmpeg3_s cinelerra-2.1/cinelerra/Makefile.am
--- cinelerra-2.1/cinelerra/Makefile.am.libmpeg3_s	2008-11-04 01:53:16.000000000 +0100
+++ cinelerra-2.1/cinelerra/Makefile.am	2008-11-04 01:53:16.000000000 +0100
@@ -344,7 +344,7 @@ cinelerra_SOURCES = aattachmentpoint.C \
 INCLUDES = \
 	-I$(top_srcdir)/guicast \
 	-I$(top_srcdir)/quicktime \
-	-I$(top_srcdir)/libmpeg3
+	$(LIBMPEG3_CFLAGS)
 
 AM_CXXFLAGS = \
 	$(LARGEFILE_CFLAGS) \
@@ -678,7 +678,7 @@ cinelerra_LDADD = \
 	$(top_builddir)/toolame-02l/libtoolame.la \
 	$(top_builddir)/quicktime/libquicktimehv.la \
 	$(top_builddir)/guicast/libguicast.la \
-	$(top_builddir)/libmpeg3/libmpeg3hv.la \
+	$(LIBMPEG3_LIBS) \
 	$(XIPH_LIBS) \
 	$(A52DEC_LIBS) \
 	$(OPENEXR_LIBS) \
