diff -Nuar cinelerra.orig/cinelerra/ffmpeg.C cinelerra/cinelerra/ffmpeg.C
--- cinelerra.orig/cinelerra/ffmpeg.C	2009-06-01 16:41:26.000000000 +0300
+++ cinelerra/cinelerra/ffmpeg.C	2009-06-02 00:59:56.015594065 +0300
@@ -6,6 +6,13 @@
 }
 #endif
 
+/* fix for newer ffmpeg */
+#ifndef PIX_FMT_YUV422
+#define PIX_FMT_YUV422 PIX_FMT_YUYV422
+#endif
+#ifndef PIX_FMT_RGBA32 
+#define PIX_FMT_RGBA32 PIX_FMT_RGB32
+#endif 
 
 #include "filebase.h"
 #include "quicktime.h"
diff -Nuar cinelerra.orig/quicktime/qtffmpeg.c cinelerra/quicktime/qtffmpeg.c
--- cinelerra.orig/quicktime/qtffmpeg.c	2009-06-01 16:41:28.000000000 +0300
+++ cinelerra/quicktime/qtffmpeg.c	2009-06-02 01:02:42.474592672 +0300
@@ -8,9 +8,10 @@
 // FFMPEG front end for quicktime.
 
 
-
-
-
+// fix for newer ffmpeg, don't break w/ older ones...
+#ifndef PIX_FMT_YUV422
+#define PIX_FMT_YUV422 PIX_FMT_YUYV422
+#endif
 
 
 
