--- configure.in.orig	2009-12-30 15:34:03.000000000 +0200
+++ configure.in	2010-03-13 22:58:11.809948287 +0200
@@ -35,9 +35,9 @@
 AH_TEMPLATE(HAVE_VIDEO4LINUX2, [Define to 1 if Video4Linux2 is available.])
 AC_CHECK_HEADER([linux/videodev2.h], 
 	[AC_CHECK_MEMBER(struct v4l2_buffer.index, [AC_DEFINE(HAVE_VIDEO4LINUX2)], [],
-  		[#include <linux/time.h>
-#include <linux/videodev2.h>])],
-	[],[#include <linux/time.h>] )
+                [#include <linux/videodev2.h>])],
+
+        [],[] )
 
 ############## SPECIAL DIRECTORIES
 AC_ARG_WITH(plugindir,
