--- configure.in.orig	2009-06-02 12:49:09.149533952 +0300
+++ configure.in	2009-06-02 12:49:09.153536341 +0300
@@ -35,9 +35,9 @@
 AH_TEMPLATE(HAVE_VIDEO4LINUX2, [Define to 1 if Video4Linux2 is available.])
 AC_CHECK_HEADER([linux/videodev2.h], 
 	[AC_CHECK_MEMBER(struct v4l2_buffer.index, [AC_DEFINE(HAVE_VIDEO4LINUX2)], [],
-  		[#include <linux/time.h>
+  		[#include <time.h>
 #include <linux/videodev2.h>])],
-	[],[#include <linux/time.h>] )
+	[],[#include <time.h>] )
 
 ############## SPECIAL DIRECTORIES
 AC_ARG_WITH(plugindir,
