diff -Naur KDEVisualBoyAdvance-0.3(beta)-orig/src/sdl_link.cpp KDEVisualBoyAdvance-0.3(beta)/src/sdl_link.cpp
--- KDEVisualBoyAdvance-0.3(beta)-orig/src/sdl_link.cpp	2006-09-21 16:54:18.000000000 +0300
+++ KDEVisualBoyAdvance-0.3(beta)/src/sdl_link.cpp	2006-09-21 16:58:43.000000000 +0300
@@ -6,8 +6,7 @@
  * <achrafcherti@gmail.com>
  *
  *****************************************
- * Ce code source est distribué
- * selon la licence de la GNU GPL.
+ * Ce code source est distribuï¿½ * selon la licence de la GNU GPL.
  *****************************************
  * http://vbaexpress.tuxfamily.org/
  *****************************************
@@ -32,7 +31,7 @@
 #include <qdir.h>
 #include <qstring.h>
 #include <kmessagebox.h>
-#include <SDL/DT_drawtext.h>
+#include <SDL_console/DT_drawtext.h>
 #include "Text.h"
 
 static bool stop = false;
