--- jabbin.desktop.orig	2006-07-12 01:22:17.000000000 +0300
+++ jabbin.desktop	2006-10-29 11:35:24.000000000 +0200
@@ -1,10 +1,12 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Jabbin
+Name[tr]=Jabbin
 GenericName=Jabber Client
+GenericName[tr]=Jabber İstemcisi
 Exec=jabbin -caption "%c" %i %m
-Icon=jabbin
+Icon=jabbin.png
 Terminal=0
+Categories=Qt;KDE;Network;
 Type=Application
-X-KDE-StartupNotify=true
-
+X-KDE-StartupNotify=true
\ Dosya sonunda yenisatır yok.
