--- configure.in.orig	2008-10-13 14:59:07.000000000 +0300
+++ configure.in	2008-10-13 15:12:15.000000000 +0300
@@ -625,11 +625,7 @@
   AC_OUTPUT([
     doc/Makefile
     doc/gretl.sty
-    doc/commands/Makefile
     doc/tex/Makefile
-    doc/tex_it/Makefile
-    doc/tex_es/Makefile
-    doc/tex_pt/Makefile
   ])
 fi
 
--- share/Makefile.in.orig	2008-10-13 15:16:45.000000000 +0300
+++ share/Makefile.in	2008-10-13 15:17:07.000000000 +0300
@@ -41,7 +41,6 @@
 all: $(query_make_help)
 
 help:
-	make -C ../doc/commands help-all
 
 .PHONY :
 
