diff -Naur kiba-dock-0.0_20071118.orig/files/kiba-dock.desktop.in.in kiba-dock-0.0_20071118/files/kiba-dock.desktop.in.in
--- kiba-dock-0.0_20071118.orig/files/kiba-dock.desktop.in.in	2007-11-18 03:40:05.000000000 +0200
+++ kiba-dock-0.0_20071118/files/kiba-dock.desktop.in.in	2007-11-18 03:32:51.000000000 +0200
@@ -2,7 +2,9 @@
 Type=Application
 Encoding=UTF-8
 Name=Kiba-Dock
+Name[tr]=Kiba-Dock
 _Comment=Gtk+ Dock with smooth graphics.
+GenericName[tr]=Dinamik Panel
 Exec=kiba-dock
 Icon=%prefix%/%DATADIRNAME%/pixmaps/kiba-dock.png
-Categories=Desktop;Application;Utility;GNOME;
+Categories=Application;Compiz;
diff -Naur kiba-dock-0.0_20071118.orig/kiba-settings/kiba-settings.desktop.in.in kiba-dock-0.0_20071118/kiba-settings/kiba-settings.desktop.in.in
--- kiba-dock-0.0_20071118.orig/kiba-settings/kiba-settings.desktop.in.in	2007-11-18 03:40:05.000000000 +0200
+++ kiba-dock-0.0_20071118/kiba-settings/kiba-settings.desktop.in.in	2007-11-18 03:35:05.000000000 +0200
@@ -2,7 +2,9 @@
 Type=Application
 Encoding=UTF-8
 Name=Kiba-Settings
+Name[tr]=Kiba-Settings
 _Comment=Kiba-Dock Settings Editor.
+GenericName[tr]=Kiba-Dock Yapılandırma Yöneticisi
 Exec=kiba-settings
 Icon=%prefix%/%DATADIRNAME%/pixmaps/kiba-dock.png
-Categories=Settings;Application;Utility;GNOME;
+Categories=Application;Compiz;
