--- src/Makefile.in.orig	2005-10-16 23:11:50.000000000 +0200
+++ src/Makefile.in	2005-10-16 23:13:37.000000000 +0200
@@ -148,7 +148,7 @@
 bin_PROGRAMS = fluxconf
 fluxconf_SOURCES = fluxconf.c fluxmenu.c fluxkeys.c fluxbare.c actions.c 
 fluxconf_INCLUDES = include/globals.h include/defines.h include/functions.h xpm/dalogo.h xpm/fc_menu.xpm xpm/fc_keys.xpm xpm/fc_conf.xpm
-fluxconf_CFLAGS = -Iinclude/ -Ixpm/ -ggdb -D_GNU_SOURCE -Werror
+fluxconf_CFLAGS = -Iinclude/ -Ixpm/ -ggdb -D_GNU_SOURCE
 subdir = src
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
