diff -u dxflib-2.0.4.8-2.src-old/Makefile dxflib-2.0.4.8-2.src/Makefile
--- dxflib-2.0.4.8-2.src-old/Makefile	2005-02-12 23:19:38.000000000 +0200
+++ dxflib-2.0.4.8-2.src/Makefile	2008-02-01 23:39:57.000000000 +0200
@@ -3,7 +3,7 @@
 # Makefile for dxflib
 #
 #
-VERSION = 2.0.3.4
+VERSION = 2.0.4.8
 
 #
 # dxflib installation directories
diff -u dxflib-2.0.4.8-2.src-old/Makefile.in dxflib-2.0.4.8-2.src/Makefile.in
--- dxflib-2.0.4.8-2.src-old/Makefile.in	2005-02-09 00:40:26.000000000 +0200
+++ dxflib-2.0.4.8-2.src/Makefile.in	2008-02-01 23:40:10.000000000 +0200
@@ -3,7 +3,7 @@
 # Makefile for dxflib
 #
 #
-VERSION = 2.0.3.4
+VERSION = 2.0.4.8
 
 #
 # dxflib installation directories
Ortak alt dizinler: dxflib-2.0.4.8-2.src-old/src ve dxflib-2.0.4.8-2.src/src
Ortak alt dizinler: dxflib-2.0.4.8-2.src-old/test ve dxflib-2.0.4.8-2.src/test
