--- ../lib/libscrobble.cpp	2007-09-30 14:19:58.000000000 +0300
+++ ../lib/libscrobble2.cpp	2007-10-18 00:04:06.000000000 +0300
@@ -255,7 +255,7 @@
     scrob_entry* iTunesTable;
     string fileName;
 
-    fileName = db_folder + "iPod_Control/iTunes/iTunesDB";
+    fileName = db_folder + "/iPod_Control/iTunes/iTunesDB";
     input_stream.open(fileName.c_str(), std::ios::in | std::ios::binary);
 
     if(input_stream.fail())
@@ -376,7 +376,7 @@
     }
     input_stream.close();
 
-    fileName = db_folder + "iPod_Control/iTunes/Play Counts";
+    fileName = db_folder + "/iPod_Control/iTunes/Play Counts";
     input_stream.open(fileName.c_str(), std::ios::in | std::ios::binary);
 
     if(input_stream.fail())
