--- hardinfo.desktop.orig	2008-06-24 13:27:34.000000000 +0300
+++ hardinfo.desktop	2008-06-24 13:37:53.000000000 +0300
@@ -2,8 +2,9 @@
 Encoding=UTF-8
-Name=System Profiler and Benchmark
+Name=Hardinfo
 Name[pt_BR]=Informações e Testes do Sistema
+GenericName[tr]=Sistem Bilgileri Değerlendiricisi
 Exec=hardinfo
-Icon=/usr/share/hardinfo/pixmaps/logo.png
+Icon=/usr/share/pixmaps/hardinfo.png
 Terminal=false
 Type=Application
 StartupNotify=true
