--- audacious.desktop.orig	2008-05-24 01:00:33.000000000 +0300
+++ audacious.desktop	2008-05-31 14:53:48.000000000 +0300
@@ -1,9 +1,11 @@
 [Desktop Entry]
 Name=Audacious
 GenericName=Audio Player
+GenericName[tr]=Ses Çalıcısı
 Comment=Play music
 Comment[hu]=Zene lejátszása
 Comment[de]=Musik abspielen
+Comment[tr]=Müzik çalar
 Exec=audacious %U
 Icon=audacious
 MimeType=application/x-ogg;audio/mp3;audio/mpeg;audio/mpegurl;audio/prs.sid;audio/x-flac;audio/x-it;audio/x-mod;audio/x-mp3;audio/x-mpeg;audio/x-mpegurl;audio/x-ms-wma;audio/x-musepack;audio/x-s3m;audio/x-scpls;audio/x-stm;audio/x-wav;audio/x-xm;application/ogg;audio/x-vorbis+ogg
