diff -p -up audacious-2.0-alpha1/src/audacious/playback.c~ audacious-2.0-alpha1/src/audacious/playback.c
--- audacious-2.0-alpha1/src/audacious/playback.c~	2009-04-16 23:58:42.000000000 +0200
+++ audacious-2.0-alpha1/src/audacious/playback.c	2009-04-22 16:25:54.000000000 +0200
@@ -285,7 +285,7 @@ run_no_output_plugin_dialog(void)
                                            GTK_DIALOG_DESTROY_WITH_PARENT,
                                            GTK_MESSAGE_ERROR,
                                            GTK_BUTTONS_OK,
-                                           _(markup));
+                                           "%s", _(markup));
     gtk_dialog_run(GTK_DIALOG(dialog));
     gtk_widget_destroy(dialog);
 }
