diff -Nuar k3guitune-1.0/k3guitune/k3guitune.desktop k3guitune-1.0.patched/k3guitune/k3guitune.desktop
--- k3guitune-1.0/k3guitune/k3guitune.desktop	2006-02-09 22:25:24.000000000 +0200
+++ k3guitune-1.0.patched/k3guitune/k3guitune.desktop	2007-05-08 17:07:05.000000000 +0300
@@ -3,10 +3,10 @@
 Encoding=UTF-8
 Type=Application
 Exec=k3guitune -caption "%c" %i %m  
-Icon=
+Icon=guitune_logo
 DocPath=k3guitune/index.html
-Comment=
-Comment[de]=
+Comment=Musical Instrument Tuning Application
+Comment[tr]=Müzik Aleti Akort Etme Uygulaması
 Terminal=0
 Name=K3GuiTune
 Name[de]=K3GuiTune
