diff -u ../uisp-20050207/src/Makefile.am src/Makefile.am
--- ../uisp-20050207/src/Makefile.am	2002-10-31 00:22:07.000000000 +0100
+++ src/Makefile.am	2007-03-12 14:30:56.000000000 +0100
@@ -31,7 +31,7 @@
 # Turn on all warnings and consider them errors. This kinda forces you to have
 # to use gcc, but I'd rather have all warnings eliminated.
 
-AM_CXXFLAGS          = -Wall -Werror
+AM_CXXFLAGS          = -Wall
 
 bin_PROGRAMS         = uisp
 uisp_SOURCES         = \
diff -u ../uisp-20050207/src/Makefile.in src/Makefile.in
--- ../uisp-20050207/src/Makefile.in	2005-02-07 22:48:13.000000000 +0100
+++ src/Makefile.in	2007-03-12 14:31:08.000000000 +0100
@@ -145,7 +145,7 @@
 
 # Turn on all warnings and consider them errors. This kinda forces you to have
 # to use gcc, but I'd rather have all warnings eliminated.
-AM_CXXFLAGS = -Wall -Werror
+AM_CXXFLAGS = -Wall
 
 bin_PROGRAMS = uisp
 uisp_SOURCES = \
