--- src/systeminfo.cpp.orig	2007-10-16 14:38:42.753513426 +0300
+++ src/systeminfo.cpp	2007-10-16 14:42:29.095104274 +0300
@@ -62,6 +62,7 @@
 		LinuxConectiva,
 		LinuxCaldera,
 		LinuxLFS,
+		LinuxPardus,
 
 		LinuxASP, // Russian Linux distros
 		LinuxALT,
@@ -91,6 +92,7 @@
 		{ LinuxAurox,		OsUseName,	"/etc/aurox-release",		"Aurox Linux"		},
 		{ LinuxArch,		OsUseFile,	"/etc/arch-release",		"Arch Linux"		},
 		{ LinuxLFS,		OsAppendFile,	"/etc/lfs-release",		"LFS Linux"		},
+		{ LinuxPardus,		OsUseFile,	"/etc/pardus-release",		"Pardus Linux"		},
 
 		// untested
 		{ LinuxSuSE,		OsUseFile,	"/etc/SuSE-release",		"SuSE Linux"		},
