diff -Nur psi-0.10/src/common.cpp psi-0.10.patched/src/common.cpp
--- psi-0.10/src/common.cpp	2006-01-07 23:37:22.000000000 +0200
+++ psi-0.10.patched/src/common.cpp	2007-08-16 17:47:47.000000000 +0300
@@ -1144,7 +1144,8 @@
 
 		LinuxPLD, // Polish Linux distros
 		LinuxAurox,
-		LinuxArch
+		LinuxArch,
+		LinuxPardus
 	};
 
 	enum OsFlags {
@@ -1166,6 +1167,7 @@
 		{ LinuxPLD,		OsUseFile,	"/etc/pld-release",		"PLD Linux"		},
 		{ LinuxAurox,		OsUseName,	"/etc/aurox-release",		"Aurox Linux"		},
 		{ LinuxArch,		OsUseFile,	"/etc/arch-release",		"Arch Linux"		},
+		{ LinuxPardus,		OsUseFile,	"/etc/pardus-release",		"Pardus Linux"		},
 		{ LinuxLFS,		OsAppendFile,	"/etc/lfs-release",		"LFS Linux"		},
 
 		// untested
