--- kdesvn.desktop	2006-05-11 02:16:21.000000000 +0300
+++ src/kdesvn.desktop	2007-10-22 05:31:53.000000000 +0300
@@ -1,9 +1,10 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=kdesvn
+Name=Kdesvn
 GenericName=SVN client
 GenericName[de]=Graphische Oberfläche für SVN
 GenericName[es]=Cliente SVN
+GenericName[tr]=SVN İstemcisi
 Exec=kdesvn %i %m -caption "%c"
 Icon=kdesvn
 Type=Application
@@ -12,6 +13,7 @@
 Comment[de]=Ein Subversionclient für KDE
 Comment[nl]=Een Subversion-client voor KDE
 Comment[es]=Un cliente de Subversion para KDE
+Comment[tr]=KDE için SVN İstemcisi
 Terminal=false
 MimeType=inode/directory
 ServiceTypes=KParts/ReadOnlyPart,Browser/View
