--- /usr/share/applications/bluefish.desktop	2008-09-19 03:27:29.000000000 +0300
+++ bluefish.desktop	2008-09-19 03:46:46.000000000 +0300
@@ -2,11 +2,15 @@
 Name=Bluefish Editor
 Name[de]=Bluefish Editor
 Name[fr]=Éditeur Bluefish
+Name[tr]=Bluefish Editörü
+GenericName=Create web pages
+GenericName[tr]=Web sayfası yaratmaya yarar
 Comment=Create web pages
 Comment[de]=Webseiten erstellen
 Comment[fr]=Éditeur pour développeurs web
+Comment[tr]=Web dizaynır ve programcıları için güçlü bir editör
 Exec=bluefish %F
-Icon=/usr/share/pixmaps/bluefish-icon.png
+Icon=/usr/share/pixmaps/bluefish.png
 Terminal=false
 Type=Application
 Categories=GTK;GNOME;Application;Development;WebDevelopment;
