--- qtpfsgui2.desktop	2007-04-02 17:29:16.000000000 +0300
+++ qtpfsgui.desktop	2007-05-30 22:28:21.000000000 +0300
@@ -2,15 +2,18 @@
 Encoding=UTF-8
 Name=Qtpfsgui
 Name[en_US]=Qtpfsgui
+Name[tr]=Qtpfsgui
 GenericName=HDR imaging
 GenericName[en_US]=HDR imaging
 GenericName[ru]=Cоздание HDR-снимков
 GenericName[it]=Workflow per immagini HDR
+GenericName[tr]=HDR Resim Düzenleyici
 Categories=Graphics;RasterGraphics;Photography;Viewer;Qt;
 Comment=Qtpfsgui is a graphical user interface used to create and tonemap HDR images
 Comment[en_US]=Qtpfsgui is a graphical user interface used to create and tonemap HDR images
 Comment[ru]=Qtpfsgui — программа для создания HDR-снимков и расчёта тонов
 Comment[it]=Qtpfsgui e' una interfaccia grafica usata per creare ed eseguire il tonemapping di immagini HDR
+Comment[tr]=Qtpsgui, HDR resim dosyaları oluşturmakta ve renk tonlamasında kullanılan, grafiksel bir kullanıcı arayüzüdür.
 Exec=qtpfsgui
 Icon=qtpfsgui.png
 MimeType=
