diff -Naur k9copy-2.1.0-Source.orig/k9copy_assistant.desktop k9copy-2.1.0-Source/k9copy_assistant.desktop
--- k9copy-2.1.0-Source.orig/k9copy_assistant.desktop	2009-01-17 08:09:23.000000000 +0200
+++ k9copy-2.1.0-Source/k9copy_assistant.desktop	2009-02-17 00:01:56.000000000 +0200
@@ -2,13 +2,14 @@
 Encoding=UTF-8
 Type=Application
 Name=k9copy assistant
+Name[tr]=k9copy sihirbazı
 Exec=k9copy --assistant -caption "%c" %i %m  %u
 Comment=DVD9 to DVD5 backup tool
 Comment[fr]=Backup de DVD9 en DVD5
 Comment[tr]=DVD9 DVD5 yedekleme aracı
 GenericName=Video DVD Backup
 GenericName[fr]=Backup de DVD Video
-GenericName[tr]=DVD Vidyo Yedekleme Aracı
+GenericName[tr]=DVD Video Yedekleyici
 MimeTypes=
 Terminal=false
 Icon=k9copy
diff -Naur k9copy-2.1.0-Source.orig/k9copy.desktop k9copy-2.1.0-Source/k9copy.desktop
--- k9copy-2.1.0-Source.orig/k9copy.desktop	2009-01-17 08:09:23.000000000 +0200
+++ k9copy-2.1.0-Source/k9copy.desktop	2009-02-16 23:58:46.000000000 +0200
@@ -8,7 +8,7 @@
 Comment[tr]=DVD9 DVD5 yedekleme aracı
 GenericName=Video DVD Backup
 GenericName[fr]=Backup de DVD Video
-GenericName[tr]=DVD Vidyo Yedekleme Aracı
+GenericName[tr]=DVD Video Yedekleyici
 MimeTypes=
 Terminal=false
 Icon=k9copy
