--- eric/Preferences/__init__.py.orig	2008-02-24 16:48:50.000000000 +0200
+++ eric/Preferences/__init__.py	2008-02-24 16:48:59.000000000 +0200
@@ -119,7 +119,7 @@
         },
         "ShowSplash" : 1,
         
-        "PerformVersionCheck" : 4,      # 0 = off
+        "PerformVersionCheck" : 0,      # 0 = off
                                         # 1 = at startup
                                         # 2 = daily
                                         # 3 = weekly
@@ -177,7 +177,7 @@
         
         "AutoPrepareAPIs" : 0,
         
-        "AutoCompletionEnabled" : 0,
+        "AutoCompletionEnabled" : 1,
         "AutoCompletionCaseSensitivity" : 1,
         "AutoCompletionReplaceWord" : 0,
         "AutoCompletionShowSingle" : 0,
@@ -429,12 +429,12 @@
     
     # defaults for the help settings
     helpDefaults = {
-        "HelpViewerType" : 1,      # this coresponds with the radio button id
+        "HelpViewerType" : 2,      # this coresponds with the radio button id
         "CustomViewer" : "",
-        "PythonDocDir" : "",
+        "PythonDocDir" : "/usr/share/doc/python-docs/2.4.4/html/",
         "QtDocDir" : "",
-        "Qt4DocDir" : "",
-        "PyQt4DocDir" : "",
+        "Qt4DocDir" : "/usr/qt/4/doc/html/",
+        "PyQt4DocDir" : "/usr/qt/4/doc/PyQt4/",
         "SingleHelpWindow" : 1,
         "SaveGeometry" : 1,
     }
@@ -465,7 +465,7 @@
         "QtToolsPrefix3" : "", 
         "QtToolsPostfix3" : "", 
         "QtToolsPrefix4" : "", 
-        "QtToolsPostfix4" : "", 
+        "QtToolsPostfix4" : "-qt4", 
     }
     
     # defaults for corba related stuff
