diff -up eric4-4.1.0/eric/uninstall.py.patch eric4-4.1.0/eric/uninstall.py
--- eric4-4.1.0/eric/uninstall.py.patch	2008-02-04 20:59:23.000000000 +0100
+++ eric4-4.1.0/eric/uninstall.py	2008-02-04 20:59:31.000000000 +0100
@@ -1,3 +1,4 @@
+#!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
 # Copyright (c) 2002-2008 Detlev Offenbach <detlev@die-offenbachs.de>
diff -up eric4-4.1.0/eric/patch_modpython.py.patch eric4-4.1.0/eric/patch_modpython.py
--- eric4-4.1.0/eric/patch_modpython.py.patch	2008-02-04 20:59:04.000000000 +0100
+++ eric4-4.1.0/eric/patch_modpython.py	2008-02-04 20:59:16.000000000 +0100
@@ -1,3 +1,4 @@
+#!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
 # Copyright (c) 2003-2008 Detlev Offenbach <detlev@die-offenbachs.de>
diff -up eric4-4.1.0/eric/__init__.py.patch eric4-4.1.0/eric/__init__.py
--- eric4-4.1.0/eric/__init__.py.patch	2008-02-04 20:58:41.000000000 +0100
+++ eric4-4.1.0/eric/__init__.py	2008-02-04 20:58:49.000000000 +0100
@@ -1,3 +1,4 @@
+#!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
 # Copyright (c) 2003 - 2008 Detlev Offenbach <detlev@die-offenbachs.de>
