diff -Naur arora-0.4.orig/src/arora.desktop arora-0.4/src/arora.desktop
--- arora-0.4.orig/src/arora.desktop	2008-10-01 04:17:22.000000000 +0300
+++ arora-0.4/src/arora.desktop	2008-10-08 21:48:23.000000000 +0300
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Exec=arora %U
 GenericName=Web Browser
+GenericName[tr]=Web Tarayıcı
 GenericName[en_US]=Web Browser
 GenericName[cs_CZ]=Webový prohlížeč
 GenericName[nl]=Webbrowser
@@ -8,6 +9,7 @@
 MimeType=text/html;
 Name=Arora
 Name[en_US]=Arora
+Name[tr]=Arora
 StartupNotify=true
 Terminal=false
 Type=Application
diff -Naur arora-0.4.orig/src/arora.desktop.rej arora-0.4/src/arora.desktop.rej
--- arora-0.4.orig/src/arora.desktop.rej	1970-01-01 02:00:00.000000000 +0200
+++ arora-0.4/src/arora.desktop.rej	2008-10-08 20:41:05.000000000 +0300
@@ -0,0 +1,32 @@
+***************
+*** 1,6 ****
+  [Desktop Entry]
+  Exec=arora %U
+  GenericName=Web Browser
+  GenericName[en_US]=Web Browser
+  GenericName[cs_CZ]=Webový prohlížeč
+  GenericName[nl]=Webbrowser
+--- 1,7 ----
+  [Desktop Entry]
+  Exec=arora %U
+  GenericName=Web Browser
++ GenericName[tr]=Web Tarayıcı
+  GenericName[en_US]=Web Browser
+  GenericName[cs_CZ]=Webový prohlížeč
+  GenericName[nl]=Webbrowser
+***************
+*** 8,13 ****
+  MimeType=text/html;
+  Name=Arora
+  Name[en_US]=Arora
+  StartupNotify=true
+  Terminal=false
+  Type=Application
+--- 9,15 ----
+  MimeType=text/html;
+  Name=Arora
+  Name[en_US]=Arora
++ Name[tr]=Arora
+  StartupNotify=true
+  Terminal=false
+  Type=Application
