diff -Naur arora-0.4.orig/src/browserapplication.cpp arora-0.4/src/browserapplication.cpp
--- arora-0.4.orig/src/browserapplication.cpp	2008-10-01 04:17:22.000000000 +0300
+++ arora-0.4/src/browserapplication.cpp	2008-10-11 20:13:33.000000000 +0300
@@ -131,11 +131,6 @@
     m_lastSession = settings.value(QLatin1String("lastSession")).toByteArray();
     settings.endGroup();

-    settings.beginGroup(QLatin1String("LanguageManager"));
-    if (settings.contains(QLatin1String("language")))
-        languageManager()->setCurrentLanguage(settings.value(QLatin1String("language")).toString());
-    settings.endGroup();
-
 #if defined(Q_WS_MAC)
     connect(this, SIGNAL(lastWindowClosed()),
             this, SLOT(lastWindowClosed()));
@@ -144,6 +139,7 @@
 #ifndef AUTOTESTS
     QTimer::singleShot(0, this, SLOT(postLaunch()));
 #endif
+    LanguageManager();
 }

 BrowserApplication::~BrowserApplication()
--- arora-0.4.orig/src/languagemanager.cpp  2008-10-01 04:17:22.000000000 +0300
+++ arora-0.4/src/languagemanager.cpp   2008-10-17 17:17:06.000000000 +0300
@@ -48,6 +48,13 @@
     , m_appTranslator(0)
     , m_loaded(false)
 {
+    QSettings settings;
+    settings.beginGroup(QLatin1String("LanguageManager"));
+    if (settings.contains(QLatin1String("language"))) {
+        setCurrentLanguage(settings.value(QLatin1String("language")).toString());
+    } else if (!currentLanguage().isEmpty()) {
+        setCurrentLanguage(currentLanguage());
+    }
 }

 QString LanguageManager::currentLanguage() const
@@ -126,26 +133,24 @@
     loadAvailableLanguages();

     QStringList items;
-    QString systemLocaleString = QLocale::system().name();
-    systemLocaleString = tr("System locale (%1) %2")
-        .arg(systemLocaleString)
-        // this is for pretty RTL support
-        .arg(QChar(0x200E)); // LRM = 0x200E;
-    items << systemLocaleString;
-
-    int defaultItem = 0;
-    foreach (QString name, m_languages) {
+    int defaultItem = -1;
+    QString current = currentLanguage();
+    foreach (const QString name, m_languages) {
         QLocale locale(name);
-        if (name == m_currentLanguage)
-            defaultItem = items.count();
         QString string = QString(QLatin1String("%1, %2 (%3) %4"))
             .arg(QLocale::languageToString(locale.language()))
             .arg(QLocale::countryToString(locale.country()))
             .arg(name)
             // this is for pretty RTL support
             .arg(QChar(0x200E)); // LRM = 0x200E
+        if (name == current)
+            defaultItem = items.count();
         items << string;
     }
+    if (defaultItem == -1) {
+        defaultItem = items.count();
+        items << tr("Default");
+    }

     bool ok;
     QString item = QInputDialog::getItem(0,
@@ -158,11 +163,8 @@
         return;

     int selection = items.indexOf(item);
-    if (selection == 0) {
-        setCurrentLanguage(QString());
-    } else {
-        setCurrentLanguage(m_languages.value(selection - 1));
-    }
+    if (selection != m_languages.count())
+        setCurrentLanguage(m_languages.value(selection));
 }

 QString LanguageManager::translationLocation() const
