--- rawstudio.desktop.orig	2008-09-25 00:48:39.000000000 +0300
+++ rawstudio.desktop	2008-09-25 00:50:32.000000000 +0300
@@ -9,6 +9,7 @@
 Name[pl]=Rawstudio
 Name[ru]=Rawstudio
 Name[sv]=Rawstudio
+Name[tr]=Rawstudio
 GenericName=Raw image converter
 GenericName[de]=Ein RAW Bildkonverter
 GenericName[es]=Un convertidor de imágenes raw
@@ -16,6 +17,7 @@
 GenericName[pl]=Konwerter plików RAW
 GenericName[ru]=Проявка цифровых негативов
 GenericName[sv]=Bildkonverterare
+GenericName[tr]=Rawstudio Resim Dönüştürücü
 Comment=Read, manipulate and convert digital camera raw-images
 Comment[de]=Lesen, manipulieren und konvertieren die Bilder der Digitalkamera
 Comment[es]=Leído, manipule y convierta las raw imágenes digitales de la cámara fotográfica
@@ -23,6 +25,8 @@
 Comment[pl]=Program Open Source do konwersji plików RAW i manipulacji na nich
 Comment[ru]=Свободная программа для проявки цифровых негативов
 Comment[sv]=Läs, bearbeta och konvertera råbilder från digitalkameror
+Comment[tr]=Raw dosyalarını okur, işler ve dönüştürür
+Encoding=UTF-8
 Exec=rawstudio %F
 StartupNotify=true
 Icon=rawstudio
