--- src/vfs/vfs-file-task.c.orig	2008-06-28 12:45:22.000000000 +0200
+++ src/vfs/vfs-file-task.c	2009-03-06 22:50:02.000000000 +0100
@@ -456,7 +456,7 @@
         {
             goto on_error;
         }
-        g_debug( dest_file );
+        g_debug( "%s", dest_file );
     }
     else
         dest_file = g_build_filename( task->dest_dir, file_name, NULL );
--- src/vfs/vfs-volume-hal.c.orig	2008-06-11 20:38:25.000000000 +0200
+++ src/vfs/vfs-volume-hal.c	2009-03-06 22:55:05.000000000 +0100
@@ -2195,7 +2195,7 @@
   if (G_UNLIKELY (message == NULL))
     {
       /* out of memory */
-oom:  g_set_error (error, G_FILE_ERROR, G_FILE_ERROR_NOMEM, g_strerror (ENOMEM));
+oom:  g_set_error ("%s", error, G_FILE_ERROR, G_FILE_ERROR_NOMEM, g_strerror (ENOMEM));
       vfs_volume_hal_free( device );
       return FALSE;
     }
@@ -2522,7 +2522,7 @@
       message = dbus_message_new_method_call ("org.freedesktop.Hal", device->udi, "org.freedesktop.Hal.Device.Volume", "Mount");
       if (G_UNLIKELY (message == NULL))
         {
-oom:      g_set_error (error, G_FILE_ERROR, G_FILE_ERROR_NOMEM, g_strerror (ENOMEM));
+oom:      g_set_error ("%s", error, G_FILE_ERROR, G_FILE_ERROR_NOMEM, g_strerror (ENOMEM));
           g_strfreev (options);
           g_free (mount_point);
           g_free (fstype);
@@ -2757,7 +2757,7 @@
   if (G_UNLIKELY (message == NULL))
     {
       /* out of memory */
-oom:  g_set_error (error, G_FILE_ERROR, G_FILE_ERROR_NOMEM, g_strerror (ENOMEM));
+oom:  g_set_error ("%s", error, G_FILE_ERROR, G_FILE_ERROR_NOMEM, g_strerror (ENOMEM));
     vfs_volume_hal_free( device );
       return FALSE;
     }
--- src/ptk/ptk-utils.c.orig	2008-07-14 11:43:55.000000000 +0200
+++ src/ptk/ptk-utils.c	2009-03-06 22:58:40.000000000 +0100
@@ -190,7 +190,7 @@
 
 void ptk_show_error(GtkWindow* parent, const char* title, const char* message )
 {
-  GtkWidget* dlg = gtk_message_dialog_new_with_markup(parent, GTK_DIALOG_MODAL, GTK_MESSAGE_ERROR, GTK_BUTTONS_OK, message);
+  GtkWidget* dlg = gtk_message_dialog_new_with_markup(parent, GTK_DIALOG_MODAL, GTK_MESSAGE_ERROR, GTK_BUTTONS_OK, "%s", message);
   if( title )
     gtk_window_set_title( (GtkWindow*)dlg, title );
   gtk_dialog_run( GTK_DIALOG(dlg) );