--- synfigstudio.desktop.in_orj	2007-10-10 04:03:29.000000000 +0300
+++ synfigstudio.desktop.in	2007-11-28 21:21:41.000000000 +0200
@@ -1,10 +1,13 @@
 [Desktop Entry]
-Name=@PACKAGE_NAME@
-Comment=Create and edit 2D animations and compositions
-Exec=synfigstudio %F
-Terminal=false
 Type=Application
+Encoding=UTF-8
+Name=Synfig Studio
+GenericName=2D Animation Editor
+GenericName[tr]=2D Animasyon Editörü
+Comment=Create and edit 2D animations and compositions
+Comment[tr]=2D Animastonlar oluşturur ve düzenler
 Icon=synfig_icon.png
-Categories=Graphics;2DGraphics;VectorGraphics;
+Exec=synfigstudio %F
 StartupNotify=true
 MimeType=image/sif
+Categories=Graphics;2DGraphics;VectorGraphics;
