--- src/kdesvn.desktop.orig	2008-09-24 11:40:22.000000000 +0300
+++ src/kdesvn.desktop	2008-09-24 11:41:52.000000000 +0300
@@ -2,9 +2,11 @@
 Encoding=UTF-8
 Name=kdesvn
 GenericName=SVN client
+Name[tr]=Kdesvnd Subversion Modülü
 GenericName[de]=Graphische Oberfläche für SVN
 GenericName[es]=Cliente SVN
 GenericName[lt]=SVN klientas
+GenericName[tr]=SVN İstemcisi
 Exec=kdesvn %i %m -caption "%c"
 Icon=kdesvn
 Type=Application
@@ -14,6 +16,7 @@
 Comment[nl]=Een Subversion-client voor KDE
 Comment[es]=Un cliente de Subversion para KDE
 Comment[lt]=Subversion klientas skirtas KDE
+Comment[tr]=KDE için SVN İstemcisi
 Terminal=false
 MimeType=inode/directory;
 ServiceTypes=KParts/ReadOnlyPart,Browser/View
