diff -Nur ksquirrel-libs-0.7.0/kernel/kls_camera/dcraw.c ksquirrel-libs-0.7.0.patched/kernel/kls_camera/dcraw.c
--- ksquirrel-libs-0.7.0/kernel/kls_camera/dcraw.c	2007-04-09 19:41:44.000000000 +0300
+++ ksquirrel-libs-0.7.0.patched/kernel/kls_camera/dcraw.c	2007-06-03 10:47:45.000000000 +0300
@@ -52,7 +52,7 @@
 #ifndef inline
 #define inline
 #endif
-#include <lcms.h>
+#include <lcms/lcms.h>
 #endif
 #ifdef LOCALEDIR
 #include <libintl.h>
