--- fontmatrix.desktop.orig	2010-01-09 13:39:20.046970373 +0200
+++ fontmatrix.desktop	2010-01-09 13:38:46.004221026 +0200
@@ -4,6 +4,7 @@
 Type=Application
 Name=Fontmatrix
 Comment=A font manager
+Comment[tr]=Bir font yöneticisi
 Comment[ru]=Программа для управления шрифтами
 Comment[fi]=Työkalu fonttien hallintaan
 Categories=Graphics;VectorGraphics;
@@ -11,6 +12,7 @@
 Icon=fontmatrix
 Terminal=false
 GenericName=Font Manager
+GenericName[tr]=Font Yönetici
 GenericName[fr]=Gestion des Polices
 GenericName[ru]=Управление шрифтами
 GenericName[fi]=Fonttien hallinta
