diff -Nur gnome-cups-manager-0.31/po/POTFILES.in gnome-cups-manager-0.31.new/po/POTFILES.in
--- gnome-cups-manager-0.31/po/POTFILES.in	2005-03-07 17:39:10.000000000 +0100
+++ gnome-cups-manager-0.31.new/po/POTFILES.in	2006-05-19 00:07:49.000000000 +0200
@@ -1,5 +1,7 @@
 # List of source files containing translatable strings.
 # Please keep this file sorted alphabetically.
+debian/gnome-cups-icon.desktop.in
+debian/gnome-cups-manager.desktop.in
 gnome-cups-add/add-printer.c
 gnome-cups-add/gnome-cups-add.glade
 gnome-cups-manager/eggtrayicon.c
