--- data/streamtuner.desktop.in	2004-07-23 21:04:08.000000000 +0300
+++ streamtuner-yeni.desktop.in	2006-05-13 04:46:31.000000000 +0300
@@ -2,8 +2,10 @@
 Encoding=UTF-8
 Type=Application
 Name=streamtuner
-_GenericName=Stream Directory Browser
-_Comment=Browse Internet stream directories
+GenericName=Stream Directory Browser
+GenericName[tr]=İnternet Radyo tarayıcısı
+Comment=Browse Internet stream directories
+Comment[tr]=İnternet Radyo tarayıcısı
 Icon=streamtuner.png
 TryExec=streamtuner
 Exec=streamtuner
