--- src/plugins/Input/mad/decoder_mad.cpp.uint	2009-05-04 08:10:32.000000000 +0200
+++ src/plugins/Input/mad/decoder_mad.cpp	2009-05-04 08:11:26.000000000 +0200
@@ -265,7 +265,7 @@
             {
                 //skip ID3v2 tag
                 uint tagSize = findID3v2((uchar *)stream.this_frame,
-                                         (uint) stream.bufend - (uint) stream.this_frame);
+                                         (ulong) stream.bufend - (ulong) stream.this_frame);
                 if (tagSize > 0)
                 {
                     mad_stream_skip(&stream, tagSize);
