diff -Naur pdfedit-0.2.5-orig/doc/Makefile pdfedit-0.2.5/doc/Makefile
--- pdfedit-0.2.5-orig/doc/Makefile	2007-04-07 23:35:57.000000000 +0300
+++ pdfedit-0.2.5/doc/Makefile	2007-04-07 23:54:18.000000000 +0300
@@ -4,7 +4,6 @@
 
 #Build all documentation, including doxygen programmer documentation and doxygen's default stylesheet
 all:	doc_core
-	doxygen -w html /dev/null /dev/null programmer/html/doxygen.css || true
 	doxygen || true
 
 #Core documentation, without PDF
