--- mypaint.desktop.orig	2007-07-31 12:42:03.000000000 +0300
+++ desktop/mypaint.desktop	2007-07-31 12:44:04.000000000 +0300
@@ -1,11 +1,14 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=MyPaint
-Exec=mypaint %U
+Name[tr]=MyPaint
+Exec=mypaint
 Comment=Paint images
+Comment[tr]=Resim Boyama
 GenericName=Painting Images
+GenericName[tr]=Resim Boyama
 MimeType=application/x-mypaint
 Type=Application
-Icon=mypaint_48
+Icon=mypaint
 Categories=Graphics;
 
