diff -Naur ksniffer-0.2-orig/ksniffer/ksniffer.desktop ksniffer-0.2/ksniffer/ksniffer.desktop
--- ksniffer-0.2/ksniffer/ksniffer.desktop	2007-08-10 03:41:24.000000000 +0300
+++ ksniffer-0.2/ksniffer/ksniffer.desktop	2007-08-10 03:42:12.000000000 +0300
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Name=KSniffer
 Name[sv]=Ksniffer
-Name[tr]=K Koklayıcı
+Name[tr]=KSniffer
 Name[xx]=xxKSnifferxx
 GenericName=Network Analyzer
 GenericName[ar]=محلل الشبكة
