--- share/hedgewars/Data/Locale/hedgewars_tr_TR.ts	2009-04-13 19:00:56.000000000 +0300
+++ share/hedgewars/Data/Locale/hedgewars_tr_TR.ts	2009-04-16 13:11:10.000000000 +0300
@@ -1,17 +1,18 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!DOCTYPE TS><TS version="1.1" language="tr_TR">
+<defaultcodec></defaultcodec>
 <context>
     <name>FreqSpinBox</name>
     <message>
         <location filename="../../../../QTfrontend/misc.h" line="38"/>
         <source>Never</source>
-        <translation type="unfinished"></translation>
+        <translation>Asla</translation>
     </message>
     <message numerus="yes">
         <location filename="../../../../QTfrontend/misc.h" line="40"/>
         <source>Every %1 turn</source>
-        <translation type="unfinished">
-            <numerusform></numerusform>
+        <translation>
+            <numerusform>%1 sırada bir</numerusform>
             <numerusform></numerusform>
         </translation>
     </message>
@@ -222,12 +223,12 @@
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1063"/>
         <source>Server message:</source>
-        <translation type="unfinished"></translation>
+        <translation>Sunucu iletisi:</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1069"/>
         <source>Set message</source>
-        <translation type="unfinished"></translation>
+        <translation>İletiyi ayarla</translation>
     </message>
 </context>
 <context>
@@ -235,7 +236,7 @@
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="848"/>
         <source>Connecting...</source>
-        <translation type="unfinished"></translation>
+        <translation>Bağlanıyor...</translation>
     </message>
 </context>
 <context>
@@ -322,12 +323,12 @@
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1097"/>
         <source>LAN game</source>
-        <translation type="unfinished"></translation>
+        <translation>LAN oyunu</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1098"/>
         <source>Official server</source>
-        <translation type="unfinished"></translation>
+        <translation>Resmi sunucu</translation>
     </message>
 </context>
 <context>
@@ -454,7 +455,7 @@
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1009"/>
         <source>Delete</source>
-        <translation type="unfinished">Sil</translation>
+        <translation>Sil</translation>
     </message>
 </context>
 <context>
@@ -682,12 +683,12 @@
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="866"/>
         <source>Game Modifiers</source>
-        <translation type="unfinished"></translation>
+        <translation>Oyun Değişkenleri</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="867"/>
         <source>Basic Settings</source>
-        <translation type="unfinished"></translation>
+        <translation>Basit Ayarlar</translation>
     </message>
 </context>
 <context>
@@ -785,37 +786,37 @@
     <message>
         <location filename="../../../../QTfrontend/gamecfgwidget.cpp" line="53"/>
         <source>Game scheme</source>
-        <translation type="unfinished"></translation>
+        <translation>Oyun düzeni</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="923"/>
         <source>Damage Modifier</source>
-        <translation type="unfinished"></translation>
+        <translation>Hasar Değişkeni</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="938"/>
         <source>Turn Time</source>
-        <translation type="unfinished"></translation>
+        <translation>Tur Süresi</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="953"/>
         <source>Initial Health</source>
-        <translation type="unfinished"></translation>
+        <translation>Başlangıç Sağlığı</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="968"/>
         <source>Sudden Death Timeout</source>
-        <translation type="unfinished"></translation>
+        <translation>Ani Ölüm Zamanı</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="983"/>
         <source>Crate Drops</source>
-        <translation type="unfinished"></translation>
+        <translation>Sandık Düşüşü</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="998"/>
         <source>Scheme Name:</source>
-        <translation type="unfinished"></translation>
+        <translation>Düzen Adı:</translation>
     </message>
 </context>
 <context>
@@ -1046,42 +1047,42 @@
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="888"/>
         <source>Fort Mode</source>
-        <translation type="unfinished"></translation>
+        <translation>Kaleler Modu</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="892"/>
         <source>Divide Teams</source>
-        <translation type="unfinished"></translation>
+        <translation>Takımları Böl</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="896"/>
         <source>Solid Land</source>
-        <translation type="unfinished"></translation>
+        <translation>Parçalanmaz Yüzey</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="900"/>
         <source>Add Border</source>
-        <translation type="unfinished">Sınır Ekle</translation>
+        <translation>Sınır Ekle</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="904"/>
         <source>Low Gravity</source>
-        <translation type="unfinished">Azaltılmış Yerçekimi</translation>
+        <translation>Azaltılmış Yerçekimi</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="908"/>
         <source>Laser Sight</source>
-        <translation type="unfinished">Lazer Görüşü</translation>
+        <translation>Lazer Görüşü</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="912"/>
         <source>Invulnerable</source>
-        <translation type="unfinished">Ölümsüzlük</translation>
+        <translation>Ölümsüzlük</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="916"/>
         <source>Add Mines</source>
-        <translation type="unfinished">Mayın Ekle</translation>
+        <translation>Mayın Ekle</translation>
     </message>
 </context>
 <context>
