--- lxappearance.desktop.orig	2009-11-13 12:41:30.626302540 +0200
+++ lxappearance.desktop	2009-11-13 12:42:25.238555820 +0200
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Appearance
+Name[tr]=Görünüm
 Name[ar]=المظهر
 Name[be@latin]=Vyhlad
 Name[bg]=Външен вид
@@ -46,6 +47,7 @@
 Name[zh_CN]=外观
 Name[zh_TW]=外觀設定
 Comment=Customize the look of the desktop
+Comment[tr]=Masaüstü görünümünü değiştirir
 Comment[ar]=خصّص مظهر سطح المكتب
 Comment[be@latin]=Źmiani vyhlad stała
 Comment[bg]=Настройване на външния вид на работния плот
