diff -Nuar murrine-0.90.3.orig/Makefile.in murrine-0.90.3/Makefile.in
--- murrine-0.90.3.orig/Makefile.in	2009-06-11 02:53:51.000000000 +0300
+++ murrine-0.90.3/Makefile.in	2009-06-11 02:56:34.000000000 +0300
@@ -267,7 +267,7 @@
 	./src/murrine_draw.h \
 	./src/murrine_types.h
 
-libmurrine_la_LDFLAGS = -module -avoid-version -no-undefined
+libmurrine_la_LDFLAGS = -module -avoid-version -no-undefined -lm
 libmurrine_la_LIBADD = $(GTK_LIBS)
 all: all-recursive
 
