diff -Naur qdvdauthor-1.7.0.orig/configure qdvdauthor-1.7.0/configure
--- qdvdauthor-1.7.0.orig/configure	2009-04-13 22:47:46.000000000 +0300
+++ qdvdauthor-1.7.0/configure	2009-04-13 23:20:33.000000000 +0300
@@ -521,10 +521,10 @@
 	then
 		check_vlclibs;
 	fi;
-	if [ $WITH_MPLAYER_SUPPORT -eq 1 ];
-	then
-		check_mplayerlibs;
-	fi;
+# 	if [ $WITH_MPLAYER_SUPPORT -eq 1 ];
+# 	then
+# 		check_mplayerlibs;
+# 	fi;
 
 	# next step is we should start compilation ...
 	echo "configured with XINE_SUPPORT=$WITH_XINE_SUPPORT  MPLAYER_SUPPORT=$WITH_MPLAYER_SUPPORT VLC_SUPPORT=$WITH_VLC_SUPPORT QTDIR3=$QTDIR3 QTDIR4=$QTDIR4 QT_LIB=$QT_LIB"
