--- firefoxqs.desktop.orig	2007-09-02 00:14:06.000000000 +0300
+++ src/firefoxqs.desktop	2007-11-19 18:25:35.000000000 +0200
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 Name=Firefox Quickstarter
 Exec=firefoxqs
-Icon=firefox
+Icon=firefoxqs
 Type=Application
 Comment=Firefox Quickstarter
 Comment[br]=Ur meziant eeun evit KDE
@@ -34,5 +34,6 @@
 Comment[tg]=Гузориши оддиKDE 
 Comment[tr]=Basit bir KDE Uygulaması
 Comment[zh_CN]=一个简单的KDE应用程序
+GenericName[tr]=Firefox Hızlı Başlatıcı
 X-KDE-autostart-condition=firefoxqsrc:General:Autostart:true
 X-KDE-autostart-after=panel
\ Dosya sonunda yenisatır yok.
