diff -Nuar wicd-1.6.2.orig/curses/wicd-curses.py wicd-1.6.2/curses/wicd-curses.py
--- wicd-1.6.2.orig/curses/wicd-curses.py	2009-07-19 01:34:04.000000000 +0300
+++ wicd-1.6.2/curses/wicd-curses.py	2009-07-19 13:33:27.000000000 +0300
@@ -1037,6 +1037,23 @@
 ##### MAIN ENTRY POINT
 ########################################
 if __name__ == '__main__':
+    import wicd.startservice as pardus
+
+    if pardus.infoService("wicd") == "stopped":
+        try:
+            pardus.startService("wicd")
+
+        except:
+            from os import getenv
+
+            if getenv("LANG").startswith("tr"):
+                print("\nWicd servisi şuan etkin değil! Servis Yöneticisi'nden\nWicd ağ yöneticisini çalıştırın ve tekrar deneyin.")
+
+            else:
+                print("\nService Wicd is not started! Please run Wicd connection\nmanager from Service Manager and try again.")
+
+            sys.exit(1)
+
     try:
         parser = OptionParser(version="wicd-curses-%s (using wicd %s)" % (CURSES_REV,daemon.Hello()))
     except Exception, e:
diff -Nuar wicd-1.6.2.orig/wicd/wicd-client.py wicd-1.6.2/wicd/wicd-client.py
--- wicd-1.6.2.orig/wicd/wicd-client.py	2009-07-19 01:34:04.000000000 +0300
+++ wicd-1.6.2/wicd/wicd-client.py	2009-07-19 13:14:42.000000000 +0300
@@ -824,4 +824,15 @@
 
 
 if __name__ == '__main__':
-    main(sys.argv)
+    import wicd.startservice as pardus
+
+    if pardus.infoService("wicd") == "stopped":
+        try:
+            pardus.startService("wicd")
+            main(sys.argv)
+
+        except:
+            pardus.warningWidget()
+
+    else:
+        main(sys.argv)
