--- a/metadata/decoration.xml.in.orig	2007-10-24 01:55:25.000000000 +0100
+++ b/metadata/decoration.xml.in	2007-10-24 01:56:45.000000000 +0100
@@ -59,7 +59,7 @@
 	    <option name="decoration_match" type="match">
 		<_short>Decoration windows</_short>
 		<_long>Windows that should be decorated</_long>
-		<default>any</default>
+		<default>any &amp; !name=kdesktop_lock</default>
 	    </option>
 	    <option name="shadow_match" type="match">
 		<_short>Shadow windows</_short>
