diff -Naur kmess-1.5pre1-orig/kmess/account.cpp kmess-1.5pre1/kmess/account.cpp
--- kmess-1.5pre1-orig/kmess/account.cpp	2006-10-12 19:55:47.000000000 +0300
+++ kmess-1.5pre1/kmess/account.cpp	2006-10-12 21:18:42.000000000 +0300
@@ -44,7 +44,7 @@
    customImageIndex_(0),
    deleted_(false),
    dirty_(false),
-   emailCommand_("kmail"),
+   emailCommand_("firefox www.hotmail.com"),
    emailSupported_(true),  // Assume true (for new accounts), until the msn server proves the opposite
    emoticonStyle_("KMess-new"),
    fontColor_("#000000"),
@@ -70,7 +70,7 @@
    useContactFont_(false),
    useEmoticons_(true),
    useFontEffects_(true),
-   useHotmail_(true),
+   useHotmail_(false),
    useIdleTimer_(true),
    useNotifier_(true),
 
