diff -Nuar glade3-3.2.0.orig/help/Makefile.in glade3-3.2.0/help/Makefile.in
--- glade3-3.2.0.orig/help/Makefile.in	2007-03-30 16:54:16.000000000 +0300
+++ glade3-3.2.0/help/Makefile.in	2007-03-30 16:54:33.000000000 +0300
@@ -880,7 +880,6 @@
 	done
 	@if test "x$(_ENABLE_SK)" = "xtrue"; then \
 	  echo "scrollkeeper-update -p $(DESTDIR)$(_sklocalstatedir) -o $(DESTDIR)$(OMF_DIR)/$(DOC_MODULE)"; \
-	  scrollkeeper-update -p "$(DESTDIR)$(_sklocalstatedir)" -o "$(DESTDIR)$(OMF_DIR)/$(DOC_MODULE)"; \
 	fi;
 
 install-doc-dsk:
