diff -Nuar playonlinux.orig/lib/check_depend playonlinux/lib/check_depend
--- playonlinux/lib/check_depend	2008-06-30 19:15:20.000000000 +0300
+++ playonlinux/lib/check_depend	2008-07-21 09:05:59.000000000 +0300
@@ -52,7 +52,7 @@
	check_one wget "wget"
 	check_one wine "wine"
 	check_one unzip "unzip"
 	check_one ar "binutils" 
-	check_one xterm "xterm"
+	check_one konsole "konsole"
 	check_one convert "imagemagick"
 	check_one cabextract "cabextract" 
 	check_one lzma "lzma"
 
