diff -Naur compiz.orig/metadata/Makefile.am compiz/metadata/Makefile.am
--- compiz.orig/metadata/Makefile.am	2007-09-04 15:46:57.000000000 +0300
+++ compiz/metadata/Makefile.am	2007-09-04 15:49:18.000000000 +0300
@@ -59,17 +59,6 @@
 	$(kde_kcfg_files) \
 	$(kde_config_files)
 
-if USE_KCONFIG
-	if mkdir -p $(kde_kcfgdir) && test -w $(kde_kcfgdir); then ( \
-		$(INSTALL) -m 644 $(kde_kcfg_files) $(kde_kcfgdir) \
-		); \
-	fi
-	if mkdir -p $(kde_configdir) && test -w $(kde_configdir); then ( \
-		$(INSTALL) -m 644 $(kde_config_files) $(kde_configdir) \
-		); \
-	fi
-endif
-
 uninstall-local:
 if USE_KCONFIG
 	if test -w $(kde_kcfgdir); then ( \
