diff -Naur geda-gattrib-1.4.0_old/data/geda-gattrib.desktop geda-gattrib-1.4.0_new/data/geda-gattrib.desktop
--- data/geda-gattrib.desktop	2008-01-27 18:58:14.000000000 +0200
+++ data/geda-gattrib.desktop	2008-01-28 00:42:37.000000000 +0200
@@ -6,12 +6,14 @@
 Name[de_DE]=gEDA Attributeditor
 Name[es_ES]=Editor de Propiedades de gEDA
 Name[nl_NL]=gEDA Attribuut Bewerking
+Name[tr]=gEDA Özellik Editörü
 Comment=Manipulate component attributes with gattrib
 Comment[de_DE]=Editieren von Bauteilattributen mit gattrib
 Comment[es_ES]=Manejar propiedades de componentes con gattrib
 Comment[nl_NL]=Manipuleer componenten attributen met gattrib
+Comment[tr]=Bileşenlerin özelliklerini değiştirir
 Type=Application
 Exec=gattrib %F
 Icon=geda-gattrib
 #MimeType=application/x-geda-schematic;application/x-geda-symbol;
-Categories=Engineering;Electronics
+Categories=Application;Education;Science;
diff -Naur geda-gattrib-1.4.0_old/data/geda-gattrib.desktop.in geda-gattrib-1.4.0_new/data/geda-gattrib.desktop.in
--- data/geda-gattrib.desktop.in	2008-01-04 19:58:05.000000000 +0200
+++ data/geda-gattrib.desktop.in	2008-01-28 00:48:28.000000000 +0200
@@ -8,4 +8,4 @@
 Exec=gattrib %F
 Icon=geda-gattrib
 #MimeType=application/x-geda-schematic;application/x-geda-symbol;
-Categories=Engineering;Electronics
+Categories=Application;Education;Science;
