--- azureus	2006-01-03 18:11:54.000000000 +0200
+++ azureus.orig	2006-03-13 22:11:03.000000000 +0200
@@ -1,8 +1,8 @@
 #!/bin/bash
 
 ######## CONFIGURATION OPTIONS ########
-JAVA_PROGRAM_DIR=""				# use full path to java bin dir, ex. "/usr/java/j2sdk1.4.2/bin/"
-#PROGRAM_DIR="/home/username/apps/azureus"	# use full path to Azureus bin dir
+JAVA_PROGRAM_DIR="/opt/sun-jre/bin/"				# use full path to java bin dir, ex. "/usr/java/j2sdk1.4.2/bin/"
+PROGRAM_DIR="/usr/share/azureus"                        	# use full path to Azureus bin dir
 #######################################
 
 
