diff -Naur linden.orig/indra/SConstruct linden/indra/SConstruct
--- linden.orig/indra/SConstruct	2007-08-30 02:33:27.000000000 +0300
+++ linden/indra/SConstruct	2007-08-30 02:35:15.000000000 +0300
@@ -739,7 +739,7 @@
 
         # Dataserver
         external_libs = common_external_libs + [
-            'boost_regex-gcc-mt', 'mysqlclient', 'tcmalloc', 'stacktrace',
+            'boost_regex', 'mysqlclient', 'tcmalloc', 'stacktrace',
             ]
         internal_libs = [ 'llcharacter', 'lldatabase', 'llimage', 'llimagej2coj', 'llinventory',
                 'llscene', 'llmessage', 'llvfs', 'llxml', 'llcommon', 'llmath' ]
@@ -762,7 +762,7 @@
 
         # Mapserver
         external_libs = common_external_libs + ['OSMesa16', 'kdu', 
-                'boost_regex-gcc-mt', 'iconv', 'jpeg', 'openjpeg', 'GL',
+                'boost_regex', 'iconv', 'jpeg', 'openjpeg', 'GL',
                 'mysqlclient', 'pthread', 'dl']
         internal_libs = ['llrender', 'llwindow', 'llimage', 'llimagej2coj', 'lldatabase', 'llprimitive', 'llmessage', 'llkdustatic',
                 'llxml', 'llvfs', 'llmath', 'llcommon']
@@ -772,7 +772,7 @@
         # Simulator
         external_libs = common_external_libs + [
             'hkdynamics', 'hkgeometry', 'hkmath', 'hkbase', 'hkcollide',
-            'hkactions', 'boost_regex-gcc-mt', 'openjpeg', 'dl', 'kdu',
+            'hkactions', 'boost_regex', 'openjpeg', 'dl', 'kdu',
             'mysqlclient', 'iconv', 'tcmalloc', 'stacktrace',
             ]
         internal_libs = [ 'lscript', 'llprimitive',
