diff -urNad linden~/indra/llvfs/lldir_linux.cpp linden/indra/llvfs/lldir_linux.cpp
--- linden~/indra/llvfs/lldir_linux.cpp	2007-04-04 02:55:51.000000000 +1000
+++ linden/indra/llvfs/lldir_linux.cpp	2007-04-08 01:14:23.000000000 +1000
@@ -80,7 +80,7 @@
 	mExecutablePathAndName = "";
 	mExecutableDir = tmp_str;
 	mWorkingDir = tmp_str;
-	mAppRODataDir = tmp_str;
+	mAppRODataDir = "/usr/share/secondlife";
 	mOSUserDir = getCurrentUserHome(tmp_str);
 	mOSUserAppDir = "";
 	mLindenUserDir = tmp_str;
