diff -Naur linden.orig/indra/newview/viewer_manifest.py linden/indra/newview/viewer_manifest.py
--- linden.orig/indra/newview/viewer_manifest.py	2007-06-13 20:10:29.000000000 +0300
+++ linden/indra/newview/viewer_manifest.py	2007-06-18 22:21:34.000000000 +0300
@@ -446,7 +446,7 @@
                 # TODO: I get the sense that this isn't fully fleshed out
                 if self.prefix("../../libraries/x86_64-linux/lib_release_client", "lib"):
 #                        self.path("libkdu_v42R.so")
-                        self.path("libxmlrpc.so.0")
+                        self.path("libxmlrpc-epi.so.0")
 #                        # self.path("libllkdu.so", "../bin/libllkdu.so") # llkdu goes in bin for some reason
                         self.end_prefix("lib")
 
