diff -Naur linden.orig/indra/SConstruct linden/indra/SConstruct
--- linden.orig/indra/SConstruct	2007-06-27 02:46:56.000000000 +0300
+++ linden/indra/SConstruct	2007-07-03 00:27:13.000000000 +0300
@@ -511,7 +511,7 @@
     if standalone and fedora:
         common_external_libs += [ 'xmlrpc-epi' ]
     else:
-        common_external_libs += [ 'xmlrpc' ]
+        common_external_libs += [ 'xmlrpc-epi' ]
 
     if build_target == 'client':
         if platform == 'linux':
@@ -659,7 +659,7 @@
 
         # Rpcserver
         Depends('rpcserver/rpcserver', 'userserver/userserver' + file_suffix)
-        external_libs = common_external_libs + ['xmlrpc', 'mysqlclient']
+        external_libs = common_external_libs + ['xmlrpc-epi', 'mysqlclient']
         internal_libs = ['llscene', 'llmessage', 'lldatabase', 'llvfs', 
                 'llmath', 'llcommon']
         create_executable('rpcserver/rpcserver' + file_suffix, 'rpcserver',
