--- linden/indra/newview/viewer.cpp	2007-02-21 09:49:25.000000000 +0200
+++ linden/indra/newview/viewer.cpp	2007-02-22 11:36:46.507634061 +0200
@@ -71,7 +71,7 @@
   #if LL_LINUX
   #	include <dlfcn.h>		// RTLD_LAZY
   #     include <execinfo.h>            // backtrace - glibc only
-  #define LL_ELFBIN 1
+  #define LL_ELFBIN 0
   #     if LL_ELFBIN
   #          include <cxxabi.h>         // for symbol demangling
   #          include "ELFIO.h"          // for better backtraces
