diff -Naur indra.orig/newview/viewer.cpp indra/newview/viewer.cpp
--- linden/indra.orig/newview/viewer.cpp	2007-08-30 00:10:43.000000000 +0300
+++ linden/indra/newview/viewer.cpp	2007-08-30 02:27:07.000000000 +0300
@@ -82,7 +82,7 @@
   #	include <dlfcn.h>		// RTLD_LAZY
   #     include <execinfo.h>            // backtrace - glibc only
   #     ifndef LL_ELFBIN
-  #define LL_ELFBIN 1
+  #define LL_ELFBIN 0
   #     endif // LL_ELFBIN
   #     if LL_ELFBIN
   #          include <cxxabi.h>         // for symbol demangling
diff -Naur indra.orig/SConstruct indra/SConstruct
--- linden/indra.orig/SConstruct	2007-08-30 00:10:39.000000000 +0300
+++ linden/indra/SConstruct	2007-08-30 02:26:08.000000000 +0300
@@ -93,7 +93,7 @@
         allowed_values=('default', 'aditi', 'agni', 'dmz', 'durga', 'firstlook', 'ganga', 'shakti', 'siva', 'soma', 'uma', 'vaak')),
     EnumOption('CHANNEL', 'Client package default channel', 'default',
         allowed_values=('default', 'Release', 'Release Candidate', 'WindLight')),
-    BoolOption('ELFIO', 'Enabled enhanced backtraces with libELFIO symbol extraction support', True),
+    BoolOption('ELFIO', 'Enabled enhanced backtraces with libELFIO symbol extraction support', False),
     BoolOption('STANDALONE', 'Build using system packages (implies OPENSOURCE)', False),
     BoolOption('OPENSOURCE', 'Build using only non-proprietary dependencies', True) # OPENSOURCE: do not edit this line
 )
