--- linden/indra/SConstruct.orig	2007-08-29 13:55:19.000000000 +0300
+++ linden/indra/SConstruct	2007-08-29 14:12:33.000000000 +0300
@@ -97,7 +97,7 @@
     EnumOption('BTARGET', 'Set build target', 'server',
         allowed_values=('client', 'server', 'all')),
     BoolOption('DISTCC', 'Enabled distcc', True),
-    BoolOption('MOZLIB', 'Enabled llmozlib/mozilla support', True),
+    BoolOption('MOZLIB', 'Enabled llmozlib/mozilla support', False),
     BoolOption('FMOD', 'Enabled FMOD audio support', True),
     BoolOption('GSTREAMER', 'Enabled GStreamer support', True),
     BoolOption('COLORGCC', 'Enabled colorgcc', True),
