diff -Naur linden.orig/indra/SConstruct linden/indra/SConstruct
--- linden.orig/indra/SConstruct	2007-04-05 00:45:31.000000000 +0900
+++ linden/indra/SConstruct	2007-04-05 00:49:09.000000000 +0900
@@ -56,6 +56,8 @@
 			allowed_values=('yes', 'no')))
 opts.Add(EnumOption('MOZLIB', 'Enabled llmozlib/mozilla support', 'yes',
 			allowed_values=('yes', 'no')))
+opts.Add(EnumOption('FMOD', 'Enabled fmod support', 'yes',
+			allowed_values=('yes', 'no')))
 opts.Add(EnumOption('COLORGCC', 'Enabled colorgcc', 'yes',
 			allowed_values=('yes', 'no')))
 opts.Add(EnumOption('GRID', 'Client package\'s default grid', 'default',
@@ -71,6 +73,7 @@
 target_param = ARGUMENTS.get('BTARGET', 'server')
 enable_distcc = ARGUMENTS.get('DISTCC', 'yes')
 enable_mozlib = ARGUMENTS.get('MOZLIB', 'yes')
+enable_fmod = ARGUMENTS.get('FMOD', 'yes')
 enable_colorgcc = ARGUMENTS.get('COLORGCC', 'yes')
 grid = ARGUMENTS.get('GRID', 'default')
 # OPENSOURCE: do not edit the following line:
@@ -184,7 +187,7 @@
 		flags += '-DLL_LINUX=1 '
 		if build_target == 'client':
 			flags += '-DAPPID=secondlife -DLL_SDL=1 '
-			if arch == 'x86_64' or arch == 'x86_64cross':
+			if enable_fmod == 'no' or arch == 'x86_64' or arch == 'x86_64cross':
 				flags += '-DLL_FMOD=0 '
 			flags += '-DLL_X11=1 -DLL_GTK=1 '
 			client_external_libs += [ 'gtk-x11-2.0', 'elfio' ]
@@ -419,7 +422,7 @@
 
 		external_libs = client_external_libs + common_external_libs + [ 'freetype', 'jpeg', 'SDL', 'GL', 'GLU', 'ogg', 'vorbisenc', 'vorbisfile', 'vorbis', 'db-4.2', 'openjpeg' ]
 
-		if arch != 'x86_64' and arch != 'x86_64cross':
+		if enable_fmod == 'yes' and arch != 'x86_64' and arch != 'x86_64cross':
 			external_libs += [ 'fmod-3.75' ]
 
 		external_libs.remove('cares')
