--- share/hedgewars/Data/Locale/tr.txt	2009-05-24 18:47:12.000000000 +0300
+++ share/hedgewars/Data/Locale/tr.txt	2009-05-31 13:54:43.000000000 +0300
@@ -37,8 +37,8 @@
 00:34=Ölümsüzlük
 00:35=Arttırılmış Zaman
 00:36=Lazer Görüşü
-00:37=Vampirism
-00:38=Sniper Rifle
+00:37=Vampirizm
+00:38=Keskin Nişancı
 
 01:00=Savaş başlasın!
 01:01=Beraberlik
--- share/hedgewars/Data/Locale/hedgewars_tr_TR.ts	2009-05-24 18:47:12.000000000 +0300
+++ share/hedgewars/Data/Locale/hedgewars_tr_TR.ts	2009-05-31 13:51:16.000000000 +0300
@@ -1,11 +1,12 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!DOCTYPE TS><TS version="1.1" language="tr_TR">
+<defaultcodec></defaultcodec>
 <context>
     <name>AmmoSchemeModel</name>
     <message>
         <location filename="../../../../QTfrontend/ammoSchemeModel.cpp" line="246"/>
         <source>new</source>
-        <translation type="unfinished">yeni</translation>
+        <translation>yeni</translation>
     </message>
 </context>
 <context>
@@ -13,13 +14,13 @@
     <message>
         <location filename="../../../../QTfrontend/misc.h" line="38"/>
         <source>Never</source>
-        <translation type="unfinished"></translation>
+        <translation>Asla</translation>
     </message>
     <message numerus="yes">
         <location filename="../../../../QTfrontend/misc.h" line="40"/>
         <source>Every %1 turn</source>
-        <translation type="unfinished">
-            <numerusform></numerusform>
+        <translation>
+            <numerusform>%1 turda bir</numerusform>
             <numerusform></numerusform>
         </translation>
     </message>
@@ -29,7 +30,7 @@
     <message>
         <location filename="../../../../QTfrontend/gamecfgwidget.cpp" line="69"/>
         <source>Edit weapons</source>
-        <translation type="unfinished"></translation>
+        <translation>Silahları düzenle</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/gamecfgwidget.cpp" line="148"/>
@@ -167,7 +168,6 @@
 <context>
     <name>HWNewNet</name>
     <message>
-        <location filename="" line="0"/>
         <source>Error</source>
         <translation type="obsolete">Hata</translation>
     </message>
@@ -235,12 +235,12 @@
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1137"/>
         <source>Server message:</source>
-        <translation type="unfinished"></translation>
+        <translation>Sunucu iletisi:</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1143"/>
         <source>Set message</source>
-        <translation type="unfinished"></translation>
+        <translation>İletiyi ayarla</translation>
     </message>
 </context>
 <context>
@@ -248,7 +248,7 @@
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="857"/>
         <source>Connecting...</source>
-        <translation type="unfinished"></translation>
+        <translation>Bağlanıyor...</translation>
     </message>
 </context>
 <context>
@@ -335,12 +335,12 @@
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1171"/>
         <source>LAN game</source>
-        <translation type="unfinished"></translation>
+        <translation>LAN oyunu</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1172"/>
         <source>Official server</source>
-        <translation type="unfinished"></translation>
+        <translation>Resmi sunucu</translation>
     </message>
 </context>
 <context>
@@ -452,67 +452,67 @@
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="898"/>
         <source>Defend your fort and destroy the opponents, two team colours max!</source>
-        <translation type="unfinished"></translation>
+        <translation>Kaleni koru ve düşmanları yok et, en fazla iki takım rengi!</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="903"/>
         <source>Teams will start on opposite sides of the terrain, two team colours max!</source>
-        <translation type="unfinished"></translation>
+        <translation>Takımlar bölgenin faklı taraflarında başlarlar, en fazla iki takım rengi!</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="908"/>
         <source>Land can not be destroyed!</source>
-        <translation type="unfinished"></translation>
+        <translation>Zemin yok edilemez!</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="913"/>
         <source>Add an indestructable border around the terrain</source>
-        <translation type="unfinished"></translation>
+        <translation>Bölgenin etrafına yok edilemez bir sınır ekle</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="918"/>
         <source>Lower gravity</source>
-        <translation type="unfinished"></translation>
+        <translation>Düşük yer çekimi</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="923"/>
         <source>Assisted aiming with laser sight</source>
-        <translation type="unfinished"></translation>
+        <translation>Lazer görüşü destekli nişan alma</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="928"/>
         <source>All hogs have a personal forcefield</source>
-        <translation type="unfinished"></translation>
+        <translation>Tüm kirpilerin kişisel güç kalkanı olur</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="933"/>
         <source>Enable random mines</source>
-        <translation type="unfinished"></translation>
+        <translation>Rastgele mayınlar</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="938"/>
         <source>Gain 80% of the damage you do back in health</source>
-        <translation type="unfinished"></translation>
+        <translation>Verdiğin hasarın %%80&apos;ini sağlık olarak kazan</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="943"/>
         <source>Share your opponents pain, share their damage</source>
-        <translation type="unfinished"></translation>
+        <translation>Rakibin acısını paylaş, hasarını paylaş</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="948"/>
         <source>Your hogs are unable to move, put your artillery skills to the test</source>
-        <translation type="unfinished"></translation>
+        <translation>Kirpiler hareket edemez, topçuluk becerini ölç</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1040"/>
         <source>Random</source>
-        <translation type="unfinished"></translation>
+        <translation>Rastgele</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1041"/>
         <source>Seconds</source>
-        <translation type="unfinished"></translation>
+        <translation>Saniye</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1069"/>
@@ -520,19 +520,17 @@
         <translation>Yeni</translation>
     </message>
     <message>
-        <location filename="" line="0"/>
         <source>Prev</source>
         <translation type="obsolete">Önce</translation>
     </message>
     <message>
-        <location filename="" line="0"/>
         <source>Next</source>
         <translation type="obsolete">Sonra</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1070"/>
         <source>Delete</source>
-        <translation type="unfinished">Sil</translation>
+        <translation>Sil</translation>
     </message>
 </context>
 <context>
@@ -612,17 +610,14 @@
 <context>
     <name>QCheckBox</name>
     <message>
-        <location filename="" line="0"/>
         <source>Forts mode</source>
         <translation type="obsolete">Kaleler modu</translation>
     </message>
     <message>
-        <location filename="" line="0"/>
         <source>Divide teams</source>
         <translation type="obsolete">Takımları böl</translation>
     </message>
     <message>
-        <location filename="" line="0"/>
         <source>Solid land</source>
         <translation type="obsolete">Parçalanmaz yüzey</translation>
     </message>
@@ -662,7 +657,6 @@
         <translation>Kayıt dosyasının sonuna tarihi ve saati ekle</translation>
     </message>
     <message>
-        <location filename="" line="0"/>
         <source>Add Border</source>
         <translation type="obsolete">Sınır Ekle</translation>
     </message>
@@ -672,29 +666,25 @@
         <translation>Kaliteyi Düşür</translation>
     </message>
     <message>
-        <location filename="" line="0"/>
         <source>Low Gravity</source>
         <translation type="obsolete">Azaltılmış Yerçekimi</translation>
     </message>
     <message>
-        <location filename="" line="0"/>
         <source>Laser Sight</source>
         <translation type="obsolete">Lazer Görüşü</translation>
     </message>
     <message>
-        <location filename="" line="0"/>
         <source>Invulnerable</source>
         <translation type="obsolete">Ölümsüzlük</translation>
     </message>
     <message>
-        <location filename="" line="0"/>
         <source>Add Mines</source>
         <translation type="obsolete">Mayın Ekle</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="412"/>
         <source>Frontend Effects (Requires Restart)</source>
-        <translation type="unfinished"></translation>
+        <translation>Arayüz Efektleri (Baştan Başlatmayı Gerektirir)</translation>
     </message>
 </context>
 <context>
@@ -765,12 +755,12 @@
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="875"/>
         <source>Game Modifiers</source>
-        <translation type="unfinished"></translation>
+        <translation>Oyun Değişkenleri</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="876"/>
         <source>Basic Settings</source>
-        <translation type="unfinished"></translation>
+        <translation>Basit Ayarlar</translation>
     </message>
 </context>
 <context>
@@ -778,12 +768,12 @@
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1029"/>
         <source>Mines Time</source>
-        <translation type="unfinished"></translation>
+        <translation>Mayın Süresi</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1045"/>
         <source>Mines</source>
-        <translation type="unfinished"></translation>
+        <translation>Mayınlar</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/about.cpp" line="43"/>
@@ -821,12 +811,10 @@
         <translation>Özel teşekkür:</translation>
     </message>
     <message>
-        <location filename="" line="0"/>
         <source>Turn time</source>
         <translation type="obsolete">Tur süresi</translation>
     </message>
     <message>
-        <location filename="" line="0"/>
         <source>Initial health</source>
         <translation type="obsolete">Başlangıç sağlığı</translation>
     </message>
@@ -878,37 +866,37 @@
     <message>
         <location filename="../../../../QTfrontend/gamecfgwidget.cpp" line="53"/>
         <source>Game scheme</source>
-        <translation type="unfinished"></translation>
+        <translation>Oyun teması</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="955"/>
         <source>Damage Modifier</source>
-        <translation type="unfinished"></translation>
+        <translation>Hasar Çarpanı</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="970"/>
         <source>Turn Time</source>
-        <translation type="unfinished"></translation>
+        <translation>Tur Süresi</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="985"/>
         <source>Initial Health</source>
-        <translation type="unfinished"></translation>
+        <translation>Başlangıç Sağlığı</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1000"/>
         <source>Sudden Death Timeout</source>
-        <translation type="unfinished"></translation>
+        <translation>Ani Ölüm Süresi</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1015"/>
         <source>Crate Drops</source>
-        <translation type="unfinished"></translation>
+        <translation>Yardım Kasaları</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="1059"/>
         <source>Scheme Name:</source>
-        <translation type="unfinished"></translation>
+        <translation>Tema Adı:</translation>
     </message>
 </context>
 <context>
@@ -1030,7 +1018,6 @@
         <translation>Elle Belirt</translation>
     </message>
     <message>
-        <location filename="" line="0"/>
         <source>Join official server</source>
         <translation type="obsolete">Resmi sunucuya bağlan</translation>
     </message>
@@ -1139,57 +1126,57 @@
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="937"/>
         <source>Vampirism</source>
-        <translation type="unfinished"></translation>
+        <translation>Vampircilik</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="942"/>
         <source>Karma</source>
-        <translation type="unfinished"></translation>
+        <translation>Karma</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="947"/>
         <source>Artillery</source>
-        <translation type="unfinished"></translation>
+        <translation>Topçuluk</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="897"/>
         <source>Fort Mode</source>
-        <translation type="unfinished"></translation>
+        <translation>Kale Modu</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="902"/>
         <source>Divide Teams</source>
-        <translation type="unfinished"></translation>
+        <translation>Takımları Böl</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="907"/>
         <source>Solid Land</source>
-        <translation type="unfinished"></translation>
+        <translation>Parçalanmaz Yüzey</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="912"/>
         <source>Add Border</source>
-        <translation type="unfinished">Sınır Ekle</translation>
+        <translation>Sınır Ekle</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="917"/>
         <source>Low Gravity</source>
-        <translation type="unfinished">Azaltılmış Yerçekimi</translation>
+        <translation>Azaltılmış Yerçekimi</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="922"/>
         <source>Laser Sight</source>
-        <translation type="unfinished">Lazer Görüşü</translation>
+        <translation>Lazer Görüşü</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="927"/>
         <source>Invulnerable</source>
-        <translation type="unfinished">Ölümsüzlük</translation>
+        <translation>Ölümsüzlük</translation>
     </message>
     <message>
         <location filename="../../../../QTfrontend/pages.cpp" line="932"/>
         <source>Add Mines</source>
-        <translation type="unfinished">Mayın Ekle</translation>
+        <translation>Mayın Ekle</translation>
     </message>
 </context>
 <context>
