--- Makefile	1995-12-05 14:26:11.000000000 +0200
+++ Makefile-yeni	2006-09-07 14:27:42.000000000 +0300
@@ -16,20 +16,20 @@
 PERL = /usr/bin/perl
 
 # Where do you want the sgmlspl executable script to be installed?
-BINDIR = /usr/local/bin
+BINDIR = /var/tmp/pisi/SGMLSpm-1.03.5-1/install/usr/bin
 
 # Where do you put local perl5 libaries?
-PERL5DIR = /usr/local/lib/perl5
-MODULEDIR = ${PERL5DIR}/SGMLS
+PERL5DIR = /var/tmp/pisi/SGMLSpm-1.03.5-1/install/usr/lib/perl5
+MODULEDIR = /var/tmp/pisi/SGMLSpm-1.03.5-1/install/usr/lib/perl5/vendor_perl/5.8.8/SGMLS
 
 # Where do you want to put sgmlspl specifications?
-SPECDIR = ${PERL5DIR}
+SPECDIR = /var/tmp/pisi/SGMLSpm-1.03.5-1/install/usr/share/SGMLS
 
 # If you plan to install the HTML version of the documentation, where
 # do you intend to put it?  'make html' will create two
 # subdirectories, ${HTMLDIR}/SGMLSpm and ${HTMLDIR}/sgmlspl, and place
 # its files there.
-HTMLDIR = /usr/local/lib/www/docs
+HTMLDIR = /var/tmp/pisi/SGMLSpm-1.03.5-1/install/usr/share/doc/SGMLSpm-1.03.5-1/html
 
 
 #
@@ -39,12 +39,12 @@
 HTML_SOURCES = DOC/HTML/SGMLSpm/ DOC/HTML/sgmlspl/
 
 FILES =	${BINDIR}/sgmlspl \
-	${PERL5DIR}/SGMLS.pm \
+	/usr/lib/perl5/vendor_perl/5.8.8/SGMLS.pm \
 	${MODULEDIR}/Output.pm \
 	${MODULEDIR}/Refs.pm \
 	${SPECDIR}/skel.pl
 
-all: install docs
+all:
 
 install: install_system # install_html
 
@@ -56,9 +56,9 @@
 	sed -e 's!/usr/bin/perl!${PERL}!' sgmlspl.pl > ${BINDIR}/sgmlspl
 	chmod a+x,a+r ${BINDIR}/sgmlspl
 
-${PERL5DIR}/SGMLS.pm: SGMLS.pm
-	cp SGMLS.pm ${PERL5DIR}/SGMLS.pm
-	chmod a+r ${PERL5DIR}/SGMLS.pm
+${PERL5DIR}/vendor_perl/5.8.8/SGMLS.pm: SGMLS.pm
+	cp SGMLS.pm /usr/lib/perl5/vendor_perl/5.8.8/SGMLS.pm
+	chmod a+r /usr/lib/perl5/vendor_perl/5.8.8/SGMLS.pm
 
 ${MODULEDIR}:
 	if [ ! -d ${MODULEDIR} ]; then\
