diff -Nuar exo-0.3.101.orig/python/Makefile.in exo-0.3.101/python/Makefile.in
--- exo-0.3.101.orig/python/Makefile.in	2009-04-13 00:37:15.000000000 +0300
+++ exo-0.3.101/python/Makefile.in	2009-05-29 18:21:03.495512228 +0300
@@ -333,7 +333,7 @@
 	-avoid-version							\
 	-export-symbols-regex init_exo					\
 	-module								\
-	-no-undefined
+	-no-undefined `python-config --libs`
 
 _exo_la_LIBADD = \
 	$(top_builddir)/exo/libexo-$(LIBEXO_VERSION_API).la		\
