--- ./icons/peg-e.desktop	2009-04-21 14:26:48.000000000 +0300
+++ ./icons/peg-e.desktop	2010-01-26 17:27:55.984866852 +0200
@@ -1,8 +1,15 @@
 [Desktop Entry]
+Type=Application
+Version=1.0
+Encoding=UTF-8
 Name=Peg-E
+Name[tr]=Peg-E
+GenericName=Peg Solitaire Game
+GenericName[tr]=Solo Test Oyunu
 Comment=Peg elimination game
+Comment[tr]=Bir süper beyin türevi
 Exec=peg-e
 Icon=peg-e
 Terminal=false
-Type=Application
+StartupNotify=false
 Categories=Qt;Game;
