--- kmess/account.cpp	2008-01-10 08:01:49.000000000 +0200
+++ kmess/account.cpp	2008-01-10 08:05:12.000000000 +0200
@@ -50,7 +50,7 @@
    doNotifyContactsChatMessages_(false),
    doNotifyContactsOffline_(true),
    doNotifyEmails_(true),
-   emailCommand_("kmail"),
+   emailCommand_("xdg-open http://www.hotmail.com"),
    emailSupported_(true),  // Assume true (for new accounts), until the msn server proves the opposite
    emoticonStyle_("KMess-new"),
    fontColor_("#000000"),
