--- data.bos/scripts/menus/options.lua.orig	2006-12-04 23:05:50.000000000 +0200
+++ data.bos/scripts/menus/options.lua	2007-01-24 00:25:46.000000000 +0200
@@ -250,14 +250,15 @@
   end
      
   print(StratagusTranslation)
-  b = AddLanguage("English", "en", 1.5)
+  b = AddLanguage("English", "en", 1.6)
   if StratagusTranslation == "" then
      b:setMarked(true)
   end
-  AddLanguage("Français", "fr", 2.5)
-  AddLanguage("Suomi", "fi", 3.5)
-  AddLanguage("Deutsch", "de", 4.5)
-  AddLanguage("Polski", "pl", 5.5)
+  AddLanguage("Türkçe", "tr", 2.6)
+  AddLanguage("Français", "fr", 3.6)
+  AddLanguage("Suomi", "fi", 4.6)
+  AddLanguage("Deutsch", "de", 5.6)
+  AddLanguage("Polski", "pl", 6.6)
 
   menu:addButton(_("~!OK"), "o", Video.Width / 2 - 100, Video.Height - 100,
     function() menu:stop() end)
