diff -Naur firestarter-1.0.3.orig/Makefile.in firestarter-1.0.3/Makefile.in
--- firestarter-1.0.3.orig/Makefile.in	2008-06-10 18:33:03.000000000 +0300
+++ firestarter-1.0.3/Makefile.in	2008-06-10 18:36:32.000000000 +0300
@@ -479,7 +479,7 @@
 @INTLTOOL_SCHEMAS_RULE@
 
 install-data-local:
-	-GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(srcdir)/$(schema_DATA)
+	# -GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(srcdir)/$(schema_DATA)
 
 @INTLTOOL_DESKTOP_RULE@
 
