diff -Naur twitux-0.68.orig/data/twitux.desktop.in.in twitux-0.68/data/twitux.desktop.in.in
--- twitux-0.68.orig/data/twitux.desktop.in.in	2008-12-13 23:36:37.000000000 +0200
+++ twitux-0.68/data/twitux.desktop.in.in	2008-12-26 17:57:33.000000000 +0200
@@ -1,7 +1,10 @@
 [Desktop Entry]
 Version=1.0
 _Name=Twitux
+_GenericName=Twitter Client
+_GenericName[tr]=Twitter İstemcisi
 _Comment=A Twitter client
+_Comment[tr]=Twitter istemcisi
 Exec=twitux
 Icon=twitux
 StartupNotify=true
