diff -ur gquilt-0.20.orig/Makefile gquilt-0.20/Makefile
--- gquilt-0.20.orig/Makefile	2007-03-05 07:38:46.000000000 +0200
+++ gquilt-0.20/Makefile	2007-04-04 23:47:21.472617187 +0300
@@ -1,7 +1,7 @@
 MAJVER=0
 MINVER=20
 VERSION=$(MAJVER).$(MINVER)
-PREFIX=/usr/local
+PREFIX=/usr
 MODULE_PREFIX=share
 MODULES=gquilt gquilt_gtk gquilt_utils gquilt_icons gquilt_pfuns gquilt_tool
 MODULES+=gquilt_quilt gquilt_mq gquilt_const
