--- src/kirocker.desktop.orig	2007-03-05 10:34:01.000000000 +0200
+++ src/kirocker.desktop	2007-03-05 10:35:39.000000000 +0200
@@ -2,8 +2,10 @@
 Encoding=UTF-8
 Comment=Display cover and information about the currently played music
 Comment[fr]=Affiche la pochette et les informations sur la musique en cours de lecture
+Comment[tr]=Çalan parçanın kapak dosyasını ve mevcut bilgilerini gösterir.
 Comment[xx]=xxDisplay cover and information about the currently played musicxx
 Name=Kirocker Music Display
+Name[tr]=Kirocker Müzik Paneli
 Name[fr]=Afficheur de musique Kirocker
 Name[xx]=xxKirocker Music Displayxx
 Icon=kirocker
