--- CHANGES.orig	2010-05-07 16:35:13.485650530 +0300
+++ CHANGES	2010-05-07 16:37:44.012774589 +0300
@@ -3,6 +3,13 @@
 Changes since Lynx 2.8 release
 ===============================================================================
 
+2010-05-07
+* modify patch for CVE-2005-2929 to prompt user before executing command via
+  a lynxcgi link even in advanced mode, as the actual URL may not be shown but
+  hidden behind an HTTP redirect
+* set TRUSTED_LYNXCGI:none in lynx.cfg to disable all lynxcgi URLs by default
+  [CVE-2008-4690]
+
 2009-07-05 (2.8.7rel.1)
 * update metrics for 2.8.7 release.
 * fix ifdef'ing for cfg_bad_html (report by Gabor Z Papp) -TD
--- lynx.cfg.orig	2010-05-07 13:23:16.869711592 +0300
+++ lynx.cfg	2010-05-07 16:33:07.191650813 +0300
@@ -1078,7 +1078,7 @@
 #
 # The default TRUSTED_LYNXCGI rule is "none".
 #
-#TRUSTED_LYNXCGI:none
+TRUSTED_LYNXCGI:none
 
 
 .h2 LYNXCGI_ENVIRONMENT
