diff -Nuar Thunar-0.8.0/Thunar-bulk-rename.desktop.in.in Thunar-0.8.0.patched/Thunar-bulk-rename.desktop.in.in
--- Thunar-0.8.0/Thunar-bulk-rename.desktop.in.in	2007-01-20 22:39:10.000000000 +0200
+++ Thunar-0.8.0.patched/Thunar-bulk-rename.desktop.in.in	2007-05-08 17:41:52.000000000 +0300
@@ -8,6 +8,6 @@
 Terminal=false
 StartupNotify=true
 Type=Application
-Categories=Application;System;Utility;Core;GTK;Filesystem;
+Categories=Application;Utility;Core;GTK;Filesystem;
 
 # vi:set encoding=UTF-8:
diff -Nuar Thunar-0.8.0/Thunar.desktop.in.in Thunar-0.8.0.patched/Thunar.desktop.in.in
--- Thunar-0.8.0/Thunar.desktop.in.in	2007-01-20 22:39:10.000000000 +0200
+++ Thunar-0.8.0.patched/Thunar.desktop.in.in	2007-05-08 17:41:50.000000000 +0300
@@ -8,6 +8,6 @@
 Terminal=false
 StartupNotify=true
 Type=Application
-Categories=Application;System;Utility;Core;GTK;FileManager;
+Categories=Application;Utility;Core;GTK;FileManager;
 
 # vi:set encoding=UTF-8:
diff -Nuar Thunar-0.8.0/Thunar-folder-handler.desktop.in.in Thunar-0.8.0.patched/Thunar-folder-handler.desktop.in.in
--- Thunar-0.8.0/Thunar-folder-handler.desktop.in.in	2007-01-20 22:39:10.000000000 +0200
+++ Thunar-0.8.0.patched/Thunar-folder-handler.desktop.in.in	2007-05-08 17:41:45.000000000 +0300
@@ -10,7 +10,7 @@
 Terminal=false
 StartupNotify=true
 Type=Application
-Categories=System;Utility;Core;GTK;FileManager;
+Categories=Application;Utility;Core;GTK;Filesystem
 MimeType=x-directory/gnome-default-handler;x-directory/normal;inode/directory;
 
 # vi:set encoding=UTF-8:
