diff -Naur kgmailnotifier-0.5.0.orig/src/kgmailnotifier.desktop kgmailnotifier-0.5.0/src/kgmailnotifier.desktop
--- kgmailnotifier-0.5.0.orig/src/kgmailnotifier.desktop	2009-09-11 00:44:03.665416139 +0300
+++ kgmailnotifier-0.5.0/src/kgmailnotifier.desktop	2009-09-11 00:46:37.398416065 +0300
@@ -1,6 +1,8 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=KGmailNotifier
+GenericName=Gmail Notifier
+GenericName[tr]=Gmail Uygulaması
 Exec=kgmailnotifier
 Icon=kgmailnotifier
 Type=Application
@@ -36,7 +38,7 @@
 Comment[sv]=Ett enkelt KDE-program
 Comment[ta]=ஒரு சாதாரண கெடிஇ பயன்பாடு
 Comment[tg]=Гузориши оддиKDE 
-Comment[tr]=Basit bir KDE Uygulaması
+Comment[tr]=KDE için Gmail uyarı uygulaması
 Comment[zh_CN]=一个简单的 KDE 应用程序
 Comment[zh_TW]=簡單的 KDE 應用程式
 
