Index: enlightenment-0.16.999.49539/config/standard/e.src
===================================================================
--- enlightenment-0.16.999.49539.orig/config/standard/e.src
+++ enlightenment-0.16.999.49539/config/standard/e.src
@@ -18,7 +18,7 @@ group "E_Config" struct {
   value "zone_desks_x_count" int: 4;
   value "zone_desks_y_count" int: 1;
   value "use_virtual_roots" int: 0;
-  value "show_desktop_icons" int: 1;
+  value "show_desktop_icons" int: 0;
   value "edge_flip_dragging" int: 1;
   value "evas_engine_default" int: 1;
   value "evas_engine_container" int: 0;
@@ -458,6 +458,14 @@ group "E_Config" struct {
   }
   group "modules" list {
     group "E_Config_Module" struct {
+      value "name" string: "systray";
+      value "enabled" uchar: 1;
+      value "delayed" uchar: 1;
+      value "priority" int: 0;
+    }
+  }
+  group "modules" list {
+    group "E_Config_Module" struct {
       value "name" string: "everything";
       value "enabled" uchar: 1;
       value "delayed" uchar: 0;
@@ -1635,7 +1643,7 @@ group "E_Config" struct {
   value "focus_revert_on_hide_or_close" int: 1;
   value "pointer_slide" int: 1;
   value "use_e_cursor" int: 1;
-  value "cursor_size" int: 32;
+  value "cursor_size" int: 24;
   value "menu_autoscroll_margin" int: 0;
   value "menu_autoscroll_cursor_margin" int: 1;
   value "transient.move" int: 1;
@@ -2086,8 +2094,8 @@ group "E_Config" struct {
   value "border_raise_on_focus" int: 1;
   value "desk_flip_wrap" int: 0;
   value "fullscreen_flip" int: 1;
-  value "icon_theme" string: "Tango";
-  value "icon_theme_overrides" uchar: 0;
+  value "icon_theme" string: "Milky";
+  value "icon_theme_overrides" uchar: 1;
   value "desk_flip_animate_mode" int: 1;
   value "desk_flip_animate_interpolation" int: 0;
   value "desk_flip_animate_time" double: 0.2000000000000000111022302;
