--- e-0.16.999.063_20100215/data/other/Makefile.am.orig	2010-02-25 16:55:11.599927215 +0200
+++ e-0.16.999.063_20100215/data/other/Makefile.am	2010-02-25 16:55:16.123892631 +0200
@@ -1,9 +1,7 @@
 MAINTAINERCLEANFILES = Makefile.in
 filesdir = $(datadir)/enlightenment/data/other
 files_DATA = \
-desktop_files.tar.gz \
 desktop_order.tar.gz \
-efm_favorites.tar.gz \
 icon_example.tar.gz
 
 EXTRA_DIST = $(files_DATA)
