--- e-0.16.999.063_20100215/config/minimalist/profile.desktop.orig	2010-02-15 20:26:07.000000000 +0200
+++ e-0.16.999.063_20100215/config/minimalist/profile.desktop	2010-02-18 16:09:33.559663122 +0200
@@ -7,6 +7,7 @@
 Name[fr]=Minimaliste
 Name[el]=Μινιμαλιστικό
 Name[eo]=Minimuma
+Name[tr_TR]=Minimal
 Name[zh_CN]=极简
 Icon=icon.png
 Comment=A mininimalist setup<br>with very few things on<br>the screen. Should<br>appeal to Those<br>that like as little<br>clutter as possible.
@@ -16,4 +17,5 @@
 Comment[fr]=Une configuration minimale avec<br>très peu de choses à l'écran.<br>Pour ceux qui apprécient<br>un bureau dégagé.
 Comment[el]=Ένα μινιμαλιστικό προφίλ<br>με πολύ λίγα στοιχεία στην<br>οθόνη. Θα αρέσει σε<br>αυτούς που θέλουν τα απολύτως<br>απαραίτητα.
 Comment[eo]=Minimuma agordo kun<br>malmulte da objektoj sur<br>la ekrano. Taŭgas<br>por ĉiuj kiuj ŝatas<br>senĝenajn labortablojn.
+Comment[tr_TR]=Ekranda en az ögenin bulunduğu minimal kurulumdur.<br>
 Comment[zh_CN]=一个只在屏幕上放置极少项目的极简设置。
