--- data/lmms.desktop	2007-07-16 22:26:05.000000000 +0300
+++ data/lmms.desktop.patched	2007-09-14 20:18:33.000000000 +0300
@@ -1,6 +1,8 @@
 [Desktop Entry]
 Name=Linux MultiMedia Studio
-GenericName=software for music-production
+Name[tr]=Linux Çokluortam Stüdyosu
+GenericName=Music Production Software
+GenericName[tr]=Müzik Yapım Yazılımı
 GenericName[ca]=Programari de producció musical
 GenericName[de]=Software zur Musik-Produktion
 Comment=easy music-production for everyone!
