diff -dur xfce4-taskmanager-0.4.0.orig/src/functions.c xfce4-taskmanager-0.4.0/src/functions.c
--- xfce4-taskmanager-0.4.0.orig/src/functions.c	2008-05-17 10:20:35.000000000 +0200
+++ xfce4-taskmanager-0.4.0/src/functions.c	2008-05-24 01:32:25.000000000 +0200
@@ -55,7 +55,7 @@
 				tmp->time = new_tmp->time;
 
 				tmp->old_time_percentage = tmp->time_percentage;
-				tmp->time_percentage = (gdouble)(tmp->time - tmp->old_time) * (gdouble)(1/num_cpus);
+				tmp->time_percentage = (gdouble)(tmp->time - tmp->old_time) * ((gdouble)1/num_cpus);
 
 				if(
 				    tmp->ppid != new_tmp->ppid ||
