diff -Nuar anjuta-2.1.2.orig/gnome-doc-utils.make anjuta-2.1.2/gnome-doc-utils.make
--- anjuta-2.1.2.orig/gnome-doc-utils.make	2007-03-30 17:00:53.000000000 +0300
+++ anjuta-2.1.2/gnome-doc-utils.make	2007-03-30 17:02:40.000000000 +0300
@@ -662,7 +662,6 @@
 	done
 	@if test "x$(_ENABLE_SK)" = "xtrue"; then \
 	  echo "scrollkeeper-update -p $(DESTDIR)$(_sklocalstatedir) -o $(DESTDIR)$(OMF_DIR)/$(DOC_MODULE)"; \
-	  scrollkeeper-update -p "$(DESTDIR)$(_sklocalstatedir)" -o "$(DESTDIR)$(OMF_DIR)/$(DOC_MODULE)"; \
 	fi;
 
 install-doc-dsk:
diff -Nuar anjuta-2.1.2.orig/manuals/anjuta-faqs/Makefile.in anjuta-2.1.2/manuals/anjuta-faqs/Makefile.in
--- anjuta-2.1.2.orig/manuals/anjuta-faqs/Makefile.in	2007-03-30 17:00:52.000000000 +0300
+++ anjuta-2.1.2/manuals/anjuta-faqs/Makefile.in	2007-03-30 17:03:50.000000000 +0300
@@ -1059,7 +1059,6 @@
 	done
 	@if test "x$(_ENABLE_SK)" = "xtrue"; then \
 	  echo "scrollkeeper-update -p $(DESTDIR)$(_sklocalstatedir) -o $(DESTDIR)$(OMF_DIR)/$(DOC_MODULE)"; \
-	  scrollkeeper-update -p "$(DESTDIR)$(_sklocalstatedir)" -o "$(DESTDIR)$(OMF_DIR)/$(DOC_MODULE)"; \
 	fi;
 
 install-doc-dsk:
diff -Nuar anjuta-2.1.2.orig/manuals/anjuta-manual/Makefile.in anjuta-2.1.2/manuals/anjuta-manual/Makefile.in
--- anjuta-2.1.2.orig/manuals/anjuta-manual/Makefile.in	2007-03-30 17:00:52.000000000 +0300
+++ anjuta-2.1.2/manuals/anjuta-manual/Makefile.in	2007-03-30 17:03:24.000000000 +0300
@@ -1133,7 +1133,6 @@
 	done
 	@if test "x$(_ENABLE_SK)" = "xtrue"; then \
 	  echo "scrollkeeper-update -p $(DESTDIR)$(_sklocalstatedir) -o $(DESTDIR)$(OMF_DIR)/$(DOC_MODULE)"; \
-	  scrollkeeper-update -p "$(DESTDIR)$(_sklocalstatedir)" -o "$(DESTDIR)$(OMF_DIR)/$(DOC_MODULE)"; \
 	fi;
 
 install-doc-dsk:
