diff -Naur texmaker-1.8.orig/utilities/texmaker.desktop texmaker-1.8/utilities/texmaker.desktop
--- texmaker-1.8.orig/utilities/texmaker.desktop	2008-06-28 01:42:57.000000000 +0300
+++ texmaker-1.8/utilities/texmaker.desktop	2008-11-04 22:59:17.000000000 +0200
@@ -1,14 +1,17 @@
 [Desktop Entry]
-Categories=Office;Publishing;Qt;X-SuSE-Core-Office;X-Mandriva-Office-Publishing;X-Misc;
+Categories=Application;Office;Publishing;Qt;X-Misc;
 Encoding=UTF-8
 Exec=texmaker
 GenericName=LaTeX Editor
+GenericName[tr]=LaTeX Düzenleyici
 GenericName[fr]=Editeur LaTeX
 Comment=LaTeX development environment
+Comment[tr]=LaTeX Geliştirme Araçları
 Comment[fr]=Environnement de développement LaTeX
 Icon=texmaker
 MimeType=text/x-tex;
 Name=Texmaker
+Name[tr]=Texmaker
 Name[fr]=Texmaker
 StartupNotify=false
 Terminal=false
