diff -Nuar listen-0.6.2.orig/misc/listen.desktop.in listen-0.6.2/misc/listen.desktop.in
--- listen-0.6.2.orig/misc/listen.desktop.in	2009-07-05 22:31:13.000000000 +0300
+++ listen-0.6.2/misc/listen.desktop.in	2009-07-05 22:38:17.000000000 +0300
@@ -1,8 +1,10 @@
 [Desktop Entry]
 Version=1.0
-_Name=Listen Music Player
+Name=Listen
 _GenericName=Music Player
+GenericName[tr]=Müzik Çalar
 _Comment=Listen, browse, or edit your audio collection
+Comment[tr]=Dinle, tara ve ses kolleksiyonunu düzenle
 Exec=listen
 Terminal=false
 Type=Application
