diff -Nuar gst-python-0.10.15.orig/gst/Makefile.in gst-python-0.10.15/gst/Makefile.in
--- gst-python-0.10.15.orig/gst/Makefile.in	2009-05-11 02:00:49.000000000 +0300
+++ gst-python-0.10.15/gst/Makefile.in	2009-06-16 22:47:02.000000000 +0300
@@ -388,7 +388,7 @@
 top_srcdir = @top_srcdir@
 common_cflags = $(PYGOBJECT_CFLAGS) $(GST_CFLAGS) $(GST_OPTION_CFLAGS) -fno-strict-aliasing
 common_libadd = $(GST_LIBS) $(GST_OPTION_LIBS)
-common_ldflags = -module -avoid-version
+common_ldflags = -module -avoid-version `python-config --libs`
 
 # we install everything in pyexecdir; otherwise you end up with a mess for
 # multilib
@@ -431,8 +431,10 @@
 # GStreamer bindings
 _gst_la_CFLAGS = $(common_cflags)
 _gst_la_LIBADD = $(common_libadd) $(GST_BASE_LIBS)
+
 _gst_la_LDFLAGS = $(common_ldflags) -export-symbols-regex "^(init_gst|_PyGObject_API).*" \
-	$(GST_BASE_LIBS) $(GST_CONTROLLER_LIBS) $(GST_NET_LIBS) $(GST_DP_LIBS)
+				  $(GST_BASE_LIBS) $(GST_CONTROLLER_LIBS) $(GST_NET_LIBS) $(GST_DP_LIBS)
+
 
 _gst_la_SOURCES = \
 	gst-argtypes.c 		\
@@ -693,7 +695,7 @@
 @am__fastdepCC_FALSE@	$(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(_gst_la_CFLAGS) $(CFLAGS) -c -o _gst_la-pygstexception.lo `test -f 'pygstexception.c' || echo '$(srcdir)/'`pygstexception.c
 
 _gst_la-gst.lo: gst.c
-@am__fastdepCC_TRUE@	$(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(_gst_la_CFLAGS) $(CFLAGS) -MT _gst_la-gst.lo -MD -MP -MF $(DEPDIR)/_gst_la-gst.Tpo -c -o _gst_la-gst.lo `test -f 'gst.c' || echo '$(srcdir)/'`gst.c
+@am__fastdepCC_TRUE@	$(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(_gst_la_CFLAGS) $(CFLAGS) -lxml2 -MT _gst_la-gst.lo -MD -MP -MF $(DEPDIR)/_gst_la-gst.Tpo -c -o _gst_la-gst.lo `test -f 'gst.c' || echo '$(srcdir)/'`gst.c
 @am__fastdepCC_TRUE@	mv -f $(DEPDIR)/_gst_la-gst.Tpo $(DEPDIR)/_gst_la-gst.Plo
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	source='gst.c' object='_gst_la-gst.lo' libtool=yes @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
