diff -p -up gtkpod/src/prefs.c.cover gtkpod/src/prefs.c
--- gtkpod/src/prefs.c.cover	2008-12-28 14:59:12.000000000 +0100
+++ gtkpod/src/prefs.c	2009-01-07 21:58:05.000000000 +0100
@@ -283,7 +283,7 @@ static void set_default_preferences()
     prefs_set_string("parsetags_template", "%a - %A/%T %t.mp3;%t.wav");
     prefs_set_int("coverart_apic", TRUE);
     prefs_set_int("coverart_file", TRUE);
-    prefs_set_string("coverart_template", "%A;folder.jpg");
+    prefs_set_string("coverart_template", "%A;folder.jpg;cover.jpg;cover.png;cover.gif");
     prefs_set_string("video_thumbnailer_prog", "totem-video-thumbnailer %f %o");
     prefs_set_int("mserv_use", FALSE);
     prefs_set_string("mserv_username", "");
