--- src/loaders.c.orig	2008-12-11 18:06:15.000000000 +0200
+++ src/loaders.c	2008-12-11 18:17:02.000000000 +0200
@@ -202,13 +202,13 @@
   /* HACK hard-coded for Debian (and current exact font names): */ 
 
   if (strncmp(font_name, "AndikaDesRevG.ttf", FNLEN ) == 0)
-    sprintf(fn, "/usr/share/fonts/truetype/ttf-sil-andika/AndikaDesRevG.ttf");
+    sprintf(fn, "/usr/share/tuxtype/fonts/AndikaDesRevG.ttf");
   else if (strncmp(font_name, "DoulosSILR.ttf", FNLEN ) == 0)
-    sprintf(fn, "/usr/share/fonts/truetype/ttf-sil-doulos/DoulosSILR.ttf");
+    sprintf(fn, "/usr/share/tuxtype/fonts/DoulosSILR.ttf");
   else if (strncmp(font_name, "lohit_hi.ttf", FNLEN ) == 0)
-    sprintf(fn, "/usr/share/fonts/truetype/ttf-devanagari-fonts/lohit_hi.ttf");
+    sprintf(fn, "/usr/share/tuxtype/fonts/lohit_hi.ttf");
   else if (strncmp(font_name, "Rachana_w01.ttf", FNLEN ) == 0)
-    sprintf(fn, "/usr/share/fonts/truetype/ttf-malayalam-fonts/Rachana_w01.ttf");
+    sprintf(fn, "/usr/share/tuxtype/fonts/Rachana_w01.ttf");
 
 
 
