--- setup.py	2007-07-19 08:24:47.000000000 +0300
+++ setup2.py	2007-08-29 10:10:32.000000000 +0300
@@ -69,32 +69,32 @@
 
 
 # rapyd:
-os.chdir(os.getcwd() + os.sep + 'rapyd-0.1.1')
-rc = os.popen('python setup.py install', 'w') 
-rc.close()
-os.chdir('..')
+#os.chdir(os.getcwd() + os.sep + 'rapyd-0.1.1')
+#rc = os.popen('python setup.py install', 'w') 
+#rc.close()
+#os.chdir('..')
 
 
 # tk_happy:
-os.chdir(os.getcwd() + os.sep + 'tk_happy-0.3')
-rc = os.popen('python setup.py install', 'w') 
-rc.close()
-os.chdir('..')
+#os.chdir(os.getcwd() + os.sep + 'tk_happy-0.3')
+#rc = os.popen('python setup.py install', 'w') 
+#rc.close()
+#os.chdir('..')
 
 #Pmw:
-try:
-    import Pmw
-except:
-    os.chdir(os.getcwd() + os.sep + 'Pmw-1.1.2')
-    rc = os.popen('python setup.py install', 'w') 
-    rc.close()
-    os.chdir('..')
+#try:
+#    import Pmw
+#except:
+#    os.chdir(os.getcwd() + os.sep + 'Pmw-1.1.2')
+#    rc = os.popen('python setup.py install', 'w') 
+#    rc.close()
+#    os.chdir('..')
 
 #pychecker:
-try:
-    import pychecker
-except:
-    os.chdir(os.getcwd() + os.sep + 'pychecker-0.8.17')
-    rc = os.popen('python setup.py install', 'w') 
-    rc.close()
-    os.chdir('..')
+#try:
+#    import pychecker
+#except:
+#    os.chdir(os.getcwd() + os.sep + 'pychecker-0.8.17')
+#    rc = os.popen('python setup.py install', 'w') 
+#    rc.close()
+#    os.chdir('..')
