diff -Nuar libsift-1.8.orig/src/Makefile libsift-1.8/src/Makefile
--- libsift-1.8.orig/src/Makefile	2005-10-31 14:48:14.000000000 +0200
+++ libsift-1.8/src/Makefile	2006-12-05 11:58:31.000000000 +0200
@@ -11,7 +11,7 @@
 LIBS_WIN=/r:gtk-sharp.dll /r:System.Drawing.dll /r:ICSharpCode.SharpZipLib.dll \
 	/r:glib-sharp.dll /r:gdk-sharp.dll
 MAKE_MONO=
-LIBS_MONO=/pkg:gtk-sharp /r:System.Drawing /r:ICSharpCode.SharpZipLib
+LIBS_MONO=/pkg:gtk-sharp-2.0 /r:System.Drawing /r:ICSharpCode.SharpZipLib
 
 ifeq ($(windows),yes)
 	CSC=$(CSC_WINDOWS)
