--- src/kstreamripper.desktop	2008-12-31 08:23:13.000000000 +0200
+++ src/kstreamripper.desktop	2008-12-31 08:28:41.000000000 +0200
@@ -1,16 +1,12 @@
 [Desktop Entry]
+Version=1.0
 Encoding=UTF-8
 Name=KStreamRipper
-Name[xx]=xxKStreamRipperxx
+GenericName=Streamripper Frontend
+GenericName[de]=Frontend für streamripper
+GenericName[tr]=Streamripper Önyüzü
 Exec=kstreamripper
 Icon=kstreamripper
 Type=Application
-Comment=A simple KDE Application
-Comment[da]=Et simpelt KDE program
-Comment[es]=Una aplicación de KDE sencilla
-Comment[hu]=Egyszerű KDE-alkalmazás
-Comment[pt]=Uma aplicação KDE simples
-Comment[pt_BR]=Um simples Aplicativo do KDE
-Comment[sl]=Preprost program za KDE
-Comment[sv]=Ett enkelt KDE-program
-Comment[xx]=xxA simple KDE Applicationxx
+Terminal=false
+Categories=AudioVideo;Audio;Tuner;Recorder;
