Index: plugin.c
===================================================================
--- xfce4-mixer/panel-plugin/plugin.c	(révision 23116)
+++ xfce4-mixer/panel-plugin/plugin.c	(copie de travail)
@@ -23,6 +23,7 @@
 
 #define PLUGIN_NAME "xfce4-mixer-plugin"
 #define MIXER_RC_GROUP "mixer-plugin"
+#define BORDER 8
 
 /* Panel Plugin Interface */
 static void mixer_construct (XfcePanelPlugin *plugin);
@@ -65,11 +66,11 @@
 	if (orientation == GTK_ORIENTATION_VERTICAL) {
 		gtk_widget_set_size_request (GTK_WIDGET(plugin), size, size);
 		gtk_widget_set_size_request (GTK_WIDGET(mixer->ib), -1, -1);
-		gtk_widget_set_size_request (GTK_WIDGET(mixer->slider), slider_width, -1);
+		gtk_widget_set_size_request (GTK_WIDGET(mixer->slider), BORDER, size - BORDER);
 	} else {
 		gtk_widget_set_size_request (GTK_WIDGET(plugin), -1, size);
 		gtk_widget_set_size_request (GTK_WIDGET(mixer->ib), size, -1);
-		gtk_widget_set_size_request (GTK_WIDGET(mixer->slider), slider_width, -1);
+		gtk_widget_set_size_request (GTK_WIDGET(mixer->slider), BORDER, size - BORDER);
 	}
 }
 
@@ -438,6 +439,7 @@
 	t_mixer *mixer;
 	GdkPixbuf *pb;
 	GtkWidget *align;
+    static GtkWidget *box;
 	
 	mixer = g_new0 (t_mixer, 1);
 
@@ -463,6 +465,12 @@
 
 	gtk_box_pack_start (GTK_BOX (mixer->box), GTK_WIDGET (mixer->ib), TRUE, TRUE, 0);
 
+    if (GTK_IS_WIDGET (box))
+        gtk_widget_destroy (box);
+    box = gtk_hbox_new (FALSE, 0);
+    gtk_container_set_border_width (GTK_CONTAINER (box), BORDER / 2);
+    gtk_widget_show (box);
+
 	mixer->slider = XFCE_MIXER_CONTROL (xfce_mixer_slider_tiny_new ());
 	g_signal_connect (
 		G_OBJECT (mixer->slider), "notify::value", 
@@ -470,7 +478,8 @@
 	);
 	
 	gtk_widget_show (GTK_WIDGET (mixer->slider));
-	gtk_box_pack_start (GTK_BOX (mixer->box), GTK_WIDGET (mixer->slider), FALSE, TRUE, 0);
+	gtk_box_pack_start (GTK_BOX (box), GTK_WIDGET (mixer->slider), FALSE, FALSE, 0);
+	gtk_box_pack_start (GTK_BOX (mixer->box), GTK_WIDGET (box), FALSE, FALSE, 0);
 		
 	g_signal_connect_swapped (
 		G_OBJECT (mixer->ib), "scroll-event", 
