diff -Naur cinepaint-0.21-0/cinepaint.desktop cinepaint-0.21-0-yeni/cinepaint.desktop
--- cinepaint-0.21-0/cinepaint.desktop	2006-01-27 15:31:52.000000000 +0200
+++ cinepaint-0.21-0-yeni/cinepaint.desktop	2006-05-11 21:16:52.000000000 +0300
@@ -2,12 +2,12 @@
 Encoding=UTF-8
 Name=CinePaint
 Name[de]=CinePaint
-
+Name[tr]=CinePaint
 GenericName=CinePaint
 
 Comment=Edit high resolution and HDR pictures
 Comment[de]=Bearbeiten Sie farblich und geometrisch hochauflösende Bilder
-
+Comment[tr]=Yüksek çözünürlükteki resimler üzerinde renkli ve geometrik çalışın
 Type=Application
 Categories=Application;Graphics;RasterGraphics;
 
