--- mysql-workbench/source/lua/Workbench.lua.orig	2008-05-26 21:40:58.000000000 +0300
+++ mysql-workbench/source/lua/Workbench.lua	2008-05-26 21:41:09.000000000 +0300
@@ -1683,10 +1683,10 @@
   local x= grtV.toLua(args[1])
   local y= grtV.toLua(args[2])
   local curView= grtV.getGlobal("/workbench/model/currentView")
-  local lowestX= 9999999999.0
-  local lowestY= 9999999999.0
-  local highestX= 0.0
-  local highestY= 0.0
+  local lowestX= 9999999999
+  local lowestY= 9999999999
+  local highestX= 0
+  local highestY= 0
 
   local i, c
 
--- mysql-workbench/source/lua/WorkbenchImport.lua.orig	2008-05-26 21:41:18.000000000 +0300
+++ mysql-workbench/source/lua/WorkbenchImport.lua	2008-05-26 21:41:53.000000000 +0300
@@ -55,7 +55,7 @@
 function importDbd4(args)
   local filename= grtV.toLua(args[1])
   local p
-  local xF= 1.5
+  local xF= 1 + 1 / 2
   local yF= 2
   
   local callbacks= {
