diff -Naur fusion-icon-0.0_20071028.orig/FusionIcon/interface.py fusion-icon-0.0_20071028/FusionIcon/interface.py
--- fusion-icon-0.0_20071028.orig/FusionIcon/interface.py	2007-10-03 15:28:26.000000000 +0300
+++ fusion-icon-0.0_20071028/FusionIcon/interface.py	2007-10-30 12:54:27.000000000 +0200
@@ -73,14 +73,14 @@
 #		elif 'qt3' in interfaces and env.desktop == 'kde':
 #			chosen_interface = 'qt3'
 
-		if 'gtk' in interfaces:
-			chosen_interface = 'gtk'
+		if 'qt4' in interfaces:
+			chosen_interface = 'qt4'
 
 		# try qt* for non-kde:
-		elif 'qt4' in interfaces:
-			chosen_interface = 'qt4'
 		elif 'qt3' in interfaces:
 			chosen_interface = 'qt3'
+		elif 'gtk' in interfaces:
+			chosen_interface = 'gtk'
 
 		# interfaces is empty
 		else:
diff -Naur fusion-icon-0.0_20071028.orig/FusionIcon/interface_qt4/main.py fusion-icon-0.0_20071028/FusionIcon/interface_qt4/main.py
--- fusion-icon-0.0_20071028.orig/FusionIcon/interface_qt4/main.py	2007-10-03 15:28:26.000000000 +0300
+++ fusion-icon-0.0_20071028/FusionIcon/interface_qt4/main.py	2007-10-30 12:55:17.000000000 +0200
@@ -19,6 +19,7 @@
 import sys, os, time
 from PyQt4 import QtGui, QtCore
 from FusionIcon.start import wms, apps, options, decorators, init
+from FusionIcon.execute import run
 
 class Build(QtGui.QApplication):
 	def reload_wm(self):
