--- configure.orig	2006-08-20 10:35:58.000000000 +0300
+++ configure	2006-11-01 22:44:47.000000000 +0200
@@ -7253,7 +7253,7 @@
 ac_lib_var=`echo readline'_'readline | sed 'y%./+-%__p_%'`
 
 ac_save_LIBS="$LIBS"
-LIBS="-lreadline  $LIBS"
+LIBS="-lreadline -lncurses  $LIBS"
 cat > conftest.$ac_ext <<EOF
 #line 7259 "configure"
 #include "confdefs.h"
@@ -7280,7 +7280,7 @@
 
 if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
   echo "$ac_t""yes" 1>&6
-  READLINELIB="$READLINELIB -lreadline "
+  READLINELIB="$READLINELIB -lreadline -lncurses "
 else
   echo "$ac_t""no" 1>&6
 
