--- configure.in.orig   2008-11-24 15:10:28.000000000 +0100
+++ configure.in        2009-03-21 16:19:52.485799982 +0100
@@ -462,9 +462,9 @@
        valgrind="yes"
 fi
 
-AC_CHECK_LIB(bfd, bfd_get_error, [PLUGIN_VALGRIND_LIBS="-Wl,-Bstatic -lbfd -liberty -Wl,-Bdynamic"],
+AC_CHECK_LIB(bfd, bfd_get_error, [PLUGIN_VALGRIND_LIBS="-Wl,-Bstatic -lbfd -liberty -lz -Wl,-Bdynamic"],
    valgrind="no",
-  -liberty)
+  [-liberty -lz])
 
 AC_SUBST(PLUGIN_VALGRIND_LIBS)
 AM_CONDITIONAL(HAVE_PLUGIN_VALGRIND, [test x$valgrind = xyes])