--- src/kcm/kcm_knemo.desktop	2006-08-27 17:51:49.000000000 +0300
+++ src/kcm/kcm_knemo.desktop	2006-08-28 07:08:35.000000000 +0300
@@ -78,4 +78,4 @@
 Keywords[sv]=knemo,nätverksövervakning,WLAN-övervakning,WIFI-övervakning,Ethernet-övervakning,trådlös övervakning,systembricka
 Keywords[xx]=xxknemo,network monitor,wlan monitor,wifi monitor,ethernet monitor,wireless monitor,systrayxx
 
-Categories=Qt;KDE;X-KDE-settings-network
+Categories=Qt;KDE;X-KDE-settings-network;Network
