--- src/xfce4-dict.desktop.in.orig	2008-07-06 18:03:00.000000000 +0300
+++ src/xfce4-dict.desktop.in	2008-07-06 18:04:59.000000000 +0300
@@ -1,9 +1,12 @@
 [Desktop Entry]
 Type=Application
 Version=1.0
-_Name=Xfce4 Dictionary
-_GenericName=Dictionary Client
-_Comment=A client program to query different dictionaries.
+Name=Xfce4 Dictionary
+Name[tr]=Xfce4 Sözlük
+GenericName=Dictionary Client
+GenericName[tr]=Sözlük İstemcisi
+Comment=A client program to query different dictionaries.
+Comment[tr]=Farklı sözlükleri sorgulamak için istemci programdır.
 Icon=xfce4-dict
 Exec=xfce4-dict
 Terminal=false
