--- panel-plugin/xfce4-dict-plugin.desktop.in.in.orig	2008-07-06 18:06:04.000000000 +0300
+++ panel-plugin/xfce4-dict-plugin.desktop.in.in	2008-07-06 18:06:45.000000000 +0300
@@ -1,7 +1,9 @@
 [Xfce Panel]
 Type=X-XFCE-PanelPlugin
 Encoding=UTF-8
-_Name=Dictionary
-_Comment=A plugin to query different dictionaries.
+Name=Dictionary
+Name[tr]=Sözlük
+Comment=A plugin to query different dictionaries
+Comment[tr]=Farklı sözlükleri sorgulamaya yarayan eklenti
 Icon=xfce4-dict
 X-XFCE-Exec=@PLUGIN_PATH@/xfce4-dict-plugin
