--- src/main.cpp	2007-05-23 13:54:56.000000000 +0300
+++ src/main2.cpp	2007-06-30 21:48:34.000000000 +0300
@@ -116,7 +116,7 @@
 localedirfile = QString("%1/locale/edit_%2.qm").arg(QCoreApplication::applicationDirPath()).arg(UserLanguage());
 #endif
 #if defined Q_WS_X11
-localedirfile = QString("%1/locale/edit_%2.qm").arg(WORK_CACHEDIR).arg(UserLanguage()); 
+localedirfile = QString("/usr/share/qt_xhtmledit/locale/edit_%1.qm").arg(UserLanguage()); 
 #endif
     
     
