--- src/plugins/FileDialogs/QmmpFileDialog/qmmpfiledialogimpl.cpp.str	2009-05-04 07:57:28.000000000 +0200
+++ src/plugins/FileDialogs/QmmpFileDialog/qmmpfiledialogimpl.cpp	2009-05-04 07:59:00.000000000 +0200
@@ -102,7 +102,7 @@
     if (m_mode == FileDialog::SaveFile)
     {
         l << m_model->filePath(fileListView->rootIndex()) + "/" + fileNameLineEdit->text();
-        qDebug(qPrintable(l[0]));
+        qDebug("%s",qPrintable(l[0]));
     }
     else
     {
--- src/plugins/General/scrobbler/scrobbler.cpp.str	2009-05-04 07:56:48.000000000 +0200
+++ src/plugins/General/scrobbler/scrobbler.cpp	2009-05-04 07:57:00.000000000 +0200
@@ -261,7 +261,7 @@
     header.setValue("Accept", "*/*");
     header.setContentLength(QUrl::toPercentEncoding(body,":/[]&=").size());
     qDebug("Scrobbler: submit request header");
-    qDebug(qPrintable(header.toString()));
+    qDebug("%s",qPrintable(header.toString()));
     qDebug("*****************************");
     m_submitid = m_http->request(header, QUrl::toPercentEncoding(body,":/[]&="));
 }
--- src/plugins/Input/vorbis/decoder_vorbis.cpp.str	2009-05-04 08:01:34.000000000 +0200
+++ src/plugins/Input/vorbis/decoder_vorbis.cpp	2009-05-04 08:01:49.000000000 +0200
@@ -192,7 +192,7 @@
     {
         if (! input()->open(QIODevice::ReadOnly))
         {
-            qWarning(qPrintable("DecoderVorbis: failed to open input. " +
+            qWarning("%s",qPrintable("DecoderVorbis: failed to open input. " +
                                 input()->errorString () + "."));
             return FALSE;
         }
--- src/plugins/Output/alsa/settingsdialog.cpp.str	2009-05-04 08:00:58.000000000 +0200
+++ src/plugins/Output/alsa/settingsdialog.cpp	2009-05-04 08:01:11.000000000 +0200
@@ -145,7 +145,7 @@
         QString str;
         str =  QString(card_name) + ": "+
                snd_pcm_info_get_name(pcm_info)+" ("+device+")";
-        qDebug(str.toAscii());
+        qDebug("%s",str.toAscii());
         ui.deviceComboBox->addItem(str);
     }
 
--- src/ui/mainwindow.cpp.str	2009-05-04 07:54:41.000000000 +0200
+++ src/ui/mainwindow.cpp	2009-05-04 07:54:55.000000000 +0200
@@ -704,7 +704,7 @@
             QStringList full_path_list;
             foreach(QString s,slist)
             {
-                qWarning(qPrintable(cwd + "/" + s));
+                qWarning("%s",qPrintable(cwd + "/" + s));
                 if (s.left(1) == "/")   //is it absolute path?
                     full_path_list << s;
                 else
