diff -Nuar qmmp-0.3.0.orig/src/plugins/FileDialogs/QmmpFileDialog/translations/qmmp_file_dialog_plugin_tr.ts qmmp-0.3.0/src/plugins/FileDialogs/QmmpFileDialog/translations/qmmp_file_dialog_plugin_tr.ts
--- qmmp-0.3.0.orig/src/plugins/FileDialogs/QmmpFileDialog/translations/qmmp_file_dialog_plugin_tr.ts	2009-01-23 19:57:34.000000000 +0200
+++ qmmp-0.3.0/src/plugins/FileDialogs/QmmpFileDialog/translations/qmmp_file_dialog_plugin_tr.ts	2009-07-23 13:13:53.553752316 +0300
@@ -1,56 +1,57 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!DOCTYPE TS><TS version="1.1">
+<!DOCTYPE TS>
+<TS version="2.0" language="tr_TR">
 <context>
     <name>QmmpFileDialog</name>
     <message>
         <location filename="../qmmpfiledialog.ui" line="13"/>
         <source>Add Files</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Dosya Ekle</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialog.ui" line="37"/>
         <source>Up</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Yukarı</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialog.ui" line="97"/>
         <source>...</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">...</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialog.ui" line="50"/>
         <source>List view</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Liste görünümü</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialog.ui" line="69"/>
         <source>Detailed view</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Detaylı görünüm</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialog.ui" line="94"/>
         <source>Close dialog on add</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Ekledikten sonra diyoloğu kapat</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialog.ui" line="177"/>
         <source>File name:</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Dosya adı:</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialog.ui" line="193"/>
         <source>Add</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Ekle</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialog.ui" line="200"/>
         <source>Files of type:</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Dosya tipi:</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialog.ui" line="226"/>
         <source>Close</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Kapat</translation>
     </message>
 </context>
 <context>
@@ -58,24 +59,26 @@
     <message>
         <location filename="../qmmpfiledialog.cpp" line="118"/>
         <source>Qmmp File Dialog</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Qmmp Dosya Diyaloğu</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialog.cpp" line="117"/>
         <source>About Qmmp File Dialog</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Qmmp Dosya Diyaloğu Hakkında</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialog.cpp" line="121"/>
         <source>Writen by:
 Vladimir Kuznetsov &lt;vovanec@gmail.com&gt;
 Ilya Kotov &lt;forkotov02@hotmail.ru&gt;</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Yazan:
+Vladimir Kuznetsov &lt;vovanec@gmail.com&gt;
+Ilya Kotov &lt;forkotov02@hotmail.ru&gt;</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialog.cpp" line="122"/>
         <source>Some code is copied from the Qt library</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Bazı kodlar Qt kitaplığından kopyalandı</translation>
     </message>
 </context>
 <context>
@@ -83,27 +86,27 @@
     <message>
         <location filename="../qmmpfiledialogimpl.cpp" line="241"/>
         <source>Add</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Ekle</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialogimpl.cpp" line="254"/>
         <source>Save</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Kaydet</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialogimpl.cpp" line="270"/>
         <source>Directories</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Dizinler</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialogimpl.cpp" line="439"/>
         <source>already exists.</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">zaten mevcut.</translation>
     </message>
     <message>
         <location filename="../qmmpfiledialogimpl.cpp" line="440"/>
         <source>Do you want to replace it?</source>
-        <translation type="unfinished"></translation>
+        <translation type="unfinished">Yerini değiştirmek ister misiniz?</translation>
     </message>
 </context>
 </TS>
diff -Nuar qmmp-0.3.0.orig/src/qmmpui/translations/libqmmpui_tr.ts qmmp-0.3.0/src/qmmpui/translations/libqmmpui_tr.ts
--- qmmp-0.3.0.orig/src/qmmpui/translations/libqmmpui_tr.ts	2009-01-23 19:56:32.000000000 +0200
+++ qmmp-0.3.0/src/qmmpui/translations/libqmmpui_tr.ts	2009-07-23 13:13:53.553752316 +0300
@@ -1,36 +1,37 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!DOCTYPE TS><TS version="1.1">
+<!DOCTYPE TS>
+<TS version="2.0" language="tr_TR">
 <context>
     <name>PlayListModel</name>
     <message>
         <location filename="../playlistmodel.cpp" line="341"/>
         <source>Url:</source>
-        <translation type="unfinished"></translation>
+        <translation>Url:</translation>
     </message>
     <message>
         <location filename="../playlistmodel.cpp" line="342"/>
         <source>Title:</source>
-        <translation type="unfinished"></translation>
+        <translation>Başlık:</translation>
     </message>
     <message>
         <location filename="../playlistmodel.cpp" line="343"/>
         <source>Artist:</source>
-        <translation type="unfinished"></translation>
+        <translation>Artist:</translation>
     </message>
     <message>
         <location filename="../playlistmodel.cpp" line="344"/>
         <source>Album:</source>
-        <translation type="unfinished"></translation>
+        <translation>Albüm:</translation>
     </message>
     <message>
         <location filename="../playlistmodel.cpp" line="345"/>
         <source>Genre:</source>
-        <translation type="unfinished"></translation>
+        <translation>Tür:</translation>
     </message>
     <message>
         <location filename="../playlistmodel.cpp" line="346"/>
         <source>Comment:</source>
-        <translation type="unfinished"></translation>
+        <translation>Yorum:</translation>
     </message>
 </context>
 <context>
@@ -38,7 +39,7 @@
     <message>
         <location filename="../qtfiledialog.cpp" line="33"/>
         <source>Qt File Dialog</source>
-        <translation type="unfinished"></translation>
+        <translation>Qt Dosya Diyaloğu</translation>
     </message>
 </context>
 </TS>
diff -Nuar qmmp-0.3.0.orig/src/ui/stuff.qrc qmmp-0.3.0/src/ui/stuff.qrc
--- qmmp-0.3.0.orig/src/ui/stuff.qrc	2009-05-12 11:56:46.000000000 +0300
+++ qmmp-0.3.0/src/ui/stuff.qrc	2009-07-23 13:46:41.261511795 +0300
@@ -43,6 +43,11 @@
     <file>txt/translators_it.txt</file>
     <file>txt/description_it.txt</file>
 
+    <file>txt/authors_tr.txt</file>
+    <file>txt/thanks_tr.txt</file>
+    <file>txt/translators_tr.txt</file>
+    <file>txt/description_tr.txt</file>
+
     <file>default/balance.png</file>
     <file>default/eqmain.png</file>
     <file>default/numbers.png</file>
diff -Nuar qmmp-0.3.0.orig/src/ui/translations/qmmp_tr.ts qmmp-0.3.0/src/ui/translations/qmmp_tr.ts
--- qmmp-0.3.0.orig/src/ui/translations/qmmp_tr.ts	2009-03-10 12:26:26.000000000 +0200
+++ qmmp-0.3.0/src/ui/translations/qmmp_tr.ts	2009-07-23 13:21:16.125753411 +0300
@@ -31,62 +31,62 @@
     <message>
         <location filename="../aboutdialog.cpp" line="61"/>
         <source>:/txt/authors_en.txt</source>
-        <translation type="unfinished"></translation>
+        <translation>:/txt/authors_tr.txt</translation>
     </message>
     <message>
         <location filename="../aboutdialog.cpp" line="62"/>
         <source>:/txt/thanks_en.txt</source>
-        <translation type="unfinished"></translation>
+        <translation>:/txt/thanks_tr.txt</translation>
     </message>
     <message>
         <location filename="../aboutdialog.cpp" line="81"/>
         <source>Qt-based Multimedia Player (Qmmp)</source>
-        <translation type="unfinished"></translation>
+        <translation>Qt tabanlı Çokluortam Oynatıcısı (Qmmp)</translation>
     </message>
     <message>
         <location filename="../aboutdialog.cpp" line="82"/>
         <source>Version:</source>
-        <translation type="unfinished"></translation>
+        <translation>Sürüm:</translation>
     </message>
     <message>
         <location filename="../aboutdialog.cpp" line="83"/>
         <source>:txt/description_en.txt</source>
-        <translation type="unfinished"></translation>
+        <translation>:txt/description_tr.txt</translation>
     </message>
     <message>
         <location filename="../aboutdialog.cpp" line="84"/>
         <source>Input plugins:</source>
-        <translation type="unfinished"></translation>
+        <translation>Girdi eklentileri:</translation>
     </message>
     <message>
         <location filename="../aboutdialog.cpp" line="93"/>
         <source>Output plugins:</source>
-        <translation type="unfinished"></translation>
+        <translation>Çıktı eklentileri:</translation>
     </message>
     <message>
         <location filename="../aboutdialog.cpp" line="102"/>
         <source>Visual plugins:</source>
-        <translation type="unfinished"></translation>
+        <translation>Görsel eklentiler:</translation>
     </message>
     <message>
         <location filename="../aboutdialog.cpp" line="111"/>
         <source>Effect plugins:</source>
-        <translation type="unfinished"></translation>
+        <translation>Efekt eklentileri:</translation>
     </message>
     <message>
         <location filename="../aboutdialog.cpp" line="120"/>
         <source>General plugins:</source>
-        <translation type="unfinished"></translation>
+        <translation>Genel eklentiler:</translation>
     </message>
     <message>
         <location filename="../forms/aboutdialog.ui" line="115"/>
         <source>Translators</source>
-        <translation type="unfinished"></translation>
+        <translation>Çevirmenler</translation>
     </message>
     <message>
         <location filename="../aboutdialog.cpp" line="63"/>
         <source>:/txt/translators_en.txt</source>
-        <translation type="unfinished"></translation>
+        <translation>:/txt/translators_tr.txt</translation>
     </message>
 </context>
 <context>
@@ -94,22 +94,22 @@
     <message>
         <location filename="../addurldialog.cpp" line="107"/>
         <source>Error</source>
-        <translation type="unfinished"></translation>
+        <translation>Hata</translation>
     </message>
     <message>
         <location filename="../forms/addurldialog.ui" line="13"/>
         <source>Enter URL to add</source>
-        <translation type="unfinished"></translation>
+        <translation>Eklenecek URL girin</translation>
     </message>
     <message>
         <location filename="../forms/addurldialog.ui" line="51"/>
         <source>&amp;Add</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Ekle</translation>
     </message>
     <message>
         <location filename="../forms/addurldialog.ui" line="58"/>
         <source>&amp;Cancel</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;İptal</translation>
     </message>
 </context>
 <context>
@@ -117,57 +117,57 @@
     <message>
         <location filename="../builtincommandlineoption.cpp" line="68"/>
         <source>Skip forward in playlist</source>
-        <translation type="unfinished"></translation>
+        <translation>İleri atla</translation>
     </message>
     <message>
         <location filename="../builtincommandlineoption.cpp" line="69"/>
         <source>Skip backwards in playlist</source>
-        <translation type="unfinished"></translation>
+        <translation>Geri atla</translation>
     </message>
     <message>
         <location filename="../builtincommandlineoption.cpp" line="62"/>
         <source>Start playing current song</source>
-        <translation type="unfinished"></translation>
+        <translation>Mevcut şarkıyı çalmaya başla</translation>
     </message>
     <message>
         <location filename="../builtincommandlineoption.cpp" line="63"/>
         <source>Pause current song</source>
-        <translation type="unfinished"></translation>
+        <translation>Mevcut şarkıyı duraklat</translation>
     </message>
     <message>
         <location filename="../builtincommandlineoption.cpp" line="64"/>
         <source>Pause if playing, play otherwise</source>
-        <translation type="unfinished"></translation>
+        <translation>Çalıyorsa duraklat, değilse oynat</translation>
     </message>
     <message>
         <location filename="../builtincommandlineoption.cpp" line="65"/>
         <source>Stop current song</source>
-        <translation type="unfinished"></translation>
+        <translation>Mevcut şarkıyı durdur</translation>
     </message>
     <message>
         <location filename="../builtincommandlineoption.cpp" line="67"/>
         <source>Set playback volume(example: qmmp --volume20, qmmp --volume100)</source>
-        <translation type="unfinished"></translation>
+        <translation>Sesi ayarla(örnek: qmmp --volume, qmmp --volume100)</translation>
     </message>
     <message>
         <location filename="../builtincommandlineoption.cpp" line="66"/>
         <source>Display Jump to File dialog</source>
-        <translation type="unfinished"></translation>
+        <translation>Dosyaya atlama diyaloğunu göster</translation>
     </message>
     <message>
         <location filename="../builtincommandlineoption.cpp" line="70"/>
         <source>Show/hide application</source>
-        <translation type="unfinished"></translation>
+        <translation>Uygulamayı göster/gizle</translation>
     </message>
     <message>
         <location filename="../builtincommandlineoption.cpp" line="71"/>
         <source>Display Add File dialog</source>
-        <translation type="unfinished"></translation>
+        <translation>Dosya ekleme diyaloğunu göster</translation>
     </message>
     <message>
         <location filename="../builtincommandlineoption.cpp" line="72"/>
         <source>Display Add Directory dialog</source>
-        <translation type="unfinished"></translation>
+        <translation>Dizin ekleme diyaloğunu göster</translation>
     </message>
 </context>
 <context>
@@ -353,154 +353,154 @@
     <message>
         <location filename="../configdialog.cpp" line="177"/>
         <source>Archived skin</source>
-        <translation type="unfinished"></translation>
+        <translation>Arşivlenmiş kabuk</translation>
     </message>
     <message>
         <location filename="../configdialog.cpp" line="197"/>
         <source>Unarchived skin</source>
-        <translation type="unfinished"></translation>
+        <translation>Arşivlenmemiş kabuk</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="119"/>
         <source>Connectivity</source>
-        <translation type="unfinished"></translation>
+        <translation>Bağlanırlık</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="512"/>
         <source>Visualization</source>
-        <translation type="unfinished"></translation>
+        <translation>Görsellik</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="529"/>
         <source>Effects</source>
-        <translation type="unfinished"></translation>
+        <translation>Efektler</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="546"/>
         <source>General</source>
-        <translation type="unfinished"></translation>
+        <translation>Genel</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="609"/>
         <source>File Dialog</source>
-        <translation type="unfinished"></translation>
+        <translation>Dosya Diyaloğu</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="570"/>
         <source>Audio</source>
-        <translation type="unfinished"></translation>
+        <translation>Ses</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="576"/>
         <source>Use software volume control</source>
-        <translation type="unfinished"></translation>
+        <translation>Yazılımsal ses kontrolünü kullan</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="586"/>
         <source>Visibility Control</source>
-        <translation type="unfinished"></translation>
+        <translation>Görünürlük Kontrolü</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="592"/>
         <source>Hide on close</source>
-        <translation type="unfinished"></translation>
+        <translation>Kapatınca saklan</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="599"/>
         <source>Start hidden</source>
-        <translation type="unfinished"></translation>
+        <translation>Gizli başlat</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="744"/>
         <source>Proxy</source>
-        <translation type="unfinished"></translation>
+        <translation>Vekil sunucu</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="756"/>
         <source>Enable proxy usage</source>
-        <translation type="unfinished"></translation>
+        <translation>Vekil sunucu kullanımını etkinleştir</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="763"/>
         <source>Proxy host name:</source>
-        <translation type="unfinished"></translation>
+        <translation>Vekil sunucu adı:</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="776"/>
         <source>Proxy port:</source>
-        <translation type="unfinished"></translation>
+        <translation>Vekil sunucu portu:</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="789"/>
         <source>Use authentication with proxy</source>
-        <translation type="unfinished"></translation>
+        <translation>Vekil sunucu yetkilendirmesi kullan</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="796"/>
         <source>Proxy user name:</source>
-        <translation type="unfinished"></translation>
+        <translation>Vekil sunucu kullanıcı adı:</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="809"/>
         <source>Proxy password:</source>
-        <translation type="unfinished"></translation>
+        <translation>Vekil sunucu parolası:</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="355"/>
         <source>Convert underscores to blanks</source>
-        <translation type="unfinished"></translation>
+        <translation>Alt çizgileri boşluğa çevir</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="362"/>
         <source>Convert %20 to blanks</source>
-        <translation type="unfinished"></translation>
+        <translation>%20 yi boşluğa çevir</translation>
     </message>
     <message>
         <location filename="../configdialog.cpp" line="645"/>
         <source>Select Skin Files</source>
-        <translation type="unfinished"></translation>
+        <translation>Kabuk Dosyası Seç</translation>
     </message>
     <message>
         <location filename="../configdialog.cpp" line="646"/>
         <source>Skin files</source>
-        <translation type="unfinished"></translation>
+        <translation>Kabuk dosyaları</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="193"/>
         <source>Add...</source>
-        <translation type="unfinished"></translation>
+        <translation>Ekle...</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="206"/>
         <source>Refresh</source>
-        <translation type="unfinished">Yenile</translation>
+        <translation>Yenile</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="369"/>
         <source>Show protocol</source>
-        <translation type="unfinished"></translation>
+        <translation>Protokolü göster</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="641"/>
         <source>Transparency</source>
-        <translation type="unfinished"></translation>
+        <translation>Transparanlık</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="647"/>
         <source>Main window</source>
-        <translation type="unfinished"></translation>
+        <translation>Ana pencere</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="670"/>
         <location filename="../forms/configdialog.ui" line="694"/>
         <location filename="../forms/configdialog.ui" line="718"/>
         <source>0</source>
-        <translation type="unfinished"></translation>
+        <translation>0</translation>
     </message>
     <message>
         <location filename="../forms/configdialog.ui" line="677"/>
         <source>Equalizer</source>
-        <translation type="unfinished"></translation>
+        <translation>Ekolayzır</translation>
     </message>
 </context>
 <context>
@@ -603,62 +603,62 @@
     <message>
         <location filename="../display.cpp" line="61"/>
         <source>Previous</source>
-        <translation type="unfinished"></translation>
+        <translation>Önceki</translation>
     </message>
     <message>
         <location filename="../display.cpp" line="65"/>
         <source>Play</source>
-        <translation type="unfinished"></translation>
+        <translation>Oynat</translation>
     </message>
     <message>
         <location filename="../display.cpp" line="69"/>
         <source>Pause</source>
-        <translation type="unfinished"></translation>
+        <translation>Duraklat</translation>
     </message>
     <message>
         <location filename="../display.cpp" line="73"/>
         <source>Stop</source>
-        <translation type="unfinished"></translation>
+        <translation>Durdur</translation>
     </message>
     <message>
         <location filename="../display.cpp" line="77"/>
         <source>Next</source>
-        <translation type="unfinished"></translation>
+        <translation>Sonraki</translation>
     </message>
     <message>
         <location filename="../display.cpp" line="81"/>
         <source>Add file</source>
-        <translation type="unfinished"></translation>
+        <translation>Dosya ekle</translation>
     </message>
     <message>
         <location filename="../display.cpp" line="96"/>
         <source>Equalizer</source>
-        <translation type="unfinished"></translation>
+        <translation>Ekolayzır</translation>
     </message>
     <message>
         <location filename="../display.cpp" line="101"/>
         <source>Playlist</source>
-        <translation type="unfinished">Çalma Listesi</translation>
+        <translation>Çalma Listesi</translation>
     </message>
     <message>
         <location filename="../display.cpp" line="109"/>
         <source>Repeat playlist</source>
-        <translation type="unfinished"></translation>
+        <translation>Çalma Listesini Yinele</translation>
     </message>
     <message>
         <location filename="../display.cpp" line="113"/>
         <source>Shuffle</source>
-        <translation type="unfinished"></translation>
+        <translation>Rastgele</translation>
     </message>
     <message>
         <location filename="../display.cpp" line="144"/>
         <source>Volume</source>
-        <translation type="unfinished"></translation>
+        <translation>Ses</translation>
     </message>
     <message>
         <location filename="../display.cpp" line="150"/>
         <source>Balance</source>
-        <translation type="unfinished"></translation>
+        <translation>Denge</translation>
     </message>
 </context>
 <context>
@@ -666,82 +666,82 @@
     <message>
         <location filename="../mainvisual.cpp" line="294"/>
         <source>Visualization Mode</source>
-        <translation type="unfinished"></translation>
+        <translation>Görselleştirme Modu</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="297"/>
         <source>Analyzer</source>
-        <translation type="unfinished"></translation>
+        <translation>Çözümleyici</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="298"/>
         <source>Scope</source>
-        <translation type="unfinished"></translation>
+        <translation>Kapsam</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="299"/>
         <source>Off</source>
-        <translation type="unfinished"></translation>
+        <translation>Kapat</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="306"/>
         <source>Analyzer Mode</source>
-        <translation type="unfinished"></translation>
+        <translation>Çözümleyici Modu</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="309"/>
         <source>Normal</source>
-        <translation type="unfinished"></translation>
+        <translation>Normal</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="310"/>
         <source>Fire</source>
-        <translation type="unfinished"></translation>
+        <translation>Ateş</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="311"/>
         <source>Vertical Lines</source>
-        <translation type="unfinished"></translation>
+        <translation>Dikey Satırlar</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="312"/>
         <source>Lines</source>
-        <translation type="unfinished"></translation>
+        <translation>Satırlar</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="313"/>
         <source>Bars</source>
-        <translation type="unfinished"></translation>
+        <translation>Çubuklar</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="326"/>
         <source>Peaks</source>
-        <translation type="unfinished"></translation>
+        <translation>Uçlar</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="330"/>
         <source>Refresh Rate</source>
-        <translation type="unfinished"></translation>
+        <translation>Tazeleme Oranı</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="333"/>
         <source>50 fps</source>
-        <translation type="unfinished"></translation>
+        <translation>50 fps</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="334"/>
         <source>25 fps</source>
-        <translation type="unfinished"></translation>
+        <translation>25 fps</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="335"/>
         <source>10 fps</source>
-        <translation type="unfinished"></translation>
+        <translation>10 fps</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="336"/>
         <source>5 fps</source>
-        <translation type="unfinished"></translation>
+        <translation>5 fps</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="343"/>
@@ -752,31 +752,31 @@
         <location filename="../mainvisual.cpp" line="346"/>
         <location filename="../mainvisual.cpp" line="360"/>
         <source>Slowest</source>
-        <translation type="unfinished"></translation>
+        <translation>En yavaş</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="347"/>
         <location filename="../mainvisual.cpp" line="361"/>
         <source>Slow</source>
-        <translation type="unfinished"></translation>
+        <translation>Yavaş</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="348"/>
         <location filename="../mainvisual.cpp" line="362"/>
         <source>Medium</source>
-        <translation type="unfinished"></translation>
+        <translation>Orta</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="349"/>
         <location filename="../mainvisual.cpp" line="363"/>
         <source>Fast</source>
-        <translation type="unfinished"></translation>
+        <translation>Hızlı</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="350"/>
         <location filename="../mainvisual.cpp" line="364"/>
         <source>Fastest</source>
-        <translation type="unfinished"></translation>
+        <translation>En hızlı</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="357"/>
@@ -786,12 +786,12 @@
     <message>
         <location filename="../mainvisual.cpp" line="370"/>
         <source>Background</source>
-        <translation type="unfinished"></translation>
+        <translation>Arkaplan</translation>
     </message>
     <message>
         <location filename="../mainvisual.cpp" line="371"/>
         <source>Transparent</source>
-        <translation type="unfinished"></translation>
+        <translation>Transparan</translation>
     </message>
 </context>
 <context>
@@ -916,57 +916,57 @@
     <message>
         <location filename="../mainwindow.cpp" line="422"/>
         <source>Space</source>
-        <translation type="unfinished"></translation>
+        <translation>Boşluk</translation>
     </message>
     <message>
         <location filename="../mainwindow.cpp" line="449"/>
         <source>&amp;About Qt</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Qt Hakkında</translation>
     </message>
     <message>
         <location filename="../mainwindow.cpp" line="422"/>
         <source>&amp;Play/Pause</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Oynat/Duraklat</translation>
     </message>
     <message>
         <location filename="../mainwindow.cpp" line="286"/>
         <source>All Supported Bitstreams</source>
-        <translation type="unfinished"></translation>
+        <translation>Tüm Desteklenen Bitstreamler</translation>
     </message>
     <message>
         <location filename="../mainwindow.cpp" line="425"/>
         <source>&amp;Repeat Track</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Parçayı Yinele</translation>
     </message>
     <message>
         <location filename="../mainwindow.cpp" line="426"/>
         <source>&amp;Shuffle</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Rastgele</translation>
     </message>
     <message>
         <location filename="../mainwindow.cpp" line="430"/>
         <source>R</source>
-        <translation type="unfinished"></translation>
+        <translation>R</translation>
     </message>
     <message>
         <location filename="../mainwindow.cpp" line="431"/>
         <source>Ctrl+R</source>
-        <translation type="unfinished"></translation>
+        <translation>Ctrl+R</translation>
     </message>
     <message>
         <location filename="../mainwindow.cpp" line="432"/>
         <source>S</source>
-        <translation type="unfinished"></translation>
+        <translation>S</translation>
     </message>
     <message>
         <location filename="../mainwindow.cpp" line="424"/>
         <source>&amp;Repeat Playlist</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Çalma Listesini Yinele</translation>
     </message>
     <message>
         <location filename="../mainwindow.cpp" line="444"/>
         <source>Tools</source>
-        <translation type="unfinished"></translation>
+        <translation>Araçlar</translation>
     </message>
 </context>
 <context>
@@ -1118,33 +1118,33 @@
     <message>
         <location filename="../playlist.cpp" line="147"/>
         <source>&amp;Add Url</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Url Ekle</translation>
     </message>
     <message>
         <location filename="../playlist.cpp" line="148"/>
         <source>U</source>
-        <translation type="unfinished"></translation>
+        <translation>U</translation>
     </message>
     <message>
         <location filename="../playlist.cpp" line="201"/>
         <location filename="../playlist.cpp" line="228"/>
         <source>By Track Number</source>
-        <translation type="unfinished"></translation>
+        <translation>Parça Numarasına Göre</translation>
     </message>
     <message>
         <location filename="../playlist.cpp" line="247"/>
         <source>&amp;Queue</source>
-        <translation type="unfinished">&amp;Kuyruğa ekle</translation>
+        <translation>&amp;Kuyruğa ekle</translation>
     </message>
     <message>
         <location filename="../playlist.cpp" line="247"/>
         <source>Q</source>
-        <translation type="unfinished">Q</translation>
+        <translation>Q</translation>
     </message>
     <message>
         <location filename="../playlist.cpp" line="244"/>
         <source>Actions</source>
-        <translation type="unfinished"></translation>
+        <translation>Eylemler</translation>
     </message>
 </context>
 <context>
@@ -1180,42 +1180,42 @@
     <message>
         <location filename="../qmmpstarter.cpp" line="142"/>
         <source>Usage: qmmp [options] [files]</source>
-        <translation type="unfinished"></translation>
+        <translation>Kullanım:qmmp [seçenek] [dosyalar]</translation>
     </message>
     <message>
         <location filename="../qmmpstarter.cpp" line="143"/>
         <source>Options:</source>
-        <translation type="unfinished"></translation>
+        <translation>Seçenekler:</translation>
     </message>
     <message>
         <location filename="../qmmpstarter.cpp" line="149"/>
         <source>Print version number and exit</source>
-        <translation type="unfinished"></translation>
+        <translation>Sürüm numarasını yazdır ve çık</translation>
     </message>
     <message>
         <location filename="../qmmpstarter.cpp" line="150"/>
         <source>Ideas, patches, bugreports send to forkotov02@hotmail.ru</source>
-        <translation type="unfinished"></translation>
+        <translation>Fikirleinizi, yamalarınızı, hata raporlarınızı forkotov02@hotmail.ru adresine gönderin</translation>
     </message>
     <message>
         <location filename="../qmmpstarter.cpp" line="148"/>
         <source>Display this text and exit</source>
-        <translation type="unfinished"></translation>
+        <translation>Bu metni göster ve çık</translation>
     </message>
     <message>
         <location filename="../qmmpstarter.cpp" line="155"/>
         <source>QMMP version:</source>
-        <translation type="unfinished"></translation>
+        <translation>QMMP sürümü:</translation>
     </message>
     <message>
         <location filename="../qmmpstarter.cpp" line="145"/>
         <source>Don&apos;t clear the playlist</source>
-        <translation type="unfinished"></translation>
+        <translation>Çalma listesini temizleme</translation>
     </message>
     <message>
         <location filename="../qmmpstarter.cpp" line="156"/>
         <source>Qt version:</source>
-        <translation type="unfinished"></translation>
+        <translation>Qt sürümü:</translation>
     </message>
 </context>
 <context>
@@ -1223,12 +1223,12 @@
     <message>
         <location filename="../textscroller.cpp" line="176"/>
         <source>Buffering:</source>
-        <translation type="unfinished"></translation>
+        <translation>Arabelleğe alınıyor:</translation>
     </message>
     <message>
         <location filename="../textscroller.cpp" line="62"/>
         <source>Autoscroll Songname</source>
-        <translation type="unfinished"></translation>
+        <translation>Şarkı Adını Otomatik Kaydır</translation>
     </message>
 </context>
 <context>
@@ -1236,7 +1236,7 @@
     <message>
         <location filename="../visualmenu.cpp" line="30"/>
         <source>Visualization</source>
-        <translation type="unfinished"></translation>
+        <translation>Görüntüleme</translation>
     </message>
 </context>
 </TS>
diff -Nuar qmmp-0.3.0.orig/src/ui/txt/authors_tr.txt qmmp-0.3.0/src/ui/txt/authors_tr.txt
--- qmmp-0.3.0.orig/src/ui/txt/authors_tr.txt	1970-01-01 02:00:00.000000000 +0200
+++ qmmp-0.3.0/src/ui/txt/authors_tr.txt	2009-07-23 13:17:09.841507867 +0300
@@ -0,0 +1,13 @@
+Çekirdek Geliştiriciler:
+
+    Ilya Kotov <forkotov02@hotmail.ru> (fikirler ve temel kod)
+    Vladimir Kuznetsov <vovanec@gmail.com> (görünüm ve birçok iyileştirme)
+
+Eklenti Geliştiricileri:
+
+    Yuriy Zhuravlev <stalkerg@gmail.com> (jack & oss eklentisi)
+
+Artwork:
+
+    Andrey Adreev <andreev00@gmail.com>
+
diff -Nuar qmmp-0.3.0.orig/src/ui/txt/description_tr.txt qmmp-0.3.0/src/ui/txt/description_tr.txt
--- qmmp-0.3.0.orig/src/ui/txt/description_tr.txt	1970-01-01 02:00:00.000000000 +0200
+++ qmmp-0.3.0/src/ui/txt/description_tr.txt	2009-07-23 13:17:50.026533724 +0300
@@ -0,0 +1 @@
+Bu program Qt kitaplığı kullanılarak yazılmış bir çoklu ortam oynatıcısıdır.
diff -Nuar qmmp-0.3.0.orig/src/ui/txt/thanks_tr.txt qmmp-0.3.0/src/ui/txt/thanks_tr.txt
--- qmmp-0.3.0.orig/src/ui/txt/thanks_tr.txt	1970-01-01 02:00:00.000000000 +0200
+++ qmmp-0.3.0/src/ui/txt/thanks_tr.txt	2009-07-23 13:18:07.638005852 +0300
@@ -0,0 +1,3 @@
+Teşekkürler:
+
+    Vadim Kalinnikov <moose@ylsoftware.com> (proje barındırma)
diff -Nuar qmmp-0.3.0.orig/src/ui/txt/translators_ru.txt qmmp-0.3.0/src/ui/txt/translators_ru.txt
--- qmmp-0.3.0.orig/src/ui/txt/translators_ru.txt	2009-06-23 17:27:51.000000000 +0300
+++ qmmp-0.3.0/src/ui/txt/translators_ru.txt	2009-07-23 13:20:05.340148157 +0300
@@ -1,31 +1,35 @@
-Бразильский португальский:
+Brezilya Portekizcesi:
     Klaos Lacerda
     Bruno Gonçalves
 
-Итальянский:
-    Gian Paolo Renello <emmerkar@gmail.com>
-
-Китайский традиционный:
+Geleneksel Çince:
     lon <lon83129@126.com>
 
-Китайский упрощённый:
+Basitleştirilmiş Çince:
     lon <lon83129@126.com>
 
-Польский
-   Grzegorz Gibas <amigib@gmail.com>
-
-Чешский:
+Çekce:
     Karel Volný <kvolny@redhat.com>
 
-Немецкий:
+Fransızca:
+    Stanislas Zeller <uncensored.assault@gmail.com>
+
+Almanca:
     Stefan Koelling <stefankoelling.ext@googlemail.com>
     Panagiotis Papadopoulos <pano_90@gmx.net>
 
-Русский:
-    Илья Котов <forkotov02@hotmail.ru>
+İtalyanca:
+    Gian Paolo Renello <emmerkar@gmail.com>
+
+Polonyaca:
+    Grzegorz Gibas <amigib@gmail.com>
+
+Rusça:
+    Ilya Kotov <forkotov02@hotmail.ru>
 
-Турецкий:
-    Mustafa GUNAY <mustafagunay@pgmail.com>
+Türkçe:
+    Mustafa GUNAY <mustafagunay@gmail.com>
+    Bilgesu Güngör <bilgesugungor@gmail.com>
 
-Украинский:
-    Геннадий Моцьо <drool@altlinux.ru>
+Ukraynaca:
+    Gennadi Motsyo <drool@altlinux.ru>
diff -Nuar qmmp-0.3.0.orig/src/ui/txt/translators_tr.txt qmmp-0.3.0/src/ui/txt/translators_tr.txt
--- qmmp-0.3.0.orig/src/ui/txt/translators_tr.txt	1970-01-01 02:00:00.000000000 +0200
+++ qmmp-0.3.0/src/ui/txt/translators_tr.txt	2009-07-23 13:52:03.054944042 +0300
@@ -0,0 +1,35 @@
+Brezilya Portekizcesi:
+    Klaos Lacerda
+    Bruno Gonçalves
+
+Geleneksel Çince:
+    lon <lon83129@126.com>
+
+Basitleştirilmiş Çince:
+    lon <lon83129@126.com>
+
+Çekce:
+    Karel Volný <kvolny@redhat.com>
+
+Fransızca:
+    Stanislas Zeller <uncensored.assault@gmail.com>
+
+Almanca:
+    Stefan Koelling <stefankoelling.ext@googlemail.com>
+    Panagiotis Papadopoulos <pano_90@gmx.net>
+
+İtalyanca:
+    Gian Paolo Renello <emmerkar@gmail.com>
+
+Polonyaca:
+    Grzegorz Gibas <amigib@gmail.com>
+
+Rusça:
+    Ilya Kotov <forkotov02@hotmail.ru>
+
+Türkçe:
+    Mustafa GUNAY <mustafagunay@gmail.com>
+    Bilgesu Güngör <bilgesugungor@gmail.com>
+
+Ukraynaca:
+    Gennadi Motsyo <drool@altlinux.ru>
