--- juffed.desktop.orig	2009-05-20 20:33:52.672792967 +0200
+++ juffed.desktop	2009-05-20 20:36:50.922797206 +0200
@@ -1,17 +1,13 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=Text editor JuffEd
-Name[ru]=Текстовый редактор JuffEd
-Name[pl]=Edytor tekstu JuffEd
-Name[de]=Texteditor JuffEd
-Name[fr]=Éditeur de texte JuffEd
-Name[pt]=Editor de texto JuffEd
-Name[zh_CN]=文本编辑器 JuffEd
-Comment=Advanced text editor
+Name=JuffEd
+Comment=Simple text editor
 Comment[ru]=Продвинутый текстовый редактор
 Comment[pl]=Zaawansowany edytor tekstu
 Comment[de]=Ein erweiterter Texteditor
 Comment[fr]=Un avancé éditeur de texte
+Comment[pt]=Editor de texto simples
+Comment[tr]=Basit metin editörü
 Comment[zh_CN]=先进的文本编辑器
 Exec=juffed
 Terminal=false
@@ -26,5 +22,6 @@
 GenericName[de]=Ein erweiterter Texteditor
 GenericName[fr]=Un avancé éditeur de texte
 GenericName[pt]=Editor de texto simples
+GenericName[tr]=Basit metin editörü
 GenericName[zh_CN]=先进的文本编辑器
 X-SuSE-translate=true
