diff -p -up qtpfsgui-1.9.3/src/Batch/batch_dialog.cpp.str qtpfsgui-1.9.3/src/Batch/batch_dialog.cpp
--- qtpfsgui-1.9.3/src/Batch/batch_dialog.cpp.str	2008-09-24 22:34:14.000000000 +0200
+++ qtpfsgui-1.9.3/src/Batch/batch_dialog.cpp	2009-06-11 18:42:12.000000000 +0200
@@ -356,7 +356,7 @@ void BatchTMDialog::newResult(const QIma
 }
 
 void BatchTMDialog::add_log_message(const QString& message) {
-	qDebug(qPrintable(message));
+	qDebug("%s",qPrintable(message));
 	full_Log_Model->insertRows(full_Log_Model->rowCount(),1);
 	full_Log_Model->setData(full_Log_Model->index(full_Log_Model->rowCount()-1), message, Qt::DisplayRole);
 	Log_Widget->scrollToBottom();
diff -p -up qtpfsgui-1.9.3/src/Common/commandline.cpp.str qtpfsgui-1.9.3/src/Common/commandline.cpp
--- qtpfsgui-1.9.3/src/Common/commandline.cpp.str	2008-09-05 18:38:40.000000000 +0200
+++ qtpfsgui-1.9.3/src/Common/commandline.cpp	2009-06-11 18:42:12.000000000 +0200
@@ -37,16 +37,16 @@ void writeRGBEfile (pfs::Frame* inputpfs
 void writeEXRfile  (pfs::Frame* inputpfsframe, const char* outfilename);
 
 #if defined(__FreeBSD__) || defined(WIN32) || defined(Q_WS_MAC)
-#define error(Z) { fprintf(stderr,Z); exit(1); }
+#define error(Z) { fprintf(stderr,"%s",Z); exit(1); }
 #else
 #include <error.h>
-#define error(Z) error(1,0,Z);
+#define error(Z) error(1,0,"%s",Z);
 #endif
 
 ///string is a QString with a %1 in it
 #define VERBOSEPRINT( string, argument ) \
 if (verbose) { \
-	fprintf(stdout, qPrintable(tr( string "\n" ).arg( argument )) ); \
+	fprintf(stdout, "%s", qPrintable(tr( string "\n" ).arg( argument )) ); \
 }
 
 
@@ -488,5 +488,5 @@ tr("Usage: %1 [OPTIONS]... [INPUTFILES].
 "\t" + tr("-o --output LDR_FILE   File name you want to save your tone mapped LDR to.") + "\n" +
 "\t" + tr("                       (No tonemapping is performed unless -o is specified).") + "\n\n" +
 tr("You must either load an existing HDR file (via the -l option) or specify INPUTFILES to create a new HDR.\n");
-	fprintf(stderr,qPrintable(help));
+	fprintf(stderr,"%s",qPrintable(help));
 }
diff -p -up qtpfsgui-1.9.3/src/Libpfs/pfs.cpp.str qtpfsgui-1.9.3/src/Libpfs/pfs.cpp
--- qtpfsgui-1.9.3/src/Libpfs/pfs.cpp.str	2009-06-11 18:42:44.000000000 +0200
+++ qtpfsgui-1.9.3/src/Libpfs/pfs.cpp	2009-06-11 18:43:27.000000000 +0200
@@ -590,7 +590,7 @@ static void writeTags( const TagContaine
   TagList::const_iterator it;
   fprintf( out, "%d" PFSEOL, tags->getSize() );
   for( it = tags->tagsBegin(); it != tags->tagsEnd(); it++ ) {
-    fprintf( out, it->c_str() );
+    fprintf( out, "%s", it->c_str() );
     fprintf( out, PFSEOL );
   }
 }
