--- menu-entries/LiVES.desktop.orig	2009-06-16 21:25:22.604686025 +0300
+++ menu-entries/LiVES.desktop	2009-06-16 21:26:14.608685954 +0300
@@ -1,6 +1,10 @@
 [Desktop Entry]
 Name=LiVES 
+Name[tr]=LiVES
+GenericName=Video Editor/VJ tool
+GenericName[tr]=Video Düzenleyici
 Comment=Video Editor/VJ tool
+Comment[tr]=Videolarınızı düzenlemeye yarar
 Categories=AudioVideo;AudioVideoEditing;
 Exec=lives
 Icon=lives
