--- sqliteman.desktop	2007-09-25 10:23:33.000000000 +0300
+++ sqliteman.desktop	2008-06-10 07:10:14.000000000 +0300
@@ -2,16 +2,15 @@
 Type=Application
 Encoding=UTF-8
 Name=Sqliteman
-GenericName=Sqlite admin tool
+GenericName=Sqlite Admin Tool
 Comment=Administer and develop your Sqlite3 databases
 Icon=sqliteman.png
 Exec=sqliteman %f
 Terminal=false
-Categories=Office;Database
+Categories=Qt;Utility;Database
 StartupNotify=true
 MimeType=application/x-sqlite3
 
 # Translations
-GenericName[cs]=Sqlite administrace
-
-X-SuSE-translate=true
+GenericName[cs]=Sqlite Administrace
+GenericName[tr]=Sqlite Yönetim Aracı
