diff -Nur liferea-1.4.1/opml/feedlist_tr.opml liferea-1.4.1.patched/opml/feedlist_tr.opml
--- liferea-1.4.1/opml/feedlist_tr.opml	2007-09-03 20:25:52.000000000 +0300
+++ liferea-1.4.1.patched/opml/feedlist_tr.opml	2007-09-05 07:56:06.000000000 +0300
@@ -14,7 +14,6 @@
       <outline title="&#xD6;zg&#xFC;rl&#xFC;k i&#xE7;in... - Anasayfa" text="&#xD6;zg&#xFC;rl&#xFC;k i&#xE7;in... - Anasayfa" description="&#xD6;zg&#xFC;rl&#xFC;k i&#xE7;in... - Anasayfa" type="rss" id="fpwuash" sortColumn="time" viewMode="0" htmlUrl="http://www.ozgurlukicin.com" xmlUrl="http://www.ozgurlukicin.com/rss" updateInterval="-1"/>
       <outline title="Pardus Gezegeni" text="Pardus Gezegeni" description="Pardus Gezegeni" type="rss" id="vmaorqs" sortColumn="time" viewMode="0" htmlUrl="http://gezegen.pardus.org.tr" xmlUrl="http://gezegen.pardus.org.tr/rss20.xml" updateInterval="-1"/>
       <outline title="&#xD6;zg&#xFC;rl&#xFC;k&#x130;&#xE7;in Gezegeni" text="&#xD6;zg&#xFC;rl&#xFC;k&#x130;&#xE7;in Gezegeni" description="&#xD6;zg&#xFC;rl&#xFC;k&#x130;&#xE7;in Gezegeni" type="rss" id="vslcibm" sortColumn="time" viewMode="0" htmlUrl="http://www.ozgurlukicin.com/gezegen" xmlUrl="http://ozgurlukicin.com/rss/gezegen/" updateInterval="-1"/>
-      <outline title="Linux Gezegeni" text="Linux Gezegeni" description="Linux Gezegeni" type="rss" id="onddcru" sortColumn="time" viewMode="0" htmlUrl="http://gezegen.linux.org.tr" xmlUrl="http://gezegen.linux.org.tr/rss20.xml" updateInterval="-1"/>
     </outline>
   </body>
 </opml>
