--- k9copy.desktop.orig	2007-02-18 11:58:59.000000000 +0200
+++ src/k9copy.desktop	2007-03-28 22:13:49.000000000 +0300
@@ -4,9 +4,11 @@
 Name=k9copy
 Exec=k9copy -caption "%c" %i %m  %u
 Comment=DVD9 to DVD5 backup tool
-GenericName=Video DVD Backup
 Comment[fr]=Backup de DVD9 en DVD5
+Comment[tr]=DVD9 DVD5 yedekleme aracı
+GenericName=Video DVD Backup
 GenericName[fr]=Backup de DVD Video
+GenericName[tr]=DVD Vidyo Yedekleme Aracı
 MimeTypes=
 Terminal=false
 Icon=k9copy.png
