--- dataloader.cpp.orig	2007-12-06 22:13:50.000000000 +0200
+++ src/dataloader.cpp	2007-12-06 22:14:10.000000000 +0200
@@ -47,7 +47,7 @@
 	if ( !m_file->open ( QFile::ReadOnly ) )
 	{
 		QMessageBox::warning (0, QString ( "Fontmatrix" ),
-				       QString ( "Seems that’s the first time you run Fontmatrix, if not there’s a problem loading the data file %1." ).arg(m_file->fileName()) );
+				       QString ( "Seems that's the first time you run Fontmatrix, if not there's a problem loading the data file %1." ).arg(m_file->fileName()) );
 		return;
 	}
 	if ( !doc.setContent ( m_file ) )
