--- linden/indra/SConstruct.orig	2007-11-30 10:32:15.000000000 +0200
+++ linden/indra/SConstruct	2007-11-30 10:34:05.000000000 +0200
@@ -95,7 +95,7 @@
     EnumOption('CHANNEL', 'Client package\'s default channel', 'Release',
         allowed_values=('Release', 'Release Candidate', 'WindLight')),
     BoolOption('ELFIO', 'Enabled enhanced backtraces with libELFIO symbol extraction support', False),
-    BoolOption('STANDALONE', 'Build using system packages (implies OPENSOURCE)', False),
+    BoolOption('STANDALONE', 'Build using system packages (implies OPENSOURCE)', True),
     BoolOption('OPENSOURCE', 'Build using only non-proprietary dependencies', True) # OPENSOURCE: do not edit this line
 )
 optenv = Environment(options = opts)
