--- wammu.desktop~	2007-10-05 04:13:51.000000000 +0300
+++ wammu.desktop	2008-02-03 13:01:49.000000000 +0200
@@ -1,11 +1,14 @@
 [Desktop Entry]
 Name=Wammu
+Name[tr]=Wammu
 GenericName=Mobile Phone Manager
+GenericName[tr]=Mobil Telefon Yöneticisi
 Comment=Application for mobile phones - frontend for Gammu
 Comment[pl]=Aplikacja do obsługi telefonów komórkowych - nakładka graficzna dla Gammu
 Comment[cz]=Aplikace pro práci s mobilními telefony - grafické rozhaní pro Gammu
+Comment[tr]=Gammu tabanlı mobil telefon yöneticisi
 Exec=wammu
 Icon=wammu
 Terminal=false
 Type=Application
-Categories=Utility;ContactManagement;Office;GTK;
+Categories=TelephonyTools;Calendar;ContactManagement;Office;GTK;
