--- Miro-1.2.8.orig/platform/gtk-x11/miro.desktop	2008-10-24 20:49:17.000000000 +0300
+++ Miro-1.2.8/platform/gtk-x11/miro.desktop	2008-10-25 13:39:30.000000000 +0300
@@ -2,7 +2,9 @@
 Type=Application
 Name=Miro Internet TV
 GenericName=Internet TV
+GenericName[tr]=İnternet TV
 Comment=Watch online video
+Comment[tr]=İnternetten  video izleyin, indirin
 Comment[de]=Multimedia Player
 Comment[es]=Reproductor multimedia
 Comment[fr]=Lecteur multimedia
