--- Miro-1.2.1/portable/frontends/html/main.py	2008-03-26 08:50:44.000000000 -0700
+++ Miro-1.2.1/portable/frontends/html/main.py.new	2008-03-31 08:49:15.000000000 -0700
@@ -63,7 +63,7 @@
     templates for the display (i.e. the Miro frontend for version 1.0)
     """
 
-    AUTOUPDATE_SUPPORTED = True
+    AUTOUPDATE_SUPPORTED = False
 
     def __init__(self):
         self.ignoreErrors = False
