--- polyxmass-bin.desktop.orig	2006-07-10 19:45:18.000000000 +0300
+++ polyxmass-bin.desktop	2007-07-28 13:55:07.000000000 +0300
@@ -5,10 +5,11 @@
 Encoding=UTF-8
 Name=GNU polyxmass
 GenericName=Mass Spectrometry Software Suite
-GenericName[fr]=Suite Logicielle de Spectrométrie de Masse 
+GenericName[fr]=Suite Logicielle de Spectrométrie de Masse
+GenericName[tr]=Kütle spektrometri uygulaması
 Comment=Simulation/analysis of (bio-) polymer mass spectrometric data
 Comment[fr]=Simulation/analyse de données de spectrométrie de masse de (bio-polymères)
 Type=Application
 Exec=polyxmass
-Icon=/usr/share/pixmaps/polyxmass-icon-32x32.png
+Icon=polyxmass
 Categories=Application;Science;Education;Chemistry;Biology;
