--- xfce4-utils-4.4.2/scripts/xflock4.orig	2007-11-17 20:31:32.000000000 +0100
+++ xfce4-utils-4.4.2/scripts/xflock4	2008-01-08 23:49:22.000000000 +0100
@@ -19,9 +19,9 @@
 #  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 #
 
-if test x"`which xscreensaver-command 2>/dev/null`" != x""; then
+if ps aux | grep x[s]creensaver > /dev/null 2>&1; then
     xscreensaver-command -lock 
-elif test x"`which gnome-screensaver-command 2>/dev/null`" != x""; then
+elif ps aux | grep gnome-[s]creensaver > /dev/null 2>&1; then
     gnome-screensaver-command --lock
 else
     xlock $*
