Index: gdk/gdkinputcommon.h
===================================================================
--- gdk/gdkinputcommon.h.orig
+++ gdk/gdkinputcommon.h
@@ -25,6 +25,7 @@
  */
 
 #if defined(XINPUT_GXI) || defined(XINPUT_XFREE)
+#include <string.h>
 
 /* Forward declarations */
 static void gdk_input_get_root_relative_geometry (Display *dpy, Window w, 
Index: gtk/gtkfontsel.c
===================================================================
--- gtk/gtkfontsel.c.orig
+++ gtk/gtkfontsel.c
@@ -3025,6 +3025,8 @@ gtk_font_selection_insert_font (GSList		
   /* Add another entry to the table for this new font family */
   temp_info.family = family_exists ? family : g_strdup(family);
   temp_info.foundry = foundry;
+  temp_info.nstyles = 0;
+  temp_info.style_index = 0;
   temp_fontname = g_slist_prepend (NULL, fontname);
   
   (*ntable)++;
Index: gtk/testgtk.c
===================================================================
--- gtk/testgtk.c.orig
+++ gtk/testgtk.c
@@ -3808,9 +3808,11 @@ insert_row_clist (GtkWidget *widget, gpo
       col1.red   = 0;
       col1.green = 56000;
       col1.blue  = 0;
+      col1.pixel = 0;
       col2.red   = 32000;
       col2.green = 0;
       col2.blue  = 56000;
+      col2.pixel = 0;
 
       style1 = gtk_style_copy (GTK_WIDGET (data)->style);
       style1->base[GTK_STATE_NORMAL] = col1;
@@ -4067,9 +4069,11 @@ create_clist (void)
       col1.red   = 56000;
       col1.green = 0;
       col1.blue  = 0;
+      col1.pixel = 0;
       col2.red   = 0;
       col2.green = 56000;
       col2.blue  = 32000;
+      col2.pixel = 0;
 
       style = gtk_style_new ();
       style->fg[GTK_STATE_NORMAL] = col1;
@@ -4234,9 +4238,11 @@ void change_style (GtkWidget *widget, Gt
       col1.red   = 0;
       col1.green = 56000;
       col1.blue  = 0;
+      col1.pixel = 0;
       col2.red   = 32000;
       col2.green = 0;
       col2.blue  = 56000;
+      col2.pixel = 0;
 
       style1 = gtk_style_new ();
       style1->base[GTK_STATE_NORMAL] = col1;
