--- Makefile.orig	2006-10-23 18:53:36.678885256 +0300
+++ Makefile	2006-10-23 18:53:52.117538224 +0300
@@ -2,10 +2,10 @@
 CC=gcc
 
 # Set PREFIX to the root of where you want to install wordplay
-PREFIX=/usr/local
+PREFIX=/usr
 
 # Installation points for the wordplay binaries and data
-BINDIR=$(PREFIX)/games
+BINDIR=$(PREFIX)/bin
 DATADIR=$(PREFIX)/share/wordplay
 
 # Optimization and debug flags
