--- emerald_20061023.orig/misc/Makefile.am	2006-10-23 20:22:55.000000000 +0300
+++ emerald_20061023/misc/Makefile.am	2006-10-23 20:23:29.000000000 +0300
@@ -15,6 +15,6 @@
 if MIMEUPDATE
 install-data-local: install-desktopDATA
 	$(UPDATE_MIME_DATABASE) $(datadir)/mime
-	$(UPDATE_ICON_CACHE) $(datadir)/icons/gnome
+	$(UPDATE_ICON_CACHE) $(datadir)/icons/Tulliana-2.0
 	$(UPDATE_DESKTOP_DATABASE)
 endif
