diff -Naur ncmpc-0.11.1.orig/doc/Makefile.in ncmpc-0.11.1/doc/Makefile.in
--- ncmpc-0.11.1.orig/doc/Makefile.in	2005-01-23 19:39:49.000000000 +0200
+++ ncmpc-0.11.1/doc/Makefile.in	2008-01-31 22:04:27.000000000 +0200
@@ -108,7 +108,7 @@
 install_sh = @install_sh@
 
 man_MANS = ncmpc.1
-docdir = $(prefix)/share/doc/$(PACKAGE)
+docdir = $(prefix)/share/doc/srcTAG
 doc_DATA = config.sample keys.sample ncmpc.lirc
 EXTRA_DIST = $(man_MANS) $(doc_DATA)
 subdir = doc
diff -Naur ncmpc-0.11.1.orig/Makefile.in ncmpc-0.11.1/Makefile.in
--- ncmpc-0.11.1.orig/Makefile.in	2005-01-23 19:39:48.000000000 +0200
+++ ncmpc-0.11.1/Makefile.in	2008-01-31 22:14:39.000000000 +0200
@@ -111,8 +111,8 @@
 
 SUBDIRS = src doc po
 pkgdata_DATA = 
-docdir = $(prefix)/share/doc/$(PACKAGE)
-doc_DATA = AUTHORS README NEWS
+docdir = $(prefix)/share/doc/srcTAG
+doc_DATA = AUTHORS README NEWS COPYING
 EXTRA_DIST = COPYING $(pkgdata_DATA) $(doc_DATA)
 subdir = .
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
