--- configure	2006-11-09 01:03:49.000000000 +0200
+++ configure_yeni	2007-01-22 14:15:29.000000000 +0200
@@ -12,9 +12,7 @@
 	rm -f src/Makefile.in
 	rm -f contrib/rtlbrowse/Makefile.in
 
-	X=""
-	echo "Build GTKWave for GTK+-1.x or 2.x? Please enter '1' or '2'"
-	read X
+	X="2"
 
 	if [ "$X" = "" ]
 		then X="2"
@@ -47,9 +45,7 @@
 
 # in case of automatic build, default to gtk+-1.x...
 
-X=""
-echo "Build GTKWave for GTK+-1.x or 2.x? Please enter '1' or '2'"
-read X
+X="2"
 
 if [ "$X" = "" ]
 	then X="1"
