diff -Nuar quodlibet-2.1.orig/exfalso.desktop.in quodlibet-2.1/exfalso.desktop.in
--- quodlibet-2.1.orig/exfalso.desktop.in	2009-07-23 01:39:17.486988916 +0300
+++ quodlibet-2.1/exfalso.desktop.in	2009-07-23 01:39:41.560992039 +0300
@@ -6,5 +6,5 @@
 Exec=exfalso
 Terminal=false
 Type=Application
-Icon=exfalso.png
+Icon=exfalso
 Categories=Application;AudioVideo;GTK
diff -Nuar quodlibet-2.1.orig/quodlibet.desktop.in quodlibet-2.1/quodlibet.desktop.in
--- quodlibet-2.1.orig/quodlibet.desktop.in	2009-07-23 01:39:17.533236710 +0300
+++ quodlibet-2.1/quodlibet.desktop.in	2009-07-23 01:39:31.630992481 +0300
@@ -6,5 +6,5 @@
 Exec=quodlibet
 Terminal=false
 Type=Application
-Icon=quodlibet.png
+Icon=quodlibet
 Categories=Application;AudioVideo;AudioPlayer;GTK
