diff -Naur rlplot.orig/Makefile rlplot/Makefile
--- rlplot.orig/Makefile	2008-05-02 18:48:33.000000000 +0300
+++ rlplot/Makefile	2010-01-29 15:17:01.325377164 +0200
@@ -18,10 +18,12 @@
 #    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 #
 #
-CC = g++
 X11LIBS = -lX11 -lm
 SRCDIR = ./
 
+BIN = $(DESTDIR)/usr/bin
+MAN = $(DESTDIR)/usr/share/man/man1
+
 #######
 #######   Declarations for Qt3
 QT3DIR = /usr/lib/qt-3.3
@@ -39,10 +41,10 @@
 
 #######
 #######   Declarations for Qt4
-QT4DIR = /usr/lib/qt4
-QT4MOC = /usr/lib/qt4/bin/moc-qt4
+QT4DIR = /usr/qt/4
+QT4MOC = /usr/qt/4/bin/moc
 MOC4FLAGS = "MOCFLAGS=-DQT_VERSION=0x040000"
-QT4CFLAGS = "-I/usr/include/Qt -pipe -O2"
+QT4CFLAGS = "$(CFLAGS) $(LDFLAGS) -I/usr/qt/4/include/Qt -I/usr/qt/4/include -pipe -O2"
 QT4H = QT_Spec.h
 QT4LIBS = "-L$(QT4DIR)/lib -L/usr/X11R6/lib -lQtCore -lQtGui"
 #
@@ -79,6 +81,11 @@
 exprlp: $(GENOBJ)
 	$(CC) -o exprlp $(GENOBJ)
 
+install:
+	install -d $(BIN) $(MAN)
+	install ./rlplot $(BIN)
+	install ./exprlp $(BIN)
+
 clean:
 	rm -f *.o *~
 	rm -f moc_QT_Spec.cpp
