--- gperiodic.desktop.orig	2007-07-11 15:58:05.000000000 +0300
+++ gperiodic.desktop	2007-08-06 11:51:11.000000000 +0300
@@ -5,6 +5,7 @@
 Comment[da]=Et program til at vise data om grundstofferne i det priodiske system.
 Comment[pl]=Program do wyszukiwania danych pierwiastków z tablicy okresowej.
 Comment[sv]=En lathund för information om grundämnena i den periodiska tabellen.
+Comment[tr]=Periyodik Tablo Uygulaması
 Exec=gperiodic
 Icon=gperiodic
 Name=GPeriodic
