diff -Naur glchess-1.0.1-orig/data/glchess.desktop glchess-1.0.1/data/glchess.desktop
--- data/glchess.desktop	2007-01-14 08:43:00.000000000 +0200
+++ data/glchess.desktop	2007-01-14 08:44:11.000000000 +0200
@@ -5,7 +5,8 @@
 GenericName=3D Chess Game
 GenericName[pt_BR]=Jogo de Xadrez 3D
 GenericName[de]=3D Schach
-Icon=glchess
+GenericName[tr]=Satranç oyunu
+Icon=/usr/share/pixmaps/glchess.svg
 Exec=glchess
 Terminal=false
 Categories=Application;Game;BoardGame;
