--- ooRexx-3.1.0/kernel/platform/unix/FileSystem.cpp.paths	2006-09-25 19:11:07.000000000 +0200
+++ ooRexx-3.1.0/kernel/platform/unix/FileSystem.cpp	2006-09-25 19:12:09.000000000 +0200
@@ -427,6 +427,9 @@
 //  fullname = (PSZ)string_data(imgpath);
 
 //if ( imgpath && fullname )                         /* seg faultn          */
+  if (fullname == 0) {
+    fullname = SearchFileName((PCHAR)"/usr/share/ooRexx/" BASEIMAGE, 'P');
+  }
   if ( fullname != OREF_NULL )
     image = fopen((PSZ)fullname, "rb");/* try to open the file              */
   else
--- ooRexx-3.1.0/Makefile.in.paths	2006-09-25 19:12:21.000000000 +0200
+++ ooRexx-3.1.0/Makefile.in	2006-09-25 19:14:15.000000000 +0200
@@ -60,7 +60,7 @@
 top_srcdir = @top_srcdir@
 VPATH = @srcdir@
 pkgdatadir = $(datadir)/@PACKAGE@
-pkglibdir = $(libdir)/@PACKAGE@
+pkglibdir = $(libdir)
 pkgincludedir = $(includedir)/@PACKAGE@
 top_builddir = .
 am__cd = CDPATH="$${ZSH_VERSION+.}$(PATH_SEPARATOR)" && cd
@@ -415,7 +415,7 @@
 			compile @REXXCAT@ rexx.img \
 			stamp-h.in cscope.files cscope.out $(distdir).tar.gz
 
-SUBDIRS = samples
+SUBDIRS =
 DIST_SUBDIRS = samples
 
 # This defines the current version of ooRexx
@@ -462,7 +462,7 @@
 samples_dir = $(top_srcdir)/samples
 samples_windows_dir = $(top_srcdir)/samples/windows
 samples_unix_dir = $(top_srcdir)/samples/unix
-catdir = $(prefix)/bin
+catdir = $(datadir)/ooRexx
 COMMON_CPPFLAGS = -DORX_VER=$(ORX_MAJOR) -DORX_REL=$(ORX_MINOR) -DORX_MOD=$(ORX_MOD_LVL) -DORX_FIX=0 \
                   -DORX_SYS_STR=\"@ORX_SYS_STR@\" \
                   -DORX_CATDIR=\"$(catdir)\" \
@@ -3170,7 +3170,7 @@
 	echo $(DESTDIR) is destdir
 	echo $(prefix) is prefix
 	echo $(install_sh_PROGRAM)
-	$(install_sh_SCRIPT) rexx.img $(DESTDIR)$(prefix)/bin/rexx.img
+	$(install_sh_SCRIPT) rexx.img $(DESTDIR)$(datadir)/ooRexx/rexx.img
 	if test -z "$(GENCAT)"; then \
 		echo "No rexx.cat to install"; \
 	else \
@@ -3181,9 +3181,9 @@
 	$(install_sh_SCRIPT) $(rexutils_dir)/rxftp.cls $(DESTDIR)$(prefix)/bin/rxftp.cls
 	for a in $(samples_dir)/*.rex; do \
 		bn=`basename $$a`; \
-		$(install_sh_SCRIPT) $$a $(DESTDIR)$(prefix)/share/ooRexx/$$bn; \
+		$(install_sh_SCRIPT) $$a $(DESTDIR)$(datadir)/ooRexx/$$bn; \
 	done;
-	$(install_sh_SCRIPT) $(samples_dir)/readme $(DESTDIR)$(prefix)/share/ooRexx/readme
+	$(install_sh_SCRIPT) $(samples_dir)/readme $(DESTDIR)$(datadir)/ooRexx/readme
 
 package:
 	./rexx $(top_srcdir)/platform/unix/makepkg.rex @PACKAGETYPE@ $(ORX_MAJOR).$(ORX_MINOR).$(ORX_MOD_LVL) $(target) $(prefix) $(DESTDIR)
