--- Makefile	2010-01-31 03:30:20.445026895 +0200
+++ Makefile	2010-01-31 03:31:00.740781244 +0200
@@ -21,11 +21,11 @@
 
 # Standard pbzip2 compile
 pbzip2: pbzip2.cpp
-	$(CC) $(CFLAGS) $^ -o pbzip2 -pthread -lpthread -lbz2
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -o pbzip2 -pthread -lpthread -lbz2
 
 # Choose this if you want to compile in a static version of the libbz2 library
 pbzip2-static: pbzip2.cpp libbz2.a
-	$(CC) $(CFLAGS) $^ -o pbzip2 -pthread -lpthread -I. -L. -lbz2
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -o pbzip2 -pthread -lpthread -I. -L. -lbz2
 
 # Install the binary pbzip2 program and man page
 install: pbzip2
