diff -Naur eog-2.14.3-orig/libeog/eog-image.c eog-2.14.3/libeog/eog-image.c
--- eog-2.14.3-orig/libeog/eog-image.c	2006-08-29 12:01:41.000000000 +0300
+++ eog-2.14.3/libeog/eog-image.c	2006-08-29 12:24:47.000000000 +0300
@@ -28,7 +28,7 @@
 #endif
 
 #ifdef HAVE_LCMS
- #include <lcms.h>
+ #include <lcms/lcms.h>
  #ifndef EXIF_TAG_GAMMA
   #define EXIF_TAG_GAMMA 0xa500
  #endif
diff -Naur eog-2.14.3-orig/libeog/eog-image.h eog-2.14.3/libeog/eog-image.h
--- eog-2.14.3-orig/libeog/eog-image.h	2006-08-29 12:01:41.000000000 +0300
+++ eog-2.14.3/libeog/eog-image.h	2006-08-29 12:25:33.000000000 +0300
@@ -11,7 +11,7 @@
 #include "eog-job.h"
 
 #ifdef HAVE_LCMS
-#include <lcms.h>
+#include <lcms/lcms.h>
 #endif
 
 G_BEGIN_DECLS
diff -Naur eog-2.14.3-orig/libeog/eog-image-private.h eog-2.14.3/libeog/eog-image-private.h
--- eog-2.14.3-orig/libeog/eog-image-private.h	2006-08-29 12:01:41.000000000 +0300
+++ eog-2.14.3/libeog/eog-image-private.h	2006-08-29 12:26:07.000000000 +0300
@@ -7,7 +7,7 @@
 #endif
 #include "eog-image.h"
 #ifdef HAVE_LCMS
-#include <lcms.h>
+#include <lcms/lcms.h>
 #endif
 
 typedef enum {
diff -Naur eog-2.14.3-orig/shell/eog-window.c eog-2.14.3/shell/eog-window.c
--- eog-2.14.3-orig/shell/eog-window.c	2006-08-29 12:01:41.000000000 +0300
+++ eog-2.14.3/shell/eog-window.c	2006-08-29 12:26:36.000000000 +0300
@@ -62,7 +62,7 @@
 #if HAVE_LCMS
 #include <X11/Xlib.h>
 #include <X11/Xatom.h>
-#include <lcms.h>
+#include <lcms/lcms.h>
 #endif
 
 #include <libgnomeprint/gnome-print.h>
