diff -Nuar wicd-1.6.1.orig/other/wicd.desktop wicd-1.6.1/other/wicd.desktop
--- wicd-1.6.1.orig/other/wicd.desktop	2009-06-24 19:04:01.000000000 +0300
+++ wicd-1.6.1/other/wicd.desktop	2009-06-24 19:08:50.000000000 +0300
@@ -2,12 +2,12 @@
 Categories=Application;Network;
 Exec=wicd-client --no-tray
 GenericName=Network Manager
+GenericName[tr]=Ağ Yöneticisi
 Icon=wicd-client
-Icon[en_US]=wicd-client
-Name=Wicd Network Manager
-Name[en_US]=Wicd Network Manager
+Name=Wicd
 Comment=Start the Wicd client without system tray icon
 Comment[en_US]=Start the Wicd client without system tray icon
+Comment[tr]=Sistem çekmecesine yerleştirmeden Wicd'i başlat
 Terminal=false
 Type=Application
 Version=1.0
diff -Nuar wicd-1.6.1.orig/other/wicd-tray.desktop wicd-1.6.1/other/wicd-tray.desktop
--- wicd-1.6.1.orig/other/wicd-tray.desktop	2009-06-24 19:04:01.000000000 +0300
+++ wicd-1.6.1/other/wicd-tray.desktop	2009-06-24 19:12:50.000000000 +0300
@@ -2,12 +2,12 @@
 Categories=Application;Network;
 Exec=wicd-client
 GenericName=Network Manager
+GenericName[tr]=Ağ Yöneticisi
 Icon=wicd-client
-Icon[en_US]=wicd-client
-Name=Wicd Network Manager Tray
-Name[en_US]=Wicd Network Manager Tray
+Name=Wicd
 Comment=Display network connection status in the system tray
 Comment[en_US]=Display network connection status in the system tray
+Comment[tr]=Sistem çekmecesinde bağlantı durumunu göster
 Terminal=false
 Type=Application
 Version=1.0
