--- wicd/gui.py.orig	2008-12-10 04:25:06.000000000 +0200
+++ wicd/gui.py	2008-12-10 04:29:56.000000000 +0200
@@ -171,13 +171,20 @@
 
 def setup_dbus():
     global proxy_obj, daemon, wireless, wired, config, dbus_ifaces
-    proxy_obj = bus.get_object("org.wicd.daemon", '/org/wicd/daemon')
-    daemon = dbus.Interface(proxy_obj, 'org.wicd.daemon')
-    wireless = dbus.Interface(proxy_obj, 'org.wicd.daemon.wireless')
-    wired = dbus.Interface(proxy_obj, 'org.wicd.daemon.wired')
-    config = dbus.Interface(proxy_obj, 'org.wicd.daemon.config')
-    dbus_ifaces = {"daemon" : daemon, "wireless" : wireless, "wired" : wired, 
-                   "config" : config}
+    try:
+        proxy_obj = bus.get_object("org.wicd.daemon", '/org/wicd/daemon')
+        daemon = dbus.Interface(proxy_obj, 'org.wicd.daemon')
+        wireless = dbus.Interface(proxy_obj, 'org.wicd.daemon.wireless')
+        wired = dbus.Interface(proxy_obj, 'org.wicd.daemon.wired')
+        config = dbus.Interface(proxy_obj, 'org.wicd.daemon.config')
+        dbus_ifaces = {"daemon" : daemon, "wireless" : wireless, "wired" : wired, "config" : config}
+    except:
+        import wicd.pardusservice as pardusService
+        import sys
+
+        print "Service wicd is not started! Exiting..."
+        pardusService.main()
+        sys.exit()
 
 ########################################
 ##### GTK EXTENSION CLASSES
