=== modified file 'curses/wicd-curses.py'
--- curses/wicd-curses.py	2009-06-21 02:44:32 +0000
+++ curses/wicd-curses.py	2009-06-21 18:45:44 +0000
@@ -202,7 +202,7 @@
 # Modeled after htop's help
 def help_dialog(body):
     textT  = urwid.Text(('header','wicd-curses help'),'right') 
-    textSH = urwid.Text(['This is ',('blue','wicd-curses-'+CURSES_REVNO),' using wicd ',unicode(daemon.Hello()),'\n'])
+    textSH = urwid.Text(['This is ',('blue','wicd-curses-'+CURSES_REV),' using wicd ',unicode(daemon.Hello()),'\n'])
 
     textH = urwid.Text([
 "For more detailed help, consult the wicd-curses(8) man page.\n",

