--- isomaster.desktop.src.orig	2009-01-14 17:30:12.000000000 +0200
+++ isomaster.desktop.src	2009-01-14 17:31:34.000000000 +0200
@@ -3,12 +3,13 @@
 GenericName=ISO File Editor
 GenericName[ca]=Editor de fitxers ISO
 GenericName[es]=Editor de ficheros ISO
+GenericName[tr]=ISO Düzenleyici
 Comment=Read, write and modify ISO images
 Comment[ca]=Llegiu, escriviu i modifiqueu imatges ISO
 Comment[es]=Leer, escribir i modificar imagenes ISO
 Exec=isomaster
 Terminal=false
-StartupNotify=true
+StartupNotify=false
 Type=Application
-Categories=AudioVideo;DiscBurning;
+Categories=Application;GTK;Utility;
 MimeType=application/x-iso;
