--- bin/jokosher.desktop.orig	2008-09-01 21:59:22.000000000 +0300
+++ bin/jokosher.desktop	2008-09-01 22:11:35.000000000 +0300
@@ -2,12 +2,15 @@
 Version=1.0
 Encoding=UTF-8
 Name=Jokosher Audio Editor
+Name[tr]= Jokosher
 GenericName=Jokosher audio editor
+GenericName[tr]=Jokosher ses düzenleyicisi
 Comment=Simply and easily create multi-track audio
+Comment[tr]=Basit ve kolayca ses izi oluşturur
 Exec=jokosher
 Icon=jokosher
 Terminal=false
 Type=Application
 MimeType=application/x-jokosher;
-Categories=Application;AudioVideo;
+Categories=GTK;AudioVideo;Recorder
 GenericName[en_CA]=Jokosher audio editor
