--- AcetoneISO-6.7/AcetoneISO.kmdr	2007-03-05 17:05:59.000000000 +0200
+++ AcetoneISO-6.7/AcetoneISO.kmdr	2007-04-01 19:24:23.000000000 +0300
@@ -497,7 +497,7 @@
 						if test -e $HOME/Desktop/Acetone-Mounted
 						then kdialog --error "Sorry, AcetoneISO wasn't able to Umount ~/Desktop/Acetone-Mounted/"
 						else
-   			artsplay /usr/share/sounds/KDE_Star_Ship_1.ogg
+                                                artsplay /usr/share/sounds/KDE_Star_Ship_1.ogg
 						fi
 			fi
    
@@ -512,8 +512,9 @@
 							then kdialog --error "Sorry, AcetoneISO wasn't able to Mount the Image"
 							rm -r -f $HOME/Desktop/Acetone-Mounted
 							else
-							artsplay /usr/share/sounds/KDE_Window_Open.wav  
-   							konqueror $HOME/Desktop/Acetone-Mounted
+							artsplay /usr/share/sounds/KDE_Window_Open.wav
+                                                        kdialog --caption "Completed" --msgbox 'Process Completed!'
+   							konqueror $HOME/Desktop/Acetone-Mounted&amp;
 							fi
 
 				else
@@ -716,7 +717,8 @@
 			then exit
 			else
 
-			konsole --noclose -e /opt/acetoneiso/bin/poweriso extract "$OPENFILE" / -od "$VAROUT" 
+			konsole -e /opt/acetoneiso/bin/poweriso extract "$OPENFILE" / -od "$VAROUT"
+                        kdialog --caption "Completed" --msgbox 'Process Completed!'
 			konqueror "$VAROUT"
 			fi
 		fi
@@ -849,7 +851,7 @@
 								then kdialog --error "Sorry, AcetoneISO wasn't able to Mount the Image"
 								else
 								artsplay /usr/share/sounds/KDE_Window_Open.wav 
-  								konqueror "$VAROUT"
+  								konqueror "$VAROUT"&amp;
 								fi
 							
 
@@ -863,7 +865,8 @@
 					then kdialog --error "Sorry, AcetoneISO wasn't able to Mount the Image"
 					else
 					artsplay /usr/share/sounds/KDE_Window_Open.wav  
-  					konqueror "$VAROUT"
+                                        kdialog --caption "Completed" --msgbox 'Process Completed!'
+  					konqueror "$VAROUT"&amp;
 					fi
 					
   			fi
@@ -1711,7 +1714,8 @@
 else
 
 
-konsole --noclose -e cdrdao read-cd --read-raw --datafile $SAVE.bin --device /dev/cdrom --driver generic-mmc-raw $SAVE.toc
+konsole -e cdrdao read-cd --read-raw --datafile $SAVE.bin --device /dev/cdrom --driver generic-mmc-raw $SAVE.toc
+kdialog --caption "Completed" --msgbox 'Process Completed!'
 fi
 exit</string>
                         </stringlist>
@@ -1781,7 +1785,8 @@
 
 if test $lol = "mdf"
 then
-konsole --noclose -e /opt/acetoneiso/bin/mdf2iso "$OPENFILE" "$SAVE"
+konsole -e /opt/acetoneiso/bin/mdf2iso "$OPENFILE" "$SAVE"
+kdialog --caption "Completed" --msgbox 'Process Completed!'
 else 
 	if test $lol = "img"
 	then
@@ -1793,32 +1798,39 @@
 		else
 			if test $lol = "nrg"
 			then
-			konsole --noclose -e /opt/acetoneiso/bin/nrg2iso "$OPENFILE" "$SAVE"
+			konsole -e /opt/acetoneiso/bin/nrg2iso "$OPENFILE" "$SAVE"
+                        kdialog --caption "Completed" --msgbox 'Process Completed!'
 			else
 				if test $lol = "cdi"
 				then
-				konsole --noclose -e /opt/acetoneiso/bin/cdi2iso "$OPENFILE" "$SAVE"
+				konsole -e /opt/acetoneiso/bin/cdi2iso "$OPENFILE" "$SAVE"
+                                kdialog --caption "Completed" --msgbox 'Process Completed!'
 				else
 					if test $lol = "b5i"
 					then
-					konsole --noclose -e /opt/acetoneiso/bin/b5i2iso "$OPENFILE" "$SAVE"
+					konsole -e /opt/acetoneiso/bin/b5i2iso "$OPENFILE" "$SAVE"
+                                        kdialog --caption "Completed" --msgbox 'Process Completed!'
 					else
 						if test $lol = "bwi"
 						then
-						konsole --noclose -e /opt/acetoneiso/bin/b5i2iso "$OPENFILE" "$SAVE"
+						konsole -e /opt/acetoneiso/bin/b5i2iso "$OPENFILE" "$SAVE"
+                                                kdialog --caption "Completed" --msgbox 'Process Completed!'
 						else
 							if test $lol = "xbx"
 							then
-							konsole --noclose -e /opt/acetoneiso/bin/xbiso "$OPENFILE" -d "$SAVE"
+							konsole -e /opt/acetoneiso/bin/xbiso "$OPENFILE" -d "$SAVE"
+                                                        kdialog --caption "Completed" --msgbox 'Process Completed!'
 							else
 								if test $lol = "pdi"
 								then
-								konsole --noclose -e /opt/acetoneiso/bin/pdi2iso  "$OPENFILE" "$SAVE"
+								konsole -e /opt/acetoneiso/bin/pdi2iso  "$OPENFILE" "$SAVE"
+                                                                kdialog --caption "Completed" --msgbox 'Process Completed!'
 								else
 						
 									if test $lol = "daa"
 									then
-									konsole --noclose -e /opt/acetoneiso/bin/poweriso convert "$OPENFILE" -o "$SAVE"
+									konsole -e /opt/acetoneiso/bin/poweriso convert "$OPENFILE" -o "$SAVE"
+                                                                        kdialog --caption "Completed" --msgbox 'Process Completed!'
 									else 
 									kdialog --error "ERROR.\nPlease select a file with a *. extension written under lowercase"
 									
@@ -1958,11 +1970,13 @@
 
 				if test $sev7 = Ultra-High-VerySlow
 				then
-				konsole --noclose -e 7z a -t7z -m0=lzma -mx=9 -mfb=64 -md=32m -ms=on "$SAVE7Z" "$OPENFILE"
+				konsole -e 7z a -t7z -m0=lzma -mx=9 -mfb=64 -md=32m -ms=on "$SAVE7Z" "$OPENFILE"
+                                kdialog --caption "Completed" --msgbox 'Process Completed!'
 
 				else
 
-                 		konsole --noclose -e 7z a -t7z -m0=lzma -mx=4 -mfb=32 -md=4m -ms=on "$SAVE7Z" "$OPENFILE" 
+                 		konsole -e 7z a -t7z -m0=lzma -mx=4 -mfb=32 -md=4m -ms=on "$SAVE7Z" "$OPENFILE"
+                                kdialog --caption "Completed" --msgbox 'Process Completed!'
 				fi
 				fi
 				exit
@@ -2045,7 +2059,8 @@
 		then exit
 		else
 
-		cd "$VAROUT7" &amp;&amp; konsole --noclose -e 7z e "$OPENFILE7"
+		cd "$VAROUT7" &amp;&amp; konsole -e 7z e "$OPENFILE7"
+                kdialog --caption "Completed" --msgbox 'Process Completed!'
 
 		fi
 fi
@@ -2111,7 +2126,8 @@
 then exit
 else
 
-konsole -e dd if=/dev/cdrom of="$VARIN" bs=1024 
+konsole -e dd if=/dev/cdrom of="$VARIN" bs=1024
+kdialog --caption "Completed" --msgbox 'Process Completed!'
 
 
 
@@ -2185,7 +2201,8 @@
 		then exit
 		else
 
- 		konsole --noclose -e mkisofs -ldots -allow-multidot -relaxed-filenames -iso-level 3 -allow-lowercase -o "$VAROUT" "$VARIN" 
+ 		konsole -e mkisofs -ldots -allow-multidot -relaxed-filenames -iso-level 3 -allow-lowercase -o "$VAROUT" "$VARIN"
+                kdialog --caption "Completed" --msgbox 'Process Completed!'
       
 		fi
 	fi
@@ -2884,7 +2901,7 @@
 	then exit
 	else 
 
- 	konsole -e gpg "$OPENFILE" &amp;&amp;
+ 	konsole -e gpg "$OPENFILE"&amp;
 
 	kdialog --msgbox "ISO Succesfully Decrypted"
 	fi
