diff -Naur idjc-0.7.10.orig/idjcpython/Makefile.in idjc-0.7.10/idjcpython/Makefile.in
--- idjc-0.7.10.orig/idjcpython/Makefile.in	2008-09-27 21:30:50.000000000 +0300
+++ idjc-0.7.10/idjcpython/Makefile.in	2008-09-28 19:48:56.000000000 +0300
@@ -56,7 +56,7 @@
 	"$(DESTDIR)$(pkgdatadir)"
 pkgpythonPYTHON_INSTALL = $(INSTALL_DATA)
 pythonPYTHON_INSTALL = $(INSTALL_DATA)
-py_compile = $(top_srcdir)/py-compile
+# py_compile = $(top_srcdir)/py-compile
 dist_pkgdataDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(dist_pkgdata_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
@@ -238,13 +238,13 @@
 	    $(pkgpythonPYTHON_INSTALL) "$$b$$p" "$(DESTDIR)$(pkgpythondir)/$$f"; \
 	  else :; fi; \
 	done; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pkgpythondir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pkgpythondir)" $$dlist; \
-	  fi; \
-	else :; fi
+# 	if test -n "$$dlist"; then \
+# 	  if test -z "$(DESTDIR)"; then \
+# 	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pkgpythondir)" $$dlist; \
+# 	  else \
+# 	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pkgpythondir)" $$dlist; \
+# 	  fi; \
+# 	else :; fi
 
 uninstall-pkgpythonPYTHON:
 	@$(NORMAL_UNINSTALL)
@@ -266,13 +266,13 @@
 	    $(pythonPYTHON_INSTALL) "$$b$$p" "$(DESTDIR)$(pythondir)/$$f"; \
 	  else :; fi; \
 	done; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pythondir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pythondir)" $$dlist; \
-	  fi; \
-	else :; fi
+# 	if test -n "$$dlist"; then \
+# 	  if test -z "$(DESTDIR)"; then \
+# 	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pythondir)" $$dlist; \
+# 	  else \
+# 	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pythondir)" $$dlist; \
+# 	  fi; \
+# 	else :; fi
 
 uninstall-pythonPYTHON:
 	@$(NORMAL_UNINSTALL)
