diff -Naur xcircuit-3.6.127/lib/tcl/tkcon.tcl xcircuit-3.6.127_yeni/lib/tcl/tkcon.tcl
--- lib/tcl/tkcon.tcl	2006-04-11 00:52:19.000000000 +0300
+++ lib/tcl/tkcon.tcl	2007-11-28 14:17:11.000000000 +0200
@@ -44,7 +44,7 @@
 if {$tcl_version < 8.0} {
     return -code error "tkcon requires at least Tcl/Tk8"
 } else {
-    package require -exact Tk $tcl_version
+    package require Tk $tcl_version
 }
 
 catch {package require bogus-package-name}
