diff -Nuar liferea-1.2.12.orig/opml/feedlist.opml liferea-1.2.12/opml/feedlist.opml
--- liferea-1.2.12.orig/opml/feedlist.opml	2007-04-29 22:52:59.000000000 +0300
+++ liferea-1.2.12/opml/feedlist.opml	2007-04-29 22:53:09.000000000 +0300
@@ -1,32 +1,17 @@
 <?xml version="1.0"?>
 <opml version="1.0">
   <head>
-    <title>Liferea Default Feed List</title>
+    <title>Liferea Feed List Export</title>
   </head>
   <body>
-      <!-- lets keep this English block in sync over all feed lists! -->
-      <outline text="international">
-	      <outline text="OSS">
-        	<outline text="art.gnome.org" htmlUrl="http://art.gnome.org/" xmlUrl="http://art.gnome.org/backend.php" />
-        	<outline text="Debian Package a Day" htmlUrl="http://debaday.debian.net/" xmlUrl="http://debaday.debian.net/feed/atom/" />
-        	<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
-        	<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
-        	<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
-        	<outline text="Linux.com" htmlUrl="http://www.linux.com/" xmlUrl="http://www.linux.com/index.rss" />
-        	<outline text="mozillaZine" htmlUrl="http://www.mozillazine.org/" xmlUrl="http://www.mozillazine.org/contents.rdf" />
-        	<outline text="Newsforge" htmlUrl="http://www.newsforge.com" xmlUrl="http://www.newsforge.com/newsforge.rss" />
-        	<outline text="Planet Debian" htmlUrl="http://planet.debian.org/" xmlUrl="http://planet.debian.org/rss20.xml" />
-        	<outline text="Slashdot" htmlUrl="http://slashdot.org"  xmlUrl="http://slashdot.org/slashdot.rss" />
-        	<outline text="TuxMobil" htmlUrl="http://tuxmobil.org" xmlUrl="http://tuxmobil.org/tuxmobil_rss.rdf" />      
-	      </outline>	
-      </outline>
-
-    <outline text="Unread" title="Unread" description="Unread" type="vfolder" htmlUrl="" xmlUrl="vfolder">
-
-      <outline type="rule" text="Read status" rule="unread" value="" additive="true"/>
+    <outline title="Unread" text="Unread" description="Unread" type="vfolder" id="hlwekgo" sortColumn="time" viewMode="0">
+      <outline type="rule" text="Okunma durumu" rule="unread" value="" additive="true"/>
     </outline>
-    <outline text="Important" title="Important" description="Important" type="vfolder" htmlUrl="" xmlUrl="vfolder">
-      <outline type="rule" text="Flag status" rule="flagged" value="" additive="true"/>
+    <outline title="Important" text="Important" description="Important" type="vfolder" id="gnnymgu" sortColumn="time" viewMode="0">
+      <outline type="rule" text="&#x130;&#x15F;aretlenme durumu" rule="flagged" value="" additive="true"/>
+    </outline>
+    <outline title="Open Source" text="Open Source" description="Open Source" type="folder" id="xgxxhcy" sortColumn="time" viewMode="0" expanded="true">
+      <outline title="Planet Pardus" text="Planet Pardus" description="Planet Pardus" type="rss" id="vmaorqs" sortColumn="time" viewMode="0" htmlUrl="http://planet.pardus.org.tr" xmlUrl="http://planet.pardus.org.tr/rss20.xml" updateInterval="-1" etag="&quot;cc354e-12b39-fc0f8600&quot;" lastModified="Sun, 29 Apr 2007 19:00:40 GMT" lastPollTime="1177876350" lastFaviconPollTime="1177876353"/>
     </outline>
   </body>
 </opml>
