diff -Nuar liferea-1.2.15/src/htmlview.c liferea-1.2.15.new/src/htmlview.c
--- liferea-1.2.15/src/htmlview.c	2007-05-17 19:37:55.000000000 +0300
+++ liferea-1.2.15.new/src/htmlview.c	2007-05-18 01:49:28.000000000 +0300
@@ -216,7 +216,7 @@
 	switch(mode) {
 		case ITEMVIEW_SINGLE_ITEM:
 			item = itemlist_get_selected();
-			if(item)
+			if(item && item->sourceNode->itemSet)
 				baseURL = (gchar *)itemset_get_base_url(item->sourceNode->itemSet);
 			break;
 		default:
