diff -ur ifolder3-3.4.6167.1.orig/src/LinuxClient/application/ifolder.in ifolder3-3.4.6167.1/src/LinuxClient/application/ifolder.in
--- ifolder3-3.4.6167.1.orig/src/LinuxClient/application/ifolder.in	2006-06-16 10:05:21.000000000 +0300
+++ ifolder3-3.4.6167.1/src/LinuxClient/application/ifolder.in	2006-06-30 17:27:12.000000000 +0300
@@ -25,6 +25,7 @@
 # Startup iFolder
 export MONO_PATH=@_prefix_@/lib:@_webbindir_@:@_casadir_@:$MONO_PATH
 export LD_LIBRARY_PATH=@_prefix_@/lib:@_webbindir_@:/usr/lib:$LD_LIBRARY_PATH
+export LC_ALL=C
 cd @_prefix_@/bin
 mono iFolderClient.exe $@
 
