diff -p -up gtkpod-0.99.12/scripts/sync-webcalendar.sh.mountpoint gtkpod-0.99.12/scripts/sync-webcalendar.sh
--- gtkpod-0.99.12/scripts/sync-webcalendar.sh.mountpoint	2007-12-18 05:37:41.000000000 +0100
+++ gtkpod-0.99.12/scripts/sync-webcalendar.sh	2007-12-18 09:27:00.000000000 +0100
@@ -9,7 +9,7 @@
 # with the following defaults:
 
 # mount point of ipod
-IPOD_MOUNT='/mnt/ipod'
+IPOD_MOUNT='/media/ipod'
 # uri for webcalendar (example)
 DATAFILE='https://somewhere.local/calendar.ics'
 # calendar name
diff -p -up gtkpod-0.99.12/scripts/sync-thunderbird.sh.mountpoint gtkpod-0.99.12/scripts/sync-thunderbird.sh
--- gtkpod-0.99.12/scripts/sync-thunderbird.sh.mountpoint	2007-12-18 05:37:41.000000000 +0100
+++ gtkpod-0.99.12/scripts/sync-thunderbird.sh	2007-12-18 09:27:00.000000000 +0100
@@ -18,7 +18,7 @@
 
 # with the following defaults: 
 
-IPOD_MOUNT=/mnt/ipod         # mountpoint of ipod
+IPOD_MOUNT=/media/ipod         # mountpoint of ipod
 ENCODING=ISO-8859-15         # encoding used by ipod
 NAME=thunderbird             # default file export name
 
diff -p -up gtkpod-0.99.12/scripts/sync-thunderbird-nano.sh.mountpoint gtkpod-0.99.12/scripts/sync-thunderbird-nano.sh
--- gtkpod-0.99.12/scripts/sync-thunderbird-nano.sh.mountpoint	2007-12-18 05:37:41.000000000 +0100
+++ gtkpod-0.99.12/scripts/sync-thunderbird-nano.sh	2007-12-18 09:27:00.000000000 +0100
@@ -25,7 +25,7 @@
 
 # with the following defaults: 
 
-IPOD_MOUNT=/mnt/ipod         # mountpoint of ipod
+IPOD_MOUNT=/media/ipod         # mountpoint of ipod
 ENCODING=ISO-8859-15         # encoding used by ipod
 NAME=thunderbird             # default file export name
 FILE_FLAG=''		     # flag used to determine end of file
diff -p -up gtkpod-0.99.12/scripts/sync-ldif.sh.mountpoint gtkpod-0.99.12/scripts/sync-ldif.sh
--- gtkpod-0.99.12/scripts/sync-ldif.sh.mountpoint	2007-12-18 05:37:41.000000000 +0100
+++ gtkpod-0.99.12/scripts/sync-ldif.sh	2007-12-18 09:27:00.000000000 +0100
@@ -15,7 +15,7 @@
 # differently...This is still Chinese for me!! :)
 
 export LDIFAMILYNAME=contactIPOD	# Filenames will look like $LDIFAMILYNAMEXX.vcf, X=[0-9]
-export IPOD_MOUNT=/mnt/ipod		# Mount point of the ipod
+export IPOD_MOUNT=/media/ipod		# Mount point of the ipod
 declare LDIFILE=addressbook.ldif	# default filename 'addressbook.ldif'
 declare ENCODING=ISO-8859-15            # To try others encodings : 'iconv --list'
 declare DELETE="NO"			# To delete old .vcf files by default? 'NO'!!
diff -p -up gtkpod-0.99.12/scripts/sync-korganizer.sh.mountpoint gtkpod-0.99.12/scripts/sync-korganizer.sh
--- gtkpod-0.99.12/scripts/sync-korganizer.sh.mountpoint	2007-12-18 05:37:41.000000000 +0100
+++ gtkpod-0.99.12/scripts/sync-korganizer.sh	2007-12-18 09:27:00.000000000 +0100
@@ -9,7 +9,7 @@
 #
 # with the following defaults: 
 
-IPOD_MOUNT=/mnt/ipod                          # mountpoint of ipod
+IPOD_MOUNT=/media/ipod                          # mountpoint of ipod
 DATAFILE=~/.kde/share/apps/korganizer/std.ics # calendar data file
 ENCODING=ISO-8859-15                          # encoding used by ipod
 
diff -p -up gtkpod-0.99.12/scripts/sync-knotes.sh.mountpoint gtkpod-0.99.12/scripts/sync-knotes.sh
--- gtkpod-0.99.12/scripts/sync-knotes.sh.mountpoint	2007-12-18 05:37:41.000000000 +0100
+++ gtkpod-0.99.12/scripts/sync-knotes.sh	2007-12-18 09:27:00.000000000 +0100
@@ -6,7 +6,7 @@
 # sync-notes.sh [-i <ipod mountpoint>] [-e <encoding>]
 #
 # defaults:
-IPOD_MOUNT=/media/iPod
+IPOD_MOUNT=/media/ipod
 ENCODING=ISO-8859-15                          # encoding used by ipod
 
 #First start knotes if it is not started already:
diff -p -up gtkpod-0.99.12/scripts/sync-kaddressbook.sh.mountpoint gtkpod-0.99.12/scripts/sync-kaddressbook.sh
--- gtkpod-0.99.12/scripts/sync-kaddressbook.sh.mountpoint	2007-12-18 05:37:41.000000000 +0100
+++ gtkpod-0.99.12/scripts/sync-kaddressbook.sh	2007-12-18 09:27:00.000000000 +0100
@@ -9,7 +9,7 @@
 #
 # with the following defaults: 
 
-IPOD_MOUNT=/mnt/ipod                          # mountpoint of ipod
+IPOD_MOUNT=/media/ipod                          # mountpoint of ipod
 DATAFILE=~/.kde/share/apps/kabc/std.vcf       # vcard file
 ENCODING=ISO-8859-15                          # encoding used by ipod
 
diff -p -up gtkpod-0.99.12/scripts/sync-evocalendar.sh.mountpoint gtkpod-0.99.12/scripts/sync-evocalendar.sh
--- gtkpod-0.99.12/scripts/sync-evocalendar.sh.mountpoint	2007-12-18 05:37:41.000000000 +0100
+++ gtkpod-0.99.12/scripts/sync-evocalendar.sh	2007-12-18 09:27:00.000000000 +0100
@@ -10,7 +10,7 @@
 #
 # with the following defaults: 
 
-IPOD_MOUNT=/mnt/ipod                          # mountpoint of ipod
+IPOD_MOUNT=/media/ipod                          # mountpoint of ipod
 
 #the path to a script that will be passed the ical information from STDIN and filter, if needed
 #FILTER_SCRIPT=
diff -p -up gtkpod-0.99.12/scripts/sync-abook.sh.mountpoint gtkpod-0.99.12/scripts/sync-abook.sh
--- gtkpod-0.99.12/scripts/sync-abook.sh.mountpoint	2007-12-18 05:37:41.000000000 +0100
+++ gtkpod-0.99.12/scripts/sync-abook.sh	2007-12-18 09:27:00.000000000 +0100
@@ -9,7 +9,7 @@
 #
 # with the following defaults: 
 
-IPOD_MOUNT='/mnt/ipod'				# mount point of ipod
+IPOD_MOUNT='/media/ipod'			# mount point of ipod
 DATAFILE='~/.abook/addressbook'	                # the abook db
 ENCODING_FROM=UTF-8                             # encoding used by abook
 ENCODING=ISO-8859-15                            # encoding used by ipod
diff -p -up gtkpod-0.99.12/scripts/sync-evolution.sh.mountpoint gtkpod-0.99.12/scripts/sync-evolution.sh
--- gtkpod-0.99.12/scripts/sync-evolution.sh.mountpoint	2007-12-18 05:37:41.000000000 +0100
+++ gtkpod-0.99.12/scripts/sync-evolution.sh	2007-12-18 09:27:24.000000000 +0100
@@ -13,7 +13,7 @@
 #
 # with the following defaults: 
 
-IPOD_MOUNT=/mnt/ipod         # mountpoint of ipod
+IPOD_MOUNT=/media/ipod         # mountpoint of ipod
 ENCODING=ISO-8859-15         # encoding used by ipod
 
 # try to find the path to evolution-addressbook-export
