diff -Naur gedit-plugins-2.22.3.orig/plugins/bracketcompletion/Makefile.in gedit-plugins-2.22.3/plugins/bracketcompletion/Makefile.in
--- gedit-plugins-2.22.3.orig/plugins/bracketcompletion/Makefile.in	2008-09-28 21:43:56.000000000 +0300
+++ gedit-plugins-2.22.3/plugins/bracketcompletion/Makefile.in	2008-09-28 21:45:54.000000000 +0300
@@ -54,7 +54,7 @@
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
 am__installdirs = "$(DESTDIR)$(plugindir)" "$(DESTDIR)$(plugindir)"
 pluginPYTHON_INSTALL = $(INSTALL_DATA)
-py_compile = $(top_srcdir)/py-compile
+py_compile = echo
 pluginDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(plugin_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
diff -Naur gedit-plugins-2.22.3.orig/plugins/codecomment/Makefile.in gedit-plugins-2.22.3/plugins/codecomment/Makefile.in
--- gedit-plugins-2.22.3.orig/plugins/codecomment/Makefile.in	2008-09-28 21:43:56.000000000 +0300
+++ gedit-plugins-2.22.3/plugins/codecomment/Makefile.in	2008-09-28 21:45:54.000000000 +0300
@@ -54,7 +54,7 @@
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
 am__installdirs = "$(DESTDIR)$(plugindir)" "$(DESTDIR)$(plugindir)"
 pluginPYTHON_INSTALL = $(INSTALL_DATA)
-py_compile = $(top_srcdir)/py-compile
+py_compile = echo
 pluginDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(plugin_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
diff -Naur gedit-plugins-2.22.3.orig/plugins/colorpicker/Makefile.in gedit-plugins-2.22.3/plugins/colorpicker/Makefile.in
--- gedit-plugins-2.22.3.orig/plugins/colorpicker/Makefile.in	2008-09-28 21:43:56.000000000 +0300
+++ gedit-plugins-2.22.3/plugins/colorpicker/Makefile.in	2008-09-28 21:45:54.000000000 +0300
@@ -54,7 +54,7 @@
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
 am__installdirs = "$(DESTDIR)$(plugindir)" "$(DESTDIR)$(plugindir)"
 pluginPYTHON_INSTALL = $(INSTALL_DATA)
-py_compile = $(top_srcdir)/py-compile
+py_compile = echo
 pluginDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(plugin_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
diff -Naur gedit-plugins-2.22.3.orig/plugins/drawspaces/Makefile.in gedit-plugins-2.22.3/plugins/drawspaces/Makefile.in
--- gedit-plugins-2.22.3.orig/plugins/drawspaces/Makefile.in	2008-09-28 21:43:56.000000000 +0300
+++ gedit-plugins-2.22.3/plugins/drawspaces/Makefile.in	2008-09-28 21:45:54.000000000 +0300
@@ -56,7 +56,7 @@
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
 am__installdirs = "$(DESTDIR)$(plugindir)" "$(DESTDIR)$(plugindir)"
 pluginPYTHON_INSTALL = $(INSTALL_DATA)
-py_compile = $(top_srcdir)/py-compile
+py_compile = echo
 pluginDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(plugin_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
diff -Naur gedit-plugins-2.22.3.orig/plugins/joinlines/Makefile.in gedit-plugins-2.22.3/plugins/joinlines/Makefile.in
--- gedit-plugins-2.22.3.orig/plugins/joinlines/Makefile.in	2008-09-28 21:43:56.000000000 +0300
+++ gedit-plugins-2.22.3/plugins/joinlines/Makefile.in	2008-09-28 21:45:54.000000000 +0300
@@ -54,7 +54,7 @@
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
 am__installdirs = "$(DESTDIR)$(plugindir)" "$(DESTDIR)$(plugindir)"
 pluginPYTHON_INSTALL = $(INSTALL_DATA)
-py_compile = $(top_srcdir)/py-compile
+py_compile = echo
 pluginDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(plugin_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
diff -Naur gedit-plugins-2.22.3.orig/plugins/Makefile.in gedit-plugins-2.22.3/plugins/Makefile.in
--- gedit-plugins-2.22.3.orig/plugins/Makefile.in	2008-09-28 21:43:56.000000000 +0300
+++ gedit-plugins-2.22.3/plugins/Makefile.in	2008-09-28 21:45:54.000000000 +0300
@@ -60,7 +60,7 @@
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
 am__installdirs = "$(DESTDIR)$(plugindir)"
 pluginPYTHON_INSTALL = $(INSTALL_DATA)
-py_compile = $(top_srcdir)/py-compile
+py_compile = echo
 RECURSIVE_CLEAN_TARGETS = mostlyclean-recursive clean-recursive	\
   distclean-recursive maintainer-clean-recursive
 ETAGS = etags
diff -Naur gedit-plugins-2.22.3.orig/plugins/sessionsaver/Makefile.in gedit-plugins-2.22.3/plugins/sessionsaver/Makefile.in
--- gedit-plugins-2.22.3.orig/plugins/sessionsaver/Makefile.in	2008-09-28 21:43:56.000000000 +0300
+++ gedit-plugins-2.22.3/plugins/sessionsaver/Makefile.in	2008-09-28 21:45:54.000000000 +0300
@@ -57,7 +57,7 @@
 am__installdirs = "$(DESTDIR)$(sessionsaverdir)" \
 	"$(DESTDIR)$(plugindir)" "$(DESTDIR)$(sessionsaverdir)"
 sessionsaverPYTHON_INSTALL = $(INSTALL_DATA)
-py_compile = $(top_srcdir)/py-compile
+py_compile = echo
 pluginDATA_INSTALL = $(INSTALL_DATA)
 sessionsaverDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(plugin_DATA) $(sessionsaver_DATA)
diff -Naur gedit-plugins-2.22.3.orig/plugins/smartspaces/Makefile.in gedit-plugins-2.22.3/plugins/smartspaces/Makefile.in
--- gedit-plugins-2.22.3.orig/plugins/smartspaces/Makefile.in	2008-09-28 21:43:56.000000000 +0300
+++ gedit-plugins-2.22.3/plugins/smartspaces/Makefile.in	2008-09-28 21:45:54.000000000 +0300
@@ -56,7 +56,7 @@
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
 am__installdirs = "$(DESTDIR)$(plugindir)" "$(DESTDIR)$(plugindir)"
 pluginPYTHON_INSTALL = $(INSTALL_DATA)
-py_compile = $(top_srcdir)/py-compile
+py_compile = echo
 pluginDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(plugin_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
diff -Naur gedit-plugins-2.22.3.orig/plugins/terminal/Makefile.in gedit-plugins-2.22.3/plugins/terminal/Makefile.in
--- gedit-plugins-2.22.3.orig/plugins/terminal/Makefile.in	2008-09-28 21:43:56.000000000 +0300
+++ gedit-plugins-2.22.3/plugins/terminal/Makefile.in	2008-09-28 21:45:54.000000000 +0300
@@ -56,7 +56,7 @@
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
 am__installdirs = "$(DESTDIR)$(plugindir)" "$(DESTDIR)$(plugindir)"
 pluginPYTHON_INSTALL = $(INSTALL_DATA)
-py_compile = $(top_srcdir)/py-compile
+py_compile = echo
 pluginDATA_INSTALL = $(INSTALL_DATA)
 DATA = $(plugin_DATA)
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
