--- setup.py.orig	2007-04-16 02:05:17.000000000 +0300
+++ setup.py	2007-08-22 13:13:36.000000000 +0300
@@ -46,8 +46,6 @@
     package_dir = {'osh': '.'},
     packages = ['osh', 'osh.command', 'osh.oshtypes'],
     data_files = [('/usr/bin', ['bin/osh', 'bin/remoteosh', 'bin/oshtestssh']),
-                  ('/usr/share/osh', ['bin/remoteinstall', '../osh-0.8.1.tar.gz']),
-                  ('/usr/share/osh/scripts', scripts),
                   ('/usr/share/osh/test', test),
                   ('/usr/share/doc/osh', doc),
                   ('/usr/share/doc/osh/web', doc_web)]
