--- panel-plugin/diskperf.desktop.in.in.orig	2008-07-18 21:21:07.000000000 +0300
+++ panel-plugin/diskperf.desktop.in.in	2008-07-18 21:21:42.000000000 +0300
@@ -1,8 +1,10 @@
 [Xfce Panel]
 Type=X-XFCE-PanelPlugin
 Encoding=UTF-8
-_Name=Disk Performance Monitor
-_Comment=Show disk performance
+Name=Disk Performance Monitor
+Name[tr]=Disk Performans Ekranı
+Comment=Show disk performance
+Comment[tr]=Disk performansını gösterir
 Icon=gtk-harddisk
 X-XFCE-Exec=@libexecdir@/xfce4/panel-plugins/xfce4-diskperf-plugin
 
