--- qdvdauthor.desktop	2007-05-13 06:14:12.000000000 +0300
+++ qdvdauthor2.desktop	2007-09-13 02:01:30.000000000 +0300
@@ -1,8 +1,12 @@
 [Desktop Entry]
 Name=QDVDAuthor
+Name[tr]=QDVDAuthor
+GenericName=DVD Author
+GenericName[tr]=DVD Oluşturma
 Comment=DVD Author GUI
-Icon=/usr/share/pixmaps/qdvdauthor.png
-Exec=/usr/bin/qdvdauthor
+Comment[tr]=DVD Oluşturmak İçin Arayüz
+Icon=qdvdauthor
+Exec=qdvdauthor
 Terminal=false
 Type=Application
 Encoding=UTF-8
