diff -Nuar comedilib-0.7.22.orig/comedi_calibrate/Makefile.am comedilib-0.7.22/comedi_calibrate/Makefile.am
--- comedilib-0.7.22.orig/comedi_calibrate/Makefile.am	2003-10-13 21:52:23.000000000 +0300
+++ comedilib-0.7.22/comedi_calibrate/Makefile.am	2007-07-18 11:06:34.000000000 +0300
@@ -8,8 +8,6 @@
 comedi_calibrate_CFLAGS = $(COMEDILIB_CFLAGS)
 comedi_calibrate_LDADD = $(COMEDILIB_LIBS)
 
-install-data-local:
-	$(INSTALL) -d $(localstatedir)/lib/comedi/calibrations
 
 EXTRA_DIST = results
 
diff -Nuar comedilib-0.7.22.orig/comedi_calibrate/Makefile.in comedilib-0.7.22/comedi_calibrate/Makefile.in
--- comedilib-0.7.22.orig/comedi_calibrate/Makefile.in	2004-10-03 01:10:28.000000000 +0300
+++ comedilib-0.7.22/comedi_calibrate/Makefile.in	2007-07-18 11:07:12.000000000 +0300
@@ -638,8 +638,6 @@
 	uninstall-info-am
 
 
-install-data-local:
-	$(INSTALL) -d $(localstatedir)/lib/comedi/calibrations
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
