--- src/setup/mime/codeblocks.desktop.orig	2008-07-06 19:24:20.000000000 +0300
+++ src/setup/mime/codeblocks.desktop	2008-07-06 19:26:37.000000000 +0300
@@ -3,12 +3,12 @@
 Encoding=UTF-8
 Name=Code::Blocks IDE
 Comment=Configurable and extensible IDE
+Comment[tr]=Yapılandırılabilir ve eklentilerle genişletilebilir IDE
 Exec=codeblocks %F
-Icon=codeblocks.png
+Icon=codeblocks
 Terminal=false
 X-MultipleArgs=false
 Type=Application
 Categories=Application;Development;
 StartupNotify=true
 MimeType=application/x-codeblocks;application/x-codeblocks-workspace;
-GenericName[en_GR]=
