diff -Nuar wbar-1.3.3-orig/IconLoader.cc wbar-1.3.3/IconLoader.cc
--- wbar-1.3.3-orig/IconLoader.cc	2010-04-12 21:17:18.940296039 +0300
+++ wbar-1.3.3/IconLoader.cc	2010-04-12 21:18:35.221046331 +0300
@@ -2,6 +2,7 @@
 #include <unistd.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <cstdlib>
 };
 
 #include "IconLoader.h"
diff -Nuar wbar-1.3.3-orig/SuperBar.cc wbar-1.3.3/SuperBar.cc
--- wbar-1.3.3-orig/SuperBar.cc	2010-04-12 21:17:18.943295839 +0300
+++ wbar-1.3.3/SuperBar.cc	2010-04-12 21:18:53.483067109 +0300
@@ -1,5 +1,6 @@
 #include "SuperBar.h"
 #include "SuperIcon.h"
+#include <cstdlib>
 
 using namespace std;
 
