diff -Nuar wbar-1.3.3-orig/Makefile wbar-1.3.3/Makefile
--- wbar-1.3.3-orig/Makefile	2010-04-12 21:17:18.942295759 +0300
+++ wbar-1.3.3/Makefile	2010-04-14 00:00:34.681126284 +0300
@@ -1,9 +1,8 @@
 TARGET=wbar
-CXXFLAGS=`imlib2-config --cflags` -Wall -O2 #-DCOS_ZOOM #-DLINEAR_TRASL -DNO_EXPAND #-DAVGFILTER
-LDFLAGS=`imlib2-config --libs`  -Wl,-O2
+CXXFLAGS=`imlib2-config --cflags` -Wall -O2 #-DCOS_ZOOM #-DLINEAR_TRASL -DNO_EXPAND #-DAVGFILTER#
+LDFLAGS=-L/usr/lib -lImlib2 -L/usr/X11R6/lib -lX11 -lm -Wl,-O2
 PREFIX=/usr/share/wbar
 
-	
 sources= XWin.cc Icon.cc Bar.cc IconLoader.cc \
 	 OptParser.cc SuperBar.cc SuperIcon.cc
 objects= $(sources:.cc=.o) Main.o
