diff -Naur compiz-0.5.4.orig/configure.ac compiz-0.5.4/configure.ac
--- compiz-0.5.4.orig/configure.ac	2007-08-20 19:14:10.000000000 +0300
+++ compiz-0.5.4/configure.ac	2007-08-21 20:59:28.000000000 +0300
@@ -176,26 +176,6 @@
   AC_DEFINE(USE_GLIB, 1, [Build glib plugin])
 fi
 
-AC_ARG_ENABLE(gconf,
-  [  --disable-gconf         Disable gconf plugin],
-  [use_gconf=$enableval], [use_gconf=yes])
-
-if test "x$use_gconf" = "xyes"; then
-  PKG_CHECK_MODULES(GCONF, gconf-2.0)
-
-  AC_PATH_PROG(GCONFTOOL, gconftool-2, no)
-  if test x"$GCONFTOOL" = xno; then
-    AC_MSG_ERROR([gconftool-2 executable not found in your path - should be installed with GConf])
-  fi
-fi
-
-AM_GCONF_SOURCE_2
-
-AM_CONDITIONAL(USE_GCONF, test "x$use_gconf" = "xyes")
-if test "$use_gconf" = yes; then
-  AC_DEFINE(USE_GCONF, 1, [Build gconf plugin])
-fi
-
 stylesheetdir=$datadir/compiz
 AC_SUBST(stylesheetdir)
 
diff -Naur compiz-0.5.4.orig/gtk/window-decorator/Makefile.am compiz-0.5.4/gtk/window-decorator/Makefile.am
--- compiz-0.5.4.orig/gtk/window-decorator/Makefile.am	2007-05-11 20:01:09.000000000 +0300
+++ compiz-0.5.4/gtk/window-decorator/Makefile.am	2007-08-21 21:01:21.000000000 +0300
@@ -9,7 +9,6 @@
 gtk_window_decorator_LDADD	  =		       \
 	$(top_builddir)/libdecoration/libdecoration.la \
 	@GTK_WINDOW_DECORATOR_LIBS@		       \
-	@GCONF_LIBS@				       \
 	@DBUS_GLIB_LIBS@			       \
 	$(metacitylibs)
 gtk_window_decorator_SOURCES	  = \
@@ -17,21 +16,7 @@
 gtk_window_decorator_program = gtk-window-decorator
 endif
 
-if USE_GCONF
-schemadir       = $(GCONF_SCHEMA_FILE_DIR)
-schema_in_files = gwd.schemas.in
-schema_files    = $(schema_in_files:.schemas.in=.schemas)
-schema_DATA     = $(schema_files)
-
-@INTLTOOL_SCHEMAS_RULE@
-
-install-data-local:
-	if test -z "$(DESTDIR)" ; then \
-		GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(schema_DATA); \
-	fi
-endif
-
-INCLUDES = -I$(top_srcdir)/include @GTK_WINDOW_DECORATOR_CFLAGS@ @METACITY_CFLAGS@ @GCONF_CFLAGS@ @DBUS_GLIB_CFLAGS@
+INCLUDES = -I$(top_srcdir)/include @GTK_WINDOW_DECORATOR_CFLAGS@ @METACITY_CFLAGS@ @DBUS_GLIB_CFLAGS@
 
 bin_PROGRAMS = $(gtk_window_decorator_program)
 
diff -Naur compiz-0.5.4.orig/Makefile.am compiz-0.5.4/Makefile.am
--- compiz-0.5.4.orig/Makefile.am	2007-08-15 07:54:27.000000000 +0300
+++ compiz-0.5.4/Makefile.am	2007-08-21 21:02:02.000000000 +0300
@@ -23,10 +23,6 @@
 	po/stamp-it	 \
 	po/.intltool-merge-cache
 
-if USE_GCONF
-gconfdata = compiz-gconf.pc
-endif
-
 if USE_KCONFIG
 kconfigdata = compiz-kconfig.pc
 endif
diff -Naur compiz-0.5.4.orig/metadata/Makefile.am compiz-0.5.4/metadata/Makefile.am
--- compiz-0.5.4.orig/metadata/Makefile.am	2007-08-20 19:48:36.000000000 +0300
+++ compiz-0.5.4/metadata/Makefile.am	2007-08-21 21:03:27.000000000 +0300
@@ -35,18 +35,6 @@
 
 @INTLTOOL_XML_RULE@
 
-if USE_GCONF
-schemadir    = $(GCONF_SCHEMA_FILE_DIR)
-schema_files = $(patsubst %.xml.in,compiz-%.schemas,$(xml_in_files))
-schema_DATA  = $(schema_files)
-
-%.schemas: $(xml_files)
-	xsltproc -o $@ $(srcdir)/schemas.xslt $(subst compiz-,,$*).xml;
-
-schemas_stylesheets = schemas.xslt
-
-endif
-
 if USE_KCONFIG
 kde_kcfgdir    = $(DESTDIR)$(KDE_KCFG_DIR)
 kde_kcfg_files = $(patsubst %.xml.in,compiz-%.kcfg,$(xml_in_files))
@@ -71,14 +59,6 @@
 	$(kde_kcfg_files) \
 	$(kde_config_files)
 
-install-data-local:
-if USE_GCONF
-if GCONF_SCHEMAS_INSTALL
-	if test -z "$(DESTDIR)" ; then \
-		GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(schema_DATA); \
-        fi
-endif
-endif
 if USE_KCONFIG
 	if mkdir -p $(kde_kcfgdir) && test -w $(kde_kcfgdir); then ( \
 		$(INSTALL) -m 644 $(kde_kcfg_files) $(kde_kcfgdir) \
diff -Naur compiz-0.5.4.orig/plugins/Makefile.am compiz-0.5.4/plugins/Makefile.am
--- compiz-0.5.4.orig/plugins/Makefile.am	2007-08-13 19:11:16.000000000 +0300
+++ compiz-0.5.4/plugins/Makefile.am	2007-08-21 21:04:08.000000000 +0300
@@ -89,13 +89,6 @@
 libglib_module = libglib.la
 endif
 
-if USE_GCONF
-libgconf_la_LDFLAGS = -module -avoid-version -no-undefined
-libgconf_la_LIBADD = @GCONF_LIBS@
-libgconf_la_SOURCES = gconf.c
-libgconf_module = libgconf.la
-endif
-
 if USE_KCONFIG
 libkconfig_la_LDFLAGS = -module -avoid-version -no-undefined
 libkconfig_la_LIBADD = @KCONFIG_LIBS@
@@ -135,7 +128,6 @@
 	@LIBPNG_CFLAGS@				\
 	@LIBRSVG_CFLAGS@			\
 	@ANNOTATE_CFLAGS@			\
-	@GCONF_CFLAGS@				\
 	@KCONFIG_CFLAGS@			\
 	@PLACE_CFLAGS@				\
 	@DBUS_CFLAGS@				\
