--- src/KLNfancywidgets.old.cpp	2004-04-27 05:28:22.000000000 +0300
+++ src/KLNfancywidgets.cpp	2006-06-26 11:48:46.000000000 +0300
@@ -43,7 +43,7 @@
   colorchanged=false;
 }
 
-FancyButton::~FancyButton() {};
+FancyButton::~FancyButton() {}
 
 void FancyButton::drawButton ( QPainter * p)
 {
@@ -81,7 +81,7 @@
   color=QColor(0,0,0);
 }
  
-FancyQLCDNumber::~FancyQLCDNumber() {};
+FancyQLCDNumber::~FancyQLCDNumber() {}
   
 void FancyQLCDNumber::setOK(bool isItOK)  // isItOK=> green, !isItOK=>red
 {
