diff -Nuar autopano-sift-2.4.orig/src/util/Makefile autopano-sift-2.4/src/util/Makefile
--- autopano-sift-2.4.orig/src/util/Makefile	2005-01-15 18:33:38.000000000 +0200
+++ autopano-sift-2.4/src/util/Makefile	2006-12-04 23:20:54.000000000 +0200
@@ -10,7 +10,7 @@
 MAKE_MONO=
 LIBS_MONO=/r:System.Drawing
 
-USING_GTK=/define:USING_GTK /pkg:gtk-sharp /pkg:glade-sharp
+USING_GTK=/define:USING_GTK /pkg:gtk-sharp-2.0 /pkg:glade-sharp-2.0
 
 LIBSIFT=/r:../libsift.dll
 SOURCES=DrawingPrimitives.cs BasicImagingInterface.cs