diff -Nuar ltsp-utils.orig/ltspcfg ltsp-utils/ltspcfg
--- ltsp-utils.orig/ltspcfg	2006-04-17 15:41:57.000000000 +0300
+++ ltsp-utils/ltspcfg	2006-11-24 09:31:23.000000000 +0200
@@ -56,7 +56,8 @@
                        );
 
 my @kdm_locs         = ( "/usr/bin/kdm",
-                         "/opt/kde/bin/kdm",
+                         "/usr/kde/3.5/bin",
+			 "/opt/kde/bin/kdm",
                          "/opt/kde2/bin/kdm",
                          "/opt/kde3/bin/kdm",
                        );
@@ -64,6 +65,7 @@
 my @kdmrc_locs       = ( "/etc/kde/kdm/kdmrc",
                          "/etc/kde3/kdm/kdmrc",
                          "/etc/X11/xdm/kdmrc",
+			 "/etc/X11/kdm/kdmrc",
                          "/usr/share/config/kdm/kdmrc",
                          "/etc/opt/kde3/share/config/kdm/kdmrc",
                          "/opt/kde3/share/config/kdm/kdmrc",
