diff -Naur lives-0.9.8.6-org/Makefile.am lives-0.9.8.6/Makefile.am
--- lives-0.9.8.6-org/Makefile.am	2007-06-08 05:28:04.000000000 +0300
+++ lives-0.9.8.6/Makefile.am	2007-11-01 14:23:02.000000000 +0200
@@ -34,11 +34,14 @@
 	$(doc_DATA)
 
 installdirs: 
-	rm -rf $(datadir)/$(PACKAGE)/*
-	mkinstalldirs
-	$(srcdir)/mkinstalldirs \
-        $(DESTDIR)$(bindir) $(DESTDIR)$(datadir) \
-        $(DESTDIR)$(docdir) $(DESTDIR)/usr/share/applications/ $(DESTDIR)/usr/share/pixmaps/
+#	rm -rf $(datadir)/$(PACKAGE)/*
+#	mkinstalldirs
+#	$(srcdir)/mkinstalldirs \
+        mkdir -p $(DESTDIR)$(bindir) \
+	mkdir -p $(DESTDIR)$(datadir) \
+	mkdir -p $(DESTDIR)$(docdir) \
+	mkdir -p $(DESTDIR)/usr/share/applications/ \
+	mkdir -p $(DESTDIR)/usr/share/pixmaps/
 
 
 install-exec-local:
diff -Naur lives-0.9.8.6-org/Makefile.in lives-0.9.8.6/Makefile.in
--- lives-0.9.8.6-org/Makefile.in	2007-07-11 08:17:00.000000000 +0300
+++ lives-0.9.8.6/Makefile.in	2007-11-01 14:23:55.000000000 +0200
@@ -748,11 +748,14 @@
 
 
 installdirs: 
-	rm -rf $(datadir)/$(PACKAGE)/*
-	mkinstalldirs
-	$(srcdir)/mkinstalldirs \
-        $(DESTDIR)$(bindir) $(DESTDIR)$(datadir) \
-        $(DESTDIR)$(docdir) $(DESTDIR)/usr/share/applications/ $(DESTDIR)/usr/share/pixmaps/
+#	rm -rf $(datadir)/$(PACKAGE)/*
+#	mkinstalldirs
+#	$(srcdir)/mkinstalldirs \
+        mkdir -p $(DESTDIR)$(bindir) \
+	mkdir -p $(DESTDIR)$(datadir) \
+	mkdir -p $(DESTDIR)$(docdir) \
+	mkdir -p $(DESTDIR)/usr/share/applications/ \
+	mkdir -p $(DESTDIR)/usr/share/pixmaps/
 
 install-exec-local:
 	rm -rf $(DESTDIR)$(bindir)/lives
