--- configure~	2008-05-24 01:44:25.000000000 +0300
+++ configure	2008-10-11 14:40:29.000000000 +0300
@@ -11195,14 +11195,14 @@
 
 
 if test x$artsc_config_exec_prefix != x; then
-    artsc_config_args="$artsc_config_args --exec-prefix=$artsc_config_exec_prefix"
+#     artsc_config_args="$artsc_config_args --exec-prefix=$artsc_config_exec_prefix"
     if test x${ARTSC_CONFIG+set} != xset; then
 	ARTSC_CONFIG=$artsc_config_exec_prefix/bin/artsc-config
     fi
 fi
 
 if test x$artsc_config_prefix != x; then
-    artsc_config_args="$artsc_config_args --prefix=$artsc_config_prefix"
+#     artsc_config_args="$artsc_config_args --prefix=$artsc_config_prefix"
     if test x${ARTSC_CONFIG+set} != xset; then
   	ARTSC_CONFIG=$artsc_config_prefix/bin/artsc-config
     fi
