--- ristretto.desktop.in.orig	2008-01-21 08:50:09.000000000 +0200
+++ ristretto.desktop.in	2008-01-21 09:14:28.000000000 +0200
@@ -1,8 +1,11 @@
 [Desktop Entry]
 Version=1.0
-_Name=Ristretto Image Viewer
+_Name=Ristretto
+Name[tr]=Ristretto
 _Comment=Xfce Image viewer
+Comment[tr]=Xfce Resim Görüntüleyici
 _GenericName=Image Viewer
+GenericName[tr]=Resim Görüntüleyici
 Exec=ristretto %F
 Icon=ristretto
 Terminal=false
