--- compiz-manager.orig	2007-10-28 01:34:29.000000000 +0300
+++ compiz-manager	2007-10-28 01:40:29.000000000 +0300
@@ -27,11 +27,11 @@
 # 
 
 
-COMPIZ_BIN_PATH="/usr/local/bin/" # For window decorators and compiz
-PLUGIN_PATH="/usr/local/lib/compiz/" 
+COMPIZ_BIN_PATH="/usr/bin/" # For window decorators and compiz
+PLUGIN_PATH="/usr/lib/compiz/" 
 GLXINFO="/usr/bin/glxinfo"
 KWIN="/usr/bin/kwin"
-METACITY="/usr/bin/metacity"
+#METACITY="/usr/bin/metacity"
 COMPIZ_NAME="compiz" # Final name for compiz (compiz.real) 
 
 # For Xgl LD_PRELOAD
@@ -47,11 +47,11 @@
 # For detecting what driver is in use, the + is for one or more /'s
 XORG_DRIVER_PATH="/usr/lib/xorg/modules/drivers/+"
 
-FALLBACKWM="${METACITY}"
+#FALLBACKWM="${METACITY}"
 FALLBACKWM_OPTIONS="--replace $@"
 
 # Driver whitelist
-WHITELIST="nvidia intel ati radeon i810"
+WHITELIST="nvidia intel ati radeon i810 fglrx"
 
 # blacklist based on the pci ids 
 # See http://wiki.compiz-fusion.org/Hardware/Blacklist for details
@@ -262,14 +262,14 @@
 # check pciid blacklist
 have_blacklisted_pciid()
 {
-	OUTPUT=$(lspci -n)
+	OUTPUT=$(/usr/sbin/lspci -n)
 	for ID in ${BLACKLIST_PCIIDS}; do
 		if echo "$OUTPUT" | egrep -q "$ID"; then
 			verbose "Blacklisted PCIID '$ID' found \n"
 			return 0
 		fi
 	done
-	OUTPUT=$(lspci -vn | grep -i VGA)
+	OUTPUT=$(/usr/sbin/lspci -vn | grep -i VGA)
 	verbose "Detected PCI ID for VGA: $OUTPUT\n"
 	return 1
 }
