--- menu-entries/LiVES.desktop.orig	2008-10-26 09:50:29.000000000 +0200
+++ menu-entries/LiVES.desktop	2008-12-04 17:09:46.000000000 +0200
@@ -1,6 +1,10 @@
 [Desktop Entry]
 Name=LiVES 
+Name[tr]=LiVES
+GenericName=Video Editor/VJ tool
+GenericName[tr]=Görüntü işleme aracı
 Comment=Video Editor/VJ tool
+Comment[tr]=Görüntü işleme aracı
 #Categories=Application;AudioVideo;Multimedia-Video;
 Categories=Application;AudioVideo;AudioVideoEditing;
 Encoding=UTF-8
