--- menu-entries/LiVES.desktop.orig	2008-07-22 05:17:00.000000000 +0300
+++ menu-entries/LiVES.desktop	2008-10-02 00:30:47.000000000 +0300
@@ -1,6 +1,10 @@
 [Desktop Entry]
 Name=LiVES 
+Name[tr]=LiVES
+GenericName=Video Editor/VJ tool
+GenericName[tr]=Görüntü işleme aracı
 Comment=Video Editor/VJ tool
+Comment[tr]=Görüntü işleme aracı
 Categories=Application;AudioVideo;Multimedia-Video;
 Encoding=UTF-8
 Exec=lives
