--- Amaya/batch/script_install_gnomekde.in.orig	2006-12-12 10:25:07.000000000 +0200
+++ Amaya/batch/script_install_gnomekde.in	2007-01-10 22:49:21.000000000 +0200
@@ -57,27 +57,32 @@
   exit
 fi
 
-# --- creating /usr/bin/amaya-version
+# --- creating /usr/bin/amaya
 $ECHO "#!/bin/sh" >> /usr/bin/amaya
 $ECHO "Exec=env XLIB_SKIP_ARGB_VISUALS=1 $PREFIX/$AMAYA_INSTALLDIR/$AMAYAGUI/bin/amaya %u" >> /usr/bin/amaya
 $CHMOD /usr/bin/amaya
 
-# --- creating amaya-version.desktop
-$ECHO "--> Creating $GNOMEKDEDIR/applications/amaya-@APP_VER@.desktop file"
-$ECHO "[Desktop Entry]" > $GNOMEKDEDIR/applications/amaya-@APP_VER@.desktop
-$ECHO "Encoding=UTF-8" >> $GNOMEKDEDIR/applications/amaya-@APP_VER@.desktop
-$ECHO "Name=Amaya @APP_VER@" >> $GNOMEKDEDIR/applications/amaya-@APP_VER@.desktop
-$ECHO "Comment=Edit the Web" >> $GNOMEKDEDIR/applications/amaya-@APP_VER@.desktop
-$ECHO "Comment[fr]=Editez le Web" >> $GNOMEKDEDIR/applications/amaya-@APP_VER@.desktop
-$ECHO "Categories=Application;Network;Development;" >> $GNOMEKDEDIR/applications/amaya-@APP_VER@.desktop
-$ECHO "Exec=env XLIB_SKIP_ARGB_VISUALS=1 $PREFIX/$AMAYA_INSTALLDIR/$AMAYAGUI/bin/amaya %u" >> $GNOMEKDEDIR/applications/amaya-@APP_VER@.desktop
-$ECHO "Icon=amaya-@APP_VER@.png" >> $GNOMEKDEDIR/applications/amaya-@APP_VER@.desktop
-$ECHO "Terminal=false" >> $GNOMEKDEDIR/applications/amaya-@APP_VER@.desktop
-$ECHO "Type=Application" >> $GNOMEKDEDIR/applications/amaya-@APP_VER@.desktop
-$ECHO "MimeType=text/html;text/xml;application/xhtml+xml;application/xml" >> $GNOMEKDEDIR/applications/amaya-@APP_VER@.desktop
+# --- creating amaya.desktop
+$ECHO "--> Creating $GNOMEKDEDIR/applications/amaya.desktop file"
+$ECHO "[Desktop Entry]" > $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "Encoding=UTF-8" >> $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "Name=Amaya" >> $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "Comment=Edit the Web" >> $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "Comment[fr]=Editez le Web" >> $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "Comment[nl]=Bewerk het web" >> $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "Comment[tr]=Web Düzenleyicisi" >> $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "GenericName=Edit the Web" >> $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "GenericName[nl]=Bewerk het web" >> $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "GenericName[tr]=Web Düzenleyicisi" >> $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "Categories=Application;Network;Development;" >> $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "Exec=env XLIB_SKIP_ARGB_VISUALS=1 $PREFIX/$AMAYA_INSTALLDIR/$AMAYAGUI/bin/amaya %u" >> $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "Icon=amaya.png" >> $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "Terminal=false" >> $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "Type=Application" >> $GNOMEKDEDIR/applications/amaya.desktop
+$ECHO "MimeType=text/html;text/xml;application/xhtml+xml;application/xml" >> $GNOMEKDEDIR/applications/amaya.desktop
 
-# --- creating amaya-version.png
-$ECHO "--> Creating $GNOMEKDEDIR/pixmaps/amaya-@APP_VER@.png"
-$CP $AMAYA_SOURCEBINDIR/../../resources/bundle/logo.png $GNOMEKDEDIR/pixmaps/amaya-@APP_VER@.png
+# --- creating amaya.png
+$ECHO "--> Creating $GNOMEKDEDIR/pixmaps/amaya.png"
+$CP $AMAYA_SOURCEBINDIR/../../resources/bundle/logo.png $GNOMEKDEDIR/pixmaps/amaya.png
 
 $ECHO "-- Installing Gnome/Kde integration for $AMAYA_INSTALLDIR into $GNOMEKDEDIR done --"
