diff -Naur Amaya.orig/configure.in Amaya/configure.in
--- Amaya.orig/configure.in	2007-04-18 20:15:56.000000000 +0300
+++ Amaya/configure.in	2007-04-18 20:22:41.000000000 +0300
@@ -1416,7 +1416,6 @@
 
 EXTRA_MAKEFILE_IN=
 make_output="Makefile Options.orig:Options.in tools/Makefile \
-       libjpeg/Makefile libpng/Makefile libpng/zlib/Makefile \
        tools/mkdep/Makefile \
        tools/cextract-1.7/Makefile thotlib/Makefile \
        thotlib/include/Makefile batch/Makefile \
diff -Naur Amaya.orig/Makefile.in Amaya/Makefile.in
--- Amaya.orig/Makefile.in	2007-04-18 20:04:02.000000000 +0300
+++ Amaya/Makefile.in	2007-04-18 20:23:46.000000000 +0300
@@ -235,9 +235,6 @@
 $(THOTDIR)/Makefile.in \
 $(THOTDIR)/amaya/Makefile.in \
 $(THOTDIR)/batch/Makefile.in \
-$(THOTDIR)/libjpeg/Makefile.in \
-$(THOTDIR)/libpng/Makefile.in \
-$(THOTDIR)/libpng/zlib/Makefile.in \
 $(THOTDIR)/thotlib/Makefile.in \
 $(THOTDIR)/thotlib/include/Makefile.in \
 $(THOTDIR)/tools/Makefile.in \
