--- /dev/null	2006-09-13 23:21:02.789316912 +0300
+++ prokyon3.desktop	2006-09-13 23:25:20.000000000 +0300
@@ -0,0 +1,13 @@
+[Desktop Entry]
+Categories=Qt;Application;AudioVideo;Audio;Player;
+Comment=Müzik Yöneticisi
+Comment[tr]=Müzik Yöneticisi
+Encoding=UTF-8
+Exec=prokyon3
+GenericName=Music Manager
+GenericName[tr]=Müzik Yöneticisi
+Icon=prokyon.png
+Name=Prokyon
+Name[tr]=Prokyon
+Terminal=false
+Type=Application
