diff -p -up xfce4-settings-4.5.92/xfce4-settings-helper/xfce4-settings-helper-autostart.desktop~ xfce4-settings-4.5.92/xfce4-settings-helper/xfce4-settings-helper-autostart.desktop
--- xfce4-settings-4.5.92/xfce4-settings-helper/xfce4-settings-helper-autostart.desktop~	2008-08-31 01:40:21.000000000 +0200
+++ xfce4-settings-4.5.92/xfce4-settings-helper/xfce4-settings-helper-autostart.desktop	2009-01-07 07:33:21.000000000 +0100
@@ -1,9 +1,8 @@
 [Desktop Entry]
-Encoding=UTF-8
 Version=1.0
 Name=Xfce Settings Helper
 Exec=xfce4-settings-helper
 Terminal=false
 Type=Application
 StartupNotify=false
-Terminal=false
+OnlyShowIn=XFCE;
