diff -Naur subtitleeditor-0.21.2_org/share/subtitleeditor.desktop subtitleeditor-0.21.2_new/share/subtitleeditor.desktop
--- share/subtitleeditor.desktop	2008-06-12 21:16:37.000000000 +0300
+++ share/subtitleeditor.desktop	2008-06-18 14:54:33.000000000 +0300
@@ -1,8 +1,11 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Subtitle Editor
+Name[tr]=Subtitle Editor
 GenericName=Subtitle Editor
+GenericName[tr]=Altyazı Düzenleyici
 Comment=A subtitle editor based to GStreamer and Gtk+
+Comment[tr]=Altyazı dosyalarını düzenlemeye yarayan GStreamer ve Gtk+ tabanlı uygulama
 Exec=subtitleeditor %F
 Terminal=false
 Type=Application
