--- bin/jokosher.desktop.orig	2009-06-16 13:54:24.000000000 +0300
+++ bin/jokosher.desktop	2009-06-16 13:56:52.000000000 +0300
@@ -1,12 +1,15 @@
 [Desktop Entry]
 Version=1.0
-Name=Jokosher Audio Editor
+Name=Jokosher
+Name[tr]=Jokosher
 GenericName=Jokosher audio editor
+GenericName[en_CA]=Jokosher audio editor
+GenericName[tr]=Jokosher ses düzenleyicisi
 Comment=Simply and easily create multi-track audio
+Comment[tr]=Basit ve kolayca ses izi oluştur
 Exec=jokosher
 Icon=jokosher
 Terminal=false
 Type=Application
 MimeType=application/x-jokosher;
-Categories=Application;AudioVideo;
-GenericName[en_CA]=Jokosher audio editor
+Categories=GTK;Application;AudioVideo;Recorder;
