diff -urNad firestarter-1.0.3~/src/menus.c firestarter-1.0.3/src/menus.c
--- firestarter-1.0.3~/src/menus.c	2007-08-24 16:19:10.000000000 +0100
+++ firestarter-1.0.3/src/menus.c	2007-08-24 16:21:06.000000000 +0100
@@ -548,7 +548,7 @@
 static void
 open_manual (void)
 {
-	open_browser ("http://www.fs-security.com/docs/");
+	open_browser ("http://www.fs-security.com/docs.php");
 }
 
 /* [ copy_selected ]
