--- ccsm.desktop.in.orig	2008-09-26 21:55:43.000000000 +0300
+++ ccsm.desktop.in	2008-09-26 21:56:12.000000000 +0300
@@ -2,9 +2,9 @@
 Encoding=UTF-8
 _Name=CompizConfig Settings Manager
 _Comment=Configure Compiz with CompizConfig
-Icon=ccsm
+Icon=compizconfig
 Exec=ccsm
 Terminal=false
 Type=Application
-Categories=Compiz;Settings;DesktopSettings;
-StartupNotify=true
+Categories=Application;Compiz;
+StartupNotify=false
