--- 3party/qca/qca-tls.cpp.orig	2006-10-26 23:26:49.000000000 +0300
+++ 3party/qca/qca-tls.cpp	2006-10-28 14:30:46.000000000 +0300
@@ -458,7 +458,7 @@
 				r = d2i_RSA_PUBKEY(NULL, (const unsigned char **)&p, len);
 #else
     #if OPENSSL_VERSION_NUMBER >  0x0090707fL
-				r = d2i_RSA_PUBKEY(NULL, (const unsigned char **)&p, len);
+				r = d2i_RSA_PUBKEY(NULL, (unsigned char **)&p, len);
     #else
 				r = d2i_RSA_PUBKEY(NULL, (unsigned char **)&p, len);
     #endif
@@ -811,7 +811,7 @@
 		X509 *t = d2i_X509(NULL, (const unsigned char **)&p, len);
 #else
     #if OPENSSL_VERSION_NUMBER >  0x0090707fL
-		X509 *t = d2i_X509(NULL, (const unsigned char **)&p, len);
+		X509 *t = d2i_X509(NULL, (unsigned char **)&p, len);
     #else
 		X509 *t = d2i_X509(NULL, (unsigned char **)&p, len);
     #endif
