diff -Naur kirocker-4.0.orig/src/kirocker-applet.desktop kirocker-4.0/src/kirocker-applet.desktop
--- kirocker-4.0.orig/src/kirocker-applet.desktop	2007-11-19 12:53:01.000000000 +0200
+++ kirocker-4.0/src/kirocker-applet.desktop	2007-12-09 02:43:03.000000000 +0200
@@ -4,10 +4,12 @@
 Comment=Display cover and information about the currently played music
 Comment[fr]=Affiche la pochette et les informations sur la musique en cours de lecture
 Comment[xx]=xxDisplay cover and information about the currently played musicxx
+Comment[tr]=Çalan parçanın kapak dosyasını ve mevcut bilgilerini gösterir
 
 Name=Kirocker Music Display
 Name[fr]=Afficheur de musique Kirocker
 Name[xx]=xxKirocker Music Displayxx
+Name[tr]=Kirocker Müzik Paneli
 
 Icon=kirocker
 
diff -Naur kirocker-4.0.orig/src/kirocker.desktop kirocker-4.0/src/kirocker.desktop
--- kirocker-4.0.orig/src/kirocker.desktop	2007-11-19 12:54:03.000000000 +0200
+++ kirocker-4.0/src/kirocker.desktop	2007-12-09 02:42:27.000000000 +0200
@@ -10,7 +10,9 @@
 Name=Kirocker Music Display
 Name[fr]=Afficheur de musique Kirocker
 Name[xx]=xxKirocker Music Displayxx
+Name[tr]=Kirocker Müzik Paneli
 
 Comment=Display cover and information about the currently played music
 Comment[fr]=Affiche la pochette et les informations sur la musique en cours de lecture
 Comment[xx]=xxDisplay cover and information about the currently played musicxx
+Comment[tr]=Çalan parçanın kapak dosyasını ve mevcut bilgilerini gösterir
