Index: SHERPA-1.0.10/PerturbativePhysics/Cluster_Partons_Base.H
===================================================================
--- SHERPA-1.0.10/PerturbativePhysics/Cluster_Partons_Base.H	(Revision 7396)
+++ SHERPA-1.0.10/PerturbativePhysics/Cluster_Partons_Base.H	(Arbeitskopie)
@@ -72,7 +72,7 @@
     void FixJetvetoPt2(double &);
     void JetvetoPt2(double &, double &,double &,double &);
     
-    virtual EXTRAXS::XS_Base * GetXS(EXTRAXS::XS_Group *,ATOOLS::Flavour *)=NULL;
+    virtual EXTRAXS::XS_Base * GetXS(EXTRAXS::XS_Group *,ATOOLS::Flavour *) = 0;
     virtual int                SetColours(EXTRAXS::XS_Base *,ATOOLS::Vec4D *,ATOOLS::Flavour *) = 0;
     virtual void               CalculateWeight(const double &meweight) = 0;
     void SetQMin(Combine_Table_Base *const ref=NULL);
