--- vmpk_tr2.ts	2009-09-26 22:15:36.000000000 +0300
+++ vmpk_tr.ts	2009-10-24 22:04:56.000000000 +0300
@@ -35,7 +35,11 @@
 p, li { white-space: pre-wrap; }
 &lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:&apos;Sans Serif&apos;; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
 &lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:9pt;&quot;&gt;&lt;span style=&quot; font-size:16pt; font-weight:600;&quot;&gt;Virtual MIDI Piano Keyboard&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
-        <translation type="unfinished"></translation>
+        <translation>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
+&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
+p, li { white-space: pre-wrap; }
+&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:&apos;Sans Serif&apos;; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
+&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:9pt;&quot;&gt;&lt;span style=&quot; font-size:16pt; font-weight:600;&quot;&gt;Sanal MIDI Piyano Klavyesi&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
     </message>
     <message>
         <location filename="../src/about.ui" line="139"/>
@@ -44,7 +48,11 @@
 p, li { white-space: pre-wrap; }
 &lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:&apos;Sans Serif&apos;; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
 &lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;a href=&quot;http://vmpk.sourceforge.net&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0057ae;&quot;&gt;http://vmpk.sourceforge.net&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
-        <translation type="unfinished"></translation>
+        <translation>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
+&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
+p, li { white-space: pre-wrap; }
+&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:&apos;Sans Serif&apos;; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
+&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;a href=&quot;http://vmpk.sourceforge.net&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0057ae;&quot;&gt;http://vmpk.sourceforge.net&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
     </message>
     <message utf8="true">
         <location filename="../src/about.ui" line="108"/>
@@ -65,103 +73,103 @@
     <message>
         <location filename="../src/extracontrols.cpp" line="66"/>
         <source>New Control</source>
-        <translation type="unfinished"></translation>
+        <translation>Yeni Denetleyici</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="14"/>
         <source>Extra Controls Editor</source>
-        <translation type="unfinished"></translation>
+        <translation>Ek Denetim Düzenleyici</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="42"/>
         <source>Label:</source>
-        <translation type="unfinished"></translation>
+        <translation>Etiket:</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="55"/>
         <source>MIDI Controller:</source>
-        <translation type="unfinished"></translation>
+        <translation>MIDI Denetleyici:</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="81"/>
         <source>Add</source>
-        <translation type="unfinished"></translation>
+        <translation>Ekle</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="95"/>
         <source>Remove</source>
-        <translation type="unfinished"></translation>
+        <translation>Kaldır</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="109"/>
         <source>Up</source>
-        <translation type="unfinished"></translation>
+        <translation>Yukarı</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="119"/>
         <source>Down</source>
-        <translation type="unfinished"></translation>
+        <translation>Aşağı</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="146"/>
         <source>Switch</source>
-        <translation type="unfinished"></translation>
+        <translation>Seç</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="151"/>
         <source>Knob</source>
-        <translation type="unfinished"></translation>
+        <translation>Top</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="156"/>
         <source>Spin box</source>
-        <translation type="unfinished"></translation>
+        <translation>Dönen kutucuk</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="161"/>
         <source>Slider</source>
-        <translation type="unfinished"></translation>
+        <translation>Kaydırıcı</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="185"/>
         <source>Default ON</source>
-        <translation type="unfinished"></translation>
+        <translation>Öntanımlı AÇIK</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="192"/>
         <source>value ON:</source>
-        <translation type="unfinished"></translation>
+        <translation>değer AÇIK:</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="209"/>
         <source>value OFF:</source>
-        <translation type="unfinished"></translation>
+        <translation>değer KAPALI:</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="230"/>
         <location filename="../src/extracontrols.ui" line="285"/>
         <location filename="../src/extracontrols.ui" line="357"/>
         <source>Min. value:</source>
-        <translation type="unfinished"></translation>
+        <translation>En küçük değer:</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="247"/>
         <location filename="../src/extracontrols.ui" line="302"/>
         <location filename="../src/extracontrols.ui" line="374"/>
         <source>Max. value:</source>
-        <translation type="unfinished"></translation>
+        <translation>En büyük değer:</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="264"/>
         <location filename="../src/extracontrols.ui" line="319"/>
         <location filename="../src/extracontrols.ui" line="391"/>
         <source>Default value:</source>
-        <translation type="unfinished"></translation>
+        <translation>Öntanımlı değer:</translation>
     </message>
     <message>
         <location filename="../src/extracontrols.ui" line="340"/>
         <source>Display size:</source>
-        <translation type="unfinished"></translation>
+        <translation>Görünüm boyutu:</translation>
     </message>
 </context>
 <context>
@@ -169,48 +177,48 @@
     <message>
         <location filename="../src/kmapdialog.cpp" line="31"/>
         <source>Open...</source>
-        <translation type="unfinished"></translation>
+        <translation>Aç...</translation>
     </message>
     <message>
         <location filename="../src/kmapdialog.cpp" line="32"/>
         <source>Save As...</source>
-        <translation type="unfinished"></translation>
+        <translation>Farklı Kaydet...</translation>
     </message>
     <message>
         <location filename="../src/kmapdialog.cpp" line="43"/>
         <source>Raw Key Map Editor</source>
-        <translation type="unfinished"></translation>
+        <translation>Ham Klavye Düzenleyici</translation>
     </message>
     <message>
         <location filename="../src/kmapdialog.cpp" line="43"/>
         <source>Key Map Editor</source>
-        <translation type="unfinished">Klayve Düzeni Düzenleyici</translation>
+        <translation>Klayve Düzeni Düzenleyici</translation>
     </message>
     <message>
         <location filename="../src/kmapdialog.cpp" line="44"/>
         <source>Key Code</source>
-        <translation type="unfinished"></translation>
+        <translation>Klavye Kodu</translation>
     </message>
     <message>
         <location filename="../src/kmapdialog.cpp" line="44"/>
         <source>Key</source>
-        <translation type="unfinished">Tuş</translation>
+        <translation>Kod</translation>
     </message>
     <message>
         <location filename="../src/kmapdialog.cpp" line="87"/>
         <source>Open keyboard map definition</source>
-        <translation type="unfinished">Klavye düzeni tanımlaması aç</translation>
+        <translation>Klavye düzeni tanımlaması aç</translation>
     </message>
     <message>
         <location filename="../src/kmapdialog.cpp" line="89"/>
         <location filename="../src/kmapdialog.cpp" line="102"/>
         <source>Keyboard map (*.xml)</source>
-        <translation type="unfinished">Klavye düzeni (*.xml)</translation>
+        <translation>Klavye düzeni (*.xml)</translation>
     </message>
     <message>
         <location filename="../src/kmapdialog.cpp" line="100"/>
         <source>Save keyboard map definition</source>
-        <translation type="unfinished">Klavye düzeni tanımlamasını kaydet</translation>
+        <translation>Klavye düzeni tanımlamasını kaydet</translation>
     </message>
 </context>
 <context>
@@ -868,7 +876,7 @@
     <message>
         <location filename="../src/kmapdialog.ui" line="33"/>
         <source>This box displays the name of the current mapping file</source>
-        <translation type="unfinished"></translation>
+        <translation>Bu kutu geçerli düzenin adını gösterir</translation>
     </message>
     <message>
         <location filename="../src/kmapdialog.ui" line="46"/>
@@ -931,7 +939,7 @@
     <message>
         <location filename="../src/midisetup.ui" line="26"/>
         <source>Check this box to enable MIDI input for the program. In Linux and Mac OSX the input port is always enabled and can&apos;t be un-ckecked</source>
-        <translation type="unfinished"></translation>
+        <translation>Uygulama için MIDI girişini etkinleştirmek için bu kutucuğu işaretleyin. Linux ve Mac OSX için giriş portu herzaman etkindir ve pasifleştirilemez</translation>
     </message>
     <message>
         <location filename="../src/midisetup.ui" line="39"/>
@@ -946,7 +954,7 @@
     <message>
         <location filename="../src/midisetup.ui" line="112"/>
         <source>Use this control to change the connection for the MIDI output port</source>
-        <translation type="unfinished"></translation>
+        <translation>MIDI çıktı portunun bağlantısını değiştirmek için bu denetimi kullanın</translation>
     </message>
 </context>
 <context>
@@ -955,93 +963,93 @@
         <location filename="../src/pianoscene.cpp" line="298"/>
         <location filename="../src/pianoscene.cpp" line="301"/>
         <source>C</source>
-        <translation type="unfinished"></translation>
+        <translation>C</translation>
     </message>
     <message utf8="true">
         <location filename="../src/pianoscene.cpp" line="298"/>
         <source>C♯</source>
-        <translation type="unfinished"></translation>
+        <translation>C♯</translation>
     </message>
     <message>
         <location filename="../src/pianoscene.cpp" line="298"/>
         <location filename="../src/pianoscene.cpp" line="301"/>
         <source>D</source>
-        <translation type="unfinished"></translation>
+        <translation>D</translation>
     </message>
     <message utf8="true">
         <location filename="../src/pianoscene.cpp" line="298"/>
         <source>D♯</source>
-        <translation type="unfinished"></translation>
+        <translation>D♯</translation>
     </message>
     <message>
         <location filename="../src/pianoscene.cpp" line="298"/>
         <location filename="../src/pianoscene.cpp" line="301"/>
         <source>E</source>
-        <translation type="unfinished"></translation>
+        <translation>E</translation>
     </message>
     <message>
         <location filename="../src/pianoscene.cpp" line="299"/>
         <location filename="../src/pianoscene.cpp" line="302"/>
         <source>F</source>
-        <translation type="unfinished"></translation>
+        <translation>F</translation>
     </message>
     <message utf8="true">
         <location filename="../src/pianoscene.cpp" line="299"/>
         <source>F♯</source>
-        <translation type="unfinished"></translation>
+        <translation>F♯</translation>
     </message>
     <message>
         <location filename="../src/pianoscene.cpp" line="299"/>
         <location filename="../src/pianoscene.cpp" line="302"/>
         <source>G</source>
-        <translation type="unfinished"></translation>
+        <translation>G</translation>
     </message>
     <message utf8="true">
         <location filename="../src/pianoscene.cpp" line="299"/>
         <source>G♯</source>
-        <translation type="unfinished"></translation>
+        <translation>G♯</translation>
     </message>
     <message>
         <location filename="../src/pianoscene.cpp" line="300"/>
         <location filename="../src/pianoscene.cpp" line="303"/>
         <source>A</source>
-        <translation type="unfinished"></translation>
+        <translation>A</translation>
     </message>
     <message utf8="true">
         <location filename="../src/pianoscene.cpp" line="300"/>
         <source>A♯</source>
-        <translation type="unfinished"></translation>
+        <translation>A♯</translation>
     </message>
     <message>
         <location filename="../src/pianoscene.cpp" line="300"/>
         <location filename="../src/pianoscene.cpp" line="303"/>
         <source>B</source>
-        <translation type="unfinished"></translation>
+        <translation>B</translation>
     </message>
     <message utf8="true">
         <location filename="../src/pianoscene.cpp" line="301"/>
         <source>D♭</source>
-        <translation type="unfinished"></translation>
+        <translation>D♭</translation>
     </message>
     <message utf8="true">
         <location filename="../src/pianoscene.cpp" line="301"/>
         <source>E♭</source>
-        <translation type="unfinished"></translation>
+        <translation>E♭</translation>
     </message>
     <message utf8="true">
         <location filename="../src/pianoscene.cpp" line="302"/>
         <source>G♭</source>
-        <translation type="unfinished"></translation>
+        <translation>G♭</translation>
     </message>
     <message utf8="true">
         <location filename="../src/pianoscene.cpp" line="302"/>
         <source>A♭</source>
-        <translation type="unfinished"></translation>
+        <translation>A♭</translation>
     </message>
     <message utf8="true">
         <location filename="../src/pianoscene.cpp" line="303"/>
         <source>B♭</source>
-        <translation type="unfinished"></translation>
+        <translation>B♭</translation>
     </message>
 </context>
 <context>
@@ -1060,13 +1068,13 @@
         <location filename="../src/preferences.cpp" line="168"/>
         <location filename="../src/preferences.cpp" line="179"/>
         <source>Open keyboard map definition</source>
-        <translation type="unfinished">Klavye düzeni tanımlaması aç</translation>
+        <translation>Klavye düzeni tanımlaması aç</translation>
     </message>
     <message>
         <location filename="../src/preferences.cpp" line="170"/>
         <location filename="../src/preferences.cpp" line="181"/>
         <source>Keyboard map (*.xml)</source>
-        <translation type="unfinished">Klavye düzeni (*.xml)</translation>
+        <translation>Klavye düzeni (*.xml)</translation>
     </message>
 </context>
 <context>
@@ -1149,12 +1157,12 @@
     <message>
         <location filename="../src/preferences.ui" line="116"/>
         <source>The instruments definition file currently loaded</source>
-        <translation type="unfinished"></translation>
+        <translation>Şimdi yüklü olan enstrüman tanımlama dosyaları</translation>
     </message>
     <message>
         <location filename="../src/preferences.ui" line="129"/>
         <source>Press this button to load an instruments definition file from disk.</source>
-        <translation type="unfinished"></translation>
+        <translation>Diskten bir enstrüman tanımlaması yüklemek için buna tıklayın.</translation>
     </message>
     <message>
         <location filename="../src/preferences.ui" line="164"/>
@@ -1164,18 +1172,18 @@
     <message>
         <location filename="../src/preferences.ui" line="171"/>
         <source>Keyboard Map</source>
-        <translation type="unfinished"></translation>
+        <translation>Klavye Düzeni</translation>
     </message>
     <message>
         <location filename="../src/preferences.ui" line="181"/>
         <location filename="../src/preferences.ui" line="198"/>
         <source>Load...</source>
-        <translation type="unfinished">Yükle...</translation>
+        <translation>Yükle...</translation>
     </message>
     <message>
         <location filename="../src/preferences.ui" line="188"/>
         <source>Raw Keyboard Map</source>
-        <translation type="unfinished"></translation>
+        <translation>Ham Klavye Düzeni</translation>
     </message>
     <message>
         <location filename="../src/preferences.ui" line="205"/>
@@ -1185,7 +1193,7 @@
     <message>
         <location filename="../src/preferences.ui" line="208"/>
         <source>Styled Widgets</source>
-        <translation type="unfinished"></translation>
+        <translation>Biçimlendirilmiş Parçacıklar</translation>
     </message>
     <message>
         <location filename="../src/preferences.ui" line="242"/>
@@ -1207,7 +1215,7 @@
     <message>
         <location filename="../src/preferences.ui" line="221"/>
         <source>Always On Top</source>
-        <translation type="unfinished"></translation>
+        <translation>Herzaman Üstte</translation>
     </message>
     <message>
         <location filename="../src/preferences.ui" line="228"/>
@@ -1221,12 +1229,12 @@
     <message>
         <location filename="../src/preferences.ui" line="235"/>
         <source>Grab Computer Keyboard</source>
-        <translation type="unfinished"></translation>
+        <translation>Bilgisayar Klavyesini Yakala</translation>
     </message>
     <message>
         <location filename="../src/preferences.ui" line="252"/>
         <source>Raw Computer Keyboard</source>
-        <translation type="unfinished"></translation>
+        <translation>Ham Bilgisayar Klavyesi</translation>
     </message>
     <message>
         <location filename="../src/preferences.ui" line="259"/>
@@ -1236,7 +1244,7 @@
     <message>
         <location filename="../src/preferences.ui" line="262"/>
         <source>Show Note Names</source>
-        <translation type="unfinished"></translation>
+        <translation>Nota İsimlerini Göster</translation>
     </message>
 </context>
 <context>
@@ -1262,48 +1270,48 @@
     <message>
         <location filename="../src/riffimportdlg.cpp" line="49"/>
         <source>Input SoundFont</source>
-        <translation type="unfinished"></translation>
+        <translation>Girdi Ses Yazı Tipi</translation>
     </message>
     <message>
         <location filename="../src/riffimportdlg.cpp" line="49"/>
         <source>SoundFonts (*.sf2 *.sbk *.dls)</source>
-        <translation type="unfinished"></translation>
+        <translation>Ses Yazı Tipleri (*.sf2 *.sbk *.dls)</translation>
     </message>
     <message>
         <location filename="../src/riffimportdlg.cpp" line="59"/>
         <source>Output</source>
-        <translation type="unfinished"></translation>
+        <translation>Çıkış</translation>
     </message>
     <message>
         <location filename="../src/riffimportdlg.ui" line="29"/>
         <source>Input File</source>
-        <translation type="unfinished"></translation>
+        <translation>Girdi Dosyası</translation>
     </message>
     <message>
         <location filename="../src/riffimportdlg.ui" line="67"/>
         <location filename="../src/riffimportdlg.ui" line="140"/>
         <source>...</source>
-        <translation type="unfinished"></translation>
+        <translation>...</translation>
     </message>
     <message>
         <location filename="../src/riffimportdlg.ui" line="74"/>
         <source>Name</source>
-        <translation type="unfinished"></translation>
+        <translation>İsim</translation>
     </message>
     <message>
         <location filename="../src/riffimportdlg.ui" line="88"/>
         <source>Version</source>
-        <translation type="unfinished"></translation>
+        <translation>Sürüm</translation>
     </message>
     <message>
         <location filename="../src/riffimportdlg.ui" line="98"/>
         <source>Copyright</source>
-        <translation type="unfinished"></translation>
+        <translation>Telif Hakları</translation>
     </message>
     <message>
         <location filename="../src/riffimportdlg.ui" line="108"/>
         <source>Output File</source>
-        <translation type="unfinished"></translation>
+        <translation>Çıktı Dosyası</translation>
     </message>
     <message>
         <location filename="../src/riffimportdlg.ui" line="118"/>
@@ -1318,7 +1326,7 @@
     <message>
         <location filename="../src/riffimportdlg.cpp" line="59"/>
         <source>Instrument definitions (*.ins)</source>
-        <translation type="unfinished">Enstrüman tanımlaması (*.ins)</translation>
+        <translation>Enstrüman tanımlamaları (*.ins)</translation>
     </message>
     <message>
         <location filename="../src/riffimportdlg.ui" line="14"/>
@@ -1357,12 +1365,12 @@
     <message>
         <location filename="../src/vpiano.cpp" line="199"/>
         <source>Channel:</source>
-        <translation type="unfinished"></translation>
+        <translation>Kanal:</translation>
     </message>
     <message>
         <location filename="../src/vpiano.cpp" line="207"/>
         <source>Base Octave:</source>
-        <translation type="unfinished"></translation>
+        <translation>Temel Oktav:</translation>
     </message>
     <message>
         <location filename="../src/vpiano.cpp" line="215"/>
@@ -1372,17 +1380,17 @@
     <message>
         <location filename="../src/vpiano.cpp" line="223"/>
         <source>Velocity:</source>
-        <translation type="unfinished"></translation>
+        <translation>Hız:</translation>
     </message>
     <message>
         <location filename="../src/vpiano.cpp" line="245"/>
         <source>Control:</source>
-        <translation type="unfinished"></translation>
+        <translation>Denetle:</translation>
     </message>
     <message>
         <location filename="../src/vpiano.cpp" line="251"/>
         <source>Value:</source>
-        <translation type="unfinished"></translation>
+        <translation>Değer:</translation>
     </message>
     <message>
         <location filename="../src/vpiano.cpp" line="267"/>
@@ -1392,12 +1400,12 @@
     <message>
         <location filename="../src/vpiano.cpp" line="281"/>
         <source>Bank:</source>
-        <translation type="unfinished"></translation>
+        <translation></translation>
     </message>
     <message>
         <location filename="../src/vpiano.cpp" line="287"/>
         <source>Program:</source>
-        <translation type="unfinished"></translation>
+        <translation>Program:</translation>
     </message>
     <message>
         <source> Channel: </source>
@@ -1492,7 +1500,7 @@
         <location filename="../src/vpiano.ui" line="179"/>
         <location filename="../src/vpiano.ui" line="346"/>
         <source>&amp;Extra Controls</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Ek Denetimler</translation>
     </message>
     <message>
         <location filename="../src/vpiano.ui" line="195"/>
@@ -1603,13 +1611,13 @@
         <location filename="../src/vpiano.ui" line="351"/>
         <location filename="../src/vpiano.ui" line="362"/>
         <source>Edit</source>
-        <translation type="unfinished"></translation>
+        <translation>Düzen</translation>
     </message>
     <message>
         <location filename="../src/vpiano.ui" line="354"/>
         <location filename="../src/vpiano.ui" line="379"/>
         <source>Open the Extra Controls editor</source>
-        <translation type="unfinished"></translation>
+        <translation>Ek Denetim Düzenleyiciyi Aç</translation>
     </message>
     <message>
         <location filename="../src/vpiano.ui" line="365"/>
@@ -1619,7 +1627,7 @@
     <message>
         <location filename="../src/vpiano.ui" line="376"/>
         <source>&amp;Extra Controllers</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;k Denetleyiciler</translation>
     </message>
     <message>
         <source>Load...</source>
@@ -1640,7 +1648,7 @@
     <message>
         <location filename="../src/vpiano.ui" line="314"/>
         <source>&amp;Keyboard Map</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Klavye Düzeni</translation>
     </message>
     <message>
         <source>Edit...</source>
@@ -1654,32 +1662,32 @@
     <message>
         <location filename="../src/vpiano.ui" line="322"/>
         <source>&amp;Contents</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;İçindekiler</translation>
     </message>
     <message>
         <location filename="../src/vpiano.ui" line="325"/>
         <source>Open the index of the help document</source>
-        <translation type="unfinished"></translation>
+        <translation>Yardım belgesinin indeksini aç</translation>
     </message>
     <message>
         <location filename="../src/vpiano.ui" line="330"/>
         <source>VMPK &amp;Web site</source>
-        <translation type="unfinished"></translation>
+        <translation>VMPK &amp;Web Sayfası</translation>
     </message>
     <message>
         <location filename="../src/vpiano.ui" line="333"/>
         <source>Open the VMPK web site address using a web browser</source>
-        <translation type="unfinished"></translation>
+        <translation>Bir web tarayıcı kullanarak VMPK web sitesini aç</translation>
     </message>
     <message>
         <location filename="../src/vpiano.ui" line="338"/>
         <source>&amp;Import SoundFont...</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;İçeriye Ses Yazı Tipi Aktar...</translation>
     </message>
     <message>
         <location filename="../src/vpiano.cpp" line="1228"/>
         <source>No help file found</source>
-        <translation type="unfinished"></translation>
+        <translation>Yardım dosyası bulunamadı</translation>
     </message>
 </context>
 </TS>
