--- deluge/data/share/applications/deluge.desktop.orig	2009-04-30 12:52:46.000000000 +0300
+++ deluge/data/share/applications/deluge.desktop	2009-04-30 12:53:44.000000000 +0300
@@ -1,8 +1,10 @@
 [Desktop Entry]
 Version=1.0
-Name=Deluge BitTorrent Client
+Name=Deluge
 GenericName=Bittorrent Client
+GenericName[tr]=Bittorrent İstemcisi
 Comment=Transfer files using the Bittorrent protocol
+Comment[tr]=Bittorrent protokolüyle dosyaları paylaş
 Exec=deluge
 Icon=deluge
 Terminal=false
