diff -Naur fluxbox-1.0.1_20080818.orig/data/init.in fluxbox-1.0.1_20080818/data/init.in
--- fluxbox-1.0.1_20080818.orig/data/init.in	2008-08-18 17:50:59.000000000 +0300
+++ fluxbox-1.0.1_20080818/data/init.in	2008-08-18 21:48:25.000000000 +0300
@@ -1,3 +1,4 @@
+session.screen0.rootCommand: wpsetters=feh fbsetbg -f "/usr/share/fluxbox/backgrounds/default.png"
 session.screen0.slit.placement:	RightBottom
 session.screen0.slit.onTop:	False
 session.screen0.slit.autoHide:	False
@@ -7,7 +8,7 @@
 session.screen0.toolbar.placement:	BottomCenter
 session.screen0.toolbar.widthPercent:	66
 session.screen0.workspaceNames:	one,two,three,four
-session.screen0.strftimeFormat:	%k:%M
+session.screen0.strftimeFormat: %T %d %b %a
 session.screen0.focusNewWindows:	True
 session.screen0.focusModel:	ClickToFocus
 session.screen0.fullMaximization:	False
@@ -22,6 +23,8 @@
 session.opaqueMove:	False
 session.autoRaiseDelay:	250
 session.menuFile:	~/.@pkgprefix@fluxbox@pkgsuffix@/menu
+session.styleOverlay:	~/.@pkgprefix@fluxbox@pkgsuffix@/overlay
+session.groupFile:	~/.@pkgprefix@fluxbox@pkgsuffix@/groups
 session.cacheLife:	5
 session.styleFile:	@default_style@
 session.keyFile: ~/.@pkgprefix@fluxbox@pkgsuffix@/keys
diff -Naur fluxbox-1.0.1_20080818.orig/data/overlay fluxbox-1.0.1_20080818/data/overlay
--- fluxbox-1.0.1_20080818.orig/data/overlay	2008-08-16 21:57:43.000000000 +0300
+++ fluxbox-1.0.1_20080818/data/overlay	2008-08-18 21:49:07.000000000 +0300
@@ -1,2 +1,20 @@
 ! The following line will prevent styles from setting the background.
 ! background: none
+
+menu.frame.font:                        Dejavu Sans-9:bold
+menu.title.font:                        Dejavu Sans-9:bold
+toolbar.clock.font:                     Dejavu Sans-9:bold
+toolbar.workspace.font:                 Dejavu Sans-12:bold
+toolbar.iconbar.focused.font:           Dejavu Sans-9:bold
+toolbar.iconbar.unfocused.font:         Dejavu Sans-9:bold
+window.font:                            Dejavu Sans-9:bold
+window.label.focused.font:              Dejavu Sans-10:bold
+window.label.unfocused.font:            Dejavu Sans-10:bold
+
+toolbar.height:                         20
+toolbar.borderWidth:                    0
+toolbar.workspace.textColor:            #ffffff
+
+background:                             flat
+background.color:                       #B2B2BF
+background.colorTo:                     #B2B2BF
