--- devede.desktop.orig	2009-01-25 02:26:40.000000000 +0200
+++ devede.desktop	2009-01-26 10:33:29.000000000 +0200
@@ -7,11 +7,13 @@
 GenericName[es]=Creador de CD/DVDs de vídeo
 GenericName[gl]=Creador de CD/DVDs de vídeo
 GenericName[pl]=Tworzenie wideo na CD/DVD
+GenericName[tr]=DVD/CD Video Oluşturucu
 Comment=Allows to create video CDs and DVDs, suitable to be reproduced in home DVD players
 Comment[ca]=Permet crear CDs i DVDs de vídeo, que poden reproduir-se en reproductors domèstics
 Comment[es]=Permite crear video CDs y DVDs, que pueden reproducirse en reproductores domésticos
 Comment[gl]=Permite crear video CDs e DVDs, que poden reproducirse en reproductores domésticos
 Comment[pl]=Pozwala stworzyć wideo na płytach CD lub DVD, do otwarzania w domowych odtwarzaczach DVD
+Comment[tr]=Videolarınızı DVD, SVCD yapmanıza yarar
 TryExec=devede
 Exec=devede
 Categories=AudioVideo;
