--- bin/jokosher.desktop.orig	2008-07-31 13:56:49.000000000 +0300
+++ bin/jokosher.desktop	2008-07-31 16:06:52.000000000 +0300
@@ -2,12 +2,15 @@
 Version=0.9
 Encoding=UTF-8
 Name=Jokosher Audio Editor
+Name[tr]=Jokosher Ses Düzenleyicisi
 GenericName=Jokosher audio editor
+GenericName[en_CA]=Jokosher audio editor
+GenericName[tr]=Jokosher ses düzenleyicisi
 Comment=Simply and easily create multi-track audio
+Comment[tr]Basit ve kolayca çoklu ses izi oluşturur
 Exec=jokosher
 Icon=/usr/share/pixmaps/jokosher-icon.png
 Terminal=false
 Type=Application
 MimeType=application/x-jokosher;
 Categories=AudioVideoEditing;GTK;Recorder
-GenericName[en_CA]=Jokosher audio editor
