diff -ur ifolder3-3.4.6167.1.orig/configure.in ifolder3-3.4.6167.1/configure.in
--- ifolder3-3.4.6167.1.orig/configure.in	2006-06-16 10:05:52.000000000 +0300
+++ ifolder3-3.4.6167.1/configure.in	2006-06-29 20:41:49.000000000 +0300
@@ -555,7 +555,6 @@
 package/linux/Makefile
 package/darwin/Makefile
 doc/Makefile
-help/Makefile
 help-linux/Makefile
 po/Makefile
 ])
diff -ur ifolder3-3.4.6167.1.orig/Makefile.am ifolder3-3.4.6167.1/Makefile.am
--- ifolder3-3.4.6167.1.orig/Makefile.am	2006-06-16 10:05:52.000000000 +0300
+++ ifolder3-3.4.6167.1/Makefile.am	2006-06-29 20:41:53.000000000 +0300
@@ -1,11 +1,5 @@
-if LINUX
 SUBDIRS = src help-linux po
 DIST_SUBDIRS = $(SUBDIRS) doc package test
-else
-SUBDIRS = src help po
-DIST_SUBDIRS = $(SUBDIRS) doc package test help-linux
-endif
-
 
 # remove all of the CVS directories
 dist-hook:
