diff -ru libgksu-2.0.5/libgksu/libgksu.c libgksu-2.0.5.new/libgksu/libgksu.c
--- libgksu-2.0.5/libgksu/libgksu.c	2007-06-10 20:28:43.000000000 +0200
+++ libgksu-2.0.5.new/libgksu/libgksu.c	2007-09-12 23:13:53.315398336 +0200
@@ -869,12 +869,12 @@
 				     "The application '%s' lets you "
 				     "modify essential parts of your "
 				     "system."),
-				   command);
+				   g_markup_escape_text(command, -1));
 	  else
 	    msg = g_strdup_printf (_("<b><big>Enter your password to run "
 				     "the application '%s' as user %s"
 				     "</big></b>"),
-				   command, context->user);
+				   g_markup_escape_text(command, -1), context->user);
 	}
       else
 	{
@@ -884,12 +884,12 @@
                                    "The application '%s' lets you "
                                    "modify essential parts of your "
                                    "system."),
-				   command);
+				   g_markup_escape_text(command, -1));
         else
           msg = g_strdup_printf (_("<b><big>Enter the password of %s to run "
                                    "the application '%s'"
                                    "</big></b>"),
-				   command, context->user);
+				   context->user, g_markup_escape_text(command, -1));
       }
 
       gksuui_dialog_set_message (GKSUUI_DIALOG(dialog), msg);
