diff -Nur lynx2-8-6-old/lynx.cfg lynx2-8-6/lynx.cfg
--- lynx2-8-6-old/lynx.cfg	2008-11-03 19:50:57.000000000 +0200
+++ lynx2-8-6/lynx.cfg	2008-11-03 19:50:59.000000000 +0200
@@ -127,7 +127,7 @@
 # pointers to lots of interesting places on the web.
 #
 #DEFAULT_INDEX_FILE:http://www.ncsa.uiuc.edu/SDG/Software/Mosaic/MetaIndex.html
-DEFAULT_INDEX_FILE:http://lynx.isc.org/
+DEFAULT_INDEX_FILE:http://www.google.com/
 
 
 .h1 Interaction
