diff -Nur lynx2-8-6-old/lynx.cfg lynx2-8-6/lynx.cfg
--- lynx2-8-6-old/lynx.cfg	2008-11-03 16:39:15.000000000 +0200
+++ lynx2-8-6/lynx.cfg	2008-11-03 16:39:17.000000000 +0200
@@ -406,7 +406,7 @@
 # standardized, the return values and their relationship to the locale value is
 # not.  GNU libiconv happens to give useful values, but other implementations
 # are not guaranteed to do this.
-#LOCALE_CHARSET:FALSE
+LOCALE_CHARSET:TRUE
 
 
 .h2 ASSUME_CHARSET
