--- fityk.desktop	2007-01-20 12:45:59.000000000 +0200
+++ fityk2.desktop	2007-07-22 13:51:54.000000000 +0300
@@ -2,11 +2,14 @@
 Type=Application
 Encoding=UTF-8
 Name=fityk
+Name[tr]=fityk
 Comment=peak fitting and data analysis
+Comment[tr]=Zirve provası ve veri analizi
 GenericName=Curve Fitter
+GenericName[tr]=Eğri Prova Uygulaması
 Icon=fityk
 Exec=fityk %F
 Terminal=False
 StartupNotify=true
 MimeType=application/x-fityk;
-Categories=Application;Science;GTK;
+Categories=GTK;KDE;Education;Math;
