diff -Nur gnome-cups-manager-0.31/libgnomecups/gnome-cups-ui-driver.c gnome-cups-manager-0.31.new/libgnomecups/gnome-cups-ui-driver.c
--- gnome-cups-manager-0.31/libgnomecups/gnome-cups-ui-driver.c	2005-02-03 21:10:04.000000000 +0100
+++ gnome-cups-manager-0.31.new/libgnomecups/gnome-cups-ui-driver.c	2006-04-07 17:15:15.000000000 +0200
@@ -770,7 +770,7 @@
 	g_signal_emit (G_OBJECT (ds), signals [CHANGED], 0);
 }
 
-#define CUPS_DIR  "/usr/share/cups/model"
+#define CUPS_DIR  "/usr/share/ppd/custom"
 
 static gboolean
 cb_find_ppd_by_filename (gpointer key, gpointer value, gpointer filename)
