--- Makefile.orig	2006-11-21 06:12:47.000000000 +0200
+++ Makefile	2008-01-13 00:40:55.000000000 +0200
@@ -59,10 +59,10 @@
 STRIP =
 else
 OPT = $(POPT)
-STRIP = -s
+STRIP =
 endif
 INCLUDE = -Iinclude
-CFLAGS = $(OPT) $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
+CFLAGS += $(OPT) $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
 
 LIB.SRCS = src/glew.c
 LIB.OBJS = $(LIB.SRCS:.c=.o)
@@ -76,16 +76,13 @@
 VISUALINFO.BIN = visualinfo$(BIN.SUFFIX)
 VISUALINFO.BIN.SRCS = src/visualinfo.c
 VISUALINFO.BIN.OBJS = $(VISUALINFO_BIN.SRCS:.c=.o)
-BIN.LIBS = -Llib $(LDFLAGS.STATIC) -l$(NAME) $(LDFLAGS.EXTRA) $(LDFLAGS.DYNAMIC) $(LDFLAGS.GL)
+BIN.LIBS = -Llib -l$(NAME) $(LDFLAGS.EXTRA) $(LDFLAGS.DYNAMIC) $(LDFLAGS.GL)
 
-all debug: lib/$(LIB.SHARED) lib/$(LIB.STATIC) bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN)
+all debug: lib/$(LIB.SHARED) bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN)
 
 lib:
 	mkdir lib
 
-lib/$(LIB.STATIC): $(LIB.OBJS)
-	$(AR) cr $@ $^
-
 lib/$(LIB.SHARED): $(LIB.SOBJS)
 	$(LD) $(LDFLAGS.SO) -o $@ $^ $(LIB.LDFLAGS) $(LIB.LIBS)
 ifeq ($(patsubst mingw%,mingw,$(SYSTEM)), mingw)
@@ -123,7 +120,7 @@
 	$(INSTALL) -m 0644 lib/$(LIB.SHARED) $(LIBDIR)/
 	$(LN) $(LIB.SHARED) $(LIBDIR)/$(LIB.SONAME)
   else
-	$(INSTALL) $(STRIP) -m 0644 lib/$(LIB.SHARED) $(LIBDIR)/
+	$(INSTALL) $(STRIP) -m 0755 lib/$(LIB.SHARED) $(LIBDIR)/
 	$(LN) $(LIB.SHARED) $(LIBDIR)/$(LIB.SONAME)
   endif
 endif
@@ -139,7 +136,6 @@
 	$(INSTALL) -m 0644 lib/$(LIB.STATIC) $(LIBDIR)/
 	$(LN) $(LIB.SHARED) $(LIBDIR)/$(LIB.DEVLNK)
   else
-	$(INSTALL) $(STRIP) -m 0644 lib/$(LIB.STATIC) $(LIBDIR)/
 	$(LN) $(LIB.SHARED) $(LIBDIR)/$(LIB.DEVLNK)
   endif
 endif
@@ -156,14 +152,13 @@
 else
 	$(RM) $(LIBDIR)/$(LIB.SONAME)
 	$(RM) $(LIBDIR)/$(LIB.SHARED)
-	$(RM) $(LIBDIR)/$(LIB.STATIC)
 endif
 	$(RM) $(GLEW_DEST)/bin/$(GLEWINFO.BIN) $(GLEW_DEST)/bin/$(VISUALINFO.BIN)
 
 clean:
 	$(RM) $(LIB.OBJS)
 	$(RM) $(LIB.SOBJS)
-	$(RM) lib/$(LIB.STATIC) lib/$(LIB.SHARED) lib/$(LIB.DEVLNK) lib/$(LIB.SONAME) $(LIB.STATIC)
+	$(RM) lib/$(LIB.SHARED) lib/$(LIB.DEVLNK) lib/$(LIB.SONAME)
 	$(RM) $(GLEWINFO.BIN.OBJS) bin/$(GLEWINFO.BIN) $(VISUALINFO.BIN.OBJS) bin/$(VISUALINFO.BIN)
 # Compiler droppings
 	$(RM) so_locations
