--- extern_libs/Makefile.in	2009-06-04 23:40:59.000000000 +0300
+++ extern_libs/Makefile.in	2009-06-05 23:09:59.000000000 +0300
@@ -458,9 +458,9 @@
 @BUILD_LIBMPEG3_LIB_TRUE@	$(EXGMAKE)
 
 install-data-hook:
-	rm -f $(GAPLIBDIR)/libavformat.a \
-	rm -f $(GAPLIBDIR)/libavcodec.a \
-	rm -f $(GAPLIBDIR)/libmpeg3.a
+	rm -f ${D}$(GAPLIBDIR)/libavformat.a \
+	rm -f ${D}$(GAPLIBDIR)/libavcodec.a \
+	rm -f ${D}$(GAPLIBDIR)/libmpeg3.a
 
 # eof (without this line Makefile generation produces syntax error)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
