--- launchy-2.1.2/platforms/unix/platform_unix.cpp.orig	2009-06-05 18:20:13.000000000 +0300
+++ launchy-2.1.2/platforms/unix/platform_unix.cpp	2009-06-05 18:21:11.000000000 +0300
@@ -52,8 +52,8 @@
     QList<Directory> list;
     const char *dirs[] = { "/usr/share/applications/",
 			   "/usr/local/share/applications/",
-			   "/usr/share/gdm/applications/",
-			   "/usr/share/applications/kde/",
+			   "/usr/kde/4/share/applications/kde4",
+			   "/usr/share/applications/",
 			   "~/.local/share/applications/"};
     QStringList l;
     l << "*.desktop";
