diff -Nur dia-0.96.1-old/configure.in dia-0.96.1/configure.in
--- dia-0.96.1-old/configure.in	2008-10-13 10:42:20.000000000 +0300
+++ dia-0.96.1/configure.in	2008-10-13 10:42:23.000000000 +0300
@@ -435,7 +435,6 @@
 
 if test "x$with_python" != xno; then
   AM_PATH_PYTHON(1.5.2)
-  AM_CHECK_PYMOD(gtk,,,[AC_MSG_ERROR(could not find pygtk module)])
   AM_CHECK_PYTHON_HEADERS
   AM_CHECK_PYTHON_LIB
 
