diff -Naur qcad-2.0.5.0-1-community.src/qcadactions/src/rs_actionzoompan.cpp qcad-2.0.5.0-1-community.src_patched/qcadactions/src/rs_actionzoompan.cpp
--- qcad-2.0.5.0-1-community.src/qcadactions/src/rs_actionzoompan.cpp	2005-11-22 13:51:46.000000000 +0200
+++ qcad-2.0.5.0-1-community.src_patched/qcadactions/src/rs_actionzoompan.cpp	2008-08-11 10:50:07.000000000 +0300
@@ -24,6 +24,8 @@
 **
 **********************************************************************/
 
+#include <cstdlib>
+
 #include "rs_actionzoompan.h"
 #include "rs_snapper.h"
 #include "rs_point.h"
diff -Naur qcad-2.0.5.0-1-community.src/qcadlib/src/information/rs_information.cpp qcad-2.0.5.0-1-community.src_patched/qcadlib/src/information/rs_information.cpp
--- qcad-2.0.5.0-1-community.src/qcadlib/src/information/rs_information.cpp	2005-11-22 13:52:39.000000000 +0200
+++ qcad-2.0.5.0-1-community.src_patched/qcadlib/src/information/rs_information.cpp	2008-08-11 10:35:38.000000000 +0300
@@ -24,6 +24,8 @@
 **
 **********************************************************************/
 
+#include <cstdlib>
+
 #include "rs_information.h"
 
 #include "rs_constructionline.h"
