diff -Naur klineakconfig-0.8-beta2-orig/klineakconfig/klineakconfig.desktop klineakconfig-0.8-beta2/klineakconfig/klineakconfig.desktop
--- klineakconfig-0.8-beta2-orig/klineakconfig/klineakconfig.desktop	2007-04-23 17:35:26.000000000 +0300
+++ klineakconfig-0.8-beta2/klineakconfig/klineakconfig.desktop	2007-04-23 18:00:25.000000000 +0300
@@ -1,12 +1,10 @@
-# KDE Config File  
 [Desktop Entry]
-Type=Application
-Exec=klineakconfig -caption "%c" %i %m  
-Icon=klineakconfig.png
-MiniIcon=klineakconfig.png
+Categories=System;HardwareSettings;KDE;
 DocPath=klineakconfig/index.html
-Comment=KDE Configuration program for LinEAK 0.2
-Comment[de]=
-Terminal=0
+Exec=klineakconfig -caption "%c" %i %m
+GenericName=KDE Configuration program for LinEAK
+GenericName[tr]=LinEAK için KDE arayüzü
+Icon=klineakconfig
 Name=KlineakConfig
-Name[de]=KlineakConfig
+Type=Application
+Terminal=False
