--- amarokFS.desktop.orig	2008-02-19 21:53:31.000000000 +0200
+++ amarokFS.desktop	2008-02-19 21:54:36.000000000 +0200
@@ -5,10 +5,12 @@
 Exec=amarokFS
 GenericName=Full screen mode for the Amarok player
 GenericName[en_US]=Full screen mode for the Amarok player
+GenericName[tr]=Amarok için tam ekran modu
 Icon=/usr/share/icons/amarokFS.png
 MimeType=
 Name=Amarok Full Screen
 Name[en_US]=Amarok Full Screen
+Name[tr]=Amarok Tam Ekran
 Path=
 StartupNotify=true
 Terminal=false
