--- libgeotiff-1.2.4/Doxyfile	2001-03-05 06:36:54.000000000 +0200
+++ libgeotiff.work/Doxyfile	2008-01-06 05:28:45.000000000 +0200
@@ -49,7 +49,7 @@
 # each generated HTML page. If it is left blank doxygen will generate a 
 # standard footer.
 
-HTML_FOOTER          =
+HTML_FOOTER          = no_date_footer.html
 
 # The QUIET tag can be used to turn on/off the messages that are generated
 # by doxygen. Possible values are YES and NO. If left blank NO is used.
--- libgeotiff-1.2.4/no_date_footer.html	1970-01-01 02:00:00.000000000 +0200
+++ libgeotiff.work/no_date_footer.html	2008-01-06 05:33:09.000000000 +0200
@@ -0,0 +1,6 @@
+<hr size="1"><address style="text-align: right;"><small>
+Generated for $projectname by&nbsp;<a href="http://www.doxygen.org/
+index.html"><img src="doxygen.png" alt="doxygen" align="middle" border="0"></a> 
+$doxygenversion</small></address>
+</body>
+</html>
