--- py-compile.orig	2008-10-12 12:21:13.000000000 +0300
+++ py-compile	2008-10-12 12:21:17.000000000 +0300
@@ -1,7 +1,7 @@
 #!/bin/sh
 # py-compile - Compile a Python program
 
-scriptversion=2005-05-14.22
+scriptversion=2008-10-12.12
 
 # Copyright (C) 2000, 2001, 2003, 2004, 2005  Free Software Foundation, Inc.
 
@@ -102,40 +102,7 @@
     filetrans="filepath = os.path.normpath('$destdir' + os.sep + path)"
 fi
 
-$PYTHON -c "
-import sys, os, string, py_compile
-
-files = '''$files'''
-
-print 'Byte-compiling python modules...'
-for file in string.split(files):
-    $pathtrans
-    $filetrans
-    if not os.path.exists(filepath) or not (len(filepath) >= 3
-                                            and filepath[-3:] == '.py'):
-	continue
-    print file,
-    sys.stdout.flush()
-    py_compile.compile(filepath, filepath + 'c', path)
-print" || exit $?
-
 # this will fail for python < 1.5, but that doesn't matter ...
-$PYTHON -O -c "
-import sys, os, string, py_compile
-
-files = '''$files'''
-print 'Byte-compiling python modules (optimized versions) ...'
-for file in string.split(files):
-    $pathtrans
-    $filetrans
-    if not os.path.exists(filepath) or not (len(filepath) >= 3
-                                            and filepath[-3:] == '.py'):
-	continue
-    print file,
-    sys.stdout.flush()
-    py_compile.compile(filepath, filepath + 'o', path)
-print" 2>/dev/null || :
-
 # Local Variables:
 # mode: shell-script
 # sh-indentation: 2
