--- gajim-0.11.1.orig/src/common/passwords.py
+++ gajim-0.11.1/src/common/passwords.py
@@ -81,9 +81,14 @@
 	def save_password(self, account_name, password, update=True):
 		display_name = _('Gajim account %s') % account_name
 		attributes = dict(account_name=str(account_name), gajim=1)
-		auth_token = gnomekeyring.item_create_sync(
-			self.keyring, gnomekeyring.ITEM_GENERIC_SECRET,
-			display_name, attributes, password, update)
+		try:
+			auth_token = gnomekeyring.item_create_sync(
+				self.keyring, gnomekeyring.ITEM_GENERIC_SECRET,
+				display_name, attributes, password, update)
+		except gnomekeyring.DeniedError:
+			set_storage(SimplePasswordStorage()) 
+			storage.save_password(account_name, password) 
+			return 
 		token = 'gnomekeyring:%i' % auth_token
 		gajim.config.set_per('accounts', account_name, 'password', token)
 
@@ -111,6 +116,8 @@
 				storage = GnomePasswordStorage()
 			except gnomekeyring.NoKeyringDaemonError:
 				storage = SimplePasswordStorage()
+			except gnomekeyring.DeniedError: 
+				storage = SimplePasswordStorage()
 		else:
 			storage = SimplePasswordStorage()
 	return storage
