diff -Nuar gajim-0.11.orig/scripts/gajim.in gajim-0.11/scripts/gajim.in
--- gajim-0.11.orig/scripts/gajim.in	2007-03-04 20:43:48.000000000 +0200
+++ gajim-0.11/scripts/gajim.in	2007-03-04 20:44:54.000000000 +0200
@@ -32,4 +32,4 @@
 
 cd ${datadir}/gajim/src
 export PYTHONPATH="$PYTHONPATH:@LIBDIR@/gajim"
-exec -a gajim ${PYTHON_EXEC} -OO gajim.py $@
+LC_ALL=en_US.UTF8 exec -a gajim ${PYTHON_EXEC} -OO gajim.py $@
