--- rawstudio.desktop.orig	2008-09-02 01:24:42.000000000 +0300
+++ rawstudio.desktop	2008-09-02 01:57:52.000000000 +0300
@@ -6,16 +6,19 @@
 Name[it]=Rawstudio
 Name[pl]=Rawstudio
 Name[ru]=Rawstudio
+Name[tr]=Rawstudio
 GenericName=Raw image converter
 GenericName[it]=Convertitore Raw
 GenericName[pl]=Konwerter plików RAW
 GenericName[ru]=Проявка цифровых негативов
+GenericName[tr]=Raw Resim Dönüştürücü
 Comment=Read, manipulate and convert digital camera raw-images
 Comment[it]=Converte e manipola immagini raw
 Comment[pl]=Program Open Source do konwersji plików RAW i manipulacji na nich
 Comment[ru]=Свободная программа для проявки цифровых негативов
+Comment[tr]=Raw dosyalarını okur, işler ve dönüştürür
+Encoding=UTF-8
 Exec=rawstudio %F
 StartupNotify=true
-Icon=rawstudio.png
+Icon=rawstudio
 MimeType=image/x-dcraw;
-
