diff -Naur Miro-1.2.6.orig/portable/setup_portable.py Miro-1.2.6/portable/setup_portable.py
--- Miro-1.2.6.orig/portable/setup_portable.py	2008-08-15 07:50:42.000000000 +0300
+++ Miro-1.2.6/portable/setup_portable.py	2008-08-24 03:06:45.000000000 +0300
@@ -106,10 +106,10 @@
             EXTRA_COMPILE_ARGS.append("-DAMD64")
     
         if OS == "linux":
-            if os.WEXITSTATUS(os.system('grep -iq "Debian GNU/Linux 4.0\|Ubuntu 7.04\|Ubuntu 6.06\|Ubuntu 6.10\|Fedora Core release 6\|openSUSE 10.2\|openSUSE 10.3\|Mandriva Linux release 2007.1\|Fedora release 7\|BLAG release 60001\|Yellow Dog Linux release 5.0 (Phoenix)\|CentOS release 5 (Final)" /etc/issue')) == 0:
+            #if os.WEXITSTATUS(os.system!('grep -iq "Debian GNU/Linux 4.0\|Ubuntu 7.04\|Ubuntu 6.06\|Ubuntu 6.10\|Fedora Core release 6\|openSUSE 10.2\|openSUSE 10.3\|Mandriva Linux release 2007.1\|Fedora release 7\|BLAG release 60001\|Yellow Dog Linux release 5.0 (Phoenix)\|CentOS release 5 (Final)" /etc/issue')) == 0:
                 boosttype = 'nomt'
-            else:
-                boosttype = 'mt'
+            #else:
+                #boosttype = 'mt'
         elif OS == "freebsd":
             boosttype = 'nomt'
         else:
