Index: ecore/src/lib/ecore_data/Makefile.am
===================================================================
--- ecore/src/lib/ecore_data/Makefile.am	(revision 46192)
+++ ecore/src/lib/ecore_data/Makefile.am	(revision 46208)
@@ -1,8 +1,14 @@
 MAINTAINERCLEANFILES = Makefile.in
 
-AM_CPPFLAGS = -I$(top_srcdir)/src/lib/ecore \
-@EVIL_CFLAGS@ @EINA_CFLAGS@  @WIN32_CPPFLAGS@ @EFL_ECORE_BUILD@
+AM_CPPFLAGS = \
+-I$(top_srcdir)/src/lib/ecore \
+@EVIL_CFLAGS@ \
+@EINA_CFLAGS@ \
+@WIN32_CPPFLAGS@ \
+@EFL_ECORE_BUILD@
 
+if BUILD_ECORE_DATA
+
 lib_LTLIBRARIES = libecore_data.la
 include_HEADERS = \
 Ecore_Data.h \
@@ -22,3 +28,4 @@
 libecore_data_la_LIBADD = @dlopen_libs@ @EINA_LIBS@ @EVIL_LIBS@ @WIN32_LIBS@ @LTLIBINTL@ @EFL_PTHREAD_LIBS@ -lm
 libecore_data_la_LDFLAGS = -no-undefined @lt_enable_auto_import@ -version-info @version_info@ @ecore_release_info@
 
+endif
Index: ecore/src/lib/ecore_txt/Makefile.am
===================================================================
--- ecore/src/lib/ecore_txt/Makefile.am	(revision 46192)
+++ ecore/src/lib/ecore_txt/Makefile.am	(revision 46208)
@@ -16,8 +16,7 @@
 libecore_txt_la_SOURCES = \
 ecore_txt.c
 
-libecore_txt_la_LIBADD = @iconv_libs@ \
-@EINA_LIBS@
+libecore_txt_la_LIBADD = @iconv_libs@
 
 libecore_txt_la_LDFLAGS = -no-undefined @lt_enable_auto_import@ -version-info @version_info@ @ecore_txt_release_info@
 
Index: ecore/src/lib/ecore_evas/ecore_evas_x.c
===================================================================
--- ecore/src/lib/ecore_evas/ecore_evas_x.c	(revision 46192)
+++ ecore/src/lib/ecore_evas/ecore_evas_x.c	(revision 46208)
@@ -1811,6 +1811,7 @@
 
    if (!strcmp(ee->driver, "software_x11"))
      {
+#ifdef BUILD_ECORE_EVAS_SOFTWARE_X11
 	Evas_Engine_Info_Software_X11 *einfo;
 
 	einfo = (Evas_Engine_Info_Software_X11 *)evas_engine_info_get(ee->evas);
@@ -1820,6 +1821,7 @@
 	einfo->info.destination_alpha = transparent;
 	evas_engine_info_set(ee->evas, (Evas_Engine_Info *)einfo);
 	evas_damage_rectangle_add(ee->evas, 0, 0, ee->w, ee->h);
+#endif        
      }
 }
 #endif /* BUILD_ECORE_EVAS_X11 */
@@ -2372,7 +2374,7 @@
 
 #if defined (BUILD_ECORE_EVAS_SOFTWARE_X11) || defined (BUILD_ECORE_EVAS_OPENGL_X11) || defined (BUILD_ECORE_EVAS_XRENDER_X11) || defined (BUILD_ECORE_EVAS_XRENDER_XCB) || defined (BUILD_ECORE_EVAS_SOFTWARE_16_X11)
 static void
-_ecore_evas_x_flush_pre(void *data, Evas *e, Evas_Object *obj, void *event_info)
+_ecore_evas_x_flush_pre(void *data, Evas *e, void *event_info)
 {
    Ecore_Evas *ee = data;
 
@@ -2391,7 +2393,7 @@
 }
 
 static void
-_ecore_evas_x_flush_post(void *data, Evas *e, Evas_Object *obj, void *event_info)
+_ecore_evas_x_flush_post(void *data, Evas *e, void *event_info)
 {
    Ecore_Evas *ee = data;
    
Index: ecore/src/lib/ecore_evas/ecore_evas_private.h
===================================================================
--- ecore/src/lib/ecore_evas/ecore_evas_private.h	(revision 46192)
+++ ecore/src/lib/ecore_evas/ecore_evas_private.h	(revision 46208)
@@ -192,7 +192,7 @@
 {
    Ecore_Evas_Engine_Func *func;
 
-#ifdef BUILD_ECORE_EVAS_SOFTWARE_X11
+#ifdef BUILD_ECORE_EVAS_X11
    struct {
       Ecore_X_Window win_root;
       Eina_List     *win_extra;
Index: ecore/configure.ac
===================================================================
--- ecore/configure.ac	(revision 46192)
+++ ecore/configure.ac	(revision 46208)
@@ -96,6 +96,7 @@
 want_glib="no"
 
 # core modules
+want_ecore_data="yes"
 want_ecore_txt="yes"
 want_ecore_con="no"
 want_ecore_ipc="no"
@@ -321,7 +322,6 @@
       requirements_ecore_file="evil ${requirements_ecore_file}"
       requirements_ecore_imf="evil ${requirements_ecore_imf}"
       requirements_ecore_imf_evas="evil ${requirements_ecore_imf_evas}"
-      requirements_ecore_txt="evil ${requirements_ecore_txt}"
       EFL_ECORE_BUILD="-DEFL_ECORE_BUILD"
       EFL_ECORE_TXT_BUILD="-DEFL_ECORE_TXT_BUILD"
       EFL_ECORE_FILE_BUILD="-DEFL_ECORE_FILE_BUILD"
@@ -968,6 +968,9 @@
 
 ## Core modules
 
+# ecore_data
+ECORE_CHECK_MODULE([data], [${want_ecore_data}], [Data])
+
 # ecore_txt
 ECORE_CHECK_MODULE([txt], [${want_ecore_txt}], [Txt], [$have_iconv])
 
@@ -1383,6 +1386,7 @@
 echo "  Ecore........................: always"
 echo "    Thread support.............: $have_pthread"
 echo "    GLib support...............: $have_glib"
+echo "  Ecore_Data...................: $have_ecore_data"
 echo "  Ecore_Txt....................: $have_ecore_txt"
 echo "  Ecore_Con....................: $have_ecore_con"
 if test "x$have_ecore_con" = "xyes" ; then
Index: ecore/Makefile.am
===================================================================
--- ecore/Makefile.am	(revision 46192)
+++ ecore/Makefile.am	(revision 46208)
@@ -83,8 +83,12 @@
 	     ecore.spec.in ecore.spec
 
 pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = ecore.pc ecore-data.pc
+pkgconfig_DATA = ecore.pc
 
+if BUILD_ECORE_DATA
+pkgconfig_DATA += ecore-data.pc
+endif
+
 if BUILD_ECORE_CON
 pkgconfig_DATA += ecore-con.pc
 endif
