--- src/kleansweep.desktop	2006-10-29 22:05:18.000000000 +0200
+++ src/kleansweep.desktop	2007-12-18 16:46:07.000000000 +0200
@@ -1,9 +1,12 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=KleanSweep
+GenericName=Disc Cleanup Tool
+GenericName[tr]=Disk Temizleme Aracı
 Exec=kleansweep
-Icon=kleansweep.png
+Icon=kleansweep
 Categories=Qt;KDE;Utility;System;
 Type=Application
 Comment=Utility which finds unneeded files
 Comment[pl]=Narzędzie odnajdujące niepotrzebne pliki
+Comment[tr]=Diskinizdeki gereksiz dosyaları bulur
