--- wammu.desktop.in~	2007-12-19 14:16:01.000000000 +0200
+++ wammu.desktop.in	2008-03-04 20:57:03.000000000 +0200
@@ -1,9 +1,12 @@
 [Desktop Entry]
 _Name=Wammu ; EDIT IN wammu_setup/msgfmt.py
+Name[tr]=Wammu
 _GenericName=Mobile Phone Manager ; EDIT IN wammu_setup/msgfmt.py
+GenericName[tr]=Mobil Telefon Yöneticisi
 _Comment=Application for mobile phones - frontend for Gammu ; EDIT IN wammu_setup/msgfmt.py
+Comment[tr]=Gammu tabanlı mobil telefon yöneticisi
 Exec=wammu
 Icon=wammu
 Terminal=false
 Type=Application
-Categories=Utility;ContactManagement;Office;GTK;
+Categories=TelephonyTools;Calendar;ContactManagement;Office;GTK;
