diff -Naur xfce4-mixer-4.6.1/panel-plugin/xfce-volume-button.c xfce4-mixer-4.6.1.tpg/panel-plugin/xfce-volume-button.c
--- xfce4-mixer-4.6.1/panel-plugin/xfce-volume-button.c	2009-01-12 21:42:56.000000000 +0000
+++ xfce4-mixer-4.6.1.tpg/panel-plugin/xfce-volume-button.c	2009-05-30 10:43:26.000000000 +0000
@@ -202,7 +202,7 @@
   button->pixbufs = g_new0 (GdkPixbuf*, G_N_ELEMENTS (icons)-1);
 
   /* Create adjustment for the button (from 0.0 to 1.0 in 5% steps) */
-  button->adjustment = gtk_adjustment_new (0.0, 0.0, 1.0, 0.05, 0.05, 0.1);
+  button->adjustment = gtk_adjustment_new (0.0, 0.0, 1.0, 0.05, 0.05, 0.0);
 
   /* Create a new scaled image for the button icon */
   button->image = xfce_scaled_image_new ();
