diff -Naur flickrapi-1.2/_setup.py flickrapi-1.2_patched/_setup.py
--- flickrapi-1.2/_setup.py	2008-04-02 20:59:48.000000000 +0300
+++ flickrapi-1.2_patched/_setup.py	2009-02-03 02:17:46.000000000 +0200
@@ -16,7 +16,8 @@
 except ImportError:
     docutils = None
 
-from flickrapi import __version__
+#from flickrapi import __version__
+__version__ = "1.2"
 
 # This will be set to True when either the documentation is already
 # there, or if we can build it.
