--- Makefile	2006-04-02 14:40:53.000000000 +0300
+++ Makefile	2007-01-11 20:19:19.568908056 +0200
@@ -20,7 +20,7 @@
 # This Makefile assumes that you installed Trolltechs QT in /usr/local/qt
 #   visit: http://www.trolltech.com for download and licence information
 #
-QTDIR = /usr/lib/qt3
+QTDIR = /usr/qt/3
 CC = g++
 #CFLAGS = -I$(QTDIR)/include -pipe -O2 -g
 CFLAGS = -I$(QTDIR)/include -pipe -O2
