--- desktop/graveman.desktop.in	2005-05-24 00:07:08.000000000 +0300
+++ graveman.desktop-yeni.in	2006-06-19 11:16:44.000000000 +0300
@@ -15,6 +15,7 @@
 Name[no]=Graveman
 Name[hu]=Graveman
 Name[ja]=Graveman
+Name[tr]=Graveman
 GenericName=CD Creator
 GenericName[cs]=CD/DVD vypalovací program
 GenericName[de]=CD ersteller
@@ -30,6 +31,7 @@
 GenericName[no]=CD Brenner
 GenericName[hu]=CD/DVD íróprogram
 GenericName[ja]=CD/DVD作成ツール
+GenericName[tr]=CD/DVD yazıcısı
 Comment=Create and copy data and audio CD
 Comment[cs]=Vytváří a kopíruje datová a hudební CD/DVD
 Comment[de]=erstellt und kopiert Daten und Audio-CDs
@@ -45,8 +47,9 @@
 Comment[no]=Lage og kopiere data og lyd CD/DVD
 Comment[hu]=Adat és Audio CD/DVD létrehozása és másolása egyszerűen
 Comment[ja]=データ/オーディォCDの作成やコピー
+Comment[tr]=Veri ve müzik CD'si oluştur yada kopyala
 Exec=graveman
 Terminal=false
 Type=Application
 Icon=graveman48.png
-Categories=Application;Utility;
+Categories=AudioVideo;Player;GTK;
