--- data/amyedit.desktop.in.orig	2006-01-24 14:14:30.000000000 +0200
+++ data/amyedit.desktop.in	2007-08-16 04:19:57.000000000 +0300
@@ -1,9 +1,12 @@
 [Desktop Entry]
 Encoding=UTF-8
-_Name=AmyEdit LaTex Editor
-_Comment=A LaTeX Editor
+Name=AmyEdit LaTex Editor
+Name[tr]=AmyEdit LaTex Düzenleyicisi
+Comment=A LaTex Editor
+Comment[tr]=LaTex Düzenleyicisi
 Exec=amyedit
+Icon=package_editors
 Terminal=false
 Type=Application
+StartupNotify=false
-Categories=Application;Utility;TextEditor;
+Categories=KDE;Applications;Utility;TextEditor;
-
