diff -urNad firestarter-1.0.0/src/scriptwriter.c /tmp/dpep.d731CV/firestarter-1.0.0/src/scriptwriter.c
--- firestarter-1.0.0/src/scriptwriter.c	2004-11-25 07:53:37.000000000 +0100
+++ /tmp/dpep.d731CV/firestarter-1.0.0/src/scriptwriter.c	2004-11-25 08:01:07.000000000 +0100
@@ -425,7 +425,7 @@
 	
 	return TRUE;
 	
-	gchar *path = g_strconcat ("/var/run/dhclient-",
+	gchar *path = g_strconcat ("/var/run/dhclient.",
 				preferences_get_string (PREFS_FW_EXT_IF),
 				".pid", NULL);
 
