diff -Naur texmaker-1.7.1.orig/utilities/texmaker.desktop texmaker-1.7.1/utilities/texmaker.desktop
--- texmaker-1.7.1.orig/utilities/texmaker.desktop	2006-12-28 18:50:30.000000000 +0200
+++ texmaker-1.7.1/utilities/texmaker.desktop	2008-09-29 02:57:49.000000000 +0300
@@ -1,11 +1,13 @@
 [Desktop Entry]
-Categories=Application;Office;Publishing;X-SuSE-Core-Office;X-Mandriva-Office-Publishing;Qt;X-Misc;
+Categories=Application;Office;Publishing;Qt;
 Comment=
+Comment[tr]=LaTeX Editörü
 Comment[fr]=
 DocPath=
 Encoding=UTF-8
 Exec=texmaker
 GenericName=LaTeX Editor
+GenericName[tr]=LaTeX Editörü
 GenericName[fr]=Editeur LaTeX
 Icon=texmaker
 MimeType=text/x-tex
