--- Amaya/config/unix-thot.rc.orig	2005-10-05 14:23:22.000000000 +0300
+++ Amaya/config/unix-thot.rc	2006-09-23 20:16:34.000000000 +0300
@@ -14,7 +14,7 @@
 ENABLE_CACHE=Yes
 THOTSYSSCH=$(THOTDIR)/amaya:$(THOTDIR)/annotlib
 #TEMPLATE_URL=$(THOTDIR)/templates
-LANG=en
+LANG=tr
 # AMAYA_PLUGIN_DIR=$(THOTDIR)/plugintest/$(MACHINE)
 # URL of your HTTP proxy server  (server.address:port :port is optional)
 # ex: HTTP_PROXY=http://www.imag.fr:3128
@@ -34,7 +34,7 @@
 AUTO_SAVE=50
 GTM_TIME=No
 ISO_DATE=Yes
-DOCUMENT_CHARSET=iso-8859-1
+DOCUMENT_CHARSET=utf-8
 SHOW_SPECIAL_CHARS=yes
 FONT_ALIASING=yes
 
