diff -Nur aqsis-1_1_0.orig/platform/default/SConscript aqsis-1_1_0/platform/default/SConscript
--- aqsis-1_1_0.orig/platform/default/SConscript	2006-09-02 06:00:05.000000000 +0300
+++ aqsis-1_1_0/platform/default/SConscript	2006-10-21 13:29:42.322940241 +0300
@@ -79,8 +79,8 @@
 display_libs = ['tiff', 'z']
 
 # These are place holders at the moment, we need to work out the proper defines?
-env.AppendUnique(CPPDEFINES=[('DEFAULT_RC_PATH', '\\"' + env.Dir('$install_prefix').abspath + os.sep + 'etc' + '\\"')])
-env.AppendUnique(CPPDEFINES=[('DEFAULT_SHADER_PATH', '\\"' + env.Dir('$install_prefix').abspath + os.sep + 'shaders' + '\\"')])
+env.AppendUnique(CPPDEFINES=[('DEFAULT_RC_PATH', '\\"/etc\\"')])
+env.AppendUnique(CPPDEFINES=[('DEFAULT_SHADER_PATH', '\\"/usr/share/aqsis/shaders\\"')])
 
 # System configuration location
 env.Replace(SYSCONFDIR = env.Dir('$install_prefix').abspath + os.sep + 'etc')
diff -Nur aqsis-1_1_0.orig/SConstruct aqsis-1_1_0/SConstruct
--- aqsis-1_1_0.orig/SConstruct	2006-08-22 12:39:50.000000000 +0300
+++ aqsis-1_1_0/SConstruct	2006-10-21 13:33:39.646861521 +0300
@@ -143,7 +143,7 @@
 
 # Setup common environment settings to allow includes from the various local folders
 env.AppendUnique(CPPPATH = [target_dir.abspath, target_dir.abspath + '/aqsistypes', target_dir.abspath + '/renderer/render', target_dir.abspath + '/shadercompiler/shaderexecenv', target_dir.abspath + '/rib/rib2', target_dir.abspath + '/shadercompiler/shadervm', target_dir.abspath + '/rib/rib2ri', target_dir.abspath + '/argparse', target_dir.abspath + '/shadercompiler/slparse', target_dir.abspath + '/shadercompiler/codegenvm', target_dir.abspath + '/rib/api', '$zlib_include_path', '$tiff_include_path', '$jpeg_include_path', '$boost_include_path', '$fltk_include_path', '$exr_include_path'])
-env.AppendUnique(CPPDEFINES=[('DEFAULT_PLUGIN_PATH', '\\"' + env.Dir('${BINDIR}').abspath + '\\"')])
+env.AppendUnique(CPPDEFINES=[('DEFAULT_PLUGIN_PATH', '\\"/usr/lib/aqsis\\"')])
 env.AppendUnique(CPPDEFINES=['SCONS_BUILD'])
 
 # Setup the include path to the tiff headers (should have been determined in the system specific sections above).
@@ -221,8 +221,8 @@
 		"xpmlib": xpmlib,
 		"bmplib": bmplib,
 		"win32lib": win32lib,
-		"shaderpath": env.Dir('$SHADERDIR').abspath,
-		"displaypath": env.Dir('$BINDIR').abspath,
+		"shaderpath": '/usr/share/aqsis/shaders',
+		"displaypath": '/usr/lib/aqsis',
 		"exrlib": ""
 	    }
 	if not env['no_exr']:
