--- compiz-0.3.6/kde/window-decorator/options.cpp--	2007-03-16 11:44:44.000000000 +0100
+++ compiz-0.3.6/kde/window-decorator/options.cpp	2007-03-16 11:45:08.000000000 +0100
@@ -32,6 +32,11 @@
     updateSettings ();
 }
 
+KWD::Options::~Options()
+{
+    delete d;
+}
+
 unsigned long
 KWD::Options::updateSettings (void)
 {
--- compiz-0.3.6/kde/window-decorator/options.h--	2007-03-16 11:45:13.000000000 +0100
+++ compiz-0.3.6/kde/window-decorator/options.h	2007-03-16 11:45:25.000000000 +0100
@@ -77,7 +77,7 @@
 	};
 
 	Options (KConfig *config);
-
+	~Options();
 	virtual unsigned long updateSettings (void);
 
 	WindowOperation operationTitlebarDblClick (void)
