Index: xfce4-session-4.5.99.1/xfce4-tips/xfce4-tips-autostart.desktop.in
===================================================================
--- xfce4-session-4.5.99.1.orig/xfce4-tips/xfce4-tips-autostart.desktop.in	2009-02-15 17:06:23.000000000 -0600
+++ xfce4-session-4.5.99.1/xfce4-tips/xfce4-tips-autostart.desktop.in	2009-02-15 17:06:32.000000000 -0600
@@ -7,4 +7,4 @@
 Terminal=false
 Type=Application
 OnlyShowIn=XFCE;
-Hidden=false
+Hidden=true
