--- AcetoneISO-6.7/SConstruct	1970-01-01 02:00:00.000000000 +0200
+++ AcetoneISO-6.7/SConstruct	2007-04-21 15:58:40.000000000 +0300
@@ -0,0 +1,13 @@
+import os
+
+prefix = ARGUMENTS.get('prefix', 'install')
+bindir = os.path.join(prefix, 'opt/acetoneiso/bin')
+
+env= Environment()
+pdi2iso= env.Program("src/pdi2iso.c")
+cdi2iso= env.Program("src/cdi2iso.c")
+b5i2iso= env.Program("src/b5i2iso.c")
+mdf2iso= env.Program("src/mdf2iso.c")
+nrg2iso= env.Program("src/nrg2iso.c")
+env.Install(bindir, [pdi2iso, cdi2iso, b5i2iso, mdf2iso, nrg2iso])
+env.Alias('install', bindir)
\ Dosya sonunda yenisatır yok.
