diff -ur msmtp-1.4.13.orig/src/conf.c msmtp-1.4.13/src/conf.c
--- msmtp-1.4.13.orig/src/conf.c	2007-07-31 00:41:09.000000000 +0300
+++ msmtp-1.4.13/src/conf.c	2008-02-15 18:38:30.000000000 +0200
@@ -32,6 +32,7 @@
 #include <ctype.h>
 #include <errno.h>
 
+#include "c-ctype.h"
 #include "gettext.h"
 #include "xalloc.h"
 #include "xvasprintf.h"
@@ -287,7 +288,7 @@
     	l = strlen(arg);
 	for (i = 0; i < l; i++)
 	{
-	    arg[i] = toupper((unsigned char)arg[i]);
+	    arg[i] = c_toupper((unsigned char)arg[i]);
 	}
 	return 0;
     }
@@ -341,7 +342,7 @@
     l = strlen(arg);
     for (i = 0; i < l; i++)
     {
-	arg[i] = toupper((unsigned char)arg[i]);
+	arg[i] = c_toupper((unsigned char)arg[i]);
     }
     return 0;
 }
diff -ur msmtp-1.4.13.orig/src/smtp.c msmtp-1.4.13/src/smtp.c
--- msmtp-1.4.13.orig/src/smtp.c	2007-07-16 23:43:39.000000000 +0300
+++ msmtp-1.4.13/src/smtp.c	2008-02-15 18:39:08.000000000 +0200
@@ -40,6 +40,7 @@
 # include "hmac.h"
 #endif
 
+#include "c-ctype.h"
 #include "gettext.h"
 #include "xalloc.h"
 #include "xvasprintf.h"
@@ -445,7 +446,7 @@
 	/* make line uppercase */
     	for (i = 4; (size_t)i < len; i++)
 	{
-	    s[i] = toupper((unsigned char)s[i]);
+	    s[i] = c_toupper((unsigned char)s[i]);
 	}
 	/* search capabilities */
 	if (strncmp(s + 4, "STARTTLS", 8) == 0)
