diff -Naur PackageKit-0.4.9-orig/etc/Vendor.conf PackageKit-0.4.9/etc/Vendor.conf
--- PackageKit-0.4.9-orig/etc/Vendor.conf	2009-08-27 09:27:36.904467882 +0300
+++ PackageKit-0.4.9/etc/Vendor.conf	2009-08-28 10:14:32.006209060 +0300
@@ -12,7 +12,7 @@
 # If the value is set to 'none' then no link is shown.
 #
 # default=http://www.packagekit.org/pk-package-not-found.html
-DefaultUrl=http://www.packagekit.org/pk-package-not-found.html
+DefaultUrl=http://www.pardus-wiki.org
 
 # The URL which is shown to the user when a codec could not be found.
 # It should explain why certain codecs cannot be used, and perhaps show
@@ -54,7 +54,7 @@
 # The vendor name shown for the 'vendor' group
 #
 # default=My Vendor
-VendorName=My Vendor
+VendorName=Pardus
 
 # The vendor icon name shown for the 'vendor' group
 #
