diff -Naur opera-10.60-6386.i386.linux.orig/share/opera/defaults/pluginpath.ini opera-10.60-6386.i386.linux/share/opera/defaults/pluginpath.ini
--- opera-10.60-6386.i386.linux.orig/share/opera/defaults/pluginpath.ini	2010-06-30 14:23:43.000000000 +0300
+++ opera-10.60-6386.i386.linux/share/opera/defaults/pluginpath.ini	2010-07-06 09:14:44.882016870 +0300
@@ -61,6 +61,7 @@
 $HOME/.netscape/plugins=1
 /opt/netscape/plugins=1
 /usr/lib/netscape/plugins=1
+/usr/lib/nsbrowser/plugins=1
 /usr/local/netscape/plugins=1
 /usr/local/lib/netscape/plugins=1
 $HOME/.kde/.konqueror/nsplugins=1
