diff -Naur opera-10.60-6386.i386.linux.orig/share/applications/opera-browser.desktop opera-10.60-6386.i386.linux/share/applications/opera-browser.desktop
--- opera-10.60-6386.i386.linux.orig/share/applications/opera-browser.desktop	2010-06-30 14:33:07.000000000 +0300
+++ opera-10.60-6386.i386.linux/share/applications/opera-browser.desktop	2010-07-06 09:36:34.535016105 +0300
@@ -21,6 +21,7 @@
 GenericName[ro]=Navigator de web
 GenericName[ru]=Веб-браузер
 GenericName[sl]=Spletni brskalnik
+GenericName[tr]=Web Tarayıcı
 GenericName[ven]=Buronza ya Webu
 GenericName[xh]=Umkhangeli Zincwadi Zokubhaliweyo
 GenericName[zu]=Umkhangeli zincwadi we Web
