diff -Naur libQGLViewer-2.2.6-3.old/QGLViewer/QGLViewer.pro libQGLViewer-2.2.6-3/QGLViewer/QGLViewer.pro
--- libQGLViewer-2.2.6-3.old/QGLViewer/QGLViewer.pro	2007-08-28 21:38:55.000000000 +0000
+++ libQGLViewer-2.2.6-3/QGLViewer/QGLViewer.pro	2008-03-01 23:24:12.973961031 +0000
@@ -55,9 +55,9 @@
   }
 }
 
-!isEmpty( QGLVIEWER_STATIC ) {
-  CONFIG *= staticlib
-}
+# !isEmpty( QGLVIEWER_STATIC ) {
+#  CONFIG *= staticlib
+# }
 
 
 #		--  U n i x  --
@@ -74,9 +74,9 @@
     INCLUDE_DIR = $${PREFIX}/include
   }
 
-  isEmpty( DOC_DIR ) {
-    DOC_DIR = $${PREFIX}/share/doc
-  }
+  # isEmpty( DOC_DIR ) {
+  #  DOC_DIR = $${PREFIX}/share/doc
+  # }
 
   # GLUT for Unix architecture
   !isEmpty( USE_GLUT ) {
@@ -101,31 +101,31 @@
   include.files = $${HEADERS} qglviewer.cw
 
   # install documentation html
-  documentation.path = $${DOC_DIR}/QGLViewer
-  documentation.files = ../doc/*.html ../doc/*.css
+  #documentation.path = $${DOC_DIR}/QGLViewer
+  #documentation.files = ../doc/*.html ../doc/*.css
 
   # install documentation images
-  docImages.path = $${DOC_DIR}/QGLViewer/images
-  docImages.files = ../doc/images/*
+  #docImages.path = $${DOC_DIR}/QGLViewer/images
+  #docImages.files = ../doc/images/*
 
   # install documentation examples
   #docExamples.path = $${DOC_DIR}/QGLViewer/examples
   #docExamples.files = ../examples/*../examples/*/*
 
   # install documentation refManual
-  docRefManual.path = $${DOC_DIR}/QGLViewer/refManual
-  docRefManual.files = ../doc/refManual/*
+  #docRefManual.path = $${DOC_DIR}/QGLViewer/refManual
+  #docRefManual.files = ../doc/refManual/*
 
   # install static library
-  staticlib.extra = make -f Makefile.Release staticlib
-  staticlib.path = $${LIB_DIR}
-  staticlib.files = lib$${TARGET}.a
+  #staticlib.extra = make -f Makefile.Release staticlib
+  #staticlib.path = $${LIB_DIR}
+  #staticlib.files = lib$${TARGET}.a
 
   # install library
   target.path = $${LIB_DIR}
 
   # "make install" configuration options
-  INSTALLS *= target staticlib include documentation docImages docRefManual
+  INSTALLS *= target include
 }
 
 
