--- gnome-bluetooth-0.7.0/obex/gnome-obex-server.desktop.in	2005-11-20 07:20:01.000000000 +1100
+++ gnome-bluetooth-0.7.0/obex/gnome-obex-server.desktop.in	2006-05-18 10:01:38.000000000 +1000
@@ -4,9 +4,9 @@
 _Comment=Receive files sent by Bluetooth devices
 Icon=blueradio-48
 Exec=gnome-obex-server
-Terminal=0
+Terminal=false
 Type=Application
-Categories=Application;System;
+Categories=Application;Utility;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-bluetooth
 X-GNOME-Bugzilla-Component=obex-server
