diff -up gamin-0.1.9/server/gam_fs.c.filesystem-leak gamin-0.1.9/server/gam_fs.c
--- gamin-0.1.9/server/gam_fs.c.filesystem-leak	2007-09-13 00:21:18.000000000 -0400
+++ gamin-0.1.9/server/gam_fs.c	2007-09-13 00:21:25.000000000 -0400
@@ -40,13 +40,13 @@ gam_fs_free_filesystems (void)
 	{
 		fs = iterator->data;
 
+		iterator = g_list_next (iterator);
+
 		filesystems = g_list_remove (filesystems, fs);
 
 		g_free (fs->path);
 		g_free (fs->fsname);
 		g_free (fs);
-
-		iterator = g_list_next (filesystems);
 	}
 }
 
