--- soundkonverter.desktop.orig	2008-05-19 12:59:01.000000000 +0300
+++ src/soundkonverter.desktop	2008-05-19 13:00:08.000000000 +0300
@@ -9,7 +9,9 @@
 Comment=An audio converter and CD ripper
 Comment[de]=Ein Audiokonverter und CD Ripper
 Comment[fr]=Convertisseur de fichiers audio
+Comment[tr]=Ses ve CD Dönüştürücü
 GenericName=Audio files converter and CD ripper
 GenericName[de]=Audiodateien Konvertierer und CD Ripper
 GenericName[fr]=Convertisseur Audio
+GenericName[tr]=Ses ve CD Dönüştürücü
 Terminal=false
