--- src/gtk-recordmydesktop.desktop.orig	2008-10-06 08:40:01.000000000 +0300
+++ src/gtk-recordmydesktop.desktop	2008-10-06 08:42:12.000000000 +0300
@@ -2,8 +2,11 @@
 Version=1.0
 Encoding=UTF-8
 Name=gtk-recordMyDesktop
+GenericName=Frontend for recordMyDesktop
+GenericName[tr]=recordMyDesktop için arayüz
 Comment=Frontend for recordMyDesktop
 Comment[fr]=Interface pour recordmydesktop
+Comment[tr]=recordMyDesktop için arayüz
 Exec=gtk-recordMyDesktop
 Terminal=false
 Type=Application
