diff -p -up audacious-plugins-1.5.1/src/adplug/Makefile.linking audacious-plugins-1.5.1/src/adplug/Makefile
--- audacious-plugins-1.5.1/src/adplug/Makefile.linking	2008-05-24 00:44:19.000000000 +0200
+++ audacious-plugins-1.5.1/src/adplug/Makefile	2008-05-26 15:47:28.000000000 +0200
@@ -61,4 +61,4 @@ plugindir := ${plugindir}/${INPUT_PLUGIN
 CFLAGS += ${PLUGIN_CFLAGS}
 CXXFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${MOWGLI_CFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS}  ${BINIO_CFLAGS} -I../.. -I./core -Dstricmp=strcasecmp
-LIBS += ${BINIO_LIBS} ${GTK_LIBS} ${GLIB_LIBS}  ${MOWGLI_LIBS}
+LIBS += ${BINIO_LIBS} ${GTK_LIBS} ${GLIB_LIBS}  ${MOWGLI_LIBS} -lstdc++ -lm
diff -p -up audacious-plugins-1.5.1/src/amidi-plug/i_configure.c.linking audacious-plugins-1.5.1/src/amidi-plug/i_configure.c
--- audacious-plugins-1.5.1/src/amidi-plug/i_configure.c.linking	2008-05-24 00:44:19.000000000 +0200
+++ audacious-plugins-1.5.1/src/amidi-plug/i_configure.c	2008-05-26 15:47:04.000000000 +0200
@@ -29,6 +29,7 @@
 #include "i_configure-dummy.h"
 #include "i_utils.h"
 #include <audacious/auddrct.h>
+#include <audacious/plugin.h>
 
 
 amidiplug_cfg_backend_t * amidiplug_cfg_backend;
diff -p -up audacious-plugins-1.5.1/src/hotkey/Makefile.linking audacious-plugins-1.5.1/src/hotkey/Makefile
--- audacious-plugins-1.5.1/src/hotkey/Makefile.linking	2008-05-24 00:44:19.000000000 +0200
+++ audacious-plugins-1.5.1/src/hotkey/Makefile	2008-05-26 15:47:04.000000000 +0200
@@ -9,4 +9,4 @@ plugindir := ${plugindir}/${GENERAL_PLUG
 
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${GTK_CFLAGS} ${MOWGLI_CFLAGS} ${GLIB_CFLAGS}  -I../.. -I..
-LIBS += ${GLIB_LIBS}  ${GTK_LIBS}
+LIBS += ${GLIB_LIBS}  ${GTK_LIBS} -lX11
diff -p -up audacious-plugins-1.5.1/src/jack/Makefile.linking audacious-plugins-1.5.1/src/jack/Makefile
--- audacious-plugins-1.5.1/src/jack/Makefile.linking	2008-05-24 00:44:19.000000000 +0200
+++ audacious-plugins-1.5.1/src/jack/Makefile	2008-05-26 15:47:04.000000000 +0200
@@ -11,4 +11,4 @@ plugindir := ${plugindir}/${OUTPUT_PLUGI
 
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${MOWGLI_CFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS}  ${SAMPLERATE_CFLAGS} -I../..
-LIBS += ${GTK_LIBS} ${GLIB_LIBS}  ${JACK_LIBS} ${SAMPLERATE_LIBS} -pthread
+LIBS += ${GTK_LIBS} ${GLIB_LIBS}  ${JACK_LIBS} ${SAMPLERATE_LIBS} -pthread -lm
diff -p -up audacious-plugins-1.5.1/src/modplug/Makefile.linking audacious-plugins-1.5.1/src/modplug/Makefile
--- audacious-plugins-1.5.1/src/modplug/Makefile.linking	2008-05-24 00:44:19.000000000 +0200
+++ audacious-plugins-1.5.1/src/modplug/Makefile	2008-05-26 15:47:04.000000000 +0200
@@ -53,4 +53,4 @@ plugindir := ${plugindir}/${INPUT_PLUGIN
 CFLAGS += ${PLUGIN_CFLAGS}
 CXXFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${MOWGLI_CFLAGS}  ${GTK_CFLAGS} ${GLIB_CFLAGS}  -I.. -I../.. -I../../intl
-LIBS += ${MOWGLI_LIBS}  ${GTK_LIBS} ${GLIB_LIBS}  -lstdc++
+LIBS += ${MOWGLI_LIBS}  ${GTK_LIBS} ${GLIB_LIBS}  -lstdc++ -lm
diff -p -up audacious-plugins-1.5.1/src/musepack/Makefile.linking audacious-plugins-1.5.1/src/musepack/Makefile
--- audacious-plugins-1.5.1/src/musepack/Makefile.linking	2008-05-24 00:44:20.000000000 +0200
+++ audacious-plugins-1.5.1/src/musepack/Makefile	2008-05-26 15:47:04.000000000 +0200
@@ -9,4 +9,4 @@ plugindir := ${plugindir}/${INPUT_PLUGIN
 
 CXXFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${MOWGLI_CFLAGS}  ${GTK_CFLAGS} ${GLIB_CFLAGS}  ${TAGLIB_CFLAGS} -I../..
-LIBS += ${TAGLIB_LIBS} ${GTK_LIBS} ${GLIB_LIBS}  ${MOWGLI_LIBS} -lmpcdec
+LIBS += ${TAGLIB_LIBS} ${GTK_LIBS} ${GLIB_LIBS}  ${MOWGLI_LIBS} -lmpcdec -lstdc++ -lm
diff -p -up audacious-plugins-1.5.1/src/projectm-1.0/Makefile.linking audacious-plugins-1.5.1/src/projectm-1.0/Makefile
--- audacious-plugins-1.5.1/src/projectm-1.0/Makefile.linking	2008-05-24 00:44:20.000000000 +0200
+++ audacious-plugins-1.5.1/src/projectm-1.0/Makefile	2008-05-26 15:47:04.000000000 +0200
@@ -11,4 +11,4 @@ plugindir := ${plugindir}/${VISUALIZATIO
 CFLAGS += ${PLUGIN_CFLAGS}
 CXXFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${MOWGLI_CFLAGS}  ${GTK_CFLAGS} ${GLIB_CFLAGS}  ${ARCH_DEFINES} ${XML_CPPFLAGS} ${GTKGLEXT_CFLAGS} ${LIBPROJECTM1_CFLAGS} -I../..
-LIBS += ${GTK_LIBS} ${GLIB_LIBS}  ${GTKGLEXT_LIBS} ${LIBPROJECTM1_LIBS}
+LIBS += ${GTK_LIBS} ${GLIB_LIBS}  ${GTKGLEXT_LIBS} ${LIBPROJECTM1_LIBS} -lstdc++
diff -p -up audacious-plugins-1.5.1/src/rootvis/Makefile.linking audacious-plugins-1.5.1/src/rootvis/Makefile
--- audacious-plugins-1.5.1/src/rootvis/Makefile.linking	2008-05-24 00:44:20.000000000 +0200
+++ audacious-plugins-1.5.1/src/rootvis/Makefile	2008-05-26 15:47:51.000000000 +0200
@@ -14,4 +14,4 @@ plugindir := ${plugindir}/${VISUALIZATIO
 
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${MOWGLI_CFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS}  ${IMLIB2_CFLAGS} -I.
-LIBS += ${GTK_LIBS} ${GLIB_LIBS}  ${IMLIB2_LIBS}
+LIBS += ${GTK_LIBS} ${GLIB_LIBS}  ${IMLIB2_LIBS} -lm -lX11
diff -p -up audacious-plugins-1.5.1/src/sid/Makefile.linking audacious-plugins-1.5.1/src/sid/Makefile
--- audacious-plugins-1.5.1/src/sid/Makefile.linking	2008-05-24 00:44:20.000000000 +0200
+++ audacious-plugins-1.5.1/src/sid/Makefile	2008-05-26 15:47:04.000000000 +0200
@@ -26,4 +26,4 @@ plugindir := ${plugindir}/${INPUT_PLUGIN
 CFLAGS += ${PLUGIN_CFLAGS}
 CXXFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} -D_REENTRANT -I../.. -DAUDACIOUS_PLUGIN ${MOWGLI_CFLAGS} ${SIDPLAY1_CFLAGS} ${SIDPLAY2_CFLAGS} ${BUILDERS_CFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS}
-LIBS += ${BUILDERS_LDFLAGS} ${SIDPLAY1_LIBS} ${SIDPLAY2_LIBS} ${BUILDERS_LIBS} ${GTK_LIBS} ${GLIB_LIBS} -lstdc++
+LIBS += ${BUILDERS_LDFLAGS} ${SIDPLAY1_LIBS} ${SIDPLAY2_LIBS} ${BUILDERS_LIBS} ${GTK_LIBS} ${GLIB_LIBS} -lstdc++ -lm
diff -p -up audacious-plugins-1.5.1/src/sndfile/Makefile.linking audacious-plugins-1.5.1/src/sndfile/Makefile
--- audacious-plugins-1.5.1/src/sndfile/Makefile.linking	2008-05-24 00:44:20.000000000 +0200
+++ audacious-plugins-1.5.1/src/sndfile/Makefile	2008-05-26 15:47:04.000000000 +0200
@@ -10,4 +10,4 @@ plugindir := ${plugindir}/${INPUT_PLUGIN
 
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${MOWGLI_CFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS}   -I../.. 
-LIBS += ${MOWGLI_LIBS} ${SNDFILE_LIBS} ${GTK_LIBS} ${GLIB_LIBS} 
+LIBS += ${MOWGLI_LIBS} ${SNDFILE_LIBS} ${GTK_LIBS} ${GLIB_LIBS} -lm
diff -p -up audacious-plugins-1.5.1/src/sndstretch/Makefile.linking audacious-plugins-1.5.1/src/sndstretch/Makefile
--- audacious-plugins-1.5.1/src/sndstretch/Makefile.linking	2008-05-24 00:44:20.000000000 +0200
+++ audacious-plugins-1.5.1/src/sndstretch/Makefile	2008-05-26 15:47:04.000000000 +0200
@@ -10,4 +10,4 @@ plugindir := ${plugindir}/${EFFECT_PLUGI
 
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS} ${MOWGLI_CFLAGS}  -I../..
-LIBS += ${GTK_LIBS} ${GLIB_LIBS} ${MOWGLI_LIBS} 
+LIBS += ${GTK_LIBS} ${GLIB_LIBS} ${MOWGLI_LIBS} -lm
diff -p -up audacious-plugins-1.5.1/src/spectrum/Makefile.linking audacious-plugins-1.5.1/src/spectrum/Makefile
--- audacious-plugins-1.5.1/src/spectrum/Makefile.linking	2008-05-24 00:44:20.000000000 +0200
+++ audacious-plugins-1.5.1/src/spectrum/Makefile	2008-05-26 15:47:04.000000000 +0200
@@ -9,4 +9,4 @@ plugindir := ${plugindir}/${VISUALIZATIO
 
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${MOWGLI_CFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS}  -I../..
-LIBS += ${GTK_LIBS} ${GLIB_LIBS} 
+LIBS += ${GTK_LIBS} ${GLIB_LIBS} -lm
diff -p -up audacious-plugins-1.5.1/src/statusicon/Makefile.linking audacious-plugins-1.5.1/src/statusicon/Makefile
--- audacious-plugins-1.5.1/src/statusicon/Makefile.linking	2008-05-24 00:44:20.000000000 +0200
+++ audacious-plugins-1.5.1/src/statusicon/Makefile	2008-05-26 15:47:04.000000000 +0200
@@ -12,4 +12,4 @@ plugindir := ${plugindir}/${GENERAL_PLUG
 
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${BEEP_DEFINES} ${MOWGLI_CFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS}  -I../..
-LIBS += ${GTK_LIBS} ${GLIB_LIBS} 
+LIBS += ${GTK_LIBS} ${GLIB_LIBS} -lX11
diff -p -up audacious-plugins-1.5.1/src/tonegen/Makefile.linking audacious-plugins-1.5.1/src/tonegen/Makefile
--- audacious-plugins-1.5.1/src/tonegen/Makefile.linking	2008-05-24 00:44:20.000000000 +0200
+++ audacious-plugins-1.5.1/src/tonegen/Makefile	2008-05-26 15:47:04.000000000 +0200
@@ -9,4 +9,4 @@ plugindir := ${plugindir}/${INPUT_PLUGIN
 
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${MOWGLI_CFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS}  -I../..
-LIBS += ${GTK_LIBS} ${GLIB_LIBS} 
+LIBS += ${GTK_LIBS} ${GLIB_LIBS} -lm
diff -p -up audacious-plugins-1.5.1/src/wma/Makefile.linking audacious-plugins-1.5.1/src/wma/Makefile
--- audacious-plugins-1.5.1/src/wma/Makefile.linking	2008-05-24 00:44:20.000000000 +0200
+++ audacious-plugins-1.5.1/src/wma/Makefile	2008-05-26 15:47:04.000000000 +0200
@@ -28,4 +28,4 @@ plugindir := ${plugindir}/${INPUT_PLUGIN
 
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${MOWGLI_CFLAGS}  ${GTK_CFLAGS} ${GLIB_CFLAGS}  ${SIMD_CFLAGS} -I../.. -I./libffwma -std=c99 ${GCC42_CFLAGS} -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE
-LIBS += ${GTK_LIBS} ${GLIB_LIBS} 
+LIBS += ${GTK_LIBS} ${GLIB_LIBS} -lm
diff -p -up audacious-plugins-1.5.1/src/alsa/Makefile~ audacious-plugins-1.5.1/src/alsa/Makefile
--- audacious-plugins-1.5.1/src/alsa/Makefile~	2008-05-24 00:44:19.000000000 +0200
+++ audacious-plugins-1.5.1/src/alsa/Makefile	2008-07-18 10:33:21.000000000 +0200
@@ -12,4 +12,4 @@ plugindir := ${plugindir}/${OUTPUT_PLUGI
 
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${MOWGLI_CFLAGS}  ${GTK_CFLAGS} ${GLIB_CFLAGS}  ${ALSA_CFLAGS} -I../..
-LIBS += ${GTK_LIBS} ${GLIB_LIBS}  ${ALSA_LIBS}
+LIBS += ${GTK_LIBS} ${GLIB_LIBS}  ${ALSA_LIBS} -ldl
diff -p -up audacious-plugins-1.5.1/src/ladspa/Makefile~ audacious-plugins-1.5.1/src/ladspa/Makefile
--- audacious-plugins-1.5.1/src/ladspa/Makefile~	2008-05-24 00:44:19.000000000 +0200
+++ audacious-plugins-1.5.1/src/ladspa/Makefile	2008-07-18 10:34:00.000000000 +0200
@@ -9,4 +9,4 @@ plugindir := ${plugindir}/${EFFECT_PLUGI
 
 CFLAGS += ${PLUGIN_CFLAGS}
 CPPFLAGS += ${PLUGIN_CPPFLAGS} ${MOWGLI_CFLAGS} ${GTK_CFLAGS} ${GLIB_CFLAGS}  -I../..
-LIBS += ${GTK_LIBS} ${GLIB_LIBS} 
+LIBS += ${GTK_LIBS} ${GLIB_LIBS} -ldl
