--- src/LinConverter.kmdr~	2008-02-09 22:52:12.000000000 +0200
+++ src/LinConverter.kmdr	2008-02-09 23:13:48.000000000 +0200
@@ -239,7 +239,7 @@
                     </property>
                     <property name="associations" stdset="0">
                         <stringlist>
-                            <string>konsole -e ffmpeg -i "@fileSelector.text" -s qcif -r 12 -ac 1 -ar 8000 -b 30 -ab 12 "@outputselector.text".3gp</string>
+                            <string>konsole -e ffmpeg -i "@fileSelector.text" -s qcif -vcodec h263 -acodec aac -ac 1 -ar 8000 -r 25 -ab 32 -y "@outputselector.text".3gp</string>
                         </stringlist>
                     </property>
                 </widget>
@@ -503,7 +503,7 @@
                     </property>
                     <property name="associations" stdset="0">
                         <stringlist>
-                            <string>konsole -e ffmpeg -i "@fileSelector.text" -s qcif -r 12 -ac 1 -ar 8000 -b 30 -ab 12 "@outputselector.text".3gp</string>
+                            <string>konsole -e ffmpeg -i "@fileSelector.text" -s qcif -vcodec h263 -acodec aac -ac 1 -ar 8000 -r 25 -ab 32 -y "@outputselector.text".3gp</string>
                         </stringlist>
                     </property>
                 </widget>
@@ -935,7 +935,7 @@
                     </property>
                     <property name="associations" stdset="0">
                         <stringlist>
-                            <string>konsole -e ffmpeg -i "@fileSelector.text" -s qcif -r 12 -b 30 -ab 12 -ac 1 -ar 8000 "@outputselector.text".3gp</string>
+                            <string>konsole -e ffmpeg -i "@fileSelector.text" -s qcif -vcodec h263 -acodec aac -ac 1 -ar 8000 -r 25 -ab 32 -y "@outputselector.text".3gp</string>
                         </stringlist>
                     </property>
                 </widget>
@@ -1415,7 +1415,7 @@
                     </property>
                     <property name="associations" stdset="0">
                         <stringlist>
-                            <string>konsole -e ffmpeg -i "@fileSelector.text" -s qcif -r 12 -b 30 -ab 12 -ac 1 -ar 8000 "@outputselector.text".3gp</string>
+                            <string>konsole -e ffmpeg -i "@fileSelector.text" -s qcif -vcodec h263 -acodec aac -ac 1 -ar 8000 -r 25 -ab 32 -y "@outputselector.text".3gp</string>
                         </stringlist>
                     </property>
                 </widget>
