diff -Nuar xfce4-panel-4.6.1.orig/plugins/clock/Makefile.in xfce4-panel-4.6.1/plugins/clock/Makefile.in
--- xfce4-panel-4.6.1.orig/plugins/clock/Makefile.in	2009-06-05 20:21:03.000000000 +0300
+++ xfce4-panel-4.6.1/plugins/clock/Makefile.in	2009-06-05 20:22:03.000000000 +0300
@@ -331,7 +331,7 @@
 	$(GTK_LIBS)							\
 	$(CAIRO_LIBS)							\
 	$(LIBXFCE4UTIL_LIBS)						\
-	$(LIBXFCEGUI4_LIBS)
+	$(LIBXFCEGUI4_LIBS) -lm
 
 libclock_la_DEPENDENCIES = \
 	$(top_builddir)/libxfce4panel/libxfce4panel.la
