diff -uNr lives-0.9.8.9/lives-plugins/marcos-encoders/avi_encoder.py lives-0.9.8.9-fixed/lives-plugins/marcos-encoders/avi_encoder.py
--- lives-0.9.8.9/lives-plugins/marcos-encoders/avi_encoder.py	2007-12-10 22:38:37.000000000 -0500
+++ lives-0.9.8.9-fixed/lives-plugins/marcos-encoders/avi_encoder.py	2008-03-16 09:15:22.000000000 -0400
@@ -22,7 +22,7 @@
 See my vids at http://amv.reimeika.ca !
 """
 
-version = '0.0.9'
+version = '0.0.10'
 convert = 'convert'
 mencoder = 'mencoder'
 lameenc = 'lame'
@@ -372,7 +372,7 @@
         cpass = ''
     elif vtype[-1] == 'h':
         codec_vopts = '-ovc x264 -x264encopts ' + \
-                      'frameref=15:deblockalpha=3:4x4mv:subq=5' + hinfo + \
+                      'frameref=15:subq=5' + hinfo + \
                       ':bitrate='
         cpass = 'pass='
     else:
