diff -Nuar tork-0.21.orig/src/tsocks/Makefile.in tork-0.21/src/tsocks/Makefile.in
--- tork-0.21.orig/src/tsocks/Makefile.in	2007-10-20 21:23:06.000000000 +0300
+++ tork-0.21/src/tsocks/Makefile.in	2007-10-29 19:56:32.000000000 +0200
@@ -532,7 +532,7 @@
 %.o: %.c
 	${SHCC} ${CFLAGS} ${INCLUDES} -c ${CC_SWITCHES} $< -o $@
 
-install: ${TARGETS} installscript installtorkify installtorkifyconf installlib installman
+install: ${TARGETS} installscript installtorkify
 
 installscript:
 	${MKINSTALLDIRS} "${TSOCKSDESTDIR}${bindir}"
diff -Nuar tork-0.21.orig/src/tsocks/torkify.in tork-0.21/src/tsocks/torkify.in
--- tork-0.21.orig/src/tsocks/torkify.in	2007-10-07 21:04:15.000000000 +0300
+++ tork-0.21/src/tsocks/torkify.in	2007-10-29 19:53:27.000000000 +0200
@@ -1,20 +1,21 @@
 #! /bin/sh
 
-# Wrapper script for use of the torksocks(8) transparent socksification library
-# See the torksocks(1) and torkify(1) manpages.
+# Wrapper script for use of the tsocks(8) transparent socksification library
+# See the tsocks(1) and torify(1) manpages.
 
 # Copyright (c) 2004, 2006 Peter Palfrader
 # Modified by Jacob Appelbaum <jacob@appelbaum.net> April 16th 2006
 # Modified by Robert Hogan <robert@roberthogan.net> March 2007
+# Modified by Ertuğrul Erata <ertugerata@gmail.com> june 2007 for external tsocks
 # May be distributed under the same terms as Tor itself
 
 
-# Define and ensure we have torksocks
+# Define and ensure we have tsocks
 # XXX: what if we don't have which?
-TSOCKS="`which torksocks`"
+TSOCKS="`which tsocks`"
 if [ ! -x "$TSOCKS" ]
 then
-        echo "$0: Can't find torksocks in PATH. Perhaps you haven't installed it?" >&2
+        echo "$0: Can't find tsocks in PATH. Perhaps you haven't installed it?" >&2
         exit 1
 fi
 
@@ -30,17 +31,17 @@
         exit 0
 fi
 
-# Define our torksocks config file
-TSOCKS_CONF_FILE="@CONFDIR@/tork-tsocks.conf"
+# Define our tsocks config file
+TSOCKS_CONF_FILE="/etc/tor/tor-tsocks.conf"
 export TSOCKS_CONF_FILE
 
-# Check that we've got a torksocks config file
+# Check that we've got a tsocks config file
 if [ -r "$TSOCKS_CONF_FILE" ]
 then
-	exec torksocks "$@"
-	echo "$0: Failed to exec torksocks $@" >&2
+	exec tsocks "$@"
+	echo "$0: Failed to exec tsocks $@" >&2
 	exit 1
 else
-	echo "$0: Missing torksocks configuration file \"$TSOCKS_CONF_FILE\"." >&2
+	echo "$0: Missing tsocks configuration file \"$TSOCKS_CONF_FILE\"." >&2
 	exit 1
 fi
