--- image/cinelerra.desktop.orig	2009-08-05 13:04:36.111372179 +0300
+++ image/cinelerra.desktop	2009-08-05 13:08:39.356374409 +0300
@@ -4,7 +4,11 @@
 Name[sl]=Cinelerra
 Name[fr]=Cinelerra
 Name[es]=Cinelerra
+Name[tr]=Cinelerra
+GenericName=Video Editor
+GenericName[tr]=Video Düzenleyici
 Comment=Video Editor
+Comment[tr]=Videolarınızı düzenlemeye yarar
 Categories=Application;AudioVideo;Multimedia;
 Encoding=UTF-8
 Exec=cinelerra
