diff -uNrBbE cinelerra-2.1/configure.in cinelerra-2.1-new/configure.in
--- cinelerra-2.1/configure.in	2009-02-01 22:24:07 +0300
+++ cinelerra-2.1-new/configure.in	2009-02-01 21:36:06 +0300
@@ -568,7 +568,6 @@
 					plugins/timefront/Makefile \
 					plugins/svg/Makefile \
 					plugins/diffkey/Makefile \
-					plugins/fonts/Makefile \
 					image/Makefile)
 
 succeeded=yes
diff -uNrBbE cinelerra-2.1/plugins/Makefile.am cinelerra-2.1-new/plugins/Makefile.am
--- cinelerra-2.1/plugins/Makefile.am	2009-01-07 21:51:51 +0300
+++ cinelerra-2.1-new/plugins/Makefile.am	2009-02-01 21:33:43 +0300
@@ -102,8 +102,7 @@
 	whirl \
 	wipe \
 	yuv \
-	zoomblur \
-	fonts
+	zoomblur
 
 AUTOMAKE_OPTIONS=foreign
 
