Index: edje/src/lib/edje_entry.c
===================================================================
--- edje/src/lib/edje_entry.c	(revision 46192)
+++ edje/src/lib/edje_entry.c	(revision 46215)
@@ -1509,7 +1509,7 @@
 }
 
 static void
-_evas_focus_in_cb(void *data, Evas *e, Evas_Object *obj, void *event_info)
+_evas_focus_in_cb(void *data, Evas *e, void *event_info)
 {
    Edje *ed = (Edje *)data;
 
@@ -1519,7 +1519,7 @@
 }
 
 static void
-_evas_focus_out_cb(void *data, Evas *e, Evas_Object *obj, void *event_info)
+_evas_focus_out_cb(void *data, Evas *e, void *event_info)
 {
    Edje *ed = (Edje *)data;
 
