Index: src/testgtkhtml.c
===================================================================
--- src/testgtkhtml.c	(revision 8407)
+++ src/testgtkhtml.c	(working copy)
@@ -327,7 +327,7 @@ draw_page_cb (GtkPrintOperation *operati
 
 	metrics = pango_context_get_metrics (
 		pango_layout_get_context (layout),
-		desc, pango_language_get_default ());
+		desc, gtk_get_default_language ());
 	footer_height = (pango_font_metrics_get_ascent (metrics) +
 		pango_font_metrics_get_descent (metrics)) / PANGO_SCALE;
 	pango_font_metrics_unref (metrics);
