--- Makefile.in.orig	2006-02-14 14:14:22.000000000 +0200
+++ Makefile.in	2006-10-25 13:14:06.000000000 +0300
@@ -206,28 +206,12 @@
 	pvm3/bin/@PVMARCH@/*
 
 INSTBINDISTFILES = \
-	.pvmd.conf \
-	ACKNOWLEDGEMENTS \
-	CHANGES \
-	Makefile \
-        Makefile.OBJ \
-	Makefile.incl \
-	Makemex \
-        Path.incl \
-	README_Unix \
-	Version.incl \
-	configure \
-	libtool \
-	license.txt \
-	licence.txt \
 	scilab.quit \
 	scilab.star \
 	X11_defaults \
 	bin \
-	config \
 	contrib \
 	demos \
-	examples \
 	imp/NperiPos.ps \
 	imp/giffonts \
 	macros \
@@ -258,27 +242,23 @@
 DOCNAME=$(SCIBASE)
 
 install:
-	@if test `pwd`/..  != $(PREFIX)/lib ; then \
-		install -d $(PREFIX)/lib/$(SCIBASE); \
-		( for file in $(INSTBINDISTFILES) @PVMINST1@; do tar cf - $${file}  | (cd $(PREFIX)/lib/$(SCIBASE); tar xf -); done); \
-		(cd $(PREFIX)/lib/$(SCIBASE);touch .binary; make postinstall PREFIX=$(PREFIX) ) \
+	@if test `pwd`/..  != $(PREFIX)/share ; then \
+		install -d $(PREFIX)/share/$(SCIBASE); \
+		( for file in $(INSTBINDISTFILES) @PVMINST1@; do tar cf - $${file}  | (cd $(PREFIX)/share/$(SCIBASE); tar xf -); done); \
 	fi
-	install -d $(PREFIX)/share/doc/$(DOCNAME)/
-	install ACKNOWLEDGEMENTS CHANGES README_Unix Version.incl \
-		licence.txt license.txt  $(PREFIX)/share/doc/$(DOCNAME)
 	install -d $(PREFIX)/bin 
 	$(RM)  $(PREFIX)/bin/scilab
-	ln -fs $(PREFIX)/lib/$(SCIBASE)/bin/scilab $(PREFIX)/bin/scilab
+	ln -fs /usr/share/$(SCIBASE)/bin/scilab $(PREFIX)/bin/scilab
 	$(RM)  $(PREFIX)/bin/intersci
-	ln -fs $(PREFIX)/lib/$(SCIBASE)/bin/intersci $(PREFIX)/bin/intersci
+	ln -fs /usr/share/$(SCIBASE)/bin/intersci $(PREFIX)/bin/intersci
 	$(RM)  $(PREFIX)/bin/intersci-n
-	ln -fs $(PREFIX)/lib/$(SCIBASE)/bin/intersci-n $(PREFIX)/bin/intersci-n
+	ln -fs /usr/share/$(SCIBASE)/bin/intersci-n $(PREFIX)/bin/intersci-n
 
 # DESTDIR is the final destination of the file 
 # it can be different from $(PREFIX)/lib
 # when we use package builders for rpm or deb files 
 
-DESTDIR=$(PREFIX)/lib
+DESTDIR=$(PREFIX)/share
 
 postinstall: 
 	@echo "creating Path.incl SCIDIR=$(DESTDIR)/$(SCIBASE)"
