--- /dev/null	2006-10-25 13:55:08.864481206 +0300
+++ scilab.desktop	2006-10-25 14:33:38.000000000 +0300
@@ -0,0 +1,12 @@
+[Desktop Entry]
+Name=SciLab
+Name[tr]=SciLab
+Comment=Numerical Computations Software
+comment[tr]=Rakamsal Hesaplama Yazılımı
+GenericName=Numerical Computation
+GenericName[tr]=Rakamsal Hesaplama
+Encoding=UTF-8
+Exec=scilab
+Icon=scilab.xpm
+Type=Application
+Categories=Application;Education;Science


