diff -Naur kbackup-0.6.1.orig/src/kbackup.desktop kbackup-0.6.1/src/kbackup.desktop
--- kbackup-0.6.1.orig/src/kbackup.desktop	2009-07-04 21:43:19.109414101 +0300
+++ kbackup-0.6.1/src/kbackup.desktop	2009-07-04 21:55:19.050387699 +0300
@@ -2,18 +2,20 @@
 Type=Application
 Encoding=UTF-8
 Exec=kbackup %f
-Name=KBackup (Backup Program)
+Name=KBackup
+GenericName=Backup Program
+GenericName[tr]=Yedekleme Uygulaması
 Comment=Backup your data with an easy to use user interface
 Icon=kbackup
 MimeType=text/x-kbp;
 Terminal=false
 Categories=System;KDE;X-SuSE-Backup;X-MandrivaLinux-System-Archiving-Backup;
 
-Name[de]=KBackup (Datensicherungs Programm)
+Name[de]=KBackup
 Comment[de]=Datensicherung mit einem einfach zu bedienended User Interface
 
-Name[fr]=KBackup (Programme de sauvegarde)
+Name[fr]=KBackup
 Comment[fr]=Sauvegarder vos données avec une interface utilisateur facile à utiliser
 
-Name[es]=KBackup (Programa de Copias de Seguridad)
+Name[es]=KBackup
 Comment[es]=Realiza copias de seguridad de tus datos con un programa fácil de usar
