--- isomaster.desktop.src.orig	2007-02-09 22:13:57.000000000 +0200
+++ isomaster.desktop.src	2007-05-08 10:51:19.000000000 +0300
@@ -1,11 +1,13 @@
 [Desktop Entry]
 Name=ISO Master
 GenericName=CD image editor
+GenericName[tr]=ISO Düzenleyici
 Comment=Graphical CD image editor for reading, modifying and writing ISO images
 Exec=isomaster
 Encoding=UTF-8
 Terminal=false
 StartupNotify=true
 Type=Application
-Categories=Utility;DiscBurning;AudioVideo;AudioVideoEditing;
+Categories=KDE;Application;Utility
+Icon=/usr/share/pixmaps/isomaster.png
 MimeType=application/x-iso
