--- xpad-3.1.orig/xpad.desktop.in	2009-01-11 18:27:39.000000000 +0200
+++ xpad-3.1/xpad.desktop.in	2009-03-27 20:32:53.000000000 +0200
@@ -4,6 +4,9 @@
 _Name=Xpad
 _GenericName=Sticky Notes
 _Comment=Jot down notes for later
+Name[tr]=Xpad
+GenericName[tr]=Yapışkan Notlar
+Comment[tr]=Daha sonrası için not alın
 
 TryExec=xpad
 Exec=xpad
@@ -15,4 +18,3 @@
 
 Type=Application
 Categories=GTK;Utility;
-
