diff -Naur fusion-icon-0.1.0_20080711.orig/FusionIcon/interface.py fusion-icon-0.1.0_20080711/FusionIcon/interface.py
--- fusion-icon-0.1.0_20080711.orig/FusionIcon/interface.py	2008-07-11 23:16:10.000000000 +0300
+++ fusion-icon-0.1.0_20080711/FusionIcon/interface.py	2008-07-11 23:21:01.000000000 +0300
@@ -23,9 +23,9 @@
 import start
 
 interfaces={
-	'gtk': 'GTK',
 	'qt4': 'Qt4',
 	'qt3': 'Qt3',
+	'gtk': 'GTK',
 }
 
 def import_interface(interface):	
@@ -73,14 +73,14 @@
 #		elif 'qt3' in interfaces and env.desktop == 'kde':
 #			chosen_interface = 'qt3'
 
-		if 'gtk' in interfaces:
-			chosen_interface = 'gtk'
+		if 'qt4' in interfaces:
+			chosen_interface = 'qt4'
 
 		# try qt* for non-kde:
-		elif 'qt4' in interfaces:
-			chosen_interface = 'qt4'
 		elif 'qt3' in interfaces:
 			chosen_interface = 'qt3'
+		elif 'gtk' in interfaces:
+			chosen_interface = 'gtk'
 
 		# interfaces is empty
 		else:
diff -Naur fusion-icon-0.1.0_20080711.orig/FusionIcon/interface_qt4/main.py fusion-icon-0.1.0_20080711/FusionIcon/interface_qt4/main.py
--- fusion-icon-0.1.0_20080711.orig/FusionIcon/interface_qt4/main.py	2008-07-11 23:16:10.000000000 +0300
+++ fusion-icon-0.1.0_20080711/FusionIcon/interface_qt4/main.py	2008-07-11 23:18:00.000000000 +0300
@@ -18,6 +18,7 @@
 
 import sys, os, time
 from PyQt4 import QtGui, QtCore
+from FusionIcon.execute import run
 from FusionIcon.start import wms, apps, options, decorators, init
 
 class Build(QtGui.QApplication):
