diff -Naur plugins-unsupported.orig/configure.ac plugins-unsupported/configure.ac
--- plugins-unsupported.orig/configure.ac	2007-08-08 01:26:04.000000000 +0300
+++ plugins-unsupported/configure.ac	2007-08-09 23:15:35.000000000 +0300
@@ -124,32 +124,6 @@
 AC_SUBST(GL_CFLAGS)
 AC_SUBST(GL_LIBS)
 
-AC_ARG_ENABLE(schemas,
-  [  --enable-schemas         Build gconf schemas],
-  [use_schemas=$enableval], [use_schemas=no])
-
-if test x"$use_schemas" = x"yes"; then
-  AC_PATH_PROG(GCONFTOOL, gconftool-2, no)
-  PKG_CHECK_MODULES(SCHEMAS, compiz-gconf, [use_schemas=yes], [use_schemas=no])
-  if test x"$GCONFTOOL" = xno; then
-    AC_MSG_ERROR([gconftool-2 executable not found in your path - should be installed with GConf])
-  fi
-  AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
-  if test x"$PKG_CONFIG" = xno ; then
-    AC_MSG_ERROR([pkg-config executable not found in your path, unable to determine schema directory])
-  fi
-  xsltdir=`$PKG_CONFIG --variable=xsltdir compiz-gconf`
-  AC_SUBST(xsltdir)
-fi
-
-AM_GCONF_SOURCE_2
-
-
-AM_CONDITIONAL(USE_SCHEMAS, test "x$use_schemas" = "xyes")
-if test "$use_schemas" = yes; then
-  AC_DEFINE(USE_SCHEMAS, 1, [Build gconf schemas])
-fi
-
 AC_OUTPUT([
 Makefile
 images/Makefile
diff -Naur plugins-unsupported.orig/metadata/Makefile.am plugins-unsupported/metadata/Makefile.am
--- plugins-unsupported.orig/metadata/Makefile.am	2007-08-08 01:26:04.000000000 +0300
+++ plugins-unsupported/metadata/Makefile.am	2007-08-09 23:15:50.000000000 +0300
@@ -9,22 +9,6 @@
 
 @INTLTOOL_XML_RULE@
 
-if USE_SCHEMAS
-schemadir   = $(GCONF_SCHEMA_FILE_DIR)
-schema_files = $(patsubst %.xml.in,compiz-%.schemas,$(xml_in_files))
-schema_DATA  = $(schema_files)
-
-%.schemas: $(xml_files)
-	xsltproc -o $@ $(xsltdir)/schemas.xslt $(subst compiz-,,$*).xml;
-
-if GCONF_SCHEMAS_INSTALL
-install-data-local:
-	if test -z "$(DESTDIR)" ; then \
-               GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(schema_DATA); \
-       fi
-endif
-endif
-
 EXTRA_DIST =		\
 	$(xml_files)    \
 	$(xml_in_files) \
