--- hardinfo.desktop.orig	2008-10-06 21:05:18.000000000 +0300
+++ hardinfo.desktop	2008-10-06 21:07:21.000000000 +0300
@@ -1,7 +1,9 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=System Profiler and Benchmark
-Name[pt_BR]=Informações e Testes do Sistema
+Name=Hardinfo
+GenericName=System Profiler and Benchmark
+GenericName[pt_BR]=Informações e Testes do Sistema
+GenericName[tr]=Sistem Bilgileri Değerlendiricisi
 Exec=hardinfo
 Icon=/usr/share/hardinfo/pixmaps/logo.png
 Terminal=false
