diff -Naur k9copy-2.3.5-Source.orig/k9copy_assistant.desktop k9copy-2.3.5-Source/k9copy_assistant.desktop
--- k9copy-2.3.5-Source.orig/k9copy_assistant.desktop	2010-06-12 13:16:16.385338929 +0300
+++ k9copy-2.3.5-Source/k9copy_assistant.desktop	2010-06-12 13:17:06.017189227 +0300
@@ -3,12 +3,14 @@
 Encoding=UTF-8
 Type=Application
 Name=k9copy assistant
+Name[tr]=k9copy sihirbazı
 Exec=k9copy --assistant -caption "%c" %i %m  %u
 Comment=DVD9 to DVD5 backup tool
 Comment[fr]=Backup de DVD9 en DVD5
 Comment[tr]=DVD9 DVD5 yedekleme aracı
 Comment[ru]=Инструмент для копирования с DVD9 на DVD5
 GenericName=Video DVD Backup
+GenericName[tr]=DVD Video Yedekleyici
 GenericName[fr]=Backup de DVD Video
 GenericName[tr]=DVD Vidyo Yedekleme Aracı
 GenericName[ru]=Копирование с видео-DVD
diff -Naur k9copy-2.3.5-Source.orig/k9copy_assistant_open.desktop k9copy-2.3.5-Source/k9copy_assistant_open.desktop
--- k9copy-2.3.5-Source.orig/k9copy_assistant_open.desktop	2010-06-12 13:16:16.387338276 +0300
+++ k9copy-2.3.5-Source/k9copy_assistant_open.desktop	2010-06-12 13:18:04.138189679 +0300
@@ -2,6 +2,7 @@
 Exec=k9copy --input %f --assistant
 Icon=k9copy
 Name=Copy with k9copy (wizard)
+Name[tr]=k9copy ile kopyala (sihirbaz)
 Name[fr]=Copier avec k9copy (wizard)
 Name[de]=Kopieren mit k9copy (wizard)
 Name[tr]=k9copy ile kopyala (wizard)
diff -Naur k9copy-2.3.5-Source.orig/k9copy.desktop k9copy-2.3.5-Source/k9copy.desktop
--- k9copy-2.3.5-Source.orig/k9copy.desktop	2010-06-12 13:16:16.386339475 +0300
+++ k9copy-2.3.5-Source/k9copy.desktop	2010-06-12 13:17:18.441189593 +0300
@@ -9,6 +9,7 @@
 Comment[tr]=DVD9 DVD5 yedekleme aracı
 Comment[ru]=Инструмент для копирования с DVD9 на DVD5
 GenericName=Video DVD Backup
+GenericName[tr]=DVD Video Yedekleyici
 GenericName[fr]=Backup de DVD Video
 GenericName[tr]=DVD Vidyo Yedekleme Aracı
 GenericName[ru]=Копирование с видео-DVD
