diff -Naur k9copy-2.3.2-Source.orig/src/import/k9avidecode.cpp k9copy-2.3.2-Source/src/import/k9avidecode.cpp
--- k9copy-2.3.2-Source.orig/src/import/k9avidecode.cpp	2009-05-29 00:48:40.000000000 +0300
+++ k9copy-2.3.2-Source/src/import/k9avidecode.cpp	2009-06-13 03:21:44.000000000 +0300
@@ -9,6 +9,7 @@
 // Copyright: See COPYING file that comes with this distribution
 //
 //
+#include <stdlib.h>
 #include "k9avidecode.h"
 #include <avcodec.h>
 #include <qimage.h>
