--- configure.orig	2006-11-13 19:14:20.000000000 +0200
+++ configure	2007-04-30 21:22:14.000000000 +0300
@@ -3379,8 +3379,8 @@
 
 # Shared library section
 #-------------------------------------------------------------------------------
-SHLIB_LD=:
-SHLIB_SUFFIX=".so"
+SHLIB_LD="cc -shared -Wl,-soname,libcfitsio.so.0"
+SHLIB_SUFFIX=".so.0"
 lhea_shlib_cflags=
 case $EXT in
   cygwin)
@@ -3418,7 +3418,7 @@
 # Darwin uses gcc (=cc), but needs different flags (see above)
 # if test "x$GCC" = xyes; then
 if test "x$GCC" = xyes && test "x$EXT" != xdarwin && test "x$EXT" != xcygwin; then
-  SHLIB_LD="$CC -shared"
+  SHLIB_LD="cc -shared -Wl,-soname,libcfitsio.so.0"
   lhea_shlib_cflags='-fPIC'
 fi
 if test "x$lhea_shlib_cflags" != x; then
