--- usr/share/applications/frostwire.desktop	2006-02-20 00:18:27.000000000 +0200
+++ frostwire-yeni.desktop	2006-06-30 11:10:07.000000000 +0300
@@ -1,7 +1,10 @@
 [Desktop Entry]
 Name=FrostWire
+Name[tr]=FrostWire
 GenericName=P2P Gnutella client
+GenericName[tr]=P2P Gnutella istemcisi
 Comment=Search and share all kinds of files on the Gnutella network
+Comment[tr]=Gnutella ağında her çesit dosya tipini ara ve paylaş
 Exec=frostwire
 Icon=frostwire.png
 Terminal=false
