--- src/Makefile	2006-04-16 19:12:56.000000000 +0300
+++ Makefile-yeni	2006-06-15 11:14:45.000000000 +0300
@@ -31,7 +31,7 @@
                 -DICEWMBGEXE='"icewmbg$(EXEEXT)"' \
                 -DICESMEXE='"icewm-session$(EXEEXT)"' \
                 -DICEHELPEXE='"icehelp$(EXEEXT)"' \
-                -DICEHELPIDX='"$(DOCDIR)/icewm-$(VERSION)/icewm.html"'
+                -DICEHELPIDX='"$(DOCDIR)/icewm.html"'
 
 CXXFLAGS =      -fpermissive -Wall -Wpointer-arith -Wconversion -Wwrite-strings -Winline -Woverloaded-virtual -W -fno-exceptions -fno-rtti -g -O2 $(DEBUG) $(DEFS) \
 	         -I/usr/include -I/usr/include/freetype2   # `fc-config --cflags`
@@ -39,7 +39,7 @@
 LIBS =          
 
 CORE_LIBS =     -lXinerama  -lSM -lICE  -lX11  -L/usr/X11R6/lib  -lXrandr -lXrender -lXext -lXft -lX11 -lfreetype -lz -lfontconfig -lXrender -lX11 -ldl # `fc-config --libs`
-IMAGE_LIBS =    -L/usr/lib -lImlib -ljpeg -ltiff -lungif -lpng10 -lz -lm -lXext -L/usr/X11R6/lib -lSM -lICE -lXext -lX11
+IMAGE_LIBS =    -L/usr/lib -lImlib -ljpeg -ltiff -lgif -lpng10 -lz -lm -lXext -L/usr/X11R6/lib -lSM -lICE -lXext -lX11
 AUDIO_LIBS =    
 GNOME1_LIBS =    
 GNOME2_LIBS =    
