--- setup.cfg.old	2008-02-13 00:07:12.594722128 +0000
+++ setup.cfg	2008-02-13 00:14:45.830819768 +0000
@@ -1,12 +1,9 @@
 [bdist_rpm]
 doc_files = LICENSE,docs
 
-### Enable the following two items for unattended install.
-### The scripts location is mandatory. Change it so something sensible.
-
-#[install-options]
-#unattended=1
+[install-options]
+unattended=1
 
 [install]
 optimize=1
-#install-scripts=/foo/bar
+install-scripts=/usr/bin
