--- common/general/circular_iterator.h	2004-04-09 02:54:10.000000000 +0300
+++ common/general/circular_iterator.h	2006-05-06 16:59:30.000000000 +0300
@@ -137,7 +137,7 @@
 inline circular_iterator<It>& circular_iterator<It>::operator += ( circular_iterator<It>::difference_type n )
 {
 	n %= last - first; // put it in range
-	circular_iterator::difference_type d = last - cur;
+	typename circular_iterator::difference_type d = last - cur;
 	if ( d > n ) {
 		cur += n;
 		return *this;

--- common/network/message.h	2003-11-26 21:07:57.000000000 +0200
+++ common/network/message.h	2006-05-06 17:00:42.000000000 +0300
@@ -129,7 +129,7 @@
 }
 
 template <typename T>
-inline const T Message::arg( size_t idx ) const
+inline const T Message::arg( unsigned idx ) const
 {
 	QString rep = arg( idx );
 	return to<T>( rep );
