--- src/audacious/main.c~	2007-04-05 08:44:05.000000000 +0200
+++ src/audacious/main.c	2007-04-17 22:51:33.000000000 +0200
@@ -291,7 +291,7 @@
     {"playlist_shaded", &cfg.playlist_shaded, TRUE},
     {"playlist_visible", &cfg.playlist_visible, TRUE},
     {"use_fontsets", &cfg.use_fontsets, TRUE},
-    {"mainwin_use_xfont", &cfg.mainwin_use_xfont, FALSE},
+    {"mainwin_use_xfont", &cfg.mainwin_use_xfont, TRUE},
     {"equalizer_visible", &cfg.equalizer_visible, TRUE},
     {"equalizer_active", &cfg.equalizer_active, TRUE},
     {"equalizer_shaded", &cfg.equalizer_shaded, TRUE},
