--- launchy-2.1.2/platforms/unix/platform_unix.cpp.orig	2009-06-18 21:32:19.000000000 +0300
+++ launchy-2.1.2/platforms/unix/platform_unix.cpp	2009-06-18 21:33:25.000000000 +0300
@@ -52,13 +52,14 @@
     QList<Directory> list;
     const char *dirs[] = { "/usr/share/applications/",
 			   "/usr/local/share/applications/",
-			   "/usr/share/gdm/applications/",
+			   "/usr/kde/3.5/share/applications/kde",
+			   "/usr/kde/4/share/applications/kde4",
 			   "/usr/share/applications/kde/",
 			   "~/.local/share/applications/"};
     QStringList l;
     l << "*.desktop";
     
-    for(int i = 0; i < 5; i++)
+    for(int i = 0; i < 6; i++)
 	list.append(Directory(dirs[i],l,false,false,100));
 
     list.append(Directory("~",QStringList(),true,false,0));
