Index: translations/qwit_en_US.ts
===================================================================
--- translations/qwit_en_US.ts	(revision 0)
+++ translations/qwit_en_US.ts	(revision 38)
@@ -0,0 +1,313 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE TS><TS version="1.1" language="en">
+<context>
+    <name>LogsDialog</name>
+    <message>
+        <location filename="../src/LogsDialog.ui" line="13"/>
+        <source>Logs</source>
+        <translation type="unfinished"></translation>
+    </message>
+</context>
+<context>
+    <name>MainWindow</name>
+    <message>
+        <location filename="../src/MainWindow.cpp" line="167"/>
+        <source>&amp;Show / Hide</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.cpp" line="169"/>
+        <source>&amp;Quit</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.cpp" line="545"/>
+        <source>Qwit updates</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="13"/>
+        <source>Qwit</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="39"/>
+        <source>What are you doing?</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="205"/>
+        <source>Left characters count - please, do not exceed this number</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="214"/>
+        <source>140</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="236"/>
+        <source>Home</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="239"/>
+        <source>Your and your friends&apos; timeline</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="244"/>
+        <source>Public</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="247"/>
+        <source>Public timeline</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="252"/>
+        <source>Replies</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="255"/>
+        <source>Replies to you</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="260"/>
+        <source>Custom</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="263"/>
+        <source>Custom user timeline</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="273"/>
+        <source>Username:</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="324"/>
+        <source>Refresh</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="334"/>
+        <source>Ctrl+R</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="353"/>
+        <source>Options</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="396"/>
+        <source>Quit</source>
+        <translation type="unfinished"></translation>
+    </message>
+</context>
+<context>
+    <name>OptionsDialog</name>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="13"/>
+        <source>Options</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="25"/>
+        <source>General</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="63"/>
+        <source>Update interval:</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="76"/>
+        <source>Number of messages per page:</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="115"/>
+        <source>Connection</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="123"/>
+        <source>use proxy</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="130"/>
+        <source>Proxy address:</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="143"/>
+        <source>Proxy port:</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="156"/>
+        <source>Proxy username:</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="169"/>
+        <source>Proxy password:</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="186"/>
+        <source>save proxy password</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="209"/>
+        <source>Authentication</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="217"/>
+        <source>Username or email:</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="230"/>
+        <source>Password:</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="247"/>
+        <source>save password</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="292"/>
+        <source>Save options</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="299"/>
+        <source>Reset options</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="43"/>
+        <source>Service base URL:</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="53"/>
+        <source>Service API URL:</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="89"/>
+        <source>Notify about updates</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="33"/>
+        <source>Greeting message:</source>
+        <translation type="unfinished"></translation>
+    </message>
+</context>
+<context>
+    <name>Twitter</name>
+    <message>
+        <location filename="../src/Twitter.cpp" line="122"/>
+        <source>Error sending status: %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="125"/>
+        <source>Status sent: %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="131"/>
+        <source>Error while updating timeline: %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="135"/>
+        <source>Timeline updated: %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="117"/>
+        <source>Updating timeline: %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="88"/>
+        <source>Sending status: %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+</context>
+<context>
+    <name>TwitterWidget</name>
+    <message numerus="yes">
+        <location filename="../src/TwitterWidget.cpp" line="227"/>
+        <source>about %n second(s) ago</source>
+        <translation>
+            <numerusform>about %n second ago</numerusform>
+            <numerusform>about %n seconds ago</numerusform>
+        </translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../src/TwitterWidget.cpp" line="229"/>
+        <source>about %n minute(s) ago</source>
+        <translation>
+            <numerusform>about %n minute ago</numerusform>
+            <numerusform>about %n minutes ago</numerusform>
+        </translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../src/TwitterWidget.cpp" line="231"/>
+        <source>about %n hour(s) ago</source>
+        <translation>
+            <numerusform>about %n hour ago</numerusform>
+            <numerusform>about %n hours ago</numerusform>
+        </translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../src/TwitterWidget.cpp" line="233"/>
+        <source>about %n day(s) ago</source>
+        <translation>
+            <numerusform>about %n day ago</numerusform>
+            <numerusform>about %n days ago</numerusform>
+        </translation>
+    </message>
+    <message>
+        <location filename="../src/TwitterWidget.cpp" line="226"/>
+        <source>Just now</source>
+        <translation type="unfinished"></translation>
+    </message>
+</context>
+<context>
+    <name>UserpicsDownloader</name>
+    <message>
+        <location filename="../src/UserpicsDownloader.cpp" line="43"/>
+        <source>Downloading: %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/UserpicsDownloader.cpp" line="57"/>
+        <source>Error while downloading userpic: %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+    <message>
+        <location filename="../src/UserpicsDownloader.cpp" line="62"/>
+        <source>Downloaded: %1</source>
+        <translation type="unfinished"></translation>
+    </message>
+</context>
+</TS>
Index: translations/qwit_tr_TR.ts
===================================================================
--- translations/qwit_tr_TR.ts	(revision 0)
+++ translations/qwit_tr_TR.ts	(revision 38)
@@ -0,0 +1,313 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE TS><TS version="1.1" language="tr_TR">
+<context>
+    <name>LogsDialog</name>
+    <message>
+        <location filename="../src/LogsDialog.ui" line="13"/>
+        <source>Logs</source>
+        <translation>Kayıtlar</translation>
+    </message>
+</context>
+<context>
+    <name>MainWindow</name>
+    <message>
+        <location filename="../src/MainWindow.cpp" line="167"/>
+        <source>&amp;Show / Hide</source>
+        <translation>&amp;Göster / Gizle</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.cpp" line="169"/>
+        <source>&amp;Quit</source>
+        <translation>Çı&amp;kış</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="13"/>
+        <source>Qwit</source>
+        <translation>Qwit</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="39"/>
+        <source>What are you doing?</source>
+        <translation>Ne yapıyorsun?</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="205"/>
+        <source>Left characters count - please, do not exceed this number</source>
+        <translation>Kalan karakter sayısı - lütfen bu sayıyı geçmeyin</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="214"/>
+        <source>140</source>
+        <translation>140</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="236"/>
+        <source>Home</source>
+        <translation>Ev</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="239"/>
+        <source>Your and your friends&apos; timeline</source>
+        <translation>Sizin ve arkadaşlarınızın yazdıkları</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="244"/>
+        <source>Public</source>
+        <translation>Herkes</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="247"/>
+        <source>Public timeline</source>
+        <translation>Herkesin yazdıkarı</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="252"/>
+        <source>Replies</source>
+        <translation>Yanıtlar</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="255"/>
+        <source>Replies to you</source>
+        <translation>Size verilen yanıtlar</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="260"/>
+        <source>Custom</source>
+        <translation>Özel</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="263"/>
+        <source>Custom user timeline</source>
+        <translation>Özel bir kullanıcının yazdıkları</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="273"/>
+        <source>Username:</source>
+        <translation>Kullanıcı adı:</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="324"/>
+        <source>Refresh</source>
+        <translation>Yenile</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="334"/>
+        <source>Ctrl+R</source>
+        <translation>Ctrl+R</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="353"/>
+        <source>Options</source>
+        <translation>Seçenekler</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="396"/>
+        <source>Quit</source>
+        <translation>Çıkış</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.cpp" line="545"/>
+        <source>Qwit updates</source>
+        <translation>Qwit güncellemeleri</translation>
+    </message>
+</context>
+<context>
+    <name>OptionsDialog</name>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="13"/>
+        <source>Options</source>
+        <translation>Seçenekler</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="25"/>
+        <source>General</source>
+        <translation>Genel</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="63"/>
+        <source>Update interval:</source>
+        <translation>Güncelleme sıklığı:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="76"/>
+        <source>Number of messages per page:</source>
+        <translation>Sayfa başına düşen ileti sayısı:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="115"/>
+        <source>Connection</source>
+        <translation>Bağlantı</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="123"/>
+        <source>use proxy</source>
+        <translation>proxy kullan</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="130"/>
+        <source>Proxy address:</source>
+        <translation>Proxy adresi:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="143"/>
+        <source>Proxy port:</source>
+        <translation>Proxy portu:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="156"/>
+        <source>Proxy username:</source>
+        <translation>Proxy kullanıcı adı:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="169"/>
+        <source>Proxy password:</source>
+        <translation>Proxy parolası:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="186"/>
+        <source>save proxy password</source>
+        <translation>proxy parolasını kaydet</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="209"/>
+        <source>Authentication</source>
+        <translation>Giriş</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="217"/>
+        <source>Username or email:</source>
+        <translation>Kullanıcı adı veya e-posta:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="230"/>
+        <source>Password:</source>
+        <translation>Parola:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="247"/>
+        <source>save password</source>
+        <translation>parolayı kaydet</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="292"/>
+        <source>Save options</source>
+        <translation>Seçenekleri kaydet</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="299"/>
+        <source>Reset options</source>
+        <translation>Seçenekleri sıfırla</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="33"/>
+        <source>Greeting message:</source>
+        <translation>Karşılama iletisi:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="43"/>
+        <source>Service base URL:</source>
+        <translation>Servisin taban URL&apos;si:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="53"/>
+        <source>Service API URL:</source>
+        <translation>Servisin API URL&apos;si:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="89"/>
+        <source>Notify about updates</source>
+        <translation>Güncellemeleri bildir</translation>
+    </message>
+</context>
+<context>
+    <name>Twitter</name>
+    <message>
+        <location filename="../src/Twitter.cpp" line="122"/>
+        <source>Error sending status: %1</source>
+        <translation>Durum gönderilemedi: %1</translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="125"/>
+        <source>Status sent: %1</source>
+        <translation>Durum gönderildi: %1</translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="131"/>
+        <source>Error while updating timeline: %1</source>
+        <translation>Yazılar güncellenemedi: %1</translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="135"/>
+        <source>Timeline updated: %1</source>
+        <translation>Yazılar güncellendi: %1</translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="88"/>
+        <source>Sending status: %1</source>
+        <translation type="unfinished">Durum gönderiliyor: %1</translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="117"/>
+        <source>Updating timeline: %1</source>
+        <translation type="unfinished">Yazılar güncelleniyor: %1</translation>
+    </message>
+</context>
+<context>
+    <name>TwitterWidget</name>
+    <message>
+        <location filename="../src/TwitterWidget.cpp" line="226"/>
+        <source>Just now</source>
+        <translation>Şimdi</translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../src/TwitterWidget.cpp" line="227"/>
+        <source>about %n second(s) ago</source>
+        <translation>
+            <numerusform>yaklaşık %n saniye önce</numerusform>
+            <numerusform></numerusform>
+        </translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../src/TwitterWidget.cpp" line="229"/>
+        <source>about %n minute(s) ago</source>
+        <translation>
+            <numerusform>yaklaşık %n dakika önce</numerusform>
+            <numerusform></numerusform>
+        </translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../src/TwitterWidget.cpp" line="231"/>
+        <source>about %n hour(s) ago</source>
+        <translation>
+            <numerusform>yaklaşık %n saat önce</numerusform>
+            <numerusform></numerusform>
+        </translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../src/TwitterWidget.cpp" line="233"/>
+        <source>about %n day(s) ago</source>
+        <translation>
+            <numerusform>yaklaşık %n gün önce</numerusform>
+            <numerusform></numerusform>
+        </translation>
+    </message>
+</context>
+<context>
+    <name>UserpicsDownloader</name>
+    <message>
+        <location filename="../src/UserpicsDownloader.cpp" line="43"/>
+        <source>Downloading: %1</source>
+        <translation>İndirilen: %1</translation>
+    </message>
+    <message>
+        <location filename="../src/UserpicsDownloader.cpp" line="57"/>
+        <source>Error while downloading userpic: %1</source>
+        <translation>Kullanıcı resmi indirilemedi: %1</translation>
+    </message>
+    <message>
+        <location filename="../src/UserpicsDownloader.cpp" line="62"/>
+        <source>Downloaded: %1</source>
+        <translation>İndirildi: %1</translation>
+    </message>
+</context>
+</TS>

Property changes on: translations/qwit_tr_TR.ts
___________________________________________________________________
Added: svn:mergeinfo

Index: translations/qwit_ru_RU.ts
===================================================================
--- translations/qwit_ru_RU.ts	(revision 0)
+++ translations/qwit_ru_RU.ts	(revision 38)
@@ -0,0 +1,317 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE TS><TS version="1.1" language="ru">
+<context>
+    <name>LogsDialog</name>
+    <message>
+        <location filename="../src/LogsDialog.ui" line="13"/>
+        <source>Logs</source>
+        <translation>Логи</translation>
+    </message>
+</context>
+<context>
+    <name>MainWindow</name>
+    <message>
+        <location filename="../src/MainWindow.cpp" line="167"/>
+        <source>&amp;Show / Hide</source>
+        <translation>&amp;Показать / Скрыть</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.cpp" line="169"/>
+        <source>&amp;Quit</source>
+        <translation>Выход</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.cpp" line="545"/>
+        <source>Qwit updates</source>
+        <translation>Обновления Qwit</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="13"/>
+        <source>Qwit</source>
+        <translation>Qwit</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="39"/>
+        <source>What are you doing?</source>
+        <translation>Что вы сейчас делаете?</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="205"/>
+        <source>Left characters count - please, do not exceed this number</source>
+        <translation>Количество оставшихся символов - пожалуйста, не превышайте это число</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="214"/>
+        <source>140</source>
+        <translation>140</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="236"/>
+        <source>Home</source>
+        <translation>Главная</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="239"/>
+        <source>Your and your friends&apos; timeline</source>
+        <translation>Сообщения от вас и ваших друзей</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="244"/>
+        <source>Public</source>
+        <translation>Все</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="247"/>
+        <source>Public timeline</source>
+        <translation>Сообщения ото всех</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="252"/>
+        <source>Replies</source>
+        <translation>Ответы</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="255"/>
+        <source>Replies to you</source>
+        <translation>Ответы вам</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="260"/>
+        <source>Custom</source>
+        <translation>Отдельно</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="263"/>
+        <source>Custom user timeline</source>
+        <translation>Сообщения отдельного пользователя</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="273"/>
+        <source>Username:</source>
+        <translation>Имя пользователя:</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="324"/>
+        <source>Refresh</source>
+        <translation>Обновить</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="334"/>
+        <source>Ctrl+R</source>
+        <translation>Ctrl+R</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="353"/>
+        <source>Options</source>
+        <translation>Настройки</translation>
+    </message>
+    <message>
+        <location filename="../src/MainWindow.ui" line="396"/>
+        <source>Quit</source>
+        <translation>Выход</translation>
+    </message>
+</context>
+<context>
+    <name>OptionsDialog</name>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="13"/>
+        <source>Options</source>
+        <translation>Настройки</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="25"/>
+        <source>General</source>
+        <translation>Общие</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="63"/>
+        <source>Update interval:</source>
+        <translation>Интервал обновлений:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="76"/>
+        <source>Number of messages per page:</source>
+        <translation>Количество сообщений на странице:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="115"/>
+        <source>Connection</source>
+        <translation>Соединение</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="123"/>
+        <source>use proxy</source>
+        <translation>использовать прокси</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="130"/>
+        <source>Proxy address:</source>
+        <translation>Адрес прокси-сервера:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="143"/>
+        <source>Proxy port:</source>
+        <translation>Порт прокси-сервера:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="156"/>
+        <source>Proxy username:</source>
+        <translation>Имя пользователя:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="169"/>
+        <source>Proxy password:</source>
+        <translation>Пароль:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="186"/>
+        <source>save proxy password</source>
+        <translation>сохранять пароль для прокси</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="209"/>
+        <source>Authentication</source>
+        <translation>Аутентификация</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="217"/>
+        <source>Username or email:</source>
+        <translation>Имя пользователя или email:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="230"/>
+        <source>Password:</source>
+        <translation>Пароль:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="247"/>
+        <source>save password</source>
+        <translation>сохранять пароль</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="292"/>
+        <source>Save options</source>
+        <translation>Сохранить настройки</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="299"/>
+        <source>Reset options</source>
+        <translation>Сбросить настройки</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="43"/>
+        <source>Service base URL:</source>
+        <translation>Базовый адрес сервиса:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="53"/>
+        <source>Service API URL:</source>
+        <translation>Адрес API сервиса:</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="89"/>
+        <source>Notify about updates</source>
+        <translation>Информировать об обновлениях</translation>
+    </message>
+    <message>
+        <location filename="../src/OptionsDialog.ui" line="33"/>
+        <source>Greeting message:</source>
+        <translation>Приветствие:</translation>
+    </message>
+</context>
+<context>
+    <name>Twitter</name>
+    <message>
+        <location filename="../src/Twitter.cpp" line="122"/>
+        <source>Error sending status: %1</source>
+        <translation>Ошибка при отправлении статуса: %1</translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="125"/>
+        <source>Status sent: %1</source>
+        <translation>Статус отправлен: %1</translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="131"/>
+        <source>Error while updating timeline: %1</source>
+        <translation>Ошибка обновления: %1</translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="135"/>
+        <source>Timeline updated: %1</source>
+        <translation>Сообщения обновлены: %1</translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="117"/>
+        <source>Updating timeline: %1</source>
+        <translation>Обновление сообщения: %1</translation>
+    </message>
+    <message>
+        <location filename="../src/Twitter.cpp" line="88"/>
+        <source>Sending status: %1</source>
+        <translation>Отправка статуса: %1</translation>
+    </message>
+</context>
+<context>
+    <name>TwitterWidget</name>
+    <message numerus="yes">
+        <location filename="../src/TwitterWidget.cpp" line="227"/>
+        <source>about %n second(s) ago</source>
+        <translation>
+            <numerusform>примерно %n секунду назад</numerusform>
+            <numerusform>примерно %n секунды назад</numerusform>
+            <numerusform>примерно %n секунд назад</numerusform>
+        </translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../src/TwitterWidget.cpp" line="229"/>
+        <source>about %n minute(s) ago</source>
+        <translation>
+            <numerusform>примерно %n минуту назад</numerusform>
+            <numerusform>примерно %n минуты назад</numerusform>
+            <numerusform>примерно %n минут назад</numerusform>
+        </translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../src/TwitterWidget.cpp" line="231"/>
+        <source>about %n hour(s) ago</source>
+        <translation>
+            <numerusform>примерно %n час назад</numerusform>
+            <numerusform>примерно %n часа назад</numerusform>
+            <numerusform>примерно %n часов назад</numerusform>
+        </translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../src/TwitterWidget.cpp" line="233"/>
+        <source>about %n day(s) ago</source>
+        <translation>
+            <numerusform>примерно %n день назад</numerusform>
+            <numerusform>примерно %n дня назад</numerusform>
+            <numerusform>примерно %n дней назад</numerusform>
+        </translation>
+    </message>
+    <message>
+        <location filename="../src/TwitterWidget.cpp" line="226"/>
+        <source>Just now</source>
+        <translation>Только что</translation>
+    </message>
+</context>
+<context>
+    <name>UserpicsDownloader</name>
+    <message>
+        <location filename="../src/UserpicsDownloader.cpp" line="43"/>
+        <source>Downloading: %1</source>
+        <translation>Загрузка: %1</translation>
+    </message>
+    <message>
+        <location filename="../src/UserpicsDownloader.cpp" line="57"/>
+        <source>Error while downloading userpic: %1</source>
+        <translation>Ошибка при загрузке изображения пользователя: %1</translation>
+    </message>
+    <message>
+        <location filename="../src/UserpicsDownloader.cpp" line="62"/>
+        <source>Downloaded: %1</source>
+        <translation>Загружено: %1</translation>
+    </message>
+</context>
+</TS>
Index: qwit.pro
===================================================================
--- qwit.pro	(revision 21)
+++ qwit.pro	(revision 38)
@@ -29,8 +29,12 @@
            src/Twitter.cpp \
            src/TwitterWidget.cpp \
            src/UserpicsDownloader.cpp
+TRANSLATIONS += \
+           translations/qwit_tr_TR.ts \
+           translations/qwit_ru_RU.ts \
+           translations/qwit_en_US.ts
 QT += network xml
 RESOURCES = qwit.qrc
 target.path = /usr/bin
 INSTALLS = target
 CONFIG += release
Index: src/Twitter.cpp
===================================================================
--- src/Twitter.cpp	(revision 21)
+++ src/Twitter.cpp	(revision 38)
@@ -37,6 +37,22 @@
 	connect(&timelineHttp, SIGNAL(done(bool)), this, SLOT(timelineHttpDone(bool)));
 }
 
+void Twitter::setServiceBaseURL(const QString &url) {
+	serviceBaseURL = url;
+}
+
+void Twitter::setServiceAPIURL(const QString &url) {
+	serviceAPIURL = url;
+}
+
+QString Twitter::getServiceBaseURL() {
+	return serviceBaseURL;
+}
+
+QString Twitter::getServiceAPIURL() {
+	return serviceAPIURL;
+}
+
 void Twitter::useProxy(const QString &address, int port, const QString &username, const QString &password) {
 	proxyAddress = address;
 	proxyPort = port;
@@ -48,8 +64,8 @@
 	proxyAddress = "";
 }
 
-void Twitter::sendStatus(QString username, QString password, QString status) {
-	QUrl url(STATUS_UPDATE_URL);
+void Twitter::sendStatus(QString username, QString password, QString status, QString replyID) {
+	QUrl url(serviceAPIURL + STATUS_UPDATE_URL);
 
 	QHttpRequestHeader header;
 	header.setRequest("POST", url.path());
@@ -66,14 +82,21 @@
 	
 	QByteArray data = "status=";
 	data += QUrl::toPercentEncoding(status);
+        if (replyID != "") {
+                data += "&in_reply_to_status_id=";
+                data += replyID;
+        }
 	data += "&source=qwit";
 	statusHttp.request(header, data);
 
-	emit stateChanged("Sending status...");
+	emit stateChanged(tr("Sending status: %1").arg(serviceAPIURL + STATUS_UPDATE_URL));
 }
 
 void Twitter::update(QString username, QString password, int lastStatusId, int type, int count) {
-	if (urls[type] == "") return;
+	if (urls[type] == "") {
+		cerr << "No url defined" << endl;
+		return;
+	}
 	
 	if (timelineHttp.state() != QHttp::Unconnected) {
 		timelineHttp.abort();
@@ -81,7 +104,7 @@
 	
 	currentType = type;
 	
-	QUrl url(urls[type]);
+	QUrl url(serviceAPIURL + urls[type]);
 	
 	if (proxyAddress != "") {
 		timelineHttp.setProxy(proxyAddress, proxyPort, proxyUsername, proxyPassword);
@@ -95,25 +118,25 @@
 	buffer.open(QIODevice::WriteOnly);
 	timelineHttp.get(url.path() + "?count=" + QString::number(count) + (lastStatusId ? "&since_id=" + QString::number(lastStatusId) : ""), &buffer);
 	
-	emit stateChanged("Updating timeline...");
+	emit stateChanged(tr("Updating timeline: %1").arg(serviceAPIURL + urls[type]));
 }
 
 void Twitter::statusHttpDone(bool error) {
 	if (error) {
-		emit stateChanged("Error sending status: " + statusHttp.errorString());
+		emit stateChanged(tr("Error sending status: %1").arg(statusHttp.errorString()));
 		return;
 	}
-	emit stateChanged("Status send at " + QDateTime::currentDateTime().toString("hh:mm:ss"));
+	emit stateChanged(tr("Status sent: %1").arg(QDateTime::currentDateTime().toString("hh:mm:ss")));
 	emit statusUpdated();
 }
 
 void Twitter::timelineHttpDone(bool error) {
 	if (error) {
-		emit stateChanged("Error while updating timeline: " + timelineHttp.errorString());
+		emit stateChanged(tr("Error while updating timeline: %1").arg(timelineHttp.errorString()));
 		return;
 	}
 	buffer.close();
-	emit stateChanged("Timeline updated at " + QDateTime::currentDateTime().toString("hh:mm:ss"));
+	emit stateChanged(tr("Timeline updated: %1").arg(QDateTime::currentDateTime().toString("hh:mm:ss")));
 	emit updated(buffer.data(), currentType);
 }
 
Index: src/OptionsDialog.ui
===================================================================
--- src/OptionsDialog.ui	(revision 21)
+++ src/OptionsDialog.ui	(revision 38)
@@ -6,7 +6,7 @@
     <x>0</x>
     <y>0</y>
     <width>251</width>
-    <height>371</height>
+    <height>417</height>
    </rect>
   </property>
   <property name="windowTitle" >
@@ -28,6 +28,36 @@
          <item row="0" column="0" >
           <layout class="QVBoxLayout" >
            <item>
+            <widget class="QLabel" name="label_11" >
+             <property name="text" >
+              <string>Greeting message:</string>
+             </property>
+            </widget>
+           </item>
+           <item>
+            <widget class="QLineEdit" name="greetingMessageLineEdit" />
+           </item>
+           <item>
+            <widget class="QLabel" name="label_5" >
+             <property name="text" >
+              <string>Service base URL:</string>
+             </property>
+            </widget>
+           </item>
+           <item>
+            <widget class="QLineEdit" name="serviceBaseURLLineEdit" />
+           </item>
+           <item>
+            <widget class="QLabel" name="label_10" >
+             <property name="text" >
+              <string>Service API URL:</string>
+             </property>
+            </widget>
+           </item>
+           <item>
+            <widget class="QLineEdit" name="serviceAPIURLLineEdit" />
+           </item>
+           <item>
             <widget class="QLabel" name="label_3" >
              <property name="text" >
               <string>Update interval:</string>
@@ -54,11 +84,21 @@
             <widget class="QLineEdit" name="messagesPerPageLineEdit" />
            </item>
            <item>
+            <widget class="QCheckBox" name="updatesNotificationCheckBox" >
+             <property name="text" >
+              <string>Notify about updates</string>
+             </property>
+             <property name="checked" >
+              <bool>true</bool>
+             </property>
+            </widget>
+           </item>
+           <item>
             <spacer>
              <property name="orientation" >
               <enum>Qt::Vertical</enum>
              </property>
-             <property name="sizeHint" >
+             <property name="sizeHint" stdset="0" >
               <size>
                <width>20</width>
                <height>40</height>
@@ -152,7 +192,7 @@
              <property name="orientation" >
               <enum>Qt::Vertical</enum>
              </property>
-             <property name="sizeHint" >
+             <property name="sizeHint" stdset="0" >
               <size>
                <width>20</width>
                <height>40</height>
@@ -213,7 +253,7 @@
              <property name="orientation" >
               <enum>Qt::Vertical</enum>
              </property>
-             <property name="sizeHint" >
+             <property name="sizeHint" stdset="0" >
               <size>
                <width>20</width>
                <height>40</height>
@@ -236,7 +276,7 @@
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
-           <property name="sizeHint" >
+           <property name="sizeHint" stdset="0" >
             <size>
              <width>40</width>
              <height>20</height>
@@ -267,7 +307,7 @@
            <property name="orientation" >
             <enum>Qt::Horizontal</enum>
            </property>
-           <property name="sizeHint" >
+           <property name="sizeHint" stdset="0" >
             <size>
              <width>40</width>
              <height>20</height>
@@ -282,7 +322,7 @@
          <property name="orientation" >
           <enum>Qt::Vertical</enum>
          </property>
-         <property name="sizeHint" >
+         <property name="sizeHint" stdset="0" >
           <size>
            <width>20</width>
            <height>40</height>
Index: src/TwitterWidget.cpp
===================================================================
--- src/TwitterWidget.cpp	(revision 21)
+++ src/TwitterWidget.cpp	(revision 38)
@@ -19,18 +19,34 @@
 #include <QScrollBar>
 #include <QDesktopServices>
 #include <QPalette>
+#include <QChar>
 
 #include "TwitterWidget.h"
+#include "QwitException.h"
 
 #include <iostream>
 
 using namespace std;
 
+void TwitterWidgetItem::loadIcon() {
+	QPixmap pixmap(iconFileName);
+// Fix for stupid formats
+	if (pixmap.isNull()) {
+		pixmap.load(iconFileName, "JPG");
+	}
+	icon->setPixmap(pixmap);
+	icon->resize(ICON_SIZE, ICON_SIZE);
+}
+
 TwitterWidget::TwitterWidget(): QWidget() {
 	QDesktopServices::setUrlHandler("reply", this, "replyClicked");
 }
 
-QString TwitterWidget::prepare(const QString &text, const int &replyStatusId) {
+bool TwitterWidget::isUsernameChar(const QChar &c) const {
+	return ((c >= 'a') && (c <= 'z')) || ((c >= 'A') && (c <= 'Z')) || ((c >= '0') && (c <= '9')) || (c == '_');
+}
+
+QString TwitterWidget::prepare(const QString &text, const int &replyStatusId, const QString &serviceBaseURL) {
 	QString s = text;
 	s.replace(" www.", " http://www.");
 	if (s.startsWith("www.")) s = "http://" + s;
@@ -50,8 +66,24 @@
 		int i = 1;
 		while ((i < s.length()) && (QChar(s[i]).isLetterOrNumber() || (s[i] == '_'))) ++i;
 		QString username = s.mid(1, i - 1);
-		t = "@<a href=\"http://twitter.com/" + username + "/statuses/" + QString::number(replyStatusId) + "\" style=\"text-decoration:none\">" + username + "</a>" + s.mid(i);
+		t = "@<a href=\"" + serviceBaseURL + "/" + username + "/statuses/" + QString::number(replyStatusId) + "\" style=\"text-decoration:none;font-weight:bold;\">" + username + "</a>" + s.mid(i);
 	}
+	s = t;
+	t = "";
+	for (int i = 0; i < s.length(); ++i) {
+		t += s[i];
+		if ((s[i] == '@') && (!i || (s[i - 1] == ' '))) {
+			int j = i + 1;
+			while ((j < s.length()) && isUsernameChar(s[j])) {
+				++j;
+			}
+			if (j - i - 1 > 0) {
+				QString username = s.mid(i + 1, j - i - 1);
+				t += "<a href=\"" + serviceBaseURL + "/" + username + "\" style=\"text-decoration:none;font-weight:bold;\">" + username + "</a>";
+				i = j - 1;
+			}
+		}
+	}
 	return t;
 }
 	
@@ -64,7 +96,7 @@
 	items.clear();
 }
 	
-void TwitterWidget::addItem(const QString &userpic, const QString &username, const QString &status, const QDateTime &time, int messageId, int replyStatusId, int i) {
+void TwitterWidget::addItem(const QString &userpic, const QString &username, const QString &status, const QDateTime &time, int messageId, int replyStatusId, int i, const QString &serviceBaseURL) {
 	TwitterWidgetItem item = TwitterWidgetItem();
 	
 	item.time = time;
@@ -82,7 +114,7 @@
 	item.cacheIndex = i;
 	
 	item.status = new QTextBrowser(this);
-	item.status->setHtml(prepare(status, replyStatusId));
+	item.status->setHtml(prepare(status, replyStatusId, serviceBaseURL));
 	item.status->setReadOnly(true);
 	item.status->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
 	item.status->setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
@@ -93,14 +125,16 @@
 	item.status->document()->setDefaultFont(font);
 	
 	item.icon = new QLabel(this);
-	item.icon->setPixmap(QPixmap(userpic));
 	item.iconFileName = userpic;
-	
-	item.sign = new QLabel("<a href=\"http://twitter.com/" + username + "\" style=\"font-weight:bold;text-decoration:none\">" + username + "</a> - <a href=\"http://twitter.com/" + username + "/statuses/" + QString::number(messageId) + "\" style=\"font-size:70%;text-decoration:none\">" + formatDateTime(time) + "</a> <a href=\"reply://" + username + "\" style=\"text-decoration:none\"><img src=\":/images/reply.png\"/></a>", this);
+	item.loadIcon();
+	item.sign = new QLabel("<a href=\"http://twitter.com/" + username + "\" style=\"font-weight:bold;text-decoration:none\">" + username + "</a> - <a href=\"http://twitter.com/" + username + "/statuses/" + QString::number(messageId) + "\" style=\"font-size:70%;text-decoration:none\">" + formatDateTime(time) + "</a> <a href=\"reply://" + username + ":" + QString::number(messageId) + "\" style=\"text-decoration:none\"><img src=\":/images/reply.png\"/></a>", this);
 	item.sign->setAlignment(Qt::AlignRight);
 	item.sign->setOpenExternalLinks(true);
-	if (i == -1) i = items.size();
-	items.insert(i, item);
+	if (i == -1) {
+		items.push_back(item);
+	} else {
+		items.insert(i, item);
+	}
 	
 	item.status->show();
 	item.icon->show();
@@ -133,7 +167,7 @@
 		item.status->resize(statusItemWidth, statusItemHeight);
 		item.icon->move(MARGIN, height + MARGIN);
 		
-		item.sign->setText("<a href=\"http://twitter.com/" + item.username + "\" style=\"font-weight:bold;text-decoration:none\">" + item.username + "</a> - <a href=\"http://twitter.com/" + item.username + "/statuses/" + QString::number(item.messageId) + "\" style=\"font-size:70%;text-decoration:none\">" + formatDateTime(item.time) + "</a> <a href=\"reply://" + item.username + "\" style=\"text-decoration:none\"><img src=\":/images/reply.png\"/></a>");
+		item.sign->setText("<a href=\"http://twitter.com/" + item.username + "\" style=\"font-weight:bold;text-decoration:none\">" + item.username + "</a> - <a href=\"http://twitter.com/" + item.username + "/statuses/" + QString::number(item.messageId) + "\" style=\"font-size:70%;text-decoration:none\">" + formatDateTime(item.time) + "</a> <a href=\"reply://" + item.username + ":" + QString::number(item.messageId) + "\" style=\"text-decoration:none\"><img src=\":/images/reply.png\"/></a>");
 		item.sign->adjustSize();
 		
 		item.sign->move(width() - item.sign->width() - MARGIN, height + statusItemHeight + MARGIN);
@@ -164,12 +198,13 @@
 
 void TwitterWidget::replyClicked(const QUrl &url) {
 	emit reply(url.host());
+        emit replyID(QString::number(url.port()));
 }
 
 void TwitterWidget::reloadUserpic(const QString &userpic) {
 	for (int i = 0; i < items.size(); ++i) if (items[i].iconFileName == userpic) {
 		TwitterWidgetItem &item = items[i];
-		item.icon->setPixmap(QPixmap(item.iconFileName));
+		item.loadIcon();
 		item.icon->resize(ICON_SIZE, ICON_SIZE);
 	}
 }
@@ -190,17 +225,21 @@
 	
 QString TwitterWidget::formatDateTime(const QDateTime &time) {
 	int seconds = time.secsTo(QDateTime::currentDateTime());
-	if (seconds <= 15) return "Just now";
-	if (seconds <= 45) return "about " + QString::number(seconds) + " second" + (seconds == 1 ? "" : "s") + " ago";
+	if (seconds <= 15) return tr("Just now");
+	if (seconds <= 45) return tr("about %n second(s) ago", "", seconds);
 	int minutes = (seconds - 45 + 59) / 60;
-	if (minutes <= 45) return "about " + QString::number(minutes) + " minute" + (minutes == 1 ? "" : "s") + " ago";
+	if (minutes <= 45) return tr("about %n minute(s) ago", "", minutes);
 	int hours = (seconds - 45 * 60 + 3599) / 3600;
-	if (hours <= 18) return "about " + QString::number(hours) + " hour" + (hours == 1 ? "" : "s") + " ago";
+	if (hours <= 18) return tr("about %n hour(s) ago", "", hours);
 	int days = (seconds - 18 * 3600 + 24 * 3600 - 1) / (24 * 3600);
-	return "about " + QString::number(days) + " day" + (days == 1 ? "" : "s") + " ago";
+	return tr("about %n day(s) ago", "", days);
 }
 
 const TwitterWidgetItem TwitterWidget::getItem(int index) {
+// Possible fix for issue 10
+	if (index >= items.size()) {
+		throw QwitException("Trying to get a nonexisting item from TwitterWidget::items");
+	}
 	return items[index];
 }
 
Index: src/Twitter.h
===================================================================
--- src/Twitter.h	(revision 21)
+++ src/Twitter.h	(revision 38)
@@ -21,16 +21,19 @@
 #include <QString>
 #include <QBuffer>
 
-#define HOME_XML_URL "http://twitter.com/statuses/friends_timeline.xml"
-#define PUBLIC_XML_URL "http://twitter.com/statuses/public_timeline.xml"
-#define REPLIES_XML_URL "http://twitter.com/statuses/replies.xml"
-#define CUSTOM_XML_URL "http://twitter.com/statuses/user_timeline/"
+#define HOME_XML_URL "/statuses/friends_timeline.xml"
+#define PUBLIC_XML_URL "/statuses/public_timeline.xml"
+#define REPLIES_XML_URL "/statuses/replies.xml"
+#define CUSTOM_XML_URL "/statuses/user_timeline/"
 
-#define STATUS_UPDATE_URL "http://twitter.com/statuses/update.xml"
+#define STATUS_UPDATE_URL "/statuses/update.xml"
 
-//#define HOME_XML_URL "http://twitter.com/statuses/user_timeline/freetonik.xml"
-//#define HOME_XML_URL "http://twitter.lo/друзья.php"
-//#define STATUS_UPDATE_URL "http://twitter.lo/update.php"
+// #define HOME_XML_URL "http://twitter.com/statuses/friends_timeline.xml"
+// #define PUBLIC_XML_URL "http://twitter.com/statuses/public_timeline.xml"
+// #define REPLIES_XML_URL "http://twitter.com/statuses/replies.xml"
+// #define CUSTOM_XML_URL "http://twitter.com/statuses/user_timeline/"
+// 
+// #define STATUS_UPDATE_URL "http://twitter.com/statuses/update.xml"
 
 class Twitter: public QObject {
 	Q_OBJECT
@@ -44,16 +47,22 @@
 	QString proxyPassword;
 	int currentType;
 	QString urls[4];
+	QString serviceBaseURL;
+	QString serviceAPIURL;
 	
 public:
 
 	Twitter();
 	void useProxy(const QString &address, int port, const QString &username, const QString &password);
 	void dontUseProxy();
-	void sendStatus(QString username, QString password, QString status);
+	void sendStatus(QString username, QString password, QString status, QString replyID);
 	void update(QString username, QString password, int lastStatusId, int type, int count);
 	void setUrl(int index, const QString &url);
 	void abort();
+	QString getServiceBaseURL();
+	QString getServiceAPIURL();
+	void setServiceBaseURL(const QString &url);
+	void setServiceAPIURL(const QString &url);
 
 public slots:
 
Index: src/MainWindow.ui
===================================================================
--- src/MainWindow.ui	(revision 21)
+++ src/MainWindow.ui	(revision 38)
@@ -13,7 +13,8 @@
    <string>Qwit</string>
   </property>
   <property name="windowIcon" >
-   <iconset resource="qwit.qrc" >:/images/qwit.png</iconset>
+   <iconset resource="../qwit.qrc" >
+    <normaloff>:/images/qwit.png</normaloff>:/images/qwit.png</iconset>
   </property>
   <layout class="QGridLayout" >
    <item row="0" column="0" >
@@ -21,7 +22,7 @@
      <item>
       <layout class="QHBoxLayout" >
        <item>
-        <widget class="QLabel" name="label" >
+        <widget class="QLabel" name="greetingMessageLabel" >
          <property name="sizePolicy" >
           <sizepolicy vsizetype="Preferred" hsizetype="Preferred" >
            <horstretch>0</horstretch>
@@ -304,7 +305,7 @@
           <enum>QFrame::Sunken</enum>
          </property>
          <property name="text" >
-          <string></string>
+          <string/>
          </property>
          <property name="indent" >
           <number>-1</number>
@@ -326,7 +327,8 @@
           <string/>
          </property>
          <property name="icon" >
-          <iconset resource="qwit.qrc" >:/images/reload.png</iconset>
+          <iconset resource="../qwit.qrc" >
+           <normaloff>:/images/reload.png</normaloff>:/images/reload.png</iconset>
          </property>
          <property name="shortcut" >
           <string>Ctrl+R</string>
@@ -354,7 +356,8 @@
           <string/>
          </property>
          <property name="icon" >
-          <iconset resource="qwit.qrc" >:/images/options.png</iconset>
+          <iconset resource="../qwit.qrc" >
+           <normaloff>:/images/options.png</normaloff>:/images/options.png</iconset>
          </property>
          <property name="autoDefault" >
           <bool>false</bool>
@@ -373,7 +376,8 @@
           <string/>
          </property>
          <property name="icon" >
-          <iconset resource="qwit.qrc" >:/images/logs.png</iconset>
+          <iconset resource="../qwit.qrc" >
+           <normaloff>:/images/logs.png</normaloff>:/images/logs.png</iconset>
          </property>
          <property name="autoDefault" >
           <bool>false</bool>
@@ -395,7 +399,8 @@
           <string/>
          </property>
          <property name="icon" >
-          <iconset resource="qwit.qrc" >:/images/exit.png</iconset>
+          <iconset resource="../qwit.qrc" >
+           <normaloff>:/images/exit.png</normaloff>:/images/exit.png</iconset>
          </property>
          <property name="autoDefault" >
           <bool>false</bool>
@@ -409,6 +414,7 @@
   </layout>
  </widget>
  <resources>
+  <include location="../qwit.qrc" />
   <include location="qwit.qrc" />
  </resources>
  <connections/>
Index: src/QwitException.h
===================================================================
--- src/QwitException.h	(revision 0)
+++ src/QwitException.h	(revision 38)
@@ -0,0 +1,30 @@
+/*  This file is part of Qwit.
+
+    Qwit is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Qwit is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Qwit.  If not, see <http://www.gnu.org/licenses/>. */
+
+#ifndef QwitException_h
+#define QwitException_h
+
+#include <QString>
+
+class QwitException {
+public:
+	QString message;
+	
+	QwitException(QString message) {
+		this->message = message;
+	}
+};
+
+#endif
Index: src/TwitterWidget.h
===================================================================
--- src/TwitterWidget.h	(revision 21)
+++ src/TwitterWidget.h	(revision 38)
@@ -43,6 +43,8 @@
 	int cacheMessageId;
 	int cacheReplyStatusId;
 	int cacheIndex;
+	
+	void loadIcon();
 };
 
 const int ICON_SIZE = 48;
@@ -54,17 +56,20 @@
 private:
 	int messagesPerPage;
 	QVector<TwitterWidgetItem> items;
-	QString prepare(const QString &text, const int &replyStatusId);
+	QString prepare(const QString &text, const int &replyStatusId, const QString &serviceBaseURL);
+	bool isUsernameChar(const QChar &c) const;
+	QString serviceBaseURL;
 	
 public:
 
 	TwitterWidget();
 	void clear();
-	void addItem(const QString &userpic, const QString &username, const QString &status, const QDateTime &time, int messageId, int replyStatusId, int i = -1);
+	void addItem(const QString &userpic, const QString &username, const QString &status, const QDateTime &time, int messageId, int replyStatusId, int i, const QString &serviceBaseURL);
 	void updateItems();
 	const TwitterWidgetItem getItem(int index);
 	int getItemsCount();
 	void setMessagesPerPage(int value);
+        int replyStatusID;
 	
 	static QString formatDateTime(const QDateTime &time);
 	
@@ -81,6 +86,7 @@
 signals:
 	
 	void reply(const QString &);
+        void replyID(const QString &);
 };
 
 #endif
Index: src/UserpicsDownloader.cpp
===================================================================
--- src/UserpicsDownloader.cpp	(revision 21)
+++ src/UserpicsDownloader.cpp	(revision 38)
@@ -40,7 +40,7 @@
 	}
 	http.setHost(url.host(), url.port(80));
 	http.get(QUrl::toPercentEncoding(url.path(), "/"), &file);
-	emit stateChanged("Downloading " + item.first);
+	emit stateChanged(tr("Downloading: %1").arg(item.first));
 }
 	
 void UserpicsDownloader::download(const QString &url, const QString &filename) {
@@ -54,12 +54,12 @@
 
 void UserpicsDownloader::httpDone(bool error) {
 	if (error) {
-		emit stateChanged("Error while downloading userpic: " + http.errorString());
+		emit stateChanged(tr("Error while downloading userpic: %1").arg(http.errorString()));
 	}
 	QPair<QString, QString> item = queue.dequeue();
 	file.close();
 	emit userpicDownloaded(item.second);
-	emit stateChanged("Downloaded " + item.first);
+	emit stateChanged(tr("Downloaded: %1").arg(item.first));
 	if (queue.size()) {
 		startDownload();
 	}
Index: src/MainWindow.cpp
===================================================================
--- src/MainWindow.cpp	(revision 21)
+++ src/MainWindow.cpp	(revision 38)
@@ -18,6 +18,8 @@
 
 #include "MainWindow.h"
 
+#include "QwitException.h"
+
 #include <iostream>
 
 using namespace std;
@@ -59,7 +61,7 @@
 	
 	gridLayout->addWidget(scrollArea, 0, 0, 1, 1);
 	
-	twitterTabs[HOME_TWITTER_TAB] = TwitterTab(false, scrollArea, twitterWidget, 0);
+	twitterTabs[HOME_TWITTER_TAB] = TwitterTab(scrollArea, twitterWidget, 0);
 	
 	
 	twitterWidget = new TwitterWidget();
@@ -76,7 +78,7 @@
 	
 	gridLayout->addWidget(scrollArea, 0, 0, 1, 1);
 	
-	twitterTabs[PUBLIC_TWITTER_TAB] = TwitterTab(true, scrollArea, twitterWidget, 0);
+	twitterTabs[PUBLIC_TWITTER_TAB] = TwitterTab(scrollArea, twitterWidget, 0);
 	
 	
 	twitterWidget = new TwitterWidget();
@@ -93,7 +95,7 @@
 	
 	gridLayout->addWidget(scrollArea, 0, 0, 1, 1);
 	
-	twitterTabs[REPLIES_TWITTER_TAB] = TwitterTab(false, scrollArea, twitterWidget, 0);
+	twitterTabs[REPLIES_TWITTER_TAB] = TwitterTab(scrollArea, twitterWidget, 0);
 	
 	
 	twitterWidget = new TwitterWidget();
@@ -107,7 +109,7 @@
 	
 	vboxLayout1->addWidget(scrollArea);
 	
-	twitterTabs[CUSTOM_TWITTER_TAB] = TwitterTab(true, scrollArea, twitterWidget, 1);
+	twitterTabs[CUSTOM_TWITTER_TAB] = TwitterTab(scrollArea, twitterWidget, 0);
 	
 	
 	statusTextEdit = new StatusTextEdit(this);
@@ -143,6 +145,7 @@
 	
 	for (int i = 0; i < TWITTER_TABS; ++i) {
 		connect(twitterTabs[i].twitterWidget, SIGNAL(reply(const QString &)), statusTextEdit, SLOT(reply(const QString &)));
+                connect(twitterTabs[i].twitterWidget, SIGNAL(replyID(const QString &)), this, SLOT(setReplyID(const QString &)));
 	}
 	
 	connect(statusTextEdit, SIGNAL(leftCharsNumberChanged(int)), this, SLOT(leftCharsNumberChanged(int)));
@@ -189,6 +192,10 @@
 	password = settings.value("password", "").toString();
 	interval = settings.value("interval", DEFAULT_UPDATE_INTERVAL).toInt();
 	messagesPerPage = settings.value("messagesPerPage", DEFAULT_MESSAGES_PER_PAGE).toInt();
+	updatesNotification = settings.value("updatesNotification", true).toBool();
+	twitter.setServiceAPIURL(settings.value("serviceAPIURL", "http://twitter.com").toString());
+	twitter.setServiceBaseURL(settings.value("serviceBaseURL", "http://twitter.com").toString());
+	greetingMessageLabel->setText(settings.value("greetingMessage", "What are you doing?").toString());
 	settings.endGroup();
 	settings.beginGroup("Proxy");
 	useProxy = settings.value("useProxy", "").toBool();
@@ -205,10 +212,14 @@
 	twitter.setUrl(CUSTOM_TWITTER_TAB, QString(CUSTOM_XML_URL) + customUsernameLineEdit->text() + ".xml");
 	
 	optionsDialog->messagesPerPageLineEdit->setText(QString::number(messagesPerPage));
+	optionsDialog->updatesNotificationCheckBox->setCheckState(updatesNotification ? Qt::Checked : Qt::Unchecked);
+	optionsDialog->serviceBaseURLLineEdit->setText(twitter.getServiceBaseURL());
+	optionsDialog->serviceAPIURLLineEdit->setText(twitter.getServiceAPIURL());
 	optionsDialog->usernameLineEdit->setText(username);
 	optionsDialog->passwordLineEdit->setText(password);
 	optionsDialog->savePasswordCheckBox->setCheckState(savePassword ? Qt::Checked : Qt::Unchecked);
 	optionsDialog->intervalLineEdit->setText(QString::number(interval));
+	optionsDialog->greetingMessageLineEdit->setText(greetingMessageLabel->text());
 	optionsDialog->useProxyCheckBox->setCheckState(useProxy ? Qt::Checked : Qt::Unchecked);
 	optionsDialog->proxyAddressLineEdit->setText(proxyAddress);
 	optionsDialog->proxyPortLineEdit->setText(QString::number(proxyPort));
@@ -227,6 +238,10 @@
 	if ((username == "") || (password == "")) {
 		statusTextEdit->setDisabled(true);
 	}
+
+	for (int tab = 0; tab < TWITTER_TABS; ++tab) {
+		twitterTabs[tab].twitterWidget->setMessagesPerPage(messagesPerPage);
+	}
 	
 	settings.beginGroup("Twits");
 	for (int tab = 0; tab < TWITTER_TABS; ++tab) {
@@ -243,20 +258,31 @@
 				QDateTime::fromString(settings.value("time").toString(), "yyyy-MM-dd hh:mm:ss"),
 				settings.value("messageId").toInt(),
 				settings.value("replyStatusId").toInt(),
-				-1
+				-1,
+				twitter.getServiceBaseURL()
 			);
 		}
 		settings.endArray();
 	}
 	settings.endGroup();
 }
+
+void MainWindow::setReplyID(const QString &replyID) {
+        replyStatusID = replyID;
+        isReply = true;
+}
 	
 void MainWindow::sendStatus() {
 	QString status = statusTextEdit->toPlainText().simplified();
 	if (status == "") {
 		return;
 	}
-	twitter.sendStatus(username, password, status);
+        if (isReply) {
+                twitter.sendStatus(username, password, status, replyStatusID);
+                isReply = false;
+                replyStatusID = "";
+        }
+        else twitter.sendStatus(username, password, status, "");
 	statusTextEdit->setText("");
 	charsLeftLabel->setText(QString::number(statusTextEdit->getMaxStatusCharacter()));
 	charsLeftLabel->setForegroundRole(QPalette::Light);
@@ -282,6 +308,8 @@
 	password = optionsDialog->passwordLineEdit->text();
 	bool savePassword = optionsDialog->savePasswordCheckBox->checkState() == Qt::Checked;
 	interval = optionsDialog->intervalLineEdit->text().toInt();
+	twitter.setServiceBaseURL(optionsDialog->serviceBaseURLLineEdit->text());
+	twitter.setServiceAPIURL(optionsDialog->serviceAPIURLLineEdit->text());
 	
 	useProxy = optionsDialog->useProxyCheckBox->checkState() == Qt::Checked;
 	proxyAddress = optionsDialog->proxyAddressLineEdit->text();
@@ -290,6 +318,8 @@
 	proxyPassword = optionsDialog->proxyPasswordLineEdit->text();
 	messagesPerPage = optionsDialog->messagesPerPageLineEdit->text().toInt();
 	bool proxySavePassword = optionsDialog->proxySavePasswordCheckBox->checkState() == Qt::Checked;
+	updatesNotification = optionsDialog->updatesNotificationCheckBox->checkState() == Qt::Checked;
+	greetingMessageLabel->setText(optionsDialog->greetingMessageLineEdit->text());
 	
 	settings.beginGroup("MainWindow");
 	settings.setValue("size", size());
@@ -305,6 +335,10 @@
 	}
 	settings.setValue("interval", interval);
 	settings.setValue("messagesPerPage", messagesPerPage);
+	settings.setValue("updatesNotification", updatesNotification);
+	settings.setValue("serviceBaseURL", twitter.getServiceBaseURL());
+	settings.setValue("serviceAPIURL", twitter.getServiceAPIURL());
+	settings.setValue("greetingMessage", greetingMessageLabel->text());
 	settings.endGroup();
 	settings.beginGroup("Proxy");
 	settings.setValue("useProxy", useProxy);
@@ -328,13 +362,18 @@
 	for (int tab = 0; tab < TWITTER_TABS; ++tab) {
 		settings.beginWriteArray("Twits" + QString::number(tab));
 		for (int i = 0; i < twitterTabs[tab].twitterWidget->getItemsCount(); ++i) {
-			settings.setArrayIndex(i);
-			settings.setValue("userpic", twitterTabs[tab].twitterWidget->getItem(i).cacheUserpic);
-			settings.setValue("username", twitterTabs[tab].twitterWidget->getItem(i).cacheUsername);
-			settings.setValue("status", twitterTabs[tab].twitterWidget->getItem(i).cacheStatus);
-			settings.setValue("time", twitterTabs[tab].twitterWidget->getItem(i).cacheTime.toString("yyyy-MM-dd hh:mm:ss"));
-			settings.setValue("messageId", twitterTabs[tab].twitterWidget->getItem(i).cacheMessageId);
-			settings.setValue("replyStatusId", twitterTabs[tab].twitterWidget->getItem(i).cacheReplyStatusId);
+			try {
+				const TwitterWidgetItem &item = twitterTabs[tab].twitterWidget->getItem(i);
+				settings.setArrayIndex(i);
+				settings.setValue("userpic", item.cacheUserpic);
+				settings.setValue("username", item.cacheUsername);
+				settings.setValue("status", item.cacheStatus);
+				settings.setValue("time", item.cacheTime.toString("yyyy-MM-dd hh:mm:ss"));
+				settings.setValue("messageId", item.cacheMessageId);
+				settings.setValue("replyStatusId", item.cacheReplyStatusId);
+			} catch (const QwitException &exception) {
+				cerr << "Exception catched: " << qPrintable(exception.message) << endl;
+			}
 		}
 		settings.endArray();
 	}
@@ -431,10 +470,6 @@
 }
 
 void MainWindow::updated(const QByteArray &buffer, int type) {
-	if (twitterTabs[type].clear) {
-		twitterTabs[type].twitterWidget->clear();
-	}
-	
 	QDomDocument document;
 	
 	document.setContent(buffer);
@@ -511,15 +546,14 @@
 			dir.mkdir(".qwit");
 			imageFileName = dir.absolutePath() + "/.qwit/" + imageFileName;
 			userpicsDownloader.download(image, imageFileName);
-			
-			twitterTabs[type].twitterWidget->addItem(imageFileName, user, message.simplified(), time.toLocalTime(), id, replyStatusId, j++);
+			twitterTabs[type].twitterWidget->addItem(imageFileName, user, message.simplified(), time.toLocalTime(), id, replyStatusId, j++, twitter.getServiceBaseURL());
 		}
 		node = node.nextSibling();
 	}
 	twitterTabs[type].lastId = maxId;
 	twitterTabs[type].lastUpdateTime = QDateTime::currentDateTime().toTime_t();
-	if (trayMessage != "") {
-		trayIcon->showMessage("Qwit updates", trayMessage);
+	if ((trayMessage != "") && updatesNotification) {
+		trayIcon->showMessage(tr("Qwit updates"), trayMessage);
 	}
 	saveState();
 }
@@ -574,6 +608,9 @@
 void MainWindow::customUsernameChanged() {
 	twitter.setUrl(CUSTOM_TWITTER_TAB, QString(CUSTOM_XML_URL) + customUsernameLineEdit->text() + ".xml");
 	
+	twitterTabs[CUSTOM_TWITTER_TAB].twitterWidget->clear();
+	twitterTabs[CUSTOM_TWITTER_TAB].lastId = 0;
+	
 	saveState();
 	
 	updateTimeline();
Index: src/MainWindow.h
===================================================================
--- src/MainWindow.h	(revision 21)
+++ src/MainWindow.h	(revision 38)
@@ -63,14 +63,12 @@
 
 class TwitterTab {
 public:
-	bool clear;
 	TwitterWidget *twitterWidget;
 	int lastId;
 	QScrollArea *scrollArea;
 	int lastUpdateTime;
 	
-	TwitterTab(bool clear, QScrollArea *scrollArea, TwitterWidget *twitterWidget, int lastId) {
-		this->clear = clear;
+	TwitterTab(QScrollArea *scrollArea, TwitterWidget *twitterWidget, int lastId) {
 		this->scrollArea = scrollArea;
 		this->twitterWidget = twitterWidget;
 		this->lastId = lastId;
@@ -113,6 +111,10 @@
 	TwitterTab twitterTabs[TWITTER_TABS];
 	Twitter twitter;
 	QMap<QString, int> monthes;
+	bool updatesNotification;
+
+        QString replyStatusID;
+        bool isReply;
 	
 	void setupTrayIcon();
 	void loadState();
@@ -135,6 +137,7 @@
 	void tabChanged(int index);
 	void updateItems();
 	void customUsernameChanged();
+        void setReplyID(const QString &replyID);
 
 protected:
 	
Index: src/qwit.cpp
===================================================================
--- src/qwit.cpp	(revision 21)
+++ src/qwit.cpp	(revision 38)
@@ -14,11 +14,19 @@
     along with Qwit.  If not, see <http://www.gnu.org/licenses/>. */
 
 #include <QApplication>
+#include <QLocale>
+#include <QTranslator>
 
 #include "MainWindow.h"
 
 int main(int argc, char *argv[]) {
 	QApplication app(argc, argv);
+
+	QString locale = QLocale::system().name();
+	QTranslator translator;
+	translator.load(QString(":/translations/qwit_") + locale);
+	app.installTranslator(&translator);
+
 	MainWindow::getInstance()->show();
 	return app.exec();
 }
Index: qwit.qrc
===================================================================
--- qwit.qrc	(revision 21)
+++ qwit.qrc	(revision 38)
@@ -6,5 +6,8 @@
         <file>images/qwit.png</file>
         <file>images/reload.png</file>
         <file>images/reply.png</file>
+        <file>translations/qwit_tr_TR.qm</file>
+        <file>translations/qwit_ru_RU.qm</file>
+        <file>translations/qwit_en_US.qm</file>
     </qresource>
 </RCC>
Index: README
===================================================================
--- README	(revision 21)
+++ README	(revision 38)
@@ -1,7 +1,7 @@
 GENERAL
 
 Qwit - free cross-platform Qt-based Twitter client.
-Copyright (C) 2008 Artem Iglikov
+Copyright (C) 2008, 2009 Artem Iglikov
 
 CONTACTS
 
@@ -24,3 +24,11 @@
 
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+THANKS
+
+...to Linus Torvalds, for Linux
+...to Richard Stallman, for GNU
+...to Uğur Çetin, for Turkish translation
+
+and to me, for this stuff :)
