diff -Nur lynx.cfg.orig lynx.cfg
--- lynx.cfg.orig	2008-11-03 19:50:57.000000000 +0200
+++ lynx.cfg	2008-11-03 19:50:59.000000000 +0200
@@ -127,7 +127,7 @@
 # pointers to lots of interesting places on the web.
 #
 #DEFAULT_INDEX_FILE:http://www.ncsa.uiuc.edu/SDG/Software/Mosaic/MetaIndex.html
-DEFAULT_INDEX_FILE:http://lynx.isc.org/
+DEFAULT_INDEX_FILE:http://www.google.com.tr
 
 
 .h1 Interaction
--- lynx.cfg.orig       2010-05-07 10:41:32.347338460 +0300
+++ lynx.cfg    2010-05-07 10:41:58.682332386 +0300
@@ -93,7 +93,7 @@
 #
 # Normally we expect you will connect to a remote site, e.g., the Lynx starting
 # site:
-STARTFILE:http://lynx.isc.org/
+STARTFILE:http://www.google.com.tr
 #
 # As an alternative, you may want to use a local URL.  A good choice for this is
 # the user's home directory:
