diff -Nuar libxfce4menu-4.6.1.orig/libxfce4menu/Makefile.in libxfce4menu-4.6.1/libxfce4menu/Makefile.in
--- libxfce4menu-4.6.1.orig/libxfce4menu/Makefile.in	2009-04-09 15:11:59.000000000 +0300
+++ libxfce4menu-4.6.1/libxfce4menu/Makefile.in	2009-06-05 02:09:24.000000000 +0300
@@ -188,7 +188,7 @@
 LD = @LD@
 LDFLAGS = @LDFLAGS@
 LIBOBJS = @LIBOBJS@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lgobject-2.0
 LIBTOOL = @LIBTOOL@
 LIBXFCE4MENU_VERINFO = @LIBXFCE4MENU_VERINFO@
 LIBXFCE4MENU_VERSION_API = @LIBXFCE4MENU_VERSION_API@
