diff -ur vnstat-1.10.orig/src/Makefile vnstat-1.10/src/Makefile
--- vnstat-1.10.orig/src/Makefile	2009-07-29 14:22:51.000000000 +0300
+++ vnstat-1.10/src/Makefile	2010-02-12 23:20:55.071160412 +0200
@@ -1,5 +1,4 @@
 CC = gcc
-CFLAGS = -O2
 LDLIBS = -lm
 OBJS = vnstat.o ifinfo.o dbxml.o dbshow.o dbaccess.o dbmerge.o common.o misc.o cfg.o traffic.o
 DOBJS = vnstatd.o ifinfo.o dbaccess.o dbcache.o common.o misc.o cfg.o
