--- htdocs/gui/smarty_template/smarty_care.class.php	2006-08-24 17:00:46.000000000 +0300
+++ smarty_care.class-yama.php	2006-08-24 16:51:34.000000000 +0300
@@ -72,7 +72,7 @@
 			# If the global config template is not available, use hard coded template theme
 			if(!isset($GLOBAL_CONFIG['template_smarty'])||empty($GLOBAL_CONFIG['template_smarty'])){
 				if(isset($template_theme)) $this->templatedir=$template_theme; // use this theme if the global item is not available
-					else $this->templatedir = $this->$default_template;
+					#else $this->templatedir = $this->$default_template;
 			}else{
 				$this->templatedir=$GLOBAL_CONFIG['template_smarty'];
 			}
