diff -p -up audacious-plugins-2.2-beta2/src/amidi-plug/i_utils.c.format-strings audacious-plugins-2.2-beta2/src/amidi-plug/i_utils.c
--- audacious-plugins-2.2-beta2/src/amidi-plug/i_utils.c.format-strings	2009-11-09 14:57:13.000000000 +0100
+++ audacious-plugins-2.2-beta2/src/amidi-plug/i_utils.c	2009-11-10 09:18:12.000000000 +0100
@@ -125,9 +125,9 @@ gpointer i_message_gui( gchar * title , 
 
   if ( parent_win != NULL )
     win = gtk_message_dialog_new( GTK_WINDOW(parent_win) , GTK_DIALOG_DESTROY_WITH_PARENT ,
-                                  mtype , GTK_BUTTONS_OK , message );
+                                  mtype , GTK_BUTTONS_OK , "%s", message );
   else
-    win = gtk_message_dialog_new( NULL , 0 , mtype , GTK_BUTTONS_OK , message );
+    win = gtk_message_dialog_new( NULL , 0 , mtype , GTK_BUTTONS_OK , "%s", message );
 
   gtk_window_set_title( GTK_WINDOW(win) , title );
   g_signal_connect_swapped( G_OBJECT(win) , "response" , G_CALLBACK(gtk_widget_destroy) , win );
diff -p -up audacious-plugins-2.2-beta2/src/bluetooth/agent.c.format-strings audacious-plugins-2.2-beta2/src/bluetooth/agent.c
--- audacious-plugins-2.2-beta2/src/bluetooth/agent.c.format-strings	2009-11-09 14:57:13.000000000 +0100
+++ audacious-plugins-2.2-beta2/src/bluetooth/agent.c	2009-11-10 09:18:12.000000000 +0100
@@ -722,7 +722,7 @@ static void bonding_created(DBusGProxy *
 			device = g_strdup_printf("%s (%s)", name, address);
 	} else
 		device = g_strdup(address);
-       bonded_dev = g_strdup_printf(address);
+       bonded_dev = g_strdup_printf("%s", address);
 
 
 	text = g_strdup_printf(_("Created bonding with %s"), device);
diff -p -up audacious-plugins-2.2-beta2/src/evdev-plug/ed_ui.c.format-strings audacious-plugins-2.2-beta2/src/evdev-plug/ed_ui.c
--- audacious-plugins-2.2-beta2/src/evdev-plug/ed_ui.c.format-strings	2009-11-09 14:57:13.000000000 +0100
+++ audacious-plugins-2.2-beta2/src/evdev-plug/ed_ui.c	2009-11-10 09:18:12.000000000 +0100
@@ -1533,7 +1533,7 @@ ed_ui_message_show ( gchar * title , gch
   message_win = gtk_message_dialog_new(
                   parent_win ,
                   ( parent_win != NULL ? GTK_DIALOG_DESTROY_WITH_PARENT : 0 ) ,
-                  GTK_MESSAGE_INFO , GTK_BUTTONS_CLOSE , message );
+                  GTK_MESSAGE_INFO , GTK_BUTTONS_CLOSE , "%s", message );
   gtk_window_set_title( GTK_WINDOW(message_win) , title );
 
   gtk_dialog_run( GTK_DIALOG(message_win) );
diff -p -up audacious-plugins-2.2-beta2/src/gtkui/ui_gtk.c.format-strings audacious-plugins-2.2-beta2/src/gtkui/ui_gtk.c
--- audacious-plugins-2.2-beta2/src/gtkui/ui_gtk.c.format-strings	2009-11-09 14:57:13.000000000 +0100
+++ audacious-plugins-2.2-beta2/src/gtkui/ui_gtk.c	2009-11-10 09:18:12.000000000 +0100
@@ -333,7 +333,7 @@ static void ui_show_error(const gchar * 
                                            GTK_DIALOG_DESTROY_WITH_PARENT,
                                            GTK_MESSAGE_ERROR,
                                            GTK_BUTTONS_OK,
-                                           _(markup));
+                                           "%s", _(markup));
 
     gtk_window_set_modal(GTK_WINDOW(dialog), TRUE);
     gtk_widget_show(GTK_WIDGET(dialog));
diff -p -up audacious-plugins-2.2-beta2/src/mtp_up/mtp.c.format-strings audacious-plugins-2.2-beta2/src/mtp_up/mtp.c
--- audacious-plugins-2.2-beta2/src/mtp_up/mtp.c.format-strings	2009-11-09 14:57:14.000000000 +0100
+++ audacious-plugins-2.2-beta2/src/mtp_up/mtp.c	2009-11-10 09:18:12.000000000 +0100
@@ -62,7 +62,7 @@ void show_dialog(const gchar* message)
             GTK_DIALOG_MODAL,
             GTK_MESSAGE_ERROR,
             GTK_BUTTONS_OK,
-            message);
+	    "%s", message);
     gtk_dialog_run (GTK_DIALOG (dialog));
     gtk_widget_show(dialog);
     /* gtk_widget_destroy(dialog); */
diff -p -up audacious-plugins-2.2-beta2/src/skins/plugin.c.format-strings audacious-plugins-2.2-beta2/src/skins/plugin.c
--- audacious-plugins-2.2-beta2/src/skins/plugin.c.format-strings	2009-11-09 14:57:15.000000000 +0100
+++ audacious-plugins-2.2-beta2/src/skins/plugin.c	2009-11-10 09:18:12.000000000 +0100
@@ -255,7 +255,7 @@ static void show_error_message(const gch
                                            GTK_DIALOG_DESTROY_WITH_PARENT,
                                            GTK_MESSAGE_ERROR,
                                            GTK_BUTTONS_OK,
-                                           _(markup));
+                                           "%s", _(markup));
 
     gtk_window_set_modal(GTK_WINDOW(dialog), TRUE);
     gtk_widget_show(GTK_WIDGET(dialog));
diff -p -up audacious-plugins-2.2-beta2/src/skins/ui_main.c.format-strings audacious-plugins-2.2-beta2/src/skins/ui_main.c
