--- gperiodic-2.0.10/Makefile	2007-07-11 14:58:05.000000000 +0200
+++ gperiodic-2.0.10/Makefile	2007-09-11 20:11:45.000000000 +0200
@@ -1,5 +1,5 @@
 CC := gcc
-CFLAGS := `pkg-config --cflags gtk+-2.0` -I. -DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGDK_PIXBUF_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED
+CFLAGS := `pkg-config --cflags gtk+-2.0` -I. -DG_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGDK_PIXBUF_DISABLE_DEPRECATED
 LIBS :=`pkg-config --libs gtk+-2.0`
 bindir ?= /usr/bin
 datadir ?= /usr/share
