diff -Naur jahshaka.orig/jahuserinstall jahshaka/jahuserinstall
--- jahshaka.orig/jahuserinstall	2006-03-30 10:32:25.000000000 +0300
+++ jahshaka/jahuserinstall	2007-02-24 17:34:42.000000000 +0200
@@ -5,7 +5,7 @@
 prefix=$1
 target=$2
 
-[ ! -d "$prefix" -o "$target" = "" ] && echo "Usage: jahinstall /usr $HOME/jahstorage" && exit 1
+[ ! -d "$prefix" -o "$target" = "" ] && echo "Usage: jahinstall /usr $HOME/.jahshaka" && exit 1
 
 cpflags="-u --no-preserve=ownership"
 [ "`uname -s`" = "Darwin" ] && cpflags=
diff -Naur jahshaka.orig/scenes/grovejet.jsf jahshaka/scenes/grovejet.jsf
--- jahshaka.orig/scenes/grovejet.jsf	2005-09-22 02:39:59.000000000 +0300
+++ jahshaka/scenes/grovejet.jsf	2007-02-24 17:32:00.000000000 +0200
@@ -11,7 +11,7 @@
         <PROPERTIES blend = '1' depth = '0' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <BACKDROP grid_setting = '0' aliasing = '0' text_safe = '0'>
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'grovejetback' clipnumber = '10'
-             location = '/jahstorage/media/comps/groovejet/'
+             location = '/.jahshaka/media/comps/groovejet/'
              filename = 'grovejetback' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '743' ypos = '86' />
@@ -25,7 +25,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '1' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'grivegirl' clipnumber = '1'
-             location = '/jahstorage/media/comps/groovejet/'
+             location = '/.jahshaka/media/comps/groovejet/'
              filename = 'grivegirl' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '79' ypos = '316' />
@@ -38,7 +38,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '1' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'CLIP' type = 'SEQUENCE' clipname = 'crazy_' clipnumber = '2'
-             location = '/jahstorage/media/comps/groovejet/crazy/'
+             location = '/.jahshaka/media/comps/groovejet/crazy/'
              filename = 'crazy_' extension = '.bmp'
              startframe = '0' endframe = '83' clipinframe = '0' clipoutframe = '83' videoext = '00000' videotex = '1'
              inframe = '0' outframe = '83' toggleends = '1' loop = '0' ping = '0' xpos = '288' ypos = '316' />
@@ -52,7 +52,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '1' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'mask' clipnumber = '7'
-             location = '/jahstorage/media/comps/groovejet/'
+             location = '/.jahshaka/media/comps/groovejet/'
              filename = 'mask' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '278' ypos = '88' />
@@ -65,7 +65,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '1' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'lines' clipnumber = '4'
-             location = '/jahstorage/media/comps/groovejet/'
+             location = '/.jahshaka/media/comps/groovejet/'
              filename = 'lines' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '506' ypos = '313' />
@@ -78,7 +78,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '0' cull = '0' lighting = '0' foreground = '1' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'groovejetcom' clipnumber = '6'
-             location = '/jahstorage/media/comps/groovejet/'
+             location = '/.jahshaka/media/comps/groovejet/'
              filename = 'groovejetcom' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '507' ypos = '154' />
@@ -93,7 +93,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '1' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'girl' clipnumber = '9'
-             location = '/jahstorage/media/images/misc/'
+             location = '/.jahshaka/media/images/misc/'
              filename = 'girl' extension = '.bmp'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '10' ypos = '10' />
@@ -107,7 +107,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '0' cull = '0' lighting = '0' foreground = '1' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'entertainment' clipnumber = '11'
-             location = '/jahstorage/media/comps/groovejet/text/'
+             location = '/.jahshaka/media/comps/groovejet/text/'
              filename = 'entertainment' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '10' ypos = '10' />
@@ -121,7 +121,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '0' cull = '0' lighting = '0' foreground = '1' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'culture' clipnumber = '12'
-             location = '/jahstorage/media/comps/groovejet/text/'
+             location = '/.jahshaka/media/comps/groovejet/text/'
              filename = 'culture' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '9' ypos = '10' />
@@ -135,7 +135,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '0' cull = '0' lighting = '0' foreground = '1' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'music' clipnumber = '13'
-             location = '/jahstorage/media/comps/groovejet/text/'
+             location = '/.jahshaka/media/comps/groovejet/text/'
              filename = 'music' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '10' ypos = '10' />
@@ -149,7 +149,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '0' cull = '0' lighting = '0' foreground = '1' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'art' clipnumber = '14'
-             location = '/jahstorage/media/comps/groovejet/text/'
+             location = '/.jahshaka/media/comps/groovejet/text/'
              filename = 'art' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '10' ypos = '10' />
@@ -163,7 +163,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '0' cull = '0' lighting = '0' foreground = '1' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'events' clipnumber = '15'
-             location = '/jahstorage/media/comps/groovejet/text/'
+             location = '/.jahshaka/media/comps/groovejet/text/'
              filename = 'events' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '10' ypos = '10' />
@@ -177,7 +177,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '0' cull = '0' lighting = '0' foreground = '1' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'fashion' clipnumber = '16'
-             location = '/jahstorage/media/comps/groovejet/text/'
+             location = '/.jahshaka/media/comps/groovejet/text/'
              filename = 'fashion' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '10' ypos = '10' />
@@ -191,7 +191,7 @@
         <TYPE layertype = 'LAYER' objtype = 'CYLINDER' />
         <PROPERTIES blend = '1' depth = '1' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'earthtex' clipnumber = '18'
-             location = '/jahstorage/media/images/textures/'
+             location = '/.jahshaka/media/images/textures/'
              filename = 'earthtex' extension = '.jpg'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '10' ypos = '10' />
diff -Naur jahshaka.orig/scenes/pluginz.jsf jahshaka/scenes/pluginz.jsf
--- jahshaka.orig/scenes/pluginz.jsf	2005-09-22 02:39:59.000000000 +0300
+++ jahshaka/scenes/pluginz.jsf	2007-02-24 17:35:10.000000000 +0200
@@ -11,7 +11,7 @@
         <PROPERTIES blend = '1' depth = '0' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <BACKDROP grid_setting = '0' aliasing = '0' text_safe = '0'>
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'plugback' clipnumber = '97'
-             location = '/jahstorage/media/comps/pluginz/'
+             location = '/.jahshaka/media/comps/pluginz/'
              filename = 'plugback' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '10' ypos = '10' />
@@ -25,7 +25,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '1' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'CLIP' type = 'SEQUENCE' clipname = 'clean_rotate.' clipnumber = '96'
-             location = '/jahstorage/media/comps/pluginz/plug/'
+             location = '/.jahshaka/media/comps/pluginz/plug/'
              filename = 'clean_rotate.' extension = '.jpg'
              startframe = '1' endframe = '150' clipinframe = '1' clipoutframe = '150' videoext = '0001' videotex = '1'
              inframe = '1' outframe = '150' toggleends = '1' loop = '0' ping = '0' xpos = '314' ypos = '359' />
@@ -39,7 +39,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '1' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'pluginztext' clipnumber = '98'
-             location = '/jahstorage/media/comps/pluginz/'
+             location = '/.jahshaka/media/comps/pluginz/'
              filename = 'pluginztext' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '11' ypos = '10' />
@@ -54,7 +54,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '1' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'fuzz' clipnumber = '101'
-             location = '/jahstorage/media/comps/pluginz/'
+             location = '/.jahshaka/media/comps/pluginz/'
              filename = 'fuzz' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '10' ypos = '10' />
@@ -69,7 +69,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '1' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'logo' clipnumber = '99'
-             location = '/jahstorage/media/comps/pluginz/'
+             location = '/.jahshaka/media/comps/pluginz/'
              filename = 'logo' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '255' ypos = '37' />
@@ -85,7 +85,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '1' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'line' clipnumber = '87'
-             location = '/jahstorage/media/comps/pluginz/'
+             location = '/.jahshaka/media/comps/pluginz/'
              filename = 'line' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '299' ypos = '277' />
@@ -100,7 +100,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '1' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'digitaltools' clipnumber = '106'
-             location = '/jahstorage/media/comps/pluginz/'
+             location = '/.jahshaka/media/comps/pluginz/'
              filename = 'digitaltools' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '10' ypos = '10' />
@@ -114,7 +114,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '1' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'genres' clipnumber = '107'
-             location = '/jahstorage/media/comps/pluginz/'
+             location = '/.jahshaka/media/comps/pluginz/'
              filename = 'genres' extension = '.png'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '10' ypos = '10' />
@@ -129,7 +129,7 @@
         <TYPE layertype = 'TEXT' objtype = 'TITLEGRID' />
         <PROPERTIES blend = '0' depth = '1' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '1' />
         <ASSET layer_status = '1' category = 'IMAGE' type = 'STILL' clipname = 'sky' clipnumber = '6'
-             location = '/jahstorage/media/images/nature/'
+             location = '/.jahshaka/media/images/nature/'
              filename = 'sky' extension = '.bmp'
              startframe = '1' endframe = '1' clipinframe = '0' clipoutframe = '0' videoext = '' videotex = '0'
              xpos = '617' ypos = '160' />
@@ -146,12 +146,12 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '0' cull = '0' lighting = '0' foreground = '0' smooth = '1' reflect = '0' />
         <ASSET layer_status = '1' category = 'CLIP' type = 'SEQUENCE' clipname = 'flame_' clipnumber = '8'
-             location = '/jahstorage/media/clips/nature/flame/'
+             location = '/.jahshaka/media/clips/nature/flame/'
              filename = 'flame_' extension = '.BMP'
              startframe = '0' endframe = '89' clipinframe = '0' clipoutframe = '89' videoext = '0000' videotex = '1'
              inframe = '0' outframe = '89' toggleends = '1' loop = '0' ping = '0' xpos = '956' ypos = '408' />
         <KEYASSET category = 'CLIP' type = 'SEQUENCE' clipname = 'flamekey_' clipnumber = '9'
-             location = '/jahstorage/media/clips/nature/flamekey/'n             filename = 'flamekey_' extension = '.BMP'
+             location = '/.jahshaka/media/clips/nature/flamekey/'n             filename = 'flamekey_' extension = '.BMP'
              startframe = '0' endframe = '89' videoext = '0000' videotex = '1' blendmode = 'GL_ZERO' inframe = '0' outframe = '89' xpos = '859' ypos = '476' />
         <KEYFRAMES count = '0'>
             <KEYFRAME TC = '0' SX = '1' SY = '1' SZ = '1' TX = '-58' TY = '143' TZ = '30' RX = '0' RY = '0' RZ = '0' AL = '100' extrude = '5' />
@@ -162,7 +162,7 @@
         <TYPE layertype = 'LAYER' objtype = 'JAHLAYER' />
         <PROPERTIES blend = '1' depth = '0' cull = '0' lighting = '0' foreground = '0' smooth = '0' reflect = '0' />
         <ASSET layer_status = '1' category = 'CLIP' type = 'SEQUENCE' clipname = 'flame_' clipnumber = '8'
-             location = '/jahstorage/media/clips/nature/flame/'
+             location = '/.jahshaka/media/clips/nature/flame/'
              filename = 'flame_' extension = '.BMP'
              startframe = '0' endframe = '89' clipinframe = '140' clipoutframe = '229' videoext = '0000' videotex = '1'
              inframe = '140' outframe = '229' toggleends = '0' loop = '0' ping = '0' xpos = '956' ypos = '408' />
diff -Naur jahshaka.orig/source/Jahshaka/JahDesktop/library/assetmanmainwidget.ui.h jahshaka/source/Jahshaka/JahDesktop/library/assetmanmainwidget.ui.h
--- jahshaka.orig/source/Jahshaka/JahDesktop/library/assetmanmainwidget.ui.h	2006-06-30 07:46:02.000000000 +0300
+++ jahshaka/source/Jahshaka/JahDesktop/library/assetmanmainwidget.ui.h	2007-02-24 21:01:59.000000000 +0200
@@ -360,7 +360,7 @@
 	/////////////////////////////////////////////////////////////////////
 	// Now deal with the media drive.
     JahPrefs& jprefs = JahPrefs::getInstance();
-    QString JahMediaPath  = jprefs.getMediaPath().c_str();
+    QString JahMediaPath  = QDir::homeDirPath( ) + "/.jahshaka/";
 
      mediaroot = new Directory( filelist, "Media", Directory::ComputerRoot );
      mediaroot->setSelected( false );
@@ -495,7 +495,7 @@
     //check to see if database has been relocated
     JahPrefs& jprefs = JahPrefs::getInstance();
     QString JahBasePath   = jprefs.getBasePath().c_str();
-    QString JahMediaPath  = jprefs.getMediaPath().c_str();
+    QString JahMediaPath  = QDir::homeDirPath( ) + "/.jahshaka/";
 	JahTrace *jtrace = JahTrace::getInstance();	//set up tracer
 
     QDir d( JahMediaPath+"/database/" );                        // "./example"
diff -Naur jahshaka.orig/source/Jahshaka/JahLibraries/jahdatabase/jahdatabase.cpp jahshaka/source/Jahshaka/JahLibraries/jahdatabase/jahdatabase.cpp
--- jahshaka.orig/source/Jahshaka/JahLibraries/jahdatabase/jahdatabase.cpp	2006-09-15 20:14:53.000000000 +0300
+++ jahshaka/source/Jahshaka/JahLibraries/jahdatabase/jahdatabase.cpp	2007-02-24 21:05:59.000000000 +0200
@@ -58,7 +58,7 @@
     //check to see if database has been relocated
     JahPrefs& jprefs = JahPrefs::getInstance();
     QString JahBasePath   = jprefs.getBasePath().c_str();
-    QString JahMediaPath  = jprefs.getMediaPath().c_str();
+    QString JahMediaPath  = QDir::homeDirPath( ) + "/.jahshaka/";
 	JahTrace *jtrace = JahTrace::getInstance();	//set up tracer
 
     QDir d( JahMediaPath+"/database/" );                        // "./example"
diff -Naur jahshaka.orig/source/Jahshaka/JahSource/jahmain/jahplayermain/jahplayer.cpp jahshaka/source/Jahshaka/JahSource/jahmain/jahplayermain/jahplayer.cpp
--- jahshaka.orig/source/Jahshaka/JahSource/jahmain/jahplayermain/jahplayer.cpp	2006-05-05 12:26:30.000000000 +0300
+++ jahshaka/source/Jahshaka/JahSource/jahmain/jahplayermain/jahplayer.cpp	2007-02-26 00:13:50.000000000 +0200
@@ -215,50 +215,12 @@
             }
     }
 
-    ///////////////////////////////////////////////////////////////
-    // now lets look to see if jahmediapath was stored in the prefs
-
-   	JahMediaPath = QString( jprefs.getMediaPath( ).c_str( ) );
-
-	// Check for the existence of the following directories in the current
-	// working directory - this will facilitate execution from the source tree
-
-	QDir media( QDir::currentDirPath( ) + "/media" );
-	QDir database( QDir::currentDirPath( ) + "/database" );
-	QDir pixmaps( QDir::currentDirPath( ) + "/Pixmaps" );
-
-	// If all 3 exist ..
-	if ( media.exists( ) && database.exists( ) && pixmaps.exists( ) )
-	{
-		// We'll use the dev tree as our base path and override media path
-		// only if the current media path is "invalid" (default from jprefs)
-		JahBasePath = QDir::currentDirPath( ) + "/";
-		if ( JahMediaPath == "invalid" )
-			JahMediaPath = QDir::currentDirPath( ) + "/";
-	}
-	else if ( !strcmp( JAHPREFIX, "" ) )
-	{
-		// On win32 and apple, JAHPREFIX is not specified
-		if ( !fileinfo.endsWith( "/" ) )
-			fileinfo.append( "/" );
-		JahBasePath = fileinfo;
-		if ( JahMediaPath == "invalid" )
-			JahMediaPath = fileinfo;
-	}
-	else
-	{
-		// Otherwise, we set the base to the installed share directory and
-		// if the current media path is invalid
-		JahBasePath = JAHPREFIX "/share/jahshaka/";
-		if ( JahMediaPath == "invalid" )
-		{
-			JahMediaPath = QDir::homeDirPath( ) + "/jahstorage/";
-			#ifndef JAHWINDOWS
-			QString command = QString( "jahuserinstall " ) + QString( JAHPREFIX ) + " " + JahMediaPath;
-			system( command );
-			#endif
-		}
-	}
+	JahBasePath = JAHPREFIX "/share/jahshaka/";
+	JahMediaPath = QDir::homeDirPath( ) + "/.jahshaka/";
+	#ifndef JAHWINDOWS
+	QString command = QString( "jahuserinstall " ) + QString( JAHPREFIX ) + " " + JahMediaPath;
+	system( command );
+	#endif
 
 	jtrace->debug( "ANTE: the application base path is " + JahBasePath );
    	jtrace->debug( "ANTE: the media path is " + JahMediaPath );
diff -Naur jahshaka.orig/source/Jahshaka/JahSource/jahmain/jahshaka.cpp jahshaka/source/Jahshaka/JahSource/jahmain/jahshaka.cpp
--- jahshaka.orig/source/Jahshaka/JahSource/jahmain/jahshaka.cpp	2006-09-15 20:14:57.000000000 +0300
+++ jahshaka/source/Jahshaka/JahSource/jahmain/jahshaka.cpp	2007-02-26 00:15:29.000000000 +0200
@@ -259,51 +259,13 @@
                 foundpath = true;
             }
     }
-
-    ///////////////////////////////////////////////////////////////
-    // now lets look to see if jahmediapath was stored in the prefs
-
-   	JahMediaPath = QString( jprefs.getMediaPath( ).c_str( ) );
-
-	// Check for the existence of the following directories in the current
-	// working directory - this will facilitate execution from the source tree
-
-	QDir media( QDir::currentDirPath( ) + "/media" );
-	QDir database( QDir::currentDirPath( ) + "/database" );
-	QDir pixmaps( QDir::currentDirPath( ) + "/Pixmaps" );
-
-	// If all 3 exist ..
-	if ( media.exists( ) && database.exists( ) && pixmaps.exists( ) )
-	{
-		// We'll use the dev tree as our base path and override media path
-		// only if the current media path is "invalid" (default from jprefs)
-		JahBasePath = QDir::currentDirPath( ) + "/";
-		if ( JahMediaPath == "invalid" )
-			JahMediaPath = QDir::currentDirPath( ) + "/";
-	}
-	else if ( !strcmp( JAHPREFIX, "" ) )
-	{
-		// On win32 and apple, JAHPREFIX is not specified
-		if ( !fileinfo.endsWith( "/" ) )
-			fileinfo.append( "/" );
-		JahBasePath = fileinfo;
-		if ( JahMediaPath == "invalid" )
-			JahMediaPath = fileinfo;
-	}
-	else
-	{
-		// Otherwise, we set the base to the installed share directory and
-		// if the current media path is invalid
-		JahBasePath = JAHPREFIX "/share/jahshaka/";
-		if ( JahMediaPath == "invalid" )
-		{
-			JahMediaPath = QDir::homeDirPath( ) + "/jahstorage/";
-			#ifndef JAHWINDOWS
-			QString command = QString( "jahuserinstall " ) + QString( JAHPREFIX ) + " " + JahMediaPath;
-			system( command );
-			#endif
-		}
-	}
+	
+	JahBasePath = JAHPREFIX "/share/jahshaka/";
+	JahMediaPath = QDir::homeDirPath( ) + "/.jahshaka/";
+	#ifndef JAHWINDOWS
+	QString command = QString( "jahuserinstall " ) + QString( JAHPREFIX ) + " " + JahMediaPath;
+	system( command );
+	#endif
 
 	jtrace->debug( "ANTE: the application base path is " + JahBasePath );
    	jtrace->debug( "ANTE: the media path is " + JahMediaPath );
