--- Makefile.orig	2008-06-10 17:23:18.000000000 +0300
+++ Makefile	2008-06-10 17:27:08.000000000 +0300
@@ -252,11 +252,11 @@
 
 install: $(PROG) nvram_export
 	@echo "Installing"
-	install -d $(DESTDIR)/bin $(DESTDIR)/man/man1 $(DESTDIR)/man/man7 $(DESTDIR)/etc
+	install -d $(DESTDIR)/bin $(DESTDIR)/share/man/man1 $(DESTDIR)/share/man/man7 $(DESTDIR)/etc
 	install dynamips nvram_export   $(DESTDIR)/bin
-	install -m644 dynamips.1        $(DESTDIR)/man/man1
-	install -m644 nvram_export.1    $(DESTDIR)/man/man1
-	install -m644 hypervisor_mode.7 $(DESTDIR)/man/man7
+	install -m644 dynamips.1        $(DESTDIR)/share/man/man1
+	install -m644 nvram_export.1    $(DESTDIR)/share/man/man1
+	install -m644 hypervisor_mode.7 $(DESTDIR)/share/man/man7
 # install -m644 example         $(DESTDIR)/etc/dynamips
 
 
