--- src/kcm/kcm_knemo.desktop	2007-03-30 00:52:13.000000000 +0300
+++ src/kcm/kcm_knemo.desktop	2007-10-22 17:01:24.000000000 +0300
@@ -9,7 +9,7 @@
 X-KDE-FactoryName=knemo
 X-KDE-ParentApp=kcontrol
 
-Name=Network Monitor
+Name=KNemo
 Name[bg]=Монитор на мрежата
 Name[br]=Eveshaer ar rouedad
 Name[cs]=Monitor sítě
@@ -33,7 +33,7 @@
 Name[sr]=Надгледање мреже
 Name[sr@Latn]=Nadgledanje mreže
 Name[sv]=Nätverksövervakare
-Name[tr]=Ağ İzleyici
+GenericName[tr]=Ağ İzleyici
 Name[uk]=Монітор мережі
 Name[xx]=xxNetwork Monitorxx
 Name[zh_CN]=网络监视器
@@ -86,4 +86,4 @@
 Keywords[xx]=xxknemo,network monitor,wlan monitor,wifi monitor,ethernet monitor,wireless monitor,systrayxx
 Keywords[zh_CN]=knemo,network monitor,wlan monitor,wifi monitor,ethernet monitor,wireless monitor,systray,网络监视器,无限局域网监视器,wifi 监视器以太网监视器,无线网络监视器,系统托盘
 
-Categories=Qt;KDE;X-KDE-settings-network;X-KDE-systemsettings-network
+Categories=Qt;KDE;X-KDE-settings-network;X-KDE-systemsettings-network;Network
--- src/knemod/knemod.desktop	2007-03-30 00:52:16.000000000 +0300
+++ src/knemod/knemod.desktop	2007-10-22 16:32:07.000000000 +0300
@@ -9,6 +9,7 @@
 Name=KNemo
 Name[de]=Netzwerküberwachung KNemo
 Name[sv]=Knemo
+GenericName[tr]=KDE Ağ İzleyici
 Name[xx]=xxKNemoxx
 Comment=KDE Network Monitor
 Comment[bg]=Монитор на мрежата

