--- data/gedit.desktop.in.orig	2007-05-06 09:28:34.000000000 +0300
+++ data/gedit.desktop.in	2007-05-06 09:42:16.000000000 +0300
@@ -7,8 +7,8 @@
 Type=Application
 StartupNotify=true
 MimeType=text/plain;
-Icon=accessories-text-editor
-Categories=GNOME;GTK;Utility;TextEditor;
+Icon=gedit
+Categories=GNOME;GTK;TextEditor;
 X-GNOME-DocPath=gedit/gedit.xml
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gedit
--- data/gedit.desktop.in.in.orig	2007-05-06 09:28:40.000000000 +0300
+++ data/gedit.desktop.in.in	2007-05-06 09:42:32.000000000 +0300
@@ -7,8 +7,8 @@
 Type=Application
 StartupNotify=true
 MimeType=text/plain;
-Icon=accessories-text-editor
-Categories=GNOME;GTK;Utility;TextEditor;
+Icon=gedit
+Categories=GNOME;GTK;TextEditor;
 X-GNOME-DocPath=gedit/gedit.xml
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gedit
