diff -Naur kicad-20090623.orig/common/gestfich.cpp kicad-20090623/common/gestfich.cpp
--- kicad-20090623.orig/common/gestfich.cpp	2009-06-23 23:35:34.000000000 +0300
+++ kicad-20090623/common/gestfich.cpp	2009-06-24 00:53:32.000000000 +0300
@@ -615,6 +615,7 @@
             /* here is a list of PDF viewers candidates */
             const static wxString tries[] =
             {
+                wxT( "/usr/bin/xdg-open" ),
                 wxT( "/usr/bin/evince" ),
                 wxT( "/usr/bin/gpdf" ),
                 wxT( "/usr/bin/konqueror" ),
