diff -Nuar gdm.orig/10XFce4.in gdm/10XFce4.in
--- gdm.orig/10XFce4.in	2008-10-10 20:41:52.000000000 +0300
+++ gdm/10XFce4.in	2008-10-10 20:34:22.000000000 +0300
@@ -1,5 +1,5 @@
 NAME=xfce4
-ICON=None
+ICON=/usr/share/xfce4/doc/xfce-mouse.png
 DESC=XFce4 Desktop Environment
 EXEC=BIN_DIR/startxfce4
 SCRIPT:
diff -Nuar gdm.orig/xfce.desktop gdm/xfce.desktop
--- gdm.orig/xfce.desktop	2008-10-10 20:41:52.000000000 +0300
+++ gdm/xfce.desktop	2008-10-10 20:36:54.000000000 +0300
@@ -9,6 +9,7 @@
 Name[ko]=Xfce 4 세션
 Name[nl]=Xfce 4 Sessie
 Name[ru]=Сеанс Xfce 4
+Name[tr]=Xfce 4 Oturumu
 Name[zh_TW]=Xfce 4 工作階段
 Comment=Use this session to run Xfce as your desktop environment
 Comment[et]=kasuta seda sessiooni, et Xfce käivituks sinu töökeskkonnana
@@ -18,7 +19,8 @@
 Comment[ko]=사용자 데스크탑 환경으로 Xfce를 이용합니다.
 Comment[nl]=Deze optie start de Xfce 4 Desktop Environment
 Comment[ru]=Запускает рабочую среду Xfce 
+Comment[tr]=Masaüstü yöneticisini Xfce olarak ayarlamak için bu oturumu seçin
 Comment[zh_TW]=使用這個工作階段載入 Xfce 4 作為您的桌面環境
 Exec=startxfce4
-Icon=
+Icon=/usr/share/xfce4/doc/xfce-mouse.png
 Type=Application
