--- wammu.desktop.in.orig	2010-01-16 15:30:30.439984752 +0200
+++ wammu.desktop.in	2010-01-16 15:32:41.953745940 +0200
@@ -1,9 +1,13 @@
 [Desktop Entry]
 _Name=Wammu ; EDIT IN wammu_setup/msgfmt.py
+Name[tr]=Wammu
 _GenericName=Mobile Phone Manager ; EDIT IN wammu_setup/msgfmt.py
+GenericName[tr]=Mobil Telefon Yöneticisi
 _Comment=Application for mobile phones - frontend for Gammu ; EDIT IN wammu_setup/msgfmt.py
+Comment[tr]=Gammu tabanlı mobil telefon yöneticisi
 Exec=wammu
 Icon=wammu
 Terminal=false
 Type=Application
 Categories=Utility;TelephonyTools;GTK;
+Categories=TelephonyTools;Calendar;ContactManagement;Office;GTK;
