diff -p -up confuse-2.6/tests/Makefile.in.Werror confuse-2.6/tests/Makefile.in
--- confuse-2.6/tests/Makefile.in.Werror	2008-02-13 13:38:52.000000000 -0200
+++ confuse-2.6/tests/Makefile.in	2008-02-13 13:38:56.000000000 -0200
@@ -244,7 +244,7 @@ TESTS = suite_single suite_dup suite_fun
 LDADD = -L../src ../src/libconfuse.la $(LTLIBINTL)
 EXTRA_DIST = a.conf b.conf check_confuse.h
 CLEANFILES = *~
-@cc_is_gcc_TRUE@AM_CFLAGS = -Wall -Werror
+@cc_is_gcc_TRUE@AM_CFLAGS = -Wall
 all: all-am
 
 .SUFFIXES:
diff -p -up confuse-2.6/src/Makefile.in.Werror confuse-2.6/src/Makefile.in
--- confuse-2.6/src/Makefile.in.Werror	2008-02-13 13:38:30.000000000 -0200
+++ confuse-2.6/src/Makefile.in	2008-02-13 13:38:35.000000000 -0200
@@ -196,7 +196,7 @@ lib_LTLIBRARIES = libconfuse.la
 libconfuse_la_SOURCES = confuse.c lexer.l
 libconfuse_la_LIBADD = $(LTLIBINTL)
 include_HEADERS = confuse.h
-@cc_is_gcc_TRUE@AM_CFLAGS = -Wall -Werror
+@cc_is_gcc_TRUE@AM_CFLAGS = -Wall
 CLEANFILES = *~ \#*\#
 AM_LFLAGS = -Pcfg_yy -olex.yy.c
 all: all-am
