diff -Nur boxes-1.1/src/boxes.h boxes-1.1.patched/src/boxes.h
--- boxes-1.1/src/boxes.h	2006-07-23 21:03:43.000000000 +0300
+++ boxes-1.1.patched/src/boxes.h	2007-06-09 11:24:02.000000000 +0300
@@ -124,7 +124,7 @@
 
 #define PROJECT "boxes"                  /* name of program */
 #define VERSION "1.1"           /* current release of project */
-#define GLOBALCONF "/usr/share/boxes"      /* name of system-wide config file */
+#define GLOBALCONF "/etc/boxes.conf"      /* name of system-wide config file */
 
 
 /*
