diff -Nuar autopano-sift-2.4.orig/src/util/autopanog/Makefile autopano-sift-2.4/src/util/autopanog/Makefile
--- autopano-sift-2.4.orig/src/util/autopanog/Makefile	2005-04-09 12:45:28.000000000 +0300
+++ autopano-sift-2.4/src/util/autopanog/Makefile	2006-12-04 23:21:23.000000000 +0200
@@ -11,7 +11,7 @@
 LIBS_WIN=/r:gtk-sharp.dll /r:glade-sharp.dll /r:atk-sharp.dll \
 	/r:gdk-sharp.dll /r:glib-sharp.dll /r:System.Drawing.dll
 MAKE_MONO=
-LIBS_MONO=/pkg:gtk-sharp /pkg:glade-sharp /resource:image-bottom-left.png \
+LIBS_MONO=/pkg:gtk-sharp-2.0 /pkg:glade-sharp-2.0 /resource:image-bottom-left.png \
 	/resource:image-bottom-right.png /resource:image-vanilla.png \
 	/r:System.Drawing.dll