--- image/cinelerra.desktop	2006-06-19 21:41:17.000000000 +0300
+++ cinelerra-yeni.desktop	2006-07-11 11:07:53.000000000 +0300
@@ -4,8 +4,10 @@
 Name[sl]=Cinelerra
 Name[fr]=Cinelerra
 Name[es]=Cinelerra
+Name[tr]=Cinelerra
 Comment=Video Editor
-Categories=Application;AudioVideo;Multimedia;
+Comment[tr]=Görüntü Düzenliyici
+Categories=AudioVideo;
 Encoding=UTF-8
 Exec=cinelerra
 Icon=cinelerra.xpm
