diff -Nuar gpdf-2.10.0.orig/help/C/Makefile.in gpdf-2.10.0/help/C/Makefile.in
--- gpdf-2.10.0.orig/help/C/Makefile.in	2007-03-07 13:08:21.000000000 +0200
+++ gpdf-2.10.0/help/C/Makefile.in	2007-03-07 13:08:30.000000000 +0200
@@ -434,7 +434,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -Nuar gpdf-2.10.0.orig/help/de/Makefile.in gpdf-2.10.0/help/de/Makefile.in
--- gpdf-2.10.0.orig/help/de/Makefile.in	2007-03-07 13:08:21.000000000 +0200
+++ gpdf-2.10.0/help/de/Makefile.in	2007-03-07 13:08:27.000000000 +0200
@@ -434,7 +434,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -Nuar gpdf-2.10.0.orig/help/es/Makefile.in gpdf-2.10.0/help/es/Makefile.in
--- gpdf-2.10.0.orig/help/es/Makefile.in	2007-03-07 13:08:21.000000000 +0200
+++ gpdf-2.10.0/help/es/Makefile.in	2007-03-07 13:08:29.000000000 +0200
@@ -434,7 +434,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -Nuar gpdf-2.10.0.orig/help/omf.make gpdf-2.10.0/help/omf.make
--- gpdf-2.10.0.orig/help/omf.make	2007-03-07 13:08:21.000000000 +0200
+++ gpdf-2.10.0/help/omf.make	2007-03-07 13:08:28.000000000 +0200
@@ -42,7 +42,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
