--- linden/indra/newview/llvoiceclient.cpp.orig	2007-08-29 13:55:21.000000000 +0300
+++ linden/indra/newview/llvoiceclient.cpp	2007-08-29 16:58:50.000000000 +0300
@@ -36,7 +36,7 @@
 #include "llvoavatar.h"
 #include "llbufferstream.h"
 #include "llfile.h"
-#include "expat/expat.h"
+#include "expat.h"
 #include "llcallbacklist.h"
 #include "llviewerregion.h"
 #include "llviewernetwork.h"		// for gUserServerChoice
