diff -Nur gnomebaker-0.6.1/src/execfunctions.c gnomebaker-0.6.1.patched/src/execfunctions.c
--- gnomebaker-0.6.1/src/execfunctions.c	2006-06-09 00:38:26.000000000 +0300
+++ gnomebaker-0.6.1.patched/src/execfunctions.c	2007-09-01 16:00:57.000000000 +0300
@@ -226,6 +226,9 @@
     GError *err = NULL;
     gchar *tmp = preferences_get_string(GB_TEMP_DIR);
     GDir *dir = g_dir_open(tmp, 0, &err);
+    GList *files_list = NULL;
+    GList *p = NULL;
+
     if(dir != NULL)
     {
         cdrecord_total_tracks_to_write = 0;
@@ -235,16 +238,23 @@
         {
             if(g_str_has_suffix(name, ".wav"))
             {
-                GB_TRACE("cdrecord_copy_audio_cd_pre_proc - adding [%s]\n", name);
+                GB_TRACE("wodim_copy_audio_cd_pre_proc - found [%s]\n", name);
                 gchar *full_path = g_build_filename(tmp, name, NULL);
-                exec_cmd_add_arg(ex, full_path);
+                files_list = g_list_insert_sorted(files_list, full_path, strcmp);
                 cdrecord_total_tracks_to_write++;
-                g_free(full_path);
             }
 
             name = g_dir_read_name(dir);
         }
         g_dir_close(dir);
+
+        for (p = files_list; p != NULL; p = g_list_next(p)) {
+            GB_TRACE("wodim_copy_audio_cd_pre_proc - adding [%s]\n", p->data);
+            exec_cmd_add_arg(ex, p->data);
+            g_free(p->data);
+            p->data = NULL;
+        }
+        g_list_free(files_list);
     }
     g_free(tmp);
 }
