--- lock-keyboard-for-baby-20080706.pl	2010-05-08 01:20:32.811580422 +0300
+++ lock-keyboard-for-baby-20080706.pl	2010-05-08 01:31:43.624329496 +0300
@@ -1,10 +1,11 @@
-#!/bin/env perl
+#!/usr/bin/perl
 use warnings;
 use strict;
+use utf8;
 use Data::Dumper;
 my $lastmod="2006/05/25";
 my $datemod="2008/07/03";
-my $defaultpassword="QuitNow";
+my $defaultpassword="Aktif";
 my $progname=$0;
 $progname =~ s%.*/%%g;
 
@@ -44,7 +44,7 @@
 
 
 my $password=$defaultpassword;
-my $message="Type the password to quit\n:";
+my $message="Klavyeyi etkinleştirmek için parolayı yazın\n:";
 my $true=1;
 my $false=0;
 my $noshow=$true;
@@ -122,7 +122,7 @@
 }
 if((!$defaults_changed || $defaults_changed_vis))
 {
-  $message="Type '$password' to quit\n";
+  $message="Klavyeyi etkinleştirmek için '$password' yazın\n";
 }
 
 use Gtk2 -init;
