diff -Nuar xfce-utils-4.4.0.orig/scripts/startxfce4.in xfce-utils-4.4.0/scripts/startxfce4.in
--- xfce-utils-4.4.0.orig/scripts/startxfce4.in	2007-02-28 00:10:12.000000000 +0200
+++ xfce-utils-4.4.0/scripts/startxfce4.in	2007-03-01 17:39:58.000000000 +0200
@@ -28,9 +28,9 @@
 
 if test x"$XDG_DATA_DIRS" = x""
 then
-  XDG_DATA_DIRS="/usr/share:/usr/local/share:@datadir@"
+  XDG_DATA_DIRS="/usr/share:/usr/local/share:/usr/kde/3.5/share:@datadir@"
 else
-  XDG_DATA_DIRS="$XDG_DATA_DIRS:@datadir@"
+  XDG_DATA_DIRS="$XDG_DATA_DIRS:/usr/kde/3.5/share:@datadir@"
 fi
 export XDG_DATA_DIRS
 
