--- scripts/startxfce4.in.orig	2007-11-17 21:31:32.000000000 +0200
+++ scripts/startxfce4.in	2007-12-06 08:48:27.000000000 +0200
@@ -28,9 +28,9 @@
 
 if test "x$XDG_DATA_DIRS" = "x"
 then
-  XDG_DATA_DIRS="/usr/local/share:/usr/share:@datadir@"
+  XDG_DATA_DIRS="/usr/local/share:/usr/share:/usr/kde/3.5/share:@datadir@"
 else
-  XDG_DATA_DIRS="$XDG_DATA_DIRS:@datadir@"
+  XDG_DATA_DIRS="$XDG_DATA_DIRS:@datadir@:/usr/kde/3.5/share"
 fi
 export XDG_DATA_DIRS
 
