--- configure.orig	2005-11-05 01:05:33.000000000 +0200
+++ configure	2006-09-30 15:58:15.000000000 +0300
    { (exit 1); exit 1; }; }
@@ -21883,7 +21883,7 @@
     fi
     GRASS_LIB="-L$with_grass/${_lib} -lgrass_vect -lgrass_dig2 -lgrass_dbmiclient -lgrass_dbmibase -lgrass_shape -lgrass_dgl -lgrass_rtree -lgrass_gis -lgrass_datetime -lgrass_linkm -lgrass_form -lgrass_gproj"
 
-    GISINC=`ls $with_grass/include/gis.h`
+    GISINC=`ls $with_grass/include/grass/gis.h`
     if test ! -f "$GISINC"; then
   { { echo "$as_me:$LINENO: error: GRASS headers not found " >&5
 echo "$as_me: error: GRASS headers not found " >&2;}
