diff -Naur xfce4-settings-4.5.99.1/dialogs/display-settings/main.c xfce4-settings-4.5.99.1.tpg/dialogs/display-settings/main.c
--- xfce4-settings-4.5.99.1/dialogs/display-settings/main.c	2009-01-25 21:27:17.000000000 +0000
+++ xfce4-settings-4.5.99.1.tpg/dialogs/display-settings/main.c	2009-01-26 23:52:49.000000000 +0000
@@ -717,7 +717,7 @@
                 /* show an error dialog the version is too old */
                 dialog = gtk_message_dialog_new (NULL, 0, GTK_MESSAGE_ERROR, GTK_BUTTONS_NONE,
                                                  _("Failed to use the RandR extension"));
-                gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dialog), error->message);
+                gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dialog), "%s", error->message);
                 gtk_dialog_add_button (GTK_DIALOG (dialog), GTK_STOCK_QUIT, GTK_RESPONSE_CLOSE);
                 gtk_dialog_run (GTK_DIALOG (dialog));
                 gtk_widget_destroy (dialog);
