--- comix.desktop.orig	2009-02-06 16:54:07.000000000 +0200
+++ comix.desktop	2009-02-06 16:55:03.000000000 +0200
@@ -2,15 +2,18 @@
 Encoding=UTF-8
 Name=Comix
 Name[it]=Comix
+Name[tr]=Comix
 GenericName=Comic Book Viewer
 GenericName[sv]=Serieboksläsare
 GenericName[fr]=Visionneur de bandes dessinées
+GenericName[tr]=Çizgi Roman Görüntüleyici
 Comment=A viewer for comic book archives
 Comment[it]=Un visualizzatore di fumetti 
 Comment[es]=Un visor de comics
 Comment[sv]=En serieboksläsare
 Comment[fr]=Visionneur d'images spécialisé dans la lecture des bandes dessinées
 Comment[pl]=Przeglądarka komiksów
+Comment[tr]=Çizgi roman arşivleri için görüntüleyici
 Exec=comix %f
 Icon=comix
 Terminal=false
