--- kmymoney2.desktop	2008-04-03 09:57:21.000000000 +0300
+++ kmymoney2/kmymoney2.desktop	2008-07-05 10:59:29.000000000 +0300
@@ -3,12 +3,14 @@
 GenericName=Personal Finance Manager
 GenericName[de]=Persönlicher Finanzassistent
 GenericName[fr]=Gestionnaire de finances personnelles
+GenericName[tr]=Kişisel Finans Yöneticisi
 GenericName[ru]=Учёт финансов
 GenericName[zh_CN]=个人财务管理软件
 Exec=kmymoney2 %u
 Comment=Personal Finance Manager
 Comment[de]=Persönlicher Finanzassistent
 Comment[fr]=Gestionnaire de finances personnelles
+Comment[tr]=Kişisel Finans Yöneticisi
 Comment[ru]=Учёт финансов
 Comment[zh_CN]=个人财务管理软件
 MimeType=application/x-kmymoney;application/vnd.intu.qfx;application/x-ofx;
