--- nt.desktop	2005-08-31 16:32:13.000000000 +0300
+++ share/nt.desktop	2006-06-28 00:06:46.000000000 +0300
@@ -1,11 +1,14 @@
 [Desktop Entry]
-Name=Downloader for X
+Name=D4X
 Name[cs]=Downloader pro X
 Name[fr]=Downloader for X
 Name[es]=Downloader para X
 Name[de]=Downloader für X
 Name[pl]=Downloader dla X
 Name[ru]=Качалка для Иксов
+Name[tr]=D4X
+GenericName=Downloader for X
+GenericName[tr]=X İçin İndirme Yöneticisi
 Comment=downloading program
 Comment[cs]=Program pro stahování souborů
 Comment[fr]=Programme de téléchargement de fichiers
@@ -13,9 +16,11 @@
 Comment[de]=Programm zum Downloaden von Dateien
 Comment[pl]=Program do pobierania plików z Internetu.
 Comment[ru]=программа для скачивания файлов
+Comment[tr]=İndirme yöneticisi
 Icon=nt.png
 Exec=d4x
 TryExec=d4x
 Terminal=false
 Type=Application
 Encoding=UTF-8
+Categories=Applications;Network;
