diff -Naur subtitleeditor-0.12.4-orig/share/subtitleeditor.desktop subtitleeditor-0.12.4/share/subtitleeditor.desktop
--- subtitleeditor-0.12.4-orig/share/subtitleeditor.desktop	2006-11-13 23:28:45.000000000 +0200
+++ subtitleeditor-0.12.4/share/subtitleeditor.desktop	2006-12-16 02:53:53.000000000 +0200
@@ -1,11 +1,14 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Subtitle Editor
+Name[tr]=Altyazı Düzenleyici
 GenericName=Subtitle Editor
+GenericName[tr]=Altyazı Düzenleyici
 Comment=A subtitle editor based to GStreamer and Gtk+
+Comment[tr]=Altyazı dosyalarını düzenlemeye yarayan GStreamer ve Gtk+ tabanlı uygulama
 Exec=subtitleeditor
 Terminal=false
 Type=Application
-Icon=subtitleeditor-icon.png
+Icon=/usr/share/subtitleeditor/subtitleeditor-icon.png
 Categories=Application;AudioVideo;
 StartupNotify=true
