--- base/autopsy.base.erdinc	2006-09-01 19:13:46.000000000 +0300
+++ base/autopsy.base	        2006-09-18 13:43:19.000000000 +0300
@@ -1,4 +1,7 @@
-#
+#!/usr/bin/perl -wT
+use lib '/usr/lib/autopsy/';
+use lib '/usr/lib/autopsy/lib/';
+
 # $Date: 2006/09/01 15:23:53 $
 #
 # autopsy gui server
@@ -57,7 +60,7 @@
 require Fs;
 require Caseman;
 
-require 'conf.pl';
+require '/etc/autopsy.pl';
 require 'lib/define.pl';
 
 # Import variables from conf.pl
