diff -Nuar libgtkhtml-2.11.1.orig/libgtkhtml/Makefile.in libgtkhtml-2.11.1/libgtkhtml/Makefile.in
--- libgtkhtml-2.11.1.orig/libgtkhtml/Makefile.in	2009-06-18 00:27:41.000000000 +0300
+++ libgtkhtml-2.11.1/libgtkhtml/Makefile.in	2009-06-18 00:29:06.000000000 +0300
@@ -239,7 +239,7 @@
 	$(top_builddir)/libgtkhtml/layout/libhtmllayout.la \
 	$(top_builddir)/libgtkhtml/util/libhtmlutil.la \
 	$(top_builddir)/libgtkhtml/view/libhtmlview.la \
-	$(A11Y_LT_LIB) $(GTKHTML2_LIBS) $(GAILUTIL_LIBS)
+	$(A11Y_LT_LIB) $(GTKHTML2_LIBS) $(GAILUTIL_LIBS) -lm
 
 MAINTAINERCLEANFILES = \
 	Makefile.in
