--- data/filezilla.desktop.orig	2008-01-12 00:46:26.000000000 +0200
+++ data/filezilla.desktop	2008-01-13 14:59:11.000000000 +0200
@@ -1,7 +1,9 @@
 [Desktop Entry]
-Name=FileZilla FTP client
+Name=FileZilla
+GenericName=FTP and SFTP client
+GenericName[tr]=FTP ve SFTP istemcisi
 Comment=Download and upload files via ftp, sftp and ftps
-Comment[fr]=Transférer des fichiers via ftp, sftp et ftps
+Comment[tr]=Dosyaları ftp, sftp, ftps ile gönderin ve indirin
 Exec=filezilla
 Terminal=false
 Icon=filezilla.png
