--- htdocs/chklang.php	2006-08-25 16:18:35.000000000 +0300
+++ chklang-patch.php	2006-08-25 16:20:01.000000000 +0300
@@ -61,6 +61,7 @@
   'se'         => array('se|swedish', 'swedish'),
   'sr'         => array('sr|serbian', 'serbian-win1250'),
   'th'         => array('th|thai', 'thai'),
+  'tr'         => array('tr|Turkish', 'iso-8859-9'),
   'zh-tw'      => array('zh[-_]tw|chinese traditional', 'chinese_big5'),
   'zh'         => array('zh|chinese simplified', 'chinese_gb')
 );
