diff -Naur Miro-2.5-rc1.orig/platform/gtk-x11/miro.desktop Miro-2.5-rc1/platform/gtk-x11/miro.desktop
--- Miro-2.5-rc1.orig/platform/gtk-x11/miro.desktop	2009-07-08 23:11:09.000000000 +0300
+++ Miro-2.5-rc1/platform/gtk-x11/miro.desktop	2009-07-08 23:13:22.000000000 +0300
@@ -2,7 +2,9 @@
 Type=Application
 Name=Miro Internet TV
 GenericName=Internet TV
+GenericName[tr]=İnternet TV
 Comment=Watch online video
+Comment[tr]=Çevrimiçi video izleyin, indirin
 Comment[de]=Multimedia Player
 Comment[es]=Reproductor multimedia
 Comment[fr]=Lecteur multimedia
