diff -Naur Miro-2.5.1.orig/platform/gtk-x11/plat/screensaver.py Miro-2.5.1/platform/gtk-x11/plat/screensaver.py
--- Miro-2.5.1.orig/platform/gtk-x11/plat/screensaver.py	2009-07-27 21:52:38.358174971 +0300
+++ Miro-2.5.1/platform/gtk-x11/plat/screensaver.py	2009-07-27 21:55:50.873175377 +0300
@@ -76,10 +76,10 @@
         return rc == 0
 
     def should_use(self):
-        return call_xss('-time')
+        return self.call_xss('-time')
 
     def deactivate(self):
-        return call_xss('-deactivate')
+        return self.call_xss('-deactivate')
 
     def disable(self):
         self.timer = gobject.timer_add(1000, self.deactivate)
