diff -Naur xmms-1.2.10-20041012/libxmms/xmmsctrl.c xmms-1.2.10-20041012.jump/libxmms/xmmsctrl.c
--- xmms-1.2.10-20041012/libxmms/xmmsctrl.c	2004-10-13 01:03:03.272232621 -0700
+++ xmms-1.2.10-20041012.jump/libxmms/xmmsctrl.c	2004-10-13 01:09:58.861918746 -0700
@@ -383,6 +383,11 @@
 	remote_cmd(session, CMD_PLAY_PAUSE);
 }
 
+void xmms_remote_show_jump_box(int session)
+{
+	remote_cmd(session, CMD_SHOW_JUMP_BOX);
+}
+
 gboolean xmms_remote_is_playing(int session)
 {
 	return remote_get_gboolean(session, CMD_IS_PLAYING);
diff -Naur xmms-1.2.10-20041012/libxmms/xmmsctrl.h xmms-1.2.10-20041012.jump/libxmms/xmmsctrl.h
--- xmms-1.2.10-20041012/libxmms/xmmsctrl.h	2004-10-13 01:03:03.274232292 -0700
+++ xmms-1.2.10-20041012.jump/libxmms/xmmsctrl.h	2004-10-13 01:08:27.542919721 -0700
@@ -61,6 +61,7 @@
 gboolean xmms_remote_is_pl_win(gint session);
 gboolean xmms_remote_is_eq_win(gint session);
 void xmms_remote_show_prefs_box(gint session);
+void xmms_remote_show_jump_box(gint session);
 void xmms_remote_toggle_aot(gint session, gboolean ontop);
 void xmms_remote_eject(gint session);
 void xmms_remote_playlist_prev(gint session);
diff -Naur xmms-1.2.10-20041012/xmms/controlsocket.c xmms-1.2.10-20041012.jump/xmms/controlsocket.c
--- xmms-1.2.10-20041012/xmms/controlsocket.c	2004-10-13 01:03:03.699162382 -0700
+++ xmms-1.2.10-20041012.jump/xmms/controlsocket.c	2004-10-13 01:08:27.545919229 -0700
@@ -523,6 +523,9 @@
 			case CMD_SET_SKIN:
 				load_skin(data);
 				break;
+            case CMD_SHOW_JUMP_BOX:
+               show_jump_to_file();
+               break;
 			case CMD_PL_WIN_TOGGLE:
 				tbool = *((gboolean *) data);
 				playlistwin_show(!!tbool);
diff -Naur xmms-1.2.10-20041012/xmms/controlsocket.h xmms-1.2.10-20041012.jump/xmms/controlsocket.h
--- xmms-1.2.10-20041012/xmms/controlsocket.h	2004-10-13 01:03:03.701162053 -0700
+++ xmms-1.2.10-20041012.jump/xmms/controlsocket.h	2004-10-13 01:11:15.408348579 -0700
@@ -46,7 +46,8 @@
 	CMD_GET_EQ, CMD_GET_EQ_PREAMP, CMD_GET_EQ_BAND,
 	CMD_SET_EQ, CMD_SET_EQ_PREAMP, CMD_SET_EQ_BAND,
 	CMD_QUIT, CMD_PLAYLIST_INS_URL_STRING, CMD_PLAYLIST_INS, CMD_PLAY_PAUSE,
-	CMD_PLAYQUEUE_ADD, CMD_GET_PLAYQUEUE_LENGTH, CMD_PLAYQUEUE_REMOVE
+	CMD_PLAYQUEUE_ADD, CMD_GET_PLAYQUEUE_LENGTH, CMD_PLAYQUEUE_REMOVE,
+	CMD_SHOW_JUMP_BOX
 };
 
 typedef struct
diff -Naur xmms-1.2.10-20041012/xmms/main.c xmms-1.2.10-20041012.jump/xmms/main.c
--- xmms-1.2.10-20041012/xmms/main.c	2004-10-13 01:03:03.716159586 -0700
+++ xmms-1.2.10-20041012.jump/xmms/main.c	2004-10-13 01:14:11.538439019 -0700
@@ -1907,6 +1907,11 @@
 	GTK_CLIST(clist)->focus_row = GPOINTER_TO_INT(GTK_CLIST(clist)->selection->data);
 }
 
+void show_jump_to_file(void)
+{
+    mainwin_jump_to_file();
+}
+
 static gboolean mainwin_configure(GtkWidget * window, GdkEventConfigure *event, gpointer data)
 {
 	if (!GTK_WIDGET_VISIBLE(window))
@@ -3269,6 +3274,7 @@
 	{"play-pause", 0, NULL, 't'},
 	{"stop", 0, NULL, 's'},
 	{"fwd", 0, NULL, 'f'},
+	{"show-jump-box", 0, NULL, 'j'},
 	{"enqueue", 0, NULL, 'e'},
 	{"queue", 0, NULL, 'Q'},
 	{"toggle-shuffle", 2, NULL, 'S'},
@@ -3310,6 +3316,9 @@
 	fprintf(stderr, "\n  -f, --fwd			");
 	/* I18N: -f, --fwd switch */
 	fprintf(stderr, _("Skip forward in playlist"));
+	fprintf(stderr, "\n-j, --show-jump-box  ");
+	/* -j, --show-jump-box switch */
+	fprintf(stderr, _("Display Jump to file dialog"));
 	fprintf(stderr, "\n  -e, --enqueue			");
 	/* I18N: -e, --enqueue switch */
 	fprintf(stderr, _("Don't clear the playlist"));
@@ -3351,7 +3360,7 @@
 struct cmdlineopt {
 	GList *filenames;
 	int session;
-	gboolean play, stop, pause, fwd, rew, play_pause;
+	gboolean play, stop, pause, fwd, rew, play_pause, show_jump_box;
 	gboolean toggle_shuffle, toggle_repeat, toggle_advance;
 	gboolean enqueue, queue, mainwin, remote, quit;
 	char *shuffle, *repeat, *advance;
@@ -3365,7 +3374,7 @@
 
 	memset(opt, 0, sizeof(struct cmdlineopt));
 	opt->session = -1;
-	while ((c = getopt_long(argc, argv, "hn:rpusfeQS::R::A::mi:vtq", long_options, NULL)) != -1)
+	while ((c = getopt_long(argc, argv, "hn:rpusfeQS::R::A::mi:vtqj", long_options, NULL)) != -1)
 	{
 		switch (c)
 		{
@@ -3410,6 +3419,9 @@
 					opt->advance = g_strdup(optarg);
 				break;
 				break;
+			case 'j':
+				opt->show_jump_box = TRUE;
+				break;
 			case 'm':
 				opt->mainwin = TRUE;
 				break;
@@ -3498,6 +3510,8 @@
 		xmms_remote_playlist_next(opt->session);
 	if (opt->play_pause)
 		xmms_remote_play_pause(opt->session);
+	if (opt->show_jump_box)
+		xmms_remote_show_jump_box(opt->session);
 	if (opt->toggle_shuffle)
 	{
 		if (opt->shuffle)
diff -Naur xmms-1.2.10-20041012/xmms/main.h xmms-1.2.10-20041012.jump/xmms/main.h
--- xmms-1.2.10-20041012/xmms/main.h	2004-10-13 01:03:03.718159257 -0700
+++ xmms-1.2.10-20041012.jump/xmms/main.h	2004-10-13 01:08:27.562916436 -0700
@@ -119,6 +119,7 @@
 void mainwin_set_shade_menu_cb(gboolean shaded);
 void mainwin_set_shade(gboolean shaded);
 void mainwin_shade_toggle(void);
+void show_jump_to_file(void);
 
 #define PLAYER_HEIGHT ((cfg.player_shaded ? 14 : 116) * (cfg.doublesize + 1))
 #define PLAYER_WIDTH (275 * (cfg.doublesize + 1))
