diff -Naurp xmms-1.2.10.orig/xmms/playlist.c xmms-1.2.10/xmms/playlist.c
--- xmms-1.2.10.orig/xmms/playlist.c	2005-05-27 02:09:59.000000000 +0000
+++ xmms-1.2.10/xmms/playlist.c	2005-05-27 02:10:47.000000000 +0000
@@ -43,6 +43,7 @@ static void playlist_load_ins_file(char 
 static int __get_playlist_length(void);
 static void playlist_generate_shuffle_list(void);
 static void __playlist_generate_shuffle_list(void);
+static GList *playlist_shuffle_list(GList *list);
 
 struct devino {
 	dev_t dev;
@@ -1576,8 +1577,12 @@ static GList* playlist_sort_selected(GLi
 		}
 		list1 = list2;
 	}
-
-	temp_list = g_list_sort(temp_list, cmpfunc);
+	
+	if (cmpfunc)
+		temp_list = g_list_sort(temp_list, cmpfunc);
+	else
+		temp_list = playlist_shuffle_list(temp_list);
+		
 	list1 = temp_list;
 	list2 = index_list;
 
@@ -1693,6 +1698,13 @@ void playlist_random(void)
 	PL_UNLOCK();
 }
 
+void playlist_random_selected(void)
+{
+	PL_LOCK();
+	playlist = playlist_sort_selected(playlist, NULL);
+	PL_UNLOCK();
+}
+
 GList * playlist_get_selected(void)
 {
 	GList *node, *list = NULL;
diff -Naurp xmms-1.2.10.orig/xmms/playlist.h xmms-1.2.10/xmms/playlist.h
--- xmms-1.2.10.orig/xmms/playlist.h	2005-05-27 02:09:59.000000000 +0000
+++ xmms-1.2.10/xmms/playlist.h	2005-05-27 02:10:47.000000000 +0000
@@ -77,6 +77,7 @@ void playlist_sort_selected_by_date(void
 void playlist_sort_selected_by_length(void);
 void playlist_reverse(void);
 void playlist_random(void);
+void playlist_random_selected();
 void playlist_remove_dead_files(void);
 void playlist_fileinfo_current(void);
 void playlist_fileinfo(gint pos);
diff -Naurp xmms-1.2.10.orig/xmms/playlistwin.c xmms-1.2.10/xmms/playlistwin.c
--- xmms-1.2.10.orig/xmms/playlistwin.c	2005-05-27 02:10:01.000000000 +0000
+++ xmms-1.2.10/xmms/playlistwin.c	2005-05-27 02:10:47.000000000 +0000
@@ -88,7 +88,8 @@ enum
 	PLAYLISTWIN_SORT_SEL_BYTITLE, PLAYLISTWIN_SORT_SEL_BYFILENAME,
 	PLAYLISTWIN_SORT_SEL_BYPATH, PLAYLISTWIN_SORT_SEL_BYDATE,
 	PLAYLISTWIN_SORT_SEL_BYLENGTH,
-	PLAYLISTWIN_SORT_RANDOMIZE, PLAYLISTWIN_SORT_REVERSE
+	PLAYLISTWIN_SORT_RANDOMIZE, PLAYLISTWIN_SORT_REVERSE,
+	PLAYLISTWIN_SORT_SEL_RANDOMIZE
 };
 
 GtkItemFactoryEntry playlistwin_sort_menu_entries[] =
@@ -118,6 +119,8 @@ GtkItemFactoryEntry playlistwin_sort_men
 	{"/-", NULL, NULL, 0, "<Separator>"},
 	{N_("/Randomize List"), NULL, playlistwin_sort_menu_callback,
 					PLAYLISTWIN_SORT_RANDOMIZE, "<Item>"},
+	{N_("/Randomize Selection"), NULL, playlistwin_sort_menu_callback,
+					PLAYLISTWIN_SORT_SEL_RANDOMIZE, "<Item>"},
 	{N_("/Reverse List"), NULL, playlistwin_sort_menu_callback,
 					PLAYLISTWIN_SORT_REVERSE, "<Item>"},
 };
@@ -1802,6 +1805,10 @@ static void playlistwin_sort_menu_callba
 			playlist_random();
 			playlistwin_update_list();
 			break;
+		case PLAYLISTWIN_SORT_SEL_RANDOMIZE:
+			playlist_random_selected();
+			playlistwin_update_list();
+			break;
 	}
 }
 
diff -Naurp xmms-1.2.10.orig/xmms/prefswin.c xmms-1.2.10/xmms/prefswin.c
--- xmms-1.2.10.orig/xmms/prefswin.c	2005-05-27 02:10:01.000000000 +0000
+++ xmms-1.2.10/xmms/prefswin.c	2005-05-27 02:10:47.000000000 +0000
@@ -1469,6 +1469,8 @@ void show_prefs_window(void)
 
 	gtk_entry_set_text(GTK_ENTRY(prefswin_audiocd_cddadirectory), cfg.cdda_directory);
 
+	gtk_entry_set_text(GTK_ENTRY(prefswin_audiocd_cddadirectory), cfg.cdda_directory);
+
  	gtk_widget_show_all(prefswin);
 	gtk_widget_grab_default(prefswin_ok);
 
