--- xmms-1.2.10-org/Input/mpg123/mpg123.c	2005-10-17 18:07:52.000000000 +0200
+++ xmms-1.2.10/Input/mpg123/mpg123.c	2005-10-24 15:19:49.000000000 +0200
@@ -696,8 +696,6 @@
 		return FALSE;
 	for (;;)
 	{
-		int offset;
-		
 		while (!mpg123_head_check(h) ||
 		       !mpg123_decode_header(frm, h))
 		{
@@ -708,15 +706,6 @@
 				return FALSE;
 		}
 
-		offset = frm->framesize;
-		if (fseek(fh, offset, SEEK_CUR) || !head_read(fh, &h2))
-			return FALSE;
-		
-		if (fseek(fh, -(offset + 4), SEEK_CUR))
-			return FALSE;
-
-		if (mpg123_head_check(h2) &&
-		    mpg123_decode_header(frm, h2))
 		{
 			if (fseek(fh, -4, SEEK_CUR))
 				return FALSE;
