diff -Nuar cppunit-1.12.0.orig/src/qttestrunner/qttestrunnerlib.pro cppunit-1.12.0/src/qttestrunner/qttestrunnerlib.pro
--- cppunit-1.12.0.orig/src/qttestrunner/qttestrunnerlib.pro	2005-12-12 09:40:09.000000000 +0200
+++ cppunit-1.12.0/src/qttestrunner/qttestrunnerlib.pro	2007-06-30 11:18:24.000000000 +0300
@@ -15,11 +15,11 @@
 CONFIG -= dll
 CONFIG -= staticlib
 
-CONFIG += qt warn_on debug staticlib
+#CONFIG += qt warn_on debug staticlib
 
 #CONFIG += qt warn_on release staticlib
 #CONFIG += qt warn_on debug dll
-#CONFIG += qt warn_on release dll
+CONFIG += qt warn_on release dll
 
 
 QTRUNNER_LIB = qttestrunner   # Name of the library
@@ -84,16 +84,15 @@
     MOC_DIR = .moc
     UI_DIR = .moc
     DESTDIR = ../../lib
+    LIBS += -L../../src/cppunit/.libs -lcppunit
     dll {
         debug {
-            TARGET = $${QTRUNNER_LIB}d_shared
+            TARGET = $${QTRUNNER_LIB}d
             OBJECTS_DIR = .obj_debug_shared
-            LIBS += -L../../lib -lcppunit
         }
         release {
-            TARGET = $${QTRUNNER_LIB}_shared
+            TARGET = $${QTRUNNER_LIB}
             OBJECTS_DIR = .obj_release_shared
-            LIBS += -L../../lib -lcppunit
         }
     }
     staticlib {
