--- libtorrent-rasterbar-0.14.1/src/Makefile.am	2008-10-18 00:09:47.000000000 -0700
+++ libtorrent-rasterbar-0.14.1/src/Makefile.am.new	2008-12-14 15:08:04.000000000 -0800
@@ -105,7 +105,7 @@
 
 
 libtorrent_rasterbar_la_LDFLAGS = $(LDFLAGS) -version-info $(INTERFACE_VERSION_INFO)
-libtorrent_rasterbar_la_LIBADD = @ZLIB@ @BOOST_SYSTEM_LIB@ @BOOST_FILESYSTEM_LIB@ @BOOST_THREAD_LIB@ @PTHREAD_LIBS@ @SSL_LIBS@
+libtorrent_rasterbar_la_LIBADD = @ZLIB@ @BOOST_SYSTEM_LIB@ @BOOST_FILESYSTEM_LIB@ @BOOST_THREAD_LIB@ @PTHREAD_LIBS@ @SSL_LIBS@ -lpthread -lrt
 
 AM_CXXFLAGS= -ftemplate-depth-100 -I$(top_srcdir)/include -I$(top_srcdir)/include/libtorrent @ZLIBINCL@ @DEBUGFLAGS@ @PTHREAD_CFLAGS@ -DBOOST_MULTI_INDEX_DISABLE_SERIALIZATION @SSL_CXXFLAGS@
 AM_CFLAGS= -I$(top_srcdir)/include -I$(top_srcdir)/include/libtorrent @ZLIBINCL@ @DEBUGFLAGS@ @PTHREAD_CFLAGS@
--- libtorrent-rasterbar-0.14.4/bindings/python/setup.py.in~	2009-05-02 20:35:38.000000000 +0200
+++ libtorrent-rasterbar-0.14.4/bindings/python/setup.py.in	2009-08-21 06:49:44.000000000 +0200
@@ -80,5 +80,5 @@
 		library_dirs = ['@top_builddir@/src/.libs'] + parse_cmd(extra_cmd, '-L'),
 		extra_link_args = '@LDFLAGS@'.split() + arch(),
 		extra_compile_args = parse_cmd(extra_cmd, '-D', True) + arch() + ['-DBOOST_MULTI_INDEX_DISABLE_SERIALIZATION'],
-		libraries = ['torrent-rasterbar','@BOOST_PYTHON_LIB@'] + parse_cmd(extra_cmd, '-l'))],
+		libraries = ['torrent-rasterbar','@BOOST_PYTHON_LIB@','pthread'] + parse_cmd(extra_cmd, '-l'))],
 )
