diff -Naur opera-10.10-4742.gcc4-qt4.i386.orig/usr/share/opera/defaults/pluginpath.ini opera-10.10-4742.gcc4-qt4.i386/usr/share/opera/defaults/pluginpath.ini
--- opera-10.10-4742.gcc4-qt4.i386.orig/usr/share/opera/defaults/pluginpath.ini	2010-01-05 02:57:56.205235185 +0200
+++ opera-10.10-4742.gcc4-qt4.i386/usr/share/opera/defaults/pluginpath.ini	2010-01-05 03:01:34.877235252 +0200
@@ -60,6 +60,7 @@
 $HOME/.netscape/plugins=1
 /opt/netscape/plugins=1
 /usr/lib/netscape/plugins=1
+/usr/lib/nsbrowser/plugins=1
 /usr/local/netscape/plugins=1
 /usr/local/lib/netscape/plugins=1
 $HOME/.kde/.konqueror/nsplugins=1
