--- install.orig	2010-07-05 15:29:13.970893819 +0300
+++ install	2010-07-05 15:48:09.646015902 +0300
@@ -75,21 +75,6 @@
 	"$@" NF bcdb530c010ed72fe6096d46366c57f5 share/opera@@{SUFFIX}/extra/svg-sebd.dat
 	"$@" NF 7a2db2de5211ab16a6712e4168d17ec7 share/opera@@{SUFFIX}/html40_entities.dtd
 	"$@" NF 87803f0514670b96c73d4043cd3568c8 share/opera@@{SUFFIX}/lngcode.txt
-	"$@" NF b589bfe4b77557c00f1ffdafeffd2beb share/opera@@{SUFFIX}/locale/be/be.lng
-	"$@" NF 19a88943998948a20598ebe4dce9b898 share/opera@@{SUFFIX}/locale/be/bookmarks.adr
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/be/license.txt
-	"$@" NF 2f3c9d71ea6cc8c7ee7a0ae56b0f2ebd share/opera@@{SUFFIX}/locale/be/search.ini
-	"$@" NF 8beaab8ab5f5af7f0b9e52b31b35bea1 share/opera@@{SUFFIX}/locale/be/standard_speeddial.ini
-	"$@" NF f7a34112990b5585d1944e42aa0ee3bc share/opera@@{SUFFIX}/locale/bg/bg.lng
-	"$@" NF fbaacb092f6fb389ba57f810d31792bd share/opera@@{SUFFIX}/locale/bg/bookmarks.adr
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/bg/license.txt
-	"$@" NF a14baaa7a62b0181553244bbbcfa6408 share/opera@@{SUFFIX}/locale/bg/search.ini
-	"$@" NF d1ed970ff7b137d478bb94301da078e3 share/opera@@{SUFFIX}/locale/bg/standard_speeddial.ini
-	"$@" NF 2218692cb6ccf82543756b594f1dd444 share/opera@@{SUFFIX}/locale/cs/bookmarks.adr
-	"$@" NF 0bbed03067f25a7a13ab6478c2b47227 share/opera@@{SUFFIX}/locale/cs/cs.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/cs/license.txt
-	"$@" NF 4ee0580554e7c1ddc24d609bbb5c7253 share/opera@@{SUFFIX}/locale/cs/search.ini
-	"$@" NF d1ed970ff7b137d478bb94301da078e3 share/opera@@{SUFFIX}/locale/cs/standard_speeddial.ini
 	"$@" NF a749b04c285fb212f8048440a1dd367d share/opera@@{SUFFIX}/locale/da/da.lng
 	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/da/license.txt
 	"$@" NF 2abbf8026ccfde44516071ca973fb618 share/opera@@{SUFFIX}/locale/de/bookmarks.adr
@@ -97,8 +82,6 @@
 	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/de/license.txt
 	"$@" NF 42efa4b67073e9fbf848505d16da9230 share/opera@@{SUFFIX}/locale/de/search.ini
 	"$@" NF f81c9efe32b2cfcc6f7a7411a3ead4bd share/opera@@{SUFFIX}/locale/de/standard_speeddial.ini
-	"$@" NF 24fff94fd9645142b6eaed75183b1ad0 share/opera@@{SUFFIX}/locale/el/el.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/el/license.txt
 	"$@" NF de7b89f83721003bbb56d81088565825 share/opera@@{SUFFIX}/locale/en-GB/bookmarks.adr
 	"$@" NF d5e9fc8d94c6f266228213781cb612a6 share/opera@@{SUFFIX}/locale/en-GB/en-GB.lng
 	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/en-GB/license.txt
@@ -118,10 +101,6 @@
 	"$@" NF 45803aa62c61987464904c6c174fa781 share/opera@@{SUFFIX}/locale/es-ES/standard_speeddial.ini
 	"$@" NF 14742d663744313cd2f9529ef9de0be7 share/opera@@{SUFFIX}/locale/es-LA/es-LA.lng
 	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/es-LA/license.txt
-	"$@" NF e77d7b922c1d62c157f02950efb19e16 share/opera@@{SUFFIX}/locale/et/et.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/et/license.txt
-	"$@" NF 283cbc5235cbe9e61682bac429473e50 share/opera@@{SUFFIX}/locale/fi/fi.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/fi/license.txt
 	"$@" NF 4967df16d3d319b67be911ad2f9d0b25 share/opera@@{SUFFIX}/locale/fr-CA/fr-CA.lng
 	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/fr-CA/license.txt
 	"$@" NF 111b339f00ca3b3aca44993f09e4fa03 share/opera@@{SUFFIX}/locale/fr/bookmarks.adr
@@ -129,47 +108,18 @@
 	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/fr/license.txt
 	"$@" NF b451253b530b6903d2acdd00391107e4 share/opera@@{SUFFIX}/locale/fr/search.ini
 	"$@" NF 25d66517b407516c3b53ed4ba8f87d27 share/opera@@{SUFFIX}/locale/fr/standard_speeddial.ini
-	"$@" NF 368e2572ea8a483801d77739f1c50b65 share/opera@@{SUFFIX}/locale/fy/fy.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/fy/license.txt
-	"$@" NF edeb4d118224ae756a280d8a43952369 share/opera@@{SUFFIX}/locale/hi/hi.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/hi/license.txt
-	"$@" NF 44030f9176f1e44283a0c4384dea0fef share/opera@@{SUFFIX}/locale/hr/hr.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/hr/license.txt
 	"$@" NF bc176094b5455b35042c535de6c99531 share/opera@@{SUFFIX}/locale/hu/bookmarks.adr
 	"$@" NF 47b6e6bc462b297c5ff0dd193625c1dc share/opera@@{SUFFIX}/locale/hu/hu.lng
 	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/hu/license.txt
 	"$@" NF f31b046abc9de096d99a4478a500ee3b share/opera@@{SUFFIX}/locale/hu/search.ini
 	"$@" NF c6010c21e9a4c1fce19c9296ca5d2593 share/opera@@{SUFFIX}/locale/hu/standard_speeddial.ini
-	"$@" NF ea56d6dc73a0fabe226f07922300bd04 share/opera@@{SUFFIX}/locale/id/id.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/id/license.txt
 	"$@" NF bbd6ee592eb08fa0019f5bf9a95664f7 share/opera@@{SUFFIX}/locale/it/bookmarks.adr
 	"$@" NF 4704266f381bee6aff1f8359fa9cb7a8 share/opera@@{SUFFIX}/locale/it/it.lng
 	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/it/license.txt
 	"$@" NF b38f6ccce5e246a77c7a97d2e0fdf7a4 share/opera@@{SUFFIX}/locale/it/search.ini
 	"$@" NF 1ffe2ccf842d816a577edab81a374a5c share/opera@@{SUFFIX}/locale/it/standard_speeddial.ini
-	"$@" NF 5e87f4e8391f055ffce6fbf64ae0d542 share/opera@@{SUFFIX}/locale/ja/bookmarks.adr
-	"$@" NF 043d5f9012fc2436ac6880d9b87c34d5 share/opera@@{SUFFIX}/locale/ja/ja.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/ja/license.txt
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/ja/license.txt
-	"$@" NF 88fa821195cec2a4a3aa9cb3bafe50de share/opera@@{SUFFIX}/locale/ja/search.ini
-	"$@" NF e241ecfa93beebc6c7e13d3ff7088f2a share/opera@@{SUFFIX}/locale/ja/standard_speeddial.ini
-	"$@" NF 51c69dcef69152782660691d05ceba7f share/opera@@{SUFFIX}/locale/ka/ka.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/ka/license.txt
-	"$@" NF 36c63111441eed98bc98685c442e9b6b share/opera@@{SUFFIX}/locale/ko/ko.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/ko/license.txt
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/lt/license.txt
-	"$@" NF 9df3711491ecdb4a8705d242c4d02e91 share/opera@@{SUFFIX}/locale/lt/lt.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/mk/license.txt
-	"$@" NF 81851c3c129f86efc665265279baa79a share/opera@@{SUFFIX}/locale/mk/mk.lng
-	"$@" NF 1be92569b91543f3e5048bd2da8761a9 share/opera@@{SUFFIX}/locale/nb/bookmarks.adr
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/nb/license.txt
-	"$@" NF 95adb924b4d0f4540022cb991579c5cf share/opera@@{SUFFIX}/locale/nb/nb.lng
-	"$@" NF c6ea0dcda91384043d5231bd81da4532 share/opera@@{SUFFIX}/locale/nb/search.ini
-	"$@" NF e5ebff2e5dc9b5cf95150b9c09031c8a share/opera@@{SUFFIX}/locale/nb/standard_speeddial.ini
 	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/nl/license.txt
 	"$@" NF 25c33cfa152771222fa6a2936bf2bb83 share/opera@@{SUFFIX}/locale/nl/nl.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/nn/license.txt
-	"$@" NF 55e76ca698fdc97009f14b3d6276f99a share/opera@@{SUFFIX}/locale/nn/nn.lng
 	"$@" NF 659304a56668ac7dcda7d5ad537ac8ff share/opera@@{SUFFIX}/locale/pl/bookmarks.adr
 	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/pl/license.txt
 	"$@" NF 5794314fcee2f328de170b2e353083c0 share/opera@@{SUFFIX}/locale/pl/pl.lng
@@ -179,32 +129,13 @@
 	"$@" NF 7770f25cb087b9285f1d772a80ef9950 share/opera@@{SUFFIX}/locale/pt-BR/pt-BR.lng
 	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/pt/license.txt
 	"$@" NF dee3ed8d19f503e659e55924e7eb6c29 share/opera@@{SUFFIX}/locale/pt/pt.lng
-	"$@" NF 7b95d32fd40dfb0471a9badd405abefc share/opera@@{SUFFIX}/locale/ro/bookmarks.adr
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/ro/license.txt
-	"$@" NF e0a17c6ebe87528faca6cf550c3582eb share/opera@@{SUFFIX}/locale/ro/ro.lng
-	"$@" NF 2824879f1758a036c151aca0895cc2b5 share/opera@@{SUFFIX}/locale/ro/search.ini
-	"$@" NF d1ed970ff7b137d478bb94301da078e3 share/opera@@{SUFFIX}/locale/ro/standard_speeddial.ini
 	"$@" NF d94fca69ab135306e4c3bc2775fd0f7f share/opera@@{SUFFIX}/locale/ru/bookmarks.adr
 	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/ru/license.txt
 	"$@" NF 5e9488ade029018e63d826452daddc22 share/opera@@{SUFFIX}/locale/ru/ru.lng
 	"$@" NF 0245db6326d69ca4324c62030caa0786 share/opera@@{SUFFIX}/locale/ru/search.ini
 	"$@" NF 8beaab8ab5f5af7f0b9e52b31b35bea1 share/opera@@{SUFFIX}/locale/ru/standard_speeddial.ini
-	"$@" NF 00143eebb2330e52ae4ec7d2dd881a8a share/opera@@{SUFFIX}/locale/sk/bookmarks.adr
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/sk/license.txt
-	"$@" NF aa0deb1d50185ee1d792c4b5a302a526 share/opera@@{SUFFIX}/locale/sk/search.ini
-	"$@" NF bda3e24b68d5d200646a5f3edda11a88 share/opera@@{SUFFIX}/locale/sk/sk.lng
-	"$@" NF d1ed970ff7b137d478bb94301da078e3 share/opera@@{SUFFIX}/locale/sk/standard_speeddial.ini
-	"$@" NF 4b674679dcc365056133fc7d34ba87f3 share/opera@@{SUFFIX}/locale/sr/bookmarks.adr
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/sr/license.txt
-	"$@" NF b9cc80f70dd1090381957288d1fbe930 share/opera@@{SUFFIX}/locale/sr/search.ini
-	"$@" NF fa3b5ad4c7eb00b49a954664d9ccf19e share/opera@@{SUFFIX}/locale/sr/sr.lng
-	"$@" NF d1ed970ff7b137d478bb94301da078e3 share/opera@@{SUFFIX}/locale/sr/standard_speeddial.ini
 	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/sv/license.txt
 	"$@" NF bfc6beb08df78e55545cec53b5fbcc10 share/opera@@{SUFFIX}/locale/sv/sv.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/ta/license.txt
-	"$@" NF 695bce3cd7870e865cf30d38537bf031 share/opera@@{SUFFIX}/locale/ta/ta.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/te/license.txt
-	"$@" NF 7c9fc6ec8cf6096cf9239fb316416046 share/opera@@{SUFFIX}/locale/te/te.lng
 	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/tr/license.txt
 	"$@" NF 623deef778bf5d28d62b8b182010f5cf share/opera@@{SUFFIX}/locale/tr/tr.lng
 	"$@" NF c37440a5621ea20a11cef2629b591464 share/opera@@{SUFFIX}/locale/uk/bookmarks.adr
@@ -212,21 +143,6 @@
 	"$@" NF 1dcd633858fbb62293f9d4f5c25f5db2 share/opera@@{SUFFIX}/locale/uk/search.ini
 	"$@" NF 8beaab8ab5f5af7f0b9e52b31b35bea1 share/opera@@{SUFFIX}/locale/uk/standard_speeddial.ini
 	"$@" NF 905ed99b891e9ec3aa590fffb9e5dbfc share/opera@@{SUFFIX}/locale/uk/uk.lng
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/vi/license.txt
-	"$@" NF fd6237ff290c81fd866ac62283e2421c share/opera@@{SUFFIX}/locale/vi/vi.lng
-	"$@" NF 8e809f73f283b7e60d934753d9120b0e share/opera@@{SUFFIX}/locale/zh-cn/bookmarks.adr
-	"$@" NF 0b9bccfadaa392d0f1b016dbf926f73b share/opera@@{SUFFIX}/locale/zh-cn/browser.js
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/zh-cn/license.txt
-	"$@" NF 4166838cc06fd3594b5f01cce030dd95 share/opera@@{SUFFIX}/locale/zh-cn/search.ini
-	"$@" NF fa3dcf7fa65c11de3faa0f4fe6319ef2 share/opera@@{SUFFIX}/locale/zh-cn/standard_speeddial.ini
-	"$@" NF d44fc31b4c96a97d22624455874764ca share/opera@@{SUFFIX}/locale/zh-cn/turbosettings.xml
-	"$@" NF 0e3efdd139ec344bd2e6f33f158ef220 share/opera@@{SUFFIX}/locale/zh-cn/zh-cn.lng
-	"$@" NF 0b9bccfadaa392d0f1b016dbf926f73b share/opera@@{SUFFIX}/locale/zh-hk/browser.js
-	"$@" NF 6ea2f98cdeaf54c398dca24aea9df644 share/opera@@{SUFFIX}/locale/zh-hk/turbosettings.xml
-	"$@" NF 0b9bccfadaa392d0f1b016dbf926f73b share/opera@@{SUFFIX}/locale/zh-tw/browser.js
-	"$@" NF 76a5919d1950d5d3171636079f0b3f1b share/opera@@{SUFFIX}/locale/zh-tw/license.txt
-	"$@" NF 6ea2f98cdeaf54c398dca24aea9df644 share/opera@@{SUFFIX}/locale/zh-tw/turbosettings.xml
-	"$@" NF 8695448706942f14942603492ff3e1f4 share/opera@@{SUFFIX}/locale/zh-tw/zh-tw.lng
 	"$@" NF 1c24deadc6027ab1b7a2cecc7b7c00e4 share/opera@@{SUFFIX}/mathml.dtd
 	"$@" NF c01c5f4345a8f8512f29b24ce85c2de2 share/opera@@{SUFFIX}/package-id.ini
 	"$@" NF a6485d9b7be876314c624d291d6575a1 share/opera@@{SUFFIX}/package/lib/binprint.sh
@@ -332,7 +248,6 @@
 	"$@" NF 1f7910393ee83d68bbc6649e82805fdb share/opera@@{SUFFIX}/unite/webserver.ua
 	"$@" GX - bin/opera@@{SUFFIX}
 	"$@" GX - bin/opera-widget-manager@@{SUFFIX}
-	"$@" GX - bin/uninstall-opera@@{SUFFIX}
 }
 
 available()
