diff -Naur mpc-0.12.1.orig/doc/Makefile.in mpc-0.12.1/doc/Makefile.in
--- mpc-0.12.1.orig/doc/Makefile.in	2007-03-21 20:22:57.000000000 +0200
+++ mpc-0.12.1/doc/Makefile.in	2008-02-01 02:44:07.000000000 +0200
@@ -78,7 +78,7 @@
 install_sh = @install_sh@
 man_MANS = mpc.1
 pkgdata_DATA = 
-docdir = $(prefix)/share/doc/$(PACKAGE)
+docdir = $(prefix)/share/doc/srcTAG
 doc_DATA = mpc-bashrc m3u-handler.sh pls-handler.sh mppledit
 EXTRA_DIST = $(pkgdata_DATA) $(man_MANS) $(doc_DATA)
 subdir = doc
diff -Naur mpc-0.12.1.orig/Makefile.in mpc-0.12.1/Makefile.in
--- mpc-0.12.1.orig/Makefile.in	2007-03-21 20:22:57.000000000 +0200
+++ mpc-0.12.1/Makefile.in	2008-02-01 02:43:29.000000000 +0200
@@ -80,8 +80,8 @@
 
 SUBDIRS = doc src
 pkgdata_DATA = 
-docdir = $(prefix)/share/doc/$(PACKAGE)
-doc_DATA = AUTHORS README
+docdir = $(prefix)/share/doc/srcTAG
+doc_DATA = AUTHORS README COPYING
 EXTRA_DIST = ChangeLog COPYING $(pkgdata_DATA) $(doc_DATA)
 subdir = .
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
