diff -Naur kicad-20090623.orig/resources/linux/mime/applications/kicad.desktop kicad-20090623/resources/linux/mime/applications/kicad.desktop
--- kicad-20090623.orig/resources/linux/mime/applications/kicad.desktop	2009-06-23 23:35:34.000000000 +0300
+++ kicad-20090623/resources/linux/mime/applications/kicad.desktop	2009-06-23 23:38:14.000000000 +0300
@@ -1,8 +1,9 @@
 [Desktop Entry]
-Categories=Development;Electronics
+Categories=Qt;KDE;Education;Science;
 Comment=Design a printed circuit board.
 Exec=kicad
 GenericName=EDA Suite
+GenericName[tr]=Elektronik şematik ve PCB dizayn araçları
 Icon=kicad
 MimeType=application/x-kicad-project;
 Name=KiCad
