--- orig.klogshow.desktop	2007-03-28 13:52:02.000000000 +0300
+++ src/klogshow.desktop	2007-04-01 16:07:37.000000000 +0300
@@ -1,8 +1,10 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=klogshow
+Name=Klogshow
 Exec=klogshow
 Icon=klogshow
 Type=Application
+GenericName=Sistem Günlüğü İzleyici
 Comment=Klogshow - Klogshow used to display the last few lines of a text files on your desktop.
-Categories=Qt;
+Comment[tr]=Klogshow -
+Categories=Qt;System;Monitor;
