diff -urNp httrack-3.43.2.orig/html/server/div/WebHTTrack.desktop httrack-3.43.2/html/server/div/WebHTTrack.desktop
--- httrack-3.43.2.orig/html/server/div/WebHTTrack.desktop	2008-12-15 02:55:45.065046535 +0530
+++ httrack-3.43.2/html/server/div/WebHTTrack.desktop	2008-12-15 03:00:53.439047073 +0530
@@ -7,4 +7,4 @@ Terminal=false
 Name=WebHTTrack Website Copier
 Comment=Copy websites to your computer
 Exec=webhttrack
-Icon=/usr/share/httrack/icons/webhttrack.xpm
+Icon=webhttrack
diff -urNp httrack-3.43.2.orig/html/server/div/WebHTTrack-Websites.desktop httrack-3.43.2/html/server/div/WebHTTrack-Websites.desktop
--- httrack-3.43.2.orig/html/server/div/WebHTTrack-Websites.desktop	2008-12-15 02:55:45.065046535 +0530
+++ httrack-3.43.2/html/server/div/WebHTTrack-Websites.desktop	2008-12-15 03:01:18.262046434 +0530
@@ -7,4 +7,4 @@ Terminal=false
 Name=Browse Mirrored Websites
 Comment=Browse Websites Mirrored by WebHTTrack
 Exec=webhttrack browse
-Icon=/usr/share/httrack/icons/webhttrack.xpm
+Icon=webhttrack
 
