--- locale/tr.ts	2007-12-06 10:50:35.000000000 +0200
+++ locale/tr.ts	2007-12-08 20:00:51.000000000 +0200
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!DOCTYPE TS><TS version="1.1" language="tr_TR">
+<defaultcodec></defaultcodec>
 <context>
     <name>@default</name>
     <message>
-        <location filename="../src/main.cpp" line="24"/>
+        <location filename="../src/main.cpp" line="27"/>
         <source>A Qt Printer Ink Level Monitor Utility&lt;br /&gt;&lt;br /&gt;Copyright (C) 2003 Moris Ravasio &amp;lt;moris_ravasio@hotmail.com&amp;gt;&lt;br /&gt;Copyright (C) 2007 Matvey Kozhev &amp;lt;sikon@lucidfox.org&amp;gt;</source>
         <translation>Qt Tabanlı Yazıcı Mürekkep Düzeyi Gösterme Aracı&lt;br /&gt;&lt;br /&gt;Telif Hakkı (C) 2003 Moris Ravasio &amp;lt;moris_ravasio@hotmail.com&amp;gt;&lt;br /&gt;Telif Hakkı (C) 2007 Matvey Kozhev &amp;lt;sikon@lucidfox.org&amp;gt;</translation>
     </message>
@@ -77,176 +78,191 @@
         <translation>Bir yazıcı seçin:</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="83"/>
+        <location filename="../src/QinkDlg.ui" line="74"/>
         <source>&amp;Detect</source>
         <translation>&amp;Bul</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="86"/>
+        <location filename="../src/QinkDlg.ui" line="77"/>
         <source>Alt+D</source>
         <translation>Alt+B</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="204"/>
+        <location filename="../src/QinkDlg.ui" line="168"/>
         <source>&lt;b&gt;&lt;font size=&quot;+3&quot;&gt;Printer error&lt;/font&gt;&lt;/b&gt;</source>
         <translation>&lt;b&gt;&lt;font size=&quot;+3&quot;&gt;Yazıcı hatası&lt;/font&gt;&lt;/b&gt;</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="233"/>
+        <location filename="../src/QinkDlg.ui" line="197"/>
         <source>[Here is the place for a description of the error that has occured]</source>
         <translation>[Bu alan oluşan hatanın açıklaması içindir]</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="1629"/>
+        <location filename="../src/QinkDlg.ui" line="1485"/>
         <source>Black</source>
         <translation>Siyah</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="1591"/>
+        <location filename="../src/QinkDlg.ui" line="1447"/>
         <source>&lt;b&gt;Black:&lt;/b&gt;</source>
         <translation>&lt;b&gt;Siyah:&lt;/b&gt;</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="478"/>
+        <location filename="../src/QinkDlg.ui" line="424"/>
         <source>Colors</source>
         <translation>Renkler</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="2551"/>
+        <location filename="../src/QinkDlg.ui" line="2326"/>
         <source>&lt;b&gt;Yellow:&lt;/b&gt;</source>
         <translation>&lt;b&gt;Sarı:&lt;/b&gt;</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="2354"/>
+        <location filename="../src/QinkDlg.ui" line="2147"/>
         <source>&lt;b&gt;Magenta:&lt;/b&gt;</source>
         <translation>&lt;b&gt;Mor:&lt;/b&gt;</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="2166"/>
+        <location filename="../src/QinkDlg.ui" line="1977"/>
         <source>&lt;b&gt;Cyan:&lt;/b&gt;</source>
         <translation>&lt;b&gt;Cyan:&lt;/b&gt;</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="1972"/>
+        <location filename="../src/QinkDlg.ui" line="1801"/>
         <source>&lt;b&gt;Color:&lt;/b&gt;</source>
         <translation>&lt;b&gt;Renkli:&lt;/b&gt;</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="841"/>
+        <location filename="../src/QinkDlg.ui" line="769"/>
         <source>Color</source>
         <translation>Renk</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="1035"/>
+        <location filename="../src/QinkDlg.ui" line="945"/>
         <source>Cyan</source>
         <translation>Cyan</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="1226"/>
+        <location filename="../src/QinkDlg.ui" line="1118"/>
         <source>Magenta</source>
         <translation>Magenta</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="1423"/>
+        <location filename="../src/QinkDlg.ui" line="1297"/>
         <source>Yellow</source>
         <translation>Sarı</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="1648"/>
+        <location filename="../src/QinkDlg.ui" line="1504"/>
         <source>Photo colors</source>
         <translation>Fotoğraf renkleri</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="2010"/>
+        <location filename="../src/QinkDlg.ui" line="1839"/>
         <source>Photo color</source>
         <translation>Fotoğraf rengi</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="2204"/>
+        <location filename="../src/QinkDlg.ui" line="2015"/>
         <source>Photo cyan</source>
         <translation>Fotoğraf cyan rengi</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="2392"/>
+        <location filename="../src/QinkDlg.ui" line="2185"/>
         <source>Photo magenta</source>
         <translation>Fotoğraf magenta rengi</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="2589"/>
+        <location filename="../src/QinkDlg.ui" line="2364"/>
         <source>Photo yellow</source>
         <translation>Fotoğraf sarısı</translation>
     </message>
     <message>
-        <location filename="../src/qinkdlg.cpp" line="77"/>
+        <location filename="../src/qinkdlg.cpp" line="88"/>
         <source>&amp;About</source>
         <translation>H&amp;akkında</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="2631"/>
+        <location filename="../src/QinkDlg.ui" line="2397"/>
         <source>Alt+A</source>
         <translation>Alt+A</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="2654"/>
+        <location filename="../src/QinkDlg.ui" line="2420"/>
         <source>&amp;Update</source>
         <translation>G&amp;üncelle</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="2657"/>
+        <location filename="../src/QinkDlg.ui" line="2423"/>
         <source>Alt+U</source>
         <translation>Alt+Ü</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="2670"/>
+        <location filename="../src/QinkDlg.ui" line="2436"/>
         <source>&amp;Close</source>
         <translation>&amp;Kapat</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="2673"/>
+        <location filename="../src/QinkDlg.ui" line="2439"/>
         <source>Alt+C</source>
         <translation>Alt+K</translation>
     </message>
     <message>
-        <location filename="../src/qinkdlg.cpp" line="74"/>
+        <location filename="../src/qinkdlg.cpp" line="84"/>
         <source>&amp;Show/hide</source>
         <translation>&amp;Göster/Gizle</translation>
     </message>
     <message>
-        <location filename="../src/qinkdlg.cpp" line="81"/>
+        <location filename="../src/qinkdlg.cpp" line="95"/>
         <source>&amp;Exit</source>
         <translation>&amp;Çık</translation>
     </message>
     <message>
-        <location filename="../src/qinkdlg.cpp" line="209"/>
+        <location filename="../src/qinkdlg.cpp" line="310"/>
         <source>&lt;b&gt;No printers detected on this system&lt;/b&gt;&lt;br&gt;&lt;br&gt;It seems that there are no printer plugged to this computer. Please verify the connections of your devices; maybe a problem of wrong permissions on the device&apos;s port.&lt;br&gt;&lt;br&gt;For parallel printers try &lt;b&gt;chmod 666 /dev/lpX&lt;/b&gt; and &lt;br&gt;&lt;b&gt;chmod 666 /dev/parportX&lt;/b&gt; (as root), where X is your printer&apos;s parallel port number.&lt;br&gt;&lt;br&gt;For USB printers, try &lt;b&gt;chmod 666 /dev/usblpX&lt;/b&gt; (as root), where X is your printer&apos;s USB port number.&lt;br&gt;&lt;br&gt;Click on &lt;b&gt;Detect&lt;/b&gt; to try again.</source>
         <translation>&lt;b&gt;Sisteme bağlı bir yazıcı bulunamadı&lt;/b&gt;&lt;br&gt;&lt;br&gt;Bilgisayara hiç yazıcı takılmamış gibi görünüyor. Lütfen aygıt bağlantılarını kontrol edin; belki aygıtın bağlatı noktalarının izinleri yanlış olabilir.&lt;br&gt;&lt;br&gt;Paralel bağlantı noktası kullanan yazıcılar için,&lt;b&gt;chmod 666 /dev/lpX&lt;/b&gt; ve &lt;br&gt;&lt;b&gt;chmod 666 /dev/parportX&lt;/b&gt; (süper kullanıcı olarak) deneyin (X yazıcınızın paralel bağlantı noktasının numarasını gösterir).&lt;br&gt;&lt;br&gt; USB bağlantı kullanan yazıcılar için, &lt;b&gt;chmod 666 /dev/usblpX&lt;/b&gt; (süper kullanıcı olarak) deneyin (X yazıcınızın USB bağlantı noktasının numarasını gösterir).&lt;br&gt;&lt;br&gt;&lt;b&gt;Bul&lt;/b&gt; düğmesine tıklayarak tekrar deneyin.</translation>
     </message>
     <message>
-        <location filename="../src/qinkdlg.cpp" line="287"/>
+        <location filename="../src/qinkdlg.cpp" line="291"/>
         <source>%1: %2%
 </source>
         <translation>%1: %2%
 </translation>
     </message>
     <message>
-        <location filename="../src/qinkdlg.cpp" line="329"/>
+        <location filename="../src/qinkdlg.cpp" line="393"/>
         <source>&lt;b&gt;Generic error&lt;/b&gt;&lt;br&gt;&lt;br&gt;An error has occured communicating with the selected device. Click &lt;b&gt;Detect&lt;/b&gt; and &lt;b&gt;Update&lt;/b&gt; to refresh QInk status and try again. If the problem persists, e-mail me a description of the situation and all that can be helpful to me for reconstructing your execution environment.</source>
         <translation>&lt;b&gt;Genel hata&lt;/b&gt;&lt;br&gt;&lt;br&gt;Seçili aygıta bağlanılırken bir hata oluştu. &lt;b&gt;Bul&lt;/b&gt; ve &lt;b&gt;Güncelle&lt;/b&gt; düğmelerini kullanarak QInk&apos;in durumunu tazeleyip tekrar deneyin. Eğer sorun devam ederse geliştiriciye sorunu açıklayan bir e-posta gönderin. Bu, sorunun daha kısa sürede çözülmesine yardımcı olacaktır.</translation>
     </message>
     <message>
-        <location filename="../src/qinkdlg.cpp" line="333"/>
+        <location filename="../src/qinkdlg.cpp" line="397"/>
         <source>&lt;b&gt;Cannot read ink level from device&lt;/b&gt;&lt;br&gt;&lt;br&gt;Cannot retrieve ink levels from the selected device. Ensure your printer is switched on and click on &lt;b&gt;Update&lt;/b&gt; to try again.</source>
         <translation>&lt;b&gt;Mürekkep seviyesi aygıttan okunamadı&lt;/b&gt;&lt;br&gt;&lt;br&gt;Seçilen aygıttaki mürekkep seviyesine bilgisine erişilemedi. Yazıcınızın açık olduğundan emin olun ve &lt;b&gt;Güncelle&lt;/b&gt; düğmesine basarak tekrar deneyin.</translation>
     </message>
     <message>
-        <location filename="../src/qinkdlg.cpp" line="337"/>
+        <location filename="../src/qinkdlg.cpp" line="401"/>
         <source>&lt;b&gt;Unsupported printer!&lt;/b&gt;&lt;br&gt;&lt;br&gt;Selected printer is not supported by this version of libinklevel. Take a look to &lt;a href=&apos;http://home.arcor.de/markusheinz&apos;&gt;http://home.arcor.de/markusheinz&lt;/a&gt; for a complete list of devices supported by your version of libinklevel.</source>
         <translation>&lt;b&gt;Desteklenmeyen yazıcı!&lt;/b&gt;&lt;br&gt;&lt;br&gt;Libinklevel&apos;in bu sürümü seçilen yazıcıyı desteklemiyor . Libinklevel&apos;in bu sürümü için desteklenen yazıcıları &lt;a href=&apos;http://libinklevel.sourceforge.net&apos;&gt;http://libinklevel.sourceforge.net&lt;/a&gt; adresinden öğrenebilirsiniz.</translation>
     </message>
     <message>
-        <location filename="../src/QinkDlg.ui" line="1445"/>
+        <location filename="../src/QinkDlg.ui" line="1319"/>
         <source>Photo black</source>
         <translation>Fotoğraf siyahı</translation>
     </message>
+    <message>
+        <location filename="../src/qinkdlg.cpp" line="91"/>
+        <source>About &amp;Qt</source>
+        <translation>&amp;Qt Hakkında</translation>
+    </message>
+    <message>
+        <location filename="../src/qinkdlg.cpp" line="276"/>
+        <source>QInk: No printers detected</source>
+        <translation>QInk: Hiç yazıcı bulunamadı</translation>
+    </message>
+    <message>
+        <location filename="../src/qinkdlg.cpp" line="299"/>
+        <source>QInk: Printer error, double click for details</source>
+        <translation>QInk: Yazıcı hatası, detaylar için çift tıklayın</translation>
+    </message>
 </context>
 </TS>
