diff -Naur compiz.orig/configure.ac compiz/configure.ac
--- compiz.orig/configure.ac	2007-08-08 01:22:38.000000000 +0300
+++ compiz/configure.ac	2007-08-09 21:20:01.000000000 +0300
@@ -176,26 +176,6 @@
   AC_DEFINE(USE_GLIB, 1, [Build glib plugin])
 fi
 
-AC_ARG_ENABLE(gconf,
-  [  --disable-gconf         Disable gconf plugin],
-  [use_gconf=$enableval], [use_gconf=yes])
-
-if test "x$use_gconf" = "xyes"; then
-  PKG_CHECK_MODULES(GCONF, gconf-2.0)
-
-  AC_PATH_PROG(GCONFTOOL, gconftool-2, no)
-  if test x"$GCONFTOOL" = xno; then
-    AC_MSG_ERROR([gconftool-2 executable not found in your path - should be installed with GConf])
-  fi
-fi
-
-AM_GCONF_SOURCE_2
-
-AM_CONDITIONAL(USE_GCONF, test "x$use_gconf" = "xyes")
-if test "$use_gconf" = yes; then
-  AC_DEFINE(USE_GCONF, 1, [Build gconf plugin])
-fi
-
 stylesheetdir=$datadir/compiz
 AC_SUBST(stylesheetdir)
 
diff -Naur compiz.orig/gtk/window-decorator/Makefile.am compiz/gtk/window-decorator/Makefile.am
--- compiz.orig/gtk/window-decorator/Makefile.am	2007-07-29 03:44:59.000000000 +0300
+++ compiz/gtk/window-decorator/Makefile.am	2007-08-09 21:21:51.000000000 +0300
@@ -9,7 +9,6 @@
 gtk_window_decorator_LDADD	  =		       \
 	$(top_builddir)/libdecoration/libdecoration.la \
 	@GTK_WINDOW_DECORATOR_LIBS@		       \
-	@GCONF_LIBS@				       \
 	@DBUS_GLIB_LIBS@			       \
 	$(metacitylibs)
 gtk_window_decorator_SOURCES	  = \
@@ -17,21 +16,7 @@
 gtk_window_decorator_program = gtk-window-decorator
 endif
 
-if USE_GCONF
-schemadir       = $(GCONF_SCHEMA_FILE_DIR)
-schema_in_files = gwd.schemas.in
-schema_files    = $(schema_in_files:.schemas.in=.schemas)
-schema_DATA     = $(schema_files)
-
-@INTLTOOL_SCHEMAS_RULE@
-
-install-data-local:
-	if test -z "$(DESTDIR)" ; then \
-		GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(schema_DATA); \
-	fi
-endif
-
-INCLUDES = -I$(top_srcdir)/include @GTK_WINDOW_DECORATOR_CFLAGS@ @METACITY_CFLAGS@ @GCONF_CFLAGS@ @DBUS_GLIB_CFLAGS@
+INCLUDES = -I$(top_srcdir)/include @GTK_WINDOW_DECORATOR_CFLAGS@ @METACITY_CFLAGS@ @DBUS_GLIB_CFLAGS@
 
 bin_PROGRAMS = $(gtk_window_decorator_program)
 
diff -Naur compiz.orig/Makefile.am compiz/Makefile.am
--- compiz.orig/Makefile.am	2007-08-05 03:09:48.000000000 +0300
+++ compiz/Makefile.am	2007-08-09 21:17:56.000000000 +0300
@@ -23,10 +23,6 @@
 	po/stamp-it	 \
 	po/.intltool-merge-cache
 
-if USE_GCONF
-gconfdata = compiz-gconf.pc
-endif
-
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = compiz.pc compiz-cube.pc compiz-scale.pc $(gconfdata)
 
diff -Naur compiz.orig/metadata/Makefile.am compiz/metadata/Makefile.am
--- compiz.orig/metadata/Makefile.am	2007-07-29 03:44:59.000000000 +0300
+++ compiz/metadata/Makefile.am	2007-08-09 21:22:29.000000000 +0300
@@ -34,26 +34,6 @@
 
 @INTLTOOL_XML_RULE@
 
-if USE_GCONF
-schemadir    = $(GCONF_SCHEMA_FILE_DIR)
-schema_files = $(patsubst %.xml.in,compiz-%.schemas,$(xml_in_files))
-schema_DATA  = $(schema_files)
-
-%.schemas: $(xml_files)
-	xsltproc -o $@ $(srcdir)/schemas.xslt $(subst compiz-,,$*).xml;
-
-xsltdir    = $(stylesheetdir)
-xslt_files = schemas.xslt
-xslt_DATA  = $(xslt_files)
-
-if GCONF_SCHEMAS_INSTALL
-install-data-local:
-	if test -z "$(DESTDIR)" ; then \
-		GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(schema_DATA); \
-        fi
-endif
-endif
-
 EXTRA_DIST =		\
 	$(xml_files)    \
 	$(xml_in_files) \
diff -Naur compiz.orig/plugins/Makefile.am compiz/plugins/Makefile.am
--- compiz.orig/plugins/Makefile.am	2007-07-29 03:44:59.000000000 +0300
+++ compiz/plugins/Makefile.am	2007-08-09 21:23:30.000000000 +0300
@@ -89,13 +89,6 @@
 libglib_module = libglib.la
 endif
 
-if USE_GCONF
-libgconf_la_LDFLAGS = -module -avoid-version -no-undefined
-libgconf_la_LIBADD = @GCONF_LIBS@
-libgconf_la_SOURCES = gconf.c
-libgconf_module = libgconf.la
-endif
-
 if PLACE_PLUGIN
 libplace_la_LDFLAGS = -module -avoid-version -no-undefined
 libplace_la_LIBADD = @PLACE_LIBS@
@@ -128,7 +121,6 @@
 	@LIBPNG_CFLAGS@				\
 	@LIBRSVG_CFLAGS@			\
 	@ANNOTATE_CFLAGS@			\
-	@GCONF_CFLAGS@				\
 	@PLACE_CFLAGS@				\
 	@DBUS_CFLAGS@				\
 	@FUSE_CFLAGS@				\
