diff -Nuar evince-0.8.0.orig/data/evince.desktop.in.in evince-0.8.0/data/evince.desktop.in.in
--- evince-0.8.0.orig/data/evince.desktop.in.in	2007-03-23 06:36:33.000000000 +0200
+++ evince-0.8.0/data/evince.desktop.in.in	2007-03-23 06:36:43.000000000 +0200
@@ -9,7 +9,7 @@
 Terminal=false
 Type=Application
 Icon=evince
-NoDisplay=true
+NoDisplay=false
 X-GNOME-DocPath=
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=evince
