--- imageshack.desktop	2007-01-08 20:12:03.000000000 +0200
+++ imageshack_new.desktop	2009-08-10 22:25:09.897260844 +0300
@@ -1,11 +1,11 @@
+[Desktop Entry]
+X-KDE-ServiceTypes=KonqPopupMenu/Plugin,image/gif,image/jpeg,image/jpg,image/png,image/bmp,image/tif,image/tiff
+Actions=imageshack;
+Type=Service
+
 [Desktop Action imageshack]
-Exec=xterm -e "imageshack_upload '%u' --kdialog"
-Icon=imageshack
 Name=Host on Imageshack
 Name[it]=Carica su Imageshack
-
-[Desktop Entry]
-Actions=imageshack
-Encoding=UTF-8
-Icon=/home/bla/.kde/share/icons/imageshack.png
-ServiceTypes=image/gif,image/jpeg,image/jpg,image/png,image/bmp,image/tif,image/tiff
+Name[tr]=Imageshack'e Yükle
+Icon=imageshack.png
+Exec=imageshack_upload '%u' --kdialog
