diff -Naur qtractor-0.4.6.orig/src/qtractor.desktop.in qtractor-0.4.6/src/qtractor.desktop.in
--- qtractor-0.4.6.orig/src/qtractor.desktop.in	2010-06-12 14:00:14.173314630 +0300
+++ qtractor-0.4.6/src/qtractor.desktop.in	2010-06-12 14:01:49.648189915 +0300
@@ -2,10 +2,11 @@
 Name=@PACKAGE_NAME@
 Version=1.0
 GenericName=Multi-track Sequencer
+GenericName[tr]=Çok kanallı ses düzenleme yazılımı
 Comment=Qtractor is an Audio/MIDI multi-track sequencer application
-Exec=@ac_prefix@/bin/qtractor
+Comment[tr]=Çok kanallı ses düzenleme yazılımı
+Exec=qtractor
 Icon=qtractor
 Categories=Audio;AudioVideo;Midi;Sequencer;X-Multitrack;X-Alsa;X-Jack;Qt;
 Terminal=false
 Type=Application
-X-SuSE-translate=true
