--- ccsm.desktop.orig	2007-07-28 15:45:23.000000000 +0300
+++ ccsm.desktop	2007-07-29 01:14:08.000000000 +0300
@@ -1,9 +1,11 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=CompizConfig Settings Manager
+Name[tr]=CompizConfig
 Comment=Configure Compiz with CompizConfig
+GenericName[tr]=Compiz Yapılandırma Yöneticisi
 Icon=ccsm
 Exec=ccsm
 Terminal=false
 Type=Application
-Categories=Compiz;Settings;DesktopSettings;
+Categories=Application;Compiz;
\ Dosya sonunda yenisatır yok.
