diff -urNad gnome-python-extras-2.19.1~/gdl/gdl.override gnome-python-extras-2.19.1/gdl/gdl.override
--- gnome-python-extras-2.19.1~/gdl/gdl.override	2007-06-05 16:52:30.000000000 +0200
+++ gnome-python-extras-2.19.1/gdl/gdl.override	2008-09-26 15:43:38.000000000 +0200
@@ -6,7 +6,6 @@
 #include <pygtk/pygtk.h>
 #include <gdl/gdl-dock.h>
 #include <gdl/libgdltypebuiltins.h>
-#include <gdl/gdl-icons.h>
 #ifndef HAVE_GDL_0_7
 # include <gdl/gdl-dock-bar.h>
 #endif
