Index: xfce4-utils-4.4.2/scripts/xinitrc.in
===================================================================
--- xfce4-utils-4.4.2.orig/scripts/xinitrc.in	2008-07-16 23:52:30.000000000 -0300
+++ xfce4-utils-4.4.2/scripts/xinitrc.in	2008-07-16 23:52:58.000000000 -0300
@@ -53,6 +53,14 @@
 # load local modmap
 test -r $HOME/.Xmodmap && xmodmap $HOME/.Xmodmap
 
+# Use dbus-launch if installed.
+if test x"$DBUS_SESSION_BUS_ADDRESS" = x""; then
+	dbuslaunch=`which dbus-launch`
+	if test x"$dbuslaunch" != x"" -a x"$dbuslaunch" != x"no"; then
+		eval `$dbuslaunch --sh-syntax --exit-with-session`
+	fi
+fi
+
 # Launch xscreensaver (if available), but only as non-root user
 if test $UID -gt 0 -a -z "$VNCSESSION"; then 
     if test x"`which xscreensaver 2>/dev/null`" != x""; then
@@ -74,14 +82,6 @@
 fi
 fi
 
-# Use dbus-launch if installed.
-if test x"$DBUS_SESSION_BUS_ADDRESS" = x""; then
-	dbuslaunch=`which dbus-launch`
-	if test x"$dbuslaunch" != x"" -a x"$dbuslaunch" != x"no"; then
-		eval `$dbuslaunch --sh-syntax --exit-with-session`
-	fi
-fi
-
 # Run xfce4-session if installed
 xfcesm=`which xfce4-session`
 case "x$xfcesm" in
