diff -Nuar dia-0.96.1.orig/doc/en/Makefile.in dia-0.96.1/doc/en/Makefile.in
--- dia-0.96.1.orig/doc/en/Makefile.in	2007-04-11 20:47:46.000000000 +0300
+++ dia-0.96.1/doc/en/Makefile.in	2007-04-11 20:49:16.000000000 +0300
@@ -600,7 +600,6 @@
 	    $(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	  done; \
 	fi
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -Nuar dia-0.96.1.orig/doc/eu/Makefile.in dia-0.96.1/doc/eu/Makefile.in
--- dia-0.96.1.orig/doc/eu/Makefile.in	2007-04-11 20:47:46.000000000 +0300
+++ dia-0.96.1/doc/eu/Makefile.in	2007-04-11 20:50:10.000000000 +0300
@@ -593,7 +593,6 @@
 	    $(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	  done; \
 	fi
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -Nuar dia-0.96.1.orig/doc/pl/Makefile.in dia-0.96.1/doc/pl/Makefile.in
--- dia-0.96.1.orig/doc/pl/Makefile.in	2007-04-11 20:47:46.000000000 +0300
+++ dia-0.96.1/doc/pl/Makefile.in	2007-04-11 20:49:52.000000000 +0300
@@ -535,7 +535,6 @@
 	    $(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	  done; \
 	fi
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -Nuar dia-0.96.1.orig/omf.make dia-0.96.1/omf.make
--- dia-0.96.1.orig/omf.make	2007-04-11 20:47:44.000000000 +0300
+++ dia-0.96.1/omf.make	2007-04-11 20:48:10.000000000 +0300
@@ -49,7 +49,6 @@
 	    $(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	  done; \
 	fi
-	-scrollkeeper-update -p $(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
