--- src/screenshooter.desktop.in.in.orig	2008-07-18 19:40:45.000000000 +0300
+++ src/screenshooter.desktop.in.in	2008-07-18 20:13:03.000000000 +0300
@@ -1,7 +1,9 @@
 [Xfce Panel]
 Type=X-XFCE-PanelPlugin
-_Name= Screenshot
-_Comment= Take a screenshot
+Name= Screenshot
+Name[tr]=Ekran Görüntüsü
+Comment= Take a screenshot
+Comment[tr]= Ekran görüntüsü alır
 Icon=applets-screenshooter
 X-XFCE-Exec=@PLUGIN_PATH@/xfce4-screenshooter-plugin
 
