--- netpanzer.desktop.orig	2005-04-19 13:15:04.000000000 +0300
+++ netpanzer.desktop	2007-04-19 00:08:17.000000000 +0300
@@ -4,7 +4,9 @@
 Type=Application
 Name=NetPanzer
 GenericName=Realtime Strategy Game
+GenericName=Gerçek Zamanlı Strateji Oyunu
 Comment=Online multiplayer tactical warfare game
+Comment=Online oynanan çok oyunculu taktiksel savaş oyunu
 Icon=netpanzer
 Exec=netpanzer
 Terminal=false
