diff -Nuar enblend-3.0.orig/include/vigra/imageiteratoradapter.hxx enblend-3.0/include/vigra/imageiteratoradapter.hxx
--- enblend-3.0.orig/include/vigra/imageiteratoradapter.hxx	2007-01-27 06:54:46.000000000 +0200
+++ enblend-3.0/include/vigra/imageiteratoradapter.hxx	2008-07-03 16:24:54.000000000 +0300
@@ -39,6 +39,7 @@
 #define VIGRA_IMAGEITERATORADAPTER_HXX
 
 #include <iterator>   // iterator tags
+#include <cstdlib>
 
 namespace vigra {
 
diff -Nuar enblend-3.0.orig/src/gpu.cc enblend-3.0/src/gpu.cc
--- enblend-3.0.orig/src/gpu.cc	2006-12-08 10:40:58.000000000 +0200
+++ enblend-3.0/src/gpu.cc	2008-07-03 16:25:40.000000000 +0300
@@ -21,6 +21,7 @@
 #include <iostream>
 
 #include "gpu.h"
+#include <cstdlib>
 
 using std::cerr;
 using std::cout;
