diff -Naur centerim-4.22.3.orig/colorscheme-example centerim-4.22.3/colorscheme-example
--- colorscheme-example	1970-01-01 02:00:00.000000000 +0200
+++ colorscheme-example	2008-03-13 16:16:47.000000000 +0200
@@ -0,0 +1,34 @@
+#~/.centerim/colorscheme
+
+status	green/black bold
+dialog_text	green/black
+dialog_menu	black/yellow
+dialog_selected	black/yellow
+dialog_highlight	yellow/black bold
+dialog_frame	green/black
+main_text	green/transparent
+main_menu	yellow/transparent
+main_selected	green/transparent bold
+main_highlight	yellow/transparent	bold
+main_frame	green/transparent	bold
+main_history_incoming	green/transparent
+main_history_outgoing	green/transparent bold
+clist_icq	green/transparent
+clist_yahoo	green/transparent
+clist_infocard	white/transparent
+clist_msn	green/transparent
+clist_aim	green/transparent
+clist_irc	green/transparent
+clist_jabber	green/transparent
+clist_rss	green/transparent	
+clist_lj	green/transparent	
+clist_gg	green/transparent	
+clist_online	green/transparent	
+clist_offline	green/transparent	
+clist_away	green/transparent	
+clist_dnd	green/transparent	
+clist_na	green/transparent	
+clist_free_for_chat	green/transparent	
+clist_invisible	green/transparent	
+clist_not_in_list	red/transparent
+
