--- komparator.desktop.orj	2007-08-27 21:00:42.000000000 +0300
+++ src/komparator.desktop	2007-10-12 01:49:29.000000000 +0300
@@ -1,13 +1,13 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Komparator
-Name[xx]=xxKomparatorxx
-Exec=komparator %U
+Name[tr]=Komparator
+Exec=komparator
 Icon=komparator
 Type=Application
 Comment=A KDE Folder Synchronization Application
 Comment[de]=Eine KDE-Anwendung zum Abgleich von Ordnern
-Comment[xx]=xxA KDE Folder Synchronization Applicationxx
+GenericName[tr]=Dizin Eşleme Uygulaması
 Terminal=false
-Categories=Qt;KDE;Utility;Archiving;
+Icon=komparator
 DocPath=komparator/index.html
