diff -Nuar comedi_calibrate-1.orig/comedi_calibrate/Makefile.in comedi_calibrate-1/comedi_calibrate/Makefile.in
--- comedi_calibrate-1.orig/comedi_calibrate/Makefile.in	2007-07-30 05:30:15.000000000 +0300
+++ comedi_calibrate-1/comedi_calibrate/Makefile.in	2007-12-13 08:53:13.000000000 +0200
@@ -429,7 +429,7 @@
 
 
 install-data-local:
-	$(INSTALL) -d $(DESTDIR)$(localstatedir)/lib/comedi/calibrations
+
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -Nuar comedi_calibrate-1.orig/comedi_soft_calibrate/Makefile.in comedi_calibrate-1/comedi_soft_calibrate/Makefile.in
--- comedi_calibrate-1.orig/comedi_soft_calibrate/Makefile.in	2007-07-30 05:30:15.000000000 +0300
+++ comedi_calibrate-1/comedi_soft_calibrate/Makefile.in	2007-12-13 08:52:31.000000000 +0200
@@ -431,7 +431,7 @@
 
 
 install-data-local:
-	$(INSTALL) -d $(localstatedir)/lib/comedi/calibrations
+	
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
