diff -Naur aldrin-0.12/share/applications/aldrin.desktop aldrin-0.12_patched/share/applications/aldrin.desktop
--- aldrin-0.12/share/applications/aldrin.desktop	2006-10-21 00:53:10.000000000 +0300
+++ aldrin-0.12_patched/share/applications/aldrin.desktop	2008-08-19 12:24:25.000000000 +0300
@@ -1,12 +1,14 @@
 [Desktop Entry]
 Categories=Application;AudioVideo;
-Comment=Aldrin Modular Sequencer
+Comment[en]=Aldrin Modular Sequencer
+Comment[tr]=Aldrin Modüler Ardıştırıcı
+GenericName[en]=Aldrin Modular Sequencer
+GenericName[tr]=Aldrin Modüler Ardıştırıcı
 Encoding=UTF-8
 Exec=aldrin
 Hidden=false
 Icon=aldrin
 Name=Aldrin
-GenericName=Aldrin Modular Sequencer
 Terminal=false
 Type=Application
 MimeType=application/x-aldrin
