--- QTfrontend/gameuiconfig.cpp.orig	2008-11-11 20:29:21.000000000 +0200
+++ QTfrontend/gameuiconfig.cpp	2008-11-11 20:29:46.000000000 +0200
@@ -41,7 +41,7 @@
 	int t = Form->ui.pageOptions->CBResolution->findText(value("video/resolution").toString());
 	Form->ui.pageOptions->CBResolution->setCurrentIndex((t < 0) ? 0 : t);
 	Form->ui.pageOptions->CBFullscreen->setChecked(value("video/fullscreen", false).toBool());
-	bool ffscr=value("video/frontendfullscreen", false).toBool();
+	bool ffscr=value("video/frontendfullscreen", true).toBool();
 	Form->ui.pageOptions->CBFrontendFullscreen->setChecked(ffscr);
 
 	Form->ui.pageOptions->CBEnableSound->setChecked(value("audio/sound", true).toBool());
