--- ccm/Window.py.orig	2007-10-22 01:19:51.000000000 +0300
+++ ccm/Window.py	2007-11-06 19:16:43.000000000 +0200
@@ -152,7 +152,8 @@
             categoryToggleLabel = Label (label)
             align = gtk.Alignment (0, 0.5, 1, 1)
             align.set_padding (0, 0, 0, 10)
-            align.add (categoryToggleIcon)
+            # Until compiz developers fix icon problem, we'd prefer to have only texts without icons.
+            #align.add (categoryToggleIcon)
             categoryToggleBox = gtk.HBox ()
             categoryToggleBox.pack_start (align, False, False)
             categoryToggleBox.pack_start (categoryToggleLabel, True, True)
