--- Synergy.desktop	2005-02-17 00:11:27.000000000 +0200
+++ Synergy-yeni.desktop	2006-05-07 17:10:12.000000000 +0300
@@ -1,8 +1,10 @@
 [Desktop Entry]
 Name="Synergy"
-Comment="Mechanical CAD/CAM program"
+Comment=Mechanical CAD/CAM program
+Comment[tr]=Synergy - CAD/CAM
 Comment[pl]=Synergy - CAD
 Icon=/usr/weber/synergy.xpm
 Exec=/usr/weber/bin/synergy
-Terminal=0
+Terminal=false
 Type=Application
+Categories=GTK;Education;Science;
