diff -Nur desktop.orig/gcompris.desktop.in desktop/gcompris.desktop.in
--- gcompris.desktop.in	2006-08-14 02:55:02.000000000 +0300
+++ gcompris.desktop.in	2006-08-19 17:02:05.000000000 +0300
@@ -1,8 +1,11 @@
 [Desktop Entry]
 Encoding=UTF-8
 _Name=Educational suite GCompris
+Name[tr]=GCompris
 _GenericName=Multi-activity educational game
+GenericName[tr]= Eğitim Amaçlı Oyunlar
 _Comment=Educational game for ages 2 to 10
+Comment[tr]=2-10 yaş arası çocuklar için eğitim amaçlı oyunlar
 Exec=gcompris
 Icon=gcompris.png
 Terminal=false
diff -Nur desktop.orig/gcompris-edit.desktop.in desktop/gcompris-edit.desktop.in
--- gcompris-edit.desktop.in	2006-07-15 21:31:20.000000000 +0300
+++ gcompris-edit.desktop.in	2006-08-19 17:04:55.000000000 +0300
@@ -1,7 +1,9 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=GCompris Administration
+Name[tr]=GCompris Yönetimi
 _Comment=Administration for gcompris
+Comment[tr]=GCompris için Yönetim
 Exec=gcompris -a
 Icon=gcompris-edit.png
 Terminal=0
