--- /var/tmp/pisi/osh-0.7.1-1/work/osh-0.7.1/setup.py	2006-11-05 03:07:23.000000000 +0200
+++ ./setup.py	2006-11-05 03:08:58.000000000 +0200
@@ -45,8 +45,6 @@
     author_email = 'jao@geophile.com',
     packages = ['', 'osh', 'oshtypes'],
     data_files = [('/usr/bin', ['bin/osh', 'bin/remoteosh', 'bin/oshtestssh']),
-                  ('/usr/share/osh', ['bin/remoteinstall', '../osh-0.7.1.tar.gz']),
-                  ('/usr/share/osh/scripts', scripts),
                   ('/usr/share/osh/test', test),
                   ('/usr/share/doc/osh', doc),
                   ('/usr/share/doc/osh/web', doc_web)]
