--- monodevelop-2.0/monodevelop.desktop.orig	2009-06-25 01:36:47.000000000 +0300
+++ monodevelop-2.0/monodevelop.desktop	2009-06-25 01:38:47.000000000 +0300
@@ -2,8 +2,11 @@
 Version=1.0
 Encoding=UTF-8
 Name=MonoDevelop
+Name[tr]=MonoDevelop
 GenericName=Integrated Development Environment
+GenericName[tr]=Bütünleşik Geliştirme Ortamı
 Comment=Develop .NET applications in an Integrated Development Environment
+Comment[tr]=Bütünleşik geliştirme ortamında .NET uygulamaları geliştir
 Exec=monodevelop
 TryExec=monodevelop
 Icon=monodevelop
