diff -Nuar gedit-2.16.2.orig/help/eu/Makefile.in gedit-2.16.2/help/eu/Makefile.in
--- gedit-2.16.2.orig/help/eu/Makefile.in	2007-02-16 23:55:52.000000000 +0200
+++ gedit-2.16.2/help/eu/Makefile.in	2007-02-16 23:57:34.000000000 +0200
@@ -515,7 +515,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(DESTDIR)$(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -Nuar gedit-2.16.2.orig/help/ko/Makefile.in gedit-2.16.2/help/ko/Makefile.in
--- gedit-2.16.2.orig/help/ko/Makefile.in	2007-02-16 23:55:52.000000000 +0200
+++ gedit-2.16.2/help/ko/Makefile.in	2007-02-16 23:57:06.000000000 +0200
@@ -515,7 +515,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(DESTDIR)$(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
diff -Nuar gedit-2.16.2.orig/help/ro/Makefile.in gedit-2.16.2/help/ro/Makefile.in
--- gedit-2.16.2.orig/help/ro/Makefile.in	2007-02-16 23:55:52.000000000 +0200
+++ gedit-2.16.2/help/ro/Makefile.in	2007-02-16 23:56:50.000000000 +0200
@@ -515,7 +515,6 @@
 	for file in $(omffile); do \
 		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
 	done
-	-scrollkeeper-update -p $(DESTDIR)$(scrollkeeper_localstate_dir) -o $(DESTDIR)$(omf_dest_dir)
 
 uninstall-local-omf:
 	-for file in $(srcdir)/*.omf; do \
