--- translations/stopmotion_tr.ts.orig	2008-07-25 13:59:02.000000000 +0300
+++ translations/stopmotion_tr.ts	2008-09-09 21:27:26.000000000 +0300
@@ -1199,7 +1199,7 @@
     <message>
         <location filename="../src/presentation/frontends/qtfrontend/toolsmenu.cpp" line="264"/>
         <source>&lt;h4&gt;Launch Gimp&lt;/h4&gt; &lt;p&gt;Click this button to open the active frame in Gimp&lt;/p&gt; &lt;p&gt;Note that you can also drag images from the frame bar and drop them on Gimp&lt;/p&gt;</source>
-        <translation type="unfinished"></translation>
+        <translation>&lt;h4&gt;Gimp'i Çalıştır&lt;/h4&gt; &lt;p&gt;Çerçeveyi Gimp'te aç.&lt;/p&gt; &lt;p&gt;Not Resimleri çerçeve çubuğundan Gimp içine sürükleyip kullanabilirsiniz&lt;/p&gt;</translation>
     </message>
 </context>
 </TS>
