--- lxappearance.desktop.orig	2010-09-19 13:54:22.762330962 +0300
+++ lxappearance.desktop	2010-09-19 13:55:17.810579678 +0300
@@ -32,6 +32,7 @@
 Name[sl]=Nastavitve izgleda
 Name[sr]=Изглед
 Name[sv]=Utseende
+Name[tr]=Görünüm
 Name[uk]=Зовнішній вигляд
 Name[vi]=Diện mạo
 Name[zh_CN]=外观
@@ -68,6 +69,7 @@
 Comment[sl]=Prilagajanje videza namizja
 Comment[sr]=Прилагодите изглед радне површи
 Comment[sv]=Anpassa utseendet för skrivbordet
+Comment[tr]=Masaüstü görünümünü değiştirir
 Comment[uk]=Налаштувати вигляд робочого столу
 Comment[vi]=Tùy chỉnh màn hình nền
 Comment[zh_CN]=自定义桌面外观
