diff -Nuar autopano-sift-2.4.orig/src/Makefile autopano-sift-2.4/src/Makefile
--- autopano-sift-2.4.orig/src/Makefile	2005-10-31 14:45:20.000000000 +0200
+++ autopano-sift-2.4/src/Makefile	2006-12-04 23:19:38.000000000 +0200
@@ -11,7 +11,7 @@
 LIBS_WIN=/r:gtk-sharp.dll /r:System.Drawing.dll /r:ICSharpCode.SharpZipLib.dll \
 	/r:glib-sharp.dll /r:gdk-sharp.dll
 MAKE_MONO=
-LIBS_MONO=/pkg:gtk-sharp /r:System.Drawing /r:ICSharpCode.SharpZipLib
+LIBS_MONO=/pkg:gtk-sharp-2.0 /r:System.Drawing /r:ICSharpCode.SharpZipLib
 
 ifeq ($(windows),yes)
 	CSC=$(CSC_WINDOWS)

