diff -Naur acetoneiso_2.1.1.orig/acetoneiso/menu/AcetoneISO.desktop acetoneiso_2.1.1/acetoneiso/menu/AcetoneISO.desktop
--- acetoneiso_2.1.1.orig/acetoneiso/menu/AcetoneISO.desktop	2009-10-10 13:09:21.126728908 +0300
+++ acetoneiso_2.1.1/acetoneiso/menu/AcetoneISO.desktop	2009-10-10 14:29:55.406854286 +0300
@@ -2,12 +2,15 @@
 Categories=Qt;KDE;Application;AudioVideo;DiscBurning;
 Encoding=UTF-8
 Exec=acetoneiso
-Icon=/usr/share/pixmaps/Acetino2.png
-MimeType=application/x-iso
+Icon=acetoneiso
+MimeType=application/x-iso;
 Name=AcetoneISO
 GenericName=CD/DVD image manipulator
+GenericName[tr]=CD/DVD İmajı Düzenleyici
 GenericName[it]=Strumenti per immagini CD/DVD
 GenericName[de]=CD/DVD-Abbildmanager
+GenericName[es]=Manipulador de imágenes de CD/DVD
+GenericName[sv]=Hanterar CD/DVD-avbilder
 StartupNotify=true
 Terminal=false
 Type=Application
