diff -Naur compiz.orig/metadata/cube.xml.in compiz/metadata/cube.xml.in
--- compiz.orig/metadata/cube.xml.in	2007-08-21 21:39:09.000000000 +0300
+++ compiz/metadata/cube.xml.in	2007-09-17 15:40:42.000000000 +0300
@@ -46,14 +46,14 @@
 	    <option name="scale_image" type="bool">
 		<_short>Scale image</_short>
 		<_long>Scale images to cover top face of cube</_long>
-		<default>false</default>
+		<default>true</default>
 	    </option>
 	    <option name="images" type="list">
 		<_short>Image files</_short>
 		<_long>List of PNG and SVG files that should be rendered on top face of cube</_long>
 		<type>string</type>
 		<default>
-		    <value>freedesktop</value>
+		    <value>/usr/share/compiz/pardus-top.png</value>
 		</default>
 	    </option>
 	    <option name="skydome" type="bool">
