--- computer.h.orig	2009-03-30 03:13:48.000000000 +0300
+++ computer.h	2009-03-30 03:14:39.000000000 +0300
@@ -37,6 +37,7 @@
     { DB_PREFIX "yellowdog-release",	"yd"   },
     { DB_PREFIX "sabayon-release",      "sbn"  },
     { DB_PREFIX "arch-release",         "arch" },
+    { DB_PREFIX "pardus-release",       "pardus" },
     { DB_PREFIX "enlisy-release",       "enlsy"},
     { DB_PREFIX "SuSE-release",		"suse" },
     { DB_PREFIX "sun-release",		"sun"  },
