--- krusader/main.cpp.str	2009-03-22 04:42:54.000000000 +0100
+++ krusader/main.cpp	2009-03-22 04:43:07.000000000 +0100
@@ -217,7 +217,7 @@
     fprintf( stderr, "DBus Error: %s, %s\n", reply.error().name().toLocal8Bit().constData(), reply.error().message().toLocal8Bit().constData() );
 
   if( reply.isValid() && (bool)reply ) {
-    fprintf( stderr, i18n( "Application already running!\n" ).toLocal8Bit() );
+    fprintf( stderr, "%s", i18n( "Application already running!\n" ).toLocal8Bit() );
     KStartupInfo::appStarted();
     return 1;
   }
