--- idjcpython/sourceclientgui.py.orig	2008-06-21 11:18:18.000000000 +0200
+++ idjcpython/sourceclientgui.py	2008-06-21 11:19:10.000000000 +0200
@@ -169,7 +169,7 @@
    def __init__(self, parent, qty, default_text=""):
       self.source_client_gui = parent
       gtk.HBox.__init__(self)
-      self.set_spacing(13)
+      self.set_spacing(12)
       left_hbox = gtk.HBox()
       label = gtk.Label(ln.metadata)
       left_hbox.pack_start(label, False, False, 0)
@@ -616,7 +616,7 @@
       scrolled = gtk.ScrolledWindow()
       Tab.add(self, scrolled)
       scrolled.show()
-      scrolled.set_size_request(-1, 355)
+      scrolled.set_size_request(-1, 162)
       scrolled.set_policy(gtk.POLICY_NEVER, gtk.POLICY_AUTOMATIC)
       scrolled.set_shadow_type(gtk.SHADOW_NONE)
       scrolled_vbox = gtk.VBox()
