diff -Naur ppracer-0.5alpha/data/config.nut.in ppracer-0.5alphamod/data/config.nut.in
--- ppracer-0.5alpha/data/config.nut.in	2005-09-27 08:12:01.000000000 +0300
+++ ppracer-0.5alphamod/data/config.nut.in	2006-06-28 11:23:13.000000000 +0300
@@ -120,7 +120,7 @@
 ui_language_description = @"
 Sets the language for the ui
 ",
-ui_language = "en_GB",
+ui_language = "tr_TR",
 
 always_save_event_race_data_description = @"
 only for cheating purpose
diff -Naur ppracer-0.5alpha/data/translations/languages.nut ppracer-0.5alphamod/data/translations/languages.nut
--- ppracer-0.5alpha/data/translations/languages.nut	2005-09-16 08:41:03.000000000 +0300
+++ ppracer-0.5alphamod/data/translations/languages.nut	2006-06-28 11:13:33.000000000 +0300
@@ -24,3 +24,4 @@
 Translation.addLanguage("ru_RU", "Русский");
 Translation.addLanguage("uk_UA", "Українська");
 Translation.addLanguage("sv_SE", "Svenska");
+Translation.addLanguage("tr_TR", "Türkçe");
diff -Naur ppracer-0.5alpha/data/translations/tr_TR.nut ppracer-0.5alphamod/data/translations/tr_TR.nut
--- ppracer-0.5alpha/data/translations/tr_TR.nut	1970-01-01 02:00:00.000000000 +0200
+++ ppracer-0.5alphamod/data/translations/tr_TR.nut	2006-06-28 11:13:57.000000000 +0300
@@ -0,0 +1,132 @@
+
+//Translator: Uğur Çetin - jnmbk@users.sourceforge.net
+
+//common strings
+Translation.set("Back","Geri");
+Translation.set("PRESS ANY KEY TO START","BAŞLAMAK İÇİN BİR TUŞA BASIN");
+
+//game_type_select
+Translation.set("Enter an event","Turnuva");
+Translation.set("Practice","Alıştırma");
+Translation.set("Configuration","Ayarlar");
+Translation.set("Credits","Hakkında");
+Translation.set("Quit","Çıkış");
+
+//configuration
+Translation.set("General","Genel");
+Translation.set("Video","Görüntü");
+Translation.set("Audio","Ses");
+Translation.set("Keyboard","Klavye");
+Translation.set("Joystick","Joystick");
+
+//config mode
+Translation.set("Cancel","İptal");
+Translation.set("Ok","Tamam");
+
+//general configuration
+Translation.set("General Configuration","Genel Ayarlar");
+Translation.set("Language:","Dil:");
+Translation.set("Show UI Snow:","Kar");
+Translation.set("Display FPS:","FPS");
+Translation.set("Display Progress Bar:","İlerleme Çubuğu");
+Translation.set("Draw Fog:","Sis");
+Translation.set("Reflections:","Yansımalar");
+Translation.set("Shadows:","Gölgeler");
+
+//video configuration
+Translation.set("Video Configuration","Görüntü Ayarları");
+Translation.set("Resolution:","Çözünürlük");
+Translation.set("Bits Per Pixel:","Piksel Başına Bit");
+Translation.set("Fullscreen:","Tam Ekran");
+Translation.set("Experimental (needs restart)","Deneysel (baştan başlatma gerektirir)");
+Translation.set("Enable FSAA:","FSAA Etkin");
+
+//audio configuration
+Translation.set("Audio Configuration","Ses Ayarları");
+Translation.set("Sound Effects:","Ses Efektleri");
+Translation.set("Music:","Müzik");
+Translation.set("(needs restart)","(baştan başlatma gerektirir)");
+Translation.set("Disable Audio:","Sesi Kapat");
+Translation.set("Stereo:","Stereo");
+Translation.set("Bits Per Sample:","Örnekleme Başına bit");
+Translation.set("Samples Per Second:","Saniye Başına Örnekleme");
+
+//keyboard configuration
+Translation.set("Keyboard Configuration","Klavye Ayarları");
+Translation.set("Turn left:","Sola dön");
+Translation.set("Turn right:","Sağa dön");
+Translation.set("Paddle:","Ayakları oynat");
+Translation.set("Brake:","Frenle");
+Translation.set("Jump:","Zıpla");
+Translation.set("Trick:","Numara yap");
+Translation.set("Reset:","Düzgün yere koy");
+Translation.set("Pause:","Duraklat");
+
+//joystick configuration
+Translation.set("Joystick Configuration","Joystick ayarları");
+Translation.set("Enable Joystick","Joystick etkin");
+
+//race select
+Translation.set("Race!","Başla!");
+Translation.set("Select a race","Yarış seç");
+Translation.set("Contributed by:","Pisti yapan:");
+Translation.set("Unknown","Bilinmiyor");
+Translation.set("Time:","Süre:");
+Translation.set("Herring:","Balık:");
+Translation.set("Score:","Puan:");
+
+//event select
+Translation.set("Continue","Devam et");
+Translation.set("Select event and cup","Turnuva ve yarış seçin");
+Translation.set("Event:","Yarış");
+Translation.set("Cup:","Turnuva");
+Translation.set("You've won this cup!","Bu turnuvayı kazandınız!");
+Translation.set("You must complete this cup next","Şimdi bu turnuvayı kazanmalısınız");
+Translation.set("You cannot enter this cup yet","Bu turnuvaya katılamazsınız");
+
+//event race select
+Translation.set("You don't have any lives left.","Hiç canınız kalmadı");
+Translation.set("Race won! Your result:","Kazandınız! Sonuçlar:");
+Translation.set("Needed to advance:","Gerekli olan:");
+Translation.set("You can't enter this race yet.","Bu yarışa henüz katılamazsınız.");
+
+//loading
+Translation.set("Loading, Please Wait...","Yükleniyor, Lütfen Bekleyin");
+
+//paused
+Translation.set("Resume","Devam Et");
+Translation.set("Paused","Duraklatıldı");
+
+//race over
+Translation.set("Race Over","Yarış Bitti");
+Translation.set("Time: %02d:%02d.%02d","Süre: %02d:%02d.%02d");
+Translation.set("Herring: %3d","Balık: %3d");
+Translation.set("Score: %6d","Puan: %6d");
+Translation.set("Max speed: %3d km/h","En fazla hız: %3d km/s");
+Translation.set("Was flying: %.01f %% of time","Uçma yüzdesi: %% %.01f");
+Translation.set("Race aborted","Yarış iptal edildi");
+Translation.set("You beat your best score!","En iyi puanınızı geçtiniz!");
+Translation.set("Congratulations! You won the event!","Tebrikler, yarışı kazandınız!");
+Translation.set("Congratulations! You won the cup!","Tebrikler, kupa sizin!");
+Translation.set("You advanced to the next race!","Sonraki yarışa katılmaya hak kazandınız!");
+Translation.set("You didn't advance.","Sonraki yarışa katılamazsınız.");
+
+//HUD
+//Strings are prefixed to prevent collisions with the UI
+Translation.set("H|FPS: %.1f","FPS: %.1f");
+Translation.set("H|Time:","Süre:");
+Translation.set("H|km/h","km/s");
+//herring counter
+Translation.set("H|%03d","%03d");
+//time counter
+Translation.set("H|%02d:%02d:%02d","%02d:%02d:%02d");
+//speed counter
+Translation.set("H|%.0f","%.0f");
+
+//Credits
+Translation.set("Development Team","Geliştirme Ekibi");
+Translation.set("Translators","Çeviriciler");
+Translation.set("Contributors","Katkıcılar");
+Translation.set("Tux Racer Development Team","Tux Racer Geliştirme Ekibi");
+Translation.set("Music","Müzik");
+Translation.set("Thanks To:","Teşekkürler:");
diff -Naur ppracer-0.5alpha/src/credits.cpp ppracer-0.5alphamod/src/credits.cpp
--- ppracer-0.5alpha/src/credits.cpp	2005-09-15 11:08:43.000000000 +0300
+++ ppracer-0.5alphamod/src/credits.cpp	2006-06-28 11:22:08.000000000 +0300
@@ -57,6 +57,7 @@
 	{false, "credits_text", "Sergej Kotliar" },	
 	{false, "credits_text", "Theo Snelleman" },	
 	{false, "credits_text", "Trygve B. Wiig" },
+	{false, "credits_text", "Uğur Çetin" },
     {false, "credits_text", "" },
 	{true, "credits_h2", "Contributors" },
 	{false, "credits_text", "Peter Reichel" },
