--- Makefile	2007-08-13 19:01:10.000000000 +0300
+++ Makefile	2007-08-15 01:16:49.000000000 +0300
@@ -7,5 +7,6 @@
 	install -m 0644 lib/PerlConsole/Commands.pm $(LIBDIR)/PerlConsole
 	install -m 0644 lib/PerlConsole/Console.pm $(LIBDIR)/PerlConsole
 	install -m 0644 lib/PerlConsole/Preferences.pm $(LIBDIR)/PerlConsole
-	install -m 0755 perlconsole $(DEST)/usr/local/bin
+	install -d -m 0755 $(DEST)/usr/bin
+	install -m 0755 perlconsole $(DEST)/usr/bin
 
