--- kxmleditor.desktop.orig	2004-07-11 19:16:58.000000000 +0300
+++ kxmleditor/kxmleditor.desktop	2007-05-08 12:44:41.000000000 +0300
@@ -4,9 +4,11 @@
 Icon=kxmleditor.png
 DocPath=kxmleditor/index.html
 Comment=XML editor
+Comment[tr]=XML Düzenleyici
 Comment[fi]=XML-editori
+GenericName[tr]=XML Düzenleyici
 Terminal=false
 Name=KXML Editor
 MimeType=text/xml;
 Encoding=UTF-8
-Categories=Application;Development;TextEditor;KDE;
+Categories=Application;Development;
