--- compiz-manager.orig	2008-06-30 23:09:47.000000000 +0300
+++ compiz-manager	2008-06-30 23:30:42.000000000 +0300
@@ -379,15 +379,15 @@
 	exit 0;
 fi
 # start the gtk-window-decorator if present
-if [ -x ${COMPIZ_BIN_PATH}emerald ] && [ "$USE_EMERALD" = "yes" ]; then
+if [ -x ${COMPIZ_BIN_PATH}kde-window-decorator ] && [ -n "$KDE_FULL_SESSION" ]; then
+	verbose "Starting kde-window-decorator\n"
+	${COMPIZ_BIN_PATH}kde-window-decorator --replace &
+elif [ -x ${COMPIZ_BIN_PATH}emerald ] && [ "$USE_EMERALD" = "yes" ]; then
 	verbose "Starting emerald\n"
 	${COMPIZ_BIN_PATH}emerald --replace &
 elif [ -x ${COMPIZ_BIN_PATH}gtk-window-decorator ] && [ -n "$GNOME_DESKTOP_SESSION_ID" ]; then
 	verbose "Starting gtk-window-decorator\n"
 	${COMPIZ_BIN_PATH}gtk-window-decorator --replace &
-elif [ -x ${COMPIZ_BIN_PATH}kde-window-decorator ] && [ -n "$KDE_FULL_SESSION" ]; then
-	verbose "Starting kde-window-decorator\n"
-	${COMPIZ_BIN_PATH}kde-window-decorator --replace &
 	FALLBACKWM="${KWIN}"
 fi
 
