--- basketorij.desktop	2006-02-01 18:21:57.000000000 +0200
+++ src/basket.desktop	2006-04-08 14:47:55.000000000 +0300
@@ -3,7 +3,7 @@
 Name=BasKet Note Pads
 Name[de]=BasKet Notizblätter
 Name[fr]=Blocs notes BasKet
-Name[xx]=xxBasKetxx
+Name[tr]=BasKet Not Tutucu
 Exec=basket
 Icon=basket
 Type=Application
@@ -16,4 +16,4 @@
 Comment[pt_BR]=
 Comment[sl]=
 Comment[sv]=
-Comment[xx]=xxA set of baskets to keep a full range of data on hand.xx
+Comment[tr]=Not almaya ve her çeşit veriyi el altında tutmaya yarayan bir KDE aracı
