diff -Nuar cgkit-2.0.0alpha7.orig/supportlib/SConstruct cgkit-2.0.0alpha7/supportlib/SConstruct
--- cgkit-2.0.0alpha7.orig/supportlib/SConstruct	2006-05-29 12:26:50.000000000 +0300
+++ cgkit-2.0.0alpha7/supportlib/SConstruct	2007-04-30 14:39:40.000000000 +0300
@@ -35,8 +35,8 @@
   env.Append(CCFLAGS = ["/GX", "/GR", "/MD", "/W3"])
   env.Append(CPPDEFINES = ["WIN32", "_LIB"])
 else:
-  env.Append(CPPPATH = ["/opt/local/include"])
-  env.Append(CCFLAGS = ["-fPIC"])
+  env.Append(CPPPATH = ["/usr/include"])
+  env.Append(CCFLAGS = [])
 
 # Setup the help message
 Help(opts.GenerateHelpText(env))
