--- data/gtkpod.desktop.in.orig	2008-08-25 16:40:48.000000000 +0300
+++ data/gtkpod.desktop.in	2008-08-25 16:42:45.000000000 +0300
@@ -1,7 +1,10 @@
 [Desktop Entry]
-_Name=GtkPod
-_GenericName=iPod Manager
-_Comment=Manage music and video on an Apple iPod
+Name=GtkPod
+Name[tr]=GtkPod
+GenericName=iPod Manager
+GenericName[tr]=iPod Yöneticisi
+Comment=Manage music and video on an Apple iPod
+Comment[tr]=iPod'unuzdaki müzik ve görüntü dosyalarınızı yönetir
 Exec=gtkpod
 Icon=gtkpod
 Type=Application
