--- data/guake.desktop.in.orig	2008-12-23 06:15:41.000000000 +0200
+++ data/guake.desktop.in	2008-12-23 06:18:51.000000000 +0200
@@ -1,14 +1,20 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=Guake Terminal
-Name[pt]=Guake Terminal
-Name[pt_BR]=Guake Terminal
+Name=Guake
+Name[pt]=Guake
+Name[pt_BR]=Guake
+Name[tr]=Guake
+GenericName=Guake Terminal
+GenericName[pt]=Guake Terminal
+GenericName[pt_BR]=Guake Terminal
+GenericName[tr]=Guake Uçbirim
 Comment=Use the command line in a Quake like terminal
 Comment[pt]=Utilizar a linha de comando em um terminal estilo Quake
 Comment[pt_BR]=Utilizar a linha de comando em um terminal estilo Quake
+Comment[tr]=Quake benzeri uçbirimde komut satırı kullanın
 TryExec=guake
 Exec=guake
-Icon=@DATADIR@/pixmaps/guake/guake.png
+Icon=guake
 Type=Application
 Categories=GNOME;GTK;Utility;TerminalEmulator;
 StartupNotify=true
