diff -Naur arora-0.6.orig/src/arora.desktop arora-0.6/src/arora.desktop
--- arora-0.6.orig/src/arora.desktop	2009-03-29 17:13:56.000000000 +0300
+++ arora-0.6/src/arora.desktop	2009-03-29 17:22:23.000000000 +0300
@@ -2,9 +2,11 @@
 Name=Arora
 Name[en_US]=Arora
 Comment=Browse the World Wide Web
+Comment[tr]=İnternette Gezinin
 Comment[nl]=Surfen op het wereldwijde web
 Comment[nl_NL]=Surfen op het wereldwijde web
 GenericName=Web Browser
+GenericName[tr]=Web Tarayıcı
 GenericName[en_US]=Web Browser
 GenericName[cs_CZ]=Webový prohlížeč
 GenericName[nl]=Webbrowser
