diff --git a/src/tabwidget.cpp b/src/tabwidget.cpp
index 099a055..afd9a4a 100644
--- a/src/tabwidget.cpp
+++ b/src/tabwidget.cpp
@@ -793,6 +793,7 @@ void TabWidget::webViewTitleChanged(const QString &title)
     if (title.isEmpty())
         tabTitle = QString::fromUtf8(webView->url().toEncoded());
     tabTitle.replace(QLatin1Char('&'), QLatin1String("&&"));
+    setTabToolTip(index, tabTitle);
     setTabText(index, tabTitle);
     if (currentIndex() == index)
         emit setCurrentTitle(title);
