diff -Naur kalbum-0.8.0/kalbum/kalbum.desktop kalbum-0.8.0_1/kalbum/kalbum.desktop
--- kalbum/kalbum.desktop	2003-03-04 17:27:56.000000000 +0200
+++ kalbum/kalbum.desktop	2007-05-03 18:43:13.000000000 +0300
@@ -1,6 +1,4 @@
 [Desktop Entry]
-Comment=create albums out of a collection of images
-Comment[en_GB]=create albums out of a collection of images
 DocPath=kalbum/index.html
 Encoding=UTF-8
 Exec=kalbum -caption "%c" %i %m
@@ -8,7 +6,11 @@
 MimeType=application/x-kalbum
 Name=Kalbum
 Name[en_GB]=Kalbum
+Name[tr]=Kalbum
+GenericName= create albums out of images
+GenericName[tr]= resimlerden albüm oluştur
 ServiceTypes=
 Terminal=0
 Type=Application
 X-KDE-NativeMimeType=application/x-kalbum
+Categories=Application;Graphics;
