--- recoll-searchgui.desktop	2007-09-11 11:23:21.000000000 +0300
+++ desktop/recoll-searchgui.desktop	2007-09-13 04:10:37.000000000 +0300
@@ -1,8 +1,10 @@
 [Desktop Entry]
 Categories=Utility;Filesystem;Database;
 Comment=Find documents by specifying search terms
+Comment[tr]=Belgeler içerisinde belirtilen ifadeleri arar
 Exec=recoll
 GenericName=Local Text Search
+GenericName[tr]=Yerel Metin Arayıcı
 Icon=recoll
 Name=Recoll
 Terminal=false
