diff -Nur libQGLViewer-2.2.6-3.orig/QGLViewer/manipulatedFrame.cpp libQGLViewer-2.2.6-3/QGLViewer/manipulatedFrame.cpp
--- libQGLViewer-2.2.6-3.orig/QGLViewer/manipulatedFrame.cpp	2008-06-03 19:05:59.000000000 +0300
+++ libQGLViewer-2.2.6-3/QGLViewer/manipulatedFrame.cpp	2008-06-03 19:07:28.000000000 +0300
@@ -28,6 +28,8 @@
 #include "qglviewer.h"
 #include "camera.h"
 
+#include <cstdlib>
+
 #if QT_VERSION >= 0x040000
 # include <QMouseEvent>
 #endif
diff -Nur libQGLViewer-2.2.6-3.orig/QGLViewer/VRender/TopologicalSortMethod.cpp libQGLViewer-2.2.6-3/QGLViewer/VRender/TopologicalSortMethod.cpp
--- libQGLViewer-2.2.6-3.orig/QGLViewer/VRender/TopologicalSortMethod.cpp	2008-06-03 19:05:59.000000000 +0300
+++ libQGLViewer-2.2.6-3/QGLViewer/VRender/TopologicalSortMethod.cpp	2008-06-03 19:09:16.000000000 +0300
@@ -53,6 +53,8 @@
 #include "SortMethod.h"
 #include "Vector2.h"
 
+#include <climits>
+
 using namespace std ;
 using namespace vrender ;
 
diff -Nur libQGLViewer-2.2.6-3.orig/QGLViewer/VRender/VRender.cpp libQGLViewer-2.2.6-3/QGLViewer/VRender/VRender.cpp
--- libQGLViewer-2.2.6-3.orig/QGLViewer/VRender/VRender.cpp	2008-06-03 19:05:59.000000000 +0300
+++ libQGLViewer-2.2.6-3/QGLViewer/VRender/VRender.cpp	2008-06-03 19:10:12.000000000 +0300
@@ -63,6 +63,9 @@
 #include "SortMethod.h"
 #include "Optimizer.h"
 
+#include <cstring>
+#include <cstdlib>
+
 using namespace vrender ;
 using namespace std ;
 
