Index: Thunar-0.9.0/thunar-vfs/Makefile.am
===================================================================
--- Thunar-0.9.0/thunar-vfs/Makefile.am
+++ Thunar-0.9.0/thunar-vfs/Makefile.am	2008-06-16 15:34:49.000000000 +0300
@@ -149,7 +149,8 @@
 	$(LIBFAM_LIBS)							\
 	$(LIBJPEG_LIBS)							\
 	$(LIBPNG_LIBS)							\
-	$(LIBSTARTUP_NOTIFICATION_LIBS)
+	$(LIBSTARTUP_NOTIFICATION_LIBS)					\
+	$(LIBX11_LIBS)
 
 libexec_PROGRAMS =							\
 	thunar-vfs-mime-cleaner-1					\
