diff -Naur lives-0.9.9.7.orig/libweed/Makefile.in lives-0.9.9.7/libweed/Makefile.in
--- lives-0.9.9.7.orig/libweed/Makefile.in	2009-04-07 20:45:03.000000000 +0300
+++ lives-0.9.9.7/libweed/Makefile.in	2009-04-08 00:56:07.000000000 +0300
@@ -637,7 +637,7 @@
 
 
 install-exec-hook:
-	/sbin/ldconfig
+# 	/sbin/ldconfig
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
