diff -Naur linden.orig/indra/newview/llcontroldef.cpp linden/indra/newview/llcontroldef.cpp
--- linden.orig/indra/newview/llcontroldef.cpp	2007-04-25 22:13:13.000000000 +0300
+++ linden/indra/newview/llcontroldef.cpp	2007-05-11 21:41:26.000000000 +0300
@@ -184,8 +184,8 @@
 	gSavedSettings.declareF32("FontSizeLarge",	12.f, "Size of large font (points, or 1/72 of an inch)");
 	gSavedSettings.declareF32("FontSizeHuge",	16.f, "Size of huge font (points, or 1/72 of an inch)");
 
-	gSavedSettings.declareString("FontMonospace",		"profontwindows.ttf", "Name of monospace font (Truetype file name)");
-	gSavedSettings.declareString("FontSansSerif",		"MtBkLfRg.ttf", "Name of san-serif font (Truetype file name)");
+	gSavedSettings.declareString("FontMonospace",		"/usr/share/fonts/dejavu/DejaVuSansMono.ttf", "Name of monospace font (Truetype file name)");
+	gSavedSettings.declareString("FontSansSerif",		"/usr/share/fonts/dejavu/DejaVuSans.ttf", "Name of san-serif font (Truetype file name)");
 #if LL_WINDOWS
 	// Lists Japanese, Korean, and Chinese sanserif fonts available in
 	// Windows XP and Vista, as well as "Arial Unicode MS".
@@ -204,7 +204,7 @@
 	gSavedSettings.declareString("FontSansSerifFallback",	"unicode.ttf", "Name of fallback san-serif font (Truetype file name)");
 #endif
 	gSavedSettings.declareF32("FontSansSerifFallbackScale", 1.0, "Scale of fallback font relative to huge font (fraction of huge font size)");
-	gSavedSettings.declareString("FontSansSerifBold",	"MtBdLfRg.ttf", "Name of bold font (Truetype file name)");
+	gSavedSettings.declareString("FontSansSerifBold",	"/usr/share/fonts/dejavu/DejaVuSans-Bold.ttf", "Name of bold font (Truetype file name)");
 
 	//------------------------------------------------------------------------
 	// Chat
