diff -Naur wlassistant-0.5.5-orig/src/wlassistant.desktop wlassistant-0.5.5/src/wlassistant.desktop
--- wlassistant-0.5.5-orig/src/wlassistant.desktop	2006-09-03 12:00:53.000000000 +0300
+++ wlassistant-0.5.5/src/wlassistant.desktop	2006-09-03 12:06:36.000000000 +0300
@@ -1,9 +1,11 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Wireless Assistant
+GenericName[tr]=Kablosuz bağlantı yardımcısı
+Comment[tr]=Kablosuz bağlantı yardımcısı
 Comment[pl]=Program do obsługi sieci bezprzewodowych
 Comment[es]=Aplicación para buscar redes inalámbricas
-Exec=ksystraycmd sudo wlassistant
+Exec=kdesu wlassistant
 Icon=wlassistant
 Type=Application
 Categories=Qt;KDE;Network;
