diff -Nuar xfce4-session-4.6.1.orig/engines/balou/Makefile.in xfce4-session-4.6.1/engines/balou/Makefile.in
--- xfce4-session-4.6.1.orig/engines/balou/Makefile.in	2009-06-05 04:35:18.000000000 +0300
+++ xfce4-session-4.6.1/engines/balou/Makefile.in	2009-06-05 18:04:28.000000000 +0300
@@ -318,7 +318,7 @@
 libbalou_la_LIBADD = \
 	$(top_builddir)/libxfsm/libxfsm-4.6.la				\
 	$(LIBXFCEGUI4_LIBS)						\
-	$(XFCONF_LIBS)
+	$(XFCONF_LIBS) -lm
 
 libbalou_la_DEPENDENCIES = \
 	$(top_builddir)/libxfsm/libxfsm-4.6.la
