--- configure.orig	2007-11-10 20:02:32.000000000 +0200
+++ configure	2007-11-10 20:03:02.000000000 +0200
@@ -23784,16 +23784,16 @@
         if test -n "$PKG_CONFIG" && \
     { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"libgnome-2.0 >= \$GTKSOURCEVIEW_GNOME_REQUIRED,
 		gtk+-2.0 >= \$GTKSOURCEVIEW_GTK_REQUIRED,
-		gtksourceview-1.0 >= \$GTKSOURCEVIEW_REQUIRED\"") >&5
+		gtksourceview-2.0 >= \$GTKSOURCEVIEW_REQUIRED\"") >&5
   ($PKG_CONFIG --exists --print-errors "libgnome-2.0 >= $GTKSOURCEVIEW_GNOME_REQUIRED,
 		gtk+-2.0 >= $GTKSOURCEVIEW_GTK_REQUIRED,
-		gtksourceview-1.0 >= $GTKSOURCEVIEW_REQUIRED") 2>&5
+		gtksourceview-2.0 >= $GTKSOURCEVIEW_REQUIRED") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
   pkg_cv_PLUGIN_SOURCEVIEW_CFLAGS=`$PKG_CONFIG --cflags "libgnome-2.0 >= $GTKSOURCEVIEW_GNOME_REQUIRED,
 		gtk+-2.0 >= $GTKSOURCEVIEW_GTK_REQUIRED,
-		gtksourceview-1.0 >= $GTKSOURCEVIEW_REQUIRED" 2>/dev/null`
+		gtksourceview-2.0 >= $GTKSOURCEVIEW_REQUIRED" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -23808,16 +23808,16 @@
         if test -n "$PKG_CONFIG" && \
     { (echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"libgnome-2.0 >= \$GTKSOURCEVIEW_GNOME_REQUIRED,
 		gtk+-2.0 >= \$GTKSOURCEVIEW_GTK_REQUIRED,
-		gtksourceview-1.0 >= \$GTKSOURCEVIEW_REQUIRED\"") >&5
+		gtksourceview-2.0 >= \$GTKSOURCEVIEW_REQUIRED\"") >&5
   ($PKG_CONFIG --exists --print-errors "libgnome-2.0 >= $GTKSOURCEVIEW_GNOME_REQUIRED,
 		gtk+-2.0 >= $GTKSOURCEVIEW_GTK_REQUIRED,
-		gtksourceview-1.0 >= $GTKSOURCEVIEW_REQUIRED") 2>&5
+		gtksourceview-2.0 >= $GTKSOURCEVIEW_REQUIRED") 2>&5
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
   pkg_cv_PLUGIN_SOURCEVIEW_LIBS=`$PKG_CONFIG --libs "libgnome-2.0 >= $GTKSOURCEVIEW_GNOME_REQUIRED,
 		gtk+-2.0 >= $GTKSOURCEVIEW_GTK_REQUIRED,
-		gtksourceview-1.0 >= $GTKSOURCEVIEW_REQUIRED" 2>/dev/null`
+		gtksourceview-2.0 >= $GTKSOURCEVIEW_REQUIRED" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -23838,11 +23838,11 @@
         if test $_pkg_short_errors_supported = yes; then
 	        PLUGIN_SOURCEVIEW_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "libgnome-2.0 >= $GTKSOURCEVIEW_GNOME_REQUIRED,
 		gtk+-2.0 >= $GTKSOURCEVIEW_GTK_REQUIRED,
-		gtksourceview-1.0 >= $GTKSOURCEVIEW_REQUIRED"`
+		gtksourceview-2.0 >= $GTKSOURCEVIEW_REQUIRED"`
         else
 	        PLUGIN_SOURCEVIEW_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgnome-2.0 >= $GTKSOURCEVIEW_GNOME_REQUIRED,
 		gtk+-2.0 >= $GTKSOURCEVIEW_GTK_REQUIRED,
-		gtksourceview-1.0 >= $GTKSOURCEVIEW_REQUIRED"`
+		gtksourceview-2.0 >= $GTKSOURCEVIEW_REQUIRED"`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$PLUGIN_SOURCEVIEW_PKG_ERRORS" >&5
