--- configure.orig.in	2007-11-03 09:46:47.000000000 +0200
+++ configure.in	2007-11-03 09:48:01.000000000 +0200
@@ -438,7 +438,7 @@
 	PKG_CHECK_MODULES(PLUGIN_SOURCEVIEW,
 		[libgnome-2.0 >= $GTKSOURCEVIEW_GNOME_REQUIRED,
 		gtk+-2.0 >= $GTKSOURCEVIEW_GTK_REQUIRED,
-		gtksourceview-1.0 >= $GTKSOURCEVIEW_REQUIRED],
+		gtksourceview-2.0 >= $GTKSOURCEVIEW_REQUIRED],
 		[sourceview="yes"],
 		[sourceview="no"])
 	AC_SUBST(PLUGIN_SOURCEVIEW_CFLAGS)
