--- cairo-dock-menu.c.orig	2008-02-19 04:04:03.000000000 +0200
+++ src/cairo-dock-menu.c	2008-02-19 16:58:06.000000000 +0200
@@ -870,11 +870,11 @@
 
 	_add_entry_in_menu (_("Manage themes"), GTK_STOCK_EXECUTE, cairo_dock_initiate_theme_management, pSubMenu);
 
-	_add_entry_in_menu (_("Check updates"), GTK_STOCK_REFRESH, cairo_dock_update, pSubMenu);
+	//_add_entry_in_menu (_("Check updates"), GTK_STOCK_REFRESH, cairo_dock_update, pSubMenu);
 
 	_add_entry_in_menu (_("About"), GTK_STOCK_ABOUT, cairo_dock_about, pSubMenu);
 
-	_add_entry_in_menu (_("Help"), GTK_STOCK_HELP, cairo_dock_help, pSubMenu);
+	//_add_entry_in_menu (_("Help"), GTK_STOCK_HELP, cairo_dock_help, pSubMenu);
 
 	if (! g_bAutoHide)
 	{
