--- enlightenment-0.16.999.062/data/etc/sysactions.conf.orig	2009-08-04 12:12:14.686452740 +0300
+++ enlightenment-0.16.999.062/data/etc/sysactions.conf	2009-08-04 12:13:30.245419438 +0300
@@ -45,8 +45,8 @@
 
 action:   halt      /sbin/shutdown -h now
 action:   reboot    /sbin/shutdown -r now
-action:   suspend   /etc/acpi/sleep.sh force
-action:   hibernate /etc/acpi/hibernate.sh force
+action:   suspend   /usr/sbin/pm-suspend
+action:   hibernate /usr/sbin/pm-hibernate
 
 # on FreeBSD use this instead of the above.
 #action suspend  /usr/sbin/zzz 
@@ -72,4 +72,4 @@
 # user:     myuser     allow:  *
 # user:     another    allow:  suspend hibernate
 # deny everyone else by default
-user:     *         deny:  *
+user:     *         allow:  halt reboot suspend hibernate
