--- e/config/standard/e.src.orig	2009-01-30 11:09:34.000000000 +0200
+++ e/config/standard/e.src	2009-01-30 15:36:14.000000000 +0200
@@ -14,12 +14,12 @@
   value "font_cache" int: 512;
   value "edje_cache" int: 32;
   value "edje_collection_cache" int: 64;
-  value "zone_desks_x_count" int: 4;
+  value "zone_desks_x_count" int: 2;
   value "zone_desks_y_count" int: 1;
   value "use_virtual_roots" int: 0;
-  value "show_desktop_icons" int: 1;
+  value "show_desktop_icons" int: 0;
   value "edge_flip_dragging" int: 1;
-  value "edge_flip_moving" int: 1;
+  value "edge_flip_moving" int: 0;
   value "edge_flip_timeout" double: 0.2500000000000000000000000;
   value "evas_engine_default" int: 1;
   value "evas_engine_container" int: 0;
@@ -1425,11 +1425,11 @@
     }
   }
   value "window_placement_policy" int: 0;
-  value "focus_policy" int: 2;
-  value "focus_setting" int: 3;
+  value "focus_policy" int: 0;
+  value "focus_setting" int: 1;
   value "pass_click_on" int: 1;
   value "always_click_to_raise" int: 0;
-  value "always_click_to_focus" int: 0;
+  value "always_click_to_focus" int: 1;
   value "use_auto_raise" int: 0;
   value "auto_raise_delay" double: 0.5000000000000000000000000;
   value "use_resist" int: 1;
@@ -1478,7 +1478,7 @@
   value "focus_revert_on_hide_or_close" int: 1;
   value "pointer_slide" int: 1;
   value "use_e_cursor" int: 1;
-  value "cursor_size" int: 32;
+  value "cursor_size" int: 28;
   value "menu_autoscroll_margin" int: 0;
   value "menu_autoscroll_cursor_margin" int: 1;
   value "transient.move" int: 1;
@@ -1546,7 +1546,7 @@
           value "geom.size_h" double: 0.0000000000000000000000000;
           value "state_info.seq" int: 3;
           value "state_info.flags" int: 0;
-          value "style" string: "inset";
+          value "style" string: "plain";
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
@@ -1574,7 +1574,7 @@
           value "name" string: "ibar";
           value "id" string: "ibar.1";
           value "geom.pos" int: 350;
-          value "geom.size" int: 200;
+          value "geom.size" int: 100;
           value "geom.res" int: 800;
           value "geom.pos_x" double: 0.0000000000000000000000000;
           value "geom.pos_y" double: 0.0000000000000000000000000;
@@ -1582,7 +1582,7 @@
           value "geom.size_h" double: 0.0000000000000000000000000;
           value "state_info.seq" int: 5;
           value "state_info.flags" int: 0;
-          value "style" string: "inset";
+          value "style" string: "plain";
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
@@ -1591,7 +1591,7 @@
         group "E_Config_Gadcon_Client" struct {
           value "name" string: "temperature";
           value "id" string: "temperature.1";
-          value "geom.pos" int: 672;
+          value "geom.pos" int: 640;
           value "geom.size" int: 32;
           value "geom.res" int: 800;
           value "geom.pos_x" double: 0.0000000000000000000000000;
@@ -1609,7 +1609,7 @@
         group "E_Config_Gadcon_Client" struct {
           value "name" string: "cpufreq";
           value "id" string: "cpufreq";
-          value "geom.pos" int: 704;
+          value "geom.pos" int: 672;
           value "geom.size" int: 32;
           value "geom.res" int: 800;
           value "geom.pos_x" double: 0.0000000000000000000000000;
@@ -1627,7 +1627,7 @@
         group "E_Config_Gadcon_Client" struct {
           value "name" string: "battery";
           value "id" string: "battery";
-          value "geom.pos" int: 736;
+          value "geom.pos" int: 704;
           value "geom.size" int: 32;
           value "geom.res" int: 800;
           value "geom.pos_x" double: 0.0000000000000000000000000;
@@ -1645,7 +1645,7 @@
         group "E_Config_Gadcon_Client" struct {
           value "name" string: "clock";
           value "id" string: "clock";
-          value "geom.pos" int: 768;
+          value "geom.pos" int: 736;
           value "geom.size" int: 32;
           value "geom.res" int: 800;
           value "geom.pos_x" double: 0.0000000000000000000000000;
@@ -1659,6 +1659,24 @@
           value "resizable" uchar: 0;
         }
       }
+      group "clients" list {
+        group "E_Config_Gadcon_Client" struct {
+          value "name" string: "mixer";
+          value "id" string: "mixer";
+          value "geom.pos" int: 768;
+          value "geom.size" int: 32;
+          value "geom.res" int: 800;
+          value "geom.pos_x" double: 0.0000000000000000000000000;
+          value "geom.pos_y" double: 0.0000000000000000000000000;
+          value "geom.size_w" double: 0.0000000000000000000000000;
+          value "geom.size_h" double: 0.0000000000000000000000000;
+          value "state_info.seq" int: 10;
+          value "state_info.flags" int: 0;
+          value "style" string: "plain";
+          value "autoscroll" uchar: 0;
+          value "resizable" uchar: 0;
+        }
+      }
     }
   }
   group "gadcons" list {
@@ -1798,7 +1816,7 @@
       value "orient" int: 6;
       value "fit_along" uchar: 1;
       value "fit_size" uchar: 0;
-      value "style" string: "default";
+      value "style" string: "alternate";
       value "size" int: 40;
       value "overlap" int: 0;
       value "autohide" int: 0;
@@ -1825,7 +1843,7 @@
       value "orient" int: 6;
       value "fit_along" uchar: 1;
       value "fit_size" uchar: 0;
-      value "style" string: "default";
+      value "style" string: "alternate";
       value "size" int: 40;
       value "overlap" int: 0;
       value "autohide" int: 0;
@@ -1852,7 +1870,7 @@
       value "orient" int: 6;
       value "fit_along" uchar: 1;
       value "fit_size" uchar: 0;
-      value "style" string: "default";
+      value "style" string: "alternate";
       value "size" int: 40;
       value "overlap" int: 0;
       value "autohide" int: 0;
@@ -1879,7 +1897,7 @@
       value "orient" int: 6;
       value "fit_along" uchar: 1;
       value "fit_size" uchar: 0;
-      value "style" string: "default";
+      value "style" string: "alternate";
       value "size" int: 40;
       value "overlap" int: 0;
       value "autohide" int: 0;
@@ -1895,7 +1913,7 @@
       }
     }
   }
-  value "font_hinting" int: 0;
+  value "font_hinting" int: 1;
   value "desklock_auth_method" int: 0;
   value "desklock_login_box_zone" int: -1;
   value "desklock_start_locked" int: 0;
@@ -1998,7 +2016,7 @@
   }
   value "desk_auto_switch" int: 0;
   value "thumb_nice" int: 0;
-  value "menu_favorites_show" int: 1;
+  value "menu_favorites_show" int: 0;
   value "menu_apps_show" int: 1;
   value "ping_clients_interval" int: 128;
   value "cache_flush_poll_interval" int: 512;
@@ -2006,13 +2024,13 @@
   value "thumbscroll_threshhold" int: 8;
   value "thumbscroll_momentum_threshhold" double: 100.0000000000000000000000000;
   value "thumbscroll_friction" double: 1.0000000000000000000000000;
-  value "hal_desktop" int: 1;
+  value "hal_desktop" int: 0;
   value "border_keyboard.timeout" double: 5.0000000000000000000000000;
   value "border_keyboard.move.dx" uchar: 5;
   value "border_keyboard.move.dy" uchar: 5;
   value "border_keyboard.resize.dx" uchar: 5;
   value "border_keyboard.resize.dy" uchar: 5;
-  value "hal_desktop" int: 1;
+  value "hal_desktop" int: 0;
   value "scale.min" double: 1.0000000000000000000000000;
   value "scale.max" double: 3.0000000000000000000000000;
   value "scale.factor" double: 1.0000000000000000000000000;
