--- enlightenment-0.16.999.061/config/standard/e.src.orig	2009-04-28 05:06:05.000000000 +0300
+++ enlightenment-0.16.999.061/config/standard/e.src	2009-06-23 13:19:27.000000000 +0300
@@ -1,5 +1,5 @@
 group "E_Config" struct {
-  value "config_version" int: 65842;
+  value "config_version" int: 65843;
   value "show_splash" int: 1;
   value "init_default_theme" string: "default.edj";
   value "desktop_default_name" string: "Desktop %i, %i";
@@ -15,7 +15,7 @@
   value "font_cache" int: 512;
   value "edje_cache" int: 32;
   value "edje_collection_cache" int: 64;
-  value "zone_desks_x_count" int: 4;
+  value "zone_desks_x_count" int: 2;
   value "zone_desks_y_count" int: 1;
   value "use_virtual_roots" int: 0;
   value "show_desktop_icons" int: 1;
@@ -34,6 +34,30 @@
   value "language" string: "en_US.UTF-8";
   group "modules" list {
     group "E_Config_Module" struct {
+      value "name" string: "gadman";
+      value "enabled" uchar: 1;
+      value "delayed" uchar: 0;
+      value "priority" int: 100;
+    }
+  }
+  group "modules" list {
+    group "E_Config_Module" struct {
+      value "name" string: "everything";
+      value "enabled" uchar: 1;
+      value "delayed" uchar: 1;
+      value "priority" int: 0;
+    }
+  }
+  group "modules" list {
+    group "E_Config_Module" struct {
+      value "name" string: "systray";
+      value "enabled" uchar: 1;
+      value "delayed" uchar: 0;
+      value "priority" int: 0;
+    }
+  }
+  group "modules" list {
+    group "E_Config_Module" struct {
       value "name" string: "mixer";
       value "enabled" uchar: 1;
       value "delayed" uchar: 0;
@@ -394,14 +418,6 @@
   }
   group "modules" list {
     group "E_Config_Module" struct {
-      value "name" string: "temperature";
-      value "enabled" uchar: 1;
-      value "delayed" uchar: 0;
-      value "priority" int: 0;
-    }
-  }
-  group "modules" list {
-    group "E_Config_Module" struct {
       value "name" string: "cpufreq";
       value "enabled" uchar: 1;
       value "delayed" uchar: 0;
@@ -410,14 +426,6 @@
   }
   group "modules" list {
     group "E_Config_Module" struct {
-      value "name" string: "battery";
-      value "enabled" uchar: 1;
-      value "delayed" uchar: 0;
-      value "priority" int: 0;
-    }
-  }
-  group "modules" list {
-    group "E_Config_Module" struct {
       value "name" string: "clock";
       value "enabled" uchar: 1;
       value "delayed" uchar: 0;
@@ -458,14 +466,6 @@
   }
   group "modules" list {
     group "E_Config_Module" struct {
-      value "name" string: "gadman";
-      value "enabled" uchar: 1;
-      value "delayed" uchar: 0;
-      value "priority" int: -100;
-    }
-  }
-  group "modules" list {
-    group "E_Config_Module" struct {
       value "name" string: "syscon";
       value "enabled" uchar: 1;
       value "delayed" uchar: 1;
@@ -480,6 +480,16 @@
   }
   group "mouse_bindings" list {
     group "E_Config_Binding_Mouse" struct {
+      value "context" int: 3;
+      value "modifiers" int: 0;
+      value "action" string: "menu_show";
+      value "params" string: "main";
+      value "button" uchar: 1;
+      value "any_mod" uchar: 0;
+    }
+  }
+  group "mouse_bindings" list {
+    group "E_Config_Binding_Mouse" struct {
       value "context" int: 2;
       value "modifiers" int: 4;
       value "action" string: "window_move";
@@ -489,9 +499,10 @@
   }
   group "mouse_bindings" list {
     group "E_Config_Binding_Mouse" struct {
-      value "context" int: 2;
-      value "modifiers" int: 4;
-      value "action" string: "window_resize";
+      value "context" int: 3;
+      value "modifiers" int: 0;
+      value "action" string: "menu_show";
+      value "params" string: "clients";
       value "button" uchar: 2;
       value "any_mod" uchar: 0;
     }
@@ -501,7 +512,7 @@
       value "context" int: 2;
       value "modifiers" int: 4;
       value "action" string: "window_menu";
-      value "button" uchar: 3;
+      value "button" uchar: 2;
       value "any_mod" uchar: 0;
     }
   }
@@ -510,198 +521,205 @@
       value "context" int: 3;
       value "modifiers" int: 0;
       value "action" string: "menu_show";
-      value "params" string: "main";
-      value "button" uchar: 1;
+      value "params" string: "favorites";
+      value "button" uchar: 3;
       value "any_mod" uchar: 0;
     }
   }
   group "mouse_bindings" list {
     group "E_Config_Binding_Mouse" struct {
-      value "context" int: 3;
-      value "modifiers" int: 0;
-      value "action" string: "menu_show";
-      value "params" string: "clients";
-      value "button" uchar: 2;
+      value "context" int: 2;
+      value "modifiers" int: 4;
+      value "action" string: "window_resize";
+      value "button" uchar: 3;
       value "any_mod" uchar: 0;
     }
   }
-  group "mouse_bindings" list {
-    group "E_Config_Binding_Mouse" struct {
-      value "context" int: 3;
+  group "key_bindings" list {
+    group "E_Config_Binding_Key" struct {
+      value "context" int: 9;
       value "modifiers" int: 0;
+      value "key" string: "Menu";
       value "action" string: "menu_show";
-      value "params" string: "favorites";
-      value "button" uchar: 3;
+      value "params" string: "main";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 5;
-      value "key" string: "Left";
-      value "action" string: "desk_flip_by";
-      value "params" string: "-1 0";
+      value "modifiers" int: 1;
+      value "key" string: "F10";
+      value "action" string: "window_maximized_toggle";
+      value "params" string: "default vertical";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 5;
-      value "key" string: "Right";
-      value "action" string: "desk_flip_by";
-      value "params" string: "1 0";
+      value "modifiers" int: 2;
+      value "key" string: "Menu";
+      value "action" string: "menu_show";
+      value "params" string: "clients";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 5;
-      value "key" string: "Up";
-      value "action" string: "desk_flip_by";
-      value "params" string: "0 -1";
+      value "modifiers" int: 4;
+      value "key" string: "F1";
+      value "action" string: "desk_linear_flip_to";
+      value "params" string: "0";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 5;
-      value "key" string: "Down";
-      value "action" string: "desk_flip_by";
-      value "params" string: "0 1";
+      value "modifiers" int: 4;
+      value "key" string: "F2";
+      value "action" string: "desk_linear_flip_to";
+      value "params" string: "1";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 6;
-      value "key" string: "Up";
-      value "action" string: "window_raise";
+      value "modifiers" int: 4;
+      value "key" string: "F3";
+      value "action" string: "desk_linear_flip_to";
+      value "params" string: "2";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 6;
-      value "key" string: "Down";
-      value "action" string: "window_lower";
+      value "modifiers" int: 4;
+      value "key" string: "F4";
+      value "action" string: "desk_linear_flip_to";
+      value "params" string: "3";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 6;
-      value "key" string: "x";
-      value "action" string: "window_close";
+      value "modifiers" int: 4;
+      value "key" string: "F5";
+      value "action" string: "desk_linear_flip_to";
+      value "params" string: "4";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 6;
-      value "key" string: "k";
-      value "action" string: "window_kill";
+      value "modifiers" int: 4;
+      value "key" string: "F6";
+      value "action" string: "desk_linear_flip_to";
+      value "params" string: "5";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 6;
-      value "key" string: "w";
-      value "action" string: "window_menu";
+      value "modifiers" int: 4;
+      value "key" string: "F7";
+      value "action" string: "desk_linear_flip_to";
+      value "params" string: "6";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 6;
-      value "key" string: "s";
-      value "action" string: "window_sticky_toggle";
+      value "modifiers" int: 4;
+      value "key" string: "F8";
+      value "action" string: "desk_linear_flip_to";
+      value "params" string: "7";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 6;
-      value "key" string: "i";
-      value "action" string: "window_iconic_toggle";
+      value "modifiers" int: 4;
+      value "key" string: "F9";
+      value "action" string: "desk_linear_flip_to";
+      value "params" string: "8";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 6;
-      value "key" string: "f";
-      value "action" string: "window_maximized_toggle";
+      value "modifiers" int: 4;
+      value "key" string: "F10";
+      value "action" string: "desk_linear_flip_to";
+      value "params" string: "9";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 1;
-      value "key" string: "F10";
-      value "action" string: "window_maximized_toggle";
-      value "params" string: "default vertical";
+      value "modifiers" int: 4;
+      value "key" string: "F11";
+      value "action" string: "desk_linear_flip_to";
+      value "params" string: "10";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 5;
-      value "key" string: "F10";
-      value "action" string: "window_maximized_toggle";
-      value "params" string: "default horizontal";
+      value "modifiers" int: 4;
+      value "key" string: "F12";
+      value "action" string: "desk_linear_flip_to";
+      value "params" string: "11";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 6;
-      value "key" string: "r";
-      value "action" string: "window_shaded_toggle";
+      value "modifiers" int: 4;
+      value "key" string: "Tab";
+      value "action" string: "winlist";
+      value "params" string: "next";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 6;
-      value "key" string: "Left";
-      value "action" string: "desk_linear_flip_by";
-      value "params" string: "-1";
+      value "modifiers" int: 4;
+      value "key" string: "Menu";
+      value "action" string: "menu_show";
+      value "params" string: "favorites";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 6;
-      value "key" string: "Right";
-      value "action" string: "desk_linear_flip_by";
-      value "params" string: "1";
+      value "modifiers" int: 4;
+      value "key" string: "Escape";
+      value "action" string: "everything";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 4;
+      value "modifiers" int: 3;
       value "key" string: "F1";
-      value "action" string: "desk_linear_flip_to";
+      value "action" string: "screen_send_to";
       value "params" string: "0";
       value "any_mod" uchar: 0;
     }
@@ -709,9 +727,9 @@
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 4;
+      value "modifiers" int: 3;
       value "key" string: "F2";
-      value "action" string: "desk_linear_flip_to";
+      value "action" string: "screen_send_to";
       value "params" string: "1";
       value "any_mod" uchar: 0;
     }
@@ -719,9 +737,9 @@
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 4;
+      value "modifiers" int: 3;
       value "key" string: "F3";
-      value "action" string: "desk_linear_flip_to";
+      value "action" string: "screen_send_to";
       value "params" string: "2";
       value "any_mod" uchar: 0;
     }
@@ -729,9 +747,9 @@
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 4;
+      value "modifiers" int: 3;
       value "key" string: "F4";
-      value "action" string: "desk_linear_flip_to";
+      value "action" string: "screen_send_to";
       value "params" string: "3";
       value "any_mod" uchar: 0;
     }
@@ -739,90 +757,80 @@
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 4;
-      value "key" string: "F5";
-      value "action" string: "desk_linear_flip_to";
-      value "params" string: "4";
-      value "any_mod" uchar: 0;
-    }
-  }
-  group "key_bindings" list {
-    group "E_Config_Binding_Key" struct {
-      value "context" int: 9;
-      value "modifiers" int: 4;
-      value "key" string: "F6";
-      value "action" string: "desk_linear_flip_to";
-      value "params" string: "5";
+      value "modifiers" int: 3;
+      value "key" string: "Left";
+      value "action" string: "screen_send_by";
+      value "params" string: "-1";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 4;
-      value "key" string: "F7";
-      value "action" string: "desk_linear_flip_to";
-      value "params" string: "6";
+      value "modifiers" int: 3;
+      value "key" string: "Right";
+      value "action" string: "screen_send_by";
+      value "params" string: "1";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 4;
-      value "key" string: "F8";
-      value "action" string: "desk_linear_flip_to";
-      value "params" string: "7";
+      value "modifiers" int: 5;
+      value "key" string: "Up";
+      value "action" string: "desk_flip_by";
+      value "params" string: "0 -1";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 4;
-      value "key" string: "F9";
-      value "action" string: "desk_linear_flip_to";
-      value "params" string: "8";
+      value "modifiers" int: 5;
+      value "key" string: "F10";
+      value "action" string: "window_maximized_toggle";
+      value "params" string: "default horizontal";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 4;
-      value "key" string: "F10";
-      value "action" string: "desk_linear_flip_to";
-      value "params" string: "9";
+      value "modifiers" int: 5;
+      value "key" string: "Tab";
+      value "action" string: "winlist";
+      value "params" string: "prev";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 4;
-      value "key" string: "F11";
-      value "action" string: "desk_linear_flip_to";
-      value "params" string: "10";
+      value "modifiers" int: 5;
+      value "key" string: "Down";
+      value "action" string: "desk_flip_by";
+      value "params" string: "0 1";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 4;
-      value "key" string: "F12";
-      value "action" string: "desk_linear_flip_to";
-      value "params" string: "11";
+      value "modifiers" int: 5;
+      value "key" string: "Left";
+      value "action" string: "desk_flip_by";
+      value "params" string: "-1 0";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 6;
-      value "key" string: "m";
-      value "action" string: "menu_show";
-      value "params" string: "main";
+      value "modifiers" int: 5;
+      value "key" string: "Right";
+      value "action" string: "desk_flip_by";
+      value "params" string: "1 0";
       value "any_mod" uchar: 0;
     }
   }
@@ -839,30 +847,27 @@
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 0;
-      value "key" string: "Menu";
-      value "action" string: "menu_show";
-      value "params" string: "main";
+      value "modifiers" int: 6;
+      value "key" string: "d";
+      value "action" string: "desk_deskshow_toggle";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 2;
-      value "key" string: "Menu";
-      value "action" string: "menu_show";
-      value "params" string: "clients";
+      value "modifiers" int: 6;
+      value "key" string: "f";
+      value "action" string: "window_maximized_toggle";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 4;
-      value "key" string: "Menu";
-      value "action" string: "menu_show";
-      value "params" string: "favorites";
+      value "modifiers" int: 6;
+      value "key" string: "i";
+      value "action" string: "window_iconic_toggle";
       value "any_mod" uchar: 0;
     }
   }
@@ -870,29 +875,27 @@
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
       value "modifiers" int: 6;
-      value "key" string: "Insert";
-      value "action" string: "exec";
-      value "params" string: "Eterm";
+      value "key" string: "k";
+      value "action" string: "window_kill";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 4;
-      value "key" string: "Tab";
-      value "action" string: "winlist";
-      value "params" string: "next";
+      value "modifiers" int: 6;
+      value "key" string: "l";
+      value "action" string: "desk_lock";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 5;
-      value "key" string: "Tab";
-      value "action" string: "winlist";
-      value "params" string: "prev";
+      value "modifiers" int: 6;
+      value "key" string: "m";
+      value "action" string: "menu_show";
+      value "params" string: "main";
       value "any_mod" uchar: 0;
     }
   }
@@ -900,8 +903,8 @@
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
       value "modifiers" int: 6;
-      value "key" string: "End";
-      value "action" string: "restart";
+      value "key" string: "r";
+      value "action" string: "window_shaded_toggle";
       value "any_mod" uchar: 0;
     }
   }
@@ -909,17 +912,17 @@
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
       value "modifiers" int: 6;
-      value "key" string: "Delete";
-      value "action" string: "syscon";
+      value "key" string: "s";
+      value "action" string: "window_sticky_toggle";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 4;
-      value "key" string: "Escape";
-      value "action" string: "exebuf";
+      value "modifiers" int: 6;
+      value "key" string: "w";
+      value "action" string: "window_menu";
       value "any_mod" uchar: 0;
     }
   }
@@ -927,8 +930,8 @@
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
       value "modifiers" int: 6;
-      value "key" string: "l";
-      value "action" string: "desk_lock";
+      value "key" string: "x";
+      value "action" string: "window_close";
       value "any_mod" uchar: 0;
     }
   }
@@ -936,47 +939,45 @@
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
       value "modifiers" int: 6;
-      value "key" string: "d";
-      value "action" string: "desk_deskshow_toggle";
+      value "key" string: "Up";
+      value "action" string: "window_raise";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 3;
-      value "key" string: "Left";
-      value "action" string: "screen_send_by";
-      value "params" string: "-1";
+      value "modifiers" int: 6;
+      value "key" string: "End";
+      value "action" string: "restart";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 3;
-      value "key" string: "Right";
-      value "action" string: "screen_send_by";
-      value "params" string: "1";
+      value "modifiers" int: 6;
+      value "key" string: "Down";
+      value "action" string: "window_lower";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 3;
-      value "key" string: "F1";
-      value "action" string: "screen_send_to";
-      value "params" string: "0";
+      value "modifiers" int: 6;
+      value "key" string: "Left";
+      value "action" string: "desk_linear_flip_by";
+      value "params" string: "-1";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 3;
-      value "key" string: "F2";
-      value "action" string: "screen_send_to";
+      value "modifiers" int: 6;
+      value "key" string: "Right";
+      value "action" string: "desk_linear_flip_by";
       value "params" string: "1";
       value "any_mod" uchar: 0;
     }
@@ -984,20 +985,19 @@
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 3;
-      value "key" string: "F3";
-      value "action" string: "screen_send_to";
-      value "params" string: "2";
+      value "modifiers" int: 6;
+      value "key" string: "Delete";
+      value "action" string: "syscon";
       value "any_mod" uchar: 0;
     }
   }
   group "key_bindings" list {
     group "E_Config_Binding_Key" struct {
       value "context" int: 9;
-      value "modifiers" int: 3;
-      value "key" string: "F4";
-      value "action" string: "screen_send_to";
-      value "params" string: "3";
+      value "modifiers" int: 6;
+      value "key" string: "Insert";
+      value "action" string: "exec";
+      value "params" string: "Eterm";
       value "any_mod" uchar: 0;
     }
   }
@@ -1008,7 +1008,7 @@
       value "action" string: "desk_flip_in_direction";
       value "edge" uchar: 1;
       value "any_mod" uchar: 0;
-      value "delay" float: 0.3000000000000000000000000;
+      value "delay" float: 0.3000000119209289550781250;
     }
   }
   group "edge_bindings" list {
@@ -1018,7 +1018,7 @@
       value "action" string: "desk_flip_in_direction";
       value "edge" uchar: 2;
       value "any_mod" uchar: 0;
-      value "delay" float: 0.3000000000000000000000000;
+      value "delay" float: 0.3000000119209289550781250;
     }
   }
   group "edge_bindings" list {
@@ -1028,7 +1028,7 @@
       value "action" string: "desk_flip_in_direction";
       value "edge" uchar: 3;
       value "any_mod" uchar: 0;
-      value "delay" float: 0.3000000000000000000000000;
+      value "delay" float: 0.3000000119209289550781250;
     }
   }
   group "edge_bindings" list {
@@ -1038,7 +1038,7 @@
       value "action" string: "desk_flip_in_direction";
       value "edge" uchar: 4;
       value "any_mod" uchar: 0;
-      value "delay" float: 0.3000000000000000000000000;
+      value "delay" float: 0.3000000119209289550781250;
     }
   }
   group "edge_bindings" list {
@@ -1048,7 +1048,7 @@
       value "action" string: "desk_deskshow_toggle";
       value "edge" uchar: 6;
       value "any_mod" uchar: 0;
-      value "delay" float: 0.3000000000000000000000000;
+      value "delay" float: 0.3000000119209289550781250;
     }
   }
   group "signal_bindings" list {
@@ -1351,24 +1351,24 @@
   }
   group "wheel_bindings" list {
     group "E_Config_Binding_Wheel" struct {
-      value "context" int: 4;
+      value "context" int: 8;
       value "direction" int: 0;
-      value "z" int: -1;
+      value "z" int: 1;
       value "modifiers" int: 4;
       value "any_mod" uchar: 0;
       value "action" string: "desk_linear_flip_by";
-      value "params" string: "-1";
+      value "params" string: "1";
     }
   }
   group "wheel_bindings" list {
     group "E_Config_Binding_Wheel" struct {
-      value "context" int: 4;
-      value "direction" int: 1;
-      value "z" int: -1;
+      value "context" int: 2;
+      value "direction" int: 0;
+      value "z" int: 1;
       value "modifiers" int: 4;
       value "any_mod" uchar: 0;
       value "action" string: "desk_linear_flip_by";
-      value "params" string: "-1";
+      value "params" string: "1";
     }
   }
   group "wheel_bindings" list {
@@ -1384,13 +1384,13 @@
   }
   group "wheel_bindings" list {
     group "E_Config_Binding_Wheel" struct {
-      value "context" int: 4;
-      value "direction" int: 1;
-      value "z" int: 1;
+      value "context" int: 8;
+      value "direction" int: 0;
+      value "z" int: -1;
       value "modifiers" int: 4;
       value "any_mod" uchar: 0;
       value "action" string: "desk_linear_flip_by";
-      value "params" string: "1";
+      value "params" string: "-1";
     }
   }
   group "wheel_bindings" list {
@@ -1406,8 +1406,8 @@
   }
   group "wheel_bindings" list {
     group "E_Config_Binding_Wheel" struct {
-      value "context" int: 2;
-      value "direction" int: 1;
+      value "context" int: 4;
+      value "direction" int: 0;
       value "z" int: -1;
       value "modifiers" int: 4;
       value "any_mod" uchar: 0;
@@ -1417,8 +1417,8 @@
   }
   group "wheel_bindings" list {
     group "E_Config_Binding_Wheel" struct {
-      value "context" int: 2;
-      value "direction" int: 0;
+      value "context" int: 8;
+      value "direction" int: 1;
       value "z" int: 1;
       value "modifiers" int: 4;
       value "any_mod" uchar: 0;
@@ -1439,13 +1439,13 @@
   }
   group "wheel_bindings" list {
     group "E_Config_Binding_Wheel" struct {
-      value "context" int: 8;
-      value "direction" int: 0;
-      value "z" int: -1;
+      value "context" int: 4;
+      value "direction" int: 1;
+      value "z" int: 1;
       value "modifiers" int: 4;
       value "any_mod" uchar: 0;
       value "action" string: "desk_linear_flip_by";
-      value "params" string: "-1";
+      value "params" string: "1";
     }
   }
   group "wheel_bindings" list {
@@ -1461,29 +1461,29 @@
   }
   group "wheel_bindings" list {
     group "E_Config_Binding_Wheel" struct {
-      value "context" int: 8;
-      value "direction" int: 0;
-      value "z" int: 1;
+      value "context" int: 2;
+      value "direction" int: 1;
+      value "z" int: -1;
       value "modifiers" int: 4;
       value "any_mod" uchar: 0;
       value "action" string: "desk_linear_flip_by";
-      value "params" string: "1";
+      value "params" string: "-1";
     }
   }
   group "wheel_bindings" list {
     group "E_Config_Binding_Wheel" struct {
-      value "context" int: 8;
+      value "context" int: 4;
       value "direction" int: 1;
-      value "z" int: 1;
+      value "z" int: -1;
       value "modifiers" int: 4;
       value "any_mod" uchar: 0;
       value "action" string: "desk_linear_flip_by";
-      value "params" string: "1";
+      value "params" string: "-1";
     }
   }
   value "window_placement_policy" int: 0;
-  value "focus_policy" int: 2;
-  value "focus_setting" int: 3;
+  value "focus_policy" int: 0;
+  value "focus_setting" int: 1;
   value "pass_click_on" int: 1;
   value "always_click_to_raise" int: 0;
   value "always_click_to_focus" int: 0;
@@ -1529,13 +1529,13 @@
   value "remember_internal_windows" int: 0;
   value "move_info_follows" int: 1;
   value "resize_info_follows" int: 1;
-  value "move_info_visible" int: 1;
-  value "resize_info_visible" int: 1;
+  value "move_info_visible" int: 0;
+  value "resize_info_visible" int: 0;
   value "focus_last_focused_per_desktop" int: 1;
   value "focus_revert_on_hide_or_close" int: 1;
-  value "pointer_slide" int: 1;
+  value "pointer_slide" int: 0;
   value "use_e_cursor" int: 1;
-  value "cursor_size" int: 32;
+  value "cursor_size" int: 24;
   value "menu_autoscroll_margin" int: 0;
   value "menu_autoscroll_cursor_margin" int: 1;
   value "transient.move" int: 1;
@@ -1577,61 +1577,64 @@
           value "name" string: "start";
           value "id" string: "start";
           value "geom.pos" int: 0;
-          value "geom.size" int: 32;
+          value "geom.size" int: 34;
           value "geom.res" int: 800;
           value "geom.pos_x" double: 0.0000000000000000000000000;
           value "geom.pos_y" double: 0.0000000000000000000000000;
           value "geom.size_w" double: 0.0000000000000000000000000;
           value "geom.size_h" double: 0.0000000000000000000000000;
-          value "state_info.seq" int: 2;
+          value "state_info.seq" int: 1;
           value "state_info.flags" int: 0;
           value "style" string: "plain";
+          value "orient" int: 0;
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
       }
       group "clients" list {
         group "E_Config_Gadcon_Client" struct {
-          value "name" string: "pager";
-          value "id" string: "pager";
-          value "geom.pos" int: 32;
-          value "geom.size" int: 120;
+          value "name" string: "ibox";
+          value "id" string: "ibox.1";
+          value "geom.pos" int: 152;
+          value "geom.size" int: 34;
           value "geom.res" int: 800;
           value "geom.pos_x" double: 0.0000000000000000000000000;
           value "geom.pos_y" double: 0.0000000000000000000000000;
           value "geom.size_w" double: 0.0000000000000000000000000;
           value "geom.size_h" double: 0.0000000000000000000000000;
-          value "state_info.seq" int: 3;
+          value "state_info.seq" int: 2;
           value "state_info.flags" int: 0;
           value "style" string: "inset";
+          value "orient" int: 0;
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
       }
       group "clients" list {
         group "E_Config_Gadcon_Client" struct {
-          value "name" string: "ibox";
-          value "id" string: "ibox.1";
-          value "geom.pos" int: 152;
-          value "geom.size" int: 32;
+          value "name" string: "ibar";
+          value "id" string: "ibar.1";
+          value "geom.pos" int: 350;
+          value "geom.size" int: 90;
           value "geom.res" int: 800;
           value "geom.pos_x" double: 0.0000000000000000000000000;
           value "geom.pos_y" double: 0.0000000000000000000000000;
           value "geom.size_w" double: 0.0000000000000000000000000;
           value "geom.size_h" double: 0.0000000000000000000000000;
-          value "state_info.seq" int: 4;
+          value "state_info.seq" int: 3;
           value "state_info.flags" int: 0;
           value "style" string: "inset";
+          value "orient" int: 0;
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
       }
       group "clients" list {
         group "E_Config_Gadcon_Client" struct {
-          value "name" string: "ibar";
-          value "id" string: "ibar.1";
-          value "geom.pos" int: 350;
-          value "geom.size" int: 200;
+          value "name" string: "cpufreq";
+          value "id" string: "cpufreq";
+          value "geom.pos" int: 704;
+          value "geom.size" int: 34;
           value "geom.res" int: 800;
           value "geom.pos_x" double: 0.0000000000000000000000000;
           value "geom.pos_y" double: 0.0000000000000000000000000;
@@ -1639,79 +1642,84 @@
           value "geom.size_h" double: 0.0000000000000000000000000;
           value "state_info.seq" int: 5;
           value "state_info.flags" int: 0;
-          value "style" string: "inset";
+          value "style" string: "plain";
+          value "orient" int: 0;
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
       }
       group "clients" list {
         group "E_Config_Gadcon_Client" struct {
-          value "name" string: "temperature";
-          value "id" string: "temperature.1";
-          value "geom.pos" int: 672;
+          value "name" string: "battery";
+          value "id" string: "battery";
+          value "geom.pos" int: 736;
           value "geom.size" int: 32;
           value "geom.res" int: 800;
           value "geom.pos_x" double: 0.0000000000000000000000000;
           value "geom.pos_y" double: 0.0000000000000000000000000;
           value "geom.size_w" double: 0.0000000000000000000000000;
           value "geom.size_h" double: 0.0000000000000000000000000;
-          value "state_info.seq" int: 6;
+          value "state_info.seq" int: 8;
           value "state_info.flags" int: 0;
           value "style" string: "plain";
+          value "orient" int: 0;
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
       }
       group "clients" list {
         group "E_Config_Gadcon_Client" struct {
-          value "name" string: "cpufreq";
-          value "id" string: "cpufreq";
-          value "geom.pos" int: 704;
-          value "geom.size" int: 32;
+          value "name" string: "clock";
+          value "id" string: "clock";
+          value "geom.pos" int: 768;
+          value "geom.size" int: 34;
           value "geom.res" int: 800;
           value "geom.pos_x" double: 0.0000000000000000000000000;
           value "geom.pos_y" double: 0.0000000000000000000000000;
           value "geom.size_w" double: 0.0000000000000000000000000;
           value "geom.size_h" double: 0.0000000000000000000000000;
-          value "state_info.seq" int: 7;
+          value "state_info.seq" int: 6;
           value "state_info.flags" int: 0;
           value "style" string: "plain";
+          value "orient" int: 0;
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
       }
       group "clients" list {
         group "E_Config_Gadcon_Client" struct {
-          value "name" string: "battery";
-          value "id" string: "battery";
-          value "geom.pos" int: 736;
-          value "geom.size" int: 32;
+          value "name" string: "systray";
+          value "id" string: "systray";
+          value "geom.pos" int: 720;
+          value "geom.size" int: 34;
           value "geom.res" int: 800;
           value "geom.pos_x" double: 0.0000000000000000000000000;
           value "geom.pos_y" double: 0.0000000000000000000000000;
           value "geom.size_w" double: 0.0000000000000000000000000;
           value "geom.size_h" double: 0.0000000000000000000000000;
-          value "state_info.seq" int: 8;
+          value "state_info.seq" int: 7;
           value "state_info.flags" int: 0;
-          value "style" string: "plain";
+          value "style" string: "inset";
+          value "orient" int: 0;
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
       }
       group "clients" list {
         group "E_Config_Gadcon_Client" struct {
-          value "name" string: "clock";
-          value "id" string: "clock";
-          value "geom.pos" int: 768;
-          value "geom.size" int: 32;
-          value "geom.res" int: 800;
+          value "name" string: "mixer";
+          value "id" string: "mixer.0";
+          value "geom.pos" int: 704;
+          value "geom.size" int: 34;
+          value "geom.res" int: 294;
           value "geom.pos_x" double: 0.0000000000000000000000000;
           value "geom.pos_y" double: 0.0000000000000000000000000;
           value "geom.size_w" double: 0.0000000000000000000000000;
           value "geom.size_h" double: 0.0000000000000000000000000;
-          value "state_info.seq" int: 9;
-          value "state_info.flags" int: 0;
+          value "state_info.seq" int: 4;
+          value "state_info.flags" int: 1;
           value "style" string: "plain";
+          value "orient" int: 0;
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
@@ -1734,6 +1742,7 @@
           value "geom.size_h" double: 0.0000000000000000000000000;
           value "state_info.seq" int: 0;
           value "state_info.flags" int: 0;
+          value "orient" int: 0;
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
@@ -1750,6 +1759,7 @@
           value "geom.size_h" double: 0.0000000000000000000000000;
           value "state_info.seq" int: 0;
           value "state_info.flags" int: 0;
+          value "orient" int: 0;
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
@@ -1772,6 +1782,7 @@
           value "geom.size_h" double: 0.0000000000000000000000000;
           value "state_info.seq" int: 0;
           value "state_info.flags" int: 0;
+          value "orient" int: 0;
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
@@ -1788,6 +1799,7 @@
           value "geom.size_h" double: 0.0000000000000000000000000;
           value "state_info.seq" int: 0;
           value "state_info.flags" int: 0;
+          value "orient" int: 0;
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
@@ -1810,6 +1822,7 @@
           value "geom.size_h" double: 0.0000000000000000000000000;
           value "state_info.seq" int: 0;
           value "state_info.flags" int: 0;
+          value "orient" int: 0;
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
@@ -1826,6 +1839,7 @@
           value "geom.size_h" double: 0.0000000000000000000000000;
           value "state_info.seq" int: 0;
           value "state_info.flags" int: 0;
+          value "orient" int: 0;
           value "autoscroll" uchar: 0;
           value "resizable" uchar: 0;
         }
@@ -1836,6 +1850,44 @@
     group "E_Config_Gadcon" struct {
       value "name" string: "gadman";
       value "id" int: 114;
+      group "clients" list {
+        group "E_Config_Gadcon_Client" struct {
+          value "name" string: "clock";
+          value "id" string: "clock";
+          value "geom.pos" int: 720;
+          value "geom.size" int: 80;
+          value "geom.res" int: 800;
+          value "geom.pos_x" double: 0.4679687499999999888977698;
+          value "geom.pos_y" double: 0.0425000000000000030531133;
+          value "geom.size_w" double: 0.1132812500000000000000000;
+          value "geom.size_h" double: 0.1812499999999999944488849;
+          value "state_info.seq" int: 0;
+          value "state_info.flags" int: 0;
+          value "style" string: "plain";
+          value "orient" int: 0;
+          value "autoscroll" uchar: 0;
+          value "resizable" uchar: 0;
+        }
+      }
+      group "clients" list {
+        group "E_Config_Gadcon_Client" struct {
+          value "name" string: "pager";
+          value "id" string: "pager";
+          value "geom.pos" int: 720;
+          value "geom.size" int: 80;
+          value "geom.res" int: 800;
+          value "geom.pos_x" double: 0.0000000000000000000000000;
+          value "geom.pos_y" double: 0.8824999999999999511501869;
+          value "geom.size_w" double: 0.2132812500000000055511151;
+          value "geom.size_h" double: 0.1174999999999999933386619;
+          value "state_info.seq" int: 0;
+          value "state_info.flags" int: 0;
+          value "style" string: "plain";
+          value "orient" int: 0;
+          value "autoscroll" uchar: 0;
+          value "resizable" uchar: 0;
+        }
+      }
     }
   }
   group "gadcons" list {
@@ -1957,12 +2009,12 @@
   value "desklock_login_box_zone" int: -1;
   value "desklock_start_locked" int: 0;
   value "desklock_autolock_screensaver" int: 0;
-  value "desklock_post_screensaver_time" double: 0.0;
+  value "desklock_post_screensaver_time" double: 0.0000000000000000000000000;
   value "desklock_autolock_idle" int: 0;
   value "desklock_autolock_idle_timeout" double: 300.0000000000000000000000000;
   value "desklock_use_custom_desklock" int: 0;
   value "desklock_ask_presentation" uchar: 1;
-  value "desklock_ask_presentation_timeout" double: 30;
+  value "desklock_ask_presentation_timeout" double: 30.0000000000000000000000000;
   value "display_res_restore" int: 0;
   value "display_res_width" int: 1;
   value "display_res_height" int: 1;
@@ -1974,7 +2026,7 @@
   value "screensaver_blanking" int: 2;
   value "screensaver_expose" int: 2;
   value "screensaver_ask_presentation" uchar: 1;
-  value "screensaver_ask_presentation_timeout" double: 30;
+  value "screensaver_ask_presentation_timeout" double: 30.0000000000000000000000000;
   value "dpms_enable" int: 0;
   value "dpms_standby_enable" int: 0;
   value "dpms_suspend_enable" int: 0;
@@ -2002,7 +2054,10 @@
   value "icon_theme_overrides" uchar: 0;
   value "desk_flip_animate_mode" int: 1;
   value "desk_flip_animate_interpolation" int: 0;
-  value "desk_flip_animate_time" double: 0.2000000000000000111022302;
+  value "desk_flip_animate_time" double: 0.5000000000000000000000000;
+  value "desk_flip_pan_bg" uchar: 0;
+  value "desk_flip_pan_x_axis_factor" double: 0.0000000000000000000000000;
+  value "desk_flip_pan_y_axis_factor" double: 0.0000000000000000000000000;
   value "wallpaper_import_last_dev" string: "~/";
   value "wallpaper_import_last_path" string: "/";
   value "wallpaper_grad_c1_r" int: 0;
@@ -2076,13 +2131,14 @@
   value "thumbscroll_threshhold" int: 8;
   value "thumbscroll_momentum_threshhold" double: 100.0000000000000000000000000;
   value "thumbscroll_friction" double: 1.0000000000000000000000000;
-  value "hal_desktop" int: 1;
+  value "hal_desktop" int: 0;
+  value "hal_auto_mount" int: 0;
+  value "hal_auto_open" int: 0;
   value "border_keyboard.timeout" double: 5.0000000000000000000000000;
   value "border_keyboard.move.dx" uchar: 5;
   value "border_keyboard.move.dy" uchar: 5;
   value "border_keyboard.resize.dx" uchar: 5;
   value "border_keyboard.resize.dy" uchar: 5;
-  value "hal_desktop" int: 1;
   value "scale.min" double: 1.0000000000000000000000000;
   value "scale.max" double: 3.0000000000000000000000000;
   value "scale.factor" double: 1.0000000000000000000000000;
@@ -2091,12 +2147,12 @@
   value "scale.use_custom" uchar: 0;
   value "show_cursor" uchar: 1;
   value "idle_cursor" uchar: 1;
-  value "default_system_menu" string: "";
+  value "default_system_menu" string: "/etc/xdg/menus/ggz.menu";
   value "cfgdlg_normal_wins" uchar: 0;
   value "syscon.main.icon_size" int: 64;
   value "syscon.secondary.icon_size" int: 48;
   value "syscon.extra.icon_size" int: 48;
-  value "syscon.timeout" double: 5.0;
+  value "syscon.timeout" double: 5.0000000000000000000000000;
   value "syscon.do_input" uchar: 1;
   group "syscon.actions" list {
     group "E_Config_Syscon_Action" struct {
