--- e-0.16.999.063_20100215/config/standard/profile.desktop.orig	2010-02-15 20:26:07.000000000 +0200
+++ e-0.16.999.063_20100215/config/standard/profile.desktop	2010-02-18 16:08:26.153431916 +0200
@@ -6,6 +6,7 @@
 Name[fr]=Standard
 Name[el]=Κανονικό
 Name[eo]=Kutima
+Name[tr_TR]=Standart
 Name[zh_CN]=标准
 Icon=icon.png
 Comment=This is the standard<br>profile that is<br>universal for most<br>desktop and laptop<br>systems. Choose this<br>if you are not<br>sure what to choose.
@@ -15,4 +16,5 @@
 Comment[fr]=C'est le profil standard<br>adapté à la quasi totalité<br>des systèmes de bureau<br>et ordinateurs portables.<br>Choisissez ce profil si<br>vous n'êtes pas sûr.
 Comment[el]=Αυτό είναι το κανονικό <br>προφίλ που είναι κοινό<br>για τα περισσότερα συστήματα. Επιλέξτε αυτό<br>αν δεν είστε σίγουροι για <br>την επιλογή σας.
 Comment[eo]=Tio estas la kutima profilo<br>taŭga por preskaŭ ĉiuj<br>oficejaj sistemoj kaj<br>porteblaj sistemoj.<br>Elektu ĝin se vi ne certas.
+Comment[tr_TR]=Çoğu masaüstü ve dizüstü sistemler<br>için tasarlanmış standart kurulumdur.<br>Hangisini seçeceğinizden emin değilseniz bu seçeneği öneririz.
 Comment[zh_CN]=适用于大多数台式机和笔记本的设置主题，如果您不清楚如何选择则请使用这个。
