--- src/Makefile	2007-12-01 10:21:19.000000000 +0200
+++ src/Makefile	2007-12-01 10:20:50.000000000 +0200
@@ -6,7 +6,7 @@
 DEFINES = -DDATA_DIR='""' -DUNIX=1 -DSOUND=$(SOUND) -DDEBUG=$(DEBUG)
 INCLUDES = -I/usr/X11R6/include -I/usr/X11R6/include/GL -I../libs
 LIBPATHS = -L/usr/X11R6/lib -L/usr/lib -L/usr/local/lib -L../libs
-LIBRARIES = -pthread -lglfw -lGL -lX11 -lXxf86vm -lGLU -lfmod-3.75
+LIBRARIES = -pthread -lglfw -lGL -lX11 -lXxf86vm -lGLU
 OBJECTS = doalloc.o doberon.o dobuildings.o doconfig.o dodata.o dodraw.o doengine.o doevents.o dofactories.o dofight.o dofile.o dofollower.o doforces.o dohost.o doipc.o dolayout.o doleader.o dologs.o domap.o domapunits.o domouse.o donet.o doplayers.o doraces.o doschemes.o doselection.o dosimpletypes.o dosound.o dosources.o dounits.o dowalk.o doworkers.o glfont.o glgui.o tga.o utils.o
 TARGETS = ../dark-oberon
 
