--- src/krd.desktop.orig	2006-06-19 23:03:27.000000000 +0300
+++ src/krd.desktop	2007-08-10 01:58:44.000000000 +0300
@@ -1,8 +1,12 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=krd
+Name[tr]=Krdm
+GenericName=KDE remote desktop manager
+GenericName[tr]=KDE uzak masaüstü yöneticisi
 Exec=krd
 Icon=krd
 Type=Application
 Comment=Remote desktop connection manager
 Comment[it]=Un manager per connessioni Remote desktop
+Categories=KDE;Application;Network;
