diff -Nuar pywebkitgtk-1.1.5.orig/Makefile.in pywebkitgtk-1.1.5/Makefile.in
--- pywebkitgtk-1.1.5.orig/Makefile.in	2009-07-05 18:06:44.000000000 +0300
+++ pywebkitgtk-1.1.5/Makefile.in	2009-07-05 18:07:31.000000000 +0300
@@ -256,7 +256,7 @@
 pkgpyexec_LTLIBRARIES = webkit.la
 webkit_la_CPPFLAGS = $(PYTHON_INCLUDES) $(DEPS_CPPFLAGS) $(WEBKIT_CPPFLAGS) $(webkit_cflags)
 webkit_la_CFLAGS = $(DEPS_CFLAGS) $(WEBKIT_CFLAGS)
-webkit_la_LDFLAGS = $(common_ldflags) -export-symbols-regex initwebkit
+webkit_la_LDFLAGS = $(common_ldflags) -export-symbols-regex initwebkit `python-config --libs`
 webkit_la_LIBADD = $(DEPS_LIBS) $(WEBKIT_LIBS)
 webkit_la_SOURCES = $(webkit_sources)
 nodist_webkit_la_SOURCES = $(nodist_webkit_sources)
