diff -urN antiword-0.37.orig/Makefile antiword-0.37/Makefile
--- antiword-0.37.orig/Makefile	2005-10-29 13:13:15.000000000 -0400
+++ antiword-0.37/Makefile	2006-01-18 07:53:02.000000000 -0500
@@ -36,7 +36,7 @@
 LOCAL_INSTALL_DIR = $(HOME)/bin
 LOCAL_RESOURCES_DIR = $(HOME)/.antiword
 
-GLOBAL_INSTALL_DIR = /usr/local/bin
+GLOBAL_INSTALL_DIR = /usr/bin
 GLOBAL_RESOURCES_DIR = /usr/share/antiword
 
 all:		$(PROGS)
diff -urN antiword-0.37.orig/Makefile.Linux antiword-0.37/Makefile.Linux
--- antiword-0.37.orig/Makefile.Linux	2005-10-29 13:13:15.000000000 -0400
+++ antiword-0.37/Makefile.Linux	2006-01-18 07:53:02.000000000 -0500
@@ -36,7 +36,7 @@
 LOCAL_INSTALL_DIR = $(HOME)/bin
 LOCAL_RESOURCES_DIR = $(HOME)/.antiword
 
-GLOBAL_INSTALL_DIR = /usr/local/bin
+GLOBAL_INSTALL_DIR = /usr/bin
 GLOBAL_RESOURCES_DIR = /usr/share/antiword
 
 all:		$(PROGS)
