--- google-perftools-0.91/Makefile.am.norpath	2007-04-16 18:34:04.000000000 -0500
+++ google-perftools-0.91/Makefile.am	2007-04-23 21:51:19.000000000 -0500
@@ -218,7 +218,7 @@
                             $(TCMALLOC_UNITTEST_INCLUDES)
 tcmalloc_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 tcmalloc_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-tcmalloc_unittest_LDADD = libtcmalloc.la liblogging.la $(PTHREAD_LIBS)
+tcmalloc_unittest_LDADD = libtcmalloc.la liblogging.la libstacktrace.la $(PTHREAD_LIBS)
 
 TESTS += tcmalloc_minimal_unittest
 tcmalloc_minimal_unittest_SOURCES = src/tests/tcmalloc_unittest.cc \
@@ -226,19 +226,19 @@
 tcmalloc_minimal_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 tcmalloc_minimal_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
 tcmalloc_minimal_unittest_LDADD = libtcmalloc_minimal.la \
-                                  liblogging.la $(PTHREAD_LIBS)
+                                  liblogging.la libstacktrace.la $(PTHREAD_LIBS)
 
 TESTS += tcmalloc_large_unittest
 tcmalloc_large_unittest_SOURCES = src/tests/tcmalloc_large_unittest.cc
 tcmalloc_large_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 tcmalloc_large_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-tcmalloc_large_unittest_LDADD = libtcmalloc.la $(PTHREAD_LIBS)
+tcmalloc_large_unittest_LDADD = libtcmalloc.la libstacktrace.la $(PTHREAD_LIBS)
 
 TESTS += tcmalloc_minimal_large_unittest
 tcmalloc_minimal_large_unittest_SOURCES = src/tests/tcmalloc_large_unittest.cc
 tcmalloc_minimal_large_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 tcmalloc_minimal_large_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-tcmalloc_minimal_large_unittest_LDADD = libtcmalloc_minimal.la $(PTHREAD_LIBS)
+tcmalloc_minimal_large_unittest_LDADD = libtcmalloc_minimal.la libstacktrace.la $(PTHREAD_LIBS)
 
 # This makes sure it's safe to link in both tcmalloc and tcmalloc_minimal.
 # (One would never do this on purpose, but perhaps by accident...)
@@ -248,7 +248,7 @@
 tcmalloc_both_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 tcmalloc_both_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
 tcmalloc_both_unittest_LDADD = libtcmalloc.la libtcmalloc_minimal.la \
-                               liblogging.la $(PTHREAD_LIBS)
+                               liblogging.la libstacktrace.la $(PTHREAD_LIBS)
 
 
 # performance/unittests originally from ptmalloc2
@@ -273,27 +273,27 @@
 frag_unittest_SOURCES = src/tests/frag_unittest.cc
 frag_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 frag_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-frag_unittest_LDADD = libtcmalloc_minimal.la $(PTHREAD_LIBS)
+frag_unittest_LDADD = libtcmalloc_minimal.la libstacktrace.la $(PTHREAD_LIBS)
 
 TESTS += markidle_unittest
 markidle_unittest_SOURCES = src/tests/markidle_unittest.cc \
                             src/tests/testutil.h src/tests/testutil.cc
 markidle_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 markidle_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-markidle_unittest_LDADD = libtcmalloc_minimal.la $(PTHREAD_LIBS)
+markidle_unittest_LDADD = libtcmalloc_minimal.la libstacktrace.la $(PTHREAD_LIBS)
 
 TESTS += memalign_unittest
 memalign_unittest_SOURCES = src/tests/memalign_unittest.cc
 memalign_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 memalign_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-memalign_unittest_LDADD = libtcmalloc_minimal.la $(PTHREAD_LIBS)
+memalign_unittest_LDADD = libtcmalloc_minimal.la libstacktrace.la $(PTHREAD_LIBS)
 
 TESTS += thread_dealloc_unittest
 thread_dealloc_unittest_SOURCES = src/tests/thread_dealloc_unittest.cc \
                                   src/tests/testutil.h src/tests/testutil.cc
 thread_dealloc_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 thread_dealloc_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-thread_dealloc_unittest_LDADD = libtcmalloc_minimal.la $(PTHREAD_LIBS)
+thread_dealloc_unittest_LDADD = libtcmalloc_minimal.la libstacktrace.la $(PTHREAD_LIBS)
 
 ### Documentation
 dist_doc_DATA += doc/tcmalloc.html \
@@ -415,7 +415,7 @@
                                  $(HEAP_PROFILER_UNITTEST_INCLUDES)
 heap_profiler_unittest_CXXFLAGS = -g
 heap_profiler_unittest_LDFLAGS = -g $(PTHREAD_CFLAGS)
-heap_profiler_unittest_LDADD = libtcmalloc.la $(PTHREAD_LIBS)
+heap_profiler_unittest_LDADD = libtcmalloc.la libstacktrace.la $(PTHREAD_LIBS)
 
 TESTS += heap-checker_unittest.sh
 heap_checker_unittest_sh_SOURCES = src/tests/heap-checker_unittest.sh
@@ -489,18 +489,18 @@
                          $(PROFILER_UNITTEST_INCLUDES)
 profiler1_unittest_SOURCES = $(PROFILER_UNITTEST_SRCS)
 profiler1_unittest_CXXFLAGS = -g
-profiler1_unittest_LDADD = libprofiler.la
+profiler1_unittest_LDADD = libprofiler.la libstacktrace.la
 profiler2_unittest_SOURCES = $(PROFILER_UNITTEST_SRCS)
 profiler2_unittest_CXXFLAGS = -g
-profiler2_unittest_LDADD = -lprofiler
+profiler2_unittest_LDADD = -lprofiler -lstacktrace
 profiler3_unittest_SOURCES = $(PROFILER_UNITTEST_SRCS)
 profiler3_unittest_CXXFLAGS = -g $(PTHREAD_CFLAGS) -DWITH_THREADS
 profiler3_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-profiler3_unittest_LDADD = libprofiler.la $(PTHREAD_LIBS)
+profiler3_unittest_LDADD = libprofiler.la libstacktrace.la $(PTHREAD_LIBS)
 profiler4_unittest_SOURCES = $(PROFILER_UNITTEST_SRCS)
 profiler4_unittest_CXXFLAGS = -g $(PTHREAD_CFLAGS) -DWITH_THREADS
 profiler4_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-profiler4_unittest_LDADD = -lprofiler $(PTHREAD_LIBS)
+profiler4_unittest_LDADD = -lprofiler -lstacktrace $(PTHREAD_LIBS)
 
 ### Documentation
 dist_doc_DATA += doc/cpuprofile.html \
--- google-perftools-0.91/Makefile.in.norpath	2007-04-18 18:45:32.000000000 -0500
+++ google-perftools-0.91/Makefile.in	2007-04-23 21:56:01.000000000 -0500
@@ -148,7 +148,7 @@
 atomicops_unittest_DEPENDENCIES = libspinlock.la liblogging.la
 am_frag_unittest_OBJECTS = frag_unittest-frag_unittest.$(OBJEXT)
 frag_unittest_OBJECTS = $(am_frag_unittest_OBJECTS)
-frag_unittest_DEPENDENCIES = libtcmalloc_minimal.la \
+frag_unittest_DEPENDENCIES = libtcmalloc_minimal.la libstacktrace.la \
 	$(am__DEPENDENCIES_1)
 am_heap_checker_death_unittest_sh_OBJECTS =
 heap_checker_death_unittest_sh_OBJECTS =  \
@@ -168,7 +168,7 @@
 	heap_profiler_unittest-heap-profiler_unittest.$(OBJEXT) \
 	$(am__objects_1)
 heap_profiler_unittest_OBJECTS = $(am_heap_profiler_unittest_OBJECTS)
-heap_profiler_unittest_DEPENDENCIES = libtcmalloc.la \
+heap_profiler_unittest_DEPENDENCIES = libtcmalloc.la libstacktrace.la \
 	$(am__DEPENDENCIES_1)
 am_heap_profiler_unittest_sh_OBJECTS =
 heap_profiler_unittest_sh_OBJECTS =  \
@@ -185,18 +185,18 @@
 	markidle_unittest-markidle_unittest.$(OBJEXT) \
 	markidle_unittest-testutil.$(OBJEXT)
 markidle_unittest_OBJECTS = $(am_markidle_unittest_OBJECTS)
-markidle_unittest_DEPENDENCIES = libtcmalloc_minimal.la \
+markidle_unittest_DEPENDENCIES = libtcmalloc_minimal.la libstacktrace.la \
 	$(am__DEPENDENCIES_1)
 am_memalign_unittest_OBJECTS =  \
 	memalign_unittest-memalign_unittest.$(OBJEXT)
 memalign_unittest_OBJECTS = $(am_memalign_unittest_OBJECTS)
-memalign_unittest_DEPENDENCIES = libtcmalloc_minimal.la \
+memalign_unittest_DEPENDENCIES = libtcmalloc_minimal.la libstacktrace.la \
 	$(am__DEPENDENCIES_1)
 am__objects_7 = profiler1_unittest-profiler_unittest.$(OBJEXT) \
 	$(am__objects_1)
 am_profiler1_unittest_OBJECTS = $(am__objects_7)
 profiler1_unittest_OBJECTS = $(am_profiler1_unittest_OBJECTS)
-profiler1_unittest_DEPENDENCIES = libprofiler.la
+profiler1_unittest_DEPENDENCIES = libprofiler.la libstacktrace.la
 am__objects_8 = profiler2_unittest-profiler_unittest.$(OBJEXT) \
 	$(am__objects_1)
 am_profiler2_unittest_OBJECTS = $(am__objects_8)
@@ -206,7 +206,7 @@
 	$(am__objects_1)
 am_profiler3_unittest_OBJECTS = $(am__objects_9)
 profiler3_unittest_OBJECTS = $(am_profiler3_unittest_OBJECTS)
-profiler3_unittest_DEPENDENCIES = libprofiler.la $(am__DEPENDENCIES_1)
+profiler3_unittest_DEPENDENCIES = libprofiler.la libstacktrace.la $(am__DEPENDENCIES_1)
 am__objects_10 = profiler4_unittest-profiler_unittest.$(OBJEXT) \
 	$(am__objects_1)
 am_profiler4_unittest_OBJECTS = $(am__objects_10)
@@ -233,36 +233,36 @@
 	$(am__objects_1)
 tcmalloc_both_unittest_OBJECTS = $(am_tcmalloc_both_unittest_OBJECTS)
 tcmalloc_both_unittest_DEPENDENCIES = libtcmalloc.la \
-	libtcmalloc_minimal.la liblogging.la $(am__DEPENDENCIES_1)
+	libtcmalloc_minimal.la liblogging.la libstacktrace.la $(am__DEPENDENCIES_1)
 am_tcmalloc_large_unittest_OBJECTS =  \
 	tcmalloc_large_unittest-tcmalloc_large_unittest.$(OBJEXT)
 tcmalloc_large_unittest_OBJECTS =  \
 	$(am_tcmalloc_large_unittest_OBJECTS)
-tcmalloc_large_unittest_DEPENDENCIES = libtcmalloc.la \
+tcmalloc_large_unittest_DEPENDENCIES = libtcmalloc.la libstacktrace.la \
 	$(am__DEPENDENCIES_1)
 am_tcmalloc_minimal_large_unittest_OBJECTS = tcmalloc_minimal_large_unittest-tcmalloc_large_unittest.$(OBJEXT)
 tcmalloc_minimal_large_unittest_OBJECTS =  \
 	$(am_tcmalloc_minimal_large_unittest_OBJECTS)
-tcmalloc_minimal_large_unittest_DEPENDENCIES = libtcmalloc_minimal.la \
+tcmalloc_minimal_large_unittest_DEPENDENCIES = libtcmalloc_minimal.la libstacktrace.la \
 	$(am__DEPENDENCIES_1)
 am_tcmalloc_minimal_unittest_OBJECTS =  \
 	tcmalloc_minimal_unittest-tcmalloc_unittest.$(OBJEXT) \
 	$(am__objects_1)
 tcmalloc_minimal_unittest_OBJECTS =  \
 	$(am_tcmalloc_minimal_unittest_OBJECTS)
-tcmalloc_minimal_unittest_DEPENDENCIES = libtcmalloc_minimal.la \
+tcmalloc_minimal_unittest_DEPENDENCIES = libtcmalloc_minimal.la libstacktrace.la \
 	liblogging.la $(am__DEPENDENCIES_1)
 am_tcmalloc_unittest_OBJECTS =  \
 	tcmalloc_unittest-tcmalloc_unittest.$(OBJEXT) $(am__objects_1)
 tcmalloc_unittest_OBJECTS = $(am_tcmalloc_unittest_OBJECTS)
-tcmalloc_unittest_DEPENDENCIES = libtcmalloc.la liblogging.la \
+tcmalloc_unittest_DEPENDENCIES = libtcmalloc.la liblogging.la libstacktrace.la \
 	$(am__DEPENDENCIES_1)
 am_thread_dealloc_unittest_OBJECTS =  \
 	thread_dealloc_unittest-thread_dealloc_unittest.$(OBJEXT) \
 	thread_dealloc_unittest-testutil.$(OBJEXT)
 thread_dealloc_unittest_OBJECTS =  \
 	$(am_thread_dealloc_unittest_OBJECTS)
-thread_dealloc_unittest_DEPENDENCIES = libtcmalloc_minimal.la \
+thread_dealloc_unittest_DEPENDENCIES = libtcmalloc_minimal.la libstacktrace.la \
 	$(am__DEPENDENCIES_1)
 binSCRIPT_INSTALL = $(INSTALL_SCRIPT)
 SCRIPTS = $(bin_SCRIPTS) $(noinst_SCRIPTS)
@@ -707,30 +707,30 @@
 
 tcmalloc_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 tcmalloc_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-tcmalloc_unittest_LDADD = libtcmalloc.la liblogging.la $(PTHREAD_LIBS)
+tcmalloc_unittest_LDADD = libtcmalloc.la liblogging.la libstacktrace.la $(PTHREAD_LIBS)
 tcmalloc_minimal_unittest_SOURCES = src/tests/tcmalloc_unittest.cc \
                                     $(TCMALLOC_UNITTEST_INCLUDES)
 
 tcmalloc_minimal_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 tcmalloc_minimal_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
 tcmalloc_minimal_unittest_LDADD = libtcmalloc_minimal.la \
-                                  liblogging.la $(PTHREAD_LIBS)
+                                  liblogging.la libstacktrace.la $(PTHREAD_LIBS)
 
 tcmalloc_large_unittest_SOURCES = src/tests/tcmalloc_large_unittest.cc
 tcmalloc_large_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 tcmalloc_large_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-tcmalloc_large_unittest_LDADD = libtcmalloc.la $(PTHREAD_LIBS)
+tcmalloc_large_unittest_LDADD = libtcmalloc.la libstacktrace.la $(PTHREAD_LIBS)
 tcmalloc_minimal_large_unittest_SOURCES = src/tests/tcmalloc_large_unittest.cc
 tcmalloc_minimal_large_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 tcmalloc_minimal_large_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-tcmalloc_minimal_large_unittest_LDADD = libtcmalloc_minimal.la $(PTHREAD_LIBS)
+tcmalloc_minimal_large_unittest_LDADD = libtcmalloc_minimal.la libstacktrace.la $(PTHREAD_LIBS)
 tcmalloc_both_unittest_SOURCES = src/tests/tcmalloc_unittest.cc \
                                  $(TCMALLOC_UNITTEST_INCLUDES)
 
 tcmalloc_both_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 tcmalloc_both_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
 tcmalloc_both_unittest_LDADD = libtcmalloc.la libtcmalloc_minimal.la \
-                               liblogging.la $(PTHREAD_LIBS)
+                               liblogging.la libstacktrace.la $(PTHREAD_LIBS)
 
 PTMALLOC_UNITTEST_INCLUDES = src/tests/ptmalloc/t-test.h \
                              src/tests/ptmalloc/thread-m.h \
@@ -753,23 +753,23 @@
 frag_unittest_SOURCES = src/tests/frag_unittest.cc
 frag_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 frag_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-frag_unittest_LDADD = libtcmalloc_minimal.la $(PTHREAD_LIBS)
+frag_unittest_LDADD = libtcmalloc_minimal.la libstacktrace.la $(PTHREAD_LIBS)
 markidle_unittest_SOURCES = src/tests/markidle_unittest.cc \
                             src/tests/testutil.h src/tests/testutil.cc
 
 markidle_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 markidle_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-markidle_unittest_LDADD = libtcmalloc_minimal.la $(PTHREAD_LIBS)
+markidle_unittest_LDADD = libtcmalloc_minimal.la libstacktrace.la $(PTHREAD_LIBS)
 memalign_unittest_SOURCES = src/tests/memalign_unittest.cc
 memalign_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 memalign_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-memalign_unittest_LDADD = libtcmalloc_minimal.la $(PTHREAD_LIBS)
+memalign_unittest_LDADD = libtcmalloc_minimal.la libstacktrace.la $(PTHREAD_LIBS)
 thread_dealloc_unittest_SOURCES = src/tests/thread_dealloc_unittest.cc \
                                   src/tests/testutil.h src/tests/testutil.cc
 
 thread_dealloc_unittest_CXXFLAGS = $(PTHREAD_CFLAGS)
 thread_dealloc_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-thread_dealloc_unittest_LDADD = libtcmalloc_minimal.la $(PTHREAD_LIBS)
+thread_dealloc_unittest_LDADD = libtcmalloc_minimal.la libstacktrace.la $(PTHREAD_LIBS)
 
 ### ------- tcmalloc (thread-caching malloc + heap profiler + heap checker)
 
@@ -840,7 +840,7 @@
 
 heap_profiler_unittest_CXXFLAGS = -g
 heap_profiler_unittest_LDFLAGS = -g $(PTHREAD_CFLAGS)
-heap_profiler_unittest_LDADD = libtcmalloc.la $(PTHREAD_LIBS)
+heap_profiler_unittest_LDADD = libtcmalloc.la libstacktrace.la $(PTHREAD_LIBS)
 heap_checker_unittest_sh_SOURCES = src/tests/heap-checker_unittest.sh
 heap_checker_death_unittest_sh_SOURCES = src/tests/heap-checker-death_unittest.sh
 HEAP_CHECKER_UNITTEST_INCLUDES = src/memory_region_map.h \
@@ -855,7 +855,7 @@
 heap_checker_unittest_CXXFLAGS = -g $(PTHREAD_CFLAGS)
 heap_checker_unittest_LDFLAGS = -g $(PTHREAD_CFLAGS)
 # tcmalloc has to be specified last!
-heap_checker_unittest_LDADD = $(PTHREAD_LIBS) liblogging.la -ltcmalloc
+heap_checker_unittest_LDADD = $(PTHREAD_LIBS) liblogging.la -ltcmalloc -lstacktrace
 
 ### ------- CPU profiler
 
@@ -886,18 +886,18 @@
 
 profiler1_unittest_SOURCES = $(PROFILER_UNITTEST_SRCS)
 profiler1_unittest_CXXFLAGS = -g
-profiler1_unittest_LDADD = libprofiler.la
+profiler1_unittest_LDADD = libprofiler.la libstacktrace.la
 profiler2_unittest_SOURCES = $(PROFILER_UNITTEST_SRCS)
 profiler2_unittest_CXXFLAGS = -g
-profiler2_unittest_LDADD = -lprofiler
+profiler2_unittest_LDADD = -lprofiler -lstacktrace
 profiler3_unittest_SOURCES = $(PROFILER_UNITTEST_SRCS)
 profiler3_unittest_CXXFLAGS = -g $(PTHREAD_CFLAGS) -DWITH_THREADS
 profiler3_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-profiler3_unittest_LDADD = libprofiler.la $(PTHREAD_LIBS)
+profiler3_unittest_LDADD = libprofiler.la libstacktrace.la $(PTHREAD_LIBS)
 profiler4_unittest_SOURCES = $(PROFILER_UNITTEST_SRCS)
 profiler4_unittest_CXXFLAGS = -g $(PTHREAD_CFLAGS) -DWITH_THREADS
 profiler4_unittest_LDFLAGS = $(PTHREAD_CFLAGS)
-profiler4_unittest_LDADD = -lprofiler $(PTHREAD_LIBS)
+profiler4_unittest_LDADD = -lprofiler -lstacktrace $(PTHREAD_LIBS)
 EXTRA_DIST = packages/rpm.sh packages/rpm/rpm.spec packages/deb.sh packages/deb \
              $(SCRIPTS) libtool \
              src/solaris/libstdc++.la
