diff -Nur qgis-0.7.4.orig/plugins/grass/qgsgrassattributes.cpp qgis-0.7.4/plugins/grass/qgsgrassattributes.cpp
--- qgis-0.7.4.orig/plugins/grass/qgsgrassattributes.cpp	2005-06-15 16:31:41.000000000 +0300
+++ qgis-0.7.4/plugins/grass/qgsgrassattributes.cpp	2006-09-29 23:20:48.000000000 +0300
@@ -47,8 +47,8 @@
 #include "../../src/qgsfeatureattribute.h"
 
 extern "C" {
-#include <gis.h>
-#include <Vect.h>
+#include <grass/gis.h>
+#include <grass/Vect.h>
 }
 
 #include "../../providers/grass/qgsgrass.h"
diff -Nur qgis-0.7.4.orig/plugins/grass/qgsgrassedit.cpp qgis-0.7.4/plugins/grass/qgsgrassedit.cpp
--- qgis-0.7.4.orig/plugins/grass/qgsgrassedit.cpp	2005-04-10 10:02:55.000000000 +0300
+++ qgis-0.7.4/plugins/grass/qgsgrassedit.cpp	2006-09-29 23:21:58.000000000 +0300
@@ -49,8 +49,8 @@
 #include "../../src/qgsfeatureattribute.h"
 
 extern "C" {
-#include <gis.h>
-#include <Vect.h>
+#include <grass/gis.h>
+#include <grass/Vect.h>
 }
 
 #include "../../providers/grass/qgsgrass.h"
diff -Nur qgis-0.7.4.orig/plugins/grass/qgsgrassmodule.cpp qgis-0.7.4/plugins/grass/qgsgrassmodule.cpp
--- qgis-0.7.4.orig/plugins/grass/qgsgrassmodule.cpp	2005-05-10 18:21:59.000000000 +0300
+++ qgis-0.7.4/plugins/grass/qgsgrassmodule.cpp	2006-09-30 00:40:33.000000000 +0300
@@ -69,8 +69,8 @@
 #include "../../src/qgsfeatureattribute.h"
 
 extern "C" {
-#include <gis.h>
-#include <Vect.h>
+#include <grass/gis.h>
+#include <grass/Vect.h>
 }
 
 #include "../../providers/grass/qgsgrass.h"
diff -Nur qgis-0.7.4.orig/plugins/grass/qgsgrassplugin.cpp qgis-0.7.4/plugins/grass/qgsgrassplugin.cpp
--- qgis-0.7.4.orig/plugins/grass/qgsgrassplugin.cpp	2005-05-09 22:19:55.000000000 +0300
+++ qgis-0.7.4/plugins/grass/qgsgrassplugin.cpp	2006-09-29 23:22:48.000000000 +0300
@@ -43,8 +43,8 @@
 #include <iostream>
 
 extern "C" {
-#include <gis.h>
-#include <Vect.h>
+#include <grass/gis.h>
+#include <grass/Vect.h>
 }
 
 #include "qgsgrassplugin.h"
diff -Nur qgis-0.7.4.orig/plugins/grass/qgsgrassregion.cpp qgis-0.7.4/plugins/grass/qgsgrassregion.cpp
--- qgis-0.7.4.orig/plugins/grass/qgsgrassregion.cpp	2005-04-27 17:37:44.000000000 +0300
+++ qgis-0.7.4/plugins/grass/qgsgrassregion.cpp	2006-09-29 23:24:04.000000000 +0300
@@ -52,7 +52,7 @@
 #include "../../src/qgspoint.h"
 
 extern "C" {
-#include <gis.h>
+#include <grass/gis.h>
 }
 
 #include "../../providers/grass/qgsgrass.h"
diff -Nur qgis-0.7.4.orig/plugins/grass/qgsgrassregion.h qgis-0.7.4/plugins/grass/qgsgrassregion.h
--- qgis-0.7.4.orig/plugins/grass/qgsgrassregion.h	2005-04-27 17:37:44.000000000 +0300
+++ qgis-0.7.4/plugins/grass/qgsgrassregion.h	2006-09-29 23:27:44.000000000 +0300
@@ -28,7 +28,7 @@
 #include "qgsgrassregionbase.h"
 
 extern "C" {
-#include <gis.h>
+#include <grass/gis.h>
 }
 
 /*! \class QgsGrassRegion
diff -Nur qgis-0.7.4.orig/plugins/grass/qgsgrassselect.cpp qgis-0.7.4/plugins/grass/qgsgrassselect.cpp
--- qgis-0.7.4.orig/plugins/grass/qgsgrassselect.cpp	2005-07-01 10:05:35.000000000 +0300
+++ qgis-0.7.4/plugins/grass/qgsgrassselect.cpp	2006-09-29 23:31:53.000000000 +0300
@@ -28,8 +28,8 @@
 #include <qsettings.h>
 
 extern "C" {
-#include <gis.h>
-#include <Vect.h>
+#include <grass/gis.h>
+#include <grass/Vect.h>
 }
 
 #include "../../providers/grass/qgsgrass.h"
diff -Nur qgis-0.7.4.orig/plugins/grass/qgsgrasstools.cpp qgis-0.7.4/plugins/grass/qgsgrasstools.cpp
--- qgis-0.7.4.orig/plugins/grass/qgsgrasstools.cpp	2005-03-23 14:03:28.000000000 +0200
+++ qgis-0.7.4/plugins/grass/qgsgrasstools.cpp	2006-09-29 23:32:28.000000000 +0300
@@ -58,8 +58,8 @@
 #include "../../src/qgsfeatureattribute.h"
 
 extern "C" {
-#include <gis.h>
-#include <Vect.h>
+#include <grass/gis.h>
+#include <grass/Vect.h>
 }
 
 #include "../../providers/grass/qgsgrass.h"
diff -Nur qgis-0.7.4.orig/providers/grass/provider.cpp qgis-0.7.4/providers/grass/provider.cpp
--- qgis-0.7.4.orig/providers/grass/provider.cpp	2005-03-29 23:59:33.000000000 +0300
+++ qgis-0.7.4/providers/grass/provider.cpp	2006-09-29 22:26:22.000000000 +0300
@@ -32,9 +32,9 @@
 #include "../../src/qgsrect.h"
 
 extern "C" {
-#include <gis.h>
-#include <dbmi.h>
-#include <Vect.h>
+#include <grass/gis.h>
+#include <grass/dbmi.h>
+#include <grass/Vect.h>
 }
 
 #include "qgsgrass.h"
diff -Nur qgis-0.7.4.orig/providers/grass/qgsgrass.h qgis-0.7.4/providers/grass/qgsgrass.h
--- qgis-0.7.4.orig/providers/grass/qgsgrass.h	2005-08-12 01:52:13.000000000 +0300
+++ qgis-0.7.4/providers/grass/qgsgrass.h	2006-09-29 22:27:04.000000000 +0300
@@ -18,8 +18,8 @@
 
 // GRASS header files
 extern "C" {
-#include <gis.h>
-#include <form.h>
+#include <grass/gis.h>
+#include <grass/form.h>
 }
 
 /*!
diff -Nur qgis-0.7.4.orig/providers/grass/qgsgrassprovider.cpp qgis-0.7.4/providers/grass/qgsgrassprovider.cpp
--- qgis-0.7.4.orig/providers/grass/qgsgrassprovider.cpp	2005-08-12 01:52:13.000000000 +0300
+++ qgis-0.7.4/providers/grass/qgsgrassprovider.cpp	2006-09-29 22:27:40.000000000 +0300
@@ -33,10 +33,10 @@
 #include "../../src/qgsfeatureattribute.h"
 
 extern "C" {
-#include <gprojects.h>
-#include <gis.h>
-#include <dbmi.h>
-#include <Vect.h>
+#include <grass/gprojects.h>
+#include <grass/gis.h>
+#include <grass/dbmi.h>
+#include <grass/Vect.h>
 }
 
 #include "qgsgrass.h"
