--- isomaster.desktop.src.orig	2007-12-20 01:06:29.000000000 +0200
+++ isomaster.desktop.src	2007-12-20 01:16:02.000000000 +0200
@@ -1,10 +1,11 @@
 [Desktop Entry]
 Name=ISO Master
 GenericName=CD image editor
+GenericName[tr]=ISO Düzenleyici
 Comment=Graphical CD image editor for reading, modifying and writing ISO images
 Exec=isomaster
 Terminal=false
 StartupNotify=true
 Type=Application
-Categories=Utility;DiscBurning;AudioVideo;AudioVideoEditing;
+Categories=Application;GTK;Utility
 MimeType=application/x-iso
