--- misc/phasex.desktop	2007-05-24 11:41:10.000000000 +0300
+++ phasex.desktop.new	2007-10-25 22:45:23.000000000 +0300
@@ -1,11 +1,11 @@
 [Desktop Entry]
 Name=PHASEX
+GenericName[tr]=Yazılımsal Ses Sentezleyici
+GenericName[en]=Software Synthesizer
 Comment=Phase Harmonic Advanced Synthesis Experiment
 Icon=phasex-icon.png
 Exec=phasex
 Terminal=false
 Type=Application
 Encoding=UTF-8
-
-X-Desktop-File-Install-Version=0.10
-Categories=X-Jack;X-MIDI;X-Digital_Processing;AudioVideo;Audio;
+Categories=Application;AudioVideo;
